--- a/src/io/input.cpp
+++ b/src/io/input.cpp
@@ -1187,6 +1187,9 @@
         process_event(&event);
     }
 
+    // processes non sdl events (e.g. manymouse/udev gun polling)
+    process_event(NULL);
+
     // added by JFA for -idleexit
     if (get_idleexit() > 0 && elapsed_ms_time(idle_timer) > get_idleexit())
         set_quitflag();
@@ -1228,6 +1231,8 @@
 {
     unsigned int i = 0;
 
+    if (event != NULL) {
+
     // by RDG2010
     // make things easier to read...
     SDL_Keycode keyPressed = event->key.key;
@@ -1469,6 +1474,8 @@
         break;
     }
 
+    } // if (event != NULL)
+
     if (g_game->get_mouse_enabled())
     {
         if (g_mouse_mode == MANY_MOUSE)
@@ -1478,6 +1485,7 @@
         }
         else
         {
+            if (event != NULL) {
             switch (event->type)
             {
                 case SDL_EVENT_MOUSE_BUTTON_DOWN:
@@ -1561,6 +1569,7 @@
                     break;
                 }
             }
+            } // if (event != NULL)
         }
     }
 
