diff --git a/drivers/amlogic/display/logo/logo.c b/drivers/amlogic/display/logo/logo.c
index 9266be3..e3e9685 100644
--- a/drivers/amlogic/display/logo/logo.c
+++ b/drivers/amlogic/display/logo/logo.c
@@ -117,16 +117,10 @@ int set_osd_freescaler(int index, enum vmode_e new_mode)
 static int refresh_mode_and_logo(bool first)
 {
 	enum vmode_e cur_mode = VMODE_MAX;
-	int hdp_state = get_hpd_state();
 
 	if (!first && osd_get_logo_index() != logo_info.index)
 		return -1;
 
-	if (hdp_state)
-		cur_mode = hdmimode;
-	else
-		cur_mode = cvbsmode;
-
 	if (first) {
 		last_mode = get_logo_vmode();
 
@@ -193,6 +187,8 @@ static int logo_info_init(char *para)
 		}
 		return 0;
 	}
+	else if (vmode_name_to_mode(para) < VMODE_MAX)
+		set_vout_init_vmode(para);
 
 	return 0;
 }
diff --git a/drivers/amlogic/display/vout/vout_serve.c b/drivers/amlogic/display/vout/vout_serve.c
index fae61b3..9196c2c 100644
--- a/drivers/amlogic/display/vout/vout_serve.c
+++ b/drivers/amlogic/display/vout/vout_serve.c
@@ -666,6 +666,17 @@ static int __init get_vout_init_mode(char *str)
 }
 __setup("vout=", get_vout_init_mode);
 
+void set_vout_init_vmode(char *str)
+{
+	char str2[1024];
+	strcpy(str2, str);
+	strcat(str2, ",en"); // logo was already displayed by uboot
+	get_vout_init_mode(str2);
+	pr_err("set_vout_init_vmode: %s\n", str2);
+
+}
+EXPORT_SYMBOL(set_vout_init_vmode);
+
 MODULE_AUTHOR("Platform-BJ <platform.bj@amlogic.com>");
 MODULE_DESCRIPTION("VOUT Server Module");
 MODULE_LICENSE("GPL");
diff --git a/include/linux/amlogic/vout/vout_notify.h b/include/linux/amlogic/vout/vout_notify.h
index 2e4922a..659da93 100644
--- a/include/linux/amlogic/vout/vout_notify.h
+++ b/include/linux/amlogic/vout/vout_notify.h
@@ -102,6 +102,7 @@ extern enum vmode_e validate_vmode2(char *);
 extern void set_vout2_mode_internal(char *name);
 extern enum vmode_e get_logo_vmode(void);
 extern int set_logo_vmode(enum vmode_e);
+extern void set_vout_init_vmode(char *);
 
 extern int vout2_suspend(void);
 extern int vout2_resume(void);
