diff --git a/core/rend/gles/gldraw.cpp b/core/rend/gles/gldraw.cpp
index b69c607e0a..2b0c3dcb35 100644
--- a/core/rend/gles/gldraw.cpp
+++ b/core/rend/gles/gldraw.cpp
@@ -863,7 +863,11 @@ bool OpenGLRenderer::GetLastFrame(std::vector<u8>& data, int& width, int& height
 
 	glBindFramebuffer(GL_FRAMEBUFFER, 0);
 	data.resize(width * height * 3);
+#ifndef GLES2
 	dstFramebuffer.bind(GL_READ_FRAMEBUFFER);
+#else
+    dstFramebuffer.bind(GL_FRAMEBUFFER);
+#endif	
 	glPixelStorei(GL_PACK_ALIGNMENT, 1);
 	if (gl.is_gles)
 	{
