diff --git a/src/video/kmsdrm/SDL_kmsdrmvideo.c b/src/video/kmsdrm/SDL_kmsdrmvideo.c
index 7223a5a..0972cb9 100644
--- a/src/video/kmsdrm/SDL_kmsdrmvideo.c
+++ b/src/video/kmsdrm/SDL_kmsdrmvideo.c
@@ -78,6 +78,7 @@
     DIR *folder;
     const char *hint;
     struct dirent *res;
+    int drmConn = 0;
 
     hint = SDL_GetHint(SDL_HINT_KMSDRM_DEVICE_INDEX);
     if (hint && *hint) {
@@ -118,13 +119,28 @@ get_driindex(void)
                                      resources->count_connectors,
                                      resources->count_encoders,
                                      resources->count_crtcs);
+			// batocera
+			{
+			  FILE* fdDrmConn;
+			  int drmConnRead;
+			  if((fdDrmConn = fopen("/var/run/drmConn", "r")) != NULL) {
+			    if(fscanf(fdDrmConn, "%i", &drmConnRead) == 1) {
+			      if(drmConnRead>=0 && drmConn<resources->count_connectors) {
+				drmConn = drmConnRead;
+			      }
+			    }
+			  }
+			}
+			//
 
                         if (resources->count_connectors > 0 &&
                             resources->count_encoders > 0 &&
                             resources->count_crtcs > 0) {
                             available = -ENOENT;
                             for (i = 0; i < resources->count_connectors; i++) {
-                                drmModeConnector *conn =
+			        drmModeConnector *conn;
+				if(i != drmConn) continue;
+				conn =
                                     KMSDRM_drmModeGetConnector(
                                         drm_fd, resources->connectors[i]);
 
@@ -753,6 +769,7 @@ KMSDRM_InitDisplays (_THIS) {
     uint64_t async_pageflip = 0;
     int ret = 0;
     int i;
+    int drmConn = 0;
 
     /* Open /dev/dri/cardNN (/dev/drmN if on OpenBSD version less than 6.9) */
     SDL_snprintf(viddata->devpath, sizeof(viddata->devpath), "%s%d",
@@ -950,11 +950,27 @@
         goto cleanup;
     }
 
+    // batocera
+    {
+        FILE* fdDrmConn;
+        int drmConnRead;
+        if((fdDrmConn = fopen("/var/run/drmConn", "r")) != NULL) {
+            if(fscanf(fdDrmConn, "%i", &drmConnRead) == 1) {
+                if(drmConnRead>=0 && drmConn<resources->count_connectors) {
+                    drmConn = drmConnRead;
+                }
+            }
+        }
+    }
+    //
+
     /* Iterate on the available connectors. For every connected connector,
        we create an SDL_Display and add it to the list of SDL Displays. */
     for (i = 0; i < resources->count_connectors; i++) {
-        drmModeConnector *connector = KMSDRM_drmModeGetConnector(viddata->drm_fd,
-                                                                 resources->connectors[i]);
+        drmModeConnector *connector;
+        if(i != drmConn) continue;
+            connector = KMSDRM_drmModeGetConnector(viddata->drm_fd,
+                                                   resources->connectors[i]);
 
         if (connector == NULL) {
             continue;
