---
 src/video/SDL_video.c | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/src/video/SDL_video.c b/src/video/SDL_video.c
index 2ad82fa..5f41911 100644
--- a/src/video/SDL_video.c
+++ b/src/video/SDL_video.c
@@ -69,12 +69,12 @@ static VideoBootStrap *bootstrap[] = {
 #ifdef SDL_VIDEO_DRIVER_COCOA
     &COCOA_bootstrap,
 #endif
-#ifdef SDL_VIDEO_DRIVER_X11
-    &X11_bootstrap,
-#endif
 #ifdef SDL_VIDEO_DRIVER_WAYLAND
     &Wayland_bootstrap,
 #endif
+#ifdef SDL_VIDEO_DRIVER_X11
+    &X11_bootstrap,
+#endif
 #ifdef SDL_VIDEO_DRIVER_VIVANTE
     &VIVANTE_bootstrap,
 #endif
-- 
2.44.0

