diff --git a/src/video/SDL_video.c b/src/video/SDL_video.c
index f0b7ce4781..0def8aec67 100644
--- a/src/video/SDL_video.c
+++ b/src/video/SDL_video.c
@@ -69,12 +69,12 @@
 #ifdef SDL_VIDEO_DRIVER_COCOA
     &COCOA_bootstrap,
 #endif
-#ifdef SDL_VIDEO_DRIVER_X11
-    &X11_bootstrap,
-#endif
 #ifdef SDL_VIDEO_DRIVER_WAYLAND
     &Wayland_bootstrap,
 #endif
+#ifdef SDL_VIDEO_DRIVER_X11
+    &X11_bootstrap,
+#endif
 #ifdef SDL_VIDEO_DRIVER_VIVANTE
     &VIVANTE_bootstrap,
 #endif
