--- a/src/joystick/SDL_joystick.c	2024-05-09 16:30:28.081019154 +0200
+++ b/src/joystick/SDL_joystick.c	2024-05-09 16:31:11.029974916 +0200
@@ -2474,7 +2474,7 @@
     /* We only need 16 bits for each of these; space them out to fill 128. */
     /* Byteswap so devices get same GUID on little/big endian platforms. */
     *guid16++ = SDL_SwapLE16(bus);
-    *guid16++ = SDL_SwapLE16(crc);
+    *guid16++ = 0;
 
     if (vendor && product) {
         *guid16++ = SDL_SwapLE16(vendor);
