--- a/src/core/linux/SDL_udev.c	2024-05-09 16:37:37.438583598 +0200
+++ b/src/core/linux/SDL_udev.c	2024-05-09 16:38:29.543745412 +0200
@@ -423,6 +423,7 @@
             devclass |= SDL_UDEV_DEVICE_JOYSTICK;
         }
 
+        SDL_SetHint(SDL_HINT_ACCELEROMETER_AS_JOYSTICK, "0");
         val = _this->syms.udev_device_get_property_value(dev, "ID_INPUT_ACCELEROMETER");
         if (SDL_GetHintBoolean(SDL_HINT_ACCELEROMETER_AS_JOYSTICK, SDL_TRUE) &&
             val && SDL_strcmp(val, "1") == 0) {
