diff --git a/src/core/linux/SDL_udev.c b/src/core/linux/SDL_udev.c
index 4c38b6e..b416a53 100644
--- a/src/core/linux/SDL_udev.c
+++ b/src/core/linux/SDL_udev.c
@@ -418,11 +418,11 @@
             devclass |= SDL_UDEV_DEVICE_JOYSTICK;
         }
 
-        val = _this->syms.udev_device_get_property_value(dev, "ID_INPUT_ACCELEROMETER");
-        if (SDL_GetHintBoolean(SDL_HINT_ACCELEROMETER_AS_JOYSTICK, SDL_TRUE) &&
-            val != NULL && SDL_strcmp(val, "1") == 0) {
-            devclass |= SDL_UDEV_DEVICE_JOYSTICK;
-        }
+        //val = _this->syms.udev_device_get_property_value(dev, "ID_INPUT_ACCELEROMETER");
+        //if (SDL_GetHintBoolean(SDL_HINT_ACCELEROMETER_AS_JOYSTICK, SDL_TRUE) &&
+        //    val != NULL && SDL_strcmp(val, "1") == 0) {
+        //    devclass |= SDL_UDEV_DEVICE_JOYSTICK;
+        //}
 
         val = _this->syms.udev_device_get_property_value(dev, "ID_INPUT_MOUSE");
         if (val != NULL && SDL_strcmp(val, "1") == 0) {
