diff --git a/src/video/x11/SDL_x11modes.c b/src/video/x11/SDL_x11modes.c
index c8f6c54..bc72de2 100644
--- a/src/video/x11/SDL_x11modes.c
+++ b/src/video/x11/SDL_x11modes.c
@@ -195,7 +195,7 @@
 
 static int CalculateXRandRRefreshRate(const XRRModeInfo *info)
 {
-    return (info->hTotal && info->vTotal) ? SDL_round(((double)info->dotClock / (double)(info->hTotal * info->vTotal))) : 0;
+    return (info->hTotal && info->vTotal) ? round(((double)info->dotClock / (double)(info->hTotal * info->vTotal))) : 0;
 }
 
 static SDL_bool SetXRandRModeInfo(Display *display, XRRScreenResources *res, RRCrtc crtc,
@@ -266,7 +266,7 @@ SetXRandRDisplayName(Display *dpy, Atom EDID, char *name, const size_t namelen,
                     dump_monitor_info(info);
 #endif
                     SDL_strlcpy(name, info->dsc_product_name, namelen);
-                    SDL_free(info);
+                    free(info);
                 }
                 X11_XFree(prop);
             }
@@ -518,29 +518,6 @@
 }
 #endif /* SDL_VIDEO_DRIVER_X11_XRANDR */
 
-static int GetXftDPI(Display *dpy)
-{
-    char *xdefault_resource;
-    int xft_dpi, err;
-
-    xdefault_resource = X11_XGetDefault(dpy, "Xft", "dpi");
-
-    if (!xdefault_resource) {
-        return 0;
-    }
-
-    /*
-     * It's possible for SDL_atoi to call SDL_strtol, if it fails due to a
-     * overflow or an underflow, it will return LONG_MAX or LONG_MIN and set
-     * errno to ERANGE. So we need to check for this so we dont get crazy dpi
-     * values
-     */
-    xft_dpi = SDL_atoi(xdefault_resource);
-    err = errno;
-
-    return err == ERANGE ? 0 : xft_dpi;
-}
-
 /* This is used if there's no better functionality--like XRandR--to use.
    It won't attempt to supply different display modes at all, but it can
    enumerate the current displays and their current sizes. */
@@ -566,7 +542,7 @@ static int X11_InitModes_StdXlib(_THIS)
     Display *dpy = data->display;
     const int default_screen = DefaultScreen(dpy);
     Screen *screen = ScreenOfDisplay(dpy, default_screen);
-    int display_mm_width, display_mm_height, xft_dpi, scanline_pad, n, i;
+    int display_mm_width, display_mm_height, scanline_pad, n, i;
     SDL_DisplayModeData *modedata;
     SDL_DisplayData *displaydata;
     SDL_DisplayMode mode;
@@ -599,12 +576,6 @@
     displaydata->vdpi = display_mm_height ? (((float)mode.h) * 25.4f / display_mm_height) : 0.0f;
     displaydata->ddpi = SDL_ComputeDiagonalDPI(mode.w, mode.h, ((float)display_mm_width) / 25.4f, ((float)display_mm_height) / 25.4f);
 
-    xft_dpi = GetXftDPI(dpy);
-    if (xft_dpi > 0) {
-        displaydata->hdpi = (float)xft_dpi;
-        displaydata->vdpi = (float)xft_dpi;
-    }
-
     scanline_pad = SDL_BYTESPERPIXEL(pixelformat) * 8;
     pixmapformats = X11_XListPixmapFormats(dpy, &n);
     if (pixmapformats) {
@@ -715,22 +686,6 @@
     }
 }
 
-#ifdef SDL_VIDEO_DRIVER_X11_XRANDR
-/* This catches an error from XRRSetScreenSize, as a workaround for now. */
-/* !!! FIXME: remove this later when we have a better solution. */
-static int (*PreXRRSetScreenSizeErrorHandler)(Display *, XErrorEvent *) = NULL;
-static int SDL_XRRSetScreenSizeErrHandler(Display *d, XErrorEvent *e)
-{
-    /* BadMatch: https://github.com/libsdl-org/SDL/issues/4561 */
-    /* BadValue: https://github.com/libsdl-org/SDL/issues/4840 */
-    if ((e->error_code == BadMatch) || (e->error_code == BadValue)) {
-        return 0;
-    }
-
-    return PreXRRSetScreenSizeErrorHandler(d, e);
-}
-#endif
-
 int X11_SetDisplayMode(_THIS, SDL_VideoDisplay *sdl_display, SDL_DisplayMode *mode)
 {
     SDL_VideoData *viddata = (SDL_VideoData *)_this->driverdata;
@@ -762,7 +715,6 @@ X11_SetDisplayMode(_THIS, SDL_VideoDisplay * sdl_display, SDL_DisplayMode * mode
     if (data->use_xrandr) {
         Display *display = viddata->display;
         SDL_DisplayModeData *modedata = (SDL_DisplayModeData *)mode->driverdata;
-        int mm_width, mm_height;
         XRRScreenResources *res;
         XRROutputInfo *output_info;
         XRRCrtcInfo *crtc;
@@ -765,45 +720,10 @@
             return SDL_SetError("Couldn't get XRandR crtc info");
         }
 
-        if (crtc->mode == modedata->xrandr_mode) {
-#ifdef X11MODES_DEBUG
-            printf("already in desired mode 0x%lx (%ux%u), nothing to do\n",
-                   crtc->mode, crtc->width, crtc->height);
-#endif
-            status = Success;
-            goto freeInfo;
-        }
-
-        X11_XGrabServer(display);
-        status = X11_XRRSetCrtcConfig(display, res, output_info->crtc, CurrentTime,
-                                      0, 0, None, crtc->rotation, NULL, 0);
-        if (status != Success) {
-            goto ungrabServer;
-        }
-
-        mm_width = mode->w * DisplayWidthMM(display, data->screen) / DisplayWidth(display, data->screen);
-        mm_height = mode->h * DisplayHeightMM(display, data->screen) / DisplayHeight(display, data->screen);
-
-        /* !!! FIXME: this can get into a problem scenario when a window is
-           bigger than a physical monitor in a configuration where one screen
-           spans multiple physical monitors. A detailed reproduction case is
-           discussed at https://github.com/libsdl-org/SDL/issues/4561 ...
-           for now we cheat and just catch the X11 error and carry on, which
-           is likely to cause subtle issues but is better than outright
-           crashing */
-        X11_XSync(display, False);
-        PreXRRSetScreenSizeErrorHandler = X11_XSetErrorHandler(SDL_XRRSetScreenSizeErrHandler);
-        X11_XRRSetScreenSize(display, RootWindow(display, data->screen), mode->w, mode->h, mm_width, mm_height);
-        X11_XSync(display, False);
-        X11_XSetErrorHandler(PreXRRSetScreenSizeErrorHandler);
-
         status = X11_XRRSetCrtcConfig(display, res, output_info->crtc, CurrentTime,
                                       crtc->x, crtc->y, modedata->xrandr_mode, crtc->rotation,
                                       &data->xrandr_output, 1);
 
-    ungrabServer:
-        X11_XUngrabServer(display);
-    freeInfo:
         X11_XRRFreeCrtcInfo(crtc);
         X11_XRRFreeOutputInfo(output_info);
         X11_XRRFreeScreenResources(res);
