diff --git a/src/joystick/SDL_gamecontroller.c b/src/joystick/SDL_gamecontroller.c
index 6ebdf66..fee2e4d 100644
--- a/src/joystick/SDL_gamecontroller.c
+++ b/src/joystick/SDL_gamecontroller.c
@@ -1371,14 +1371,14 @@ SDL_GameControllerInitMappings(void)
 {
     char szControllerMapPath[1024];
     int i = 0;
-    const char *pMappingString = NULL;
-    pMappingString = s_ControllerMappings[i];
-    while (pMappingString) {
-        SDL_PrivateGameControllerAddMapping(pMappingString, SDL_CONTROLLER_MAPPING_PRIORITY_DEFAULT);
-
-        i++;
-        pMappingString = s_ControllerMappings[i];
-    }
+    //const char *pMappingString = NULL;
+    //pMappingString = s_ControllerMappings[i];
+    //while (pMappingString) {
+    //    SDL_PrivateGameControllerAddMapping(pMappingString, SDL_CONTROLLER_MAPPING_PRIORITY_DEFAULT);
+    //
+    //    i++;
+    //    pMappingString = s_ControllerMappings[i];
+    //}
 
     if (SDL_GetControllerMappingFilePath(szControllerMapPath, sizeof(szControllerMapPath))) {
         SDL_GameControllerAddMappingsFromFile(szControllerMapPath);        
