--- a/src/joystick/linux/SDL_sysjoystick.c	2021-03-08 21:51:42.226895612 +0100
+++ b/src/joystick/linux/SDL_sysjoystick.c	2021-03-09 22:15:16.065286205 +0100
@@ -240,6 +240,13 @@
         SDL_free(name);
         return 0;
     }
+
+    // blacklist some devices recognized as pads
+    if (SDL_strcmp(name, "Sony Interactive Entertainment Wireless Controller Touchpad")       == 0) { SDL_free(name); return 0; }
+    if (SDL_strcmp(name, "Sony Interactive Entertainment Wireless Controller Motion Sensors") == 0) { SDL_free(name); return 0; }
+    if (SDL_strcmp(name, "Sony Interactive Entertainment Game Controller Motion Sensors")     == 0) { SDL_free(name); return 0; }
+    if (SDL_strcmp(name, "Wireless Controller Motion Sensors")                                == 0) { SDL_free(name); return 0; }
+
     *name_return = name;
     return 1;
 }
