diff --git a/src/joystick/linux/SDL_sysjoystick.c b/src/joystick/linux/SDL_sysjoystick.c
index 827a7e7..9f82452 100644
--- a/src/joystick/linux/SDL_sysjoystick.c
+++ b/src/joystick/linux/SDL_sysjoystick.c
@@ -877,9 +877,9 @@ ConfigJoystick(SDL_Joystick *joystick, int fd)
 #ifdef DEBUG_INPUT_EVENTS
                 printf("Joystick has button: 0x%x\n", i);
 #endif
-                joystick->hwdata->key_map[i] = joystick->nbuttons;
-                joystick->hwdata->has_key[i] = SDL_TRUE;
-                ++joystick->nbuttons;
+                //joystick->hwdata->key_map[i] = joystick->nbuttons;
+                //joystick->hwdata->has_key[i] = SDL_TRUE;
+                //++joystick->nbuttons;
             }
         }
         for (i = 0; i < ABS_MISC; ++i) {
