diff --git a/src/core/linux/SDL_udev.c b/src/core/linux/SDL_udev.c
index f61ee18..88ae58c 100644
--- a/src/core/linux/SDL_udev.c
+++ b/src/core/linux/SDL_udev.c
@@ -448,6 +448,8 @@ device_event(SDL_UDEV_deviceevent type, struct udev_device *dev)
     }
     
     subsystem = _this->syms.udev_device_get_subsystem(dev);
+    if(subsystem == NULL) return;
+
     if (SDL_strcmp(subsystem, "sound") == 0) {
         devclass = SDL_UDEV_DEVICE_SOUND;
     } else if (SDL_strcmp(subsystem, "input") == 0) {
