diff --git a/src/core/linux/SDL_evdev.c b/src/core/linux/SDL_evdev.c
index 4761f3e..caa52d6 100644
--- a/src/core/linux/SDL_evdev.c
+++ b/src/core/linux/SDL_evdev.c
@@ -325,13 +325,28 @@ static void SDL_EVDEV_do_text_input(unsigned short keycode) {
     struct kbentry kbe;
     unsigned char type;
     char text[2] = { 0 };
-    
+
+    if (_this->console_fd < 0) {
+      // in case of ioctl failure, try to reopen the tty (an other chance)
+      // it could be open too soon on the boot process
+      // tty is sometimes manipulated by an other program
+
+      /* We need a physical terminal (not PTS) to be able to translate key
+	 code to symbols via the kernel tables */
+      _this->console_fd = SDL_EVDEV_get_active_tty();
+
+      /* Mute the keyboard so keystrokes only generate evdev events and do not
+	 leak through to the console */
+      SDL_EVDEV_mute_keyboard(_this->console_fd, &_this->kb_mode);
+    }
+
     if (_this->console_fd < 0)
         return;
     
     shift_state = TIOCL_GETSHIFTSTATE;
     if (ioctl(_this->console_fd, TIOCLINUX, &shift_state) < 0) {
         /* TODO: error */
+        _this->console_fd = -1;
         return;
     }
 
@@ -340,6 +355,7 @@ static void SDL_EVDEV_do_text_input(unsigned short keycode) {
     
     if (ioctl(_this->console_fd, KDGKBENT, &kbe) < 0) {
         /* TODO: error */
+        _this->console_fd = -1;
         return;
     }
     
@@ -370,6 +386,7 @@ static void SDL_EVDEV_do_text_input(unsigned short keycode) {
     
         if (ioctl(_this->console_fd, KDGKBLED, &locks_state) < 0) {
             /* TODO: error */
+            _this->console_fd = -1;
             return;
         }
         
@@ -378,6 +395,7 @@ static void SDL_EVDEV_do_text_input(unsigned short keycode) {
             
             if (ioctl(_this->console_fd, KDGKBENT, &kbe) < 0) {
                 /* TODO: error */
+                _this->console_fd = -1;
                 return;
             }
         }
