--- a/configure.in	2014-03-15 22:31:42.000000000 -0400
+++ b/configure.in	2014-07-06 02:33:57.162736334 -0400
@@ -2629,11 +2629,11 @@
             *-raspberry-linux*)
                 # Raspberry Pi
                 ARCH=linux
-                RPI_CFLAGS="-I/opt/vc/include -I/opt/vc/include/interface/vcos/pthreads -I/opt/vc/include/interface/vmcs_host/linux"
+                RPI_CFLAGS=`pkg-config bcm_host --cflags`
                 CFLAGS="$CFLAGS $RPI_CFLAGS"
                 SDL_CFLAGS="$SDL_CFLAGS $RPI_CFLAGS"
                 EXTRA_CFLAGS="$EXTRA_CFLAGS $RPI_CFLAGS"
-                EXTRA_LDFLAGS="$EXTRA_LDFLAGS -L/opt/vc/lib -lbcm_host -ldl"
+                EXTRA_LDFLAGS="$EXTRA_LDFLAGS `pkg-config bcm_host --libs` -ldl"
 
                 if test x$enable_video = xyes; then
                     SOURCES="$SOURCES $srcdir/src/video/raspberry/*.c"
