--- SDL2-2.0.3/configure	2014-03-16 03:31:41.000000000 +0100
+++ sdl2-2.0.3/configure	2014-08-19 16:27:15.377772184 +0200
@@ -20926,32 +20926,21 @@
         { $as_echo "$as_me:${as_lineno-$LINENO}: checking for EGL support" >&5
 $as_echo_n "checking for EGL support... " >&6; }
         video_opengl_egl=no
-        cat confdefs.h - <<_ACEOF >conftest.$ac_ext
-/* end confdefs.h.  */
-
-         #include <EGL/egl.h>
-
-int
-main ()
-{
-
-
-  ;
-  return 0;
-}
-_ACEOF
-if ac_fn_c_try_compile "$LINENO"; then :
-
-        video_opengl_egl=yes
-
-fi
-rm -f core conftest.err conftest.$ac_objext conftest.$ac_ext
+        if test x$PKG_CONFIG != xno; then
+            if $PKG_CONFIG --exists egl ; then
+                EGL_CFLAGS=`$PKG_CONFIG --cflags egl`
+                EGL_LIBS=`$PKG_CONFIG --libs egl`
+                video_opengl_egl=yes
+            fi
+        fi
         { $as_echo "$as_me:${as_lineno-$LINENO}: result: $video_opengl_egl" >&5
 $as_echo "$video_opengl_egl" >&6; }
         if test x$video_opengl_egl = xyes; then
 
 $as_echo "#define SDL_VIDEO_OPENGL_EGL 1" >>confdefs.h
 
+            EXTRA_CFLAGS="$EXTRA_CFLAGS $EGL_CFLAGS"
+            EXTRA_LDFLAGS="$EXTRA_LDFLAGS $EGL_LIBS"
         fi
 
         { $as_echo "$as_me:${as_lineno-$LINENO}: checking for OpenGL ES v1 headers" >&5
@@ -22536,11 +22525,8 @@
             *-raspberry-linux*)
                 # Raspberry Pi
                 ARCH=linux
-                RPI_CFLAGS="-I/opt/vc/include -I/opt/vc/include/interface/vcos/pthreads -I/opt/vc/include/interface/vmcs_host/linux"
-                CFLAGS="$CFLAGS $RPI_CFLAGS"
-                SDL_CFLAGS="$SDL_CFLAGS $RPI_CFLAGS"
-                EXTRA_CFLAGS="$EXTRA_CFLAGS $RPI_CFLAGS"
-                EXTRA_LDFLAGS="$EXTRA_LDFLAGS -L/opt/vc/lib -lbcm_host -ldl"
+                #SDL_CFLAGS="$SDL_CFLAGS $EGL_CFLAGS"
+                #EXTRA_LDFLAGS="$EXTRA_LDFLAGS -L/opt/vc/lib -lbcm_host -ldl"
 
                 if test x$enable_video = xyes; then
                     SOURCES="$SOURCES $srcdir/src/video/raspberry/*.c"
