diff --git a/src/video/x11/SDL_x11events.c b/src/video/x11/SDL_x11events.c
index 2080096..1ae08d9 100644
--- a/src/video/x11/SDL_x11events.c
+++ b/src/video/x11/SDL_x11events.c
@@ -756,7 +756,9 @@ X11_DispatchEvent(_THIS)
             if (data->ic) {
                 X11_Xutf8LookupString(data->ic, &xevent.xkey, text, sizeof(text),
                                   &keysym, &status);
-            }
+            } else {
+	      X11_XLookupString(&xevent.xkey, text, sizeof(text), &keysym, NULL);
+	    }
 #else
             X11_XLookupString(&xevent.xkey, text, sizeof(text), &keysym, NULL);
 #endif
