diff --git a/Externals/VulkanMemoryAllocator/include/vk_mem_alloc.h b/Externals/VulkanMemoryAllocator/include/vk_mem_alloc.h
index dc116b0c1b..c3f1eeea85 100644
--- a/Externals/VulkanMemoryAllocator/include/vk_mem_alloc.h
+++ b/Externals/VulkanMemoryAllocator/include/vk_mem_alloc.h
@@ -2684,6 +2684,7 @@ remove them if not needed.
     #include <cassert> // for assert
     #include <algorithm> // for min, max
     #include <mutex>
+    #include <cstdio>
 #else
     #include VMA_CONFIGURATION_USER_INCLUDES_H
 #endif
