--- a/CMakeLists.txt	2018-02-11 23:40:33.891678012 +0800
+++ b/CMakeLists.txt	2018-02-11 23:44:05.135075200 +0800
@@ -52,6 +52,14 @@
     MESSAGE("libmali.so not found")
 endif()
 
+# libhybris
+if(EXISTS "${CMAKE_FIND_ROOT_PATH}/usr/lib/libhybris-common.so")
+    MESSAGE("libhybris-common.so found")
+    set(GLSystem "OpenGL ES")
+else()
+    MESSAGE("libhybris-common.so not found")
+endif()
+
 #finding necessary packages
 #-------------------------------------------------------------------------------
 if(${GLSystem} MATCHES "Desktop OpenGL")

