diff --git a/vendor/zip/src/zip.c b/vendor/zip/src/zip.c
index 1111111..2222222 100644
--- a/vendor/zip/src/zip.c
+++ b/vendor/zip/src/zip.c
@@ -208,7 +208,7 @@ static int zip_mkpath(char *path, size_t pos) {
         *p = '/';
       }

-      if (lstat(npath, &st) < 0) {
+      if (lstat(npath, (struct stat *)&st) < 0) {
         if (MZ_MKDIR(npath) == -1) {
           if (errno != EEXIST) {
             return ZIP_EMKDIR;
