diff --git a/src/duckstation-qt/qttranslations.cpp b/src/duckstation-qt/qttranslations.cpp
index 8052278..4f338d7 100644
--- a/src/duckstation-qt/qttranslations.cpp
+++ b/src/duckstation-qt/qttranslations.cpp
@@ -41,7 +41,7 @@ void QtHost::InstallTranslator()
   const QString language(QString::fromStdString(Host::GetBaseStringSettingValue("Main", "Language", "en")));
 
   // install the base qt translation first
-  const QString base_dir(QStringLiteral("%1/translations").arg(qApp->applicationDirPath()));
+  const QString base_dir(QStringLiteral("%1/translations").arg("/usr/share/duckstation"));
   QString base_path(QStringLiteral("%1/qtbase_%2.qm").arg(base_dir).arg(language));
   bool has_base_ts = QFile::exists(base_path);
   if (!has_base_ts)