diff --git a/src/frontend-common/evdev_input_source.cpp b/src/frontend-common/evdev_input_source.cpp
index 970e0ca4..c36f431c 100644
--- a/src/frontend-common/evdev_input_source.cpp
+++ b/src/frontend-common/evdev_input_source.cpp
@@ -328,7 +328,7 @@ std::optional<InputBindingKey> EvdevInputSource::ParseKeyString(const std::strin
       if (abinding == axis.name)
       {
         key.source_subtype = InputSubclass::ControllerAxis;
-        key.negative = (binding[0] == '-');
+        key.invert = (binding[0] == '-');
         key.data = axis.id;
         return key;
       }
@@ -365,7 +365,7 @@ std::string EvdevInputSource::ConvertKeyToString(InputBindingKey key)
         {
           if (static_cast<u32>(axis.id) == key.data)
           {
-            ret = fmt::format("{}/{}{}", cd->uniq, key.negative ? "-" : "+", axis.name);
+            ret = fmt::format("{}/{}{}", cd->uniq, key.invert ? "-" : "+", axis.name);
             break;
           }
         }
