diff --git a/arch/arm/boot/dts/meson8m2-m8s.dts b/arch/arm/boot/dts/meson8m2-m8s.dts
index d58db91a2f62..bcab72f7cdf0 100644
--- a/arch/arm/boot/dts/meson8m2-m8s.dts
+++ b/arch/arm/boot/dts/meson8m2-m8s.dts
@@ -80,7 +80,7 @@ sdio_pwrseq: sdio-pwrseq {
 
 	sound {
 		compatible = "amlogic,gx-sound-card";
-		model = "M8M2-M8S";
+		model = "M8S";
 		audio-routing = "I2S Encoder Playback", "I2S FIFO Playback",
 				"SPDIF Encoder Playback", "I2S FIFO Playback",
 				"SPDIF Encoder Playback", "SPDIF FIFO Playback";
diff --git a/arch/arm/boot/dts/meson8m2-mxiii.dtsi b/arch/arm/boot/dts/meson8m2-mxiii.dtsi
index faa835615f8a..b9e7b4ddbea1 100644
--- a/arch/arm/boot/dts/meson8m2-mxiii.dtsi
+++ b/arch/arm/boot/dts/meson8m2-mxiii.dtsi
@@ -66,7 +66,7 @@ sdio_pwrseq: sdio-pwrseq {
 
 	sound {
 		compatible = "amlogic,gx-sound-card";
-		model = "M8M2-MXIII";
+		model = "MXIII";
 		audio-routing = "I2S Encoder Playback", "I2S FIFO Playback",
 				"SPDIF Encoder Playback", "I2S FIFO Playback",
 				"SPDIF Encoder Playback", "SPDIF FIFO Playback";
