From 5d897becb14c7ef45ed4b833839ab5eb6c4be368 Mon Sep 17 00:00:00 2001
From: MrFixIt2001 <mrfixit2001@gmail.com>
Date: Sat, 4 Apr 2020 00:00:00 +0000
Subject: [PATCH] Fix the constant re-asking of GCC_PLUGIN

---
 scripts/gcc-plugin.sh | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/scripts/gcc-plugin.sh b/scripts/gcc-plugin.sh
index b79fd0bea8384..05c316511c8c1 100755
--- a/scripts/gcc-plugin.sh
+++ b/scripts/gcc-plugin.sh
@@ -17,3 +17,5 @@ public:
 	.test = 1
 };
 EOF
+
+exit 1
