diff --git a/src/common/fs/path_util.cpp b/src/common/fs/path_util.cpp
index f92cbe0f7b..9e3dcd9c18 100644
--- a/src/common/fs/path_util.cpp
+++ b/src/common/fs/path_util.cpp
@@ -122,12 +122,12 @@ public:
         GenerateSuyuPath(SuyuPath::CrashDumpsDir, suyu_path / CRASH_DUMPS_DIR);
         GenerateSuyuPath(SuyuPath::DumpDir, suyu_path / DUMP_DIR);
         GenerateSuyuPath(SuyuPath::IconsDir, suyu_path / ICONS_DIR);
-        GenerateSuyuPath(SuyuPath::KeysDir, suyu_path / KEYS_DIR);
+        GenerateSuyuPath(SuyuPath::KeysDir, "/userdata/bios/switch");
         GenerateSuyuPath(SuyuPath::LoadDir, suyu_path / LOAD_DIR);
-        GenerateSuyuPath(SuyuPath::LogDir, suyu_path / LOG_DIR);
+        GenerateSuyuPath(SuyuPath::LogDir, "/userdata/system/logs");
         GenerateSuyuPath(SuyuPath::NANDDir, suyu_path / NAND_DIR);
         GenerateSuyuPath(SuyuPath::PlayTimeDir, suyu_path / PLAY_TIME_DIR);
-        GenerateSuyuPath(SuyuPath::ScreenshotsDir, suyu_path / SCREENSHOTS_DIR);
+        GenerateSuyuPath(SuyuPath::ScreenshotsDir, "/userdata/screenshots/switch/suyu");
         GenerateSuyuPath(SuyuPath::SDMCDir, suyu_path / SDMC_DIR);
         GenerateSuyuPath(SuyuPath::ShaderDir, suyu_path / SHADER_DIR);
         GenerateSuyuPath(SuyuPath::TASDir, suyu_path / TAS_DIR);
