diff --git a/drivers/power/supply/cw221x_battery.c b/drivers/power/supply/cw221x_battery.c
index c26fac7..e1daa0c 100644
--- a/drivers/power/supply/cw221x_battery.c
+++ b/drivers/power/supply/cw221x_battery.c
@@ -776,12 +776,14 @@ static int cw_battery_get_property(struct power_supply *psy,
 			val->intval = POWER_SUPPLY_CAPACITY_LEVEL_FULL;
 		break;
 	case POWER_SUPPLY_PROP_STATUS:
-		if (cw_bat->ui_soc == 100)
-			val->intval = POWER_SUPPLY_STATUS_FULL;
-		else if (power_supply_is_system_supplied())
-			val->intval = POWER_SUPPLY_STATUS_CHARGING;
-		else
+		if (power_supply_is_system_supplied()) {
+			if (cw_bat->ui_soc == 100)
+				val->intval = POWER_SUPPLY_STATUS_FULL;
+			else
+				val->intval = POWER_SUPPLY_STATUS_CHARGING;
+		} else {
 			val->intval = POWER_SUPPLY_STATUS_DISCHARGING;
+		}
 		break;
 	case POWER_SUPPLY_PROP_CHARGE_FULL:
 	case POWER_SUPPLY_PROP_CHARGE_FULL_DESIGN:
