diff --git a/configure.ac b/configure.ac
index 27c6e20..9ba6441 100644
--- a/configure.ac
+++ b/configure.ac
@@ -1322,7 +1322,7 @@ UL_REQUIRES_LINUX([lsblk])
 UL_REQUIRES_BUILD([lsblk], [libblkid])
 UL_REQUIRES_BUILD([lsblk], [libmount])
 UL_REQUIRES_BUILD([lsblk], [libsmartcols])
-AM_CONDITIONAL([BUILD_LSBLK], [test "x$build_lsblk" = xyes])
+AM_CONDITIONAL([BUILD_LSBLK], [test "xyes" = xyes])
 
 
 UL_BUILD_INIT([lscpu], [check])
