--- a/vita3k/app/src/app_init.cpp
+++ b/vita3k/app/src/app_init.cpp
@@ -291,11 +291,7 @@
         root_paths.set_cache_path(portable_path / "cache" / "");
         root_paths.set_patch_path(portable_path / "patch" / "");
     } else {
-        // SDL_GetPrefPath is deferred as it creates the directory.
-        // When using a portable directory, it is not needed.
-        auto sdl_vita_fs_path = SDL_GetPrefPath(org_name, app_name);
-        auto vita_fs_path = fs_utils::utf8_to_path(sdl_vita_fs_path);
-        SDL_free(sdl_vita_fs_path);
+        auto vita_fs_path = "/userdata/saves/psvita/";
 
 #if defined(__APPLE__)
         // Store other data in the user-wide path. Otherwise we may end up dumping
@@ -386,12 +382,12 @@
 
         // shared path
         if (XDG_DATA_HOME != NULL)
-            root_paths.set_shared_path(fs::path(XDG_DATA_HOME) / app_name / "");
+            root_paths.set_shared_path(fs::path(XDG_DATA_HOME) / "psvita" / "");
         else if (home_path[0] != '\0')
             root_paths.set_shared_path(fs::path(home_path) / ".local/share" / app_name / "");
 
         // These default to being in shared path
-        root_paths.set_vita_fs_path(root_paths.get_shared_path() / app_name / "");
+        root_paths.set_vita_fs_path(root_paths.get_shared_path() / "");
         root_paths.set_patch_path(root_paths.get_shared_path() / "patch" / "");
 #endif
     }
