diff --git a/vita3k/app/src/app_init.cpp b/vita3k/app/src/app_init.cpp
index a31b88790a..eb0ad5dc5a 100644
--- a/vita3k/app/src/app_init.cpp
+++ b/vita3k/app/src/app_init.cpp
@@ -104,8 +104,8 @@ void update_viewport(EmuEnvState &state) {
 
 void init_paths(Root &root_paths) {
     const auto dir_sep = std::string{ fs::path::preferred_separator };
-    auto base_path = SDL_GetBasePath();
-    auto pref_path = SDL_GetPrefPath(org_name, app_name);
+    auto base_path = "/userdata/system/configs/vita3k/";
+    auto pref_path = "/userdata/saves/psvita/";
     root_paths.set_base_path(string_utils::utf_to_wide(base_path));
     root_paths.set_pref_path(string_utils::utf_to_wide(pref_path));
     root_paths.set_log_path(string_utils::utf_to_wide(base_path));
@@ -113,8 +113,6 @@ void init_paths(Root &root_paths) {
     root_paths.set_static_assets_path(string_utils::utf_to_wide(base_path));
     root_paths.set_shared_path(string_utils::utf_to_wide(base_path));
     root_paths.set_cache_path(fs::path(string_utils::utf_to_wide(base_path)) / "cache" / dir_sep);
-    SDL_free(base_path);
-    SDL_free(pref_path);
 
 #if defined(__linux__) && !defined(__ANDROID__) && !defined(__APPLE__)
     // XDG Data Dirs.
