diff --git a/vita3k/renderer/src/vulkan/context.cpp b/vita3k/renderer/src/vulkan/context.cpp
index cbe8ff0acb..2d8d618e7d 100644
--- a/vita3k/renderer/src/vulkan/context.cpp
+++ b/vita3k/renderer/src/vulkan/context.cpp
@@ -247,21 +247,21 @@ void VKContext::start_render_pass(bool create_descriptor_set) {
     if (!is_recording)
         start_recording();
 
-    curr_renderpass_info = {
+    curr_renderpass_info = vk::RenderPassBeginInfo{
         .renderPass = current_render_pass,
         .framebuffer = current_framebuffer
     };
 
     if (render_target->has_macroblock_sync && !ignore_macroblock) {
         // set the render area to the correct macroblock
-        curr_renderpass_info.renderArea = {
+        curr_renderpass_info.renderArea = vk::Rect2D{
             .offset = {
                 last_macroblock_x * render_target->macroblock_width,
                 last_macroblock_y * render_target->macroblock_height },
             .extent = { render_target->macroblock_width, render_target->macroblock_height }
         };
     } else {
-        curr_renderpass_info.renderArea = {
+        curr_renderpass_info.renderArea = vk::Rect2D{
             .offset = { 0, 0 },
             .extent = { render_target->width, render_target->height }
         };
