diff --git a/CMakeModules/FindGBM.cmake b/CMakeModules/FindGBM.cmake
index 1447b5e..5808d87 100644
--- a/CMakeModules/FindGBM.cmake
+++ b/CMakeModules/FindGBM.cmake
@@ -32,7 +32,7 @@ find_package_handle_standard_args(GBM
                                   VERSION_VAR GBM_VERSION)
 
 include(CheckCSourceCompiles)
-set(CMAKE_REQUIRED_LIBRARIES ${GBM_LIBRARY})
+set(CMAKE_REQUIRED_LIBRARIES ${GBM_LIBRARY} -lmali)
 check_c_source_compiles("#include <gbm.h>
 
                          int main()
