diff --git a/dlls/winebus.sys/main.c b/dlls/winebus.sys/main.c
index 974a554..e195a5f 100644
--- a/dlls/winebus.sys/main.c
+++ b/dlls/winebus.sys/main.c
@@ -768,7 +768,7 @@ static NTSTATUS udev_driver_init(void)
         .wait_code = udev_wait,
     };
 
-    bus_options.disable_hidraw = check_bus_option(L"DisableHidraw", 0);
+    bus_options.disable_hidraw = TRUE;
     if (bus_options.disable_hidraw) TRACE("UDEV hidraw devices disabled in registry\n");
     bus_options.disable_input = check_bus_option(L"DisableInput", 1);
     if (bus_options.disable_input) TRACE("UDEV input devices disabled in registry\n");
