diff --git a/Makefile b/Makefile
index f3c5d86..093270c 100644
--- a/Makefile
+++ b/Makefile
@@ -195,6 +195,20 @@ else ifneq (,$(findstring rpi,$(platform)))
 			endif
 			ASFLAGS += $(CFLAGS) -c -frename-registers -fno-strict-aliasing -ffast-math -ftree-vectorize
 		endif
+	else ifneq (,$(findstring rpi3,$(platform)))
+			FORCE_GLES = 1
+			ifneq (,$(findstring rpi3_64,$(platform)))
+				# 64-bit userspace
+				ARM_FLOAT_ABI_HARD = 0
+				CPUFLAGS += -DTARGET_LINUX_ARMv8 -frename-registers
+				CFLAGS += -march=armv8-a+crc -mcpu=cortex-a53 -mtune=cortex-a53 $(CPUFLAGS)
+				CXXFLAGS += -march=armv8-a+crc -mcpu=cortex-a53 -mtune=cortex-a53 $(CPUFLAGS)
+				ASFLAGS += $(CFLAGS) -c -frename-registers -fno-strict-aliasing -ffast-math -ftree-vectorize
+				WITH_DYNAREC=arm64
+			else
+				CFLAGS += -mtune=cortex-a53 -mfpu=neon-fp-armv8 -mfloat-abi=hard
+				CXXFLAGS += -mtune=cortex-a53 -mfpu=neon-fp-armv8 -mfloat-abi=hard
+			endif		
 	else
 		ifeq (,$(findstring mesa,$(platform)))
 			GLES = 1
@@ -207,9 +221,6 @@ else ifneq (,$(findstring rpi,$(platform)))
 		ifneq (,$(findstring rpi2,$(platform)))
 			CFLAGS += -mcpu=cortex-a7 -mfpu=neon-vfpv4 -mfloat-abi=hard
 			CXXFLAGS += -mcpu=cortex-a7 -mfpu=neon-vfpv4 -mfloat-abi=hard
-		else ifneq (,$(findstring rpi3,$(platform)))
-			CFLAGS += -mtune=cortex-a53 -mfpu=neon-fp-armv8 -mfloat-abi=hard
-			CXXFLAGS += -mtune=cortex-a53 -mfpu=neon-fp-armv8 -mfloat-abi=hard
 		endif
 		CORE_DEFINES += -DLOW_END
 	endif
