diff --git a/CMakeLists.txt b/CMakeLists.txt
index dc0960affc..9d4a723f3d 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -67,6 +67,7 @@ option(ENABLE_FC_PROFILER "Build with support for host app (Flycast) profiler" O
 option(USE_MALI "Use mali" OFF)
 option(RK3399 "Optimize for RK3399" OFF)
 option(RPI4 "Optimize for RPi4" OFF)
+option(RPI5 "Optimize for RPi5" OFF)
 option(RK3326 "Optimize for RK3326" OFF)
 option(S922X "Optimize for S922X" OFF)
 
@@ -159,6 +160,9 @@ elseif(LIBRETRO)
     elseif(RPI4)
         set(CMAKE_C_FLAGS_RELEASE "${CMAKE_C_FLAGS_RELEASE} -O3 -mcpu=cortex-a72 -mtune=cortex-a72")
 	    set(CMAKE_CXX_FLAGS_RELEASE "${CMAKE_CXX_FLAGS_RELEASE} -O3 -mcpu=cortex-a72 -mtune=cortex-a72")
+    elseif(RPI5)
+        set(CMAKE_C_FLAGS_RELEASE "${CMAKE_C_FLAGS_RELEASE} -O3 -mcpu=cortex-a76 -mtune=cortex-a76")
+	    set(CMAKE_CXX_FLAGS_RELEASE "${CMAKE_CXX_FLAGS_RELEASE} -O3 -mcpu=cortex-a76 -mtune=cortex-a76")
     elseif(S922X)
     	set(CMAKE_C_FLAGS_RELEASE "${CMAKE_C_FLAGS_RELEASE} -O3 -mcpu=cortex-a73 -mtune=cortex-a73.cortex-a53")
 	    set(CMAKE_CXX_FLAGS_RELEASE "${CMAKE_CXX_FLAGS_RELEASE} -O3 -mcpu=cortex-a73 -mtune=cortex-a73.cortex-a53")
