diff --git a/Makefile b/Makefile
index 619ef0c..e15d525 100644
--- a/Makefile
+++ b/Makefile
@@ -364,7 +364,7 @@ else ifeq ($(platform), odroid-n2)
 	CXXFLAGS += -mcpu=cortex-a73 -mtune=cortex-a73.cortex-a53 $(CPUFLAGS)
 	ASFLAGS += $(CFLAGS) -c -frename-registers -fno-strict-aliasing -ffast-math -ftree-vectorize
 	PLATFORM_EXT := unix
-	WITH_DYNAREC=arm64
+	WITH_DYNAREC=arm
 	HAVE_GENERIC_JIT = 0
 	HAVE_LTCG = 0
 
