diff --git a/Makefile.libretro b/Makefile.libretro
index c2e9acd..cca851c 100644
--- a/Makefile.libretro
+++ b/Makefile.libretro
@@ -281,6 +281,16 @@ else ifeq ($(platform), rpi3)
         CFLAGS += -fomit-frame-pointer -ffast-math
         HAVE_NEON = 1
 
+# Raspberry Pi 3 (64-bit)
+else ifeq ($(platform), rpi3_64)
+        TARGET := $(TARGET_NAME)_libretro.so
+        fpic := -fPIC
+        SHARED := -shared -Wl,--version-script=libretro/link.T -Wl,--no-undefined
+        CFLAGS += -DARM
+        CFLAGS += -mcpu=cortex-a53 -mtune=cortex-a53 -funsafe-math-optimizations
+        CFLAGS += -fomit-frame-pointer -ffast-math
+        HAVE_NEON = 1
+
 # Raspberry Pi 4 (64-bit)
 else ifeq ($(platform), rpi4)
         TARGET := $(TARGET_NAME)_libretro.so
