diff --git a/Makefile.libretro b/Makefile.libretro
index ef2f9e7d1c..c3394dd86e 100644
--- a/Makefile.libretro
+++ b/Makefile.libretro
@@ -313,6 +313,16 @@ else ifeq ($(platform), rpi4)
         CFLAGS += -fomit-frame-pointer -ffast-math -fPIC
         HAVE_NEON = 1
 
+# Raspberry Pi 5 (64-bit)
+else ifeq ($(platform), rpi5)
+        TARGET := $(TARGET_NAME)_libretro.so
+        fpic := -fPIC
+        SHARED := -shared -Wl,--version-script=libretro/link.T -Wl,--no-undefined
+        CFLAGS += -DARM
+        CFLAGS += -mcpu=cortex-a76 -mtune=cortex-a76 -funsafe-math-optimizations
+        CFLAGS += -fomit-frame-pointer -ffast-math -fPIC
+        HAVE_NEON = 1
+
 # Classic Platforms ####################
 # Platform affix = classic_<ISA>_<µARCH>
 # Help at https://modmyclassic.com/comp
