--- a/Makefile	2015-12-21 00:22:55.000000000 +0100
+++ b/Makefile	2015-12-21 09:44:37.642079459 +0100
@@ -20,9 +20,7 @@
 	rm -Rf bin include lib local node_modules compass/.sass-cache
 
 install:
-	virtualenv --no-site-packages .
-	bin/pip install -r pip-requirements/basic.txt
-	bin/python manage.py migrate
+	python manage.py migrate
 
 install-dev: install
 	bundle install --gemfile=compass/Gemfile
