--- a/CMakeLists.txt	2021-03-14 21:33:52.327608475 +0100
+++ b/CMakeLists.txt	2021-03-14 21:35:24.632299509 +0100
@@ -12,9 +12,9 @@
 SET(DOCDIR ${DATADIR} CACHE PATH "Docs destination")
 ELSE(WIN32)
 SET(SHAREDIR "/usr/share" CACHE PATH "System share dir location")
-SET(GAMES_SHAREDIR "${SHAREDIR}/games/" CACHE PATH "Game data root dir")
+SET(GAMES_SHAREDIR "${SHAREDIR}" CACHE PATH "Game data root dir")
 SET(DATADIR "${GAMES_SHAREDIR}/commandergenius")
-SET(APPDIR games CACHE PATH "Binary destination")
+SET(APPDIR bin CACHE PATH "Binary destination")
 SET(DOCDIR ${DATADIR} CACHE PATH "Docs destination")
 SET(ICONDIR "${SHAREDIR}/icons/hicolor" CACHE PATH "Standard icon installation dir")
 SET(DESKTOPFILESDIR "${SHAREDIR}/applications" CACHE PATH "Application installation dir")
