diff --git a/Config/Supermodel.ini b/Config/Supermodel.ini
index 5b4f46a..aa8acfe 100644
--- a/Config/Supermodel.ini
+++ b/Config/Supermodel.ini
@@ -29,6 +29,163 @@
 ;
 
 
+[ daytona2 ]
+MusicVolume = 120
+SoundVolume = 100
+Balance = -10
+InputJoy1XSaturation = 155
+
+[ dayto2pe ]
+MusicVolume = 120
+SoundVolume = 100
+Balance = -10
+InputJoy1XSaturation = 155
+
+[ dirtdvls ]
+MusicVolume = 100
+SoundVolume = 130
+Balance = -20
+
+[ getbassur ]
+WideScreen =0
+WideBackground =0
+MusicVolume = 100
+SoundVolume = 110
+Balance = 10
+
+[ eca ]
+MusicVolume = 100
+SoundVolume = 100
+Balance = -35
+
+[ fvipers2 ]
+LegacySoundDSP = 1
+MusicVolume = 100
+SoundVolume = 170
+Balance = 10
+
+[ harley ]
+MusicVolume = 100
+SoundVolume = 100
+Balance = 10
+
+[ lamachin ]
+MusicVolume = 100
+SoundVolume = 115
+Balance = 0
+InputSystem=rawinput
+
+[ lemans24 ]
+MusicVolume = 100
+SoundVolume = 110
+Balance = 0
+InputJoy1XSaturation = 110
+
+[ lostwsga ]
+MusicVolume = 100
+SoundVolume = 105
+Balance = 20
+InputSystem=rawinput
+CrosshairStyle = bmp
+
+[ magtruck ]
+MusicVolume = 100
+SoundVolume = 100
+Balance = 15
+
+[ mgtrkbad ]
+MusicVolume = 100
+SoundVolume = 100
+Balance = 15
+
+[ oceanhun ]
+MusicVolume = 100
+SoundVolume = 100
+Balance = 0
+InputSystem=rawinput
+
+[ oceanhuna ]
+MusicVolume = 100
+SoundVolume = 100
+Balance = 0
+InputSystem=rawinput
+
+[ scud ]
+MusicVolume = 200
+SoundVolume = 60
+Balance = 0
+InputJoy1XSaturation = 155
+
+[ scudplus ]
+MusicVolume = 200
+SoundVolume = 60
+Balance = 0
+InputJoy1XSaturation = 155
+
+[ skichamp ]
+MusicVolume = 100
+SoundVolume = 200
+Balance = 10
+ForceFeedback=1
+
+[ spikeofe ]
+MusicVolume = 120
+SoundVolume = 120
+Balance = 0
+
+[ spikeout ]
+MusicVolume = 90
+SoundVolume = 90
+Balance = 0
+
+[ srally2 ]
+SoundVolume = 155
+MusicVolume = 140
+Balance = 15
+PowerPCFrequency=75
+InputJoy1XSaturation = 150
+
+[ swtrilgy ]
+MusicVolume = 100
+SoundVolume = 100
+Balance = 0
+
+[ vf3 ]
+MusicVolume = 100
+SoundVolume = 110
+Balance = -5
+
+[ vf3tb ]
+MusicVolume = 100
+SoundVolume = 110
+Balance = -5
+
+[ von2 ]
+MusicVolume = 100
+SoundVolume = 100
+Balance = 0
+
+[ vs2 ]
+MusicVolume = 100
+SoundVolume = 110
+Balance = -10
+WideScreen =0
+WideBackground =0
+
+[ vs298 ]
+MusicVolume = 100
+SoundVolume = 110
+Balance = -10
+WideScreen =0
+WideBackground =0
+
+[ vs2v991 ]
+MusicVolume = 100
+SoundVolume = 110
+Balance = -10
+WideScreen =0
+WideBackground =0
+
 [ Global ]
 
 ; Graphics
