diff --git a/t83x/kernel/drivers/gpu/arm/midgard/platform/devicetree/meson_main2.c b/t83x/kernel/drivers/gpu/arm/midgard/platform/devicetree/meson_main2.c
index 7687f92..7122448 100644
--- a/t83x/kernel/drivers/gpu/arm/midgard/platform/devicetree/meson_main2.c
+++ b/t83x/kernel/drivers/gpu/arm/midgard/platform/devicetree/meson_main2.c
@@ -68,6 +68,7 @@ inline int kbase_platform_dvfs_event(struct kbase_device *kbdev, u32 utilisation
     mali_gpu_utilization_callback(utilisation*255/100);
     return 1;
 }
+EXPORT_SYMBOL(kbase_platform_dvfs_event);
 
 struct kbase_platform_funcs_conf dt_funcs_conf = {
     .platform_init_func = platform_dt_init_func,
