diff --git a/Core/Screenshot.cpp b/Core/Screenshot.cpp
index dcf5989..b31bd90 100644
--- a/Core/Screenshot.cpp
+++ b/Core/Screenshot.cpp
@@ -18,7 +18,7 @@
 #ifdef USING_QT_UI
 #include <QtGui/QImage>
 #else
-#include <libpng17/png.h>
+#include <libpng16/png.h>
 #include "ext/jpge/jpge.h"
 #endif
 
diff --git a/Core/TextureReplacer.cpp b/Core/TextureReplacer.cpp
index 3e06c73..5348d9f 100644
--- a/Core/TextureReplacer.cpp
+++ b/Core/TextureReplacer.cpp
@@ -16,7 +16,7 @@
 // https://github.com/hrydgard/ppsspp and http://www.ppsspp.org/.
 
 #ifndef USING_QT_UI
-#include <libpng17/png.h>
+#include <libpng16/png.h>
 #endif
 
 #include <algorithm>
diff --git a/ext/native/image/png_load.cpp b/ext/native/image/png_load.cpp
index 145d084..70d4695 100644
--- a/ext/native/image/png_load.cpp
+++ b/ext/native/image/png_load.cpp
@@ -5,7 +5,7 @@
 #ifdef USING_QT_UI
 #include <QtGui/QImage>
 #else
-#include "libpng17/png.h"
+#include "libpng16/png.h"
 #endif
 
 #include "png_load.h"
