diff --git a/src/api/udev/JoystickInterfaceUdev.cpp b/src/api/udev/JoystickInterfaceUdev.cpp
index 8ee6dfa..0459927 100644
--- a/src/api/udev/JoystickInterfaceUdev.cpp
+++ b/src/api/udev/JoystickInterfaceUdev.cpp
@@ -92,8 +92,13 @@ bool CJoystickInterfaceUdev::ScanForJoysticks(JoystickVector& joysticks)
 
      if (devnode != nullptr)
      {
-       JoystickPtr joystick = JoystickPtr(new CJoystickUdev(dev, devnode));
-       joysticks.push_back(joystick);
+       CJoystickUdev *j = new CJoystickUdev(dev, devnode);
+       if(j->isInitialized()) {
+	 JoystickPtr joystick = JoystickPtr(j);
+	 joysticks.push_back(joystick);
+       } else {
+	 delete j;
+       }
      }
 
      udev_device_unref(dev);
diff --git a/src/api/udev/JoystickUdev.h b/src/api/udev/JoystickUdev.h
index dd61644..6adb940 100644
--- a/src/api/udev/JoystickUdev.h
+++ b/src/api/udev/JoystickUdev.h
@@ -58,6 +58,7 @@ namespace JOYSTICK
     virtual bool Initialize(void) override;
     virtual void Deinitialize(void) override;
     virtual void ProcessEvents(void) override;
+    bool isInitialized() { return m_bInitialized; }
 
   protected:
     // implementation of CJoystick
