diff --git a/Source/Core/DolphinQt/MainWindow.cpp b/Source/Core/DolphinQt/MainWindow.cpp
index 20572d8..b79b2a7 100644
--- a/Source/Core/DolphinQt/MainWindow.cpp
+++ b/Source/Core/DolphinQt/MainWindow.cpp
@@ -1768,7 +1768,12 @@ void MainWindow::Show()
   // If the booting of a game was requested on start up, do that now
   if (m_pending_boot != nullptr)
   {
+    m_menu_bar->hide();
+    m_tool_bar->hide();
+    m_game_list->hide();
     StartGame(std::move(m_pending_boot));
     m_pending_boot.reset();
+  } else {
+    m_tool_bar->show();
   }
 }
