--- a/Source/Core/VideoBackends/OGL/OGLRender.cpp	2020-12-31 00:27:53.998709857 +0100
+++ b/Source/Core/VideoBackends/OGL/OGLRender.cpp	2020-12-31 00:28:40.414557344 +0100
@@ -736,10 +736,6 @@
   g_Config.VerifyValidity();
   UpdateActiveConfig();
 
-  OSD::AddMessage(fmt::format("Video Info: {}, {}, {}", g_ogl_config.gl_vendor,
-                              g_ogl_config.gl_renderer, g_ogl_config.gl_version),
-                  5000);
-
   if (!g_ogl_config.bSupportsGLBufferStorage && !g_ogl_config.bSupportsGLPinnedMemory)
   {
     OSD::AddMessage(fmt::format("Your OpenGL driver does not support {}_buffer_storage.",
