diff --git a/pcsx2/CDVD/CDVD.cpp b/pcsx2/CDVD/CDVD.cpp
index 757d7ae..79c4937 100644
--- a/pcsx2/CDVD/CDVD.cpp
+++ b/pcsx2/CDVD/CDVD.cpp
@@ -25,7 +25,7 @@
 #include "CDVD_internal.h"
 #include "CDVDisoReader.h"
 
-#include "GS.h" // for gsVideoMode
+#include "../GS.h" // for gsVideoMode
 #include "Elfheader.h"
 #include "ps2/BiosTools.h"
 
diff --git a/pcsx2/DebugTools/DebugInterface.cpp b/pcsx2/DebugTools/DebugInterface.cpp
index d29c2b0..2a89378 100644
--- a/pcsx2/DebugTools/DebugInterface.cpp
+++ b/pcsx2/DebugTools/DebugInterface.cpp
@@ -20,7 +20,7 @@
 #include "R5900.h"
 #include "Debug.h"
 #include "VU.h"
-#include "GS.h" // Required for gsNonMirroredRead()
+#include "../GS.h" // Required for gsNonMirroredRead()
 #include "Counters.h"
 
 #include "R3000A.h"
diff --git a/pcsx2/PAD/Linux/Global.h b/pcsx2/PAD/Linux/Global.h
index 71e4a8a..7754542 100644
--- a/pcsx2/PAD/Linux/Global.h
+++ b/pcsx2/PAD/Linux/Global.h
@@ -15,6 +15,9 @@
 
 #pragma once
 
+static const int GAMEPAD_NUMBER = 2;
+static const int MAX_KEYS = 24;
+
 #include <stdio.h>
 #include <assert.h>
 #include <array>
@@ -34,9 +37,6 @@
 
 #define PADdefs
 
-static const u32 GAMEPAD_NUMBER = 2;
-static const u32 MAX_KEYS = 24;
-
 #include "Config.h"
 
 enum gamePadValues
diff --git a/pcsx2/gui/App.h b/pcsx2/gui/App.h
index d9decd7..ade53b9 100644
--- a/pcsx2/gui/App.h
+++ b/pcsx2/gui/App.h
@@ -36,7 +36,7 @@
 class DisassemblyDialog;
 struct HostKeyEvent;
 
-#include "GS.h"
+#include "../GS.h"
 #include "System.h"
 #include "System/SysThreads.h"
 
diff --git a/pcsx2/gui/CpuUsageProvider.cpp b/pcsx2/gui/CpuUsageProvider.cpp
index d5db2b1..17fd536 100644
--- a/pcsx2/gui/CpuUsageProvider.cpp
+++ b/pcsx2/gui/CpuUsageProvider.cpp
@@ -22,7 +22,7 @@
 #include "System/SysThreads.h"
 #endif
 
-#include "GS.h"
+#include "../GS.h"
 #include "MTVU.h"
 
 void AllPCSX2Threads::LoadWithCurrentTimes()
