diff --git a/cmake/BuildParameters.cmake b/cmake/BuildParameters.cmake
index bb6108619..9c1294a4a 100644
--- a/cmake/BuildParameters.cmake
+++ b/cmake/BuildParameters.cmake
@@ -214,7 +214,7 @@ option(USE_PGO_OPTIMIZE "Enable PGO optimization (use profile)")
 if(MSVC)
 	add_compile_options("$<$<COMPILE_LANGUAGE:CXX>:/Zc:externConstexpr>")
 else()
-	add_compile_options(-pipe -fvisibility=hidden -pthread -fno-builtin-strcmp -fno-builtin-memcmp -mfpmath=sse)
+	add_compile_options(-pipe -pthread -fno-builtin-strcmp -fno-builtin-memcmp -mfpmath=sse -fno-operator-names)
 
 	# -fno-operator-names should only be for C++ files, not C files.
 	add_compile_options($<$<COMPILE_LANGUAGE:CXX>:-fno-operator-names>)
