diff --git a/common/Pcsx2Defs.h b/common/Pcsx2Defs.h
index 557713c0c6..6d0659a297 100644
--- a/common/Pcsx2Defs.h
+++ b/common/Pcsx2Defs.h
@@ -57,7 +57,7 @@ static constexpr unsigned int __pagemask = __pagesize - 1;
 // __forceinline_odr is for member functions that are defined in headers. MSVC can't specify
 // inline and __forceinline at the same time, but it required to not get ODR errors in GCC.
 
-#define __forceinline __attribute__((always_inline, unused))
+#define __forceinline __attribute__((unused))
 #define __forceinline_odr __forceinline inline
 #define __noinline __attribute__((noinline))
 #define __noreturn __attribute__((noreturn))
