diff --git a/pcsx2/gui/AppConfig.cpp b/pcsx2/gui/AppConfig.cpp
index db789f2587..e4e4ecb068 100644
--- a/pcsx2/gui/AppConfig.cpp
+++ b/pcsx2/gui/AppConfig.cpp
@@ -196,12 +196,12 @@ namespace PathDefs
 
 	wxDirName GetSnapshots()
 	{
-		return GetDocuments() + Base::Snapshots();
+		return wxDirName("/userdata/screenshots");
 	}
 
 	wxDirName GetBios()
 	{
-		return GetDocuments() + Base::Bios();
+		return wxDirName("/userdata/bios");
 	}
 
 	wxDirName GetCheats()
@@ -225,7 +225,7 @@ namespace PathDefs
 
 	wxDirName GetSavestates()
 	{
-		return GetDocuments() + Base::Savestates();
+		return wxDirName("/userdata/saves/ps2");
 	}
 
 	wxDirName GetMemoryCards()
