diff --git a/pcsx2/gui/AppConfig.cpp b/pcsx2/gui/AppConfig.cpp
index 507348e73..71cd04d64 100644
--- a/pcsx2/gui/AppConfig.cpp
+++ b/pcsx2/gui/AppConfig.cpp
@@ -202,22 +202,22 @@ namespace PathDefs
 
 	wxDirName GetSnapshots()
 	{
-		return GetDocuments() + Base::Snapshots();
+		return wxDirName("/userdata/screenshots");
 	}
 
 	wxDirName GetBios()
 	{
-		return GetDocuments() + Base::Bios();
+		return wxDirName("/userdata/bios");
 	}
 
 	wxDirName GetCheats()
 	{
-		return GetDocuments() + Base::Cheats();
+		return wxDirName("/userdata/cheats/ps2/cheats");
 	}
 
 	wxDirName GetCheatsWS()
 	{
-		return GetDocuments() + Base::CheatsWS();
+		return wxDirName("/userdata/cheats/ps2/cheats_ws");
 	}
 
 	wxDirName GetDocs()
@@ -231,12 +231,12 @@ namespace PathDefs
 
 	wxDirName GetSavestates()
 	{
-		return GetDocuments() + Base::Savestates();
+		return wxDirName("/userdata/saves/ps2");
 	}
 
 	wxDirName GetMemoryCards()
 	{
-		return GetDocuments() + Base::MemoryCards();
+		return wxDirName("/userdata/saves/ps2");
 	}
 
 	wxDirName GetSettings()
@@ -246,7 +246,7 @@ namespace PathDefs
 
 	wxDirName GetLogs()
 	{
-		return GetDocuments() + Base::Logs();
+		return wxDirName("/userdata/system/logs");
 	}
 
 	wxDirName GetResources()
@@ -266,7 +266,7 @@ namespace PathDefs
 
 	wxDirName GetCache()
 	{
-		return GetDocuments() + Base::Cache();
+		return wxDirName("/userdata/system/cache/ps2");
 	}
 
 	wxDirName GetTextures()
@@ -1092,7 +1092,7 @@ void RelocateLogfile()
 {
 	g_Conf->Folders.Logs.Mkdir();
 
-	std::string newlogname(StringUtil::wxStringToUTF8String(Path::CombineWx(g_Conf->Folders.Logs.ToString(), L"emuLog.txt")));
+	std::string newlogname(StringUtil::wxStringToUTF8String(Path::CombineWx(g_Conf->Folders.Logs.ToString(), L"PCSX2-emuLog.txt")));
 
 	if ((emuLog != NULL) && (emuLogName != newlogname))
 	{
