diff --git a/cmake/Pcsx2Utils.cmake b/cmake/Pcsx2Utils.cmake.new
index 4c2d126e95..9f00c3be0f 100644
--- a/cmake/Pcsx2Utils.cmake
+++ b/cmake/Pcsx2Utils.cmake.new
@@ -3,7 +3,7 @@ function(detect_operating_system)
 	message(STATUS "CMake System Name: ${CMAKE_SYSTEM_NAME}")
 
 	# LINUX wasn't added until CMake 3.25.
-	if (CMAKE_VERSION VERSION_LESS 3.25.0 AND CMAKE_SYSTEM_NAME MATCHES "Linux")
+	if (CMAKE_SYSTEM_NAME MATCHES "Linux")
 		# Have to make it visible in this scope as well for below.
 		set(LINUX TRUE PARENT_SCOPE)
 		set(LINUX TRUE)
