diff --git a/pcsx2/SaveState.cpp b/pcsx2/SaveState.cpp
index 8e54f7b..e38d270 100644
--- a/pcsx2/SaveState.cpp
+++ b/pcsx2/SaveState.cpp
@@ -79,11 +79,8 @@ wxString SaveStateBase::GetFilename( int slot )
 	wxString serialName( DiscSerial );
 	if (serialName.IsEmpty()) serialName = L"BIOS";
 
-	return (EmuFolders::Savestates +
-		pxsFmt( L"%s (%08X).%02d.p2s", WX_STR(serialName), ElfCRC, slot )).GetFullPath();
-
-	//return (g_Conf->Folders.Savestates +
-	//	pxsFmt( L"%08X.%03d", ElfCRC, slot )).GetFullPath();
+	return (g_Conf->Folders.Savestates +
+		pxsFmt( L"%08X.%03d", ElfCRC, slot )).GetFullPath();
 }
 
 SaveStateBase::SaveStateBase( SafeArray<u8>& memblock )
@@ -923,4 +920,4 @@ void SaveState_UnzipFromDisk(const wxString& filename)
 	reader->Read(buffer.GetPtr(), foundInternal->GetSize());
 
 	memLoadingState(buffer).FreezeBios().FreezeInternals();
-}
\ No newline at end of file
+}
