diff --git a/pcsx2/SaveState.cpp b/pcsx2/SaveState.cpp
index 540e0fe..b6a69ab 100644
--- a/pcsx2/SaveState.cpp
+++ b/pcsx2/SaveState.cpp
@@ -112,17 +112,8 @@ wxString SaveStateBase::GetSavestateFolder( int slot, bool isSavingOrLoading )
 		CRCvalue = L"None";
 	}
 
-	wxFileName dirname = wxFileName::DirName(g_Conf->FullpathToSaveState(serialName, CRCvalue));
-
-	if (isSavingOrLoading)
-	{
-		if (!wxDirExists(g_Conf->FullpathToSaveState(serialName, CRCvalue)))
-		{
-			wxMkdir(g_Conf->FullpathToSaveState(serialName, CRCvalue));
-		}
-	}
-	return (dirname.GetPath() + "/" +
-			pxsFmt( L"%s (%s).%02d.p2s", WX_STR(serialName), WX_STR(CRCvalue), slot ));
+	return (g_Conf->Folders.Savestates +
+		pxsFmt( L"%08X.%03d", ElfCRC, slot )).GetFullPath();
 }
 
 SaveStateBase::SaveStateBase( SafeArray<u8>& memblock )
