diff --git a/cmake/SearchForStuff.cmake b/cmake/SearchForStuff.cmake
index d0fc74480e..2c5b4102a3 100644
--- a/cmake/SearchForStuff.cmake
+++ b/cmake/SearchForStuff.cmake
@@ -17,7 +17,7 @@ find_package(ZLIB REQUIRED) # v1.3, but Mac uses the SDK version.
 find_package(Zstd 1.5.5 REQUIRED)
 find_package(LZ4 REQUIRED)
 find_package(WebP REQUIRED) # v1.3.2, spews an error on Linux because no pkg-config.
-find_package(SDL2 2.30.2 REQUIRED)
+find_package(SDL2 2.28 REQUIRED)
 find_package(Freetype 2.11.1 REQUIRED)
 
 if(USE_VULKAN)