diff --git a/pcsx2/gui/Panels/MemoryCardListView.cpp b/pcsx2/gui/Panels/MemoryCardListView.cpp
index c1eebe3d07..fc3fcd6d5f 100644
--- a/pcsx2/gui/Panels/MemoryCardListView.cpp
+++ b/pcsx2/gui/Panels/MemoryCardListView.cpp
@@ -52,10 +52,10 @@ void BaseMcdListView::LoadSaveColumns( IniInterface& ini )
 
 		int width = GetColumnWidth(col);
 
-		ini.Entry( label, width, cinfo.width );
+		//ini.Entry( label, width, cinfo.width );
 		
-		if (ini.IsLoading())
-			SetColumnWidth(col, width);
+		//if (ini.IsLoading())
+		//	SetColumnWidth(col, width);
 	}
 }
 
