diff --git a/pcsx2/GS/Renderers/Vulkan/VKShaderCache.cpp b/pcsx2/GS/Renderers/Vulkan/VKShaderCache.cpp.new
index 905092ee68..a6d0a6c99f 100644
--- a/pcsx2/GS/Renderers/Vulkan/VKShaderCache.cpp
+++ b/pcsx2/GS/Renderers/Vulkan/VKShaderCache.cpp.new
@@ -109,19 +109,8 @@ std::optional<VKShaderCache::SPIRVCodeVector> VKShaderCache::CompileShaderToSPV(
 	options.SetSourceLanguage(shaderc_source_language_glsl);
 	options.SetTargetEnvironment(shaderc_target_env_vulkan, 0);
 
-	if (debug)
-	{
-		options.SetGenerateDebugInfo();
-		if (GSDeviceVK::GetInstance()->GetOptionalExtensions().vk_khr_shader_non_semantic_info)
-			options.SetEmitNonSemanticDebugInfo();
-
-		options.SetOptimizationLevel(shaderc_optimization_level_zero);
-	}
-	else
-	{
-		options.SetOptimizationLevel(shaderc_optimization_level_performance);
-	}
-
+	options.SetOptimizationLevel(shaderc_optimization_level_performance);
+	
 	const shaderc::SpvCompilationResult result = s_shaderc_compiler->CompileGlslToSpv(
 		source.data(), source.length(), static_cast<shaderc_shader_kind>(stage), "source", "main", options);
 	if (result.GetCompilationStatus() != shaderc_compilation_status_success)
