diff --git a/pcsx2/SaveState.cpp b/pcsx2/SaveState.cpp
index a3dd229..79a8ba8 100644
--- a/pcsx2/SaveState.cpp
+++ b/pcsx2/SaveState.cpp
@@ -121,8 +121,7 @@ wxString SaveStateBase::GetSavestateFolder( int slot, bool isSavingOrLoading )
 		CRCvalue = L"None";
 	}
 
-	return (g_Conf->Folders.Savestates +
-		pxsFmt( L"%08X.%03d", ElfCRC, slot )).GetFullPath();
+	return (g_Conf->Folders.Savestates + Path::GetFilename(g_Conf->CurrentIso)).GetFullPath() + pxsFmt( L".%d", slot );
 }
 #endif
 
