diff --git a/pcsx2/Config.h b/pcsx2/Config.h
index 3b12a96..382b19b 100644
--- a/pcsx2/Config.h
+++ b/pcsx2/Config.h
@@ -1150,6 +1150,7 @@ struct Pcsx2Config
 		BackupSavestate : 1,
 		SavestateZstdCompression : 1,
 		McdFolderAutoManage : 1,
+	        AutoIncrementSlot : 1,
 
 		HostFs : 1,
 
diff --git a/pcsx2/Pcsx2Config.cpp b/pcsx2/Pcsx2Config.cpp
index 9a715c0..ef746da 100644
--- a/pcsx2/Pcsx2Config.cpp
+++ b/pcsx2/Pcsx2Config.cpp
@@ -1699,6 +1699,7 @@ Pcsx2Config::Pcsx2Config()
 	BackupSavestate = true;
 	SavestateZstdCompression = true;
 	WarnAboutUnsafeSettings = true;
+	AutoIncrementSlot = false;
 
 	// To be moved to FileMemoryCard pluign (someday)
 	for (uint slot = 0; slot < 8; ++slot)
@@ -1734,6 +1735,7 @@ void Pcsx2Config::LoadSaveCore(SettingsWrapper& wrap)
 	SettingsWrapBitBool(EnableDiscordPresence);
 	SettingsWrapBitBool(InhibitScreensaver);
 	SettingsWrapBitBool(HostFs);
+	SettingsWrapBitBool(AutoIncrementSlot);
 
 	SettingsWrapBitBool(BackupSavestate);
 	SettingsWrapBitBool(SavestateZstdCompression);
diff --git a/pcsx2/VMManager.cpp b/pcsx2/VMManager.cpp
index 3324fc5..46e02c4 100644
--- a/pcsx2/VMManager.cpp
+++ b/pcsx2/VMManager.cpp
@@ -2022,6 +2022,12 @@ bool VMManager::SaveState(const char* filename, bool zip_on_thread, bool backup_
 
 bool VMManager::SaveStateToSlot(s32 slot, bool zip_on_thread)
 {
+	//increment the slot before saving
+	if(EmuConfig.AutoIncrementSlot) {
+	  SaveStateSelectorUI::SelectNextSlot(false);
+	  slot = SaveStateSelectorUI::GetCurrentSlot();
+	}
+
 	const std::string filename(GetCurrentSaveStateFileName(slot));
 	if (filename.empty())
 		return false;
