
--- recalbox-configgen-4.0.0/configgen/emulatorlauncher.py	2016-02-16 20:24:20.000000000 +0100
+++ recalbox-configgen-4.0.0.n/configgen/emulatorlauncher.py	2016-03-02 14:21:23.828553944 +0100
@@ -11,6 +11,7 @@ from generators.mupen.mupenGenerator imp
 from generators.kodi.kodiGenerator import KodiGenerator
 from generators.moonlight.moonlightGenerator import MoonlightGenerator
 from generators.scummvm.scummvmGenerator import ScummVMGenerator
+from generators.dosbox.dosboxGenerator import DosBoxGenerator
 from generators.configManager import ConfigManager
 import controllersConfig as controllers
 import utils.runner as runner
@@ -24,7 +25,8 @@ generators = {
     'mupen64plus': MupenGenerator(),
     'kodi': KodiGenerator(),
     'moonlight': MoonlightGenerator(),
-    'scummvm': ScummVMGenerator()
+    'scummvm': ScummVMGenerator(),
+    'dosbox': DosBoxGenerator()
 }
 
 # List emulators with their cores rest mupen64, scummvm
@@ -73,6 +75,7 @@ emulators["psx"] = Emulator(name='psx',
 emulators["cavestory"] = Emulator(name='cavestory', emulator='libretro', core='nxengine')
 emulators["imageviewer"] = Emulator(name='imageviewer', emulator='libretro', core='imageviewer')
 emulators["scummvm"] = Emulator(name='scummvm', emulator='scummvm', videomode='default')
+emulators["pc"] = Emulator(name='pc', emulator='dosbox', videomode='default')
 
 emulators["kodi"] = Emulator(name='kodi', emulator='kodi', videomode='default')
 emulators["moonlight"] = Emulator(name='moonlight', emulator='moonlight')
diff -purN recalbox-configgen-4.0.0/configgen/generators/dosbox/dosboxGenerator.py recalbox-configgen-4.0.0.n/configgen/generators/dosbox/dosboxGenerator.py
--- recalbox-configgen-4.0.0/configgen/generators/dosbox/__init__.py    1970-01-01 01:00:00.000000000 +0100
+++ recalbox-configgen-4.0.0.n/configgen/generators/dosbox/__init__.py  2016-03-02 16:23:34.700516621 +0100
@@ -0,0 +1 @@
+
--- recalbox-configgen-4.0.0/configgen/generators/dosbox/dosboxGenerator.py	1970-01-01 01:00:00.000000000 +0100
+++ recalbox-configgen-4.0.0.n/configgen/generators/dosbox/dosboxGenerator.py	2016-03-02 15:01:55.728541563 +0100
@@ -0,0 +1,27 @@
+#!/usr/bin/env python
+import Command
+import recalboxFiles
+from generators.Generator import Generator
+import os.path
+import glob
+
+
+class DosBoxGenerator(Generator):
+    # Main entry of the module
+    # Return command
+    def generate(self, system, rom, playersControllers):
+        # Find rom path
+        gameDir = rom
+        batFile = gameDir + "/dosbox.bat"
+        gameConfFile = gameDir + "/dosbox.cfg"
+           
+        commandArray = ["dosbox", 
+			"-userconf", 
+			"-exit", 
+			"""{}""".format(batFile),
+			"-c", """set ROOT=""{}""".format(gameDir)]
+        if os.path.isfile(gameConfFile):
+            commandArray.append("-conf")
+            commandArray.append("""{}""".format(gameConfFile))
+			
+        return Command.Command(videomode='default', array=commandArray)
--- recalbox-configgen-4.0.0/setup.py	2016-02-16 20:24:20.000000000 +0100
+++ recalbox-configgen-4.0.0.n/setup.py	2016-03-02 15:43:12.712528952 +0100
@@ -4,5 +4,5 @@ from distutils.core import setup
 setup(name='recalbox-configgen',
       version='1.0',
       py_modules=['configgen'],
-      packages=['configgen', 'configgen.generators', 'configgen.generators.libretro', 'configgen.generators.fba2x', 'configgen.generators.mupen', 'configgen.generators.kodi', 'configgen.generators.moonlight', 'configgen.generators.scummvm','configgen.settings','configgen.utils'],
+      packages=['configgen', 'configgen.generators', 'configgen.generators.libretro', 'configgen.generators.fba2x', 'configgen.generators.mupen', 'configgen.generators.kodi', 'configgen.generators.moonlight', 'configgen.generators.scummvm','configgen.generators.dosbox','configgen.settings','configgen.utils'],
       )
