diff --git a/CMakeLists.txt b/CMakeLists.txt
index 1074505..e4827c0 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -27,7 +27,7 @@ set(CMAKE_INCLUDE_CURRENT_DIR TRUE)
 ################################################################
 
 # Set to "-march=native" on non x86 platforms
-set(MACHINE_OPTIONS "-march=x86-64 -mtune=generic")
+set(MACHINE_OPTIONS "-march=armv7-a -mtune=cortex-a15.cortex-a7")
 
 # Base options
 set(BASE_OPTIONS "-Ofast -fomit-frame-pointer -ffast-math -Wall")
