diff --git a/projects/unix/Makefile b/projects/unix/Makefile
index a440d0d..8188030 100755
--- a/projects/unix/Makefile
+++ b/projects/unix/Makefile
@@ -122,6 +122,10 @@ ifneq ("$(filter mips,$(HOST_CPU))","")
   NO_ASM := 1
   $(warning Architecture "$(HOST_CPU)" not officially supported.')
 endif
+ifneq ("$(filter aarch64%,$(HOST_CPU))","")
+    CPU := AARCH64
+    PIC ?= 1
+endif
 ifeq ("$(CPU)","NONE")
   $(error CPU type "$(HOST_CPU)" not supported.  Please file bug report at 'http://code.google.com/p/mupen64plus/issues')
 endif
