diff --git a/engine/Makefile b/engine/Makefile
index da9bf1b50b..a76f657d2b 100644
--- a/engine/Makefile
+++ b/engine/Makefile
@@ -202,6 +202,22 @@ BUILD_DEBUG     = 1
 endif
 endif
 
+ifdef BUILD_ARM
+TARGET 	        = $(VERSION_NAME).elf
+TARGET_FINAL    = $(VERSION_NAME)
+TARGET_PLATFORM = ARM
+BUILD_LINUX     = 1
+BUILD_SDL       = 1
+BUILD_GFX       = 1
+BUILD_PTHREAD   = 1
+BUILD_SDL_IO    = 1
+BUILD_VORBIS    = 1
+BUILD_WEBM      = 1
+BUILDING        = 1
+INCLUDES        = $(SDKPATH)/include
+OBJTYPE         = elf
+endif
+
 ifdef BUILD_WII
 TARGET 	        = $(VERSION_NAME).elf
 TARGET_MAP      = $(TARGET).map
@@ -264,6 +280,9 @@ ifdef BUILD_PSP
 INCS           += psp
 endif
 
+ifdef BUILD_ARM
+CFLAGS         += -DLINUX
+endif
 
 ifdef BUILD_WII
 INCS           += wii
