diff --git a/engine/Makefile b/engine/Makefile
index 284a7d9dbb..d046b42f51 100644
--- a/engine/Makefile
+++ b/engine/Makefile
@@ -507,7 +507,7 @@ OBJS            = $(MAIN)
 #----------------------------------------------------------------------------------------------------
 
 CFLAGS 	       += $(addprefix -I", $(addsuffix ", $(INCS))) $(ARCHFLAGS) -D$(TARGET_PLATFORM)
-CFLAGS 	       += -g -Wall -Werror -Wno-format-truncation -Wno-stringop-truncation -fsigned-char -std=gnu99
+CFLAGS 	       += -g -fsigned-char -std=gnu99
 
 
 ifndef BUILD_DARWIN
