diff --git a/engine/openbor.h b/engine/openbor.h
index fbdc36a978..4c1f59b738 100644
--- a/engine/openbor.h
+++ b/engine/openbor.h
@@ -442,14 +442,10 @@ typedef enum
 // Entity types.
 typedef enum
 {
-    FLAG_MOVEUP         = (1 << SDID_MOVEUP),
-    FLAG_MOVEDOWN       = (1 << SDID_MOVEDOWN),
     FLAG_PADLEFT        = (1 << 2), // FCANEW
     FLAG_PADRIGHT       = (1 << 3), // FCANEW
     FLAG_PADUP          = (1 << 4), // FCANEW
     FLAG_PADDOWN        = (1 << 5), // FCANEW
-    FLAG_MOVELEFT       = (1 << SDID_MOVELEFT),
-    FLAG_MOVERIGHT      = (1 << SDID_MOVERIGHT),
     FLAG_ATTACK         = (1 << SDID_ATTACK),
     FLAG_ATTACK2        = (1 << SDID_ATTACK2),
     FLAG_ATTACK3        = (1 << SDID_ATTACK3),
