diff --git a/apps/ymir-sdl3/src/app/app.cpp b/apps/ymir-sdl3/src/app/app.cpp
index 9648613bec..5dc76dae1a 100644
--- a/apps/ymir-sdl3/src/app/app.cpp
+++ b/apps/ymir-sdl3/src/app/app.cpp
@@ -512,7 +512,7 @@ void App::RunEmulator() {
         const auto onboardedPath = updatesPath / ".onboarded";
         const bool onboarded = std::filesystem::is_regular_file(onboardedPath);
         if (!onboarded) {
-            m_windowManagerService.UpdateOnboardingWindow().Open = true;
+            m_windowManagerService.UpdateOnboardingWindow().Open = false;
         }
 #endif
 