diff --git a/sources/src/machdep/m68kops.h b/sources/src/machdep/m68kops.h
index 578fb8f..cff2e94 100644
--- a/sources/src/machdep/m68kops.h
+++ b/sources/src/machdep/m68kops.h
@@ -11,7 +11,7 @@
 
 #ifndef ANDROID
 
-#if defined(__CELLOS_LV2__) || defined(_WIN32) || defined(__x86_64__) || defined(ARM) || defined(WIIU)
+#if defined(__CELLOS_LV2__) || defined(_WIN32) || defined(__x86_64__) || defined(ARM) || defined(WIIU) || defined(AARCH64)
 #warning "TEST TESTFLAGS no opt"
 
 #ifdef WIIU
diff --git a/sources/src/machdep/maccess.h b/sources/src/machdep/maccess.h
index 2916cbf..4459cd0 100644
--- a/sources/src/machdep/maccess.h
+++ b/sources/src/machdep/maccess.h
@@ -6,7 +6,7 @@
   * Copyright 1996 Bernd Schmidt
   */
 
-#if defined(__CELLOS_LV2__) || defined(WIIU)
+#if defined(__CELLOS_LV2__) || defined(WIIU) || defined(AARCH64)
 
 STATIC_INLINE uae_u32 do_get_mem_long (uae_u32 *a)
 {
