--- a/Src/Graphics/New3D/R3DShaderTriangles.h	2021-03-03 14:04:46.971560780 +0100
+++ b/Src/Graphics/New3D/R3DShaderTriangles.h	2021-03-03 14:08:49.444885783 +0100
@@ -3,7 +3,7 @@
 
 static const char *vertexShaderR3D = R"glsl(
 
-#version 120
+#version 130
 
 // uniforms
 uniform float	modelScale;
@@ -61,7 +61,7 @@
 
 static const char *fragmentShaderR3D = R"glsl(
 
-#version 120
+#version 130
 
 uniform sampler2D tex1;			// base tex
 uniform sampler2D tex2;			// micro tex (optional)
