--- a/install.sh	2016-03-29 19:13:27.103062871 +0200
+++ b/install.sh	2016-03-29 19:19:06.397261127 +0200
@@ -16,12 +16,15 @@
 }
 
 # global install
-DEST_HOME=/usr/local
+if [ -z "${DEST_HOME}" ]
+then
+  DEST_HOME=/usr/local
+fi
 # for testing 
 # DEST_HOME=${HOME}/INSTALL/usr/local
 DEST_BIN=${DEST_HOME}/bin
 DEST_SHARE=${DEST_HOME}/lib/p7zip
-DEST_MAN=${DEST_HOME}/man
+DEST_MAN=${DEST_HOME}/share/man
 DEST_SHARE_DOC=${DEST_HOME}/share/doc/p7zip
 DEST_DIR=
 [ "$1" ] && DEST_BIN="$1"
