diff --git a/SConstruct b/SConstruct
index f1e4c02..43c818d 100644
--- a/SConstruct
+++ b/SConstruct
@@ -116,11 +116,11 @@ elif 'BUILD' in env and env['BUILD'] == 'custom':
 else:
     env.Append(CPPFLAGS = ['-g', '-O3', '-Wall', '-ansi', '-pedantic'])
 
-env.ParseConfig(env['PKG_CONFIG'] + " --cflags --libs dbus-glib-1 | sed 's/-I/-isystem/g'")
-env.ParseConfig(env['PKG_CONFIG'] + " --cflags --libs glib-2.0 | sed 's/-I/-isystem/g'")
-env.ParseConfig(env['PKG_CONFIG'] + " --cflags --libs gthread-2.0 | sed 's/-I/-isystem/g'")
-env.ParseConfig(env['PKG_CONFIG'] + " --cflags --libs libusb-1.0 | sed 's/-I/-isystem/g'")
-env.ParseConfig(env['PKG_CONFIG'] + " --cflags --libs libudev | sed 's/-I/-isystem/g'")
+env.ParseConfig(env['PKG_CONFIG'] + " --cflags --libs dbus-glib-1")
+env.ParseConfig(env['PKG_CONFIG'] + " --cflags --libs glib-2.0")
+env.ParseConfig(env['PKG_CONFIG'] + " --cflags --libs gthread-2.0")
+env.ParseConfig(env['PKG_CONFIG'] + " --cflags --libs libusb-1.0")
+env.ParseConfig(env['PKG_CONFIG'] + " --cflags --libs libudev")
 
 f = open("VERSION")
 package_version = f.read()
