--- a/src/evdev_helper.hpp	2015-03-02 20:23:39.061343553 +0100
+++ b/src/evdev_helper.hpp	2015-03-02 20:23:56.577132575 +0100
@@ -19,7 +19,6 @@
 #ifndef HEADER_EVDEV_HELPER_HPP
 #define HEADER_EVDEV_HELPER_HPP
 
-#include <X11/Xlib.h>
 
 #include "enum_box.hpp"
 #include "ui_event.hpp"
@@ -62,8 +61,6 @@
 public:
   X11KeysymEnum();
 
-private:
-  void process_keymap(Display* dpy);
 };
 
 extern EvDevRelEnum  evdev_rel_names;
