diff --git a/Makefile b/Makefile
index 73cc132..7383495 100644
--- a/Makefile
+++ b/Makefile
@@ -87,6 +87,14 @@ ifneq (,$(findstring unix,$(platform)))
         endif
     endif
 
+    # RockPro64
+    else ifneq (,$(findstring rockpro64,$(platform)))
+            CFLAGS += -fomit-frame-pointer -ffast-math -DARM -marm -mfloat-abi=hard -march=armv7ve -mtune=cortex-a72.cortex-a53
+
+    # Amlogic S905
+    else ifneq (,$(findstring s905,$(platform)))
+            CFLAGS = -mcpu=cortex-a53
+
 # Classic Platforms ####################
 # Platform affix = classic_<ISA>_<µARCH>
 # Help at https://modmyclassic.com/comp
