diff --git a/Makefile b/Makefile
index ef4e1d1..1795e99 100644
--- a/Makefile
+++ b/Makefile
@@ -83,11 +83,29 @@ ifneq (,$(findstring unix,$(platform)))
         # ODROID-XU3 & ODROID-XU4
         else ifneq (,$(findstring odroidxu,$(platform)))
             CFLAGS += -march=armv7ve -mtune=cortex-a15.cortex-a7
+        # ODROID-N2
+        else ifneq (,$(findstring odroidn2,$(platform)))
+            CFLAGS += -march=armv8-a+crc -mfpu=neon-fp-armv8 -mcpu=cortex-a73 -mtune=cortex-a73.cortex-a53
         # ODROID-U2, -U3, -X & -X2
         else
             CFLAGS += -mtune=cortex-a9
         endif
     endif
+    
+    # RockPro64
+    ifneq (,$(findstring rockpro64,$(platform)))
+         CFLAGS += -fomit-frame-pointer -ffast-math -DARM -marm -mfloat-abi=hard -march=armv7ve -mtune=cortex-a72.cortex-a53
+    endif
+         
+    # Amlogic S905
+    ifneq (,$(findstring s905,$(platform)))
+         CFLAGS = -mcpu=cortex-a53
+    endif
+    
+    # Amlogic S912
+    ifneq (,$(findstring s912,$(platform)))
+         CFLAGS = -mcpu=cortex-a53
+    endif    
 
 # Classic Platforms ####################
 # Platform affix = classic_<ISA>_<µARCH>
