diff --git a/Makefile b/Makefile
index a4c01b8..46d7b6e 100644
--- a/Makefile
+++ b/Makefile
@@ -71,6 +71,10 @@ ifneq (,$(findstring unix,$(platform)))
             CFLAGS = -mcpu=cortex-a53
         endif
 
+    # Amlogic S905
+    else ifneq (,$(findstring s905,$(platform)))
+        CFLAGS = -mcpu=cortex-a53
+
     # ODROIDs
     else ifneq (,$(findstring odroid,$(platform)))
         CFLAGS += -fomit-frame-pointer -ffast-math -DARM -marm -mfloat-abi=hard
