diff --git a/es-core/src/Settings.cpp b/es-core/src/Settings.cpp
index 85d80f2..ec22034 100644
--- a/es-core/src/Settings.cpp
+++ b/es-core/src/Settings.cpp
@@ -172,7 +172,7 @@ void Settings::setDefaults()
 	#ifdef _RPI_
 		mStringMap["AudioDevice"] = "PCM";
 	#else
-		mStringMap["AudioDevice"] = "Master";
+		mStringMap["AudioDevice"] = "Playback";
 	#endif
 
 	mStringMap["AudioCard"] = "default";
