From 2a5bb6a50175f4a8e725cb625f5d91f7272b066c Mon Sep 17 00:00:00 2001
From: Martin Blumenstingl <martin.blumenstingl@googlemail.com>
Date: Sat, 3 Jun 2017 20:05:48 +0200
Subject: [PATCH 077/122] ARM: dts: meson8m2-m8s: enable the NAND flash - WiP

Signed-off-by: Martin Blumenstingl <martin.blumenstingl@googlemail.com>
---
 arch/arm/boot/dts/meson8m2-m8s.dts | 74 ++++++++++++++++++++++++++++++
 1 file changed, 74 insertions(+)

diff --git a/arch/arm/boot/dts/meson8m2-m8s.dts b/arch/arm/boot/dts/meson8m2-m8s.dts
index d814b160943c..98de0eb5d67d 100644
--- a/arch/arm/boot/dts/meson8m2-m8s.dts
+++ b/arch/arm/boot/dts/meson8m2-m8s.dts
@@ -272,10 +272,84 @@ &ir_receiver {
 	pinctrl-names = "default";
 };
 
+
 &mali {
 	mali-supply = <&vddee>;
 };
 
+&nfc {
+	/* TODO: status = "okay"; */
+
+	pinctrl-0 = <&nand_pins>, <&nand_ce0_pins>, <&nand_rb0_pins>;
+	pinctrl-names = "default";
+
+	nand@0 {
+		reg = <0>;
+
+		nand-on-flash-bbt;
+
+		partitions {
+			compatible = "fixed-partitions";
+			#address-cells = <1>;
+			#size-cells = <1>;
+
+			partition@2 {
+				label = "2GiB";
+				reg = <0x0 0xffffffff>;
+				read-only;
+			};
+
+			partition@0 {
+				label = "nand_page0_cfg";
+				reg = <0x0 0x4000>;
+				read-only;
+			};
+
+			partition@4000 {
+				label = "u-boot";
+				reg = <0x4000 0x60000>;
+				read-only;
+			};
+
+			partition@400000 {
+				label = "u-boot-env";
+				reg = <0x400000 0x10000>;
+				read-only;
+			};
+
+			partition@1000000 {
+				label = "boot";
+				reg = <0x1000000 0x400000>;
+				read-only;
+			};
+
+			partition@1400000 {
+				label = "uboot_p1";
+				reg = <0x1400000 0x8000000>;
+				read-only;
+			};
+
+			partition@d000000 {
+				label = "cache";
+				reg = <0xd000000 0x19000000>;
+				read-only;
+			};
+
+			partition@33000000 {
+				label = "system";
+				reg = <0x33000000 0x1f000000>;
+				read-only;
+			};
+
+			partition@85000000 {
+				label = "data";
+				reg = <0x85000000 0xf6000000>;
+				read-only;
+			};
+		};
+	};
+};
+
 &saradc {
 	status = "okay";
 	vref-supply = <&vcc_1v8>;
-- 
2.25.1

