diff --git a/Makefile b/Makefile
index e42d169..ac1419d 100644
--- a/Makefile
+++ b/Makefile
@@ -25,11 +25,30 @@ LIBM := -lm
 endif
 LIBS :=
 
-ifeq ($(platform), unix)
+ifneq (,$(findstring unix,$(platform)))
    TARGET := $(TARGET_NAME)_libretro.so
    fpic := -fPIC
    SHARED := -shared -Wl,--version-script=libretro/link.T -Wl,--no-undefined
    CFLAGS += -fno-builtin -fno-exceptions
+   
+   # Raspberry
+   ifneq (,$(findstring rpi,$(platform)))
+      USE_DYNAREC = 1
+      ifneq (,$(findstring rpi1,$(platform)))
+         CFLAGS += -marm -march=armv6j -mfpu=vfp -mfloat-abi=hard
+      else ifneq (,$(findstring rpi2,$(platform)))
+         CFLAGS += -marm -mcpu=cortex-a7 -mfpu=neon-vfpv4 -mfloat-abi=hard
+      else ifneq (,$(findstring rpi3,$(platform)))
+         ifneq (,$(findstring rpi3_64,$(platform)))
+            CFLAGS += -mcpu=cortex-a53 -mtune=cortex-a53
+		   else
+            CFLAGS += -marm -mcpu=cortex-a53 -mfpu=neon-fp-armv8 -mfloat-abi=hard
+         endif
+      else ifneq (,$(findstring rpi4_64,$(platform)))
+         CFLAGS += -march=armv8-a+crc+simd -mtune=cortex-a72
+      endif
+   endif
+
 else ifeq ($(platform), osx)
    TARGET := $(TARGET_NAME)_libretro.dylib
    fpic := -fPIC
