diff --git a/scripts/src/3rdparty.lua b/scripts/src/3rdparty.lua
index 1ea65fa4..c2399b5a 100644
--- a/scripts/src/3rdparty.lua
+++ b/scripts/src/3rdparty.lua
@@ -1385,12 +1385,12 @@ end
 	}
 
 	if _OPTIONS["targetos"]=="linux" or _OPTIONS["targetos"]=="netbsd" or _OPTIONS["targetos"]=="openbsd" then
-		if _OPTIONS["NO_X11"]=="1" then
+--		if _OPTIONS["NO_X11"]=="1" then
 			defines {
-				"BGFX_CONFIG_RENDERER_OPENGLES=1",
+				"BGFX_CONFIG_RENDERER_OPENGLES=0",
 				"BGFX_CONFIG_RENDERER_OPENGL=0",
 			}
-		end
+--		end
 	end
 
 	files {
