diff --git a/CMakeLists.txt b/CMakeLists.txt
index b339f89..0bb3cf5 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -66,3 +66,11 @@ endif()
 add_subdirectory (libde265)
 add_subdirectory (dec265)
 add_subdirectory (enc265)
+
+set(libdir ${CMAKE_INSTALL_PREFIX}/lib)
+set(VERSION 1.0.3)
+set(LIBS ${CMAKE_INSTALL_PREFIX}/lib)
+set(includedir ${CMAKE_INSTALL_PREFIX}/include)
+
+configure_file(libde265.pc.in libde265.pc @ONLY)
+install(FILES ${CMAKE_BINARY_DIR}/libde265.pc DESTINATION lib/pkgconfig)
diff --git a/libde265/de265.h b/libde265/de265.h
index 5cbac45..b4dbbeb 100644
--- a/libde265/de265.h
+++ b/libde265/de265.h
@@ -38,9 +38,9 @@ extern "C" {
 
 #if defined(_MSC_VER) && !defined(LIBDE265_STATIC_BUILD)
   #ifdef LIBDE265_EXPORTS
-  #define LIBDE265_API __declspec(dllexport)
+  #define LIBDE265_API
   #else
-  #define LIBDE265_API __declspec(dllimport)
+  #define LIBDE265_API
   #endif
 #elif HAVE_VISIBILITY
   #ifdef LIBDE265_EXPORTS
-- 
2.25.1
