diff --git a/CMakeLists.txt b/CMakeLists.txt
index 9e1db11..9327358 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -61,6 +61,12 @@ if(MSVC)
   include_directories ("${PROJECT_SOURCE_DIR}/extra")
   add_definitions(-DHAVE_STDINT_H)
   add_definitions(-DHAVE_STDBOOL_H)
+  include(CheckSymbolExists)
+  check_symbol_exists(_ARM_ "Windows.h" _ARM_)
+  check_symbol_exists(_ARM64_ "Windows.h" _ARM64_)
+  if(_ARM_ OR _ARM64_)
+    set(DISABLE_SSE ON)
+  endif()
 endif()
 
 add_subdirectory (libde265)
-- 
2.25.1

