diff --color -uri commander-60e30e2e8f0819bc118cf546ee7fd52fc8b91e87/def.h od-commander-lbrpdx/def.h
--- commander-60e30e2e8f0819bc118cf546ee7fd52fc8b91e87/def.h    2021-07-01 00:54:17.000000000 -0700
+++ od-commander-lbrpdx/def.h   2022-05-22 08:47:55.838773761 -0700
@@ -5,7 +5,11 @@
 #define INHIBIT(X) /* X */

 #ifndef FONTS
+#ifdef BATOCERA_HANDHELD
 #define FONTS {"/usr/share/fonts/truetype/dejavu/DejaVuSansCondensed.ttf",10},{"FreeSans.ttf",10},{"DroidSansFallback.ttf",9}
+#else
+#define FONTS {"/usr/share/fonts/truetype/dejavu/DejaVuSansCondensed.ttf",20},{"FreeSans.ttf",20},{"DroidSansFallback.ttf",18}
+#endif // BATOCERA_HANDHELD
 #endif

 // Font stack to use on screens with low DPI.
@@ -14,12 +18,20 @@
 #endif

 // Panel
+#ifdef BATOCERA_HANDHELD
 #define HEADER_H 17
+#else
+#define HEADER_H 30
+#endif // BATOCERA_HANDHELD
 #define HEADER_H_PHYS static_cast<int>(HEADER_H * screen.ppu_y)
 #define HEADER_PADDING_TOP 3
 #define HEADER_PADDING_TOP_PHYS static_cast<int>(HEADER_PADDING_TOP * screen.ppu_y)

+#ifdef BATOCERA_HANDHELD
 #define FOOTER_H 13
+#else
+#define FOOTER_H 25
+#endif // BATOCERA_HANDHELD
 #define FOOTER_H_PHYS static_cast<int>(FOOTER_H * screen.ppu_y)
 #define FOOTER_PADDING_TOP 1
 #define FOOTER_PADDING_TOP_PHYS static_cast<int>(FOOTER_PADDING_TOP * screen.ppu_y)
@@ -27,14 +39,23 @@
 #define Y_LIST HEADER_H
 #define Y_LIST_PHYS static_cast<int>(Y_LIST * screen.ppu_y)

+#ifdef BATOCERA_HANDHELD
 #define LINE_HEIGHT 15
+#else
+#define LINE_HEIGHT 30
+#endif // BATOCERA_HANDHELD
 #define LINE_HEIGHT_PHYS static_cast<int>(LINE_HEIGHT * screen.ppu_y)
 #define NB_VISIBLE_LINES ((screen.actual_h - FOOTER_H_PHYS - HEADER_H_PHYS - 1) / LINE_HEIGHT_PHYS + 1)
 #define NB_FULLY_VISIBLE_LINES ((screen.actual_h - FOOTER_H_PHYS - HEADER_H_PHYS) / LINE_HEIGHT_PHYS)

 // Dialogs
+#ifdef BATOCERA_HANDHELD
 #define DIALOG_BORDER 2
 #define DIALOG_PADDING 8
+#else
+#define DIALOG_BORDER 5
+#define DIALOG_PADDING 4
+#endif // BATOCERA_HANDHELD
 // Colors
 #define COLOR_TITLE_BG 102, 85, 74
 #define COLOR_KEY 255, 0, 255
diff --color -uri commander-60e30e2e8f0819bc118cf546ee7fd52fc8b91e87/keyboard.cpp od-commander-lbrpdx/keyboard.cpp
--- commander-60e30e2e8f0819bc118cf546ee7fd52fc8b91e87/keyboard.cpp     2021-07-01 00:54:17.000000000 -0700
+++ od-commander-lbrpdx/keyboard.cpp    2022-05-22 09:43:15.816641259 -0700
@@ -23,14 +23,25 @@
 static std::string kSpace = " ";
 static std::string kTab = "\t";

+#ifdef BATOCERA_HANDHELD
 constexpr std::size_t kKeyBorderW = 1;
-constexpr std::size_t kMaxKeyW = 19 + kKeyBorderW;
-constexpr std::size_t kMaxKeyH = 17 + kKeyBorderW;
+constexpr std::size_t kMaxKeyW = 22 + kKeyBorderW;
+constexpr std::size_t kMaxKeyH = 24 + kKeyBorderW;
 constexpr std::size_t kMaxKeyGap = 2;
 constexpr std::size_t kFrameBorder = 2;
-constexpr std::size_t kTextFieldHeight = 19;
+constexpr std::size_t kTextFieldHeight = 30;
 constexpr std::size_t kTextFieldMarginBottom = 2;
 constexpr std::size_t kTextFieldBorder = 2;
+#else
+constexpr std::size_t kKeyBorderW = 2;
+constexpr std::size_t kMaxKeyW = 30 + kKeyBorderW;
+constexpr std::size_t kMaxKeyH = 30 + kKeyBorderW;
+constexpr std::size_t kMaxKeyGap = 2;
+constexpr std::size_t kFrameBorder = 2;
+constexpr std::size_t kTextFieldHeight = 35;
+constexpr std::size_t kTextFieldMarginBottom = 2;
+constexpr std::size_t kTextFieldBorder = 2;
+#endif

 KeyboardLayout UsAsciiLayout(bool support_tabs)
 {
