--- a/bauh/view/core/gems.py
+++ b/bauh/view/core/gems.py
@@ -1,6 +1,6 @@
+import importlib.util
 import inspect
 import os
-import pkgutil
 from logging import Logger
 from typing import List, Generator
 
@@ -49,10 +49,10 @@
                 logger.warning(f"gem '{f.name}' could not be loaded because it was marked as forbidden in '{FORBIDDEN_GEMS_FILE}'")
                 continue
 
-            loader = pkgutil.find_loader(f'bauh.gems.{f.name}.controller')
+            module_name = f'bauh.gems.{f.name}.controller'
 
-            if loader:
-                module = loader.load_module()
+            if importlib.util.find_spec(module_name):
+                module = importlib.import_module(module_name)
 
                 manager_class = find_manager(module)
 
