diff --git a/bauh/view/core/config.py b/bauh/view/core/config.py
index 9737bc5..873fdd2 100644
--- a/bauh/view/core/config.py
+++ b/bauh/view/core/config.py
@@ -29,7 +29,7 @@ class CoreConfigManager(YAMLConfigManager):
             },
             'suggestions': {
                 'enabled': True,
-                'by_type': 10
+                'by_type': 200
             },
             'ui': {
                 'table': {
