diff --git a/bauh/gems/flatpak/flatpak.py b/bauh/gems/flatpak/flatpak.py
index 8586af4..81a9d53 100755
--- a/bauh/gems/flatpak/flatpak.py
+++ b/bauh/gems/flatpak/flatpak.py
@@ -285,6 +285,8 @@ def search(version: Version, word: str, installation: str, app_id: bool = False)
     res = run_cmd('{} search {} --{}'.format('flatpak', word, installation))
 
     found = []
+    if res is None:
+        return found
 
     split_res = res.split('\n')
 
