--- a/src/citra/emu_window/emu_window_sdl2.cpp	2020-08-08 17:35:14.438514786 +0200
+++ b/src/citra/emu_window/emu_window_sdl2.cpp	2020-08-08 17:35:24.730700881 +0200
@@ -226,7 +226,7 @@
     SDL_GL_MakeCurrent(render_window, window_context);
     SDL_GL_SetSwapInterval(1);
     while (IsOpen()) {
-        VideoCore::g_renderer->TryPresent(100);
+        VideoCore::g_renderer->TryPresent();
         SDL_GL_SwapWindow(render_window);
     }
     SDL_GL_MakeCurrent(render_window, nullptr);
