--- a/backends/events/sdl/sdl-events.cpp	2016-03-28 12:48:42.559648267 +0200
+++ b/backends/events/sdl/sdl-events.cpp	2016-03-28 13:17:26.609933552 +0200
@@ -48,6 +48,7 @@
 #define JOY_BUT_PERIOD 1
 #define JOY_BUT_SPACE 4
 #define JOY_BUT_F5 5
+#define JOY_BUT_VKEYBOARD 7
 
 #if SDL_VERSION_ATLEAST(2, 0, 0)
 static uint32 convUTF8ToUTF32(const char *src) {
@@ -699,6 +700,9 @@
 			event.kbd.keycode = Common::KEYCODE_F5;
 			event.kbd.ascii = mapKey(SDLK_F5, (SDLMod)ev.key.keysym.mod, 0);
 			break;
+		case JOY_BUT_VKEYBOARD: // Toggles virtual keyboard
+			event.type = Common::EVENT_VIRTUAL_KEYBOARD;
+			break;
 		}
 	}
 	return true;
@@ -730,6 +734,9 @@
 			event.kbd.keycode = Common::KEYCODE_F5;
 			event.kbd.ascii = mapKey(SDLK_F5, (SDLMod)ev.key.keysym.mod, 0);
 			break;
+		case JOY_BUT_VKEYBOARD: // Toggles virtual keyboard
+			// Handled in key down
+			break;
 		}
 	}
 	return true;
