diff --git a/Makefile b/Makefile
index c3d5d3b30d..e038107a46 100644
--- a/Makefile
+++ b/Makefile
@@ -75,6 +75,8 @@ ifneq (,$(findstring unix,$(platform)))
          endif
       else ifneq (,$(findstring rpi4_64,$(platform)))
          FLAGS += -march=armv8-a+crc+simd -mtune=cortex-a72
+      else ifneq (,$(findstring rpi5_64,$(platform)))
+         FLAGS += -mcpu=cortex-a76 -mtune=cortex-a76
       endif
    endif
 
