diff --git a/pyproject.toml b/pyproject.toml
index 9ddc4025a5..4b2b5ace7a 100644
--- a/pyproject.toml
+++ b/pyproject.toml
@@ -12,7 +12,7 @@ readme = "README.md"
 requires-python = ">=3.7"
 classifiers = [
   "Programming Language :: Python :: 3",
-  "License :: OSI Approved :: GNU License",
+  "License :: OSI Approved :: GNU General Public License v2 (GPLv2)",
   "Operating System :: OS Independent",
 ]
 version = "0.0.4"
