diff --git a/main.c b/main.c
index d5ee4de..70bc160 100644
--- a/main.c
+++ b/main.c
@@ -38,7 +38,8 @@ static int opt_clear = 1,
 	   opt_stretch = 0,
 	   opt_delay = 0,
 	   opt_enlarge = 0,
-	   opt_ignore_aspect = 0;
+	   opt_ignore_aspect = 0,
+	   opt_rotate = 0;
 
 
 
@@ -205,7 +206,7 @@ int show_image(char *filename)
 	int delay = opt_delay, retransform = 1;
 	
 	int transform_stretch = opt_stretch, transform_enlarge = opt_enlarge, transform_cal = (opt_stretch == 2),
-	    transform_iaspect = opt_ignore_aspect, transform_rotation = 0;
+	    transform_iaspect = opt_ignore_aspect, transform_rotation = opt_rotate;
 	
 	struct image i;
 	
@@ -501,7 +502,7 @@ int main(int argc, char **argv)
 		return(1);
 	}
 	
-	while((c = getopt_long_only(argc, argv, "hcauifks:er", long_options, NULL)) != EOF)
+	while((c = getopt_long_only(argc, argv, "hcauifks:erxyz", long_options, NULL)) != EOF)
 	{
 		switch(c)
 		{
@@ -535,6 +536,15 @@ int main(int argc, char **argv)
 			case 'r':
 				opt_ignore_aspect = 1;
 				break;
+			case 'x':
+				opt_rotate = 1;
+				break;
+			case 'y':
+				opt_rotate = 2;
+				break;
+			case 'z':
+				opt_rotate = 3;
+				break;
 		}
 	}
 	
