diff --git a/projects/unix/Makefile b/projects/unix/Makefile
index 42295bb..3c64dee 100755
--- a/projects/unix/Makefile
+++ b/projects/unix/Makefile
@@ -117,6 +117,10 @@ ifneq ("$(filter arm%,$(HOST_CPU))","")
     $(warning Architecture "$(HOST_CPU)" not officially supported.')
   endif
 endif
+ifneq ("$(filter aarch64%,$(HOST_CPU))","")
+    CPU := AARCH64
+    PIC ?= 1
+endif
 ifeq ("$(CPU)","NONE")
   $(error CPU type "$(HOST_CPU)" not supported.  Please file bug report at 'http://code.google.com/p/mupen64plus/issues')
 endif
