diff --git a/engine/Makefile b/engine/Makefile
index 8d6a559..85ab4fe 100644
--- a/engine/Makefile
+++ b/engine/Makefile
@@ -156,6 +156,22 @@ BUILD_DEBUG     = 1
 endif
 endif
 
+ifdef BUILD_ARM
+TARGET 	        = $(VERSION_NAME).elf
+TARGET_FINAL    = $(VERSION_NAME)
+TARGET_PLATFORM = ARM
+BUILD_LINUX     = 1
+BUILD_SDL       = 1
+BUILD_GFX       = 1
+BUILD_PTHREAD   = 1
+BUILD_SDL_IO    = 1
+BUILD_VORBIS    = 1
+BUILD_WEBM      = 1
+BUILDING        = 1
+INCLUDES        = $(SDKPATH)/include
+OBJTYPE         = elf
+endif
+
 ifdef BUILD_WII
 TARGET 	        = $(VERSION_NAME).elf
 TARGET_MAP      = $(TARGET).map
@@ -485,6 +501,9 @@ ifdef BUILD_PANDORA
 CFLAGS         += -DLINUX
 endif
 
+ifdef BUILD_ARM
+CFLAGS         += -DLINUX
+endif
 
 ifdef BUILD_WII
 CFLAGS 	       += -DCACHE_BACKGROUNDS -DREVERSE_COLOR -D__ppc__ $(MACHDEP) -Wl,-Map,$(TARGET_MAP),-wrap,wiiuse_register
