diff --git a/CMakeLists.txt b/CMakeLists.txt
index 123a3082ae..da2321ac76 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -201,7 +201,7 @@ macro(yuzu_find_packages)
 endmacro()
 
 if (NOT YUZU_USE_BUNDLED_BOOST)
-    find_package(Boost 1.73.0 CONFIG COMPONENTS context headers QUIET)
+    find_package(Boost 1.75.0 COMPONENTS context REQUIRED)
 endif()
 if (Boost_FOUND)
     set(Boost_LIBRARIES Boost::boost)
