diff --git a/Makefile.libretro b/Makefile.libretro
index 5f9b7ae..3a5d121 100644
--- a/Makefile.libretro
+++ b/Makefile.libretro
@@ -93,6 +93,26 @@ else ifeq ($(platform), rpi3)
 	PLATFORM_DEFINES += -DARM
 	HAVE_NEON = 1
 
+# Raspberry Pi 3 64 bits
+else ifeq ($(platform), rpi3_64)
+	TARGET := $(TARGET_NAME)_libretro.so
+	fpic = -fPIC
+	SHARED := -shared -Wl,-version-script=link.T -Wl,-no-undefined
+	PLATFORM_DEFINES += -march=armv8-a+crc+simd -mtune=cortex-a53
+	PLATFORM_DEFINES += -fomit-frame-pointer -ffast-math
+	PLATFORM_DEFINES += -DARM
+	HAVE_NEON = 1
+
+# Raspberry Pi 4 64 bits
+else ifeq ($(platform), rpi4_64)
+	TARGET := $(TARGET_NAME)_libretro.so
+	fpic = -fPIC
+	SHARED := -shared -Wl,-version-script=link.T -Wl,-no-undefined
+	PLATFORM_DEFINES += -march=armv8-a+crc+simd -mtune=cortex-a72
+	PLATFORM_DEFINES += -fomit-frame-pointer -ffast-math
+	PLATFORM_DEFINES += -DARM
+	HAVE_NEON = 1
+
 #RETROFW
 else ifeq ($(platform), retrofw)
 	TARGET := $(TARGET_NAME)_libretro.so
