--- pcsx2x6-v0.2.22/pcsx2/BuildVersion.cpp	2025-06-29 12:47:12.000000000 -0700
+++ pcsx2x6-v0.2.22-batocera/pcsx2/BuildVersion.cpp	2026-09-10 00:00:00.000000000 -0700
@@ -5,12 +5,12 @@

 namespace BuildVersion
 {
-	const char* GitTag = GIT_TAG;
-	bool GitTaggedCommit = GIT_TAGGED_COMMIT;
-	int GitTagHi = GIT_TAG_HI;
-	int GitTagMid = GIT_TAG_MID;
-	int GitTagLo = GIT_TAG_LO;
-	const char* GitRev = GIT_REV;
-	const char* GitHash = GIT_HASH;
+	const char* GitTag = "v0.2.22";
+	bool GitTaggedCommit = 1;
+	int GitTagHi = 0;
+	int GitTagMid = 2;
+	int GitTagLo = 22;
+	const char* GitRev = "v0.2.22";
+	const char* GitHash = "64f063a6351549b0de9c97971636b23360dbcdab";
 	const char* GitDate = GIT_DATE;
 } // namespace BuildVersion
