diff --git a/SDLPoP.ini b/SDLPoP.ini
index 73c44e0b37..c361e7ea35 100644
--- a/SDLPoP.ini
+++ b/SDLPoP.ini
@@ -38,7 +38,7 @@ enable_text = true
 enable_info_screen = true;
 
 ; Start the game in fullscreen mode. (In-game fullscreen toggle: Alt+Enter)
-start_fullscreen = false
+start_fullscreen = true
 
 ; Width and height of the game window.
 ; By default, the window is 640 pixels wide and 400 pixels tall.
