diff --git a/src/backends/unix/main.c b/src/backends/unix/main.c
index 97b74a8a..392bd082 100644
--- a/src/backends/unix/main.c
+++ b/src/backends/unix/main.c
@@ -110,30 +110,6 @@ main(int argc, char **argv)
 		}
 	}
 
-#ifndef __HAIKU__
-	/* Prevent running Quake II as root. Only very mad
-	   minded or stupid people even think about it. :) */
-	if (getuid() == 0)
-	{
-		printf("Quake II shouldn't be run as root! Backing out to save your ass. If\n");
-		printf("you really know what you're doing, edit src/unix/main.c and remove\n");
-		printf("this check. But don't complain if Quake II eats your dog afterwards!\n");
-
-		return 1;
-	}
-#endif
-
-	// Enforce the real UID to prevent setuid crap
-	if (getuid() != geteuid())
-	{
-		printf("The effective UID is not the real UID! Your binary is probably marked\n");
-		printf("'setuid'. That is not good idea, please fix it :) If you really know\n");
-		printf("what you're doing edit src/unix/main.c and remove this check. Don't\n");
-		printf("complain if Quake II eats your dog afterwards!\n");
-
-		return 1;
-	}
-
 	// enforce C locale
 	setenv("LC_ALL", "C", 1);
 
