diff --git a/rocknix-joypad.c b/rocknix-joypad.c
index f500cda..9b4914e 100644
--- a/rocknix-joypad.c
+++ b/rocknix-joypad.c
@@ -11,11 +11,8 @@
 #include <linux/platform_device.h>
 #include <linux/iio/consumer.h>
 #include <linux/version.h>
-#if (LINUX_VERSION_CODE < KERNEL_VERSION(6, 3, 0))
 #include <linux/of_gpio.h>
-#else
-#include <linux/of_gpio_legacy.h>
-#endif
+#include <linux/of.h>
 #include "rocknix-joypad.h"
 
 /*----------------------------------------------------------------------------*/
diff --git a/rocknix-singleadc-joypad.c b/rocknix-singleadc-joypad.c
index 083ba90..28ff130 100644
--- a/rocknix-singleadc-joypad.c
+++ b/rocknix-singleadc-joypad.c
@@ -11,11 +11,8 @@
 #include <linux/platform_device.h>
 #include <linux/iio/consumer.h>
 #include <linux/version.h>
-#if (LINUX_VERSION_CODE < KERNEL_VERSION(6, 3, 0))
 #include <linux/of_gpio.h>
-#else
-#include <linux/of_gpio_legacy.h>
-#endif
+#include <linux/of.h>
 #include <linux/delay.h>
 #include <linux/pwm.h>
 #include "rocknix-joypad.h"
diff --git a/rocknix-singleadc-joypad.c b/rocknix-singleadc-joypad.c
index ffb0361..14c35f5 100644
--- a/rocknix-singleadc-joypad.c
+++ b/rocknix-singleadc-joypad.c
@@ -231,7 +231,8 @@ struct joypad {
 
 };
 
-extern struct input_dev * joypad_input_g;
+struct input_dev *joypad_input_g = NULL;
+EXPORT_SYMBOL_GPL(joypad_input_g);
 
 static int pwm_vibrator_start(struct joypad *joypad)
 {
