From f00ef5da61ae0d38d575856cb1352d7f57ff7cf0 Mon Sep 17 00:00:00 2001
From: Max Kellermann <max@duempel.org>
Date: Mon, 24 Apr 2017 14:38:05 +0200
Subject: [PATCH 05/62] JsonSchemaBuilder/Makefile: remove unused variable
 APPBIN

Signed-off-by: memeka <mihailescu2m@gmail.com>
---
 tools/depends/native/JsonSchemaBuilder/Makefile | 1 -
 1 file changed, 1 deletion(-)

diff --git a/tools/depends/native/JsonSchemaBuilder/Makefile b/tools/depends/native/JsonSchemaBuilder/Makefile
index cf5d20b..77d7c3f 100644
--- a/tools/depends/native/JsonSchemaBuilder/Makefile
+++ b/tools/depends/native/JsonSchemaBuilder/Makefile
@@ -17,7 +17,6 @@ SOURCE=$(ROOT_DIR)/src
 
 CONFIGURE=./configure --prefix=$(PREFIX)
 APP=$(SOURCE)/JsonSchemaBuilder
-APPBIN=$(PREFIX)/bin/JsonSchemaBuilder
 
 all: .installed-$(PLATFORM)
 
-- 
2.7.4

