diff --git a/Makefile b/Makefile
index 4872239e..8468d9fa 100644
--- a/Makefile
+++ b/Makefile
@@ -69,6 +69,12 @@ else ifeq ($(PLATFORM),orangepi-pc)
         MORE_CFLAGS += -fomit-frame-pointer
     endif
 
+# Amlogic S812 (SDL2)
+else ifeq ($(PLATFORM),s812)
+    CPUFLAGS = -mcpu=cortex-a9 -mfpu=neon-vfpv3 -mfloat-abi=hard
+    CPPFLAGS += $(CPPFLAGS32) $(NEON_FLAGS) -DUSE_RENDER_THREAD
+    HAVE_NEON = 1
+
 # Odroid XU4 (SDL2)
 else ifeq ($(PLATFORM),xu4)
     CPUFLAGS += -mcpu=cortex-a15 -mfpu=neon-vfpv4
