--- a/src/plugins/multimedia/ffmpeg/qv4l2camera.cpp	2023-09-18 19:43:44.156079103 +0200
+++ b/src/plugins/multimedia/ffmpeg/qv4l2camera.cpp	2023-09-18 19:43:56.572342172 +0200
@@ -22,6 +22,9 @@
 
 #include <linux/videodev2.h>
 
+#define V4L2_PIX_FMT_BGRA32  v4l2_fourcc('R', 'A', '2', '4') /* 32  ABGR-8-8-8-8  */
+#define V4L2_PIX_FMT_RGBA32  v4l2_fourcc('A', 'B', '2', '4') /* 32  RGBA-8-8-8-8  */
+
 #include <qloggingcategory.h>
 
 QT_BEGIN_NAMESPACE
