From 42fb1d5ac91e2fa0efd29ae68334866577fac89f Mon Sep 17 00:00:00 2001
From: jackun <jack.un@gmail.com>
Date: Tue, 7 Jun 2022 21:59:12 +0300
Subject: [PATCH 09/14] desc pool flag

---
 src/vulkan.cpp | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/vulkan.cpp b/src/vulkan.cpp
index 98ac1b4..d656fc7 100644
--- a/src/vulkan.cpp
+++ b/src/vulkan.cpp
@@ -362,6 +362,7 @@ static void setup_device_pipeline(struct device_data *device_data)
    desc_pool_info.maxSets = 3;
    desc_pool_info.poolSizeCount = 1;
    desc_pool_info.pPoolSizes = &sampler_pool_size;
+   desc_pool_info.flags = VK_DESCRIPTOR_POOL_CREATE_FREE_DESCRIPTOR_SET_BIT;
    VK_CHECK(device_data->vtable.CreateDescriptorPool(device_data->device,
                                                      &desc_pool_info,
                                                      NULL, &device_data->descriptor_pool));
-- 
2.34.1

