diff --git a/src/meson.build b/src/meson.build
index 03319831c8..3af714a6bb 100644
--- a/src/meson.build
+++ b/src/meson.build
@@ -146,12 +146,12 @@ if is_unixy
       'loaders/loader_x11.cpp',
       'shared_x11.cpp',
     )
-  endif
 
     opengl_files += files(
       'loaders/loader_glx.cpp',
       'gl/inject_glx.cpp',
     )
+  endif
 
   if get_option('with_wayland').enabled()
     pre_args += '-DHAVE_WAYLAND'
@@ -175,6 +175,7 @@ link_args = cc.get_supported_link_arguments(['-Wl,-Bsymbolic-functions', '-Wl,-z
 # meson fails to check version-script so just force add
 link_args += '-Wl,--version-script,@0@'.format(join_paths(meson.current_source_dir(), 'mangohud.version'))
 
+mangohud_static_lib = []
 if get_option('use_vulkan')
 mangohud_static_lib = static_library(
   'MangoHud',
@@ -217,8 +218,8 @@ endif
 
 mangohud_shared_lib = shared_library(
   'MangoHud',
-  objects: mangohud_static_lib.extract_all_objects(),
-  link_with: mangohud_static_lib,
+  objects: get_option('use_vulkan') ? mangohud_static_lib.extract_all_objects() : [],
+  link_with: get_option('use_vulkan') ? mangohud_static_lib : [],
   link_args : link_args,
   install_dir : libdir_mangohud,
   install: true
diff --git a/src/hud_elements.h b/src/hud_elements.h
index 5c22fe73b2..fb4c3ecb81 100644
--- a/src/hud_elements.h
+++ b/src/hud_elements.h
@@ -6,7 +6,9 @@
 #include "timing.hpp"
 #include <functional>
 #include "winesync.h"
+#ifdef HAVE_VULKAN
 #include "vulkan/vulkan.h"
+#endif
 #include <array>
 #include "net.h"
 #include "overlay_params.h"
@@ -158,6 +160,7 @@ class HudElements{
 
         void TextColored(ImVec4 col, const char *fmt, ...);
 
+#ifdef HAVE_VULKAN
         std::array<VkPresentModeKHR, 6> presentModes = {
             VK_PRESENT_MODE_FIFO_RELAXED_KHR,
             VK_PRESENT_MODE_IMMEDIATE_KHR,
@@ -176,15 +179,17 @@ class HudElements{
         };
 
         VkPresentModeKHR cur_present_mode;
+#endif
 
         std::string get_present_mode(){
+#ifdef HAVE_VULKAN
             if (is_vulkan)
                 return presentModeMap[cur_present_mode];
+#endif
             // TODO: the opengl side is probably not as solid.
             // But it also might not be possible to figure out if vsync
             // is on or off unless we specify it.
-            else
-                return params->gl_vsync == 0 ? "OFF" : "ON";
+            return params->gl_vsync == 0 ? "OFF" : "ON";
 
         }
 };
diff --git a/src/gl/gl_hud.cpp b/src/gl/gl_hud.cpp
index 83b962d6fc..05486c57b0 100644
--- a/src/gl/gl_hud.cpp
+++ b/src/gl/gl_hud.cpp
@@ -18,11 +18,12 @@
 
 #include <glad/glad.h>
 
-
+#ifdef HAVE_X11
 #define GLX_RENDERER_VENDOR_ID_MESA                      0x8183
 #define GLX_RENDERER_DEVICE_ID_MESA                      0x8184
 
 bool glx_mesa_queryInteger(int attrib, unsigned int *value);
+#endif
 
 namespace MangoHud { namespace GL {
 
@@ -148,8 +149,10 @@ void imgui_create(void *ctx, const gl_wsi plat)
     HUDElements.vendorID = vendorID;
 
     uint32_t device_id = 0;
+#ifdef HAVE_X11
     if (plat == gl_wsi::GL_WSI_GLX)
         glx_mesa_queryInteger(GLX_RENDERER_DEVICE_ID_MESA, &device_id);
+#endif
 
     SPDLOG_DEBUG("GL device id: {:04X}", device_id);
     init_gpu_stats(vendorID, device_id, params);
