From 0252f694237350292ffd6f19e982f657faaea2d4 Mon Sep 17 00:00:00 2001
From: FantasyGmm <16450052+FantasyGmm@users.noreply.github.com>
Date: Wed, 5 Feb 2025 03:46:42 +0000
Subject: [PATCH] Add Qcom Bat Support

---
 src/battery.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/battery.cpp b/src/battery.cpp
index e5d311b..d82f3ba 100644
--- a/src/battery.cpp
+++ b/src/battery.cpp
@@ -13,7 +13,7 @@ void BatteryStats::numBattery() {
     fs::path path("/sys/class/power_supply/");
     for (auto& p : fs::directory_iterator(path)) {
         string fileName = p.path().filename();
-        if (fileName.find("BAT") != std::string::npos) {
+	if (fileName.find("BAT") != std::string::npos || fileName.find("qcom-battmgr-bat") != std::string::npos) {
             battPath[batteryCount] = p.path();
             batteryCount += 1;
         }
-- 
2.34.1

