diff --git a/gensignkey.c b/gensignkey.c
index 4be3d7d..cbe3211 100644
--- a/gensignkey.c
+++ b/gensignkey.c
@@ -99,7 +99,7 @@ int signkey_generate(enum signkey_type keytype, int bits, const char* filename,
 {
 	sign_key * key = NULL;
 	buffer *buf = NULL;
-	char *fn_temp = NULL;
+	//char *fn_temp = NULL;
 	int ret = DROPBEAR_FAILURE;
 	bits = signkey_generate_get_bits(keytype, bits);
 
@@ -149,31 +149,31 @@ int signkey_generate(enum signkey_type keytype, int bits, const char* filename,
 	key = NULL;
 	buf_setpos(buf, 0);
 
-	fn_temp = m_malloc(strlen(filename) + 30);
-	snprintf(fn_temp, strlen(filename)+30, "%s.tmp%d", filename, getpid());
-	ret = buf_writefile(buf, fn_temp, 0);
-
-	if (ret == DROPBEAR_FAILURE) {
-		goto out;
-	}
-
-	if (link(fn_temp, filename) < 0) {
-		/* If generating keys on connection (skipexist) it's OK to get EEXIST 
-		- we probably just lost a race with another connection to generate the key */
-		if (!(skip_exist && errno == EEXIST)) {
-			if (errno == EPERM || errno == EACCES) {
-				/* Non-atomic fallback when hard-links not allowed or unsupported */
-				buf_setpos(buf, 0);
-				ret = buf_writefile(buf, filename, skip_exist);
-			} else {
-				dropbear_log(LOG_ERR, "Failed moving key file to %s: %s", filename,
-					strerror(errno));
-				ret = DROPBEAR_FAILURE;
-			}
-
-			goto out;
-		}
-	}
+	//fn_temp = m_malloc(strlen(filename) + 30);
+	//snprintf(fn_temp, strlen(filename)+30, "%s.tmp%d", filename, getpid());
+	ret = buf_writefile(buf, filename, 0);
+
+	//if (ret == DROPBEAR_FAILURE) {
+	//	goto out;
+	//}
+
+	//if (link(fn_temp, filename) < 0) {
+	//	/* If generating keys on connection (skipexist) it's OK to get EEXIST 
+	//	- we probably just lost a race with another connection to generate the key */
+	//	if (!(skip_exist && errno == EEXIST)) {
+	//		if (errno == EPERM || errno == EACCES) {
+	//			/* Non-atomic fallback when hard-links not allowed or unsupported */
+	//			buf_setpos(buf, 0);
+	//			ret = buf_writefile(buf, filename, skip_exist);
+	//		} else {
+	//			dropbear_log(LOG_ERR, "Failed moving key file to %s: %s", filename,
+	//				strerror(errno));
+	//			ret = DROPBEAR_FAILURE;
+	//		}
+	//
+	//		goto out;
+	//	}
+	//}
 
 	/* ensure directory update is flushed to disk, otherwise we can end up
 	with zero-byte hostkey files if the power goes off */
@@ -185,10 +185,10 @@ out:
 		buf_free(buf);
 	}
 	
-	if (fn_temp) {
-		unlink(fn_temp);
-		m_free(fn_temp);
-	}
+	//if (fn_temp) {
+	//	unlink(fn_temp);
+	//	m_free(fn_temp);
+	//}
 
 	return ret;
 }
