--- a/audio/drivers/alsa.c	2021-05-20 13:27:55.422221057 +0200
+++ b/audio/drivers/alsa.c	2021-05-20 13:29:23.728884484 +0200
@@ -37,20 +37,13 @@
 
 static bool alsa_use_float(void *data)
 {
-   alsa_t *alsa = (alsa_t*)data;
-   return alsa->has_float;
+   return false;
 }
 
 static bool find_float_format(snd_pcm_t *pcm, void *data)
 {
    snd_pcm_hw_params_t *params = (snd_pcm_hw_params_t*)data;
 
-   if (snd_pcm_hw_params_test_format(pcm, params, SND_PCM_FORMAT_FLOAT) == 0)
-   {
-      RARCH_LOG("[ALSA]: Using floating point format.\n");
-      return true;
-   }
-
    RARCH_LOG("[ALSA]: Using signed 16-bit format.\n");
    return false;
 }
