diff --git a/cmake/BuildParameters.cmake b/cmake/BuildParameters.cmake
index fe54c0aaa8..6af76006ca 100644
--- a/cmake/BuildParameters.cmake
+++ b/cmake/BuildParameters.cmake
@@ -315,7 +315,7 @@ set(COMMON_FLAG "-pipe -fvisibility=hidden -pthread -fno-builtin-strcmp -fno-bui
 if (CMAKE_BUILD_TYPE MATCHES "Debug")
 set(HARDENING_FLAG "-Wformat -Wformat-security")
 else()
-set(HARDENING_FLAG "-D_FORTIFY_SOURCE=2  -Wformat -Wformat-security")
+set(HARDENING_FLAG "-Wformat -Wformat-security")
 endif()
 
 # -Wno-attributes: "always_inline function might not be inlinable" <= real spam (thousand of warnings!!!)
