diff --git a/pcsx2/CMakeLists.txt b/pcsx2/CMakeLists.txt
index c228c94e95..92cf4014ae 100644
--- a/pcsx2/CMakeLists.txt
+++ b/pcsx2/CMakeLists.txt
@@ -347,28 +347,24 @@ if (XXD)
 	add_custom_command(
 		OUTPUT ${db_res_bin}/cheats_ws.h
 		COMMAND ${XXD} -i cheats_ws.zip cheats_ws.h
-		COMMAND cp cheats_ws.h ${db_res_bin}/cheats_ws.h
 		WORKING_DIRECTORY ${db_res_src}
 		VERBATIM
 	)
 	add_custom_command(
 		OUTPUT ${db_res_bin}/cheats_nointerlacing.h
 		COMMAND ${XXD} -i cheats_nointerlacing.zip cheats_nointerlacing.h
-		COMMAND cp cheats_nointerlacing.h ${db_res_bin}/cheats_nointerlacing.h
 		WORKING_DIRECTORY ${db_res_src}
 		VERBATIM
 	)
 	add_custom_command(
 		OUTPUT ${db_res_bin}/cheats_60fps.h
 		COMMAND ${XXD} -i cheats_60fps.zip cheats_60fps.h
-		COMMAND cp cheats_60fps.h ${db_res_bin}/cheats_60fps.h
 		WORKING_DIRECTORY ${db_res_src}
 		VERBATIM
 	)
 	add_custom_command(
 		OUTPUT  ${db_res_bin}/GameIndex.h
 		COMMAND ${XXD} -i GameIndex.yaml GameIndex.h
-		COMMAND cp GameIndex.h  ${db_res_bin}/GameIndex.h
 		WORKING_DIRECTORY  ${db_res_src}
 		VERBATIM
 	)
