From a088af6381e8c3afe3411b46c45bfd38118e9180 Mon Sep 17 00:00:00 2001
From: Max Kellermann <max@duempel.org>
Date: Mon, 24 Apr 2017 14:38:07 +0200
Subject: [PATCH 33/62] input/XKBCommonKeymap: include cleanup

Signed-off-by: memeka <mihailescu2m@gmail.com>
---
 xbmc/input/linux/XKBCommonKeymap.h | 1 -
 1 file changed, 1 deletion(-)

diff --git a/xbmc/input/linux/XKBCommonKeymap.h b/xbmc/input/linux/XKBCommonKeymap.h
index fbd6e33..d5fff46 100644
--- a/xbmc/input/linux/XKBCommonKeymap.h
+++ b/xbmc/input/linux/XKBCommonKeymap.h
@@ -19,7 +19,6 @@
  *  <http://www.gnu.org/licenses/>.
  *
  */
-#include <boost/noncopyable.hpp>
 
 #include "input/linux/Keymap.h"
 
-- 
2.7.4

