diff --git a/shell/linux/Makefile b/shell/linux/Makefile
index 2898360..fe47c48 100644
--- a/shell/linux/Makefile
+++ b/shell/linux/Makefile
@@ -98,7 +98,7 @@ ifneq (,$(findstring x86,$(platform)))
     NOT_ARM := 1
     USE_X11 := 1
     MFLAGS += -m32
-    ASFLAGS += --32
+    ASFLAGS += -c
     LDFLAGS += -m32
     CFLAGS += -m32 -D TARGET_LINUX_x86 -D TARGET_NO_AREC -fsingle-precision-constant
     CXXFLAGS += -fno-exceptions
@@ -155,10 +155,11 @@ else ifneq (,$(findstring rpi,$(platform)))
         INCS += -I/opt/vc/include/ -I/opt/vc/include/interface/vmcs_host/linux -I/opt/vc/include/interface/vcos/pthreads
         LIBS += -L/opt/vc/lib/ -lbcm_host -ldl
         LIBS += -lbrcmEGL -lbrcmGLESv2
+        INCS +=  -I../linux-deps/include
         CFLAGS += -D TARGET_VIDEOCORE
         CXXFLAGS += -DGLES
         USE_OMX := 1
-        USE_DISPMANX := 1
+        USE_GLES := 1
     endif
     undefine USE_X11
 
@@ -186,6 +187,7 @@ else ifneq (,$(findstring odroid,$(platform)))
     MFLAGS += -marm -mfpu=neon -mfloat-abi=hard -funroll-loops
     ASFLAGS += -mfpu=neon -mfloat-abi=hard
     CFLAGS += -D TARGET_BEAGLE -D TARGET_LINUX_ARMELv7 -DARM_HARDFP -fsingle-precision-constant
+    INCS +=  -I../linux-deps/include
     USE_GLES := 1
 
     # ODROID-XU3, -XU3 Lite & -XU4
@@ -193,6 +195,18 @@ else ifneq (,$(findstring odroid,$(platform)))
         MFLAGS += -march=armv7ve -mtune=cortex-a15.cortex-a7
         ASFLAGS += -march=armv7ve
 
+    # ODROIDN2
+    else ifneq (,$(findstring odroidn2,$(platform)))
+        MFLAGS += -march=armv8-a+crc -mtune=cortex-a53
+        ASFLAGS += -march=armv8-a+crc -mfpu=neon -mfloat-abi=hard
+        CFLAGS += -DGLES3 -DPNG_ARM_NEON=1 -DPNG_ARM_NEON_OPT=1
+    
+    # ODROIDGOA
+    else ifneq (,$(findstring odroidgoa,$(platform)))
+        MFLAGS += -march=armv8-a+crc -mtune=cortex-a35
+        ASFLAGS += -march=armv8-a+crc -mfpu=neon -mfloat-abi=hard
+        CFLAGS += -DGLES3 -DPNG_ARM_NEON=1 -DPNG_ARM_NEON_OPT=1
+
     # Other ODROIDs
     else
         MFLAGS += -march=armv7-a
@@ -209,6 +223,14 @@ else ifneq (,$(findstring odroid,$(platform)))
         endif
     endif
 
+# ROCKPRO64
+else ifneq (,$(findstring rockpro64,$(platform)))
+    MFLAGS += -marm -march=armv8-a+crc -mfpu=neon -mfloat-abi=hard -funroll-loops -mtune=cortex-a53
+    ASFLAGS += -march=armv8-a+crc -mfpu=neon -mfloat-abi=hard
+    CFLAGS += -D TARGET_LINUX_ARMELv7 -DARM_HARDFP -fsingle-precision-constant -DPNG_ARM_NEON=1 -DPNG_ARM_NEON_OPT=1
+    INCS +=  -I../linux-deps/include
+    USE_GLES := 1
+
 # GCW Zero
 else ifneq (,$(findstring gcwz,$(platform)))
     NOT_ARM := 1
