--- a/libswirl/libswirl.cpp	2020-08-13 19:05:57.490182180 +0200
+++ b/libswirl/libswirl.cpp	2020-08-13 19:06:41.506990719 +0200
@@ -840,10 +840,10 @@
     {
         cfgSetVirtual("config", "image", path.c_str());
 
-        if (settings.bios.UseReios || !LoadRomFiles(get_readonly_data_path(DATA_PATH)))
+        if (settings.bios.UseReios || !LoadRomFiles("/userdata/bios/"))
         {
 #ifdef USE_REIOS
-            if (!LoadHle(get_readonly_data_path(DATA_PATH)))
+            if (!LoadHle("/userdata/bios/"))
             {
                 return -5;
             }
