diff --git a/libswirl/libswirl.cpp b/libswirl/libswirl.cpp
index 3e9169f..675f3d6 100644
--- a/libswirl/libswirl.cpp
+++ b/libswirl/libswirl.cpp
@@ -318,7 +318,7 @@ int dc_start_game(const char *path)
 #if DC_PLATFORM == DC_PLATFORM_DREAMCAST
 		if (!settings.bios.UseReios)
 #endif
-			if (!LoadRomFiles(get_readonly_data_path(DATA_PATH)))
+			if (!LoadRomFiles("/userdata/bios/"))
 				return -5;
 
 #if DC_PLATFORM == DC_PLATFORM_DREAMCAST
@@ -352,10 +352,10 @@ int dc_start_game(const char *path)
 		return 0;
 	}
 
-	if (settings.bios.UseReios || !LoadRomFiles(get_readonly_data_path(DATA_PATH)))
+	if (settings.bios.UseReios || !LoadRomFiles("/userdata/bios/"))
 	{
 #ifdef USE_REIOS
-		if (!LoadHle(get_readonly_data_path(DATA_PATH)))
+		if (!LoadHle("/userdata/bios/"))
 		{
 			return -5;
 		}
