diff --git a/core/hw/sh4/sh4_mem.cpp b/core/hw/sh4/sh4_mem.cpp
index f49d41f..e139d0a 100644
--- a/core/hw/sh4/sh4_mem.cpp
+++ b/core/hw/sh4/sh4_mem.cpp
@@ -223,7 +223,7 @@ void mem_Term()
 	sh4_area0_Term();
 
 	//write back Flash/SRAM
-	SaveRomFiles(get_writable_data_path("/data/"));
+	SaveRomFiles("/userdata/bios/");
 	
 	//mem_b.Term(); // handled by vmem
 
diff --git a/core/nullDC.cpp b/core/nullDC.cpp
index 27cf443..2be2095 100755
--- a/core/nullDC.cpp
+++ b/core/nullDC.cpp
@@ -182,14 +182,14 @@ int dc_init(int argc,wchar* argv[])
 	int rv = 0;
 
 #if HOST_OS != OS_DARWIN
-    #define DATA_PATH "/data/"
+    #define DATA_PATH "/"
 #else
     #define DATA_PATH "/"
 #endif
 
-	if (settings.bios.UseReios || !LoadRomFiles(get_readonly_data_path(DATA_PATH)))
+	if (settings.bios.UseReios || !LoadRomFiles("/userdata/bios/"))
 	{
-		if (!LoadHle(get_readonly_data_path(DATA_PATH)))
+		if (!LoadHle("/userdata/bios/"))
 		{
 #if defined(_ANDROID)
 			reios_init_value = -4;
@@ -266,7 +266,7 @@ void dc_term()
 #ifndef _ANDROID
 	SaveSettings();
 #endif
-	SaveRomFiles(get_writable_data_path("/data/"));
+	SaveRomFiles("/userdata/bios/");
 
     TermAudio();
 }
@@ -274,7 +274,7 @@ void dc_term()
 #if defined(_ANDROID)
 void dc_pause()
 {
-	SaveRomFiles(get_writable_data_path("/data/"));
+	SaveRomFiles("/userdata/bios/");
 }
 #endif
 
