--- a/libswirl/core.mk	2020-08-22 12:23:31.860983061 +0200
+++ b/libswirl/core.mk	2020-08-22 12:23:57.241481897 +0200
@@ -114,7 +114,8 @@
 endif
 
 RZDCY_CFLAGS += -I$(RZDCY_SRC_DIR) -I$(RZDCY_SRC_DIR)/rend/gles -I$(RZDCY_SRC_DIR)/deps \
-		 -I$(RZDCY_SRC_DIR)/deps/vixl -I$(RZDCY_SRC_DIR)/khronos
+		 -I$(RZDCY_SRC_DIR)/deps/vixl
+#-I$(RZDCY_SRC_DIR)/khronos
 
 ifdef USE_MODEM
 	RZDCY_CFLAGS += -DENABLE_MODEM -I$(RZDCY_SRC_DIR)/gpl/deps/picotcp/include -I$(RZDCY_SRC_DIR)/gpl/deps/picotcp/modules
