diff --git a/Boss.cpp b/Boss.cpp
index 70952b2..14afee8 100644
--- a/Boss.cpp
+++ b/Boss.cpp
@@ -163,7 +163,7 @@ void Boss::Update()
 			bComingToLife = true;
 			if(!gpEngine->bDemo)
 			{
-				gpEngine->stmStage.Load("Data/Sound/Boss.ogg",true);
+				gpEngine->stmStage.Load("/usr/share/prototype/Data/Sound/Boss.ogg",true);
 				//gpEngine->stmStage.Seek(0,30);
 				gpEngine->stmStage.Play(30);
 			}
diff --git a/Demo.cpp b/Demo.cpp
index 9b26c70..c746c41 100644
--- a/Demo.cpp
+++ b/Demo.cpp
@@ -3,7 +3,7 @@
 void SaveDemo(demoSystem dem)
 {
 	CFileIO fIO;
-	fIO.GetFile("Data/Demo1.dem","wb");
+	fIO.GetFile("/usr/share/prototype/Data/Demo1.dem","wb");
 	fIO.WriteBinary(&dem.iScrollOffset,1);
 	int bufsize = (int)dem.mKeyBuffer.size();
 	fIO.WriteBinary(&bufsize,1);
@@ -21,7 +21,7 @@ void SaveDemo(demoSystem dem)
 bool LoadDemo(demoSystem& dem, int demo)
 {
 	char name[256];
-	sprintf(name,"Data/Demo%i.dem",demo);
+	sprintf(name,"/usr/share/prototype/Data/Demo%i.dem",demo);
 	dem.mKeyBuffer.clear();
 	CFileIO fIO;
 	if(!fIO.GetFile(name,"rb"))
diff --git a/Engine.cpp b/Engine.cpp
index 41db8f6..73089a2 100755
--- a/Engine.cpp
+++ b/Engine.cpp
@@ -862,311 +862,311 @@ void Engine::InitializeRenderTargets()
 //================================================================================================//
 bool Engine::LoadConstantData()
 {
-	if(!mFont1.LoadFont("Data/Gfx/Fonts.png","Data/Gfx/FontShadow.png"))
+	if(!mFont1.LoadFont("/usr/share/prototype/Data/Gfx/Fonts.png","/usr/share/prototype/Data/Gfx/FontShadow.png"))
 		return false;
-	if(!mFont2.LoadFont("Data/Gfx/Fonts2.png","Data/Gfx/Fonts2Shadow.png"))
+	if(!mFont2.LoadFont("/usr/share/prototype/Data/Gfx/Fonts2.png","/usr/share/prototype/Data/Gfx/Fonts2Shadow.png"))
 		return false;
-	if(!(imgUI=UTIL_GL::Image::LoadImage("Data/Gfx/GUI.png",GL_LINEAR,GL_LINEAR,GL_REPEAT,GL_REPEAT)))
+	if(!(imgUI=UTIL_GL::Image::LoadImage("/usr/share/prototype/Data/Gfx/GUI.png",GL_LINEAR,GL_LINEAR,GL_REPEAT,GL_REPEAT)))
 		return false;
-	if(!(imgScanlines=UTIL_GL::Image::LoadImage("Data/Gfx/Scanlines.png",GL_LINEAR,GL_LINEAR,GL_REPEAT,GL_REPEAT)))
+	if(!(imgScanlines=UTIL_GL::Image::LoadImage("/usr/share/prototype/Data/Gfx/Scanlines.png",GL_LINEAR,GL_LINEAR,GL_REPEAT,GL_REPEAT)))
 		return false;
-	if(!(imgHudBeam=UTIL_GL::Image::LoadImage("Data/Gfx/HudBeam.png",GL_NEAREST,GL_NEAREST,GL_CLAMP_TO_EDGE,GL_CLAMP_TO_EDGE)))
+	if(!(imgHudBeam=UTIL_GL::Image::LoadImage("/usr/share/prototype/Data/Gfx/HudBeam.png",GL_NEAREST,GL_NEAREST,GL_CLAMP_TO_EDGE,GL_CLAMP_TO_EDGE)))
 		return false;
 
 	unsigned int ID=0;
-	if(!(ID=UTIL_GL::Image::LoadImage("Data/Gfx/PlayerShip.png",GL_LINEAR,GL_LINEAR,GL_CLAMP_TO_EDGE,GL_CLAMP_TO_EDGE)))
+	if(!(ID=UTIL_GL::Image::LoadImage("/usr/share/prototype/Data/Gfx/PlayerShip.png",GL_LINEAR,GL_LINEAR,GL_CLAMP_TO_EDGE,GL_CLAMP_TO_EDGE)))
 		return false;
 	sprPlr.Split(ID,512,512,128,128);
 
-	if(!(ID=UTIL_GL::Image::LoadImage("Data/Gfx/ForceCharge.png",GL_LINEAR,GL_LINEAR,GL_CLAMP_TO_EDGE,GL_CLAMP_TO_EDGE)))
+	if(!(ID=UTIL_GL::Image::LoadImage("/usr/share/prototype/Data/Gfx/ForceCharge.png",GL_LINEAR,GL_LINEAR,GL_CLAMP_TO_EDGE,GL_CLAMP_TO_EDGE)))
 		return false;
 	sprForceCharge.Split(ID,512,256,64,64);
-	if(!(ID=UTIL_GL::Image::LoadImage("Data/Gfx/Orb1.png",GL_LINEAR,GL_LINEAR,GL_CLAMP_TO_EDGE,GL_CLAMP_TO_EDGE)))
+	if(!(ID=UTIL_GL::Image::LoadImage("/usr/share/prototype/Data/Gfx/Orb1.png",GL_LINEAR,GL_LINEAR,GL_CLAMP_TO_EDGE,GL_CLAMP_TO_EDGE)))
 		return false;
 	sprOrb1.Split(ID,512,512,64,64);
 
-	if(!(imgChain=UTIL_GL::Image::LoadImage("Data/Gfx/ChainParticle.png",GL_LINEAR,GL_LINEAR,GL_REPEAT,GL_REPEAT)))
+	if(!(imgChain=UTIL_GL::Image::LoadImage("/usr/share/prototype/Data/Gfx/ChainParticle.png",GL_LINEAR,GL_LINEAR,GL_REPEAT,GL_REPEAT)))
 		return false;
 
 	//entity sprites - if things get too much overhead, we may have to only load those that are neccesary
-	if(!(ID=UTIL_GL::Image::LoadImage("Data/Gfx/LadyBird.png",GL_LINEAR,GL_LINEAR,GL_CLAMP_TO_EDGE,GL_CLAMP_TO_EDGE)))
+	if(!(ID=UTIL_GL::Image::LoadImage("/usr/share/prototype/Data/Gfx/LadyBird.png",GL_LINEAR,GL_LINEAR,GL_CLAMP_TO_EDGE,GL_CLAMP_TO_EDGE)))
 		return false;
 	sprLadyBird.Split(ID,256,256,64,64);
-	if(!(ID=UTIL_GL::Image::LoadImage("Data/Gfx/turret1.png",GL_LINEAR,GL_LINEAR,GL_CLAMP_TO_EDGE,GL_CLAMP_TO_EDGE)))
+	if(!(ID=UTIL_GL::Image::LoadImage("/usr/share/prototype/Data/Gfx/turret1.png",GL_LINEAR,GL_LINEAR,GL_CLAMP_TO_EDGE,GL_CLAMP_TO_EDGE)))
 		return false;
 	sprTurret.Split(ID,256,256,64,64);
-	if(!(ID=UTIL_GL::Image::LoadImage("Data/Gfx/SuperTurret.png",GL_LINEAR,GL_LINEAR,GL_CLAMP_TO_EDGE,GL_CLAMP_TO_EDGE)))
+	if(!(ID=UTIL_GL::Image::LoadImage("/usr/share/prototype/Data/Gfx/SuperTurret.png",GL_LINEAR,GL_LINEAR,GL_CLAMP_TO_EDGE,GL_CLAMP_TO_EDGE)))
 		return false;
 	sprSuperTurret.Split(ID,256,128,64,32);
-	if(!(ID=UTIL_GL::Image::LoadImage("Data/Gfx/SerpentHead.png",GL_LINEAR,GL_LINEAR,GL_CLAMP_TO_EDGE,GL_CLAMP_TO_EDGE)))
+	if(!(ID=UTIL_GL::Image::LoadImage("/usr/share/prototype/Data/Gfx/SerpentHead.png",GL_LINEAR,GL_LINEAR,GL_CLAMP_TO_EDGE,GL_CLAMP_TO_EDGE)))
 		return false;
 	sprSerpent.Split(ID,256,256,64,64);
-	if(!(ID=UTIL_GL::Image::LoadImage("Data/Gfx/SerpentBody.png",GL_LINEAR,GL_LINEAR,GL_CLAMP_TO_EDGE,GL_CLAMP_TO_EDGE)))
+	if(!(ID=UTIL_GL::Image::LoadImage("/usr/share/prototype/Data/Gfx/SerpentBody.png",GL_LINEAR,GL_LINEAR,GL_CLAMP_TO_EDGE,GL_CLAMP_TO_EDGE)))
 		return false;
 	sprSerpentBody.Split(ID,256,256,64,64);
-	if(!(ID=UTIL_GL::Image::LoadImage("Data/Gfx/SerpentTail.png",GL_LINEAR,GL_LINEAR,GL_CLAMP_TO_EDGE,GL_CLAMP_TO_EDGE)))
+	if(!(ID=UTIL_GL::Image::LoadImage("/usr/share/prototype/Data/Gfx/SerpentTail.png",GL_LINEAR,GL_LINEAR,GL_CLAMP_TO_EDGE,GL_CLAMP_TO_EDGE)))
 		return false;
 	sprSerpentTail.Split(ID,256,256,64,64);
-	if(!(ID=UTIL_GL::Image::LoadImage("Data/Gfx/Rocket.png",GL_LINEAR,GL_LINEAR,GL_CLAMP_TO_EDGE,GL_CLAMP_TO_EDGE)))
+	if(!(ID=UTIL_GL::Image::LoadImage("/usr/share/prototype/Data/Gfx/Rocket.png",GL_LINEAR,GL_LINEAR,GL_CLAMP_TO_EDGE,GL_CLAMP_TO_EDGE)))
 		return false;
 	sprRocket.Split(ID,256,128,64,64);
-	if(!(ID=UTIL_GL::Image::LoadImage("Data/Gfx/GroundTroop_Walk.png",GL_LINEAR,GL_LINEAR,GL_CLAMP_TO_EDGE,GL_CLAMP_TO_EDGE)))
+	if(!(ID=UTIL_GL::Image::LoadImage("/usr/share/prototype/Data/Gfx/GroundTroop_Walk.png",GL_LINEAR,GL_LINEAR,GL_CLAMP_TO_EDGE,GL_CLAMP_TO_EDGE)))
 		return false;
 	sprRhinoWalk.Split(ID,256,256,64,64);
-	if(!(ID=UTIL_GL::Image::LoadImage("Data/Gfx/GroundTroop_Turn.png",GL_LINEAR,GL_LINEAR,GL_CLAMP_TO_EDGE,GL_CLAMP_TO_EDGE)))
+	if(!(ID=UTIL_GL::Image::LoadImage("/usr/share/prototype/Data/Gfx/GroundTroop_Turn.png",GL_LINEAR,GL_LINEAR,GL_CLAMP_TO_EDGE,GL_CLAMP_TO_EDGE)))
 		return false;
 	sprRhinoTurn.Split(ID,256,256,64,64);
-	if(!(ID=UTIL_GL::Image::LoadImage("Data/Gfx/GroundTroop_Aim.png",GL_LINEAR,GL_LINEAR,GL_CLAMP_TO_EDGE,GL_CLAMP_TO_EDGE)))
+	if(!(ID=UTIL_GL::Image::LoadImage("/usr/share/prototype/Data/Gfx/GroundTroop_Aim.png",GL_LINEAR,GL_LINEAR,GL_CLAMP_TO_EDGE,GL_CLAMP_TO_EDGE)))
 		return false;
 	sprRhinoAim.Split(ID,256,256,64,64);
-	if(!(ID=UTIL_GL::Image::LoadImage("Data/Gfx/PowerDroid.png",GL_LINEAR,GL_LINEAR,GL_CLAMP_TO_EDGE,GL_CLAMP_TO_EDGE)))
+	if(!(ID=UTIL_GL::Image::LoadImage("/usr/share/prototype/Data/Gfx/PowerDroid.png",GL_LINEAR,GL_LINEAR,GL_CLAMP_TO_EDGE,GL_CLAMP_TO_EDGE)))
 		return false;
 	sprPowerDroid.Split(ID,512,512,64,64);
-	if(!(ID=UTIL_GL::Image::LoadImage("Data/Gfx/Bug.png",GL_LINEAR,GL_LINEAR,GL_CLAMP_TO_EDGE,GL_CLAMP_TO_EDGE)))
+	if(!(ID=UTIL_GL::Image::LoadImage("/usr/share/prototype/Data/Gfx/Bug.png",GL_LINEAR,GL_LINEAR,GL_CLAMP_TO_EDGE,GL_CLAMP_TO_EDGE)))
 		return false;
 	sprBug.Split(ID,256,256,64,64);
-	if(!(ID=UTIL_GL::Image::LoadImage("Data/Gfx/Starburst.png",GL_LINEAR,GL_LINEAR,GL_CLAMP_TO_EDGE,GL_CLAMP_TO_EDGE)))
+	if(!(ID=UTIL_GL::Image::LoadImage("/usr/share/prototype/Data/Gfx/Starburst.png",GL_LINEAR,GL_LINEAR,GL_CLAMP_TO_EDGE,GL_CLAMP_TO_EDGE)))
 		return false;
 	sprStarBurst.Split(ID,256,256,64,64);
-	if(!(ID=UTIL_GL::Image::LoadImage("Data/Gfx/Fighter.png",GL_LINEAR,GL_LINEAR,GL_CLAMP_TO_EDGE,GL_CLAMP_TO_EDGE)))
+	if(!(ID=UTIL_GL::Image::LoadImage("/usr/share/prototype/Data/Gfx/Fighter.png",GL_LINEAR,GL_LINEAR,GL_CLAMP_TO_EDGE,GL_CLAMP_TO_EDGE)))
 		return false;
 	sprFighter.Split(ID,512,256,64,64);
-	if(!(ID=UTIL_GL::Image::LoadImage("Data/Gfx/RoboticTorso.png",GL_LINEAR,GL_LINEAR,GL_CLAMP_TO_EDGE,GL_CLAMP_TO_EDGE)))
+	if(!(ID=UTIL_GL::Image::LoadImage("/usr/share/prototype/Data/Gfx/RoboticTorso.png",GL_LINEAR,GL_LINEAR,GL_CLAMP_TO_EDGE,GL_CLAMP_TO_EDGE)))
 		return false;
 	sprRobot.Split(ID,512,256,128,128);
-	if(!(ID=UTIL_GL::Image::LoadImage("Data/Gfx/Thruster.png",GL_LINEAR,GL_LINEAR,GL_CLAMP_TO_EDGE,GL_CLAMP_TO_EDGE)))
+	if(!(ID=UTIL_GL::Image::LoadImage("/usr/share/prototype/Data/Gfx/Thruster.png",GL_LINEAR,GL_LINEAR,GL_CLAMP_TO_EDGE,GL_CLAMP_TO_EDGE)))
 		return false;
 	sprThruster.Split(ID,256,128,64,64);
-	if(!(imgBoss=UTIL_GL::Image::LoadImage("Data/Gfx/Boss.png",GL_LINEAR,GL_LINEAR,GL_CLAMP_TO_EDGE,GL_CLAMP_TO_EDGE)))
+	if(!(imgBoss=UTIL_GL::Image::LoadImage("/usr/share/prototype/Data/Gfx/Boss.png",GL_LINEAR,GL_LINEAR,GL_CLAMP_TO_EDGE,GL_CLAMP_TO_EDGE)))
 		return false;
-	if(!(imgBossTube=UTIL_GL::Image::LoadImage("Data/Gfx/BossTube.png",GL_LINEAR,GL_LINEAR,GL_CLAMP_TO_EDGE,GL_CLAMP_TO_EDGE)))
+	if(!(imgBossTube=UTIL_GL::Image::LoadImage("/usr/share/prototype/Data/Gfx/BossTube.png",GL_LINEAR,GL_LINEAR,GL_CLAMP_TO_EDGE,GL_CLAMP_TO_EDGE)))
 		return false;
-	if(!(imgBossTubeSpec=UTIL_GL::Image::LoadImage("Data/Gfx/BossTubeSpec.png",GL_LINEAR,GL_LINEAR,GL_CLAMP_TO_EDGE,GL_CLAMP_TO_EDGE)))
+	if(!(imgBossTubeSpec=UTIL_GL::Image::LoadImage("/usr/share/prototype/Data/Gfx/BossTubeSpec.png",GL_LINEAR,GL_LINEAR,GL_CLAMP_TO_EDGE,GL_CLAMP_TO_EDGE)))
 		return false;
-	if(!(imgBossTubeFluid=UTIL_GL::Image::LoadImage("Data/Gfx/BossTubeFluid.png",GL_LINEAR,GL_LINEAR,GL_REPEAT,GL_REPEAT)))
+	if(!(imgBossTubeFluid=UTIL_GL::Image::LoadImage("/usr/share/prototype/Data/Gfx/BossTubeFluid.png",GL_LINEAR,GL_LINEAR,GL_REPEAT,GL_REPEAT)))
 		return false;
-	if(!(imgBossArm=UTIL_GL::Image::LoadImage("Data/Gfx/BossArm.png",GL_LINEAR,GL_LINEAR,GL_CLAMP_TO_EDGE,GL_CLAMP_TO_EDGE)))
+	if(!(imgBossArm=UTIL_GL::Image::LoadImage("/usr/share/prototype/Data/Gfx/BossArm.png",GL_LINEAR,GL_LINEAR,GL_CLAMP_TO_EDGE,GL_CLAMP_TO_EDGE)))
 		return false;
-	if(!(imgBossPush=UTIL_GL::Image::LoadImage("Data/Gfx/BossPush.png",GL_LINEAR,GL_LINEAR,GL_REPEAT,GL_REPEAT)))
+	if(!(imgBossPush=UTIL_GL::Image::LoadImage("/usr/share/prototype/Data/Gfx/BossPush.png",GL_LINEAR,GL_LINEAR,GL_REPEAT,GL_REPEAT)))
 		return false;
-	if(!(ID=UTIL_GL::Image::LoadImage("Data/Gfx/BossEye1.png",GL_LINEAR,GL_LINEAR,GL_CLAMP_TO_EDGE,GL_CLAMP_TO_EDGE)))
+	if(!(ID=UTIL_GL::Image::LoadImage("/usr/share/prototype/Data/Gfx/BossEye1.png",GL_LINEAR,GL_LINEAR,GL_CLAMP_TO_EDGE,GL_CLAMP_TO_EDGE)))
 		return false;
 	sprBossEye1.Split(ID,128,128,64,64);
-	if(!(ID=UTIL_GL::Image::LoadImage("Data/Gfx/BossEye2.png",GL_LINEAR,GL_LINEAR,GL_CLAMP_TO_EDGE,GL_CLAMP_TO_EDGE)))
+	if(!(ID=UTIL_GL::Image::LoadImage("/usr/share/prototype/Data/Gfx/BossEye2.png",GL_LINEAR,GL_LINEAR,GL_CLAMP_TO_EDGE,GL_CLAMP_TO_EDGE)))
 		return false;
 	sprBossEye2.Split(ID,256,128,64,64);
-	if(!(ID=UTIL_GL::Image::LoadImage("Data/Gfx/BossEye3.png",GL_LINEAR,GL_LINEAR,GL_CLAMP_TO_EDGE,GL_CLAMP_TO_EDGE)))
+	if(!(ID=UTIL_GL::Image::LoadImage("/usr/share/prototype/Data/Gfx/BossEye3.png",GL_LINEAR,GL_LINEAR,GL_CLAMP_TO_EDGE,GL_CLAMP_TO_EDGE)))
 		return false;
 	sprBossEye3.Split(ID,64,64,32,32);
-	if(!(ID=UTIL_GL::Image::LoadImage("Data/Gfx/BossEye4.png",GL_LINEAR,GL_LINEAR,GL_CLAMP_TO_EDGE,GL_CLAMP_TO_EDGE)))
+	if(!(ID=UTIL_GL::Image::LoadImage("/usr/share/prototype/Data/Gfx/BossEye4.png",GL_LINEAR,GL_LINEAR,GL_CLAMP_TO_EDGE,GL_CLAMP_TO_EDGE)))
 		return false;
 	sprBossEye4.Split(ID,256,128,64,64);
-	if(!(ID=UTIL_GL::Image::LoadImage("Data/Gfx/BossEye5.png",GL_LINEAR,GL_LINEAR,GL_CLAMP_TO_EDGE,GL_CLAMP_TO_EDGE)))
+	if(!(ID=UTIL_GL::Image::LoadImage("/usr/share/prototype/Data/Gfx/BossEye5.png",GL_LINEAR,GL_LINEAR,GL_CLAMP_TO_EDGE,GL_CLAMP_TO_EDGE)))
 		return false;
 	sprBossEye5.Split(ID,128,64,32,32);
-	if(!(ID=UTIL_GL::Image::LoadImage("Data/Gfx/BossCharge.png",GL_LINEAR,GL_LINEAR,GL_CLAMP_TO_EDGE,GL_CLAMP_TO_EDGE)))
+	if(!(ID=UTIL_GL::Image::LoadImage("/usr/share/prototype/Data/Gfx/BossCharge.png",GL_LINEAR,GL_LINEAR,GL_CLAMP_TO_EDGE,GL_CLAMP_TO_EDGE)))
 		return false;
 	sprBossCharge.Split(ID,512,256,64,64);
-	if(!(imgBossBrain=UTIL_GL::Image::LoadImage("Data/Gfx/BossBrain.png",GL_LINEAR,GL_LINEAR,GL_CLAMP_TO_EDGE,GL_CLAMP_TO_EDGE)))
+	if(!(imgBossBrain=UTIL_GL::Image::LoadImage("/usr/share/prototype/Data/Gfx/BossBrain.png",GL_LINEAR,GL_LINEAR,GL_CLAMP_TO_EDGE,GL_CLAMP_TO_EDGE)))
 		return false;
-	if(!(imgBossCell=UTIL_GL::Image::LoadImage("Data/Gfx/BossBrainCell.png",GL_LINEAR,GL_LINEAR,GL_CLAMP_TO_EDGE,GL_CLAMP_TO_EDGE)))
+	if(!(imgBossCell=UTIL_GL::Image::LoadImage("/usr/share/prototype/Data/Gfx/BossBrainCell.png",GL_LINEAR,GL_LINEAR,GL_CLAMP_TO_EDGE,GL_CLAMP_TO_EDGE)))
 		return false;
-	if(!(ID=UTIL_GL::Image::LoadImage("Data/Gfx/BossSperm.png",GL_LINEAR,GL_LINEAR,GL_CLAMP_TO_EDGE,GL_CLAMP_TO_EDGE)))
+	if(!(ID=UTIL_GL::Image::LoadImage("/usr/share/prototype/Data/Gfx/BossSperm.png",GL_LINEAR,GL_LINEAR,GL_CLAMP_TO_EDGE,GL_CLAMP_TO_EDGE)))
 		return false;
 	sprBossSperm.Split(ID,128,64,32,32);
-//	if(!(ID=UTIL_GL::Image::LoadImage("Data/Gfx/SolarFlare.png",GL_LINEAR,GL_LINEAR,GL_CLAMP_TO_EDGE,GL_CLAMP_TO_EDGE)))
+//	if(!(ID=UTIL_GL::Image::LoadImage("/usr/share/prototype/Data/Gfx/SolarFlare.png",GL_LINEAR,GL_LINEAR,GL_CLAMP_TO_EDGE,GL_CLAMP_TO_EDGE)))
 //		return false;
 //	sprSolarFlare.Split(ID,128,128,64,64);
-	if(!(ID=UTIL_GL::Image::LoadImage("Data/Gfx/FireBug.png",GL_LINEAR,GL_LINEAR,GL_CLAMP_TO_EDGE,GL_CLAMP_TO_EDGE)))
+	if(!(ID=UTIL_GL::Image::LoadImage("/usr/share/prototype/Data/Gfx/FireBug.png",GL_LINEAR,GL_LINEAR,GL_CLAMP_TO_EDGE,GL_CLAMP_TO_EDGE)))
 		return false;
 	sprFireBug.Split(ID,128,128,128,128);
-	if(!(ID=UTIL_GL::Image::LoadImage("Data/Gfx/SpaceTrash.png",GL_LINEAR,GL_LINEAR,GL_CLAMP_TO_EDGE,GL_CLAMP_TO_EDGE)))
+	if(!(ID=UTIL_GL::Image::LoadImage("/usr/share/prototype/Data/Gfx/SpaceTrash.png",GL_LINEAR,GL_LINEAR,GL_CLAMP_TO_EDGE,GL_CLAMP_TO_EDGE)))
 		return false;
 	sprTrash.Split(ID,128,64,32,32);
 
 
-	if(!(imgMidBoss=UTIL_GL::Image::LoadImage("Data/Gfx/MidBoss.png",GL_LINEAR,GL_LINEAR,GL_CLAMP_TO_EDGE,GL_CLAMP_TO_EDGE)))
+	if(!(imgMidBoss=UTIL_GL::Image::LoadImage("/usr/share/prototype/Data/Gfx/MidBoss.png",GL_LINEAR,GL_LINEAR,GL_CLAMP_TO_EDGE,GL_CLAMP_TO_EDGE)))
 		return false;
-	if(!(imgMidBossSeg=UTIL_GL::Image::LoadImage("Data/Gfx/MidBossSeg.png",GL_LINEAR,GL_LINEAR,GL_CLAMP_TO_EDGE,GL_CLAMP_TO_EDGE)))
+	if(!(imgMidBossSeg=UTIL_GL::Image::LoadImage("/usr/share/prototype/Data/Gfx/MidBossSeg.png",GL_LINEAR,GL_LINEAR,GL_CLAMP_TO_EDGE,GL_CLAMP_TO_EDGE)))
 		return false;
-	if(!(imgMidBossCore=UTIL_GL::Image::LoadImage("Data/Gfx/MidBossCore.png",GL_LINEAR,GL_LINEAR,GL_CLAMP_TO_EDGE,GL_CLAMP_TO_EDGE)))
+	if(!(imgMidBossCore=UTIL_GL::Image::LoadImage("/usr/share/prototype/Data/Gfx/MidBossCore.png",GL_LINEAR,GL_LINEAR,GL_CLAMP_TO_EDGE,GL_CLAMP_TO_EDGE)))
 		return false;
-	if(!(imgMidBossGun=UTIL_GL::Image::LoadImage("Data/Gfx/MidBossGun.png",GL_LINEAR,GL_LINEAR,GL_CLAMP_TO_EDGE,GL_CLAMP_TO_EDGE)))
+	if(!(imgMidBossGun=UTIL_GL::Image::LoadImage("/usr/share/prototype/Data/Gfx/MidBossGun.png",GL_LINEAR,GL_LINEAR,GL_CLAMP_TO_EDGE,GL_CLAMP_TO_EDGE)))
 		return false;
-	if(!(imgMidBossFoot=UTIL_GL::Image::LoadImage("Data/Gfx/MidBossFoot.png",GL_LINEAR,GL_LINEAR,GL_CLAMP_TO_EDGE,GL_CLAMP_TO_EDGE)))
+	if(!(imgMidBossFoot=UTIL_GL::Image::LoadImage("/usr/share/prototype/Data/Gfx/MidBossFoot.png",GL_LINEAR,GL_LINEAR,GL_CLAMP_TO_EDGE,GL_CLAMP_TO_EDGE)))
 		return false;
 
 	//bullets
-	if(!(ID=UTIL_GL::Image::LoadImage("Data/Gfx/BulletSet1.png",GL_LINEAR,GL_LINEAR,GL_CLAMP_TO_EDGE,GL_CLAMP_TO_EDGE)))
+	if(!(ID=UTIL_GL::Image::LoadImage("/usr/share/prototype/Data/Gfx/BulletSet1.png",GL_LINEAR,GL_LINEAR,GL_CLAMP_TO_EDGE,GL_CLAMP_TO_EDGE)))
 		return false;
 	sprBulletSet1.Split(ID,64,128,32,32);
-	if(!(ID=UTIL_GL::Image::LoadImage("Data/Gfx/Forceblast.png",GL_LINEAR,GL_LINEAR,GL_CLAMP_TO_EDGE,GL_CLAMP_TO_EDGE)))
+	if(!(ID=UTIL_GL::Image::LoadImage("/usr/share/prototype/Data/Gfx/Forceblast.png",GL_LINEAR,GL_LINEAR,GL_CLAMP_TO_EDGE,GL_CLAMP_TO_EDGE)))
 		return false;
 	sprBulletForce.Split(ID,128,64,128,64);
-	if(!(ID=UTIL_GL::Image::LoadImage("Data/Gfx/GlowBullets.png",GL_LINEAR,GL_LINEAR,GL_CLAMP_TO_EDGE,GL_CLAMP_TO_EDGE)))
+	if(!(ID=UTIL_GL::Image::LoadImage("/usr/share/prototype/Data/Gfx/GlowBullets.png",GL_LINEAR,GL_LINEAR,GL_CLAMP_TO_EDGE,GL_CLAMP_TO_EDGE)))
 		return false;
 	sprGlowBullets.Split(ID,64,64,32,32);
-	if(!(ID=UTIL_GL::Image::LoadImage("Data/Gfx/BulletHit.png",GL_LINEAR,GL_LINEAR,GL_CLAMP_TO_EDGE,GL_CLAMP_TO_EDGE)))
+	if(!(ID=UTIL_GL::Image::LoadImage("/usr/share/prototype/Data/Gfx/BulletHit.png",GL_LINEAR,GL_LINEAR,GL_CLAMP_TO_EDGE,GL_CLAMP_TO_EDGE)))
 		return false;
 	sprBulletHit.Split(ID,128,64,32,32);
-	if(!(ID=UTIL_GL::Image::LoadImage("Data/Gfx/BulletHit2.png",GL_LINEAR,GL_LINEAR,GL_CLAMP_TO_EDGE,GL_CLAMP_TO_EDGE)))
+	if(!(ID=UTIL_GL::Image::LoadImage("/usr/share/prototype/Data/Gfx/BulletHit2.png",GL_LINEAR,GL_LINEAR,GL_CLAMP_TO_EDGE,GL_CLAMP_TO_EDGE)))
 		return false;
 	sprBulletHit2.Split(ID,128,64,32,32);
-	if(!(ID=UTIL_GL::Image::LoadImage("Data/Gfx/BulletHit3.png",GL_LINEAR,GL_LINEAR,GL_CLAMP_TO_EDGE,GL_CLAMP_TO_EDGE)))
+	if(!(ID=UTIL_GL::Image::LoadImage("/usr/share/prototype/Data/Gfx/BulletHit3.png",GL_LINEAR,GL_LINEAR,GL_CLAMP_TO_EDGE,GL_CLAMP_TO_EDGE)))
 		return false;
 	sprBulletHit3.Split(ID,128,64,32,32);
-	if(!(ID=UTIL_GL::Image::LoadImage("Data/Gfx/ClusterBomb.png",GL_LINEAR,GL_LINEAR,GL_CLAMP_TO_EDGE,GL_CLAMP_TO_EDGE)))
+	if(!(ID=UTIL_GL::Image::LoadImage("/usr/share/prototype/Data/Gfx/ClusterBomb.png",GL_LINEAR,GL_LINEAR,GL_CLAMP_TO_EDGE,GL_CLAMP_TO_EDGE)))
 		return false;
 	sprCluster.Split(ID,128,64,32,32);
-	if(!(ID=UTIL_GL::Image::LoadImage("Data/Gfx/Missile.png",GL_LINEAR,GL_LINEAR,GL_CLAMP_TO_EDGE,GL_CLAMP_TO_EDGE)))
+	if(!(ID=UTIL_GL::Image::LoadImage("/usr/share/prototype/Data/Gfx/Missile.png",GL_LINEAR,GL_LINEAR,GL_CLAMP_TO_EDGE,GL_CLAMP_TO_EDGE)))
 		return false;
 	sprMissile.Split(ID,256,128,64,64);
-	if(!(ID=UTIL_GL::Image::LoadImage("Data/Gfx/FireBall.png",GL_LINEAR,GL_LINEAR,GL_CLAMP_TO_EDGE,GL_CLAMP_TO_EDGE)))
+	if(!(ID=UTIL_GL::Image::LoadImage("/usr/share/prototype/Data/Gfx/FireBall.png",GL_LINEAR,GL_LINEAR,GL_CLAMP_TO_EDGE,GL_CLAMP_TO_EDGE)))
 		return false;
 	sprFireBall.Split(ID,512,512,128,128);
-	if(!(ID=UTIL_GL::Image::LoadImage("Data/Gfx/HomingMissile.png",GL_LINEAR,GL_LINEAR,GL_CLAMP_TO_EDGE,GL_CLAMP_TO_EDGE)))
+	if(!(ID=UTIL_GL::Image::LoadImage("/usr/share/prototype/Data/Gfx/HomingMissile.png",GL_LINEAR,GL_LINEAR,GL_CLAMP_TO_EDGE,GL_CLAMP_TO_EDGE)))
 		return false;
 	sprHomingMissile.Split(ID,256,256,64,64);
 
-	if(!(ID=UTIL_GL::Image::LoadImage("Data/Gfx/PowerUp1.png",GL_LINEAR,GL_LINEAR,GL_CLAMP_TO_EDGE,GL_CLAMP_TO_EDGE)))
+	if(!(ID=UTIL_GL::Image::LoadImage("/usr/share/prototype/Data/Gfx/PowerUp1.png",GL_LINEAR,GL_LINEAR,GL_CLAMP_TO_EDGE,GL_CLAMP_TO_EDGE)))
 		return false;
 	sprPowerUp.Split(ID,512,512,64,64);
-	if(!(ID=UTIL_GL::Image::LoadImage("Data/Gfx/BitUpgrade.png",GL_LINEAR,GL_LINEAR,GL_CLAMP_TO_EDGE,GL_CLAMP_TO_EDGE)))
+	if(!(ID=UTIL_GL::Image::LoadImage("/usr/share/prototype/Data/Gfx/BitUpgrade.png",GL_LINEAR,GL_LINEAR,GL_CLAMP_TO_EDGE,GL_CLAMP_TO_EDGE)))
 		return false;
 	sprBit.Split(ID,128,64,32,32);
 
 	//explosions
-	if(!(ID=UTIL_GL::Image::LoadImage("Data/Gfx/ExplosionSmall.png",GL_LINEAR,GL_LINEAR,GL_CLAMP_TO_EDGE,GL_CLAMP_TO_EDGE)))
+	if(!(ID=UTIL_GL::Image::LoadImage("/usr/share/prototype/Data/Gfx/ExplosionSmall.png",GL_LINEAR,GL_LINEAR,GL_CLAMP_TO_EDGE,GL_CLAMP_TO_EDGE)))
 		return false;
 	sprExplosionSmall.Split(ID,1024,512,128,128);
-	if(!(ID=UTIL_GL::Image::LoadImage("Data/Gfx/Explosion2.png",GL_LINEAR,GL_LINEAR,GL_CLAMP_TO_EDGE,GL_CLAMP_TO_EDGE)))
+	if(!(ID=UTIL_GL::Image::LoadImage("/usr/share/prototype/Data/Gfx/Explosion2.png",GL_LINEAR,GL_LINEAR,GL_CLAMP_TO_EDGE,GL_CLAMP_TO_EDGE)))
 		return false;
 	sprExplosion2.Split(ID,512,256,64,64);
-	if(!(ID=UTIL_GL::Image::LoadImage("Data/Gfx/Implosion.png",GL_LINEAR,GL_LINEAR,GL_CLAMP_TO_EDGE,GL_CLAMP_TO_EDGE)))
+	if(!(ID=UTIL_GL::Image::LoadImage("/usr/share/prototype/Data/Gfx/Implosion.png",GL_LINEAR,GL_LINEAR,GL_CLAMP_TO_EDGE,GL_CLAMP_TO_EDGE)))
 		return false;
 	sprImplosion.Split(ID,512,256,64,64);
-	if(!(ID=UTIL_GL::Image::LoadImage("Data/Gfx/BlueExplosion.png",GL_LINEAR,GL_LINEAR,GL_CLAMP_TO_EDGE,GL_CLAMP_TO_EDGE)))
+	if(!(ID=UTIL_GL::Image::LoadImage("/usr/share/prototype/Data/Gfx/BlueExplosion.png",GL_LINEAR,GL_LINEAR,GL_CLAMP_TO_EDGE,GL_CLAMP_TO_EDGE)))
 		return false;
 	sprBlueExplosion.Split(ID,512,256,64,64);
-	if(!(ID=UTIL_GL::Image::LoadImage("Data/Gfx/SmokePuff.png",GL_LINEAR,GL_LINEAR,GL_CLAMP_TO_EDGE,GL_CLAMP_TO_EDGE)))
+	if(!(ID=UTIL_GL::Image::LoadImage("/usr/share/prototype/Data/Gfx/SmokePuff.png",GL_LINEAR,GL_LINEAR,GL_CLAMP_TO_EDGE,GL_CLAMP_TO_EDGE)))
 		return false;
 	sprSmokePuff.Split(ID,128,128,32,32);
 
 	//sprite effects
-	if(!(ID=UTIL_GL::Image::LoadImage("Data/Gfx/PlayerSpeed.png",GL_LINEAR,GL_LINEAR,GL_CLAMP_TO_EDGE,GL_CLAMP_TO_EDGE)))
+	if(!(ID=UTIL_GL::Image::LoadImage("/usr/share/prototype/Data/Gfx/PlayerSpeed.png",GL_LINEAR,GL_LINEAR,GL_CLAMP_TO_EDGE,GL_CLAMP_TO_EDGE)))
 		return false;
 	sprSpeedBoost.Split(ID,512,256,64,64);
-	if(!(ID=UTIL_GL::Image::LoadImage("Data/Gfx/Anim_back1.png",GL_LINEAR,GL_LINEAR,GL_CLAMP_TO_EDGE,GL_CLAMP_TO_EDGE)))
+	if(!(ID=UTIL_GL::Image::LoadImage("/usr/share/prototype/Data/Gfx/Anim_back1.png",GL_LINEAR,GL_LINEAR,GL_CLAMP_TO_EDGE,GL_CLAMP_TO_EDGE)))
 		return false;
 	sprAnimBack1.Split(ID,256,128,64,128);
-	if(!(ID=UTIL_GL::Image::LoadImage("Data/Gfx/Anim_back2.png",GL_LINEAR,GL_LINEAR,GL_CLAMP_TO_EDGE,GL_CLAMP_TO_EDGE)))
+	if(!(ID=UTIL_GL::Image::LoadImage("/usr/share/prototype/Data/Gfx/Anim_back2.png",GL_LINEAR,GL_LINEAR,GL_CLAMP_TO_EDGE,GL_CLAMP_TO_EDGE)))
 		return false;
 	sprAnimBack2.Split(ID,128,128,64,64);
-	if(!(ID=UTIL_GL::Image::LoadImage("Data/Gfx/Anim_Fore1.png",GL_LINEAR,GL_LINEAR,GL_CLAMP_TO_EDGE,GL_CLAMP_TO_EDGE)))
+	if(!(ID=UTIL_GL::Image::LoadImage("/usr/share/prototype/Data/Gfx/Anim_Fore1.png",GL_LINEAR,GL_LINEAR,GL_CLAMP_TO_EDGE,GL_CLAMP_TO_EDGE)))
 		return false;
 	sprAnimFore1.Split(ID,128,128,64,64);
-	if(!(ID=UTIL_GL::Image::LoadImage("Data/Gfx/Anim_Fore2.png",GL_LINEAR,GL_LINEAR,GL_CLAMP_TO_EDGE,GL_CLAMP_TO_EDGE)))
+	if(!(ID=UTIL_GL::Image::LoadImage("/usr/share/prototype/Data/Gfx/Anim_Fore2.png",GL_LINEAR,GL_LINEAR,GL_CLAMP_TO_EDGE,GL_CLAMP_TO_EDGE)))
 		return false;
 	sprAnimFore2.Split(ID,128,128,32,32);
-	if(!(ID=UTIL_GL::Image::LoadImage("Data/Gfx/Anim_Fore3.png",GL_LINEAR,GL_LINEAR,GL_CLAMP_TO_EDGE,GL_CLAMP_TO_EDGE)))
+	if(!(ID=UTIL_GL::Image::LoadImage("/usr/share/prototype/Data/Gfx/Anim_Fore3.png",GL_LINEAR,GL_LINEAR,GL_CLAMP_TO_EDGE,GL_CLAMP_TO_EDGE)))
 		return false;
 	sprAnimFore3.Split(ID,256,128,64,32);
 	//breakable scenery
-	if(!(ID=UTIL_GL::Image::LoadImage("Data/Gfx/DamageTile1.png",GL_LINEAR,GL_LINEAR,GL_CLAMP_TO_EDGE,GL_CLAMP_TO_EDGE)))
+	if(!(ID=UTIL_GL::Image::LoadImage("/usr/share/prototype/Data/Gfx/DamageTile1.png",GL_LINEAR,GL_LINEAR,GL_CLAMP_TO_EDGE,GL_CLAMP_TO_EDGE)))
 		return false;
 	sprBreak1.Split(ID,256,128,64,128);
 
-	if(!(ID=UTIL_GL::Image::LoadImage("Data/Gfx/water.png",GL_LINEAR,GL_LINEAR,GL_REPEAT,GL_REPEAT)))
+	if(!(ID=UTIL_GL::Image::LoadImage("/usr/share/prototype/Data/Gfx/water.png",GL_LINEAR,GL_LINEAR,GL_REPEAT,GL_REPEAT)))
 		return false;
 	sprWater.Split(ID,32,32,32,32);
-	if(!(ID=UTIL_GL::Image::LoadImage("Data/Gfx/watersplash.png",GL_LINEAR,GL_LINEAR,GL_REPEAT,GL_REPEAT)))
+	if(!(ID=UTIL_GL::Image::LoadImage("/usr/share/prototype/Data/Gfx/watersplash.png",GL_LINEAR,GL_LINEAR,GL_REPEAT,GL_REPEAT)))
 		return false;
 	sprWaterSplash.Split(ID,64,64,32,32);
 
 	//splash related
-	if(!(imgSplash=UTIL_GL::Image::LoadImage("Data/Gfx/Splash.png",GL_LINEAR,GL_LINEAR,GL_REPEAT,GL_REPEAT)))
+	if(!(imgSplash=UTIL_GL::Image::LoadImage("/usr/share/prototype/Data/Gfx/Splash.png",GL_LINEAR,GL_LINEAR,GL_REPEAT,GL_REPEAT)))
 		return false;
-	if(!(imgSplashFire=UTIL_GL::Image::LoadImage("Data/Gfx/Fire.png",GL_LINEAR,GL_LINEAR,GL_REPEAT,GL_REPEAT)))
+	if(!(imgSplashFire=UTIL_GL::Image::LoadImage("/usr/share/prototype/Data/Gfx/Fire.png",GL_LINEAR,GL_LINEAR,GL_REPEAT,GL_REPEAT)))
 		return false;
-	if(!(imgSplashBack=UTIL_GL::Image::LoadImage("Data/Gfx/SplashBack.png",GL_LINEAR,GL_LINEAR,GL_REPEAT,GL_REPEAT)))
+	if(!(imgSplashBack=UTIL_GL::Image::LoadImage("/usr/share/prototype/Data/Gfx/SplashBack.png",GL_LINEAR,GL_LINEAR,GL_REPEAT,GL_REPEAT)))
 		return false;
-	if(!(imgStage1=UTIL_GL::Image::LoadImage("Data/Gfx/Stage1.png",GL_LINEAR,GL_LINEAR,GL_REPEAT,GL_REPEAT)))
+	if(!(imgStage1=UTIL_GL::Image::LoadImage("/usr/share/prototype/Data/Gfx/Stage1.png",GL_LINEAR,GL_LINEAR,GL_REPEAT,GL_REPEAT)))
 		return false;
-	if(!(imgGameOver=UTIL_GL::Image::LoadImage("Data/Gfx/GameOver.png",GL_LINEAR,GL_LINEAR,GL_REPEAT,GL_REPEAT)))
+	if(!(imgGameOver=UTIL_GL::Image::LoadImage("/usr/share/prototype/Data/Gfx/GameOver.png",GL_LINEAR,GL_LINEAR,GL_REPEAT,GL_REPEAT)))
 		return false;
-	if(!(imgProtoTypeMk1=UTIL_GL::Image::LoadImage("Data/Gfx/Prototypemk1.png",GL_LINEAR,GL_LINEAR,GL_REPEAT,GL_REPEAT)))
+	if(!(imgProtoTypeMk1=UTIL_GL::Image::LoadImage("/usr/share/prototype/Data/Gfx/Prototypemk1.png",GL_LINEAR,GL_LINEAR,GL_REPEAT,GL_REPEAT)))
 		return false;
-	if(!(imgCongrats=UTIL_GL::Image::LoadImage("Data/Gfx/Congrats.png",GL_LINEAR,GL_LINEAR,GL_REPEAT,GL_REPEAT)))
+	if(!(imgCongrats=UTIL_GL::Image::LoadImage("/usr/share/prototype/Data/Gfx/Congrats.png",GL_LINEAR,GL_LINEAR,GL_REPEAT,GL_REPEAT)))
 		return false;
-	if(!(ID=UTIL_GL::Image::LoadImage("Data/Gfx/GameOverSeq.png",GL_LINEAR,GL_LINEAR,GL_REPEAT,GL_REPEAT)))
+	if(!(ID=UTIL_GL::Image::LoadImage("/usr/share/prototype/Data/Gfx/GameOverSeq.png",GL_LINEAR,GL_LINEAR,GL_REPEAT,GL_REPEAT)))
 		return false;
 	sprGameOver.Split(ID,512,512,128,128);
-	if(!(ID=UTIL_GL::Image::LoadImage("Data/Gfx/HiScoreSeq.png",GL_LINEAR,GL_LINEAR,GL_REPEAT,GL_REPEAT)))
+	if(!(ID=UTIL_GL::Image::LoadImage("/usr/share/prototype/Data/Gfx/HiScoreSeq.png",GL_LINEAR,GL_LINEAR,GL_REPEAT,GL_REPEAT)))
 		return false;
 	sprHiScore.Split(ID,512,512,128,128);
-	if(!(ID=UTIL_GL::Image::LoadImage("Data/Gfx/LiftOff2.png",GL_LINEAR,GL_LINEAR,GL_REPEAT,GL_REPEAT)))
+	if(!(ID=UTIL_GL::Image::LoadImage("/usr/share/prototype/Data/Gfx/LiftOff2.png",GL_LINEAR,GL_LINEAR,GL_REPEAT,GL_REPEAT)))
 		return false;
 	sprIntro.Split(ID,512,512,128,128);
-	if(!(ID=UTIL_GL::Image::LoadImage("Data/Gfx/LiftOff.png",GL_LINEAR,GL_LINEAR,GL_REPEAT,GL_REPEAT)))
+	if(!(ID=UTIL_GL::Image::LoadImage("/usr/share/prototype/Data/Gfx/LiftOff.png",GL_LINEAR,GL_LINEAR,GL_REPEAT,GL_REPEAT)))
 		return false;
 	sprIntro2.Split(ID,512,256,128,128);
 
 
 	//level tiles
 	char sz[256];
-	sprintf(sz,"Data/Gfx/Stage%iLayer%i.png",iStage,1);
+	sprintf(sz,"/usr/share/prototype/Data/Gfx/Stage%iLayer%i.png",iStage,1);
 	if(!(ID=UTIL_GL::Image::LoadImage(sz,GL_LINEAR,GL_LINEAR,GL_REPEAT,GL_REPEAT)))
 		return false;
 	sprMapTiles.Split(ID,1024,1024,32,32);
 
 
 	//sounds
-	if(!smpLaser1.Load("Data/Sound/Laser.wav",false))
+	if(!smpLaser1.Load("/usr/share/prototype/Data/Sound/Laser.wav",false))
 		return false;
-	if(!smpLaser2.Load("Data/Sound/Laser2.wav",false))
+	if(!smpLaser2.Load("/usr/share/prototype/Data/Sound/Laser2.wav",false))
 		return false;
-	if(!smpLaser3.Load("Data/Sound/Laser3.wav",false))
+	if(!smpLaser3.Load("/usr/share/prototype/Data/Sound/Laser3.wav",false))
 		return false;
-	if(!smpLaser4.Load("Data/Sound/Laser4.wav",false))
+	if(!smpLaser4.Load("/usr/share/prototype/Data/Sound/Laser4.wav",false))
 		return false;
-	if(!smpLaser5.Load("Data/Sound/Laser5.wav",false))
+	if(!smpLaser5.Load("/usr/share/prototype/Data/Sound/Laser5.wav",false))
 		return false;
-	if(!smpBigLaser1.Load("Data/Sound/BigLaser1.wav",false))
+	if(!smpBigLaser1.Load("/usr/share/prototype/Data/Sound/BigLaser1.wav",false))
 		return false;
-	if(!smpForceShot.Load("Data/Sound/Force.wav",false))
+	if(!smpForceShot.Load("/usr/share/prototype/Data/Sound/Force.wav",false))
 		return false;
-	if(!smpForceCharge.Load("Data/Sound/ForceCharge.wav",true))
+	if(!smpForceCharge.Load("/usr/share/prototype/Data/Sound/ForceCharge.wav",true))
 		return false;
-	if(!smpDeflect.Load("Data/Sound/Deflection.wav",false))
+	if(!smpDeflect.Load("/usr/share/prototype/Data/Sound/Deflection.wav",false))
 		return false;
-	if(!smpSmallExplode.Load("Data/Sound/SmallExplosion.wav",false))
+	if(!smpSmallExplode.Load("/usr/share/prototype/Data/Sound/SmallExplosion.wav",false))
 		return false;
-	if(!smpMedExplode.Load("Data/Sound/MedExplosion.wav",false))
+	if(!smpMedExplode.Load("/usr/share/prototype/Data/Sound/MedExplosion.wav",false))
 		return false;
-	if(!smpLargeExplode.Load("Data/Sound/LargeExplosion.wav",false))
+	if(!smpLargeExplode.Load("/usr/share/prototype/Data/Sound/LargeExplosion.wav",false))
 		return false;
-	if(!smpLargeExplode2.Load("Data/Sound/LargeExplosion2.wav",false))
+	if(!smpLargeExplode2.Load("/usr/share/prototype/Data/Sound/LargeExplosion2.wav",false))
 		return false;
-	if(!smpPickup.Load("Data/Sound/Pickup.wav",false))
+	if(!smpPickup.Load("/usr/share/prototype/Data/Sound/Pickup.wav",false))
 		return false;
-	if(!smpOrbSpawn.Load("Data/Sound/OrbSpawn.wav",false))
+	if(!smpOrbSpawn.Load("/usr/share/prototype/Data/Sound/OrbSpawn.wav",false))
 		return false;
-	if(!smpLaser6.Load("Data/Sound/Laser6.wav",false))
+	if(!smpLaser6.Load("/usr/share/prototype/Data/Sound/Laser6.wav",false))
 		return false;
-	if(!smpBigLaser2.Load("Data/Sound/BigLaser2.wav",false))
+	if(!smpBigLaser2.Load("/usr/share/prototype/Data/Sound/BigLaser2.wav",false))
 		return false;
-	if(!smpLaser7.Load("Data/Sound/Laser7.wav",false))
+	if(!smpLaser7.Load("/usr/share/prototype/Data/Sound/Laser7.wav",false))
 		return false;
 
-	if(!smpUIselect.Load("Data/Sound/UI_select.wav",false))
+	if(!smpUIselect.Load("/usr/share/prototype/Data/Sound/UI_select.wav",false))
 		return false;
-	if(!smpUImove.Load("Data/Sound/UI_move.wav",false))
+	if(!smpUImove.Load("/usr/share/prototype/Data/Sound/UI_move.wav",false))
 		return false;
 smpUImove.Play();
 //smpUImove.Stop(1);
-	if(!stmStage.Load("Data/Sound/Stage1.ogg",true))
+	if(!stmStage.Load("/usr/share/prototype/Data/Sound/Stage1.ogg",true))
 		return false;
-	if(!stmBoss.Load("Data/Sound/Boss.ogg",true))
+	if(!stmBoss.Load("/usr/share/prototype/Data/Sound/Boss.ogg",true))
 		return false;
 
 	gLog.OutPut("Total Texture Mem usage:");
@@ -1182,13 +1182,13 @@ smpUImove.Play();
 bool Engine::LoadEditorData()
 {
 	unsigned int ID=0;
-	if(!(ID=UTIL_GL::Image::LoadImage("Data/Gfx/ED_ico.png",GL_NEAREST,GL_NEAREST,GL_CLAMP_TO_EDGE,GL_CLAMP_TO_EDGE)))
+	if(!(ID=UTIL_GL::Image::LoadImage("/usr/share/prototype/Data/Gfx/ED_ico.png",GL_NEAREST,GL_NEAREST,GL_CLAMP_TO_EDGE,GL_CLAMP_TO_EDGE)))
 		return false;
 	sprIcons.Split(ID,64,64,32,32);
-	if(!(ID=UTIL_GL::Image::LoadImage("Data/Gfx/Entities.png",GL_LINEAR,GL_LINEAR,GL_REPEAT,GL_REPEAT)))
+	if(!(ID=UTIL_GL::Image::LoadImage("/usr/share/prototype/Data/Gfx/Entities.png",GL_LINEAR,GL_LINEAR,GL_REPEAT,GL_REPEAT)))
 		return false;
 	sprEntities.Split(ID,512,512,32,32);
-	if(!(ID=UTIL_GL::Image::LoadImage("Data/Gfx/Anims.png",GL_LINEAR,GL_LINEAR,GL_REPEAT,GL_REPEAT)))
+	if(!(ID=UTIL_GL::Image::LoadImage("/usr/share/prototype/Data/Gfx/Anims.png",GL_LINEAR,GL_LINEAR,GL_REPEAT,GL_REPEAT)))
 		return false;
 	sprAnims.Split(ID,512,512,32,32);
 
@@ -1206,7 +1206,7 @@ bool Engine::LoadMapData()
 	ClearMapEntities();
 
 	CFileIO fIO;
-	sprintf(sz,"Data/Maps/Stage%i.stg",iStage);
+	sprintf(sz,"/usr/share/prototype/Data/Maps/Stage%i.stg",iStage);
 	if(!fIO.GetFile(sz,"rb"))
 		return false;
 	int x,y;
@@ -1266,7 +1266,7 @@ void Engine::SaveMap()
 {
 	CFileIO fIO;
 	char sz[256];
-	sprintf(sz,"Data/Maps/Stage%i.stg",iStage);
+	sprintf(sz,"/usr/share/prototype/Data/Maps/Stage%i.stg",iStage);
 	if(!fIO.GetFile(sz,"wb"))
 		return;
 	int x,y;
diff --git a/FrontEnd.cpp b/FrontEnd.cpp
index 01feb63..3f30875 100755
--- a/FrontEnd.cpp
+++ b/FrontEnd.cpp
@@ -23,7 +23,7 @@ void Engine::StartFrontEnd()
 	bInGame = false;
 	if(!bDemo)
 	{
-		stmSplash.Load("Data/Sound/AttractMode.ogg",true);
+		stmSplash.Load("/usr/share/prototype/Data/Sound/AttractMode.ogg",true);
 		stmSplash.Play(30);
 	}
 	bDemo = false;
@@ -621,7 +621,7 @@ void Engine::StartGameOver()
 	bBossFight = false;
 	bWimpMode = false;
 	SetState(&Engine::UpdateGameOver,&Engine::RenderGameOver);
-	stmSplash.Load("Data/Sound/GameOver.ogg",false);
+	stmSplash.Load("/usr/share/prototype/Data/Sound/GameOver.ogg",false);
 	stmSplash.Play(30);
 	StartFadeIn();
 	fGameOverFrame = 0;
@@ -1008,7 +1008,7 @@ void Engine::RenderHiScores(const float interp)
 void Engine::StartEnterHiScore()
 {
 	SetState(&Engine::UpdateEnterHiScore,&Engine::RenderEnterHiScore);
-	stmSplash.Load("Data/Sound/HiScores.ogg",false);
+	stmSplash.Load("/usr/share/prototype/Data/Sound/HiScores.ogg",false);
 	stmSplash.Play(30);
 	StartFadeIn();
 	fGameOverFrame = 0;
diff --git a/Game.cpp b/Game.cpp
index 549ce3e..f52ef20 100644
--- a/Game.cpp
+++ b/Game.cpp
@@ -86,7 +86,7 @@ void Engine::InitializeGame()
 	mPlayer.Spawn(Vec2(0,180));
 	if(!bDemo)
 	{
-		stmStage.Load("Data/Sound/Stage1.ogg",true);
+		stmStage.Load("/usr/share/prototype/Data/Sound/Stage1.ogg",true);
 		//stmStage.Seek(0,30);
 		stmStage.Play(30);
 	}
@@ -659,7 +659,7 @@ void Engine::StartCompleted()
 	gpEngine->smpForceCharge.Stop(FSOUND_ALL);
 	#endif
 	SND_LARGEEXPLODE2;
-	gpEngine->stmSplash.Load("Data/Sound/HiScores.ogg",false);
+	gpEngine->stmSplash.Load("/usr/share/prototype/Data/Sound/HiScores.ogg",false);
 	gpEngine->stmSplash.Play(30);
 
 }
