diff -ruN libdrm/data/amdgpu.ids libdrm-mod/data/amdgpu.ids
--- libdrm/data/amdgpu.ids	2020-05-25 18:24:34.384791428 +0200
+++ libdrm-mod/data/amdgpu.ids	2020-05-25 18:55:16.641564658 +0200
@@ -218,3 +218,7 @@
 9874,	85,	AMD Radeon R6 Graphics
 9874,	84,	AMD Radeon R7 Graphics
 6FDF,	EF,	AMD Radeon RX 580 2048SP
+9920,	00, AMD Liverpool (PlayStation 4)
+9922,	00, AMD Liverpool (PlayStation 4)
+9923,	00, AMD Liverpool (PlayStation 4)
+9924,	11, AMD Gladius (PlayStation 4)
diff -ruN libdrm/radeon/r600_pci_ids.h libdrm-mod/radeon/r600_pci_ids.h
--- libdrm/radeon/r600_pci_ids.h	2020-05-25 18:24:34.398124762 +0200
+++ libdrm-mod/radeon/r600_pci_ids.h	2020-05-25 18:40:57.654848297 +0200
@@ -473,6 +473,11 @@
 CHIPSET(0x131C, KAVERI_131C, KAVERI)
 CHIPSET(0x131D, KAVERI_131D, KAVERI)
 
+CHIPSET(0x9920, LIVERPOOL_9920, LIVERPOOL)
+CHIPSET(0x9922, LIVERPOOL_9922, LIVERPOOL)
+CHIPSET(0x9923, LIVERPOOL_9923, LIVERPOOL)
+CHIPSET(0x9924, GLADIUS_9924, GLADIUS)
+
 CHIPSET(0x67A0, HAWAII_67A0, HAWAII)
 CHIPSET(0x67A1, HAWAII_67A1, HAWAII)
 CHIPSET(0x67A2, HAWAII_67A2, HAWAII)
diff -ruN libdrm/radeon/radeon_surface.c libdrm-mod/radeon/radeon_surface.c
--- libdrm/radeon/radeon_surface.c	2020-05-25 18:24:34.398124762 +0200
+++ libdrm-mod/radeon/radeon_surface.c	2020-05-25 18:38:00.004838001 +0200
@@ -86,6 +86,8 @@
     CHIP_HAINAN,
     CHIP_BONAIRE,
     CHIP_KAVERI,
+    CHIP_LIVERPOOL,
+    CHIP_GLADIUS,
     CHIP_KABINI,
     CHIP_HAWAII,
     CHIP_MULLINS,
