diff --git a/src/intel/compiler/meson.build b/src/intel/compiler/meson.build
index da9a958190..f9c2a85bfe 100644
--- a/src/intel/compiler/meson.build
+++ b/src/intel/compiler/meson.build
@@ -184,7 +184,7 @@ elif with_intel_clc
     # If we can run host binaries directly, just build intel_clc for the host.
     # Most commonly this happens when doing a cross compile from an x86_64 build
     # machine to an x86 host
-    native : not meson.can_run_host_binaries(),
+    #native : not meson.can_run_host_binaries(),
     install : get_option('install-intel-clc'),
   )
   dep_prog_intel_clc = [prog_intel_clc]
