From 9a66a614c34e9f834856c4fa1ed219c431145438 Mon Sep 17 00:00:00 2001
From: Nicolas Adenis-Lamarre <nicolas.adenis-lamarre@gmail.com>
Date: Sat, 27 Feb 2016 09:31:36 +0100
Subject: [PATCH] kodi: add file based kodi power events

Signed-off-by: Nicolas Adenis-Lamarre <nicolas.adenis-lamarre@gmail.com>
---
 xbmc/ApplicationMessenger.cpp | 23 +++++++++++++++++++++++
 1 file changed, 23 insertions(+)

diff --git a/xbmc/ApplicationMessenger.cpp b/xbmc/ApplicationMessenger.cpp
index e0152aa..72e68ed 100644
--- a/xbmc/messaging/ApplicationMessenger.cpp
+++ b/xbmc/messaging/ApplicationMessenger.cpp
@@ -73,6 +73,27 @@ using namespace std;
 using namespace MUSIC_INFO;
 using namespace PERIPHERALS;
 
+void RecalboxMsg(unsigned int dwMessage) {
+  //CLog::Log(LOGNOTICE, "RECALBOX MESSAGE %i (SHUTDOWN=%i, POWERDOWN=%i, QUIT=%i, RESTART=%i)", pMsg->dwMessage, TMSG_SHUTDOWN, TMSG_POWERDOWN, TMSG_QUIT, TMSG_RESTART);
+  if(dwMessage == TMSG_QUIT || dwMessage == TMSG_RESTART || dwMessage == TMSG_POWERDOWN) {
+    FILE* fd = fopen("/var/run/kodi.msg", "a");
+    if(fd != NULL) {
+      switch(dwMessage) {
+      case TMSG_QUIT:
+	fprintf(fd, "EXIT\n");
+	break;
+      case TMSG_RESTART:
+	fprintf(fd, "RESTART\n");
+	break;
+      case TMSG_POWERDOWN:
+	fprintf(fd, "SHUTDOWN\n");
+	break;
+      }
+      fclose(fd);
+    }
+  }
+}
+
 CDelayedMessage::CDelayedMessage(ThreadMessage& msg, unsigned int delay) : CThread("DelayedMessage")
 {
   m_msg.dwMessage  = msg.dwMessage;
@@ -214,6 +235,8 @@
 
 void CApplicationMessenger::ProcessMessage(ThreadMessage *pMsg)
 {
+  RecalboxMsg(pMsg->dwMessage);
+
   //special case for this that we handle ourselves
   if (pMsg->dwMessage == TMSG_CALLBACK)
   {
