diff --git a/os_dep/ioctl_cfg80211.c b/os_dep/ioctl_cfg80211.c
index fb46f122e4..41f9a93ad7 100644
--- a/os_dep/ioctl_cfg80211.c
+++ b/os_dep/ioctl_cfg80211.c
@@ -3159,6 +3159,9 @@ static int cfg80211_rtw_set_txpower(struct wiphy *wiphy,
 static int cfg80211_rtw_get_txpower(struct wiphy *wiphy,
 #if (LINUX_VERSION_CODE >= KERNEL_VERSION(3,8,0))
 	struct wireless_dev *wdev,
+#endif
+#if (LINUX_VERSION_CODE >= KERNEL_VERSION(6,14,0))
+	unsigned int link_id,
 #endif
 	int *dbm)
 {
