diff --git a/os_dep/usb_intf.c b/os_dep/usb_intf.c
index 20b827c857..36e02a6011 100644
--- a/os_dep/usb_intf.c
+++ b/os_dep/usb_intf.c
@@ -172,7 +172,7 @@ static struct rtw_usb_drv usb_drv = {
 	.usbdrv.supports_autosuspend = 1,
 	#endif
 
-	#if (LINUX_VERSION_CODE >= KERNEL_VERSION(2, 6, 19))
+	#if (LINUX_VERSION_CODE >= KERNEL_VERSION(2, 6, 19) && LINUX_VERSION_CODE < KERNEL_VERSION(6, 8, 0))
 	.usbdrv.drvwrap.driver.shutdown = rtw_dev_shutdown,
 	#else
 	.usbdrv.driver.shutdown = rtw_dev_shutdown,
