diff --git a/core/rtw_br_ext.c b/core/rtw_br_ext.c.new
index 50063ab..c8f6230 100644
--- a/core/rtw_br_ext.c
+++ b/core/rtw_br_ext.c.new
@@ -82,7 +82,11 @@ static __inline__ unsigned char *__nat25_find_pppoe_tag(struct pppoe_hdr *ph, un
 	unsigned char *cur_ptr, *start_ptr;
 	unsigned short tagLen, tagType;
 
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(7, 1, 0)
+	start_ptr = cur_ptr = (unsigned char *)ph + sizeof(*ph);
+#else
 	start_ptr = cur_ptr = (unsigned char *)ph->tag;
+#endif
 	while((cur_ptr - start_ptr) < ntohs(ph->length)) {
 		// prevent un-alignment access
 		tagType = (unsigned short)((cur_ptr[0] << 8) + cur_ptr[1]);
@@ -108,9 +112,15 @@ static __inline__ int __nat25_add_pppoe_tag(struct sk_buff *skb, struct pppoe_ta
 
 	skb_put(skb, data_len);
 	// have a room for new tag
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(7, 1, 0)
+	memmove(((unsigned char *)ph + sizeof(*ph) + data_len), (unsigned char *)ph + sizeof(*ph), ntohs(ph->length));
+	ph->length = htons(ntohs(ph->length) + data_len);
+	memcpy((unsigned char *)ph + sizeof(*ph), tag, data_len);
+#else
 	memmove(((unsigned char *)ph->tag + data_len), (unsigned char *)ph->tag, ntohs(ph->length));
 	ph->length = htons(ntohs(ph->length) + data_len);
 	memcpy((unsigned char *)ph->tag, tag, data_len);
+#endif
 	return data_len;
 }
 
@@ -1225,8 +1235,13 @@ int nat25_db_handle(_adapter *priv, struct sk_buff *skb, int method)
 									return -1;
 								}
 
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(7, 1, 0)
+								memcpy((unsigned char *)tag + sizeof(*tag) + MAGIC_CODE_LEN+RTL_RELAY_TAG_LEN,
+									(unsigned char *)pOldTag + sizeof(*pOldTag), old_tag_len);
+#else
 								memcpy(tag->tag_data+MAGIC_CODE_LEN+RTL_RELAY_TAG_LEN,
 									pOldTag->tag_data, old_tag_len);
+#endif
 
 								if (skb_pull_and_merge(skb, (unsigned char *)pOldTag, TAG_HDR_LEN+old_tag_len) < 0) {
 									DEBUG_ERR("call skb_pull_and_merge() failed in PADI/R packet!\n");
@@ -1239,9 +1254,15 @@ int nat25_db_handle(_adapter *priv, struct sk_buff *skb, int method)
 							tag->tag_len = htons(MAGIC_CODE_LEN+RTL_RELAY_TAG_LEN+old_tag_len);
 
 							// insert the magic_code+client mac in relay tag
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(7, 1, 0)
+							pMagic = (__be16 *)((unsigned char *)tag + sizeof(*tag));
+							*pMagic = htons(MAGIC_CODE);
+							memcpy((unsigned char *)tag + sizeof(*tag) + MAGIC_CODE_LEN, skb->data+ETH_ALEN, ETH_ALEN);
+#else
 							pMagic = (__be16 *)tag->tag_data;
 							*pMagic = htons(MAGIC_CODE);
 							memcpy(tag->tag_data+MAGIC_CODE_LEN, skb->data+ETH_ALEN, ETH_ALEN);
+#endif
 
 							//Add relay tag
 							if(__nat25_add_pppoe_tag(skb, tag) < 0)
@@ -1306,14 +1327,22 @@ int nat25_db_handle(_adapter *priv, struct sk_buff *skb, int method)
 							return -1;
 						}
 
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(7, 1, 0)
+						pMagic = (__be16 *)((unsigned char *)tag + sizeof(*tag));
+#else
 						pMagic = (__be16 *)tag->tag_data;
+#endif
 						if (ntohs(*pMagic) != MAGIC_CODE) {
 							DEBUG_ERR("Can't find MAGIC_CODE in %s packet!\n",
 								(ph->code == PADO_CODE ? "PADO" : "PADS"));
 							return -1;
 						}
 
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(7, 1, 0)
+						memcpy(skb->data, (unsigned char *)tag + sizeof(*tag) + MAGIC_CODE_LEN, ETH_ALEN);
+#else
 						memcpy(skb->data, tag->tag_data+MAGIC_CODE_LEN, ETH_ALEN);
+#endif
 
 						if (tagLen > MAGIC_CODE_LEN+RTL_RELAY_TAG_LEN)
 							offset = TAG_HDR_LEN;
diff --git a/os_dep/ioctl_cfg80211.c b/os_dep/ioctl_cfg80211.c.new
index 8cd3dd9..85d0dd7 100644
--- a/os_dep/ioctl_cfg80211.c
+++ b/os_dep/ioctl_cfg80211.c.new
@@ -1428,7 +1428,11 @@ exit:
 	return ret;
 }
 
+#if (LINUX_VERSION_CODE >= KERNEL_VERSION(7, 1, 0))
+static int cfg80211_rtw_add_key(struct wiphy *wiphy, struct wireless_dev *wdev,
+#else
 static int cfg80211_rtw_add_key(struct wiphy *wiphy, struct net_device *ndev,
+#endif
 #if (LINUX_VERSION_CODE >= KERNEL_VERSION(6,1,0)) || defined(COMPAT_KERNEL_RELEASE)
 				int link_id, u8 key_index, bool pairwise, const u8 *mac_addr,
 #elif (LINUX_VERSION_CODE >= KERNEL_VERSION(2,6,37)) || defined(COMPAT_KERNEL_RELEASE)
@@ -1438,6 +1442,9 @@ static int cfg80211_rtw_add_key(struct wiphy *wiphy, struct net_device *ndev,
 #endif	// (LINUX_VERSION_CODE >= KERNEL_VERSION(2,6,37))
 				struct key_params *params)
 {
+#if (LINUX_VERSION_CODE >= KERNEL_VERSION(7, 1, 0))
+	struct net_device *ndev = wdev->netdev;
+#endif
 	char *alg_name;
 	u32 param_len;
 	struct ieee_param *param = NULL;
@@ -3332,7 +3339,11 @@ void rtw_cfg80211_indicate_sta_assoc(_adapter *padapter, u8 *pmgmt_frame, uint f
 		sinfo.filled = STATION_INFO_ASSOC_REQ_IES;
 		sinfo.assoc_req_ies = pmgmt_frame + WLAN_HDR_A3_LEN + ie_offset;
 		sinfo.assoc_req_ies_len = frame_len - WLAN_HDR_A3_LEN - ie_offset;
+#if (LINUX_VERSION_CODE >= KERNEL_VERSION(7, 1, 0))
+		cfg80211_new_sta(ndev->ieee80211_ptr, GetAddr2Ptr(pmgmt_frame), &sinfo, GFP_ATOMIC);
+#else
 		cfg80211_new_sta(ndev, GetAddr2Ptr(pmgmt_frame), &sinfo, GFP_ATOMIC);
+#endif
 	}
 #else /* defined(RTW_USE_CFG80211_STA_EVENT) */
 	channel = pmlmeext->cur_channel;
@@ -3378,7 +3389,11 @@ void rtw_cfg80211_indicate_sta_disassoc(_adapter *padapter, unsigned char *da, u
 	DBG_871X(FUNC_ADPT_FMT"\n", FUNC_ADPT_ARG(padapter));
 
 #if defined(RTW_USE_CFG80211_STA_EVENT) || defined(COMPAT_KERNEL_RELEASE)
+#if (LINUX_VERSION_CODE >= KERNEL_VERSION(7, 1, 0))
+	cfg80211_del_sta(ndev->ieee80211_ptr, da, GFP_ATOMIC);
+#else
 	cfg80211_del_sta(ndev, da, GFP_ATOMIC);
+#endif
 #else /* defined(RTW_USE_CFG80211_STA_EVENT) */
 	channel = pmlmeext->cur_channel;
 	if (channel <= RTW_CH_MAX_2G_CHANNEL)
