--- a/SDL/SDLVulkanGraphicsContext.cpp	2023-09-12 22:17:24.229662403 +0200
+++ b/SDL/SDLVulkanGraphicsContext.cpp	2023-09-12 22:17:45.522135542 +0200
@@ -112,11 +112,6 @@
 				(void *)(intptr_t)sys_info.info.x11.window);
 #endif
 		break;
-#if defined(VK_USE_PLATFORM_WAYLAND_KHR)
-	case SDL_SYSWM_WAYLAND:
-		vulkan_->InitSurface(WINDOWSYSTEM_WAYLAND, (void*)sys_info.info.wl.display, (void *)sys_info.info.wl.surface);
-		break;
-#endif
 #if defined(VK_USE_PLATFORM_METAL_EXT)
 #if PPSSPP_PLATFORM(MAC)
 	case SDL_SYSWM_COCOA:
