diff --git a/spa/plugins/support/cpu-arm.c b/spa/plugins/support/cpu-arm.c
index d80b8ad..de07ebf 100644
--- a/spa/plugins/support/cpu-arm.c
+++ b/spa/plugins/support/cpu-arm.c
@@ -80,7 +80,7 @@ arm_init(struct impl *impl)
 	int arch;
 
 	if (!(cpuinfo = get_cpuinfo())) {
-		spa_log_warn(impl->log, NAME " %p: Can't read cpuinfo", impl);
+		spa_log_warn(impl->log, "%p: Can't read cpuinfo", impl);
 		return 1;
 	}
 
