From: sunshineinabox <aqemail@gmail.com>
Subject: [PATCH] remoteproc: qcom: pas: add SM4450 ADSP and WPSS

--- a/drivers/remoteproc/qcom_q6v5_pas.c
+++ b/drivers/remoteproc/qcom_q6v5_pas.c
@@ -1455,6 +1455,40 @@
 	.ssctl_id = 0x19,
 };
 
+static const struct qcom_pas_data sm4450_adsp_resource = {
+	.crash_reason_smem = 423,
+	.firmware_name = "adsp.mbn",
+	.pas_id = 1,
+	.minidump_id = 5,
+	.auto_boot = true,
+	.proxy_pd_names = (char*[]){
+		"lcx",
+		"lmx",
+		NULL
+	},
+	.load_state = "adsp",
+	.ssr_name = "lpass",
+	.sysmon_name = "adsp",
+	.ssctl_id = 0x14,
+};
+
+static const struct qcom_pas_data sm4450_wpss_resource = {
+	.crash_reason_smem = 626,
+	.firmware_name = "wpss.mbn",
+	.pas_id = 6,
+	.minidump_id = 4,
+	.auto_boot = true,
+	.proxy_pd_names = (char*[]){
+		"cx",
+		"mx",
+		NULL
+	},
+	.load_state = "wpss",
+	.ssr_name = "wpss",
+	.sysmon_name = "wpss",
+	.ssctl_id = 0x19,
+};
+
 static const struct qcom_pas_data shikra_cdsp_resource = {
 	.crash_reason_smem = 601,
 	.firmware_name = "cdsp.mbn",
@@ -1616,6 +1650,8 @@
 	{ .compatible = "qcom,shikra-cdsp-pas", .data = &shikra_cdsp_resource },
 	{ .compatible = "qcom,shikra-lpaicp-pas", .data = &shikra_lpaicp_resource },
 	{ .compatible = "qcom,shikra-mpss-pas", .data = &shikra_mpss_resource },
+	{ .compatible = "qcom,sm4450-adsp-pas", .data = &sm4450_adsp_resource },
+	{ .compatible = "qcom,sm4450-wpss-pas", .data = &sm4450_wpss_resource },
 	{ .compatible = "qcom,sm6115-adsp-pas", .data = &adsp_resource_init },
 	{ .compatible = "qcom,sm6115-cdsp-pas", .data = &cdsp_resource_init },
 	{ .compatible = "qcom,sm6115-mpss-pas", .data = &sc8180x_mpss_resource },
