From: sunshineinabox <aqemail@gmail.com>
Subject: [PATCH] soc: qcom: pd-mapper: add SM4450 domains


--- a/drivers/soc/qcom/qcom_pd_mapper.c
+++ b/drivers/soc/qcom/qcom_pd_mapper.c
@@ -496,6 +496,15 @@
 	NULL,
 };
 
+static const struct qcom_pdm_domain_data *sm4450_domains[] = {
+	&adsp_audio_pd,
+	&adsp_root_pd,
+	&adsp_charger_pd,
+	&adsp_sensor_pd,
+	&mpss_root_pd_gps,
+	NULL,
+};
+
 static const struct qcom_pdm_domain_data *sm6115_domains[] = {
 	&adsp_audio_pd,
 	&adsp_root_pd,
@@ -606,6 +614,7 @@
 	{ .compatible = "qcom,sdm670", .data = sdm670_domains, },
 	{ .compatible = "qcom,sdm845", .data = sdm845_domains, },
 	{ .compatible = "qcom,sm4250", .data = sm6115_domains, },
+	{ .compatible = "qcom,sm4450", .data = sm4450_domains, },
 	{ .compatible = "qcom,sm6115", .data = sm6115_domains, },
 	{ .compatible = "qcom,sm6350", .data = sm6350_domains, },
 	{ .compatible = "qcom,sm7150", .data = sm7150_domains, },
