From: sunshineinabox <aqemail@gmail.com>
Subject: [PATCH] clk: qcom: gpucc-sm4450: add hlos1_vote_gpu_smmu clock

--- a/drivers/clk/qcom/gpucc-sm4450.c
+++ b/drivers/clk/qcom/gpucc-sm4450.c
@@ -484,6 +484,19 @@
 	},
 };
 
+static struct clk_branch gpu_cc_hlos1_vote_gpu_smmu_clk = {
+	.halt_reg = 0x7000,
+	.halt_check = BRANCH_HALT_VOTED,
+	.clkr = {
+		.enable_reg = 0x7000,
+		.enable_mask = BIT(0),
+		.hw.init = &(const struct clk_init_data) {
+			.name = "gpu_cc_hlos1_vote_gpu_smmu_clk",
+			.ops = &clk_branch2_ops,
+		},
+	},
+};
+
 static struct clk_branch gpu_cc_gx_cxo_clk = {
 	.halt_reg = 0x90b8,
 	.halt_check = BRANCH_HALT,
@@ -717,6 +730,7 @@
 	[GPU_CC_HUB_AON_CLK] = &gpu_cc_hub_aon_clk.clkr,
 	[GPU_CC_HUB_CLK_SRC] = &gpu_cc_hub_clk_src.clkr,
 	[GPU_CC_HUB_CX_INT_CLK] = &gpu_cc_hub_cx_int_clk.clkr,
+	[GPU_CC_HLOS1_VOTE_GPU_SMMU_CLK] = &gpu_cc_hlos1_vote_gpu_smmu_clk.clkr,
 	[GPU_CC_HUB_CX_INT_DIV_CLK_SRC] = &gpu_cc_hub_cx_int_div_clk_src.clkr,
 	[GPU_CC_MEMNOC_GFX_CLK] = &gpu_cc_memnoc_gfx_clk.clkr,
 	[GPU_CC_MND1X_0_GFX3D_CLK] = &gpu_cc_mnd1x_0_gfx3d_clk.clkr,
--- a/include/dt-bindings/clock/qcom,sm4450-gpucc.h
+++ b/include/dt-bindings/clock/qcom,sm4450-gpucc.h
@@ -41,6 +41,7 @@
 #define GPU_CC_SLEEP_CLK					31
 #define GPU_CC_XO_CLK_SRC					32
 #define GPU_CC_XO_DIV_CLK_SRC					33
+#define GPU_CC_HLOS1_VOTE_GPU_SMMU_CLK			34
 
 /* GPU_CC power domains */
 #define GPU_CC_CX_GDSC						0
