Subject: [PATCH] power: supply: qcom_battmgr: report the USB adapter type

---
--- a/drivers/power/supply/qcom_battmgr.c
+++ b/drivers/power/supply/qcom_battmgr.c
@@ -974,7 +974,7 @@
 	[POWER_SUPPLY_PROP_CURRENT_NOW] = USB_CURR_NOW,
 	[POWER_SUPPLY_PROP_CURRENT_MAX] = USB_CURR_MAX,
 	[POWER_SUPPLY_PROP_INPUT_CURRENT_LIMIT] = USB_INPUT_CURR_LIMIT,
-	[POWER_SUPPLY_PROP_USB_TYPE] = USB_TYPE,
+	[POWER_SUPPLY_PROP_USB_TYPE] = USB_ADAP_TYPE,
 };
 
 static int qcom_battmgr_usb_sm8350_update(struct qcom_battmgr *battmgr,
@@ -1543,8 +1543,11 @@
 		case USB_INPUT_CURR_LIMIT:
 			battmgr->usb.current_limit = le32_to_cpu(resp->intval.value);
 			break;
-		case USB_TYPE:
-			battmgr->usb.usb_type = le32_to_cpu(resp->intval.value);
+		case USB_ADAP_TYPE:
+			val = le32_to_cpu(resp->intval.value);
+			if (val > POWER_SUPPLY_USB_TYPE_APPLE_BRICK_ID)
+				val = POWER_SUPPLY_USB_TYPE_UNKNOWN;
+			battmgr->usb.usb_type = val;
 			break;
 		default:
 			dev_warn(battmgr->dev, "unknown property %#x\n", property);
