diff --git a/src/main.cpp b/src/main.cpp
index 62b8289..bc596b7 100644
--- a/src/main.cpp
+++ b/src/main.cpp
@@ -61,6 +61,7 @@ bool kickstart_rom = 1;
 struct gui_info gui_data;
 
 TCHAR optionsfile[256];
+TCHAR core[32];
 
 void my_trim(TCHAR *s)
 {
@@ -455,6 +456,11 @@ static void parse_cmdline(int argc, TCHAR **argv)
 			target_cfgfile_load(&currprefs, txt, -1, 0);
 			xfree(txt);
 		}
+		else if (_tcsncmp(argv[i], _T("-core="), 6) == 0) {
+		  TCHAR *txt = parsetextpath(argv[i] + 6);
+		  _tcscpy(core, txt);
+		  xfree(txt);
+		}
 		else if (_tcsncmp(argv[i], _T("-statefile="), 11) == 0) {
 			TCHAR *txt = parsetextpath(argv[i] + 11);
 			savestate_state = STATE_DORESTORE;
@@ -646,6 +652,7 @@ static int real_main2(int argc, TCHAR **argv)
 	};
 #endif
 
+	_tcscpy(core, _T(""));
 	keyboard_settrans();
 
 	if (restart_config[0]) {
@@ -676,6 +683,20 @@ static int real_main2(int argc, TCHAR **argv)
 
 	changed_prefs = currprefs;
 
+	if(_tcsncmp(core, _T("A500"), 4) == 0) {
+	  bip_a500(&currprefs, -1);
+	} else if(_tcsncmp(core, _T("A500P"), 5) == 0) {
+	  bip_a500plus(&currprefs, -1);
+	} else if(_tcsncmp(core, _T("A1200"), 5) == 0) {
+	  bip_a1200(&currprefs, -1);
+	} else if(_tcsncmp(core, _T("A2000"), 5) == 0) {
+	  bip_a2000(&currprefs, -1);
+	} else if(_tcsncmp(core, _T("A4000"), 5) == 0) {
+	  bip_a4000(&currprefs, -1);
+	} else if(_tcsncmp(core, _T("CD32"), 4) == 0) {
+	  bip_cd32(&currprefs, -1);
+	}
+
 	no_gui = !currprefs.start_gui;
 	if (restart_program == 2)
 		no_gui = 1;
