diff --git a/Makefile b/Makefile
--- a/Makefile
+++ b/Makefile
@@ -1506,6 +1511,7 @@ ifneq ($(dtstree),)
 
 %.dtb: include/config/kernel.release scripts_dtc
 	$(Q)$(MAKE) $(build)=$(dtstree) $(dtstree)/$@
+	$(Q)-cp -f $(dtstree)/$@ arch/$(SRCARCH)/boot/
 
 %.dtbo: include/config/kernel.release scripts_dtc
 	$(Q)$(MAKE) $(build)=$(dtstree) $(dtstree)/$@
@@ -2102,7 +2108,7 @@ endif # config-build
 endif # mixed-build
 endif # need-sub-make
 
-ifdef CONFIG_AMLOGIC_DRIVER
+ifdef CONFIG_AMLOGIC_DRIVER_GIT
 PHONY += FORCE
 FORCE:
 	$(Q)-cp $(srctree)/scripts/amlogic/commit-msg $(srctree)/.git/hooks/
