--- a/drivers/amlogic/Makefile	2023-06-09 02:20:40.946020445 +0200
+++ b/drivers/amlogic/Makefile	2023-06-09 02:20:46.974152815 +0200
@@ -72,9 +72,9 @@
 obj-$(CONFIG_AMLOGIC_MCLK_GPIO) += gpio/
 obj-$(CONFIG_AMLOGIC_TEE2) += tee_driver/
 obj-y   += media_modules/
-obj-y   += isp_module/
+#obj-y   += isp_module/
 obj-y	+= adla/
-obj-y   += amlisp/
+#obj-y   += amlisp/
 
 KBUILD_CFLAGS_MODULE += $(GKI_EXT_MODULE_PREDEFINE)
 
