diff --git a/drivers/i2c/busses/i2c-meson.c b/drivers/i2c/busses/i2c-meson.c
index 0133170f5..7b59ac86a 100644
--- a/drivers/i2c/busses/i2c-meson.c
+++ b/drivers/i2c/busses/i2c-meson.c
@@ -273,7 +273,7 @@ static void meson_i2c_set_clk_div(struct meson_i2c *i2c, unsigned int freq)
 }
 #endif
 
-static void meson_i2c_get_data(struct meson_i2c *i2c, char *buf, int len)
+static void meson_i2c_get_data(struct meson_i2c *i2c, u8 *buf, int len)
 {
 	u32 rdata0, rdata1;
 	int i;
@@ -291,7 +291,7 @@ static void meson_i2c_get_data(struct meson_i2c *i2c, char *buf, int len)
 		*buf++ = (rdata1 >> (i - 4) * 8) & 0xff;
 }
 
-static void meson_i2c_put_data(struct meson_i2c *i2c, char *buf, int len)
+static void meson_i2c_put_data(struct meson_i2c *i2c, u8 *buf, int len)
 {
 	u32 wdata0 = 0, wdata1 = 0;
 	int i;
