From a1a1e953f7b7a103f5445e107d9ec649c81b62b2 Mon Sep 17 00:00:00 2001
From: chewitt <christianshewitt@gmail.com>
Date: Mon, 1 Jun 2020 08:16:58 +0000
Subject: [PATCH 142/152] DEBUG: s/dev_dbg/dev_err/g drivers/net/phy/realtek.c

---
 drivers/net/phy/realtek.c | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/drivers/net/phy/realtek.c b/drivers/net/phy/realtek.c
index 2d99e9de6ee1..9864d31f41b8 100644
--- a/drivers/net/phy/realtek.c
+++ b/drivers/net/phy/realtek.c
@@ -210,11 +210,11 @@ static int rtl8211f_config_init(struct phy_device *phydev)
 		dev_err(dev, "Failed to update the TX delay register\n");
 		return ret;
 	} else if (ret) {
-		dev_dbg(dev,
+		dev_err(dev,
 			"%s 2ns TX delay (and changing the value from pin-strapping RXD1 or the bootloader)\n",
 			val_txdly ? "Enabling" : "Disabling");
 	} else {
-		dev_dbg(dev,
+		dev_err(dev,
 			"2ns TX delay was already %s (by pin-strapping RXD1 or bootloader configuration)\n",
 			val_txdly ? "enabled" : "disabled");
 	}
@@ -225,11 +225,11 @@ static int rtl8211f_config_init(struct phy_device *phydev)
 		dev_err(dev, "Failed to update the RX delay register\n");
 		return ret;
 	} else if (ret) {
-		dev_dbg(dev,
+		dev_err(dev,
 			"%s 2ns RX delay (and changing the value from pin-strapping RXD0 or the bootloader)\n",
 			val_rxdly ? "Enabling" : "Disabling");
 	} else {
-		dev_dbg(dev,
+		dev_err(dev,
 			"2ns RX delay was already %s (by pin-strapping RXD0 or bootloader configuration)\n",
 			val_rxdly ? "enabled" : "disabled");
 	}
-- 
2.17.1

