diff --git a/src/path.cpp b/src/path.cpp
index 5645c29..1bc1157 100644
--- a/src/path.cpp
+++ b/src/path.cpp
@@ -13,8 +13,6 @@ static void system_path_internal(char* buffer, size_t len)
     
     if (!path_ends_with_slash(buffer))
         strlcat(buffer, path_default_slash(), len);
-
-    strlcat(buffer, "neocd", len);
 }
 
 std::string path_replace_filename(const char* path, const char* new_filename)
