diff --git a/Makefile.libretro b/Makefile.libretro
index deb4eb6981..66397cc7d6 100644
--- a/Makefile.libretro
+++ b/Makefile.libretro
@@ -213,6 +213,33 @@ else ifeq ($(platform), rpi2)
 	CFLAGS += -marm -mcpu=cortex-a7 -mfpu=neon-vfpv4 -mfloat-abi=hard -funsafe-math-optimizations
 	CFLAGS += -fomit-frame-pointer -fstrict-aliasing -ffast-math
 
+# Raspberry Pi 3 (64 bit)
+else ifeq ($(platform), rpi3_64)
+	TARGET := $(TARGET_NAME)_libretro.so
+	fpic := -fPIC
+	SHARED := -shared -Wl,--version-script=link.T -Wl,-no-undefined
+	CFLAGS += -DARM
+	CFLAGS += -mcpu=cortex-a53 -mtune=cortex-a53
+	CFLAGS += -fomit-frame-pointer -fstrict-aliasing -ffast-math -funsafe-math-optimizations
+
+# Raspberry Pi 4 (64 bit)
+else ifeq ($(platform), rpi4)
+	TARGET := $(TARGET_NAME)_libretro.so
+	fpic := -fPIC
+	SHARED := -shared -Wl,--version-script=link.T -Wl,-no-undefined
+	CFLAGS += -DARM
+	CFLAGS += -mcpu=cortex-a72 -mtune=cortex-a72
+	CFLAGS += -fomit-frame-pointer -fstrict-aliasing -ffast-math -funsafe-math-optimizations
+
+# Raspberry Pi 5 (64 bit)
+else ifeq ($(platform), rpi5)
+	TARGET := $(TARGET_NAME)_libretro.so
+	fpic := -fPIC
+	SHARED := -shared -Wl,--version-script=link.T -Wl,-no-undefined
+	CFLAGS += -DARM
+	CFLAGS += -mcpu=cortex-a76 -mtune=cortex-a76
+	CFLAGS += -fomit-frame-pointer -fstrict-aliasing -ffast-math -funsafe-math-optimizations
+
 # Lightweight PS3 Homebrew SDK
 else ifeq ($(platform), psl1ght)
 	TARGET := $(TARGET_NAME)_libretro_$(platform).a