diff --git a/meson.build b/meson.build
index 8db4dd4..fee87b0 100644
--- a/meson.build
+++ b/meson.build
@@ -370,35 +370,19 @@ zlib_dep = dependency(
 # SpeexDSP
 # ~~~~~~~~
 # Default to the system library
+speexdsp_summary_msg = 'system library'
 speexdsp_dep = dependency(
     'speexdsp',
     allow_fallback: false,
     static: ('speexdsp' in static_libs_list or prefers_static_libs),
 )
 
-# The system has SpeexDSP, so test its floating-point handling
-if speexdsp_dep.found()
-    system_speexdsp_test = cxx.run(
-        files('contrib/check-speexdsp/test_speexdsp_float_api.cpp'),
-        dependencies: speexdsp_dep,
-        name: 'SpeexDSP system library has reliable floating-point API',
-    )
-    is_system_speexdsp_reliable = (
-        system_speexdsp_test.compiled()
-        and system_speexdsp_test.returncode() == 0
-    )
-    if is_system_speexdsp_reliable
-        speexdsp_summary_msg = 'system library'
-    endif
-endif
-
 # The system doesn't have SpeexDSP or it's unreiable, so use the wrap
-if not speexdsp_dep.found() or not is_system_speexdsp_reliable
+if not speexdsp_dep.found()
     speexdsp_dep = subproject(
         'speexdsp',
         default_options: default_wrap_options,
     ).get_variable('speexdsp_dep')
-    speexdsp_summary_msg = 'built-in'
 endif
 summary('SpeexDSP provider', speexdsp_summary_msg)
 
