diff --git a/meson.build b/meson.build
--- a/meson.build
+++ b/meson.build
@@ -1084,4 +1084,5 @@ subdir('src/misc')
 subdir('src/misc')
 subdir('src/libs/zmbv')
 subdir('src/libs/tal-chorus')
+subdir('src/libs/riffcpp')
 subdir('src/libs/glad')

diff --git a/src/libs/riffcpp/meson.build b/src/libs/riffcpp/meson.build
--- a/src/libs/riffcpp/meson.build
+++ b/src/libs/riffcpp/meson.build
@@ -1,7 +1,7 @@
 libriffcpp_sources = files(
     'membuffer.cpp',
     'riffcpp.cpp',
-
+)

 libriffcpp = static_library('riffcpp', libriffcpp_sources)

diff --git a/src/hardware/meson.build b/src/hardware/meson.build
--- a/src/hardware/meson.build
+++ b/src/hardware/meson.build
@@ -88,7 +88,7 @@ libhardware = static_library(
         libloguru_dep,
         libnuked_dep,
         libresidfp_dep,
-        libriffcpp,
+        libriffcpp_dep,
         libslirp_dep,
         libym7128bemu_dep,
         libzmbv_dep,
