diff --git a/meson.build b/meson.build
--- a/meson.build
+++ b/meson.build
@@ -362,6 +362,9 @@ git_hash = run_command(

 conf_data = configuration_data()
 conf_data.set('version', meson.project_version())
+version_parts = meson.project_version().split('.')
+version_short = version_parts.length() >= 2 ? version_parts[0] + '.' + version_parts[1] : meson.project_version()
+conf_data.set('version_short', version_short)
 conf_data.set('project_name', meson.project_name())
 conf_data.set('git_hash', git_hash)
 conf_data.set_quoted(

diff --git a/src/dosbox_config.h.in b/src/dosbox_config.h.in
--- a/src/dosbox_config.h.in
+++ b/src/dosbox_config.h.in
@@ -14,6 +14,9 @@
 // 0.81.1-alpha)
 #define DOSBOX_VERSION "@version@"

+// Current DOSBox Staging version's major.minor component only (e.g., 0.83)
+#define DOSBOX_VERSION_SHORT "@version_short@"
+
 // Git hash of the build
 #define BUILD_GIT_HASH "@git_hash@"
