diff --git a/Makefile b/Makefile
index 65ece36..0918b63 100644
--- a/Makefile
+++ b/Makefile
@@ -457,7 +457,13 @@ else ifneq (,$(findstring rpi,$(platform)))
    ifneq (,$(findstring rpi2, $(platform)))
 	CCOMFLAGS += -marm -mcpu=cortex-a7 -mfpu=neon-vfpv4 -mfloat-abi=hard
    else ifneq (,$(findstring rpi3, $(platform)))
-	CCOMFLAGS += -marm -mcpu=cortex-a53 -mfpu=neon-fp-armv8 -mfloat-abi=hard
+	   ifneq (,$(findstring rpi3_64,$(platform)))
+         CCOMFLAGS += -mcpu=cortex-a53 -mtune=cortex-a53
+      else
+         CCOMFLAGS += -marm -mcpu=cortex-a53 -mfpu=neon-fp-armv8 -mfloat-abi=hard
+      endif
+   else ifneq (,$(findstring rpi4,$(platform)))
+         CCOMFLAGS += -mcpu=cortex-a72 -mtune=cortex-a72
    endif
 
 # ARM
