diff --git a/CMakeLists.txt b/CMakeLists.txt
index 1aa93c9..8c3ef8e 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -80,16 +80,14 @@ set(BUILD_SHARED_LIBS_backup "$CACHE{BUILD_SHARED_LIBS}")
 set(BUILD_SHARED_LIBS OFF CACHE INTERNAL "" FORCE)
 
 # libressl
-if (NOT NO_SYSTEM_DEPENDENCIES)
-    list(APPEND CMAKE_MODULE_PATH "${CMAKE_CURRENT_SOURCE_DIR}/cmake")
-    find_package(OpenSSL)
-    find_package(ZLIB)
-    find_package(PNG)
-endif()
+list(APPEND CMAKE_MODULE_PATH "${CMAKE_CURRENT_SOURCE_DIR}/cmake")
+find_package(OpenSSL)
+find_package(ZLIB)
+find_package(PNG)
 
 list(APPEND CMAKE_MODULE_PATH "${CMAKE_CURRENT_SOURCE_DIR}/cmake/cmake-bundle_static_library")
 
-if (NO_SYSTEM_DEPENDENCIES OR NOT OpenSSL_FOUND)
+if (NOT OpenSSL_FOUND)
 
     if (USE_WOLFSSL)
         set (WOLFSSL_EXAMPLES "no")
@@ -139,7 +137,7 @@ else()
 endif()
 
 # zlib
-if (NO_SYSTEM_DEPENDENCIES OR NOT ZLIB_FOUND)
+if (NOT ZLIB_FOUND)
     set(ZLIB_ENABLE_TESTS OFF CACHE BOOL "Build test binaries" FORCE)
     set(ZLIB_COMPAT ON CACHE BOOL "Compile with zlib compatible API" FORCE)
     set(SKIP_INSTALL_ALL TRUE)
@@ -163,7 +161,7 @@ if (NO_SYSTEM_DEPENDENCIES OR NOT ZLIB_FOUND)
 endif()
 
 # libpng
-if (NO_SYSTEM_DEPENDENCIES OR NOT PNG_FOUND)
+if (NOT PNG_FOUND)
     set(PNG_SHARED OFF CACHE BOOL "Build libpng as a shared library" FORCE)
     set(PNG_TESTS OFF CACHE BOOL "Build the libpng tests" FORCE)
     set(PNG_TOOLS OFF CACHE BOOL "" FORCE)
diff --git a/core/CMakeLists.txt b/core/CMakeLists.txt
index 1508239..bf3914a 100644
--- a/core/CMakeLists.txt
+++ b/core/CMakeLists.txt
@@ -29,9 +29,10 @@ FetchContent_Declare(cmake_git_version_tracking
   GIT_TAG 6c0cb87edd029ddfb403a8e24577c144a03605a6
 )
 
-FetchContent_MakeAvailable(cmake_git_version_tracking)
+FetchContent_Populate(cmake_git_version_tracking)
 
 add_library(core OBJECT ${COMMON_SRC} ${BGDRTM_SRC})
+target_include_directories(core PRIVATE "${cmake_git_version_tracking_SOURCE_DIR}/")
 
 if(CMAKE_C_COMPILER_ID MATCHES "MSVC")
     set_source_files_properties("common/allocator.c" PROPERTIES COMPILE_OPTIONS "/experimental:c11atomics")
@@ -65,5 +66,5 @@ target_link_libraries(core
     ${OPENSSL_CRYPTO_LIBRARY}
     ZLIB::ZLIB
     SDL::SDL
-    cmake_git_version_tracking)
+    )
 
diff --git a/core/common/bgd_version.c b/core/common/bgd_version.c
index 24bdd6e..26e6ba1 100644
--- a/core/common/bgd_version.c
+++ b/core/common/bgd_version.c
@@ -7,7 +7,7 @@ const char* bgd_getversion()
 {
     if (*buffer==0)
     {
-        snprintf(buffer, sizeof(buffer), "1.0.0-BennuGD_libretro-%s-%s", git_Branch(), git_CommitSHA1() );
+        snprintf(buffer, sizeof(buffer), "1.0.0-BennuGD_libretro-batocera");
     }
 
     return buffer;
