diff --git a/backends/graphics/openglsdl/openglsdl-graphics.cpp b/backends/graphics/openglsdl/openglsdl-graphics.cpp
index 0d140ee..3062f48 100644
--- a/backends/graphics/openglsdl/openglsdl-graphics.cpp
+++ b/backends/graphics/openglsdl/openglsdl-graphics.cpp
@@ -506,19 +506,23 @@ bool OpenGLSdlGraphicsManager::notifyEvent(const Common::Event &event) {
 			if (event.kbd.keycode == Common::KEYCODE_s) {
 				// Alt-s creates a screenshot
 				Common::String filename;
+				char path[2048];
 
 				for (int n = 0;; n++) {
 					SDL_RWops *file;
+					strcpy(path, ConfMan.get("screenshotspath").c_str());
 
-					filename = Common::String::format("scummvm%05d.bmp", n);
-					file = SDL_RWFromFile(filename.c_str(), "r");
+					filename = Common::String::format("/scummvm%05d.bmp", n);
+					strcat(path, filename.c_str());
+
+					file = SDL_RWFromFile(path, "r");
 					if (!file)
 						break;
 					SDL_RWclose(file);
 				}
 
-				saveScreenshot(filename.c_str());
-				debug("Saved screenshot '%s'", filename.c_str());
+				saveScreenshot(path);
+				debug("Saved screenshot '%s'", path);
 
 				return true;
 			}
diff --git a/backends/graphics/surfacesdl/surfacesdl-graphics.cpp b/backends/graphics/surfacesdl/surfacesdl-graphics.cpp
index 2b9d3aa..24c0eae 100644
--- a/backends/graphics/surfacesdl/surfacesdl-graphics.cpp
+++ b/backends/graphics/surfacesdl/surfacesdl-graphics.cpp
@@ -2329,19 +2329,20 @@ bool SurfaceSdlGraphicsManager::notifyEvent(const Common::Event &event) {
 
 		// Alt-S: Create a screenshot
 		if (event.kbd.hasFlags(Common::KBD_ALT) && event.kbd.keycode == 's') {
-			char filename[20];
-
+			char filename[80];
+			char path[2048];
 			for (int n = 0;; n++) {
 				SDL_RWops *file;
-
-				sprintf(filename, "scummvm%05d.bmp", n);
-				file = SDL_RWFromFile(filename, "r");
+				strcpy(path, ConfMan.get("screenshotspath").c_str());
+				sprintf(filename, "/scummvm%05d.bmp", n);
+				strcat(path, filename);
+				file = SDL_RWFromFile(path, "r");
 				if (!file)
 					break;
 				SDL_RWclose(file);
 			}
-			if (saveScreenshot(filename))
-				debug("Saved screenshot '%s'", filename);
+			if (saveScreenshot(path))
+				debug("Saved screenshot '%s'", path);
 			else
 				warning("Could not save screenshot");
 			return true;
diff --git a/base/commandLine.cpp b/base/commandLine.cpp
index 285fd63..b3fded7 100644
--- a/base/commandLine.cpp
+++ b/base/commandLine.cpp
@@ -108,6 +108,7 @@ static const char HELP_STRING[] =
 	"                           pce, segacd, wii, windows)\n"
 	"  --savepath=PATH          Path to where saved games are stored\n"
 	"  --extrapath=PATH         Extra path to additional game data\n"
+	"  --screenshotspath=PATH   Path to where screenshots are saved\n"
 	"  --soundfont=FILE         Select the SoundFont for MIDI playback (only\n"
 	"                           supported by some MIDI drivers)\n"
 	"  --multi-midi             Enable combination AdLib and native MIDI\n"
@@ -203,6 +204,7 @@ void registerDefaults() {
 
 	ConfMan.registerDefault("enable_unsupported_game_warning", true);
 
+	ConfMan.registerDefault("screenshotspath", ".");
 	// Game specific
 	ConfMan.registerDefault("path", "");
 	ConfMan.registerDefault("platform", Common::kPlatformDOS);
@@ -544,6 +546,15 @@ Common::String parseCommandLine(Common::StringMap &settings, int argc, const cha
 				}
 			END_OPTION
 
+			DO_LONG_OPTION("screenshotspath")
+				Common::FSNode path(option);
+				if (!path.exists()) {
+					usage("Non-existent screenshots path '%s'", option);
+				} else if (!path.isWritable()) {
+					usage("Non-writable screenshots path '%s'", option);
+				}
+			END_OPTION
+
 			DO_LONG_OPTION_INT("talkspeed")
 			END_OPTION
 
