diff --git a/arch/arm/boot/dts/rk3288-miqi.dts b/arch/arm/boot/dts/rk3288-miqi.dts
index 8a5b92b..1e9aa1b 100644
--- a/arch/arm/boot/dts/rk3288-miqi.dts
+++ b/arch/arm/boot/dts/rk3288-miqi.dts
@@ -269,6 +284,37 @@
 	dynamic-power = <396>;
 };
 
+&display_subsystem {
+	status = "okay";
+	route {
+		route_hdmi: route-hdmi {
+			status = "okay";
+			connect = <&vopb_out_hdmi>;
+		};
+	};
+};
+
+&dmc {
+	upthreshold = <55>;
+	downdifferential = <10>;
+	rockchip,ddr_timing = <&ddr_timing>;
+	system-status-freq = <
+		/*system status		freq(KHz)*/
+		SYS_STATUS_NORMAL	396000
+		SYS_STATUS_REBOOT	396000
+		SYS_STATUS_SUSPEND	192000
+		SYS_STATUS_VIDEO_1080P	300000
+		SYS_STATUS_VIDEO_4K	396000
+		SYS_STATUS_VIDEO_4K_10B	528000
+		SYS_STATUS_PERFORMANCE	528000
+		SYS_STATUS_BOOST	396000
+		SYS_STATUS_DUALVIEW	396000
+		SYS_STATUS_ISP		396000
+	>;
+	auto-min-freq = <396000>;
+	auto-freq-en = <0>;
+};
+
 &emmc {
 	bus-width = <8>;
 	cap-mmc-highspeed;
