diff --git a/arch/arm/boot/dts/rk3288-miqi.dts b/arch/arm/boot/dts/rk3288-miqi.dts
index 8a5b92b..1e9aa1b 100644
--- a/arch/arm/boot/dts/rk3288-miqi.dts
+++ b/arch/arm/boot/dts/rk3288-miqi.dts
@@ -45,12 +45,15 @@
 #include "rk3288.dtsi"
 #include "rk3288cg-opp.dtsi"
 
+#include <dt-bindings/soc/rockchip-system-status.h>
+#include "rk3288-dram-default-timing.dtsi"
+
 / {
 	model = "mqmaker MiQi";
 	compatible = "rockchip,rk3288-miqi", "rockchip,rk3288";
 
 	chosen {
-		bootargs = "earlyprintk=uart8250-32bit,0xff690000";
+		bootargs = "earlycon=uart8250,mmio32,0xff690000 console=ttyS2,115200n8 swiotlb=1 coherent_pool=1m usbcore.autosuspend=-1 net.ifnames=0";
 	};
 
 	cpuinfo {
@@ -155,22 +158,22 @@
 	};
 
 	board-leds {
-			compatible = "gpio-leds";
-			status = "okay";
+		compatible = "gpio-leds";
+		status = "okay";
 
-			work {
-					label = "System";
-					linux,default-trigger = "timer";
-					default-state = "on";
-					gpios = <&gpio7 2 GPIO_ACTIVE_HIGH>;
-			};
+		work {
+			label = "System";
+			linux,default-trigger = "timer";
+			default-state = "on";
+			gpios = <&gpio7 2 GPIO_ACTIVE_HIGH>;
+		};
 
-			fan {
-					label = "Fan";
-					linux,default-trigger = "heartbeat";
-					default-state = "on";
-					gpios = <&gpio0 18 GPIO_ACTIVE_HIGH>;
-			};
+		fan {
+			label = "Fan";
+			linux,default-trigger = "heartbeat";
+			default-state = "on";
+			gpios = <&gpio0 18 GPIO_ACTIVE_HIGH>;
+		};
 	};
 };
 
@@ -197,61 +200,73 @@
 		opp-hz = /bits/ 64 <1896000000>;
 		opp-microvolt = <1425000>;
 		turbo-mode;
+		status = "disabled";
 	};
 	opp-1920000000 {
 		opp-hz = /bits/ 64 <1920000000>;
 		opp-microvolt = <1425000>;
 		turbo-mode;
+		status = "disabled";
 	};
 	opp-1992000000 {
 		opp-hz = /bits/ 64 <1992000000>;
 		opp-microvolt = <1450000>;
 		turbo-mode;
+		status = "disabled";
 	};
 	opp-2016000000 {
 		opp-hz = /bits/ 64 <2016000000>;
 		opp-microvolt = <1475000>;
 		turbo-mode;
+		status = "disabled";
 	};
 	opp-2040000000 {
 		opp-hz = /bits/ 64 <2040000000>;
 		opp-microvolt = <1475000>;
 		turbo-mode;
+		status = "disabled";
 	};
 	opp-2064000000 {
 		opp-hz = /bits/ 64 <2064000000>;
 		opp-microvolt = <1475000>;
 		turbo-mode;
+		status = "disabled";
 	};
 	opp-2088000000 {
 		opp-hz = /bits/ 64 <2088000000>;
 		opp-microvolt = <1500000>;
 		turbo-mode;
+		status = "disabled";
 	};
 	opp-2112000000 {
 		opp-hz = /bits/ 64 <2112000000>;
 		opp-microvolt = <1500000>;
 		turbo-mode;
+		status = "disabled";
 	};
 	opp-2136000000 {
 		opp-hz = /bits/ 64 <2136000000>;
 		opp-microvolt = <1500000>;
 		turbo-mode;
+		status = "disabled";
 	};
 	opp-2160000000 {
 		opp-hz = /bits/ 64 <2160000000>;
 		opp-microvolt = <1500000>;
 		turbo-mode;
+		status = "disabled";
 	};
 	opp-2184000000 {
 		opp-hz = /bits/ 64 <2184000000>;
 		opp-microvolt = <1500000>;
 		turbo-mode;
+		status = "disabled";
 	};
 	opp-2208000000 {
 		opp-hz = /bits/ 64 <2208000000>;
 		opp-microvolt = <1500000>;
 		turbo-mode;
+		status = "disabled";
 	};
 };
 
@@ -269,6 +284,37 @@
 	dynamic-power = <396>;
 };
 
+&display_subsystem {
+	status = "okay";
+	route {
+		route_hdmi: route-hdmi {
+			status = "okay";
+			connect = <&vopb_out_hdmi>;
+		};
+	};
+};
+
+&dmc {
+	upthreshold = <55>;
+	downdifferential = <10>;
+	rockchip,ddr_timing = <&ddr_timing>;
+	system-status-freq = <
+		/*system status		freq(KHz)*/
+		SYS_STATUS_NORMAL	396000
+		SYS_STATUS_REBOOT	396000
+		SYS_STATUS_SUSPEND	192000
+		SYS_STATUS_VIDEO_1080P	300000
+		SYS_STATUS_VIDEO_4K	396000
+		SYS_STATUS_VIDEO_4K_10B	528000
+		SYS_STATUS_PERFORMANCE	528000
+		SYS_STATUS_BOOST	396000
+		SYS_STATUS_DUALVIEW	396000
+		SYS_STATUS_ISP		396000
+	>;
+	auto-min-freq = <396000>;
+	auto-freq-en = <0>;
+};
+
 &emmc {
 	bus-width = <8>;
 	cap-mmc-highspeed;
@@ -286,13 +332,17 @@
 	#size-cells = <0>;
 	#sound-dai-cells = <0>;
 	status = "okay";
+	pinctrl-0 = <&hdmi_cec>;
+	ddc-i2c-bus = <&i2c5>;
+	rockchip,phy-table =
+		<74250000 0x8009 0x0004 0x0272>,
+		<165000000 0x802b 0x0004 0x0209>,
+		<297000000 0x8039 0x0005 0x028d>,
+		<594000000 0x8039 0x0000 0x019d>,
+		<000000000 0x0000 0x0000 0x0000>;
 };
 
-&hdmi_in_vopb {
-	status = "disabled";
-};
-
-&hdmi_in_vopl {
+&i2c5 {
 	status = "okay";
 };
 
@@ -565,27 +615,16 @@ I2C
 	status = "okay";
 };
 
+&video_phy {
+	status = "okay";
+};
+
 &vopb {
 	status = "okay";
-		vopb_out: port {
-			vopb_out_edp: endpoint@1 {
-				reg = <1>;
-				remote-endpoint = <&edp_in_vopb>;
-			};
-		};
 };
 
 &vopl {
 	status = "okay";
-	vopl_out: port {
-		#address-cells = <1>;
-		#size-cells = <0>;
-
-		vopl_out_hdmi: endpoint@0 {
-			reg = <0>;
-			remote-endpoint = <&hdmi_in_vopl>;
-		};
-	};
 };
 
 &vopb_mmu {
@@ -596,6 +635,14 @@ I2C
 	status = "okay";
 };
 
+&vpu {
+	status = "okay";
+};
+
+&vpu_mmu {
+	status = "okay";
+};
+
 &vpu_service {
 	status = "okay";
 };
