diff --git a/core/stdclass.cpp b/core/stdclass.cpp
index 8f95dd8..e483363 100644
--- a/core/stdclass.cpp
+++ b/core/stdclass.cpp
@@ -32,8 +32,7 @@ void add_system_data_dir(const string& dir)
 
 string get_writable_data_path(const string& filename)
 {
-   extern char game_dir_no_slash[1024];
-   return std::string(game_dir_no_slash + 
+   return std::string("/userdata/saves/dreamcast/" + 
 #ifdef _WIN32
          std::string("\\")
 #else
