diff --git a/core/libretro/libretro.cpp b/core/libretro/libretro.cpp
index 4d2bb17..2799c9a 100644
--- a/core/libretro/libretro.cpp
+++ b/core/libretro/libretro.cpp
@@ -51,7 +51,7 @@ char save_dir[PATH_MAX];
 char eeprom_file[PATH_MAX];
 char nvmem_file[PATH_MAX];
 char nvmem_file2[PATH_MAX];		// AtomisWave
-bool boot_to_bios;
+bool boot_to_bios = true;
 
 static int astick_deadzone = 0;
 static int trigger_deadzone = 0;
@@ -1300,8 +1300,8 @@ bool retro_load_game(const struct retro_game_info *game)
    if (!(environ_cb(RETRO_ENVIRONMENT_GET_SYSTEM_DIRECTORY, &dir) && dir))
       dir = game_dir;
 
-   snprintf(game_dir, sizeof(game_dir), "%s%cdc%c", dir, slash, slash);
-   snprintf(game_dir_no_slash, sizeof(game_dir_no_slash), "%s%cdc", dir, slash);
+   snprintf(game_dir, sizeof(game_dir), "%s", dir);
+   snprintf(game_dir_no_slash, sizeof(game_dir_no_slash), "%s", dir);
 
    settings.dreamcast.cable = 3;
    update_variables(true);
@@ -1335,7 +1335,7 @@ bool retro_load_game(const struct retro_game_info *game)
    {
       char data_dir[1024];
 
-      snprintf(data_dir, sizeof(data_dir), "%s%s", game_dir, "data");
+      snprintf(data_dir, sizeof(data_dir), "%s", game_dir);
 
       printf("Creating dir: %s\n", data_dir);
       struct stat buf;
