diff --git a/src/Applewin.cpp b/src/Applewin.cpp
index 71a3fa0..069e330 100644
--- a/src/Applewin.cpp
+++ b/src/Applewin.cpp
@@ -922,7 +922,7 @@ void LoadAllConfigurations(const char *userSpecifiedFilename)
   if (home) {
     // Suppport old locations under HOME.
     configFiles.push_back(std::string(home) + "/linapple/linapple.conf");
-    configFiles.push_back(std::string(home) + "/.linapple/linapple.conf");
+    configFiles.push_back(std::string(home) + "/configs/linapple/linapple.conf");
   }
   configFiles.push_back(xdgConfigHome + "/linapple/linapple.conf");
 
