diff --git a/arch/arm64/boot/dts/qcom/qcs8550-ayn-common.dtsi b/arch/arm64/boot/dts/qcom/qcs8550-ayn-common.dtsi
index a02b084f1a..f7852ab3d7 100644
--- a/arch/arm64/boot/dts/qcom/qcs8550-ayn-common.dtsi
+++ b/arch/arm64/boot/dts/qcom/qcs8550-ayn-common.dtsi
@@ -96,6 +96,9 @@ pmic_glink_sbu: endpoint {
 
 	pwm_fan: pwm-fan {
 		compatible = "pwm-fan";
+		
+		#cooling-cells = <2>;
+		cooling-levels = <0 36 72 108 144 180 216 255>;
 
 		fan-supply = <&vdd_fan_5v0>;
 		pwms = <&pm8550_pwm 3 40000>;
