diff --git a/Makefile b/Makefile
index fca49a2667..2adea5c353 100644
--- a/Makefile
+++ b/Makefile
@@ -361,6 +361,22 @@ else ifeq ($(platform), rpi4_64)
    GL_LIB := -lGLESv2
    HAVE_CDROM = 0
 
+# Raspberry Pi 5 in 64bit mode
+else ifeq ($(platform), rpi5_64)
+   TARGET := $(TARGET_NAME)_libretro.so
+   fpic   := -fPIC
+   GREP = grep
+   SHARED := -shared -Wl,--no-undefined -Wl,--version-script=link.T
+   CFLAGS   += -O3 -DNDEBUG -mcpu=cortex-a76 -mtune=cortex-a76 -fsigned-char 
+   CXXFLAGS += -O3 -DNDEBUG -mcpu=cortex-a76 -mtune=cortex-a76 -fsigned-char
+   LDFLAGS += $(PTHREAD_FLAGS) -ldl -lrt
+   HAVE_LIGHTREC = 1
+   FLAGS += -DHAVE_SHM
+   GLES = 1
+   GLES3 = 1
+   GL_LIB := -lGLESv2
+   HAVE_CDROM = 0
+
 # Windows MSVC 2017 all architectures
 else ifneq (,$(findstring windows_msvc2017,$(platform)))
 
