diff --git a/src/rf5c68/rf5c68.h b/src/rf5c68/rf5c68.h
index 60480a4e89..0fd4852318 100644
--- a/src/rf5c68/rf5c68.h
+++ b/src/rf5c68/rf5c68.h
@@ -18,6 +18,7 @@ THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
 
 
 #include <vector>
+#include <cstdint>
 #include <string>
 
 class RF5C68
diff --git a/src/rf5c68/rf5c68.cpp b/src/rf5c68/rf5c68.cpp
index bccff94654..4b00aec8fd 100644
--- a/src/rf5c68/rf5c68.cpp
+++ b/src/rf5c68/rf5c68.cpp
@@ -16,7 +16,7 @@ THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
 #include <cstring>
 #include "rf5c68.h"
 #include "cpputil.h"
-
+#include <cstdint>
 
 
 inline void WordOp_Add(unsigned char *ptr,int value)
diff --git a/src/osdependent/filesys/filesys.h b/src/osdependent/filesys/filesys.h
index 35071d1123..0f64739a7a 100644
--- a/src/osdependent/filesys/filesys.h
+++ b/src/osdependent/filesys/filesys.h
@@ -6,6 +6,7 @@
 
 #include <fstream>
 #include <vector>
+#include <cstdint>
 #include <string>
 
 // When C++17's filesystem library is commonly available and stable, I'll change it to
diff --git a/src/ym2612/ym2612.h b/src/ym2612/ym2612.h
index 85342e2253..8c270e9d68 100644
--- a/src/ym2612/ym2612.h
+++ b/src/ym2612/ym2612.h
@@ -17,6 +17,7 @@ THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
 /* { */
 
 #include <vector>
+#include <cstdint>
 #include <string>
 
 
diff --git a/src/ym2612/ym2612wave.cpp b/src/ym2612/ym2612wave.cpp
index 08917abe77..47b3c0bb89 100644
--- a/src/ym2612/ym2612wave.cpp
+++ b/src/ym2612/ym2612wave.cpp
@@ -18,7 +18,7 @@ THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
 #include <cstdlib> // for std::abs
 
 #include "ym2612.h"
-
+#include <cstdint>
 
 
 inline void WordOp_Set(unsigned char *ptr,short value)
diff --git a/src/device/device.h b/src/device/device.h
index edae7f684f..444664b002 100644
--- a/src/device/device.h
+++ b/src/device/device.h
@@ -17,10 +17,10 @@ THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
 /* { */
 
 #include <vector>
+#include <cstdint>
 #include <string>
 
 
-
 class Device
 {
 protected:
diff --git a/src/device/device.cpp b/src/device/device.cpp
index 39bf5906cd..ac6579a8ed 100644
--- a/src/device/device.cpp
+++ b/src/device/device.cpp
@@ -17,7 +17,7 @@ THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
 #include "device.h"
 #include "vmbase.h"
 #include "cpputil.h"
-
+#include <cstdint>
 
 void Device::CommonState::PushState(std::vector <unsigned char> &data) const
 {
diff --git a/src/externals/d77/d77.cpp b/src/externals/d77/d77.cpp
index 0f21d15b27..752b7b4055 100644
--- a/src/externals/d77/d77.cpp
+++ b/src/externals/d77/d77.cpp
@@ -16,7 +16,7 @@ THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
 #include <set>
 #include <string.h>
 #include "d77.h"
-
+#include <cstdint>
 
 
 std::vector <std::string> D77File::QuickParser(const char str[])
