--- a/src/burner/libretro/libretro.cpp	2021-06-09 16:51:00.542768484 +0200
+++ b/src/burner/libretro/libretro.cpp	2021-06-09 16:51:06.966101576 +0200
@@ -277,10 +277,6 @@
 {
 	char *library_version = (char*)calloc(22, sizeof(char));
 
-#ifndef GIT_VERSION
-#define GIT_VERSION ""
-#endif
-
 	sprintf(library_version, "v%x.%x.%x.%02x %s", nBurnVer >> 20, (nBurnVer >> 16) & 0x0F, (nBurnVer >> 8) & 0xFF, nBurnVer & 0xFF, GIT_VERSION);
 
 	info->library_name = APP_TITLE;
--- a/src/burner/libretro/Makefile.common	2021-05-31 08:17:51.000000000 +0200
+++ b/src/burner/libretro/Makefile.common	2021-06-09 17:01:27.016078117 +0200
@@ -468,7 +468,4 @@
 endif
 
 # GIT HASH
-GIT_VERSION := " $(shell git rev-parse --short HEAD || echo unknown)"
-ifneq ($(GIT_VERSION)," unknown")
-	CXXFLAGS += -DGIT_VERSION=\"$(GIT_VERSION)\"
-endif
+CXXFLAGS += -DGIT_VERSION=\"$(LR_FBNEO_GIT_VERSION)\"
