diff --git a/src/burner/libretro/Makefile b/src/burner/libretro/Makefile
index 2bfd706..3c2cb1c 100644
--- a/src/burner/libretro/Makefile
+++ b/src/burner/libretro/Makefile
@@ -81,7 +81,10 @@ ifneq (,$(findstring unix,$(platform)))
 	ENDIANNESS_DEFINES := -DLSB_FIRST
 
 	# Raspberry Pi
-	ifneq (,$(findstring rpi2,$(platform)))
+	ifneq (,$(findstring rpi1,$(platform)))
+		PLATFORM_DEFINES := -marm -march=armv6j -mfpu=vfp -mfloat-abi=hard
+		USE_CYCLONE = 1
+	else ifneq (,$(findstring rpi2,$(platform)))
 		PLATFORM_DEFINES := -marm -mcpu=cortex-a7 -mfpu=neon-vfpv4 -mfloat-abi=hard
 		HAVE_NEON = 1
 		USE_CYCLONE = 1
