diff --git a/src/burner/libretro/Makefile b/src/burner/libretro/Makefile
index bae2404..58b5b8e 100644
--- a/src/burner/libretro/Makefile
+++ b/src/burner/libretro/Makefile
@@ -85,9 +85,13 @@ ifneq (,$(findstring unix,$(platform)))
 		HAVE_NEON = 1
 		USE_CYCLONE = 1
 	else ifneq (,$(findstring rpi3,$(platform)))
-		PLATFORM_DEFINES := -marm -mcpu=cortex-a53 -mfpu=neon-fp-armv8 -mfloat-abi=hard
-		HAVE_NEON = 1
-		USE_CYCLONE = 1
+		ifneq (,$(findstring rpi3_64,$(platform)))
+			PLATFORM_DEFINES := -march=armv8-a+crc+simd -mtune=cortex-a53
+		else
+			PLATFORM_DEFINES := -marm -mcpu=cortex-a53 -mfpu=neon-fp-armv8 -mfloat-abi=hard
+			HAVE_NEON = 1
+			USE_CYCLONE = 1
+		endif
 	else ifneq (,$(findstring rpi4_64,$(platform)))
 		PLATFORM_DEFINES := -march=armv8-a+crc+simd -mtune=cortex-a72
 	endif
