diff --git a/Makefile.libretro b/Makefile.libretro
index 9e9d345..0ca4e5d 100644
--- a/Makefile.libretro
+++ b/Makefile.libretro
@@ -162,6 +162,35 @@ else ifeq ($(platform), wincross64)
 	LDFLAGS += -static-libgcc -static-libstdc++ 
 	EXTRA_LDF := -lwinmm -Wl,--export-all-symbols
 
+# ARM
+else ifneq (,$(findstring armv,$(platform)))
+	TARGET := $(TARGET_NAME)_libretro.so
+	SHARED := -shared -Wl,--no-undefined -Wl,--version-script=link.T
+	fpic := -fPIC -fsigned-char
+	CFLAGS += -D_GNU_SOURCE=1
+	CC = gcc
+ifneq (,$(findstring cortexa8,$(platform)))
+	CFLAGS += -marm -mcpu=cortex-a8
+	ASFLAGS += -mcpu=cortex-a8
+else ifneq (,$(findstring cortexa9,$(platform)))
+	CFLAGS += -marm -mcpu=cortex-a9
+	ASFLAGS += -mcpu=cortex-a9
+endif
+	CFLAGS += -marm
+ifneq (,$(findstring neon,$(platform)))
+	CFLAGS += -mfpu=neon
+	ASFLAGS += -mfpu=neon
+	HAVE_NEON = 1
+endif
+ifneq (,$(findstring softfloat,$(platform)))
+	CFLAGS += -mfloat-abi=softfp
+	ASFLAGS += -mfloat-abi=softfp
+else ifneq (,$(findstring hardfloat,$(platform)))
+	CFLAGS += -mfloat-abi=hard
+	ASFLAGS += -mfloat-abi=hard
+endif
+	CFLAGS += -DARM
+
 # Windows MSVC 2017 all architectures
 else ifneq (,$(findstring windows_msvc2017,$(platform)))
 
