diff --git a/configure b/configure
index 65e4731..f1bbb70 100755
--- a/configure
+++ b/configure
@@ -1490,9 +1490,6 @@ for ac_option in $@; do
 			engine_disable ${engine_name}
 		done
 		;;
-	*)
-		option_error
-		;;
 	esac;
 done;
 
@@ -3288,17 +3285,6 @@ if test -n "$_host"; then
 			;;
 		raspberrypi)
 			_libcurlpath=$RPI_ROOT/usr/bin
-			# This is needed because the official cross compiler doesn't have multiarch enabled
-			# but Raspbian does.
-			# Be careful as it's the linker (LDFLAGS) which must know about sysroot.
-			# These are needed to build against Raspbian's libSDL.
-			append_var LDFLAGS "--sysroot=$RPI_ROOT"
-			append_var LDFLAGS "-B$RPI_ROOT/usr/lib/arm-linux-gnueabihf"
-			append_var LDFLAGS "-Xlinker --rpath-link=$RPI_ROOT/usr/lib/arm-linux-gnueabihf/pulseaudio"
-			append_var LDFLAGS "-Xlinker --rpath-link=$RPI_ROOT/usr/lib/arm-linux-gnueabihf"
-			append_var LDFLAGS "-Xlinker --rpath-link=$RPI_ROOT/lib/arm-linux-gnueabihf"
-			append_var CXXFLAGS "-isystem $RPI_ROOT/usr/include/arm-linux-gnueabihf"
-			append_var CXXFLAGS "-I$RPI_ROOT/usr/include"
 			# We prefer SDL2 on the Raspberry Pi: acceleration now depends on it
 			# since SDL2 manages dispmanx/GLES2 very well internally.
 			# SDL1 is bit-rotten on this platform.
