diff --git a/base/commandLine.cpp b/base/commandLine.cpp
index 40e853f..dc1c5fa 100644
--- a/base/commandLine.cpp
+++ b/base/commandLine.cpp
@@ -151,6 +151,7 @@ static const char HELP_STRING[] =
 	"                           pce, segacd, wii, windows)\n"
 	"  --savepath=PATH          Path to where saved games are stored\n"
 	"  --extrapath=PATH         Extra path to additional game data\n"
+	"  --screenshots=PATH       Path to where screenshots are saved\n"
 	"  --iconspath=PATH         Path to additional icons for the launcher grid view\n"
 	"  --soundfont=FILE         Select the SoundFont for MIDI playback (only\n"
 	"                           supported by some MIDI drivers)\n"
@@ -308,6 +309,8 @@ void registerDefaults() {

 	ConfMan.registerDefault("enable_unsupported_game_warning", true);

+	ConfMan.registerDefault("screenshotspath", ".");
+
 	// Game specific
 	ConfMan.registerDefault("path", "");
 	ConfMan.registerDefault("platform", Common::kPlatformDOS);
@@ -857,6 +860,15 @@ Common::String parseCommandLine(Common::StringMap &settings, int argc, const cha
 			DO_LONG_OPTION("md5-engine")
 			END_OPTION

+			DO_LONG_OPTION("screenshotspath")
+				Common::FSNode path(option);
+				if (!path.exists()) {
+					usage("Non-existent screenshots path '%s'", option);
+				} else if (!path.isWritable()) {
+					usage("Non-writable screenshots path '%s'", option);
+				}
+			END_OPTION
+
 			DO_LONG_OPTION_INT("talkspeed")
 			END_OPTION

