diff --git a/src/common/file_system.cpp b/src/common/file_system.cpp
index 6a24732..a287703 100644
--- a/src/common/file_system.cpp
+++ b/src/common/file_system.cpp
@@ -1179,6 +1179,8 @@ bool FileSystem::DeleteDirectory(const char* Path, bool Recursive)
 
 std::string GetProgramPath()
 {
+  return "/usr/share/duckstation";
+
   std::wstring buffer;
   buffer.resize(MAX_PATH);
 
@@ -1549,6 +1551,8 @@ bool DeleteDirectory(const char* Path, bool Recursive)
 
 std::string GetProgramPath()
 {
+  return "/usr/share/duckstation";
+
 #if defined(__linux__)
   static const char* exeFileName = "/proc/self/exe";
 
