diff --git a/src/duckstation-qt/qttranslations.cpp b/src/duckstation-qt/qttranslations.cpp
index 84719543ef..87ff34b789 100644
--- a/src/duckstation-qt/qttranslations.cpp
+++ b/src/duckstation-qt/qttranslations.cpp
@@ -73,7 +73,7 @@ void QtHost::InstallTranslator()
 
   // install the base qt translation first
 #ifndef __APPLE__
-  const QString base_dir = QStringLiteral("%1/translations").arg(qApp->applicationDirPath());
+  const QString base_dir = QStringLiteral("%1/translations").arg("/usr/share/duckstation");
 #else
   const QString base_dir = QStringLiteral("%1/../Resources/translations").arg(qApp->applicationDirPath());
 #endif
