diff --git a/src/duckstation-nogui/wayland_nogui_platform.cpp b/src/duckstation-nogui/wayland_nogui_platform.cpp
index c6d4ec21..bc5c9e66 100644
--- a/src/duckstation-nogui/wayland_nogui_platform.cpp
+++ b/src/duckstation-nogui/wayland_nogui_platform.cpp
@@ -137,7 +137,13 @@ bool WaylandNoGUIPlatform::CreatePlatformWindow(std::string title)
     if (m_toplevel_decoration)
       zxdg_toplevel_decoration_v1_set_mode(m_toplevel_decoration, ZXDG_TOPLEVEL_DECORATION_V1_MODE_SERVER_SIDE);
   }
-
+  
+  // Set fullscreen mode
+  if (m_xdg_toplevel && m_wl_output)
+  {
+    xdg_toplevel_set_fullscreen(m_xdg_toplevel, m_wl_output);
+  }
+  
   m_window_info.surface_width = static_cast<u32>(window_width);
   m_window_info.surface_height = static_cast<u32>(window_height);
   m_window_info.surface_scale = 1.0f;
@@ -234,6 +240,11 @@ void WaylandNoGUIPlatform::GlobalRegistryHandler(void* data, wl_registry* regist
     platform->m_decoration_manager = static_cast<zxdg_decoration_manager_v1*>(
       wl_registry_bind(platform->m_registry, id, &zxdg_decoration_manager_v1_interface, 1));
   }
+  else if (std::strcmp(interface, wl_output_interface.name) == 0)
+  {
+    platform->m_wl_output = static_cast<wl_output*>(
+      wl_registry_bind(platform->m_registry, id, &wl_output_interface, std::min(version, 3u)));
+  }
   else if (std::strcmp(interface, wl_seat_interface.name) == 0)
   {
     static const wl_seat_listener seat_listener = {&WaylandNoGUIPlatform::SeatCapabilities};
diff --git a/src/duckstation-nogui/wayland_nogui_platform.h b/src/duckstation-nogui/wayland_nogui_platform.h
index 25f8b371..40996468 100644
--- a/src/duckstation-nogui/wayland_nogui_platform.h
+++ b/src/duckstation-nogui/wayland_nogui_platform.h
@@ -95,6 +95,7 @@ private:
   xkb_context* m_xkb_context = nullptr;
   xkb_keymap* m_xkb_keymap = nullptr;
   xkb_state* m_xkb_state = nullptr;
+  wl_output* m_wl_output = nullptr;
 
   std::unordered_map<s32, std::string> m_key_map;
   std::mutex m_key_map_mutex;
