--- a/CMakeModules/FindEGL.cmake	2021-02-06 11:25:26.000000000 +0100
+++ b/CMakeModules/FindEGL.cmake	2021-02-07 22:01:29.516246541 +0100
@@ -118,6 +118,7 @@
 cmake_push_check_state(RESET)
 list(APPEND CMAKE_REQUIRED_LIBRARIES "${EGL_LIBRARY}")
 list(APPEND CMAKE_REQUIRED_INCLUDES "${EGL_INCLUDE_DIR}")
+list(APPEND CMAKE_REQUIRED_DEFINITIONS "-DEGL_NO_X11")
 
 check_cxx_source_compiles("
 #include <EGL/egl.h>
