diff --git a/src/duckstation-nogui/CMakeLists.txt b/src/duckstation-nogui/CMakeLists.txt
index 2615eed09e..0c2fb5a1d9 100644
--- a/src/duckstation-nogui/CMakeLists.txt
+++ b/src/duckstation-nogui/CMakeLists.txt
@@ -57,10 +57,6 @@ if(ENABLE_WAYLAND)
   find_package(Wayland REQUIRED Client)
   find_package(WaylandScanner REQUIRED)
   find_package(WaylandProtocols 1.15 REQUIRED)
-  find_package(X11 REQUIRED)
-  if (NOT X11_xkbcommon_FOUND)
-    message(FATAL_ERROR "XKBCommon is required.")
-  endif()
 
   target_compile_definitions(duckstation-nogui PRIVATE "NOGUI_PLATFORM_WAYLAND=1")
   target_sources(duckstation-nogui PRIVATE
@@ -83,6 +79,6 @@ if(ENABLE_WAYLAND)
 
   target_link_libraries(duckstation-nogui PRIVATE
     Wayland::Client
-    X11::xkbcommon
+    xkbcommon
   )
 endif()
