diff --git a/CMakeModules/DuckStationDependencies.cmake b/CMakeModules/DuckStationDependencies.cmake
index d1f0de4545..1325bd4c6e 100644
--- a/CMakeModules/DuckStationDependencies.cmake
+++ b/CMakeModules/DuckStationDependencies.cmake
@@ -1,5 +1,5 @@
 if(ENABLE_SDL2)
-  find_package(SDL2 2.30.2 REQUIRED)
+  find_package(SDL2 2.28 REQUIRED)
 endif()
 if(NOT WIN32 AND NOT ANDROID)
   # From PCSX2: On macOS, Mono.framework contains an ancient version of libpng. We don't want that.