diff --git a/src/duckstation-qt/translations/duckstation-qt_ja.ts b/src/duckstation-qt/translations/duckstation-qt_ja.ts
index aa7c6cc..26f9a8b 100644
--- a/src/duckstation-qt/translations/duckstation-qt_ja.ts
+++ b/src/duckstation-qt/translations/duckstation-qt_ja.ts
@@ -7530,7 +7530,7 @@ Please configure a supported controller from the list above.</source>
     <message>
         <location filename="../../core/system.cpp" line="2199"/>
         <source>Switched to sub-image %s (%u) in &apos;%s&apos;.</source>
-        <translation>&apos;%s&apos;のサブイメージ %s (%u) へ切り替えました。</translation>
+        <translation>&apos;%$3s&apos;のサブイメージ %$1s (%$2u) へ切り替えました。</translation>
     </message>
     <message>
         <location filename="../../core/system.cpp" line="2055"/>
