diff --git a/src/util/vulkan_pipeline.cpp b/src/util/vulkan_pipeline.cpp
index a0a68d3ce2..02c9930455 100644
--- a/src/util/vulkan_pipeline.cpp
+++ b/src/util/vulkan_pipeline.cpp
@@ -62,20 +62,9 @@ std::unique_ptr<GPUShader> VulkanDevice::CreateShaderFromSource(GPUShaderStage s
   shaderc::CompileOptions options;
   options.SetSourceLanguage(shaderc_source_language_glsl);
   options.SetTargetEnvironment(shaderc_target_env_vulkan, 0);
-
-  if (m_debug_device)
-  {
-    options.SetGenerateDebugInfo();
-    if (m_optional_extensions.vk_khr_shader_non_semantic_info)
-      options.SetEmitNonSemanticDebugInfo();
-
-    options.SetOptimizationLevel(shaderc_optimization_level_zero);
-  }
-  else
-  {
-    options.SetOptimizationLevel(shaderc_optimization_level_performance);
-  }
-
+  
+  options.SetOptimizationLevel(shaderc_optimization_level_performance);
+  
   const shaderc::SpvCompilationResult result = s_shaderc_compiler->CompileGlslToSpv(
     source.data(), source.length(), stage_kinds[static_cast<size_t>(stage)], "source", entry_point, options);
   if (result.GetCompilationStatus() != shaderc_compilation_status_success)
