--- a/dep/vixl/include/vixl/pool-manager-impl.h	2024-07-14 01:49:39.133027213 +0200
+++ b/dep/vixl/include/vixl/pool-manager-impl.h	2024-07-14 01:53:45.782705011 +0200
@@ -487,7 +487,7 @@
 }
 
 template <typename T>
-PoolManager<T>::~PoolManager<T>() VIXL_NEGATIVE_TESTING_ALLOW_EXCEPTION {
+PoolManager<T>::~PoolManager() VIXL_NEGATIVE_TESTING_ALLOW_EXCEPTION {
 #ifdef VIXL_DEBUG
   // Check for unbound objects.
   for (objects_iter iter = objects_.begin(); iter != objects_.end(); ++iter) {
