diff --git a/src/Glide64/DepthBufferRender.cpp b/src/Glide64/DepthBufferRender.cpp
index b10ada6..3fff4f8 100644
--- a/src/Glide64/DepthBufferRender.cpp
+++ b/src/Glide64/DepthBufferRender.cpp
@@ -103,24 +103,24 @@ __inline int idiv16(int x, int y)        // (x << 16) / y
   edx = x;
   (x << 16) | ()
    */ 
-#if !defined(__GNUC__) && !defined(NO_ASM)
-  __asm {
-        mov   eax, x
-        mov   ebx, y
-        mov   edx,eax   
-        sar   edx,16
-        shl   eax,16    
-        idiv  ebx  
-        mov   x, eax
-    }
-#elif !defined(NO_ASM)
-    int reminder;
-    asm ("idivl %[divisor]"
-          : "=a" (x), "=d" (reminder)
-          : [divisor] "g" (y), "d" (x >> 16), "a" (x << 16));
-#else
+//#if !defined(__GNUC__) && !defined(NO_ASM)
+//  __asm {
+//        mov   eax, x
+//        mov   ebx, y
+//        mov   edx,eax   
+//        sar   edx,16
+//        shl   eax,16    
+//        idiv  ebx  
+//        mov   x, eax
+//    }
+//#elif !defined(NO_ASM)
+//    int reminder;
+//    asm ("idivl %[divisor]"
+//          : "=a" (x), "=d" (reminder)
+//          : [divisor] "g" (y), "d" (x >> 16), "a" (x << 16));
+//#else
 	x = (((long long)x) << 16) / ((long long)y);
-#endif
+//#endif
     return x;
 }
 
