diff --git a/loader/wine.inf.in b/loader/wine.inf.in
index 83397bbb9b..340fed68ea 100644
--- a/loader/wine.inf.in
+++ b/loader/wine.inf.in
@@ -99,6 +99,7 @@ AddReg=\
     LicenseInformation, \
     NVIDIANGX, \
     ProtonOverrides, \
+    BatoceraOverrides, \
     SteamClient
 
 [DefaultInstall.ntamd64]
@@ -128,6 +129,7 @@ AddReg=\
     LicenseInformation, \
     NVIDIANGX, \
     ProtonOverrides, \
+    BatoceraOverrides, \
     SteamClient.ntamd64
 
 [DefaultInstall.ntarm64]
@@ -173,6 +175,7 @@ AddReg=\
     LicenseInformation, \
     NVIDIANGX, \
     ProtonOverrides, \
+    BatoceraOverrides, \
     SteamClient.ntamd64
 
 [Wow64Install.ntarm64]
@@ -2884,3 +2887,8 @@ HKCU,Software\Wine\AppDefaults\Maine-Win64-Shipping.exe\DllOverrides,"SpeechSynt
 HKCU,Software\Wine\AppDefaults\rayne1.exe\DllOverrides,"d3d8",,"native"
 HKCU,Software\Wine\AppDefaults\rayne2.exe\DllOverrides,"d3d8",,"native"
 HKCU,Software\Wine\AppDefaults\RDR2.exe\DllOverrides,"vulkan-1",,"native"
+
+[BatoceraOverrides]
+HKLM,System\CurrentControlSet\Services\winebus,"DisableHidraw",0x10001,"1"
+HKLM,System\CurrentControlSet\Services\winebus,"Enable SDL",0x10001,"1"
+
