diff --git a/drivers/hid/usbhid/hid-quirks.c b/drivers/hid/usbhid/hid-quirks.c
index e6cfd32..7aa0ff0 100644
--- a/drivers/hid/usbhid/hid-quirks.c
+++ b/drivers/hid/usbhid/hid-quirks.c
@@ -163,6 +163,53 @@ static const struct hid_blacklist {
 	{ USB_VENDOR_ID_DRACAL_RAPHNET, USB_DEVICE_ID_RAPHNET_2NES2SNES, HID_QUIRK_MULTI_INPUT },
 	{ USB_VENDOR_ID_DRACAL_RAPHNET, USB_DEVICE_ID_RAPHNET_4NES4SNES, HID_QUIRK_MULTI_INPUT },
 
+// Juyao
+	{ 0x0314,0x0326,0x040 },
+	{ 0x0314,0x0328,0x040 },
+
+// 4NES4SNES
+	{ 0x1781,0x0A9D,0x040 },
+
+// Retrolink SNES
+	{ 0x1292,0x5346,0x040 },
+	{ 0x1292,0x5366,0x040 },
+
+// Retrobit Atari to USB Driver
+	{ 0x1292,0x4154,0x040 },
+
+// Retrobit Genesis to PC USB Adapter
+	{ 0x1292,0x4745,0x040 },
+
+// Retrode
+	{ 0x0403,0x97c1,0x040 },
+
+// HuiJia  USB GamePad Mayflash Sega Saturn
+	{ 0x0e8f,0x3010,0x040 },
+
+// HuiJia  USB GamePad Mayflash NES SNES
+	{ 0x0079,0x1804,0x040 },
+
+// Mayflash Dreamcast
+	{ 0x0e8f,0x3013,0x040 },
+
+// Xin-Mo 2players 20pins
+	{ 0x16c0,0x75e1,0x040 },
+
+// Xinmo 18 pins USB THT 2P arcade controller THT 2P arcade controller
+	{ 0x16c0,0x05e1,0x040 },
+
+// USB PS2 Adapter GreenAsia Electronics
+	{ 0xe8f,0x1013,0x040 },
+
+// HuiJia  USB GamePad Mayflash N64 - need test
+	{ 0xe8f,0x3013,0x040 },
+
+// XCSOURCE 2 Encoder USB - Twin USB Gamepad
+	{ 0x810,0xe001,0x040 },
+
+// 3H Dual Arcade 2Players
+	{ 0x16c0,0x05e0,0x040 },
+
 	{ 0, 0 }
 };
 
