diff --git a/Makefile b/Makefile.new
index 5e598e0410..b422cd4894 100644
--- a/Makefile
+++ b/Makefile.new
@@ -21,7 +21,7 @@ lindbergh: src/lindbergh/lindbergh.c src/lindbergh/jvs.c src/lindbergh/jvs.h src
 
 libxdiff.a: $(XDIFF_OBJS)
 	mkdir -p $(BUILD)
-	ar rcs src/libxdiff/libxdiff.a $(XDIFF_OBJS)
+	$(AR) rcs src/libxdiff/libxdiff.a $(XDIFF_OBJS)
 
 src/libxdiff/xdiff/%.o: src/libxdiff/xdiff/%.c
 	$(CC) -DHAVE_CONFIG_H -fPIC -c $< -o $@
@@ -35,7 +35,7 @@ lindbergh.so: $(OBJS)
 	$(CC) $(CFLAGS) -c $< -o $@
 
 libsegaapi.so: src/libsegaapi/libsegaapi.o
-	$(CC) -m32 -O0 -g src/libsegaapi/libsegaapi.c -lFAudio -L/app/lib32 -fPIC -shared -o $(BUILD)/libsegaapi.so
+	$(CC) $(CFLAGS) -m32 -O0 -g src/libsegaapi/libsegaapi.c -fPIC -shared -o $(BUILD)/libsegaapi.so
 
 libkswapapi.so: src/libkswapapi/libkswapapi.o
 	$(CC) -m32 src/libkswapapi/libkswapapi.o -fPIC -shared -o $(BUILD)/libkswapapi.so