--- a/src/freedreno/common/freedreno_devices.py
+++ b/src/freedreno/common/freedreno_devices.py
@@ -1359,5 +1359,7 @@
 add_gpus([
         GPUId(chip_id=0xffff44050000, name="Adreno (TM) 830"),
+        GPUId(chip_id=0xffff44050001, name="Adreno (TM) 830"),
+        GPUId(chip_id=0x44050000, name="Adreno (TM) 830"),
         GPUId(chip_id=0x44050001, name="Adreno (TM) 830"), # KGSL
     ], A6xxGPUInfo(
         CHIP.A8XX,
