--- a/shell/linux/Makefile	2021-03-31 12:20:39.327057226 +0200
+++ b/shell/linux/Makefile	2021-03-31 12:20:16.670391402 +0200
@@ -305,9 +305,9 @@
     MFLAGS += -marm -march=armv8-a+crc -mfpu=neon -mfloat-abi=hard -funroll-loops -mtune=cortex-a53
     ASFLAGS += -march=armv8-a+crc -mfpu=neon -mfloat-abi=hard
     CXXFLAGS += -fexceptions
-    CFLAGS += -DMESA_EGL_NO_X11_HEADERS=1 -DEGL_NO_X11=1 -fpermissive -fno-builtin-sqrtf
+    CFLAGS += -D TARGET_LINUX_ARMv8 -DARM_HARDFP -fsingle-precision-constant -DPNG_ARM_NEON=1 -DPNG_ARM_NEON_OPT=1 -DMESA_EGL_NO_X11_HEADERS=1 -DEGL_NO_X11=1 -fpermissive -fno-builtin-sqrtf
     INCS +=  -I/opt/vc/include/ -I/opt/vc/include/libdrm
-    LIBS += -ldrm -lGLESv2
+    LIBS += -lpng -lz -ldrm -lGLESv2
     USE_GLES := 1
     USE_EVDEV := 1
     USE_UDEV := 1
