--- a/src/main.conf	2021-02-22 21:26:59.000000000 +0100
+++ b/src/main.conf	2021-03-15 18:27:52.540388129 +0100
@@ -75,7 +75,7 @@
 # Specify the policy to the JUST-WORKS repairing initiated by peer
 # Possible values: "never", "confirm", "always"
 # Defaults to "never"
-#JustWorksRepairing = never
+JustWorksRepairing = always
 
 # How long to keep temporary devices around
 # The value is in seconds. Default is 30.
@@ -235,7 +235,7 @@
 # AutoEnable defines option to enable all controllers when they are found.
 # This includes adapters present on start as well as adapters that are plugged
 # in later on. Defaults to 'false'.
-#AutoEnable=false
+AutoEnable=true

 # Audio devices that were disconnected due to suspend will be reconnected on
 # resume. ResumeDelay determines the delay between when the controller
