diff --git a/src/adapter.c b/src/adapter.c
index a571b18..afba3ad 100644
--- a/src/adapter.c
+++ b/src/adapter.c
@@ -3039,6 +3039,9 @@ static struct link_key_info *get_key_info(GKeyFile *key_file, const char *peer)
 	struct link_key_info *info = NULL;
 	char *str;
 
+	// batocera / reusing the key sometimes causes failure
+	return NULL;
+
 	str = g_key_file_get_string(key_file, "LinkKey", "Key", NULL);
 	if (!str || strlen(str) < 32)
 		goto failed;
