diff --git a/configure b/configure
old mode 100755
new mode 100755
index bec51ffba5..f88d90a8db
--- a/configure
+++ b/configure
@@ -15978,7 +15978,7 @@ fi
 
 
 fi
-if (test "${enable_external_ell}" != "yes" &&
+if (test "${enable_external_ell}" == "yes" &&
 		(test "${enable_btpclient}" = "yes" || test "${enable_mesh}" = "yes")); then
 	if (test ! -f ${srcdir}/ell/ell.h) &&
 			(test ! -f ${srcdir}/../ell/ell/ell.h); then
