diff --git a/plugins/sixaxis.c b/plugins/sixaxis.c.new
index 544ab399ad..b68b0f46fb 100644
--- a/plugins/sixaxis.c
+++ b/plugins/sixaxis.c.new
@@ -350,7 +350,8 @@ static bool setup_device(int fd, const char *sysfs_path,
 
 	btd_device_device_set_name(device, cp->name);
 	btd_device_set_pnpid(device, cp->source, cp->vid, cp->pid, cp->version);
-	btd_device_set_temporary(device, true);
+	btd_device_set_temporary(device, false);
+	btd_device_set_trusted(device, true);
 
 	closure = g_new0(struct authentication_closure, 1);
 	if (!closure) {
