diff --git a/configure b/configure
index 82581091cd..3bc5206d26
--- a/configure
+++ b/configure
@@ -17161,10 +17161,6 @@ if test -z "${CUPS_TRUE}" && test -z "${CUPS_FALSE}"; then
   as_fn_error $? "conditional \"CUPS\" was never defined.
 Usually this means the macro was only invoked conditionally." "$LINENO" 5
 fi
-if test -z "${CUPS_SERVERBIN_TRUE}" && test -z "${CUPS_SERVERBIN_FALSE}"; then
-  as_fn_error $? "conditional \"CUPS_SERVERBIN\" was never defined.
-Usually this means the macro was only invoked conditionally." "$LINENO" 5
-fi
 if test -z "${MESH_TRUE}" && test -z "${MESH_FALSE}"; then
   as_fn_error $? "conditional \"MESH\" was never defined.
 Usually this means the macro was only invoked conditionally." "$LINENO" 5
