--- pacman-5.2.1_ORIGINAL/src/pacman/package.c	2021-04-21 15:49:43.178909492 -0700
+++ pacman-5.2.1/src/pacman/package.c	2021-04-27 17:07:32.292734413 -0700
@@ -603,6 +603,7 @@
 		  printf("    <packager>%s</packager>\n",  protected_string);
 		  printf("    <download_size>%ld</download_size>\n",   alpm_pkg_get_size(pkg)/1024);
 		  printf("    <installed_size>%ld</installed_size>\n", alpm_pkg_get_isize(pkg)/1024);
+		  printf("    <arch>%s</arch>\n",                       alpm_pkg_get_arch(pkg));
 
 		  if(installed_pkg == NULL) {
 		    printf("    <status>none</status>\n");
@@ -637,6 +638,7 @@
 					colstr->version, alpm_pkg_get_version(pkg), colstr->nocolor);
 
 			print_groups(pkg);
+			printf(" <%s>", alpm_pkg_get_arch(pkg)); // Batocera architecture
 			if(show_status) {
 				print_installed(db_local, pkg);
 			}
