--- a/SDL/SDLMain.cpp	2023-09-18 20:27:37.602845381 +0200
+++ b/SDL/SDLMain.cpp	2023-09-18 20:30:43.171046568 +0200
@@ -776,7 +776,7 @@
 #if !defined(MOBILE_DEVICE)
 	case SDL_WINDOWEVENT:
 		switch (event.window.event) {
-		case SDL_WINDOWEVENT_SIZE_CHANGED:  // better than RESIZED, more general
+		case SDL_WINDOWEVENT_RESIZED:
 		{
 			int new_width = event.window.data1;
 			int new_height = event.window.data2;
