diff --git a/bsp/drivers/drm/sunxi_drm_hdmi.c b/bsp/drivers/drm/sunxi_drm_hdmi.c
index e0b2350ee..575fcd3b4 100644
--- a/bsp/drivers/drm/sunxi_drm_hdmi.c
+++ b/bsp/drivers/drm/sunxi_drm_hdmi.c
@@ -2541,8 +2541,8 @@ static int _sunxi_drm_hdmi_check(struct drm_encoder *encoder,
 	}
 
 exit:
-	hdmi_inf("drm hdmi check mode: %s >>>>>>>>>>>>>>>>\n",
-		crtc_state->mode_changed ? "change" : "unchange");
+	//hdmi_inf("drm hdmi check mode: %s >>>>>>>>>>>>>>>>\n",
+	//	crtc_state->mode_changed ? "change" : "unchange");
 	return 0;
 }
 
