diff --git a/libgamestream/mkcert.c b/libgamestream/mkcert.c
index 64fede042a..6bfb4cb79b 100644
--- a/libgamestream/mkcert.c
+++ b/libgamestream/mkcert.c
@@ -42,7 +42,7 @@ CERT_KEY_PAIR mkcert_generate() {
     bio_err = BIO_new_fp(stderr, BIO_NOCLOSE);
 
     OpenSSL_add_all_algorithms();
-    ERR_load_crypto_strings();
+    //ERR_load_crypto_strings();
 
     mkcert(&x509, &pkey, NUM_BITS, SERIAL, NUM_YEARS);
 
