From 5b6639c8a76c4bbb5e0b4bcfe41b8df136608885 Mon Sep 17 00:00:00 2001
From: Romain Tisserand <romain.tisserand@gmail.com>
Date: Sat, 11 Jul 2020 21:29:09 +0200
Subject: [PATCH] Fix gcc-10 compilation (fno-common flag enabled by default)

---
 src/config.h | 2 +-
 src/sdl.h    | 4 ++--
 2 files changed, 3 insertions(+), 3 deletions(-)

diff --git a/src/config.h b/src/config.h
index 72c22f3f..c56c1dbe 100644
--- a/src/config.h
+++ b/src/config.h
@@ -48,7 +48,7 @@ typedef struct _CONFIGURATION {
   enum codecs codec;
 } CONFIGURATION, *PCONFIGURATION;
 
-bool inputAdded;
+extern bool inputAdded;
 
 bool config_file_parse(char* filename, PCONFIGURATION config);
 void config_parse(int argc, char* argv[], PCONFIGURATION config);
diff --git a/src/sdl.h b/src/sdl.h
index c26322ec..65345ba5 100644
--- a/src/sdl.h
+++ b/src/sdl.h
@@ -36,7 +36,7 @@
 void sdl_init(int width, int height, bool fullscreen);
 void sdl_loop();
 
-SDL_mutex *mutex;
-int sdlCurrentFrame, sdlNextFrame;
+extern SDL_mutex *mutex;
+extern int sdlCurrentFrame, sdlNextFrame;
 
 #endif /* HAVE_SDL */
