diff --git a/src/frontend/qt_sdl/Config.cpp b/src/frontend/qt_sdl/Config.cpp
index 7a0ec698ba..0b06f8f120 100644
--- a/src/frontend/qt_sdl/Config.cpp
+++ b/src/frontend/qt_sdl/Config.cpp
@@ -140,7 +140,7 @@ int DSiBatteryLevel;
 bool DSiBatteryCharging;
 
 
-const char* kConfigFile = "melonDS.ini";
+const char* kConfigFile = "/userdata/system/configs/melonDS/melonDS.ini";
 
 ConfigEntry ConfigFile[] =
 {
@@ -241,28 +241,28 @@ ConfigEntry ConfigFile[] =
 
     {"ExternalBIOSEnable", 1, &ExternalBIOSEnable, false},
 
-    {"BIOS9Path", 2, &BIOS9Path, (std::string)""},
-    {"BIOS7Path", 2, &BIOS7Path, (std::string)""},
-    {"FirmwarePath", 2, &FirmwarePath, (std::string)""},
+    {"BIOS9Path", 2, &BIOS9Path, (std::string)"/userdata/bios/biosnds9.bin"},
+    {"BIOS7Path", 2, &BIOS7Path, (std::string)"/userdata/bios/biosnds7.bin"},
+    {"FirmwarePath", 2, &FirmwarePath, (std::string)"/userdata/bios/dsfirmware.bin"},
 
-    {"DSiBIOS9Path", 2, &DSiBIOS9Path, (std::string)""},
-    {"DSiBIOS7Path", 2, &DSiBIOS7Path, (std::string)""},
-    {"DSiFirmwarePath", 2, &DSiFirmwarePath, (std::string)""},
-    {"DSiNANDPath", 2, &DSiNANDPath, (std::string)""},
+    {"DSiBIOS9Path", 2, &DSiBIOS9Path, (std::string)"/userdata/bios/biosdsi9.bin"},
+    {"DSiBIOS7Path", 2, &DSiBIOS7Path, (std::string)"/userdata/bios/biosdsi7.bin"},
+    {"DSiFirmwarePath", 2, &DSiFirmwarePath, (std::string)"/userdata/bios/dsifirmware.bin"},
+    {"DSiNANDPath", 2, &DSiNANDPath, (std::string)"/userdata/bios/dsinand.bin"},
 
     {"DLDIEnable", 1, &DLDIEnable, false},
     {"DLDISDPath", 2, &DLDISDPath, (std::string)"dldi.bin"},
     {"DLDISize", 0, &DLDISize, 0},
     {"DLDIReadOnly", 1, &DLDIReadOnly, false},
     {"DLDIFolderSync", 1, &DLDIFolderSync, false},
-    {"DLDIFolderPath", 2, &DLDIFolderPath, (std::string)""},
+    {"DLDIFolderPath", 2, &DLDIFolderPath, (std::string)"/userdata/saves/melonds"},
 
     {"DSiSDEnable", 1, &DSiSDEnable, false},
     {"DSiSDPath", 2, &DSiSDPath, (std::string)"dsisd.bin"},
     {"DSiSDSize", 0, &DSiSDSize, 0},
     {"DSiSDReadOnly", 1, &DSiSDReadOnly, false},
     {"DSiSDFolderSync", 1, &DSiSDFolderSync, false},
-    {"DSiSDFolderPath", 2, &DSiSDFolderPath, (std::string)""},
+    {"DSiSDFolderPath", 2, &DSiSDFolderPath, (std::string)"/userdata/saves/melonds"},
 
     {"FirmwareOverrideSettings", 1, &FirmwareOverrideSettings, false},
     {"FirmwareUsername", 2, &FirmwareUsername, (std::string)"melonDS"},
@@ -284,7 +284,7 @@ ConfigEntry ConfigFile[] =
     {"AudioBitrate", 0, &AudioBitrate, 0},
     {"AudioVolume", 0, &AudioVolume, 256},
     {"MicInputType", 0, &MicInputType, 1},
-    {"MicWavPath", 2, &MicWavPath, (std::string)""},
+    {"MicWavPath", 2, &MicWavPath, (std::string)"/userdata/saves/melonds"},
 
     {"LastROMFolder", 2, &LastROMFolder, (std::string)""},
 
@@ -299,9 +299,9 @@ ConfigEntry ConfigFile[] =
     {"RecentROM_8", 2, &RecentROMList[8], (std::string)""},
     {"RecentROM_9", 2, &RecentROMList[9], (std::string)""},
 
-    {"SaveFilePath", 2, &SaveFilePath, (std::string)""},
-    {"SavestatePath", 2, &SavestatePath, (std::string)""},
-    {"CheatFilePath", 2, &CheatFilePath, (std::string)""},
+    {"SaveFilePath", 2, &SaveFilePath, (std::string)"/userdata/saves/melonds"},
+    {"SavestatePath", 2, &SavestatePath, (std::string)"/userdata/saves/melonds"},
+    {"CheatFilePath", 2, &CheatFilePath, (std::string)"/userdata/cht/cheats/melonDS"},
 
     {"EnableCheats", 1, &EnableCheats, false},
