diff --git a/src/frontend/qt_sdl/Config.cpp b/src/frontend/qt_sdl/Config.cpp
index c91d6ef911..549a41ddb2 100644
--- a/src/frontend/qt_sdl/Config.cpp
+++ b/src/frontend/qt_sdl/Config.cpp
@@ -335,14 +335,7 @@ ConfigEntry ConfigFile[] =
 void LoadFile(int inst)
 {
     FILE* f;
-    if (inst > 0)
-    {
-        char name[100] = {0};
-        snprintf(name, 99, kUniqueConfigFile, inst+1);
-        f = Platform::OpenLocalFile(name, "r");
-    }
-    else
-        f = Platform::OpenLocalFile(kConfigFile, "r");
+    f = Platform::OpenLocalFile(kConfigFile, "r");
 
     if (!f) return;
 
@@ -362,9 +355,6 @@ void LoadFile(int inst)
         {
             if (!strncmp(entry->Name, entryname, 32))
             {
-                if ((inst > 0) && (!entry->InstanceUnique))
-                    break;
-
                 switch (entry->Type)
                 {
                 case 0: *(int*)entry->Value = strtol(entryval, NULL, 10); break;
@@ -394,10 +384,6 @@ void Load()
     }
 
     LoadFile(0);
-
-    int inst = Platform::InstanceID();
-    if (inst > 0)
-        LoadFile(inst);
 }
 
 void Save()
@@ -405,22 +391,12 @@ void Save()
     int inst = Platform::InstanceID();
 
     FILE* f;
-    if (inst > 0)
-    {
-        char name[100] = {0};
-        snprintf(name, 99, kUniqueConfigFile, inst+1);
-        f = Platform::OpenLocalFile(name, "w");
-    }
-    else
-        f = Platform::OpenLocalFile(kConfigFile, "w");
+    f = Platform::OpenLocalFile(kConfigFile, "w");
 
     if (!f) return;
 
     for (ConfigEntry* entry = &ConfigFile[0]; entry->Value; entry++)
     {
-        if ((inst > 0) && (!entry->InstanceUnique))
-            continue;
-
         switch (entry->Type)
         {
         case 0: fprintf(f, "%s=%d\r\n", entry->Name, *(int*)entry->Value); break;
