diff --git a/src/frontend/qt_sdl/main.cpp b/src/frontend/qt_sdl/main.cpp
index 00135b40b7..c01f6b6dc6 100644
--- a/src/frontend/qt_sdl/main.cpp
+++ b/src/frontend/qt_sdl/main.cpp
@@ -685,6 +685,8 @@ void EmuThread::emuRun()
 
     // checkme
     emit windowEmuStart();
+    // batocera - always start fullscreen
+    emit windowFullscreenToggle();
     if (audioDevice) SDL_PauseAudioDevice(audioDevice, 0);
     micOpen();
 }
