diff --git a/libretro/libretro-core.c b/libretro/libretro-core.c
index 040a49d..38c3848 100644
--- a/libretro/libretro-core.c
+++ b/libretro/libretro-core.c
@@ -563,7 +563,7 @@ void retro_get_system_info(struct retro_system_info *info)
 {
    memset(info, 0, sizeof(*info));
    info->library_name     = "Atari800";
-   info->library_version  = "3.1.0" GIT_VERSION;
+   info->library_version  = "3.1.0";
    info->valid_extensions = "xfd|atr|cdm|cas|bin|a52|zip|atx";
    info->need_fullpath    = true;
    info->block_extract = false;
@@ -742,15 +742,15 @@ bool retro_unserialize(const void *data_, size_t size)
 
 void *retro_get_memory_data(unsigned id)
 {
-  if ( id == RETRO_MEMORY_SYSTEM_RAM )
-    return MEMORY_mem;
+  if ( id == RETRO_MEMORY_SYSTEM_RAM )
+    return MEMORY_mem;
    return NULL;
 }
 
 size_t retro_get_memory_size(unsigned id)
 {
-  if ( id == RETRO_MEMORY_SYSTEM_RAM )
-    return 65536;
+  if ( id == RETRO_MEMORY_SYSTEM_RAM )
+    return 65536;
    return 0;
 }
 
