--- a/libretro/libretro-core.c	2019-05-29 06:14:47.227859884 -0300
+++ b/libretro/libretro-core.c	2019-05-29 06:15:44.721734413 -0300
@@ -563,7 +563,7 @@
 {
    memset(info, 0, sizeof(*info));
    info->library_name     = "Atari800";
-   info->library_version  = "3.1.0" GIT_VERSION;
+   info->library_version  = "3.1.0";
    info->valid_extensions = "xfd|atr|cdm|cas|bin|a52|zip|atx";
    info->need_fullpath    = true;
    info->block_extract = false;
