diff --git a/atari800/src/cfg.c b/atari800/src/cfg.c
index 224c8ff..51da289 100644
--- a/atari800/src/cfg.c
+++ b/atari800/src/cfg.c
@@ -68,7 +68,7 @@ int CFG_save_on_exit = FALSE;
 /* If another default path config path is defined use it
    otherwise use the default one */
 #ifndef DEFAULT_CFG_NAME
-#define DEFAULT_CFG_NAME ".atari800.cfg"
+#define DEFAULT_CFG_NAME "configs/retroarch/cores/atari800.cfg"
 #endif
 
 #ifndef SYSTEM_WIDE_CFG_FILE
