diff --git a/drivers/amlogic/display/logo/logo.c b/drivers/amlogic/display/logo/logo.c
index 9266be3..c2f10bd 100644
--- a/drivers/amlogic/display/logo/logo.c
+++ b/drivers/amlogic/display/logo/logo.c
@@ -117,7 +117,7 @@ int set_osd_freescaler(int index, enum vmode_e new_mode)
 static int refresh_mode_and_logo(bool first)
 {
 	enum vmode_e cur_mode = VMODE_MAX;
-	int hdp_state = get_hpd_state();
+	int hdp_state = 1;//get_hpd_state();
 
 	if (!first && osd_get_logo_index() != logo_info.index)
 		return -1;
