diff --git a/drivers/amlogic/amports/vh265.c b/drivers/amlogic/amports/vh265.c
index e962733..70d9505 100644
--- a/drivers/amlogic/amports/vh265.c
+++ b/drivers/amlogic/amports/vh265.c
@@ -215,7 +215,7 @@ static u32 dynamic_buf_num_margin;
 #else
 static u32 buf_alloc_width;
 static u32 buf_alloc_height;
-static u32 dynamic_buf_num_margin = 7;
+static u32 dynamic_buf_num_margin = 16;
 #endif
 static u32 buf_alloc_size;
 static u32 re_config_pic_flag;
diff --git a/drivers/amlogic/amports/vvp9.c b/drivers/amlogic/amports/vvp9.c
index d129327..d63bd1e 100644
--- a/drivers/amlogic/amports/vvp9.c
+++ b/drivers/amlogic/amports/vvp9.c
@@ -1691,7 +1691,7 @@ static u32 dynamic_buf_num_margin;
 #else
 static u32 buf_alloc_width;
 static u32 buf_alloc_height;
-static u32 dynamic_buf_num_margin = 7;
+static u32 dynamic_buf_num_margin = 16;
 #endif
 static u32 buf_alloc_depth = 10;
 static u32 buf_alloc_size;
