diff --git a/VENDOR/libretro-rs-ffi/include/libretro.h b/VENDOR/libretro-rs-ffi/include/libretro.h
index db5f695..220bd36 100644
--- a/VENDOR/libretro-rs-ffi/include/libretro.h
+++ b/VENDOR/libretro-rs-ffi/include/libretro.h
@@ -5275,7 +5275,12 @@
   */
  typedef unsigned (RETRO_CALLCONV *retro_get_num_images_t)(void);
  
- struct retro_game_info;
+ struct retro_game_info {
+    const char *path;
+    const void *data;
+    size_t      size;
+    const char *meta;
+ };
  
  /**
   * Replaces the disk image at the given index with a new disk.
@@ -6812,21 +6817,7 @@
  
  /** @} */
  
- struct retro_game_info
- {
-    const char *path;       /* Path to game, UTF-8 encoded.
-                             * Sometimes used as a reference for building other paths.
-                             * May be NULL if game was loaded from stdin or similar,
-                             * but in this case some cores will be unable to load `data`.
-                             * So, it is preferable to fabricate something here instead
-                             * of passing NULL, which will help more cores to succeed.
-                             * retro_system_info::need_fullpath requires
-                             * that this path is valid. */
-    const void *data;       /* Memory buffer of loaded game. Will be NULL
-                             * if need_fullpath was set. */
-    size_t      size;       /* Size of memory buffer. */
-    const char *meta;       /* String of implementation specific meta-data. */
- };
+ /* retro_game_info defined earlier */
  
  /** @defgroup GET_CURRENT_SOFTWARE_FRAMEBUFFER Frontend-Owned Framebuffers
   * @{
diff --git a/VENDOR/libretro-rs-ffi/.cargo-checksum.json b/VENDOR/libretro-rs-ffi/.cargo-checksum.json
index 12d4a90..b8f2a1c 100644
--- a/VENDOR/libretro-rs-ffi/.cargo-checksum.json
+++ b/VENDOR/libretro-rs-ffi/.cargo-checksum.json
@@ -1 +1 @@
-{"files":{"Cargo.toml":"65e2f21c32adf2d7c33b6532a08f1582411492e252265d3a0c59f969352afaef","README.md":"85b2d76a7bc85980ee692667d7842fc9e17730944be5a7078693e1d45555b710","build.rs":"6c1f303376e63a0cf8fe0315fab6263c0b50d3886b89e13a6ddbd5ab58e3d1ee","include/libretro.h":"db5f695c60359f8a9cf3bd57dde0bd2f37f91f2ac66cb4b0d76932961bdb1351","src/lib.rs":"5d6780d6f8a244fcf7fbdf06407b06fd83d0eb3aefdcafb576bc7192b10c9e4b"},"package":null}
\ No newline at end of file
+{"files":{"Cargo.toml":"65e2f21c32adf2d7c33b6532a08f1582411492e252265d3a0c59f969352afaef","README.md":"85b2d76a7bc85980ee692667d7842fc9e17730944be5a7078693e1d45555b710","build.rs":"6c1f303376e63a0cf8fe0315fab6263c0b50d3886b89e13a6ddbd5ab58e3d1ee","include/libretro.h":"220bd363eaab7d9422914a5845bfcb405d895443df09f12ccf3f49b86b87ea20","src/lib.rs":"5d6780d6f8a244fcf7fbdf06407b06fd83d0eb3aefdcafb576bc7192b10c9e4b"},"package":null}
\ No newline at end of file
