diff --git a/Makefile b/Makefile
index aadfdf5..9475da1 100644
--- a/Makefile
+++ b/Makefile
@@ -277,6 +277,25 @@ PSS_STYLE :=2
 LDFLAGS += -DLL
 CFLAGS += -D_CRT_SECURE_NO_DEPRECATE
 
+# ARM
+else ifneq (,$(findstring armv,$(platform)))
+       TARGET := $(TARGET_NAME)_libretro.so
+       fpic := -fPIC
+       SHARED := -shared -Wl,--no-undefined -Wl,--version-script=link.T
+       CC = gcc
+       CXX = g++
+       FLAGS += -marm
+       ifneq (,$(findstring softfloat,$(platform)))
+               FLAGS += -mfloat-abi=softfp
+       else ifneq (,$(findstring hardfloat,$(platform)))
+               FLAGS += -mfloat-abi=hard
+       endif
+       ifneq (,$(findstring neon,$(platform)))
+              FLAGS += -mfpu=neon
+               HAVE_NEON = 1
+       endif
+       FLAGS += -DARM
+
 # Windows
 else
    TARGET := $(TARGET_NAME)_libretro.dll
