diff --git a/Makefile b/Makefile
index 6c840aa..1749d80 100644
--- a/Makefile
+++ b/Makefile
@@ -53,6 +53,25 @@ ifeq ($(platform), unix)
    TARGET := $(TARGET_NAME)_libretro.so
    fpic := -fPIC
    SHARED := -shared -Wl,--no-undefined -Wl,--version-script=link.T
+# Raspberry Pi
+else ifneq (,$(findstring rpi,$(platform)))
+	TARGET := $(TARGET_NAME)_libretro.so
+   	fpic := -fPIC
+   	SHARED := -shared -Wl,--no-undefined -Wl,--version-script=link.T
+   
+	ifneq (,$(findstring rpi1,$(platform)))
+    	CFLAGS += -marm -mcpu=arm1176jzf-s -mfloat-abi=hard
+	else ifneq (,$(findstring rpi2,$(platform)))
+    	CFLAGS += -marm -mcpu=cortex-a7 -mfpu=neon-vfpv4 -mfloat-abi=hard
+	else ifneq (,$(findstring rpi3,$(platform)))
+		ifneq (,$(findstring rpi3_64,$(platform)))
+           CFLAGS += -mcpu=cortex-a53 -mtune=cortex-a53
+		else
+			CFLAGS += -marm -mcpu=cortex-a53 -mfpu=neon-fp-armv8 -mfloat-abi=hard
+		endif
+	else ifneq (,$(findstring rpi4,$(platform)))
+         CFLAGS += -mcpu=cortex-a72 -mtune=cortex-a72
+	endif
 else ifeq ($(platform), linux-portable)
    TARGET := $(TARGET_NAME)_libretro.so
    fpic := -fPIC -nostdlib
