--- a/core/Makefile	2021-01-20 18:41:36.873752719 +0100
+++ b/core/Makefile	2021-01-20 18:42:19.270417845 +0100
@@ -164,7 +164,7 @@
 LDSCRIPT = $(SRC)/$(ARCH)/syslinux.ld
 
 %.elf: %.o $(LIBDEP) $(LDSCRIPT) $(AUXLIBS)
-	$(LD) $(LDFLAGS) -Bsymbolic $(LD_PIE) -E --hash-style=gnu -T $(LDSCRIPT) -M -o $@ $< \
+	$(LD) $(LDFLAGS) -z muldefs -Bsymbolic $(LD_PIE) -E --hash-style=gnu -T $(LDSCRIPT) -M -o $@ $< \
 		--start-group $(LIBS) $(subst $(*F).elf,lib$(*F).a,$@) --end-group \
 		> $(@:.elf=.map)
 	$(OBJDUMP) -h $@ > $(@:.elf=.sec)
