From cbc07cc8e92d376572cee3c7ac01926925ec7337 Mon Sep 17 00:00:00 2001
From: Thomas Petazzoni <thomas.petazzoni@bootlin.com>
Date: Thu, 16 Apr 2020 16:56:52 +0200
Subject: [PATCH] mtools: build for the host

Signed-off-by: Thomas Petazzoni <thomas.petazzoni@bootlin.com>
---
 mtools/Makefile | 12 ++++++------
 1 file changed, 6 insertions(+), 6 deletions(-)

diff --git a/mtools/Makefile b/mtools/Makefile
index 70269efd..aa78311d 100755
--- a/mtools/Makefile
+++ b/mtools/Makefile
@@ -2,8 +2,8 @@ include $(MAKEDIR)/syslinux.mk
 
 OPTFLAGS = -g -Os
 INCLUDES = -I$(SRC) -I$(objdir) -I$(SRC)/../libfat -I$(SRC)/../libinstaller
-CFLAGS	 = $(GCCWARN) -D_FILE_OFFSET_BITS=64 $(OPTFLAGS) $(INCLUDES)
-LDFLAGS	 = 
+CFLAGS	 = $(CFLAGS_FOR_BUILD) $(GCCWARN) -D_FILE_OFFSET_BITS=64 $(OPTFLAGS) $(INCLUDES)
+LDFLAGS	 = $(LDFLAGS_FOR_BUILD)
 
 SRCS     = syslinux.c \
 	   ../libinstaller/fs.c \
@@ -34,16 +34,16 @@ spotless: clean
 installer: syslinux
 
 syslinux: $(OBJS)
-	$(CC) $(LDFLAGS) -o $@ $^
+	$(CC_FOR_BUILD) $(LDFLAGS) -o $@ $^
 
 strip:
 	$(STRIP) syslinux
 
 %.o: %.c
-	$(CC) $(UMAKEDEPS) $(CFLAGS) -c -o $@ $<
+	$(CC_FOR_BUILD) $(UMAKEDEPS) $(CFLAGS) -c -o $@ $<
 %.i: %.c
-	$(CC) $(UMAKEDEPS) $(CFLAGS) -E -o $@ $<
+	$(CC_FOR_BUILD) $(UMAKEDEPS) $(CFLAGS) -E -o $@ $<
 %.s: %.c
-	$(CC) $(UMAKEDEPS) $(CFLAGS) -S -o $@ $<
+	$(CC_FOR_BUILD) $(UMAKEDEPS) $(CFLAGS) -S -o $@ $<
 
 -include .*.d
-- 
2.25.2

