--- a/gpxe/src/arch/x86_64/Makefile	2021-01-20 18:52:13.670396015 +0100
+++ b/gpxe/src/arch/x86_64/Makefile	2021-01-20 18:52:20.970395739 +0100
@@ -1,6 +1,6 @@
 # Code size reduction.
 #
-CFLAGS		+= -fstrength-reduce -fomit-frame-pointer
+CFLAGS		+= -fstrength-reduce -fomit-frame-pointer -fcommon
 
 # Code size reduction.  gcc3 needs a different syntax to gcc2 if you
 # want to avoid spurious warnings.
