diff --git a/drivers/hid/hid-quirks.c b/drivers/hid/hid-quirks.c
index e92b77f..60bcce2 100644
--- a/drivers/hid/hid-quirks.c
+++ b/drivers/hid/hid-quirks.c
@@ -27,6 +27,54 @@
  */
 
 static const struct hid_device_id hid_quirks[] = {
+
+// Juyao
+	{ HID_USB_DEVICE(0x0314,0x0326),0x040 },
+	{ HID_USB_DEVICE(0x0314,0x0328),0x040 },
+
+// 4NES4SNES
+	{ HID_USB_DEVICE(0x1781,0x0A9D),0x040 },
+
+// Retrolink SNES
+	{ HID_USB_DEVICE(0x1292,0x5346),0x040 },
+	{ HID_USB_DEVICE(0x1292,0x5366),0x040 },
+
+// Retrobit Atari to USB Driver
+	{ HID_USB_DEVICE(0x1292,0x4154),0x040 },
+
+// Retrobit Genesis to PC USB Adapter
+	{ HID_USB_DEVICE(0x1292,0x4745),0x040 },
+
+// Retrode
+	{ HID_USB_DEVICE(0x0403,0x97c1),0x040 },
+
+// HuiJia  USB GamePad Mayflash Sega Saturn
+	{ HID_USB_DEVICE(0x0e8f,0x3010),0x040 },
+
+// HuiJia  USB GamePad Mayflash NES SNES
+	{ HID_USB_DEVICE(0x0079,0x1804),0x040 },
+
+// Mayflash Dreamcast
+	{ HID_USB_DEVICE(0x0e8f,0x3013),0x040 },
+
+// Xin-Mo 2players 20pins
+	{ HID_USB_DEVICE(0x16c0,0x75e1),0x040 },
+
+// Xinmo 18 pins USB THT 2P arcade controller THT 2P arcade controller
+	{ HID_USB_DEVICE(0x16c0,0x05e1),0x040 },
+
+// USB PS2 Adapter GreenAsia Electronics
+	{ HID_USB_DEVICE(0xe8f,0x1013),0x040 },
+
+// HuiJia  USB GamePad Mayflash N64 - need test
+	{ HID_USB_DEVICE(0xe8f,0x3013),0x040 },
+
+// XCSOURCE 2 Encoder USB - Twin USB Gamepad
+	{ HID_USB_DEVICE(0x810,0xe001),0x040 },
+
+// 3H Dual Arcade 2Players
+	{ HID_USB_DEVICE(0x16c0,0x05e0),0x040 },
+
 	{ HID_USB_DEVICE(USB_VENDOR_ID_AASHIMA, USB_DEVICE_ID_AASHIMA_GAMEPAD), HID_QUIRK_BADPAD },
 	{ HID_USB_DEVICE(USB_VENDOR_ID_AASHIMA, USB_DEVICE_ID_AASHIMA_PREDATOR), HID_QUIRK_BADPAD },
 	{ HID_USB_DEVICE(USB_VENDOR_ID_AFATECH, USB_DEVICE_ID_AFATECH_AF9016), HID_QUIRK_FULLSPEED_INTERVAL },
