diff --git a/xpad.c b/xpad.c
index a6f77012a6..1703485eef 100644
--- a/xpad.c
+++ b/xpad.c
@@ -1725,7 +1725,7 @@ static int xpad_suspend(struct usb_interface *intf, pm_message_t message)
 			xpad360w_poweroff_controller(xpad);
 	} else {
 		mutex_lock(&input->mutex);
-		if (input_device_enabled(input))
+		if (input->users)
 			xpad_stop_input(xpad);
 		mutex_unlock(&input->mutex);
 	}
@@ -1745,7 +1745,7 @@ static int xpad_resume(struct usb_interface *intf)
 		retval = xpad360w_start_input(xpad);
 	} else {
 		mutex_lock(&input->mutex);
-		if (input_device_enabled(input)) {
+		if (input->users) {
 			retval = xpad_start_input(xpad);
 		}
 		mutex_unlock(&input->mutex);
