diff --git a/include/SDL.h b/include/SDL.h
index 366d50f..7a66e8a 100644
--- a/include/SDL.h
+++ b/include/SDL.h
@@ -26,8 +26,8 @@
  */
 
 
-#ifndef SDL_h_
-#define SDL_h_
+#ifndef _SDL_H
+#define _SDL_H
 
 #include "SDL_main.h"
 #include "SDL_stdinc.h"
diff --git a/include/SDL_endian.h b/include/SDL_endian.h
index 3eda7f2..0b802fa 100644
--- a/include/SDL_endian.h
+++ b/include/SDL_endian.h
@@ -25,8 +25,8 @@
  *  Functions for reading and writing endian-specific values
  */
 
-#ifndef SDL_endian_h_
-#define SDL_endian_h_
+#ifndef _SDL_endian_h
+#define _SDL_endian_h
 
 #include "SDL_stdinc.h"
 
