diff --git a/CMakeLists.txt b/CMakeLists.txt
index 5df4b2e..c4b794d 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -462,14 +462,14 @@ else()
 	message("ao NOT found, disabling ao sound backend")
 endif(AO_FOUND)
 
-check_lib(BLUEZ bluez bluez QUIET)
-if(BLUEZ_FOUND)
-	add_definitions(-DHAVE_BLUEZ=1)
-	message("bluez found, enabling bluetooth support")
-else()
+#check_lib(BLUEZ bluez bluez QUIET)
+#if(BLUEZ_FOUND)
+#	add_definitions(-DHAVE_BLUEZ=1)
+#	message("bluez found, enabling bluetooth support")
+#else()
 	add_definitions(-DHAVE_BLUEZ=0)
 	message("bluez NOT found, disabling bluetooth support")
-endif(BLUEZ_FOUND)
+#endif(BLUEZ_FOUND)
 
 check_lib(PULSEAUDIO libpulse pulse QUIET)
 if(PULSEAUDIO_FOUND)
