diff --git a/Source/Core/DolphinWX/MainNoGUI.cpp b/Source/Core/DolphinWX/MainNoGUI.cpp
index 01994fe..986e914 100644
--- a/Source/Core/DolphinWX/MainNoGUI.cpp
+++ b/Source/Core/DolphinWX/MainNoGUI.cpp
@@ -215,18 +215,19 @@ class PlatformX11 : public Platform
 					key = XLookupKeysym((XKeyEvent*)&event, 0);
 					if (key == XK_Escape)
 					{
-						if (Core::GetState() == Core::CORE_RUN)
-						{
-							if (SConfig::GetInstance().bHideCursor)
-								XUndefineCursor(dpy, win);
-							Core::SetState(Core::CORE_PAUSE);
-						}
-						else
-						{
-							if (SConfig::GetInstance().bHideCursor)
-								XDefineCursor(dpy, win, blankCursor);
-							Core::SetState(Core::CORE_RUN);
-						}
+						running = false;
+						//if (Core::GetState() == Core::CORE_RUN)
+						//{
+						//	if (SConfig::GetInstance().bHideCursor)
+						//		XUndefineCursor(dpy, win);
+						//	Core::SetState(Core::CORE_PAUSE);
+						//}
+						//else
+						//{
+						//	if (SConfig::GetInstance().bHideCursor)
+						//		XDefineCursor(dpy, win, blankCursor);
+						//	Core::SetState(Core::CORE_RUN);
+						//}
 					}
 					else if ((key == XK_Return) && (event.xkey.state & Mod1Mask))
 					{
