Index: gpu-r6p1/ump/Kbuild
===================================================================
--- gpu-r6p1.orig/ump/Kbuild	2016-04-06 14:10:31.000000000 +0800
+++ gpu-r6p1/ump/Kbuild	2016-04-19 15:38:30.559918362 +0800
@@ -55,10 +55,11 @@
 ccflags-y += -DSVN_REV=$(SVN_REV)
 ccflags-y += -DSVN_REV_STRING=\"$(DRIVER_REV)\"
 
-ccflags-y += -I$(src) -I$(src)/common -I$(src)/linux -I$(src)/../mali/common -I$(src)/../mali/linux -I$(src)/include -I$(src)/../../ump/include/ump
+ccflags-y += -I$(src) -I$(src)/common -I$(src)/linux -I$(src)/../mali/common -I$(src)/../mali/linux -I$(src)/../ump/include/ump
 ccflags-y += -DMALI_STATE_TRACKING=0
 ccflags-y += -DMALI_ENABLE_CPU_CYCLES=0
 ccflags-$(CONFIG_UMP_DEBUG) += -DDEBUG
+ccflags-$(CONFIG_MALI400_DEBUG) += -DDEBUG
 
 # For customer releases the Linux Device Drivers will be provided as ARM proprietary and GPL releases:
 # The ARM proprietary product will only include the license/proprietary directory
@@ -88,7 +89,7 @@
 	linux/ump_kernel_random_mapping.o 
 
 ifneq ($(CONFIG_MALI400),y)	
-	$(UDD_FILE_PREFIX)linux/mali_osk_atomics.o \
+ump-y += $(UDD_FILE_PREFIX)linux/mali_osk_atomics.o \
 	$(UDD_FILE_PREFIX)linux/mali_osk_locks.o \
 	$(UDD_FILE_PREFIX)linux/mali_osk_memory.o \
 	$(UDD_FILE_PREFIX)linux/mali_osk_math.o \
