diff --git a/video/out/vo_rpi.c b/video/out/vo_rpi.c
index 94bb30e..554d516 100644
--- a/video/out/vo_rpi.c
+++ b/video/out/vo_rpi.c
@@ -110,7 +110,7 @@ static void *get_proc_address(const GLubyte *name)
     // EGL 1.4 (supported by the RPI firmware) does not necessarily return
     // function pointers for core functions.
     if (!p) {
-        void *h = dlopen("/opt/vc/lib/libbrcmGLESv2.so", RTLD_LAZY);
+        void *h = dlopen("/usr/lib/libGLESv2.so", RTLD_LAZY);
         if (h) {
             p = dlsym(h, name);
             dlclose(h);
