diff --git a/src/video/kmsdrm/SDL_kmsdrmmouse.c b/src/video/kmsdrm/SDL_kmsdrmmouse.c
index b22fc8e..e363be0 100644
--- a/src/video/kmsdrm/SDL_kmsdrmmouse.c
+++ b/src/video/kmsdrm/SDL_kmsdrmmouse.c
@@ -133,7 +133,7 @@ KMSDRM_CreateCursorBO (SDL_VideoDisplay *display) {
 
     dispdata->cursor_bo = KMSDRM_gbm_bo_create(viddata->gbm_dev,
         dispdata->cursor_w, dispdata->cursor_h,
-        GBM_FORMAT_ARGB8888, GBM_BO_USE_CURSOR | GBM_BO_USE_WRITE | GBM_BO_USE_LINEAR);
+        GBM_FORMAT_ARGB8888, GBM_BO_USE_CURSOR | GBM_BO_USE_WRITE);
 
     if (!dispdata->cursor_bo) {
         SDL_SetError("Could not create GBM cursor BO");
