--- a/Makefile	2015-11-02 11:41:55.000000000 +0100
+++ b/Makefile	2015-11-04 19:35:43.936903921 +0100
@@ -1,11 +1,6 @@
 OBJS=raspi2png.o
 BIN=raspi2png
 
-CFLAGS+=-Wall -g -O3 $(shell libpng-config --cflags)
-LDFLAGS+=-L/opt/vc/lib/ -lbcm_host $(shell libpng-config --ldflags) -lm
-
-INCLUDES+=-I/opt/vc/include/ -I/opt/vc/include/interface/vcos/pthreads -I/opt/vc/include/interface/vmcs_host/linux
-
 all: $(BIN)
 
 %.o: %.c
