From 29bcc5a1f939f96f3611f676bb8bb525550525b7 Mon Sep 17 00:00:00 2001
From: Max Kellermann <max@duempel.org>
Date: Mon, 24 Apr 2017 14:38:06 +0200
Subject: [PATCH 20/62] lib/gtest: move fused-src to $(builddir)

Signed-off-by: memeka <mihailescu2m@gmail.com>
---
 lib/gtest/Makefile.am | 14 +++++++-------
 1 file changed, 7 insertions(+), 7 deletions(-)

diff --git a/lib/gtest/Makefile.am b/lib/gtest/Makefile.am
index 3f565fd..886ba73 100644
--- a/lib/gtest/Makefile.am
+++ b/lib/gtest/Makefile.am
@@ -278,15 +278,15 @@ $(FUSED_GTEST_SRC): fused-gtest
 fused-gtest: $(pkginclude_HEADERS) $(pkginclude_internal_HEADERS) \
              $(GTEST_SRC) src/gtest-all.cc src/gtest_main.cc \
              scripts/fuse_gtest_files.py
-	mkdir -p "$(srcdir)/fused-src"
-	chmod -R u+w "$(srcdir)/fused-src"
-	rm -f "$(srcdir)/fused-src/gtest/gtest-all.cc"
-	rm -f "$(srcdir)/fused-src/gtest/gtest.h"
-	"$(srcdir)/scripts/fuse_gtest_files.py" "$(srcdir)/fused-src"
-	cp -f "$(srcdir)/src/gtest_main.cc" "$(srcdir)/fused-src/gtest/"
+	mkdir -p "$(builddir)/fused-src"
+	chmod -R u+w "$(builddir)/fused-src"
+	rm -f "$(builddir)/fused-src/gtest/gtest-all.cc"
+	rm -f "$(builddir)/fused-src/gtest/gtest.h"
+	"$(srcdir)/scripts/fuse_gtest_files.py" "$(builddir)/fused-src"
+	cp -f "$(srcdir)/src/gtest_main.cc" "$(builddir)/fused-src/gtest/"
 
 maintainer-clean-local:
-	rm -rf "$(srcdir)/fused-src"
+	rm -rf "$(builddir)/fused-src"
 endif
 
 # Death tests may produce core dumps in the build directory. In case
-- 
2.7.4

