From 114963e9404b588f941dd2e12d0aaec88c8372bd Mon Sep 17 00:00:00 2001
From: memeka <mihailescu2m@gmail.com>
Date: Mon, 24 Apr 2017 14:38:07 +0200
Subject: [PATCH 27/62] WINDOW_SYSTEM: add Wayland

Signed-off-by: memeka <mihailescu2m@gmail.com>
---
 xbmc/windowing/WinSystem.h | 1 +
 1 file changed, 1 insertion(+)

diff --git a/xbmc/windowing/WinSystem.h b/xbmc/windowing/WinSystem.h
index 08c28a3..e4e86dd 100644
--- a/xbmc/windowing/WinSystem.h
+++ b/xbmc/windowing/WinSystem.h
@@ -31,6 +31,7 @@ typedef enum _WindowSystemType
   WINDOW_SYSTEM_OSX,
   WINDOW_SYSTEM_IOS,
   WINDOW_SYSTEM_X11,
+  WINDOW_SYSTEM_WAYLAND,
   WINDOW_SYSTEM_SDL,
   WINDOW_SYSTEM_EGL,
   WINDOW_SYSTEM_ANDROID
-- 
2.7.4

