Index: linux-6.10/drivers/gpu/drm/drm_modes.c
===================================================================
--- linux-6.10.orig/drivers/gpu/drm/drm_modes.c
+++ linux-6.10/drivers/gpu/drm/drm_modes.c
@@ -48,6 +48,7 @@
 #include <drm/drm_print.h>
 
 #include "drm_crtc_internal.h"
+#include "drm_modes_low_dotclock.h"
 
 /**
  * drm_mode_debug_printmodeline - print a mode to dmesg
@@ -2026,7 +2027,7 @@ static int drm_mode_parse_cmdline_res_mo
 					   struct drm_cmdline_mode *mode)
 {
 	const char *str_start = str;
-	bool rb = false, cvt = false;
+	bool rb = false, cvt = false, low_dotclock = false;
 	int xres = 0, yres = 0;
 	int remaining, i;
 	char *end_ptr;
@@ -2056,6 +2057,12 @@ static int drm_mode_parse_cmdline_res_mo
 		case 'R':
 			rb = true;
 			break;
+		case 'S':
+		case 'c':
+		case 'z':
+			low_dotclock = true;
+			DRM_DEBUG_KMS("Found one of the S/c/z low dotclock mode flag");
+			break;
 		default:
 			/*
 			 * Try to pass that to our extras parsing
@@ -2080,6 +2087,7 @@ static int drm_mode_parse_cmdline_res_mo
 	mode->yres = yres;
 	mode->cvt = cvt;
 	mode->rb = rb;
+	mode->low_dotclock = low_dotclock;
 
 	return 0;
 }
@@ -2543,7 +2551,11 @@ drm_mode_create_from_cmdline_mode(struct
 
 	if (strlen(cmd->name))
 		mode = drm_named_mode(dev, cmd);
-	else if (cmd->cvt)
+	else if (cmd->low_dotclock)
+        mode = drm_mode_low_dotclock_res(dev,
+                                    cmd->xres, cmd->yres,
+                                    cmd->interlace);
+        else if (cmd->cvt)
 		mode = drm_cvt_mode(dev,
 				    cmd->xres, cmd->yres,
 				    cmd->refresh_specified ? cmd->refresh : 60,
Index: linux-6.10/drivers/gpu/drm/drm_modes_low_dotclock.c
===================================================================
--- /dev/null
+++ linux-6.10/drivers/gpu/drm/drm_modes_low_dotclock.c
@@ -0,0 +1,156 @@
+/*
+ * Permission is hereby granted, free of charge, to any person obtaining a
+ * copy of this software and associated documentation files (the "Software"),
+ * to deal in the Software without restriction, including without limitation
+ * the rights to use, copy, modify, merge, publish, distribute, sublicense,
+ * and/or sell copies of the Software, and to permit persons to whom the
+ * Software is furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice (including the next
+ * paragraph) shall be included in all copies or substantial portions of the
+ * Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.  IN NO EVENT SHALL
+ * VA LINUX SYSTEMS AND/OR ITS SUPPLIERS BE LIABLE FOR ANY CLAIM, DAMAGES OR
+ * OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE,
+ * ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+ * OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+#include <drm/drm_modes.h>
+#include <drm/drm_print.h>
+#include "drm_modes_low_dotclock.h"
+
+static struct drm_display_mode drm_low_dotclock_modes[] = {
+/* 15 kHz modes */
+	/* 320x240 60.00 - generic_15 */
+		/* Switchres: Modeline "320x240_60 15.660000KHz 60.000000Hz" 6.514560 320 333 364 416 240 241 244 261 -hsync -vsync */
+		{ DRM_MODE("320x240", DRM_MODE_TYPE_DRIVER,
+			6514, 320, 333, 364, 416, 0, 240, 241, 244, 261, 0,
+			DRM_MODE_FLAG_NHSYNC | DRM_MODE_FLAG_NVSYNC) },
+	/* 384x288 50.00 - generic_15 */
+		/* Switchres: Modeline "384x288_50 15.650000KHz 50.000000Hz" 7.809350 384 400 437 499 288 291 294 313 -hsync -vsync */
+		{ DRM_MODE("384x288", DRM_MODE_TYPE_DRIVER,
+			7809, 384, 400, 437, 499, 0, 288, 291, 294, 313, 0,
+			DRM_MODE_FLAG_NHSYNC | DRM_MODE_FLAG_NVSYNC) },
+	/* 640x240 60.00 - generic_15 */
+		/* Switchres: Modeline "640x240_60 15.660000KHz 60.000000Hz" 13.013460 640 666 727 831 240 241 244 261 -hsync -vsync */
+		{ DRM_MODE("640x240", DRM_MODE_TYPE_DRIVER,
+			13013, 640, 666, 727, 831, 0, 240, 241, 244, 261, 0,
+			DRM_MODE_FLAG_NHSYNC | DRM_MODE_FLAG_NVSYNC) },
+	/* 640x480 60.00 - generic_15 */
+		/* Switchres: Modeline "640x480_60i 15.690000KHz 60.000000Hz" 13.038390 640 666 727 831 480 483 489 523 interlace -hsync -vsync */
+		{ DRM_MODE("640x480i", DRM_MODE_TYPE_DRIVER,
+			13038, 640, 666, 727, 831, 0, 480, 483, 489, 523, 0,
+			DRM_MODE_FLAG_NHSYNC | DRM_MODE_FLAG_NVSYNC |
+		DRM_MODE_FLAG_INTERLACE) },
+       /* 648x480 60.00 - generic_15 */
+               /* Switchres: Modeline "648x480_60i 15.690000KHz 60.000000Hz" 13.210980 648 674 736 842 480 483 489 523 interlace -hsync -vsync */
+               { DRM_MODE("648x480i", DRM_MODE_TYPE_DRIVER,
+                       13210, 648, 674, 736, 842, 0, 480, 483, 489, 523, 0,
+                       DRM_MODE_FLAG_NHSYNC | DRM_MODE_FLAG_NVSYNC |
+               DRM_MODE_FLAG_INTERLACE) },
+	/* 720x480 59.95 - generic_15 */
+		/* Switchres: Modeline "720x480_59i 15.676925KHz 59.950001Hz" 14.657925 720 749 818 935 480 483 489 523 interlace -hsync -vsync */
+		{ DRM_MODE("720x480i", DRM_MODE_TYPE_DRIVER,
+			14657, 720, 749, 818, 935, 0, 480, 483, 489, 523, 0,
+			DRM_MODE_FLAG_NHSYNC | DRM_MODE_FLAG_NVSYNC |
+		DRM_MODE_FLAG_INTERLACE) },
+	/* 768x576 50.00 - generic_15 */
+		/* Switchres: Modeline "768x576_50i 15.675000KHz 50.000000Hz" 15.627975 768 799 872 997 576 583 589 627 interlace -hsync -vsync */
+		{ DRM_MODE("768x576i", DRM_MODE_TYPE_DRIVER,
+			15627, 768, 799, 872, 997, 0, 576, 583, 589, 627, 0,
+			DRM_MODE_FLAG_NHSYNC | DRM_MODE_FLAG_NVSYNC |
+		DRM_MODE_FLAG_INTERLACE) },
+	/* 800x576 50.00 - generic_15 */
+		/* Switchres: Modeline "800x576_50i 15.675000KHz 50.000000Hz" 16.302000 800 833 910 1040 576 583 589 627 interlace -hsync -vsync */
+		{ DRM_MODE("800x576i", DRM_MODE_TYPE_DRIVER,
+			16302, 800, 833, 910, 1040, 0, 576, 583, 589, 627, 0,
+			DRM_MODE_FLAG_NHSYNC | DRM_MODE_FLAG_NVSYNC |
+		DRM_MODE_FLAG_INTERLACE) },
+	/* 1280x480 60.00 - generic_15 */
+		/* Switchres: Modeline "1280x480_60i 15.690000KHz 60.000000Hz" 26.108160 1280 1332 1455 1664 480 483 489 523 interlace -hsync -vsync */
+		{ DRM_MODE("1280x480i", DRM_MODE_TYPE_DRIVER,
+			26108, 1280, 1332, 1455, 1664, 0, 480, 483, 489, 523, 0,
+			DRM_MODE_FLAG_NHSYNC | DRM_MODE_FLAG_NVSYNC |
+		DRM_MODE_FLAG_INTERLACE) },
+
+/* 25 kHz modes */
+	/* 512x384 58.59 - arcade_25ex */
+		/* Switchres: Modeline "512x384_58 24.959340KHz 58.590000Hz" 15.973977 512 525 589 640 384 391 396 426 -hsync -vsync */
+		{ DRM_MODE("512x384", DRM_MODE_TYPE_DRIVER,
+			15973, 512, 525, 589, 640, 0, 384, 391, 396, 426, 0,
+			DRM_MODE_FLAG_NHSYNC | DRM_MODE_FLAG_NVSYNC) },
+	/* 800x600 60.00 - arcade_25ex */
+		/* Switchres: Modeline "800x600_60i 24.930000KHz 60.000000Hz" 24.930000 800 820 920 1000 600 687 697 831 interlace -hsync -vsync */
+		{ DRM_MODE("800x600i", DRM_MODE_TYPE_DRIVER,
+			24930, 800, 820, 920, 1000, 0, 600, 687, 697, 831, 0,
+			DRM_MODE_FLAG_NHSYNC | DRM_MODE_FLAG_NVSYNC |
+		DRM_MODE_FLAG_INTERLACE) },
+	/* 1024x768 50.00 - arcade_25ex */
+		/* Switchres: Modeline "1024x768_50i 24.975000KHz 50.000000Hz" 31.968000 1024 1050 1178 1280 768 855 865 999 interlace -hsync -vsync */
+		{ DRM_MODE("1024x768i", DRM_MODE_TYPE_DRIVER,
+			31968, 1024, 1050, 1178, 1280, 0, 768, 855, 865, 999, 0,
+			DRM_MODE_FLAG_NHSYNC | DRM_MODE_FLAG_NVSYNC |
+		DRM_MODE_FLAG_INTERLACE) },
+	/* 1024x240 60.00 - arcade_25ex */
+		/* Switchres: Modeline "1024x240_60 24.900000KHz 60.000000Hz" 31.822200 1024 1049 1176 1278 240 313 318 415 -hsync -vsync */
+		{ DRM_MODE("1024x240", DRM_MODE_TYPE_DRIVER,
+			31822, 1024, 1049, 1176, 1278, 0, 240, 313, 318, 415, 0,
+			DRM_MODE_FLAG_NHSYNC | DRM_MODE_FLAG_NVSYNC) },
+	/* 1280x240 60.00 - arcade_25ex */
+		/* Switchres: Modeline "1280x240_60 24.900000KHz 60.000000Hz" 39.790200 1280 1312 1471 1598 240 313 318 415 -hsync -vsync */
+		{ DRM_MODE("1280x240", DRM_MODE_TYPE_DRIVER,
+			39790, 1280, 1312, 1471, 1598, 0, 240, 313, 318, 415, 0,
+			DRM_MODE_FLAG_NHSYNC | DRM_MODE_FLAG_NVSYNC) },
+
+/* 31 kHz modes */
+	/* 640x480 60.00 - arcade_31 */
+		/* Switchres: Modeline "640x480_60 31.500000KHz 60.000000Hz" 25.452000 640 664 760 808 480 491 493 525 -hsync -vsync */
+		{ DRM_MODE("640x480", DRM_MODE_TYPE_DRIVER,
+			25452, 640, 664, 760, 808, 0, 480, 491, 493, 525, 0,
+			DRM_MODE_FLAG_NHSYNC | DRM_MODE_FLAG_NVSYNC) },
+};
+
+/**
+ * drm_display_mode - get a fixed modeline
+ * @dev: drm device
+ * @hdisplay: hdisplay size
+ * @vdisplay: vdisplay size
+ * @interlaced: whether to compute an interlaced mode
+ *
+ * This function returns a modeline among predefined low dotclock modes. These are fixed
+ * modelines, until automatic mode generation gets added into kernel.
+ * No need to specify 15 or 25kHz, not the vertical refresh rate as it's only 60Hz for now.
+ * This is a very basic function. Duplicate modes (320x240 and 1024x768) are not
+ * handled, the first result will be returned.
+ *
+ * Returns:
+ * A low dotclock drm modeline
+ */
+struct drm_display_mode *drm_mode_low_dotclock_res(struct drm_device *dev,
+				int hsize, int vsize, bool interlace)
+{
+	int i;
+
+	DRM_DEBUG_KMS("Entering drm_mode_low_dotclock_res for resolution %dx%d (interlace: %s)", hsize, vsize, interlace ? "true" : "false");
+	for (i = 0; i < ARRAY_SIZE(drm_low_dotclock_modes); i++) {
+		const struct drm_display_mode *ptr = &drm_low_dotclock_modes[i];
+		if (hsize != ptr->hdisplay)
+			continue;
+		if (vsize != ptr->vdisplay)
+			continue;
+		//if ((refresh != 0) && (refresh != drm_mode_vrefresh(ptr)))
+		//	continue;
+		if (((! interlace) && (ptr->flags & DRM_MODE_FLAG_INTERLACE)) \
+		    || ((interlace) && ! (ptr->flags & DRM_MODE_FLAG_INTERLACE)))
+			continue;
+		DRM_DEBUG_KMS("Found a low dotclock mode for %dx%d (interlace: %d)", hsize, vsize, interlace);
+		drm_mode_debug_printmodeline(ptr);
+		return drm_mode_duplicate(dev, ptr);
+	}
+	return NULL;
+}
+EXPORT_SYMBOL(drm_mode_low_dotclock_res);
Index: linux-6.10/drivers/gpu/drm/drm_modes_low_dotclock.h
===================================================================
--- /dev/null
+++ linux-6.10/drivers/gpu/drm/drm_modes_low_dotclock.h
@@ -0,0 +1,23 @@
+/*
+ * Permission is hereby granted, free of charge, to any person obtaining a
+ * copy of this software and associated documentation files (the "Software"),
+ * to deal in the Software without restriction, including without limitation
+ * the rights to use, copy, modify, merge, publish, distribute, sublicense,
+ * and/or sell copies of the Software, and to permit persons to whom the
+ * Software is furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice (including the next
+ * paragraph) shall be included in all copies or substantial portions of the
+ * Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.  IN NO EVENT SHALL
+ * VA LINUX SYSTEMS AND/OR ITS SUPPLIERS BE LIABLE FOR ANY CLAIM, DAMAGES OR
+ * OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE,
+ * ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+ * OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+struct drm_display_mode *drm_mode_low_dotclock_res(struct drm_device *dev,
+				int hsize, int vsize, bool interlace);
Index: linux-6.10/drivers/gpu/drm/drm_probe_helper.c
===================================================================
--- linux-6.10.orig/drivers/gpu/drm/drm_probe_helper.c
+++ linux-6.10/drivers/gpu/drm/drm_probe_helper.c
@@ -459,21 +459,26 @@ static int __drm_helper_update_and_valid
 		mode_flags |= DRM_MODE_FLAG_3D_MASK;
 
 	list_for_each_entry(mode, &connector->modes, head) {
+		DRM_DEBUG_KMS("[CONNECTOR:%d:%s] starting with mode state : %d\n", connector->base.id, connector->name, mode->status);
 		if (mode->status != MODE_OK)
 			continue;
 
+		DRM_DEBUG_KMS("[CONNECTOR:%d:%s] before drm_mode_validate_driver mode is : %d\n", connector->base.id, connector->name, mode->status);
 		mode->status = drm_mode_validate_driver(dev, mode);
 		if (mode->status != MODE_OK)
 			continue;
 
+		DRM_DEBUG_KMS("[CONNECTOR:%d:%s] before drm_mode_validate_size mode is : %d\n", connector->base.id, connector->name, mode->status);
 		mode->status = drm_mode_validate_size(mode, maxX, maxY);
 		if (mode->status != MODE_OK)
 			continue;
 
+		DRM_DEBUG_KMS("[CONNECTOR:%d:%s] before drm_mode_validate_flag mode is : %d\n", connector->base.id, connector->name, mode->status);
 		mode->status = drm_mode_validate_flag(mode, mode_flags);
 		if (mode->status != MODE_OK)
 			continue;
 
+		DRM_DEBUG_KMS("[CONNECTOR:%d:%s] before drm_mode_validate_pipeline mode is : %d\n", connector->base.id, connector->name, mode->status);
 		ret = drm_mode_validate_pipeline(mode, connector, ctx,
 						 &mode->status);
 		if (ret) {
@@ -487,9 +492,11 @@ static int __drm_helper_update_and_valid
 				return -EDEADLK;
 		}
 
+		DRM_DEBUG_KMS("[CONNECTOR:%d:%s] before drm_mode_validate_ycbcr420 mode is : %d\n", connector->base.id, connector->name, mode->status);
 		if (mode->status != MODE_OK)
 			continue;
 		mode->status = drm_mode_validate_ycbcr420(mode, connector);
+		DRM_DEBUG_KMS("[CONNECTOR:%d:%s] in the end mode is : %d\n", connector->base.id, connector->name, mode->status);
 	}
 
 	return 0;
@@ -559,6 +566,7 @@ int drm_helper_probe_single_connector_mo
 {
 	struct drm_device *dev = connector->dev;
 	struct drm_display_mode *mode;
+	struct drm_cmdline_mode *cmdline_mode;
 	int count = 0, ret;
 	enum drm_connector_status old_status;
 	struct drm_modeset_acquire_ctx ctx;
@@ -644,10 +652,13 @@ retry:
 		goto exit;
 	}
 
+	cmdline_mode = &connector->cmdline_mode;
+
 	count = drm_helper_probe_get_modes(connector);
 
 	if (count == 0 && (connector->status == connector_status_connected ||
-			   connector->status == connector_status_unknown)) {
+			   connector->status == connector_status_unknown) &&
+			   ! cmdline_mode->low_dotclock) {
 		count = drm_add_modes_noedid(connector, 1024, 768);
 
 		/*
Index: linux-6.10/include/drm/drm_connector.h
===================================================================
--- linux-6.10.orig/include/drm/drm_connector.h
+++ linux-6.10/include/drm/drm_connector.h
@@ -1434,6 +1434,13 @@ struct drm_cmdline_mode {
 	bool interlace;
 
 	/**
+	 * @low_dotclock:
+	 *
+	 * Flag is enabled if a valid low dotclock resolution is defined.
+	 */
+	bool low_dotclock;
+
+	/**
 	 * @cvt:
 	 *
 	 * The timings will be calculated using the VESA Coordinated
Index: linux-6.10/drivers/gpu/drm/Makefile
===================================================================
--- linux-6.10.orig/drivers/gpu/drm/Makefile
+++ linux-6.10/drivers/gpu/drm/Makefile
@@ -59,6 +59,7 @@ drm-y := \
 	drm_ioctl.o \
 	drm_lease.o \
 	drm_managed.o \
+	drm_modes_low_dotclock.o \
 	drm_mm.o \
 	drm_mode_config.o \
 	drm_mode_object.o \
