Index: linux-6.10/drivers/gpu/drm/amd/display/amdgpu_dm/amdgpu_dm.c
===================================================================
--- linux-6.10.orig/drivers/gpu/drm/amd/display/amdgpu_dm/amdgpu_dm.c
+++ linux-6.10/drivers/gpu/drm/amd/display/amdgpu_dm/amdgpu_dm.c
@@ -5699,6 +5699,8 @@ static void fill_stream_properties_from_
 			timing_out->flags.HSYNC_POSITIVE_POLARITY = 1;
 		if (mode_in->flags & DRM_MODE_FLAG_PVSYNC)
 			timing_out->flags.VSYNC_POSITIVE_POLARITY = 1;
+		if (mode_in->flags & DRM_MODE_FLAG_INTERLACE)
+			timing_out->flags.INTERLACE = 1;
 	}
 
 	if (stream->signal == SIGNAL_TYPE_HDMI_TYPE_A) {
@@ -6298,6 +6300,8 @@ create_stream_for_sink(struct drm_connec
 
 	if (recalculate_timing)
 		drm_mode_set_crtcinfo(&saved_mode, 0);
+	else if (!dm_state || mode.flags & DRM_MODE_FLAG_INTERLACE)
+		drm_mode_set_crtcinfo(&mode, 0);
 
 	/*
 	 * If scaling is enabled and refresh rate didn't change
@@ -6939,8 +6943,7 @@ enum drm_mode_status amdgpu_dm_connector
 	struct dc_stream_state *stream;
 	struct amdgpu_dm_connector *aconnector = to_amdgpu_dm_connector(connector);
 
-	if ((mode->flags & DRM_MODE_FLAG_INTERLACE) ||
-			(mode->flags & DRM_MODE_FLAG_DBLSCAN))
+	if (mode->flags & DRM_MODE_FLAG_DBLSCAN)
 		return result;
 
 	/*
@@ -7647,7 +7650,7 @@ void amdgpu_dm_connector_init_helper(str
 	aconnector->connector_id = link_index;
 	aconnector->bl_idx = -1;
 	aconnector->dc_link = link;
-	aconnector->base.interlace_allowed = false;
+	aconnector->base.interlace_allowed = true;
 	aconnector->base.doublescan_allowed = false;
 	aconnector->base.stereo_allowed = false;
 	aconnector->base.dpms = DRM_MODE_DPMS_OFF;
Index: linux-6.10/drivers/gpu/drm/amd/display/dc/hwss/dcn10/dcn10_hwseq.c
===================================================================
--- linux-6.10.orig/drivers/gpu/drm/amd/display/dc/hwss/dcn10/dcn10_hwseq.c
+++ linux-6.10/drivers/gpu/drm/amd/display/dc/hwss/dcn10/dcn10_hwseq.c
@@ -2765,6 +2765,9 @@ static void update_scaler(struct pipe_ct
 	pipe_ctx->plane_res.scl_data.lb_params.alpha_en = per_pixel_alpha;
 	pipe_ctx->plane_res.scl_data.lb_params.depth = LB_PIXEL_DEPTH_36BPP;
 	/* scaler configuration */
+	pipe_ctx->plane_res.scl_data.lb_params.interleave_en =
+			pipe_ctx->stream->timing.flags.INTERLACE? 1:0;
+
 	pipe_ctx->plane_res.dpp->funcs->dpp_set_scaler(
 			pipe_ctx->plane_res.dpp, &pipe_ctx->plane_res.scl_data);
 }
@@ -3791,8 +3794,7 @@ int dcn10_get_vupdate_offset_from_vsync(
 
 	asic_blank_end = (patched_crtc_timing.v_total -
 			vesa_sync_start -
-			patched_crtc_timing.v_border_top)
-			* interlace_factor;
+			patched_crtc_timing.v_border_top); // * interlace_factor;
 
 	return asic_blank_end -
 			pipe_ctx->pipe_dlg_param.vstartup_start + 1;
Index: linux-6.10/drivers/gpu/drm/amd/display/dc/optc/dcn10/dcn10_optc.c
===================================================================
--- linux-6.10.orig/drivers/gpu/drm/amd/display/dc/optc/dcn10/dcn10_optc.c
+++ linux-6.10/drivers/gpu/drm/amd/display/dc/optc/dcn10/dcn10_optc.c
@@ -608,10 +608,6 @@ bool optc1_validate_timing(
 		timing->timing_3d_format != TIMING_3D_FORMAT_INBAND_FA)
 		return false;
 
-	/* Temporarily blocking interlacing mode until it's supported */
-	if (timing->flags.INTERLACE == 1)
-		return false;
-
 	/* Check maximum number of pixels supported by Timing Generator
 	 * (Currently will never fail, in order to fail needs display which
 	 * needs more than 8192 horizontal and
Index: linux-6.10/drivers/gpu/drm/amd/display/dc/hwss/dcn20/dcn20_hwseq.c
===================================================================
--- linux-6.10.orig/drivers/gpu/drm/amd/display/dc/hwss/dcn20/dcn20_hwseq.c
+++ linux-6.10/drivers/gpu/drm/amd/display/dc/hwss/dcn20/dcn20_hwseq.c
@@ -1733,6 +1733,9 @@ static void dcn20_update_dchubp_dpp(
 		pipe_ctx->plane_res.scl_data.lb_params.alpha_en = pipe_ctx->plane_state->per_pixel_alpha;
 		ASSERT(pipe_ctx->plane_res.scl_data.lb_params.depth == LB_PIXEL_DEPTH_36BPP);
 		/* scaler configuration */
+		pipe_ctx->plane_res.scl_data.lb_params.interleave_en =
+			pipe_ctx->stream->timing.flags.INTERLACE? 1:0;
+
 		pipe_ctx->plane_res.dpp->funcs->dpp_set_scaler(
 				pipe_ctx->plane_res.dpp, &pipe_ctx->plane_res.scl_data);
 	}
Index: linux-6.10/drivers/gpu/drm/amd/display/dc/dml/calcs/dcn_calc_auto.c
===================================================================
--- linux-6.10.orig/drivers/gpu/drm/amd/display/dc/dml/calcs/dcn_calc_auto.c
+++ linux-6.10/drivers/gpu/drm/amd/display/dc/dml/calcs/dcn_calc_auto.c
@@ -61,7 +61,8 @@ void scaler_settings_calculation(struct
 			v->v_ratio[k] = v->h_ratio[k];
 		}
 		if (v->interlace_output[k] == 1.0) {
-			v->v_ratio[k] = 2.0 * v->v_ratio[k];
+			// Commented out: this makes scale and ratio taps validation fail
+			//v->v_ratio[k] = 2.0 * v->v_ratio[k];
 		}
 		if (v->underscan_output[k] == 1.0) {
 			v->h_ratio[k] = v->h_ratio[k] * v->under_scan_factor;
Index: linux-6.10/drivers/gpu/drm/amd/display/dc/dml/calcs/dcn_calcs.c
===================================================================
--- linux-6.10.orig/drivers/gpu/drm/amd/display/dc/dml/calcs/dcn_calcs.c
+++ linux-6.10/drivers/gpu/drm/amd/display/dc/dml/calcs/dcn_calcs.c
@@ -446,6 +446,7 @@ static void pipe_ctx_to_e2e_pipe_params
 	input->dest.vupdate_offset = pipe->pipe_dlg_param.vupdate_offset;
 	input->dest.vupdate_offset = pipe->pipe_dlg_param.vupdate_offset;
 	input->dest.vupdate_width = pipe->pipe_dlg_param.vupdate_width;
+	input->dest.interlaced = pipe->stream->timing.flags.INTERLACE;
 
 }
 
@@ -900,7 +901,7 @@ bool dcn_validate_bandwidth(
 			continue;
 
 		v->underscan_output[input_idx] = false; /* taken care of in recout already*/
-		v->interlace_output[input_idx] = false;
+		v->interlace_output[input_idx] = pipe->stream->timing.flags.INTERLACE? 1.0 : false;
 
 		v->htotal[input_idx] = pipe->stream->timing.h_total;
 		v->vtotal[input_idx] = pipe->stream->timing.v_total;
@@ -1219,14 +1220,12 @@ bool dcn_validate_bandwidth(
 
 			asic_blank_end = (pipe->stream->timing.v_total -
 						vesa_sync_start -
-						pipe->stream->timing.v_border_top)
-			* (pipe->stream->timing.flags.INTERLACE ? 1 : 0);
+						pipe->stream->timing.v_border_top);
 
 			asic_blank_start = asic_blank_end +
 						(pipe->stream->timing.v_border_top +
 						pipe->stream->timing.v_addressable +
-						pipe->stream->timing.v_border_bottom)
-			* (pipe->stream->timing.flags.INTERLACE ? 1 : 0);
+						pipe->stream->timing.v_border_bottom);
 
 			pipe->pipe_dlg_param.vblank_start = asic_blank_start;
 			pipe->pipe_dlg_param.vblank_end = asic_blank_end;
Index: linux-6.10/drivers/gpu/drm/amd/display/dc/dml/display_mode_vba.c
===================================================================
--- linux-6.10.orig/drivers/gpu/drm/amd/display/dc/dml/display_mode_vba.c
+++ linux-6.10/drivers/gpu/drm/amd/display/dc/dml/display_mode_vba.c
@@ -592,9 +592,11 @@ static void fetch_pipe_params(struct dis
 		mode_lib->vba.VRatioChroma[mode_lib->vba.NumberOfActivePlanes] = scl->vscl_ratio_c;
 		mode_lib->vba.ScalerEnabled[mode_lib->vba.NumberOfActivePlanes] = scl->scl_enable;
 		mode_lib->vba.Interlace[mode_lib->vba.NumberOfActivePlanes] = dst->interlaced;
+
 		if (dst->interlaced && !ip->ptoi_supported) {
-			mode_lib->vba.VRatio[mode_lib->vba.NumberOfActivePlanes] *= 2.0;
-			mode_lib->vba.VRatioChroma[mode_lib->vba.NumberOfActivePlanes] *= 2.0;
+			// Commented out: this makes scale and ratio taps validation fail
+			//mode_lib->vba.VRatio[mode_lib->vba.NumberOfActivePlanes] *= 2.0;
+			//mode_lib->vba.VRatioChroma[mode_lib->vba.NumberOfActivePlanes] *= 2.0;
 		}
 		mode_lib->vba.htaps[mode_lib->vba.NumberOfActivePlanes] = taps->htaps;
 		mode_lib->vba.vtaps[mode_lib->vba.NumberOfActivePlanes] = taps->vtaps;
Index: linux-6.10/drivers/gpu/drm/drm_probe_helper.c
===================================================================
--- linux-6.10.orig/drivers/gpu/drm/drm_probe_helper.c
+++ linux-6.10/drivers/gpu/drm/drm_probe_helper.c
@@ -163,6 +163,9 @@ static int drm_helper_probe_add_cmdline_
 				continue;
 		}
 
+		if ((mode->flags & DRM_MODE_FLAG_INTERLACE) != cmdline_mode->interlace)
+			continue;
+
 		/* Mark the matching mode as being preferred by the user */
 		mode->type |= DRM_MODE_TYPE_USERDEF;
 		return 0;
