diff -Naur linux-6.0_stock/drivers/gpu/drm/amd/amdgpu/amdgpu_pll.c linux-6.0_15kHz/drivers/gpu/drm/amd/amdgpu/amdgpu_pll.c
--- linux-6.0_stock/drivers/gpu/drm/amd/amdgpu/amdgpu_pll.c	1970-01-01 01:00:00.000000000 +0100
+++ linux-6.0_15kHz/drivers/gpu/drm/amd/amdgpu/amdgpu_pll.c	1970-01-01 01:00:00.000000000 +0100
@@ -89,9 +89,9 @@
 
 	/* limit reference * post divider to a maximum */
 	if (adev->family == AMDGPU_FAMILY_SI)
-		ref_div_max = min(100 / post_div, ref_div_max);
+		ref_div_max = max(min(100 / post_div, ref_div_max), 1u);
 	else
-		ref_div_max = min(128 / post_div, ref_div_max);
+		ref_div_max = max(min(128 / post_div, ref_div_max), 1u);
 
 	/* get matching reference and feedback divider */
 	*ref_div = min(max(DIV_ROUND_CLOSEST(den, post_div), 1u), ref_div_max);
