Index: linux-6.4/drivers/gpu/drm/amd/amdgpu/amdgpu_pll.c
===================================================================
--- linux-6.4.orig/drivers/gpu/drm/amd/amdgpu/amdgpu_pll.c
+++ linux-6.4/drivers/gpu/drm/amd/amdgpu/amdgpu_pll.c
@@ -89,9 +89,9 @@ static void amdgpu_pll_get_fb_ref_div(st
 
 	/* limit reference * post divider to a maximum */
 	if (adev->family == AMDGPU_FAMILY_SI)
-		ref_div_max = min(100 / post_div, ref_div_max);
+		ref_div_max = max(min(100 / post_div, ref_div_max), 1u);
 	else
-		ref_div_max = min(128 / post_div, ref_div_max);
+		ref_div_max = max(min(128 / post_div, ref_div_max), 1u);
 
 	/* get matching reference and feedback divider */
 	*ref_div = min(max(DIV_ROUND_CLOSEST(den, post_div), 1u), ref_div_max);
