diff --git a/drivers/hid/hid-steam.c b/drivers/hid/hid-steam.c
index 0f364d43a..7893b4320 100644
--- a/drivers/hid/hid-steam.c
+++ b/drivers/hid/hid-steam.c
@@ -1165,8 +1165,8 @@ static int steam_input_register(struct steam_device *steam)
 			STEAM_PAD_FUZZ, 0);
 
 	if (steam->quirks & (STEAM_QUIRK_DECK | STEAM_QUIRK_IBEX)) {
-		input_set_abs_params(input, ABS_HAT2Y, 0, 32767, 0, 0);
-		input_set_abs_params(input, ABS_HAT2X, 0, 32767, 0, 0);
+		input_set_abs_params(input, ABS_Z, 0, 32767, 0, 0);
+		input_set_abs_params(input, ABS_RZ, 0, 32767, 0, 0);
 
 		input_set_abs_params(input, ABS_RX, -32767, 32767, 0, 0);
 		input_set_abs_params(input, ABS_RY, -32767, 32767, 0, 0);
@@ -1182,11 +1182,11 @@ static int steam_input_register(struct steam_device *steam)
 		input_abs_set_res(input, ABS_RY, STEAM_DECK_JOYSTICK_RESOLUTION);
 		input_abs_set_res(input, ABS_HAT1X, STEAM_PAD_RESOLUTION);
 		input_abs_set_res(input, ABS_HAT1Y, STEAM_PAD_RESOLUTION);
-		input_abs_set_res(input, ABS_HAT2Y, STEAM_DECK_TRIGGER_RESOLUTION);
-		input_abs_set_res(input, ABS_HAT2X, STEAM_DECK_TRIGGER_RESOLUTION);
+		input_abs_set_res(input, ABS_Z, STEAM_DECK_TRIGGER_RESOLUTION);
+		input_abs_set_res(input, ABS_RZ, STEAM_DECK_TRIGGER_RESOLUTION);
 	} else {
-		input_set_abs_params(input, ABS_HAT2Y, 0, 255, 0, 0);
-		input_set_abs_params(input, ABS_HAT2X, 0, 255, 0, 0);
+		input_set_abs_params(input, ABS_Z, 0, 255, 0, 0);
+		input_set_abs_params(input, ABS_RZ, 0, 255, 0, 0);
 
 		input_set_abs_params(input, ABS_RX, -32767, 32767,
 				STEAM_PAD_FUZZ, 0);
@@ -1197,8 +1197,8 @@ static int steam_input_register(struct steam_device *steam)
 		input_abs_set_res(input, ABS_Y, STEAM_JOYSTICK_RESOLUTION);
 		input_abs_set_res(input, ABS_RX, STEAM_PAD_RESOLUTION);
 		input_abs_set_res(input, ABS_RY, STEAM_PAD_RESOLUTION);
-		input_abs_set_res(input, ABS_HAT2Y, STEAM_TRIGGER_RESOLUTION);
-		input_abs_set_res(input, ABS_HAT2X, STEAM_TRIGGER_RESOLUTION);
+		input_abs_set_res(input, ABS_Z, STEAM_TRIGGER_RESOLUTION);
+		input_abs_set_res(input, ABS_RZ, STEAM_TRIGGER_RESOLUTION);
 	}
 	input_abs_set_res(input, ABS_HAT0X, STEAM_PAD_RESOLUTION);
 	input_abs_set_res(input, ABS_HAT0Y, STEAM_PAD_RESOLUTION);
@@ -1550,8 +1550,8 @@ static void steam_mode_switch_cb(struct work_struct *work)
 			input_report_abs(input, ABS_RY, 0);
 			input_report_abs(input, ABS_HAT0X, 0);
 			input_report_abs(input, ABS_HAT0Y, 0);
-			input_report_abs(input, ABS_HAT2Y, 0);
-			input_report_abs(input, ABS_HAT2X, 0);
+			input_report_abs(input, ABS_Z, 0);
+			input_report_abs(input, ABS_RZ, 0);
 
 			if (steam->quirks & (STEAM_QUIRK_DECK | STEAM_QUIRK_IBEX)) {
 				input_report_key(input, BTN_BASE, 0);
@@ -1938,8 +1938,8 @@ static void steam_map_axes(struct input_dev *input,
  * -------+-------+-----------+--------------------------
  *  4-7   | u32   | --        | sequence number
  *  8-10  | 24bit | see below | buttons
- *  11    | u8    | ABS_HAT2Y | left trigger
- *  12    | u8    | ABS_HAT2X | right trigger
+ *  11    | u8    | ABS_Z     | left trigger
+ *  12    | u8    | ABS_RZ    | right trigger
  *  13-15 | --    | --        | always 0
  *  16-17 | s16   | ABS_X/ABS_HAT0X     | X value
  *  18-19 | s16   | ABS_Y/ABS_HAT0Y     | Y value
@@ -2041,8 +2041,8 @@ static void steam_do_input_event(struct steam_device *steam,
 	s16 x, y;
 	bool lpad_touched, lpad_and_joy;
 
-	input_report_abs(input, ABS_HAT2Y, data[11]);
-	input_report_abs(input, ABS_HAT2X, data[12]);
+	input_report_abs(input, ABS_Z, data[11]);
+	input_report_abs(input, ABS_RZ, data[12]);
 
 	/*
 	 * These two bits tells how to interpret the values X and Y.
@@ -2109,8 +2109,8 @@ static void steam_do_sensors_event(struct steam_device *steam,
  *  38-39 | s16   | --        | quaternion X value
  *  40-41 | s16   | --        | quaternion Y value
  *  42-43 | s16   | --        | quaternion Z value
- *  44-45 | u16   | ABS_HAT2Y | left trigger (uncalibrated)
- *  46-47 | u16   | ABS_HAT2X | right trigger (uncalibrated)
+ *  44-45 | u16   | ABS_Z     | left trigger (uncalibrated)
+ *  46-47 | u16   | ABS_RZ    | right trigger (uncalibrated)
  *  48-49 | s16   | ABS_X     | left joystick X
  *  50-51 | s16   | ABS_Y     | left joystick Y
  *  52-53 | s16   | ABS_RX    | right joystick X
@@ -2220,8 +2220,8 @@ static const struct steam_axis_mapping steam_deck_axis_mappings[] = {
 	{ ABS_Y,	-1, 50 },
 	{ ABS_RX,	 1, 52 },
 	{ ABS_RY,	-1, 54 },
-	{ ABS_HAT2Y,	 1, 44 },
-	{ ABS_HAT2X,	 1, 46 },
+	{ ABS_Z,	 1, 44 },
+	{ ABS_RZ,	 1, 46 },
 	{ /* sentinel */ },
 };
 
@@ -2329,8 +2329,8 @@ static void steam_do_battery_event(struct steam_device *steam,
  * -------+-------+-----------+--------------------------
  *  1     | u8    | --        | sequence number
  *  2-5   | u32   | see below | buttons
- *  6-7   | s16   | ABS_HAT2Y | left trigger (uncalibrated)
- *  8-9   | s16   | ABS_HAT2X | right trigger (uncalibrated)
+ *  6-7   | s16   | ABS_Z     | left trigger (uncalibrated)
+ *  8-9   | s16   | ABS_RZ    | right trigger (uncalibrated)
  *  10-11 | s16   | ABS_X     | left joystick X
  *  12-13 | s16   | ABS_Y     | left joystick Y
  *  14-15 | s16   | ABS_RX    | right joystick X
@@ -2423,8 +2423,8 @@ static const struct steam_axis_mapping steam_ibex_axis_mappings[] = {
 	{ ABS_Y,	-1, 12 },
 	{ ABS_RX,	 1, 14 },
 	{ ABS_RY,	-1, 16 },
-	{ ABS_HAT2Y,	 1, 6 },
-	{ ABS_HAT2X,	 1, 8 },
+	{ ABS_Z,	 1, 6 },
+	{ ABS_RZ,	 1, 8 },
 	{ /* sentinel */ },
 };
 
