diff --git a/drivers/hid/hid-steam.c b/drivers/hid/hid-steam.c
index e9d9051040..2771c03dbf 100644
--- a/drivers/hid/hid-steam.c
+++ b/drivers/hid/hid-steam.c
@@ -589,14 +589,13 @@ static int steam_input_open(struct input_dev *dev)
 	 * Controller. On the Steam Deck, this is toggled manually by holding
 	 * the options button instead, handled by steam_mode_switch_cb.
 	 */
-	if (!(steam->quirks & STEAM_QUIRK_DECK)) {
-		spin_lock_irqsave(&steam->lock, flags);
-		set_lizard_mode = !steam->client_opened && lizard_mode;
-		spin_unlock_irqrestore(&steam->lock, flags);
-		if (set_lizard_mode)
-			steam_set_lizard_mode(steam, false);
-	}
 
+	spin_lock_irqsave(&steam->lock, flags);
+	set_lizard_mode = !steam->client_opened && lizard_mode;
+	spin_unlock_irqrestore(&steam->lock, flags);
+	if (set_lizard_mode)
+		steam_set_lizard_mode(steam, false);
+	
 	return 0;
 }
 
@@ -606,13 +605,11 @@ static void steam_input_close(struct input_dev *dev)
 	unsigned long flags;
 	bool set_lizard_mode;
 
-	if (!(steam->quirks & STEAM_QUIRK_DECK)) {
-		spin_lock_irqsave(&steam->lock, flags);
-		set_lizard_mode = !steam->client_opened && lizard_mode;
-		spin_unlock_irqrestore(&steam->lock, flags);
-		if (set_lizard_mode)
-			steam_set_lizard_mode(steam, true);
-	}
+	spin_lock_irqsave(&steam->lock, flags);
+	set_lizard_mode = !steam->client_opened && lizard_mode;
+	spin_unlock_irqrestore(&steam->lock, flags);
+	if (set_lizard_mode)
+		steam_set_lizard_mode(steam, true);
 }
 
 static enum power_supply_property steam_battery_props[] = {
