diff --git a/drivers/hid/hid-quirks.c b/drivers/hid/hid-quirks.c
index ee4e235..b85fc0a 100644
--- a/drivers/hid/hid-quirks.c
+++ b/drivers/hid/hid-quirks.c
@@ -73,6 +73,9 @@ static const struct hid_device_id hid_quirks[] = {
 // 3H Dual Arcade 2Players
 	{ HID_USB_DEVICE(0x16c0,0x05e0),0x040 },
 
+// MICREAL USB Controller
+	{ HID_USB_DEVICE(0x0314,0x0324),0x040 },
+
 	{ HID_USB_DEVICE(USB_VENDOR_ID_AASHIMA, USB_DEVICE_ID_AASHIMA_GAMEPAD), HID_QUIRK_BADPAD },
 	{ HID_USB_DEVICE(USB_VENDOR_ID_AASHIMA, USB_DEVICE_ID_AASHIMA_PREDATOR), HID_QUIRK_BADPAD },
 	{ HID_USB_DEVICE(USB_VENDOR_ID_AFATECH, USB_DEVICE_ID_AFATECH_AF9016), HID_QUIRK_FULLSPEED_INTERVAL },
