diff --git a/drivers/hid/hid-quirks.c b/drivers/hid/hid-quirks.c
index 4291505a9..ae74d8f3d 100644
--- a/drivers/hid/hid-quirks.c
+++ b/drivers/hid/hid-quirks.c
@@ -79,6 +79,9 @@ static const struct hid_device_id hid_quirks[] = {
 // Shanwan Twin USB Joystick (new revision)
    { HID_USB_DEVICE(0x2563,0x0555),0x111 },
 
+// MICREAL USB Controller
+	{ HID_USB_DEVICE(0x0314,0x0324),0x040 },
+
 	{ HID_USB_DEVICE(USB_VENDOR_ID_8BITDO, USB_DEVICE_ID_8BITDO_PRO_3), HID_QUIRK_ALWAYS_POLL },
 	{ HID_USB_DEVICE(USB_VENDOR_ID_AASHIMA, USB_DEVICE_ID_AASHIMA_GAMEPAD), HID_QUIRK_BADPAD },
 	{ HID_USB_DEVICE(USB_VENDOR_ID_AASHIMA, USB_DEVICE_ID_AASHIMA_PREDATOR), HID_QUIRK_BADPAD },
