--- a/interface/khronos/include/GLES2/gl2.h	2015-05-30 15:53:53.000000000 +0200
+++ b/interface/khronos/include/GLES2/gl2.h	2015-10-21 09:33:41.942996000 +0200
@@ -455,6 +455,7 @@
 #define GL_FRAMEBUFFER                    0x8D40
 #define GL_RENDERBUFFER                   0x8D41
 
+#define GL_RGBA8                          0x8058
 #define GL_RGBA4                          0x8056
 #define GL_RGB5_A1                        0x8057
 #define GL_RGB565                         0x8D62
