From f7051b10c925a0abf661ed7eb717961be537cf94 Mon Sep 17 00:00:00 2001
From: Demetris Ierokipides <ierokipides.dem@gmail.com>
Date: Tue, 26 Jan 2021 14:50:04 +0200
Subject: [PATCH] Fix khr header naming after patch

---
 meson.build | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/meson.build b/meson.build
index 7e9c0b5..45f1083 100644
--- a/meson.build
+++ b/meson.build
@@ -41,7 +41,7 @@ cl_wrappers = {'OpenCL' : '1', 'MaliOpenCL' : '1'}
 
 # Subdir : headers
 mali_headers = {
-  'KHR' : ['include/KHR/mali_khrplatform.h'],
+  'KHR' : ['include/KHR/khrplatform.h'],
 }
 egl_headers = {
   'EGL' : [
-- 
2.25.1
