diff --git a/board/rockchip/odroidgoa/odroidgoa.c b/board/rockchip/odroidgoa/odroidgoa.c
index 7704a88..106d1e9 100755
--- a/board/rockchip/odroidgoa/odroidgoa.c
+++ b/board/rockchip/odroidgoa/odroidgoa.c
@@ -146,11 +146,11 @@ void board_check_mandatory_files(void)
 	}
 
 	/* check launcher in ext4 fs of sd card */
-	if (!file_exists("mmc", "1:2", "/usr/local/bin/emulationstation/emulationstation",
-				FS_TYPE_EXT)) {
-		lcd_setfg_color("white");
-        lcd_printf(0, 0, 1, "[ EmuELEC Gameforce Chi ]");
-	}
+	// if (!file_exists("mmc", "1:2", "/usr/local/bin/emulationstation/emulationstation",
+	// 			FS_TYPE_EXT)) {
+	// 	lcd_setfg_color("white");
+    //     lcd_printf(0, 0, 1, "[ EmuELEC Gameforce Chi ]");
+	// }
 
 	return;
 
@@ -198,9 +198,9 @@ int rk_board_late_init(void)
 
 	/* show boot logo and version */
 	lcd_show_logo();
-	lcd_setfg_color("grey");
-	lcd_printf(0, 18 + disp_offs, 1, " %s", U_BOOT_VERSION);
-	lcd_printf(0, 19 + disp_offs, 1, " %s %s", U_BOOT_DATE, U_BOOT_TIME);
+	// lcd_setfg_color("grey");
+	// lcd_printf(0, 18 + disp_offs, 1, " %s", U_BOOT_VERSION);
+	// lcd_printf(0, 19 + disp_offs, 1, " %s %s", U_BOOT_DATE, U_BOOT_TIME);
 
 	if (!board_check_autotest()) {
 		board_run_autotest();
