diff --git a/board/rockchip/odroidgoa/odroidgoa.c b/board/rockchip/odroidgoa/odroidgoa.c
index 7704a88..8543951 100755
--- a/board/rockchip/odroidgoa/odroidgoa.c
+++ b/board/rockchip/odroidgoa/odroidgoa.c
@@ -146,10 +146,10 @@ void board_check_mandatory_files(void)
 	}
 
 	/* check launcher in ext4 fs of sd card */
-	if (!file_exists("mmc", "1:2", "/usr/local/bin/emulationstation/emulationstation",
+	if (file_exists("mmc", "1:2", "/usr/local/bin/emulationstation/emulationstation",
 				FS_TYPE_EXT)) {
 		lcd_setfg_color("white");
-        lcd_printf(0, 0, 1, "[ EmuELEC Gameforce Chi ]");
+		lcd_printf(0, 0, 1, "[ Batocera Gameforce Chi ]");
 	}
 
 	return;
@@ -196,11 +196,8 @@ int rk_board_late_init(void)
 		charge_display();
 #endif
 
-	/* show boot logo and version */
+	/* show boot logo */
 	lcd_show_logo();
-	lcd_setfg_color("grey");
-	lcd_printf(0, 18 + disp_offs, 1, " %s", U_BOOT_VERSION);
-	lcd_printf(0, 19 + disp_offs, 1, " %s %s", U_BOOT_DATE, U_BOOT_TIME);
 
 	if (!board_check_autotest()) {
 		board_run_autotest();
