diff -Naur bennugd-monolithic-60ee3389efcf9b402d66035e87f33d17d70cbd83.orig/projects/cmake/include/compiler_flags.cmake bennugd-monolithic-60ee3389efcf9b402d66035e87f33d17d70cbd83/projects/cmake/include/compiler_flags.cmake
--- bennugd-monolithic-60ee3389efcf9b402d66035e87f33d17d70cbd83.orig/projects/cmake/include/compiler_flags.cmake	2021-10-06 04:57:50.000000000 +0100
+++ bennugd-monolithic-60ee3389efcf9b402d66035e87f33d17d70cbd83/projects/cmake/include/compiler_flags.cmake	2021-10-20 19:32:29.039832291 +0100
@@ -8,7 +8,7 @@
         set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -std=gnu11 -fno-strict-aliasing")
         set(CMAKE_C_FLAGS_DEBUG "${CMAKE_C_FLAGS} ${CMAKE_C_FLAGS_DEBUG} -Ofast")
         set(CMAKE_C_FLAGS_RELEASE "${CMAKE_C_FLAGS} ${CMAKE_C_FLAGS_RELEASE} -Ofast")
-        set_target_properties(${PROJECT_NAME} PROPERTIES COMPILE_FLAGS "-fno-stack-protector -fno-ident -fomit-frame-pointer -fmerge-all-constants -ffast-math -funroll-all-loops -marm -mcpu=cortex-a35 -mfpu=neon-fp-armv8 -mfloat-abi=hard -Ofast -c -DHOST_CPU=0x20000002 -mfpu=neon-fp-armv8 -mfloat-abi=hard -DARM_HARDFP -mbe32" LINK_FLAGS "-marm -mcpu=cortex-a35 -mfpu=neon-fp-armv8 -mfloat-abi=hard -Ofast -mbe32")
+        set_target_properties(${PROJECT_NAME} PROPERTIES COMPILE_FLAGS "-fno-stack-protector -fno-ident -fomit-frame-pointer -fmerge-all-constants -ffast-math -funroll-all-loops -mcpu=cortex-a72 -Ofast -c -DHOST_CPU=0x20000002 -DARM_HARDFP" LINK_FLAGS "-mcpu=cortex-a72 -Ofast")
 elseif (MSVC)
         # The following flags disable warnings that suggest to use MSVC-specific functions
         set(CMAKE_C_FLAGS "/wd4996 ${CMAKE_C_FLAGS}" CACHE STRING "" FORCE)
