diff --git a/esp_sip.c b/esp_sip.c
index 632db86..9097ddd 100755
--- a/esp_sip.c
+++ b/esp_sip.c
@@ -1171,7 +1171,7 @@ sip_txdoneq_process(struct esp_sip *sip, struct sip_evt_tx_report *tx_report)
                                 tx_info->status.rates[1].idx = -1;
 #endif /* HOST_RC */
 
-                                ieee80211_tx_status(epub->hw, skb);
+                                ieee80211_tx_status_ni(epub->hw, skb);
                                 matchs++;
                                 atomic_dec(&sip->pending_tx_status);
                                 STRACE_RX_TXSTATUS_INC();
@@ -1196,7 +1196,7 @@ sip_txdoneq_process(struct esp_sip *sip)
         struct esp_pub *epub = sip->epub;
         struct sk_buff *skb;
         while ((skb = skb_dequeue(&epub->txdoneq))) {
-                ieee80211_tx_status(epub->hw, skb);
+                ieee80211_tx_status_ni(epub->hw, skb);
         }
 }
 #endif
@@ -1329,7 +1329,7 @@ _exit:
 #ifndef FAST_TX_NOWAIT 
         skb_queue_tail(&sip->epub->txdoneq, skb);
 #else
-        ieee80211_tx_status(sip->epub->hw, skb);
+        ieee80211_tx_status_ni(sip->epub->hw, skb);
 #endif
 }
 #endif /* FAST_TX_STATUS */
