diff --git a/plugins/sixaxis.c b/plugins/sixaxis.c
index fcc93bc..2dbfd74 100644
--- a/plugins/sixaxis.c
+++ b/plugins/sixaxis.c
@@ -300,6 +300,7 @@ static bool setup_device(int fd, int index, struct btd_adapter *adapter)
 	info("sixaxis: setting up new device");
 
 	btd_device_device_set_name(device, devices[index].name);
+	btd_device_set_trusted(device, true);
 	btd_device_set_pnpid(device, devices[index].source, devices[index].vid,
 				devices[index].pid, devices[index].version);
 	btd_device_set_temporary(device, false);
