diff --git a/include/my_cpu.h b/include/my_cpu.h
index bd5fca5..f11ceec 100644
--- a/include/my_cpu.h
+++ b/include/my_cpu.h
@@ -96,8 +96,6 @@ static inline void MY_RELAX_CPU(void)
 #elif defined(_ARCH_PWR8)
   /* Changed from __ppc_get_timebase for musl and clang compatibility */
   __builtin_ppc_get_timebase();
-#elif defined __GNUC__ && defined __riscv
-  __builtin_riscv_pause();
 #elif defined __GNUC__
   /* Mainly, prevent the compiler from optimizing away delay loops */
   __asm__ __volatile__ ("":::"memory");
