--- a/drivers/tty/serial/qcom_geni_serial.c
+++ b/drivers/tty/serial/qcom_geni_serial.c
@@ -1851,7 +1851,9 @@
 	struct qcom_geni_serial_port *port = dev_get_drvdata(dev);
 	struct uart_port *uport = &port->uport;
 	struct qcom_geni_private_data *private_data = uport->private_data;
+	int ret;

+	disable_irq(uport->irq);
 	/*
 	 * This is done so we can hit the lowest possible state in suspend
 	 * even with no_console_suspend
@@ -1860,7 +1862,12 @@
 		geni_icc_set_tag(&port->se, QCOM_ICC_TAG_ACTIVE_ONLY);
 		geni_icc_set_bw(&port->se);
 	}
-	return uart_suspend_port(private_data->drv, uport);
+	ret = uart_suspend_port(private_data->drv, uport);
+	if (ret) {
+		enable_irq(uport->irq);
+		return ret;
+	}
+	return 0;
 }

 static int qcom_geni_serial_resume(struct device *dev)
@@ -1875,6 +1882,7 @@
 		geni_icc_set_tag(&port->se, QCOM_ICC_TAG_ALWAYS);
 		geni_icc_set_bw(&port->se);
 	}
+	enable_irq(uport->irq);
 	return ret;
 }

