Subject: [PATCH] Bluetooth: hci_qca: keep IBS disabled for WCN3950

---
 drivers/bluetooth/hci_qca.c | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

--- a/drivers/bluetooth/hci_qca.c
+++ b/drivers/bluetooth/hci_qca.c
@@ -2025,7 +2025,8 @@ static int qca_setup(struct hci_uart *hu)
 	ret = qca_uart_setup(hdev, qca_baudrate, soc_type, ver,
 			firmware_name, rampatch_name);
 	if (!ret) {
-		clear_bit(QCA_IBS_DISABLED, &qca->flags);
+		if (soc_type != QCA_WCN3950)
+			clear_bit(QCA_IBS_DISABLED, &qca->flags);
 		qca_debugfs_init(hdev);
 		hu->hdev->hw_error = qca_hw_error;
 		hu->hdev->reset = qca_reset;
@@ -2657,6 +2658,9 @@ static int __maybe_unused qca_suspend(struct device *dev)
 	if (test_bit(QCA_ROM_FW, &qca->flags))
 		return 0;
 
+	if (qca_soc_type(hu) == QCA_WCN3950)
+		return 0;
+
 	/* During SSR after memory dump collection, controller will be
 	 * powered off and then powered on.If controller is powered off
 	 * during SSR then we should wait until SSR is completed.
