Subject: [PATCH] usb: typec: tcpm: re-queue queued control messages on -EAGAIN

---
 drivers/usb/typec/tcpm/tcpm.c | 9 ++++++---
 1 file changed, 6 insertions(+), 3 deletions(-)

diff --git a/drivers/usb/typec/tcpm/tcpm.c b/drivers/usb/typec/tcpm/tcpm.c
--- a/drivers/usb/typec/tcpm/tcpm.c
+++ b/drivers/usb/typec/tcpm/tcpm.c
@@ -3868,13 +3868,16 @@
 
 		switch (queued_message) {
 		case PD_MSG_CTRL_WAIT:
-			tcpm_pd_send_control(port, PD_CTRL_WAIT, TCPC_TX_SOP);
+			if (tcpm_pd_send_control(port, PD_CTRL_WAIT, TCPC_TX_SOP) == -EAGAIN)
+				port->queued_message = queued_message;
 			break;
 		case PD_MSG_CTRL_REJECT:
-			tcpm_pd_send_control(port, PD_CTRL_REJECT, TCPC_TX_SOP);
+			if (tcpm_pd_send_control(port, PD_CTRL_REJECT, TCPC_TX_SOP) == -EAGAIN)
+				port->queued_message = queued_message;
 			break;
 		case PD_MSG_CTRL_NOT_SUPP:
-			tcpm_pd_send_control(port, PD_CTRL_NOT_SUPP, TCPC_TX_SOP);
+			if (tcpm_pd_send_control(port, PD_CTRL_NOT_SUPP, TCPC_TX_SOP) == -EAGAIN)
+				port->queued_message = queued_message;
 			break;
 		case PD_MSG_DATA_SINK_CAP:
 			ret = tcpm_pd_send_sink_caps(port);
