--- a/drivers/power/supply/bq256xx_charger.c	2026-03-24 17:36:43
+++ b/drivers/power/supply/bq256xx_charger.c	2026-03-24 18:36:27
@@ -11,6 +11,8 @@
 #include <linux/gpio/consumer.h>
 #include <linux/power_supply.h>
 #include <linux/regmap.h>
+#include <linux/regulator/driver.h>
+#include <linux/regulator/machine.h>
 #include <linux/types.h>
 #include <linux/usb/phy.h>
 #include <linux/device.h>
@@ -75,6 +77,9 @@

 #define BQ256XX_CHG_CONFIG_MASK		BIT(4)
 #define BQ256XX_CHG_CONFIG_BIT_SHIFT	4
+#define BQ256XX_OTG_CONFIG		BIT(5)
+#define BQ256XX_BOOSTV_MASK		GENMASK(5, 4)
+#define BQ256XX_BOOST_LIM		BIT(7)

 #define BQ256XX_ITERM_MASK		GENMASK(3, 0)
 #define BQ256XX_ITERM_STEP_uA		60000
@@ -883,6 +888,73 @@

 	return regmap_update_bits(bq->regmap, BQ256XX_INPUT_CURRENT_LIMIT,
 					BQ256XX_IINDPM_MASK, iindpm_reg_code);
+}
+
+static int bq256xx_enable_vbus(struct regulator_dev *rdev)
+{
+	struct bq256xx_device *bq = rdev_get_drvdata(rdev);
+
+	return regmap_update_bits(bq->regmap, BQ256XX_CHARGER_CONTROL_0,
+				 BQ256XX_OTG_CONFIG, BQ256XX_OTG_CONFIG);
+}
+
+static int bq256xx_disable_vbus(struct regulator_dev *rdev)
+{
+	struct bq256xx_device *bq = rdev_get_drvdata(rdev);
+
+	return regmap_update_bits(bq->regmap, BQ256XX_CHARGER_CONTROL_0,
+				 BQ256XX_OTG_CONFIG, 0);
+}
+
+static int bq256xx_is_enabled_vbus(struct regulator_dev *rdev)
+{
+	struct bq256xx_device *bq = rdev_get_drvdata(rdev);
+	unsigned int val;
+	int ret;
+
+	ret = regmap_read(bq->regmap, BQ256XX_CHARGER_CONTROL_0, &val);
+	if (ret)
+		return ret;
+
+	return !!(val & BQ256XX_OTG_CONFIG);
+}
+
+static const struct regulator_ops bq256xx_vbus_ops = {
+	.enable = bq256xx_enable_vbus,
+	.disable = bq256xx_disable_vbus,
+	.is_enabled = bq256xx_is_enabled_vbus,
+};
+
+static const struct regulator_desc bq256xx_vbus_desc = {
+	.name = "usb-otg-vbus",
+	.of_match = "usb-otg-vbus",
+	.type = REGULATOR_VOLTAGE,
+	.owner = THIS_MODULE,
+	.ops = &bq256xx_vbus_ops,
+	.fixed_uV = 5000000,
+	.n_voltages = 1,
+};
+
+static int bq256xx_register_vbus_regulator(struct bq256xx_device *bq)
+{
+	struct regulator_config cfg = {
+		.dev = bq->dev,
+		.driver_data = bq,
+	};
+	struct regulator_dev *rdev;
+
+	/* Set boost voltage to 5.15V and current limit to 1.2A */
+	regmap_update_bits(bq->regmap, BQ256XX_CHARGER_CONTROL_2,
+			   BQ256XX_BOOSTV_MASK, 0x20);
+	regmap_update_bits(bq->regmap, BQ256XX_CHARGER_CONTROL_2,
+			   BQ256XX_BOOST_LIM, BQ256XX_BOOST_LIM);
+
+	rdev = devm_regulator_register(bq->dev, &bq256xx_vbus_desc, &cfg);
+	if (IS_ERR(rdev))
+		return dev_err_probe(bq->dev, PTR_ERR(rdev),
+				     "Failed to register VBUS regulator\n");
+
+	return 0;
 }

 static void bq256xx_charger_reset(void *data)
@@ -1759,6 +1831,10 @@
 		return ret;
 	}

+	ret = bq256xx_register_vbus_regulator(bq);
+	if (ret)
+		return ret;
+
 	ret = bq256xx_hw_init(bq);
 	if (ret) {
 		dev_err(dev, "Cannot initialize the chip.\n");
