diff --git a/drivers/hid/hid-input.c b/drivers/hid/hid-input.c
index faae69c..8ee0b31 100644
--- a/drivers/hid/hid-input.c
+++ b/drivers/hid/hid-input.c
@@ -603,11 +603,11 @@ static void hidinput_configure_usage(struct hid_input *hidinput, struct hid_fiel
 		case HID_GD_X: case HID_GD_Y: case HID_GD_Z:
 		case HID_GD_RX: case HID_GD_RY: case HID_GD_RZ:
 		case HID_GD_SLIDER: case HID_GD_DIAL: case HID_GD_WHEEL:
-			if (field->flags & HID_MAIN_ITEM_RELATIVE)
-				map_rel(usage->hid & 0xf);
-			else
-				map_abs(usage->hid & 0xf);
-			break;
+			//if (field->flags & HID_MAIN_ITEM_RELATIVE)
+			//	map_rel(usage->hid & 0xf);
+			//else
+			//	map_abs(usage->hid & 0xf);
+			//break;
 
 		case HID_GD_HATSWITCH:
 			usage->hat_min = field->logical_minimum;
