diff --git a/ui/xemu-settings.cc b/ui/xemu-settings.cc
index ae380fb..3962ed1 100644
--- a/ui/xemu-settings.cc
+++ b/ui/xemu-settings.cc
@@ -88,6 +88,9 @@ const char *xemu_settings_get_path(void)
 const char *xemu_settings_get_default_eeprom_path(void)
 {
     static char *eeprom_path = NULL;
+
+        return "/userdata/saves/xbox/xemu_eeprom.bin";
+
     if (eeprom_path != NULL) {
         return eeprom_path;
     }
