diff --git a/softmmu/vl.c b/softmmu/vl.c
index 2eecce1..f169ec6 100644
--- a/softmmu/vl.c
+++ b/softmmu/vl.c
@@ -2777,11 +2777,11 @@ static const char *get_eeprom_path(void)
 
     if (qemu_access(path, F_OK) == -1) {
         if (!xbox_eeprom_generate(path, XBOX_EEPROM_VERSION_R1)) {
-            char *msg = g_strdup_printf("Failed to generate EEPROM file '%s'."
-                                        "\n\nPlease check machine settings.",
-                                        path);
-            xemu_queue_error_message(msg);
-            g_free(msg);
+            //char *msg = g_strdup_printf("Failed to generate EEPROM file '%s'."
+            //                            "\n\nPlease check machine settings.",
+            //                            path);
+            //xemu_queue_error_message(msg);
+            //g_free(msg);
             return NULL;
         }
     }
@@ -2789,10 +2789,10 @@ static const char *get_eeprom_path(void)
     int size = get_image_size(path);
 
     if (size < 0) {
-        char *msg = g_strdup_printf("Failed to open EEPROM file '%s'.\n\n"
-                                    "Please check machine settings.", path);
-        xemu_queue_error_message(msg);
-        g_free(msg);
+        //char *msg = g_strdup_printf("Failed to open EEPROM file '%s'.\n\n"
+        //                            "Please check machine settings.", path);
+        //xemu_queue_error_message(msg);
+        //g_free(msg);
         return NULL;
     }
 
