--- a/softmmu/vl.c	2022-05-11 02:56:58.000000000 +0200
+++ b/softmmu/vl.c	2022-05-12 13:51:04.807691784 +0200
@@ -2845,19 +2845,19 @@
     if (strlen(bootrom_path) > 0) {
         int bootrom_size = get_image_size(bootrom_path);
         if (bootrom_size < 0) {
-            char *msg = g_strdup_printf("Failed to open BootROM file '%s'. "
+            /*char *msg = g_strdup_printf("Failed to open BootROM file '%s'. "
                                         "Please check machine settings.",
                                         bootrom_path);
             xemu_queue_error_message(msg);
-            g_free(msg);
+            g_free(msg);*/
             bootrom_path = "";
         } else if (bootrom_size != 512) {
-            char *msg = g_strdup_printf("Invalid BootROM file '%s' size %d. "
+            /*char *msg = g_strdup_printf("Invalid BootROM file '%s' size %d. "
                                         "Expected 512 bytes. Please check "
                                         "machine settings.",
                                         bootrom_path, bootrom_size);
             xemu_queue_error_message(msg);
-            g_free(msg);
+            g_free(msg);*/
             bootrom_path = "";
         } else {
             char *escaped_bootrom_path = strdup_double_commas(bootrom_path);
