--- a/ui/xemu-settings.cc	2022-05-11 02:56:58.000000000 +0200
+++ b/ui/xemu-settings.cc	2022-05-12 13:55:05.560295047 +0200
@@ -86,7 +86,9 @@
 
 const char *xemu_settings_get_default_eeprom_path(void)
 {
-    static char *eeprom_path = NULL;
+    return "/userdata/saves/xbox/xemu_eeprom.bin";
+
+    /*static char *eeprom_path = NULL;
     if (eeprom_path != NULL) {
         return eeprom_path;
     }
@@ -97,7 +99,7 @@
     assert(base != NULL);
     eeprom_path = g_strdup_printf("%s%s", base, "eeprom.bin");
     SDL_free(base);
-    return eeprom_path;
+    return eeprom_path;*/
 }
 
 static ssize_t get_file_size(FILE *fd)
