--- a/src/video/kmsdrm/SDL_kmsdrmvideo.c	2023-07-19 21:17:00.668327884 +0200
+++ b/src/video/kmsdrm/SDL_kmsdrmvideo.c	2023-07-19 21:14:18.124601452 +0200
@@ -1187,7 +1187,7 @@
     SDL_VideoDisplay *display = SDL_GetDisplayForWindow(window);
     SDL_DisplayData *dispdata = (SDL_DisplayData *)display->driverdata;
 
-    uint32_t surface_fmt = GBM_FORMAT_ARGB8888;
+    uint32_t surface_fmt = GBM_FORMAT_XRGB8888;
     uint32_t surface_flags = GBM_BO_USE_SCANOUT | GBM_BO_USE_RENDERING;
 
     EGLContext egl_context;
