diff --git a/Makefile.libretro b/Makefile.libretro
index 1ecd359..7532c00 100644
--- a/Makefile.libretro
+++ b/Makefile.libretro
@@ -305,6 +305,17 @@ else ifeq ($(platform), qnx)
 	LIBDL :=
 	LIBM :=
 
+#Raspberry Pi 1
+else ifeq ($(platform), rpi1)
+	TARGET := $(TARGET_NAME)_libretro.so
+	fpic := -fPIC
+	CFLAGS += -marm -march=armv6j -mfpu=vfp -mfloat-abi=hard
+	ASFLAGS += -march=armv6j -mfpu=vfp -mfloat-abi=hard
+	HAVE_NEON = 0
+	ARCH = arm
+	BUILTIN_GPU = peops
+	DYNAREC = ari64
+
 #Raspberry Pi 2
 else ifeq ($(platform), rpi2)
 	TARGET := $(TARGET_NAME)_libretro.so
