diff --git a/Makefile.libretro b/Makefile.libretro
index ac91cf8..1cc7f86 100644
--- a/Makefile.libretro
+++ b/Makefile.libretro
@@ -325,6 +325,15 @@ else ifeq ($(platform), rpi3)
 	BUILTIN_GPU = neon
 	DYNAREC = ari64
 
+#Raspberry Pi 3 with a 64bit GNU/Linux OS
+else ifeq ($(platform), rpi3_64)
+        TARGET := $(TARGET_NAME)_libretro.so
+        ARCH := arm64
+        BUILTIN_GPU = unai
+        DYNAREC = lightrec
+        fpic := -fPIC
+        CFLAGS += -march=armv8-a+crc+simd -mtune=cortex-a53 -ftree-vectorize
+
 #Raspberry Pi 4 with a 32bit GNU/Linux OS
 else ifeq ($(platform), rpi4)
 	TARGET := $(TARGET_NAME)_libretro.so
