diff --git a/loader/CMakeLists.txt b/loader/CMakeLists.txt
index 8c814f2234..d586763e3f 100644
--- a/loader/CMakeLists.txt
+++ b/loader/CMakeLists.txt
@@ -154,7 +154,7 @@ if(WIN32)
 
         add_executable(asm_offset asm_offset.c)
         target_link_libraries(asm_offset loader_specific_options)
-        add_custom_command(OUTPUT gen_defines.asm DEPENDS asm_offset COMMAND asm_offset MASM)
+        add_custom_command(OUTPUT gen_defines.asm DEPENDS asm_offset COMMAND ${CMAKE_CURRENT_SOURCE_DIR}/asm_offset MASM)
         add_custom_target(loader_asm_gen_files DEPENDS gen_defines.asm)
         set_target_properties(loader_asm_gen_files PROPERTIES FOLDER ${LOADER_HELPER_FOLDER})
         add_library(loader-unknown-chain OBJECT unknown_ext_chain_masm.asm)
@@ -202,7 +202,7 @@ else() # i.e.: Linux
     if(ASSEMBLER_WORKS)
         add_executable(asm_offset asm_offset.c)
         target_link_libraries(asm_offset loader_specific_options)
-        add_custom_command(OUTPUT gen_defines.asm DEPENDS asm_offset COMMAND asm_offset GAS)
+        add_custom_command(OUTPUT gen_defines.asm DEPENDS asm_offset COMMAND ${CMAKE_CURRENT_SOURCE_DIR}/asm_offset GAS)
         add_custom_target(loader_asm_gen_files DEPENDS gen_defines.asm)
     else()
         if(USE_GAS)
