diff --git a/Makefile.libretro b/Makefile.libretro
index 1409ac5..418876a 100644
--- a/Makefile.libretro
+++ b/Makefile.libretro
@@ -70,6 +70,13 @@ ifneq (,$(findstring unix,$(platform)))
       endif
    endif
    
+   # RockPro64
+   ifneq (,$(findstring rockpro64,$(platform)))
+      ENDIANNESS_DEFINES += -DALIGN_LONG
+      CFLAGS += -fomit-frame-pointer -ffast-math
+      PLATFORM_DEFINES += -DARM -marm -march=armv7ve -mcpu=cortex-a72 -mtune=cortex-a72.cortex-a53 -mfloat-abi=hard
+   endif
+   
 # (armv7 a7, hard point, neon based) ### 
 # NESC, SNESC, C64 mini 
 else ifeq ($(platform), classic_armv7_a7)
