diff --git a/Makefile.libretro b/Makefile.libretro
index 6efbeab..fa1443d 100644
--- a/Makefile.libretro
+++ b/Makefile.libretro
@@ -95,6 +95,20 @@ ifneq (,$(findstring unix,$(platform)))
          PLATFORM_DEFINES += -DARM -marm -mcpu=cortex-a53 -mfpu=neon-fp-armv8 -mfloat-abi=hard
       endif
    endif
+
+   # RockPro64
+   ifneq (,$(findstring rockpro64,$(platform)))
+      ENDIANNESS_DEFINES += -DALIGN_LONG
+      CFLAGS += -fomit-frame-pointer -ffast-math
+      PLATFORM_DEFINES += -DARM -marm -march=armv7ve -mcpu=cortex-a72 -mtune=cortex-a72.cortex-a53 -mfloat-abi=hard
+   endif
+
+   # Odroid-N2
+   ifneq (,$(findstring odroidn2,$(platform)))
+      ENDIANNESS_DEFINES += -DALIGN_LONG
+      CFLAGS += -fomit-frame-pointer -ffast-math
+      PLATFORM_DEFINES += -DARM -marm -mcpu=cortex-a73 -mtune=cortex-a73.cortex-a53 -mfloat-abi=hard
+   endif
    
 # (armv8 a35, hard point, neon based) ###
 # Playstation Classic
