diff --color -uri gsplus-orig/src/sdl2snd_driver.c gsplus-480572054518112647c8fae5d7ea7046a6d6ecfb/src/sdl2snd_driver.c
--- gsplus-orig/src/sdl2snd_driver.c	2019-08-16 12:42:35.000000000 -0700
+++ gsplus-480572054518112647c8fae5d7ea7046a6d6ecfb/src/sdl2snd_driver.c	2021-08-16 03:29:20.952071345 -0700
@@ -131,7 +131,7 @@
   wanted.freq = g_preferred_rate;
   wanted.format = AUDIO_S16SYS;
   wanted.channels = NUM_CHANNELS;
-  wanted.samples = 512;
+  wanted.samples = 2048;
   wanted.callback = _snd_callback;
   wanted.userdata = NULL;
