From a7163ecab9b2a395e809e41255f3567d7a188a5d Mon Sep 17 00:00:00 2001
From: MilhouseVH <milhouseVH.github@nmacleod.com>
Date: Fri, 14 Feb 2020 00:34:00 +0000
Subject: [PATCH] gcc-plugin.sh: use CONFIG_PLUGIN_HOSTCC="" on all distros

---
 scripts/gcc-plugin.sh | 1 +
 1 file changed, 1 insertion(+)

diff --git a/scripts/gcc-plugin.sh b/scripts/gcc-plugin.sh
index d3caefe..6ba7f13 100755
--- a/scripts/gcc-plugin.sh
+++ b/scripts/gcc-plugin.sh
@@ -1,6 +1,6 @@
 #!/bin/sh
 # SPDX-License-Identifier: GPL-2.0
-
+exit 0
 set -e

 srctree=$(dirname "$0")
-- 
2.20.1

