diff --git a/src/matchbox-remote.c b/src/matchbox-remote.c
index 9f62523..2696fd3 100644
--- a/src/matchbox-remote.c
+++ b/src/matchbox-remote.c
@@ -37,6 +37,8 @@
 #define MB_CMD_MISC          7
 #define MB_CMD_COMPOSITE     8
 #define MB_CMB_KEYS_RELOAD   9
+#define MB_CMB_MOUSE_SHOW    10
+#define MB_CMB_MOUSE_HIDE    11
 
 #define MB_CMD_PANEL_TOGGLE_VISIBILITY 1
 #define MB_CMD_PANEL_SIZE              2
@@ -268,6 +270,12 @@ int main(int argc, char* argv[])
 	case 'k':
 	  mbcommand(MB_CMB_KEYS_RELOAD, NULL);
 	  break;
+	case 's':
+	  mbcommand(MB_CMB_MOUSE_SHOW, NULL);
+	  break;
+	case 'h':
+	  mbcommand(MB_CMB_MOUSE_HIDE, NULL);
+	  break;
 	case 'p':
 	  if (!strcmp(arg+1,"panel-toggle"))
 	    {
diff --git a/src/wm.c b/src/wm.c
index 771294c..e8b546f 100644
--- a/src/wm.c
+++ b/src/wm.c
@@ -1742,6 +1742,14 @@ wm_handle_client_message(Wm *w, XClientMessageEvent *e)
 	   break;
 #endif
 
+	 case MB_CMB_MOUSE_SHOW:
+	   wm_set_cursor_visibility(w, True);	   
+	   break;
+
+	 case MB_CMB_MOUSE_HIDE:
+	   wm_set_cursor_visibility(w, False);
+	   break;
+
 #ifdef USE_COMPOSITE
 	 case MB_CMD_COMPOSITE:
 	   if (w->comp_engine_disabled)
diff --git a/src/structs.h b/src/structs.h
index ab3a6ff..0d45c85 100644
--- a/src/structs.h
+++ b/src/structs.h
@@ -136,5 +136,7 @@
 #define MB_CMD_MISC        7   /* spare, used for debugging */
 #define MB_CMD_COMPOSITE   8
 #define MB_CMB_KEYS_RELOAD 9
+#define MB_CMB_MOUSE_SHOW    10
+#define MB_CMB_MOUSE_HIDE    11
 
 /* Atoms, if you change these check ewmh_init() first */
