diff --git a/Makefile b/db9_gpio_rpi-1.2/Makefile
index 20b0d3d..59abf9d 100644
--- a/Makefile
+++ b/Makefile
@@ -1,8 +1,15 @@
-obj-m := db9_gpio_rpi.o
-KVERSION := `uname -r`
+obj-m  = db9_gpio_rpi.o
 
-all:
-	$(MAKE) -C /lib/modules/$(KVERSION)/build M=$(PWD) modules
+all: modules
+
+modules:
+	${MAKE} ARCH="${ARCH}" CROSS_COMPILE="${CROSS_COMPILE}" -C ${KERNELDIR} SUBDIRS="${MODULE_DIR}"  modules
+
+modules_install:
+	${MAKE} ARCH="${ARCH}" CROSS_COMPILE="${CROSS_COMPILE}" INSTALL_MOD_PATH="${INSTALL_MOD_PATH}" -C ${KERNELDIR} SUBDIRS="${MODULE_DIR}"  modules_install
 
 clean:
-	$(MAKE) -C /lib/modules/$(KVERSION)/build M=$(PWD) clean
+	rm -f *.o *.ko *.mod.c .*.o .*.ko .*.mod.c .*.cmd *~
+	rm -f Module.symvers Module.markers modules.order
+	rm -rf .tmp_versions
+
