diff --git a/gfx/common/x11_common.c b/gfx/common/x11_common.c
index 47b0552..098ad8a 100644
--- a/gfx/common/x11_common.c
+++ b/gfx/common/x11_common.c
@@ -783,9 +783,10 @@ void x11_update_title(void *data)
 
 bool x11_input_ctx_new(bool true_full)
 {
-   if (!x11_create_input_context(g_x11_dpy, g_x11_win,
-            &g_x11_xim, &g_x11_xic))
-      return false;
+   //if (!x11_create_input_context(g_x11_dpy, g_x11_win,
+   //         &g_x11_xim, &g_x11_xic))
+   //   return false;
+  x11_init_keyboard_lut();
 
    video_driver_display_type_set(RARCH_DISPLAY_X11);
    video_driver_display_set((uintptr_t)g_x11_dpy);
