diff --git a/gfx/common/x11_common.c b/gfx/common/x11_common.c
index f5cd494..0857f6a 100644
--- a/gfx/common/x11_common.c
+++ b/gfx/common/x11_common.c
@@ -426,25 +426,25 @@ static void x11_handle_key_event(XEvent *event, XIC ic, bool filter)
          char keybuf[32];
 
          keybuf[0] = '\0';
-#ifdef X_HAVE_UTF8_STRING
-         status = 0;
-
-         /* XwcLookupString doesn't seem to work. */
-         num = Xutf8LookupString(ic, &event->xkey, keybuf,
-               ARRAY_SIZE(keybuf), &keysym, &status);
-
-         /* libc functions need UTF-8 locale to work properly,
-          * which makes mbrtowc a bit impractical.
-          *
-          * Use custom UTF8 -> UTF-32 conversion. */
-         num = utf8_conv_utf32(chars, ARRAY_SIZE(chars), keybuf, num);
-#else
+//#ifdef X_HAVE_UTF8_STRING
+//         status = 0;
+//
+//         /* XwcLookupString doesn't seem to work. */
+//         num = Xutf8LookupString(ic, &event->xkey, keybuf,
+//               ARRAY_SIZE(keybuf), &keysym, &status);
+//
+//         /* libc functions need UTF-8 locale to work properly,
+//          * which makes mbrtowc a bit impractical.
+//          *
+//          * Use custom UTF8 -> UTF-32 conversion. */
+//         num = utf8_conv_utf32(chars, ARRAY_SIZE(chars), keybuf, num);
+//#else
          (void)ic;
          num = XLookupString(&event->xkey, keybuf,
                sizeof(keybuf), &keysym, NULL); /* ASCII only. */
          for (i = 0; i < num; i++)
             chars[i] = keybuf[i] & 0x7f;
-#endif
+//#endif
       }
       else
          keysym = XLookupKeysym(&event->xkey, (state & ShiftMask) || (state & LockMask));
