diff --git a/audio/audio_driver.c b/audio/audio_driver.c
index 57fa9ed..88bc040 100644
--- a/audio/audio_driver.c
+++ b/audio/audio_driver.c
@@ -1332,8 +1332,8 @@ static void audio_driver_load_menu_bgm_callback(retro_task_t *task,
 void audio_driver_load_system_sounds(void)
 {
    char basename_noext[256];
-   char sounds_path[PATH_MAX_LENGTH];
-   char sounds_fallback_path[PATH_MAX_LENGTH];
+   char* sounds_path = "/usr/share/libretro/assets/sounds";
+   char* sounds_fallback_path = "/userdata/sounds/retroachievements";
    settings_t *settings                  = config_get_ptr();
    const char *dir_assets                = settings->paths.directory_assets;
    const bool audio_enable_menu          = settings->bools.audio_enable_menu;
@@ -1354,19 +1354,6 @@ void audio_driver_load_system_sounds(void)
    if (!audio_enable_menu && !audio_enable_cheevo_unlock)
       goto end;
 
-   sounds_path[0] = basename_noext[0] ='\0';
-
-   fill_pathname_join_special(
-         sounds_fallback_path,
-         dir_assets,
-         "sounds",
-         sizeof(sounds_fallback_path));
-
-   fill_pathname_application_special(
-         sounds_path,
-         sizeof(sounds_path),
-         APPLICATION_SPECIAL_DIRECTORY_ASSETS_SOUNDS);
-
    list          = dir_list_new(sounds_path, MENU_SOUND_FORMATS, false, false, false, false);
    list_fallback = dir_list_new(sounds_fallback_path, MENU_SOUND_FORMATS, false, false, false, false);
 
@@ -1410,8 +1397,13 @@ void audio_driver_load_system_sounds(void)
             path_notice = path;
          else if (string_is_equal_noncase(basename_noext, "bgm"))
             path_bgm = path;
-         else if (string_is_equal_noncase(basename_noext, "unlock"))
-            path_cheevo_unlock = path;
+         else if(string_is_empty(settings->arrays.cheevos_unlock_sound)) {
+            if (string_is_equal_noncase(basename_noext, "unlock"))
+               path_cheevo_unlock = path;
+            } else {
+               if (string_is_equal_noncase(basename_noext, settings->arrays.cheevos_unlock_sound))
+                  path_cheevo_unlock = path;
+            }
       }
    }
 
diff --git a/configuration.c b/configuration.c
index 8751932..0c698a1 100644
--- a/configuration.c
+++ b/configuration.c
@@ -1406,6 +1406,7 @@ static struct config_array_setting *populate_settings_array(settings_t *settings
    SETTING_ARRAY("cheevos_custom_host",      settings->arrays.cheevos_custom_host, false, NULL, true);
    SETTING_ARRAY("cheevos_username",         settings->arrays.cheevos_username, false, NULL, true);
    SETTING_ARRAY("cheevos_password",         settings->arrays.cheevos_password, false, NULL, true);
+   SETTING_ARRAY("cheevos_unlock_sound",     settings->arrays.cheevos_unlock_sound, false, NULL, true);
    SETTING_ARRAY("cheevos_token",            settings->arrays.cheevos_token, false, NULL, true);
    SETTING_ARRAY("cheevos_leaderboards_enable", settings->arrays.cheevos_leaderboards_enable, true, "true", true);
 #endif
@@ -2672,6 +2673,7 @@ void config_set_defaults(void *data)
 #ifdef HAVE_CHEEVOS
    *settings->arrays.cheevos_username                 = '\0';
    *settings->arrays.cheevos_password                 = '\0';
+   *settings->arrays.cheevos_unlock_sound             = '\0';
    *settings->arrays.cheevos_token                    = '\0';
 #endif
 
diff --git a/configuration.h b/configuration.h
index 839fdde..aa6e709 100644
--- a/configuration.h
+++ b/configuration.h
@@ -410,6 +410,7 @@ typedef struct settings
       char menu_driver[32];
       char cheevos_username[32];
       char cheevos_password[256];
+      char cheevos_unlock_sound[128];
       char cheevos_token[32];
       char cheevos_leaderboards_enable[32];
       char cheevos_custom_host[64];
