--- a/retroarch.c	2020-08-10 10:17:29.514556581 +0200
+++ b/retroarch.c	2020-08-10 10:17:38.318728334 +0200
@@ -4727,7 +4727,7 @@
                   entry->label, entry->type, i, entry->entry_idx);
          break;
       case MENU_ACTION_SEARCH:
-         menu_input_dialog_start_search();
+         //menu_input_dialog_start_search();
          break;
       case MENU_ACTION_SCAN:
          if (cbs && cbs->action_scan)
--- a/retroarch.c	2020-08-10 10:19:57.301421374 +0200
+++ b/retroarch.c	2020-08-10 10:20:41.466270802 +0200
@@ -9788,7 +9788,7 @@
    frontend_drv =  
       (const frontend_ctx_driver_t*)frontend_driver_get_cpu_architecture_str(
             frontend_architecture_tmp, sizeof(frontend_architecture_tmp));
-   snprintf(frontend_architecture, sizeof(frontend_architecture), "%s %s",
+   snprintf(frontend_architecture, sizeof(frontend_architecture), "%s %s@BATOCERA",
          frontend_drv->ident, frontend_architecture_tmp);
 
 #ifdef HAVE_DISCORD
