diff --git a/retroarch.c b/retroarch.c
index f692376..1931d61 100644
--- a/retroarch.c
+++ b/retroarch.c
@@ -2598,7 +2598,7 @@ int generic_menu_entry_action(
                   entry->label, entry->type, i, entry->entry_idx);
          break;
       case MENU_ACTION_SEARCH:
-         menu_input_dialog_start_search();
+         //menu_input_dialog_start_search();
          break;
       case MENU_ACTION_SCAN:
          if (cbs && cbs->action_scan)
@@ -7702,7 +7702,7 @@ static void netplay_announce(struct rarch_state *p_rarch)
             frontend_architecture_tmp, sizeof(frontend_architecture_tmp));
    snprintf(frontend_architecture,
          sizeof(frontend_architecture),
-         "%s %s",
+         "%s %s @BATOCERA",
          frontend_drv->ident,
          frontend_architecture_tmp);

