diff --git a/menu/menu_driver.c b/menu/menu_driver.c
index f342977..d5b73cf 100644
--- a/menu/menu_driver.c
+++ b/menu/menu_driver.c
@@ -499,7 +499,7 @@ int generic_menu_entry_action(
                   entry->label, entry->type, i, entry->entry_idx);
          break;
       case MENU_ACTION_SEARCH:
-         menu_input_dialog_start_search();
+         //menu_input_dialog_start_search();
          break;
       case MENU_ACTION_SCAN:
          if (cbs && cbs->action_scan)
diff --git a/network/netplay/netplay_frontend.c b/network/netplay/netplay_frontend.c
index 9ece3fb..c9b6f76 100644
--- a/network/netplay/netplay_frontend.c
+++ b/network/netplay/netplay_frontend.c
@@ -852,7 +852,7 @@ static void netplay_announce(void)
    frontend_drv =  
       (const frontend_ctx_driver_t*)frontend_driver_get_cpu_architecture_str(
             frontend_architecture_tmp, sizeof(frontend_architecture_tmp));
-   snprintf(frontend_architecture, sizeof(frontend_architecture), "%s %s",
+   snprintf(frontend_architecture, sizeof(frontend_architecture), "%s %s@BATOCERA",
          frontend_drv->ident, frontend_architecture_tmp);
 
 #ifdef HAVE_DISCORD
