diff --git a/gfx/drivers/vg.c b/gfx/drivers/vg.c
index 4f08667..66322dd 100644
--- a/gfx/drivers/vg.c
+++ b/gfx/drivers/vg.c
@@ -224,7 +224,7 @@ static void *vg_init(const video_info_t *video,
          video->smooth 
          ? VG_IMAGE_QUALITY_BETTER 
          : VG_IMAGE_QUALITY_NONANTIALIASED);
-   vg_set_nonblock_state(vg, !video->vsync);
+   vg_set_nonblock_state(vg, !video->vsync, video_driver_test_all_flags(GFX_CTX_FLAGS_ADAPTIVE_VSYNC) && settings->bools.video_adaptive_vsync, 0);
 
    inp.input      = input;
    inp.input_data = input_data;
diff --git a/gfx/drivers_context/vc_egl_ctx.c b/gfx/drivers_context/vc_egl_ctx.c
index 2222ffb..9eceb5d 100644
--- a/gfx/drivers_context/vc_egl_ctx.c
+++ b/gfx/drivers_context/vc_egl_ctx.c
@@ -180,7 +180,6 @@ static void *gfx_ctx_vc_init(void *video_driver)
       EGL_NONE
    };
 #endif
-   settings_t *settings  = config_get_ptr();
    vc_ctx_data_t *vc     = NULL;
    unsigned fullscreen_x = settings->uints.video_fullscreen_x;
    unsigned fullscreen_y = settings->uints.video_fullscreen_y;
