diff --git a/gfx/drivers_context/drm_ctx.c b/gfx/drivers_context/drm_ctx.c
index 5b59e4c..6ecaa72 100644
--- a/gfx/drivers_context/drm_ctx.c
+++ b/gfx/drivers_context/drm_ctx.c
@@ -807,6 +807,8 @@ static bool gfx_ctx_drm_set_video_mode(void *data,
    settings_t *settings            = config_get_ptr();
    unsigned black_frame_insertion  = settings->uints.video_black_frame_insertion;
    float video_refresh_rate        = settings->floats.video_refresh_rate;
+   FILE* fdDrmMode;
+   int drmMode;
 
    if (!drm)
       return false;
@@ -820,58 +822,16 @@ static bool gfx_ctx_drm_set_video_mode(void *data,
    /* Find desired video mode, and use that.
     * If not fullscreen, we get desired windowed size,
     * which is not appropriate. */
-   if(gfx_ctx_drm_get_mode_from_video_state(&gfx_ctx_crt_switch_mode))
-   {
-      RARCH_DBG("[KMS]: New mode detected: %dx%d\n", gfx_ctx_crt_switch_mode.hdisplay, gfx_ctx_crt_switch_mode.vdisplay);
-      g_drm_mode = &gfx_ctx_crt_switch_mode;
-      drm->fb_width  = gfx_ctx_crt_switch_mode.hdisplay;
-      drm->fb_height = gfx_ctx_crt_switch_mode.vdisplay;
-      switch_mode = true;
-      /* Let's exit, since modeswitching will happen while swapping buffers */
-      return true;
-   }
-   if ((width == 0 && height == 0) || !fullscreen)
-      g_drm_mode                   = &g_drm_connector->modes[0];
-   else
-   {
-      /* check if custom HDMI timings were asked */
-      if (gfx_ctx_crt_switch_mode.vdisplay > 0)
-      {
-         g_drm_mode                = &gfx_ctx_crt_switch_mode;
-         RARCH_LOG("[DRM]: custom mode requested: %s\n", gfx_ctx_crt_switch_mode.name);
-      }
-      else
-      {
-         /* Try to match refresh_rate as closely as possible.
-          *
-          * Lower resolutions tend to have multiple supported
-          * refresh rates as well.
-          */
-         drmModeModeInfo *mode     = NULL;
-         float minimum_fps_diff    = 0.0f;
-         float mode_vrefresh       = 0.0f;
-         g_drm_mode                = 0;
-
-         /* Find best match. */
-         for (i = 0; i < g_drm_connector->count_modes; i++)
-         {
-            float diff;
-            mode                   = &g_drm_connector->modes[i];
-
-            if (   (width  != mode->hdisplay)
-                || (height != mode->vdisplay))
-               continue;
-
-            mode_vrefresh          = drm_calc_refresh_rate(mode);
-            diff                   = fabsf(refresh_mod * mode_vrefresh - video_refresh_rate);
-
-            if (!g_drm_mode || diff < minimum_fps_diff)
-            {
-               g_drm_mode          = mode;
-               minimum_fps_diff    = diff;
-            }
-         }
-      }
+   // on batocera, we won't guess a mode ; we want the one set by drmMode
+   g_drm_mode = &g_drm_connector->modes[0];
+   // batocera - set resolution
+   if((fdDrmMode = fopen("/var/run/drmMode", "r")) != NULL) {
+     if(fscanf(fdDrmMode, "%i", &drmMode) == 1) {
+       if(drmMode>=0 && drmMode<g_drm_connector->count_modes) {
+         g_drm_mode = &g_drm_connector->modes[drmMode];
+       }
+     }
+     fclose(fdDrmMode);
    }
 
    if (!g_drm_mode)
