--- a/cheevos/cheevos.c	2023-07-13 17:02:33.000000000 +0200
+++ b/cheevos/cheevos.c	2023-07-13 23:15:43.488414454 +0200
@@ -346,9 +346,10 @@ static void rcheevos_award_achievement(const rc_client_achievement_t* cheevo)
       {
          const char *path_directory_screenshot = settings->paths.directory_screenshot;
          video_driver_state_t* video_st = video_state_get_ptr();;
-         snprintf(shotname, shotname_len, "%s/%s-cheevo-%u",
+         snprintf(shotname, shotname_len, "%s/%s - %s (%u)",
             path_directory_screenshot,
             path_basename(path_get(RARCH_PATH_BASENAME)),
+            cheevo->title,
             (unsigned)cheevo->id);
          shotname[shotname_len - 1] = '\0';
 