--- a/gfx/video_crt_switch.c	2021-11-07 03:28:14.000000000 +0100
+++ b/gfx/video_crt_switch.c	2021-11-07 19:29:31.010213633 +0100
@@ -131,7 +131,7 @@
       RARCH_LOG("[CRT]: Resolution is stretched. Fractal scaling @ X:%f Y:%f \n", srm_xscale, srm_yscale);
    }
    else
-      RARCH_LOG("[CRT]: SR integer scaled  X:%d Y:%d \n",srm.x_scale, srm.y_scale);
+      RARCH_LOG("[CRT]: SR integer scaled  X:%d Y:%d \n",(int)srm_xscale, (int)srm_yscale);

    scaled_width = roundf(patched_width*srm_xscale);
    scaled_height = roundf(patched_height*srm_yscale);
