diff --git a/retroarch.c b/retroarch.c
index 3d2469e..c4326d8 100644
--- a/retroarch.c
+++ b/retroarch.c
@@ -30241,11 +30241,7 @@ void audio_driver_load_system_sounds(void)
    sounds_fallback_path = (char*)malloc(PATH_MAX_LENGTH * sizeof(char));
    sounds_path[0] = sounds_fallback_path[0] = '\0';
 
-   fill_pathname_join(
-         sounds_fallback_path,
-         dir_assets,
-         "sounds",
-         PATH_MAX_LENGTH);
+   sounds_fallback_path = "/userdata/sounds/retroachievements";
 
    fill_pathname_application_special(
          sounds_path,
