--- a/input/drivers/udev_input.c	2023-07-13 23:46:51.029365971 +0200
+++ b/input/drivers/udev_input.c	2023-07-13 23:49:17.936546149 +0200
@@ -3248,11 +3248,8 @@
    {
       bool mouse = 0;
       bool touch = 0;
-      /* gotta have some buttons!  return -1 to skip error logging for this:)  */
-      if (ioctl(fd, EVIOCGBIT(EV_KEY, sizeof (keycaps)), keycaps) == -1)
-      {
-         ret = -1;
-         goto end;
+      if (type == UDEV_INPUT_MOUSE) {
+         mouse = 1; // if it's a mouse, it is a mouse ; otherwise, guessing the mouse index to configure is a hell. (case of configurable keyboard which can have left/right mouse button settable)
       }
 
       if (ioctl(fd, EVIOCGBIT(EV_REL, sizeof (relcaps)), relcaps) != -1)
