diff --git a/gfx/common/wayland_common.c b/gfx/common/wayland_common.c
index bf02a4b..a181740 100644
--- a/gfx/common/wayland_common.c
+++ b/gfx/common/wayland_common.c
@@ -680,12 +680,6 @@ bool gfx_ctx_wl_init_common(
    wl_display_roundtrip(wl->input.dpy);
    xdg_wm_base_add_listener(wl->xdg_shell, &xdg_shell_listener, NULL);

-   /* Bind SHM based wl_buffer to wl_surface until the vulkan surface is ready.
-    * This shows the window which assigns us a display (wl_output)
-    *  which is usefull for HiDPI and auto selecting a display for fullscreen. */
-   if (!wl_draw_splash_screen(wl))
-      RARCH_ERR("[Wayland]: Failed to draw splash screen\n");
-
    // Make sure splash screen is on screen and sized
 #ifdef HAVE_LIBDECOR_H
    if (wl->libdecor)
@@ -695,15 +689,8 @@ bool gfx_ctx_wl_init_common(
          if (wl->libdecor_dispatch(wl->libdecor_context, 0) < 0)
             RARCH_ERR("[Wayland]: libdecor failed to dispatch\n");
    }
-   else
 #endif
-   {
-      wl->configured = true;
-
-      while (wl->configured)
-         wl_display_dispatch(wl->input.dpy);
-   }
-
+
    wl->input.fd = wl_display_get_fd(wl->input.dpy);

    wl->input.keyboard_focus  = true;
