diff --git a/gfx/drivers_context/wayland_ctx.c b/gfx/drivers_context/wayland_ctx.c
index e12a453..f5f2494 100644
--- a/gfx/drivers_context/wayland_ctx.c
+++ b/gfx/drivers_context/wayland_ctx.c
@@ -437,6 +437,10 @@ static void gfx_ctx_wl_input_driver(void *data,
       const char *joypad_name,
       input_driver_t **input, void **input_data)
 {
+   *input      = NULL;
+   *input_data = NULL;
+   return;
+
    gfx_ctx_wayland_data_t *wl = (gfx_ctx_wayland_data_t*)data;
    /* Input is heavily tied to the window stuff
     * on Wayland, so just implement the input driver here. */
