diff --git a/network/netplay/netplay_frontend.c b/network/netplay/netplay_frontend.c
index ef322cb..bc41ea8 100644
--- a/network/netplay/netplay_frontend.c
+++ b/network/netplay/netplay_frontend.c
@@ -850,7 +850,7 @@ void netplay_get_architecture(char *frontend_architecture, size_t size)
                sizeof(architecture));
          break;
    }
-   snprintf(frontend_architecture, size, "%s %s", frontend->ident, architecture);
+   snprintf(frontend_architecture, size, "%s %s@BATOCERA", frontend->ident, architecture);
 }
 
 static void netplay_announce(void)
