diff --git a/menu/widgets/menu_widgets.c b/menu/widgets/menu_widgets.c
index 0b91d19..5765eb5 100644
--- a/menu/widgets/menu_widgets.c
+++ b/menu/widgets/menu_widgets.c
@@ -1709,7 +1709,7 @@ void menu_widgets_frame(video_frame_info_t *video_info)
    if (menu_widgets_fast_forward)
       top_right_x_advance -= menu_widgets_draw_indicator(video_info,
          menu_widgets_icons_textures[MENU_WIDGETS_ICON_FAST_FORWARD], (video_info->fps_show ? simple_widget_height : 0), top_right_x_advance,
-         MSG_PAUSED);
+         MSG_FAST_FORWARD);
 
    if (menu_widgets_rewinding)
       top_right_x_advance -= menu_widgets_draw_indicator(video_info,
