--- a/Makefile
+++ b/Makefile
@@ -106,6 +106,19 @@
 	AR = psp-ar$(EXE_EXT)
    STATIC_LINKING = 1
 	FLAGS += -G0 -DLSB_FIRST
+else ifneq (,$(findstring armv,$(platform)))
+	TARGET := $(TARGET_NAME)_libretro.so
+   	fpic := -fPIC
+	SHARED := -shared -Wl,--no-undefined -Wl,--version-script=link.T
+   	CC = gcc
+   	CXX = g++
+	FLAGS += -marm
+	ifneq (,$(findstring softfloat,$(platform)))
+   		FLAGS += -mfloat-abi=softfp
+	else ifneq (,$(findstring hardfloat,$(platform)))
+   		FLAGS += -mfloat-abi=hard
+	endif
+   	FLAGS += -DARM
 else
    TARGET := $(TARGET_NAME)_libretro.dll
    CC = gcc
