diff --git a/Makefile.libretro b/Makefile.libretro
index a15d2c9..e96783d 100755
--- a/Makefile.libretro
+++ b/Makefile.libretro
@@ -75,9 +75,9 @@ ifneq (,$(findstring unix,$(platform)))
 	endif
 	CC = gcc
 	CXX = gcc
-	LDFLAGS += -I/usr/include/SDL2 -L/usr/lib/x86_64-linux-gnu -lSDL2
-	CFLAGS += -I/usr/include/SDL2 `sdl2-config --cflags`
-	CXXFLAGS += -I/usr/include/SDL2 `sdl2-config --cflags`
+	LDFLAGS += -lSDL2
+	CFLAGS += `sdl2-config --cflags`
+	CXXFLAGS += `sdl2-config --cflags`
 
 	# Raspberry Pi
 # ---FOR TEST ---
@@ -731,7 +731,7 @@ clean: cls clean-all
 rebuild: cls clean-all build-all
 
 cls:
-	@clear
+	@echo
 
 build-all: $(TARGET)
