--- a/CMakeLists.txt	2023-01-02 06:52:28.830577662 +0000
+++ b/CMakeLists.txt	2023-01-02 06:54:24.748210141 +0000
@@ -215,7 +215,7 @@
   set(_M_ARM_64 1)
   add_definitions(-D_M_ARM_64=1)
   # CRC instruction set is used in the CRC32 hash function
-  check_and_add_flag(HAVE_ARCH_ARMV8 -march=armv8-a+crc)
+  #check_and_add_flag(HAVE_ARCH_ARMV8 -march=armv8-a+crc)
 else()
   message(FATAL_ERROR "You're building on an unsupported platform: "
       "'${CMAKE_SYSTEM_PROCESSOR}' with ${CMAKE_SIZEOF_VOID_P}-byte pointers."
