diff --git a/aic8800_fdrv/rwnx_rx.c b/aic8800_fdrv/rwnx_rx.c
index 82aea2967b..6ad0f39e61 100644
--- a/aic8800_fdrv/rwnx_rx.c
+++ b/aic8800_fdrv/rwnx_rx.c
@@ -491,7 +491,7 @@ static bool rwnx_rx_data_skb(struct rwnx_hw *rwnx_hw, struct rwnx_vif *rwnx_vif,
 
     if (amsdu) {
         int count;
-#if LINUX_VERSION_CODE >= KERNEL_VERSION(6, 3, 0)
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(6, 1, 0)
         ieee80211_amsdu_to_8023s(skb, &list, rwnx_vif->ndev->dev_addr,
                                  RWNX_VIF_TYPE(rwnx_vif), 0, NULL, NULL, false);
 #else
