--- a/configure	2015-07-10 00:41:19.000000000 +0200
+++ b/configure	2015-10-06 10:35:37.741188823 +0200
@@ -492,13 +492,14 @@
 
   if is_arch_armv7():
     o['variables']['arm_version'] = '7'
+    o['variables']['arm_fpu'] = 'vfpv3'  # V8 3.18 no longer supports VFP2.
+    o['variables']['arm_neon'] = int(is_arm_neon())
   elif is_arch_armv6():
     o['variables']['arm_version'] = '6'
   else:
-    o['variables']['arm_version'] = 'default'
+    o['variables']['arm_version'] = '6'
+    o['variables']['arm_fpu'] = 'vfp'  
 
-  o['variables']['arm_fpu'] = 'vfpv3'  # V8 3.18 no longer supports VFP2.
-  o['variables']['arm_neon'] = int(is_arm_neon())
   o['variables']['arm_thumb'] = 0      # -marm
   o['variables']['arm_float_abi'] = arm_float_abi
 
