diff --git a/Makefile b/Makefile
index 15de6cf..0684e25 100644
--- a/Makefile
+++ b/Makefile
@@ -159,11 +159,19 @@ else ifneq (,$(findstring rpi,$(platform)))
       CPUFLAGS += -mcpu=cortex-a7
       ARM_CPUFLAGS = -mfpu=neon-vfpv4
    else ifneq (,$(findstring rpi3,$(platform)))
-      CPUFLAGS += -march=armv8-a+crc -mtune=cortex-a53
-      ARM_CPUFLAGS = -mfpu=neon-fp-armv8
+      ifneq (,$(findstring rpi3_64,$(platform)))
+         CPUFLAGS += -mcpu=cortex-a53 -mtune=cortex-a53
+      else
+         CPUFLAGS += -march=armv8-a+crc -mtune=cortex-a53
+         ARM_CPUFLAGS = -mfpu=neon-fp-armv8
+      endif
    else ifneq (,$(findstring rpi4,$(platform)))
-      CPUFLAGS += -march=armv8-a+crc -mtune=cortex-a72
-      ARM_CPUFLAGS = -mfpu=neon-fp-armv8
+      ifneq (,$(findstring rpi4_64,$(platform)))
+         CPUFLAGS += -mcpu=cortex-a72 -mtune=cortex-a72
+      else
+         CPUFLAGS += -march=armv8-a+crc -mtune=cortex-a72
+         ARM_CPUFLAGS = -mfpu=neon-fp-armv8
+      endif
    else ifneq (,$(findstring rpi,$(platform)))
       CPUFLAGS += -mcpu=arm1176jzf-s
       ARM_CPUFLAGS = -mfpu=vfp
