diff --git a/Makefile b/Makefile
index d6f45e6..32346e4 100644
--- a/Makefile
+++ b/Makefile
@@ -406,7 +406,7 @@ $(AWK_DEST_DIR)/asm_defines_nasm.h: $(ASM_DEFINES_OBJ)
 	$(STRINGS) "$<" | $(TR) -d '\r' | $(AWK) -v dest_dir="$(AWK_DEST_DIR)" -f $(CORE_DIR)/tools/gen_asm_defines.awk
 
 %.o: %.asm $(AWK_DEST_DIR)/asm_defines_gas.h
-	nasm -i$(AWK_DEST_DIR)/ $(ASFLAGS) $< -o $@
+	$(NASM) -i$(AWK_DEST_DIR)/ $(ASFLAGS) $< -o $@
 
 %.o: %.S $(AWK_DEST_DIR)/asm_defines_gas.h
 	$(CC_AS) $(CFLAGS) -c $< -o $@
