--- a/src/video/SDL_egl.c
+++ b/src/video/SDL_egl.c
@@ -432,6 +432,13 @@
 int
 SDL_EGL_ChooseConfig(_THIS) 
 {
+    EGLint attr[] = {       // some attributes to set up our egl-interface
+            EGL_BUFFER_SIZE, 32,
+            EGL_RENDERABLE_TYPE,
+            EGL_OPENGL_ES2_BIT,
+            EGL_NONE
+    };
+
 /* 64 seems nice. */
     EGLint attribs[64];
     EGLint found_configs = 0, value;
@@ -521,7 +528,7 @@
     attribs[i++] = EGL_NONE;
 
     if (_this->egl_data->eglChooseConfig(_this->egl_data->egl_display,
-        attribs,
+        attr,
         configs, SDL_arraysize(configs),
         &found_configs) == EGL_FALSE ||
         found_configs == 0) {
