diff --git a/src/api/Joystick.cpp b/src/api/Joystick.cpp
index 8f6492d..ca2ff0a 100644
--- a/src/api/Joystick.cpp
+++ b/src/api/Joystick.cpp
@@ -47,10 +47,11 @@ bool CJoystick::Equals(const CJoystick* rhs) const
          VendorID()      == rhs->VendorID()      &&
          ProductID()     == rhs->ProductID()     &&
          Provider()      == rhs->Provider()      &&
-         RequestedPort() == rhs->RequestedPort() &&
-         ButtonCount()   == rhs->ButtonCount()   &&
-         HatCount()      == rhs->HatCount()      &&
-         AxisCount()     == rhs->AxisCount();
+    RequestedPort() == rhs->RequestedPort() // &&
+    //ButtonCount()   == rhs->ButtonCount()   &&
+    //HatCount()      == rhs->HatCount()      &&
+    //AxisCount()     == rhs->AxisCount()
+    ;
 }
 
 void CJoystick::SetName(const std::string& strName)
diff --git a/src/storage/Device.cpp b/src/storage/Device.cpp
index 628331e..ae9718e 100644
--- a/src/storage/Device.cpp
+++ b/src/storage/Device.cpp
@@ -39,9 +39,9 @@ bool CDevice::operator==(const CDevice& rhs) const
          Provider() == rhs.Provider() &&
          VendorID() == rhs.VendorID() &&
          ProductID() == rhs.ProductID() &&
-         ButtonCount() == rhs.ButtonCount() &&
-         HatCount() == rhs.HatCount() &&
-         AxisCount() == rhs.AxisCount() &&
+  //ButtonCount() == rhs.ButtonCount() &&
+  //HatCount() == rhs.HatCount() &&
+  //AxisCount() == rhs.AxisCount() &&
          Index() == rhs.Index();
 }
 
@@ -59,14 +59,14 @@ bool CDevice::operator<(const CDevice& rhs) const
   if (ProductID() < rhs.ProductID()) return true;
   if (ProductID() > rhs.ProductID()) return false;
 
-  if (ButtonCount() < rhs.ButtonCount()) return true;
-  if (ButtonCount() > rhs.ButtonCount()) return false;
+  //if (ButtonCount() < rhs.ButtonCount()) return true;
+  //if (ButtonCount() > rhs.ButtonCount()) return false;
 
-  if (HatCount() < rhs.HatCount()) return true;
-  if (HatCount() > rhs.HatCount()) return false;
+  //if (HatCount() < rhs.HatCount()) return true;
+  //if (HatCount() > rhs.HatCount()) return false;
 
-  if (AxisCount() < rhs.AxisCount()) return true;
-  if (AxisCount() > rhs.AxisCount()) return false;
+  //if (AxisCount() < rhs.AxisCount()) return true;
+  //if (AxisCount() > rhs.AxisCount()) return false;
 
   if (Index() < rhs.Index()) return true;
   if (Index() > rhs.Index()) return false;
@@ -94,15 +94,15 @@ bool CDevice::SimilarTo(const CDevice& other) const
     }
   }
 
-  if (AreElementCountsKnown() && other.AreElementCountsKnown())
-  {
-    if (ButtonCount() != other.ButtonCount() ||
-        HatCount() != other.HatCount() ||
-        AxisCount() != other.AxisCount())
-    {
-      return false;
-    }
-  }
+  //if (AreElementCountsKnown() && other.AreElementCountsKnown())
+  //{
+  //  if (ButtonCount() != other.ButtonCount() ||
+  //      HatCount() != other.HatCount() ||
+  //      AxisCount() != other.AxisCount())
+  //  {
+  //    return false;
+  //  }
+  //}
 
   return true;
 }
