diff --git a/scraper.go b/scraper.go
index 07e0222..20a63cf 100644
--- a/scraper.go
+++ b/scraper.go
@@ -124,7 +124,7 @@ func worker(ctx context.Context, sources []ds.DS, xmlOpts *rom.XMLOpts, gameOpts
 			if done(ctx) {
 				break
 			}
-			log.Printf("INFO: Starting: %s", r.Path)
+			log.Printf("GAME: %s", r.Path)
 			if err := r.GetGame(ctx, sources, gameOpts); err != nil {
 				log.Printf("ERR: error processing %s: %s", r.Path, err)
 				res.Err = err
