From f5d43adc72ea05f990216fb8042139c7be99194e Mon Sep 17 00:00:00 2001
From: whitehax0r <allan_23_12@hotmail.com>
Date: Sat, 11 Jun 2022 21:50:15 -0600
Subject: [PATCH] tools: Fix use-after-free for realloc(..., 0)

---
 tools/lib/subcmd/subcmd-util.h | 15 ++++-----------
 1 file changed, 4 insertions(+), 11 deletions(-)

diff --git a/tools/lib/subcmd/subcmd-util.h b/tools/lib/subcmd/subcmd-util.h
index 794a375dad36..8d866fdb7e3a 100644
--- a/tools/lib/subcmd/subcmd-util.h
+++ b/tools/lib/subcmd/subcmd-util.h
@@ -49,17 +49,10 @@ static NORETURN inline void die(const char *err, ...)
 
 static inline void *xrealloc(void *ptr, size_t size)
 {
-	void *ret = realloc(ptr, size);
-	if (!ret && !size)
-		ret = realloc(ptr, 1);
-	if (!ret) {
-		ret = realloc(ptr, size);
-		if (!ret && !size)
-			ret = realloc(ptr, 1);
-		if (!ret)
-			die("Out of memory, realloc failed");
-	}
-	return ret;
+       void *ret = realloc(ptr, size);
+       if (!ret)
+		die("Out of memory, realloc failed");
+       return ret;
 }
 
 #define astrcatf(out, fmt, ...)						\
