diff --git a/Makefile.libretro b/Makefile.libretro
index 0127d63..00915ef 100755
--- a/Makefile.libretro
+++ b/Makefile.libretro
@@ -4,6 +4,13 @@ ZLIB_DIR = $(LIBRETRO_DIR)/utils/zlib
 STATIC_LINKING=0
 STATIC_LINKING_LINK=0
 
+DEBUG = 0
+IPFSUPPORT = 1
+
+ifeq ($(IPFSUPPORT), 1)
+  CAPS_LIB = -L$(STAGING_DIR)/usr/lib -lcapsimage
+endif
+
 ifeq ($(platform),)
 platform = unix
 ifeq ($(shell uname -a),)
@@ -196,6 +203,10 @@ else
 endif
 CFLAGS := -fsigned-char -D__LIBRETRO__ -fno-builtin $(CFLAGS)
 
+ifeq ($(IPFSUPPORT), 1)
+    PLATFLAGS += -DHAVE_CAPSIMAGE=1
+endif
+
 CFLAGS := $(fpic) $(CFLAGS) $(PLATFLAGS)
 CXXFLAGS := $(CFLAGS)
 CPPFLAGS := $(CFLAGS)
@@ -216,11 +227,15 @@ OBJECTS := $(SOURCES_C:.c=.o)
 
 all: $(TARGET)
 
+ifeq ($(IPFSUPPORT), 1)
+    INCFLAGS += -I$(STAGING_DIR)/usr/lib/capsimage
+endif
+
 $(TARGET): $(OBJECTS)
 ifeq ($(STATIC_LINKING_LINK),1)
 	$(AR) rcs $@ $(OBJECTS) 
 else
-	$(CC) $(CFLAGS) $(INCFLAGS) $(LDFLAGS) $(OBJECTS) -o $@ -lm -lz $(SHARED)
+	$(CC) $(CFLAGS) $(INCFLAGS) $(LDFLAGS) $(OBJECTS) -o $@ -lm -lz $(SHARED) -lpthread $(CAPS_LIB)
 endif
 
 %.o: %.c
