diff --git a/Makefile.libretro b/Makefile.libretro
index e7527c7..c5225e5 100755
--- a/Makefile.libretro
+++ b/Makefile.libretro
@@ -4,6 +4,13 @@ ZLIB_DIR = $(LIBRETRO_DIR)/utils/zlib
 STATIC_LINKING=0
 STATIC_LINKING_LINK=0
 
+DEBUG = 0
+IPFSUPPORT = 1
+
+ifeq ($(IPFSUPPORT), 1)
+CAPS_LIB = -L$(STAGING_DIR)/usr/lib -lcapsimage
+endif
+
 ifeq ($(platform),)
 platform = unix
 ifeq ($(shell uname -a),)
@@ -77,6 +84,28 @@ else ifeq ($(platform), osx)
    SHARED := -dynamiclib
    PLATFLAGS := -DLSB_FIRST -DALIGN_DWORD
 
+# ARM
+else ifneq (,$(findstring armv,$(platform)))
+   CC = gcc
+	TARGET := $(TARGET_NAME)_libretro.so
+   fpic := -fPIC -fsigned-char
+   SHARED :=  -lz -lpthread -shared -Wl,--version-script=$(LIBRETRO_DIR)/link.T -Wl,--no-undefined
+   PLATFLAGS := -DLSB_FIRST -DALIGN_DWORD
+   CFLAGS += -marm
+ifneq (,$(findstring neon,$(platform)))
+   CFLAGS += -mfpu=neon
+   ASFLAGS += -mfpu=neon
+   HAVE_NEON = 1
+endif
+ifneq (,$(findstring softfloat,$(platform)))
+   CFLAGS += -mfloat-abi=softfp
+   ASFLAGS += -mfloat-abi=softfp
+else ifneq (,$(findstring hardfloat,$(platform)))
+   CFLAGS += -mfloat-abi=hard
+   ASFLAGS += -mfloat-abi=hard
+endif
+   CFLAGS += -DARM
+
 # iOS
 else ifneq (,$(findstring ios,$(platform)))
 
@@ -149,7 +178,7 @@ endif
 ifeq ($(DEBUG), 1)
    CFLAGS += -O0 -g
 else
-   CFLAGS += -O3
+   CFLAGS += -O2 -DNDEBUG
 endif
 
 CFLAGS += \
@@ -175,12 +204,17 @@ OBJECTS := $(SOURCES_C:.c=.o)
 
 all: $(TARGET)
 
+ifeq ($(IPFSUPPORT), 1)
+INCFLAGS += -I$(STAGING_DIR)/usr/lib/capsimage
+PLATFLAGS += -DHAVE_CAPSIMAGE=1
+endif
+
 $(TARGET): $(OBJECTS)
 	@echo "** BUILDING $(TARGET) FOR PLATFORM $(platform) **"
 ifeq ($(STATIC_LINKING_LINK),1)
 	$(AR) rcs $@ $(OBJECTS) 
 else
-	$(CC) $(fpic) $(INCLUDES) -o $@ $(OBJECTS)  -lm -lz $(SHARED)
+	$(CC) $(fpic) $(SHARED) $(INCLUDES) -o $@ $(OBJECTS)  -lm -lz -lpthread $(CAPS_LIB)
 endif
 	@echo "** BUILD SUCCESSFUL! GG NO RE **"
 
