diff -Nur hatari.orig/cmake/FindCapsImage.cmake hatari/cmake/FindCapsImage.cmake
--- hatari.orig/cmake/FindCapsImage.cmake	2020-04-14 21:53:32.417557052 +0200
+++ hatari/cmake/FindCapsImage.cmake	2020-04-14 23:53:21.690847156 +0200
@@ -5,17 +5,8 @@
 ENDIF (CAPSIMAGE_INCLUDE_DIR)
 
 
-# Choose the library version to use : 4 or 5
-SET(CAPSIMAGE_VERSION 4)
-
-
-if(CAPSIMAGE_VERSION STREQUAL 4)
-	SET(CAPSIMAGE_DIR caps)
-	FIND_PATH(CAPSIMAGE_INCLUDE_DIR ${CAPSIMAGE_DIR}/capsimage.h)
-else()
-	SET(CAPSIMAGE_DIR caps5)
-	FIND_PATH(CAPSIMAGE_INCLUDE_DIR ${CAPSIMAGE_DIR}/CapsAPI.h)
-endif()
+SET(CAPSIMAGE_DIR caps5)
+FIND_PATH(CAPSIMAGE_INCLUDE_DIR ${CAPSIMAGE_DIR}/CapsAPI.h)
 
 if(WIN32)
 	FIND_LIBRARY(CAPSIMAGE_LIBRARY NAMES capsimg PATH_SUFFIXES ${CAPSIMAGE_DIR} )
diff -Nur hatari.orig/src/floppy_ipf.c hatari/src/floppy_ipf.c
--- hatari.orig/src/floppy_ipf.c	2020-04-14 21:53:32.540560349 +0200
+++ hatari/src/floppy_ipf.c	2020-04-14 23:52:52.490989256 +0200
@@ -26,13 +26,9 @@
 #include "cycles.h"
 
 #ifdef HAVE_CAPSIMAGE
-#if CAPSIMAGE_VERSION == 5
-#include <caps5/CapsLibAll.h>
-#else
-#include <caps/fdc.h>
-#define CAPS_LIB_RELEASE	4
-#define CAPS_LIB_REVISION	2
-#endif
+#include <CapsLibAll.h>
+#define CAPS_LIB_RELEASE	5
+#define CAPS_LIB_REVISION	1
 /* Macro to check release and revision */
 #define	CAPS_LIB_REL_REV	( CAPS_LIB_RELEASE * 100 + CAPS_LIB_REVISION )
 #endif
