--- a/src/uae-cpu/newcpu.c	2021-01-02 12:26:33.000000000 +0100
+++ b/src/uae-cpu/newcpu.c	2021-01-07 18:16:09.515721017 +0100
@@ -165,12 +165,6 @@ const char NewCpu_fileid[] = "Hatari newcpu.c : " __DATE__ " " __TIME__;
 #include "debugcpu.h"
 #include "68kDisass.h"
 
-#ifdef HAVE_CAPSIMAGE
-#if CAPSIMAGE_VERSION == 5
-#include <CapsLibAll.h>
-#endif
-#endif
-
 //#define DEBUG_PREFETCH
 
 struct flag_struct regflags;
