--- a/source/frontends/sdl/CMakeLists.txt	2023-08-02 00:45:55.274179545 +0200
+++ b/source/frontends/sdl/CMakeLists.txt	2023-08-02 00:46:17.498625505 +0200
@@ -10,13 +10,7 @@
 # we should use find_package, but Ubuntu does not provide it for SDL2_image
 pkg_search_module(SDL2_IMAGE REQUIRED SDL2_image)
 
-
-if (${CMAKE_SYSTEM_NAME} MATCHES "Darwin")
-  # only OpenGL supported on MacOS
-  set (SA2_OPENGL ON)
-else()
-  option(SA2_OPENGL "Prefer OpenGL over OpenGL ES" OFF)
-endif()
+option(SA2_OPENGL "Prefer OpenGL over OpenGL ES" OFF)
 
 if (SA2_OPENGL)
   find_package(OpenGL REQUIRED)
@@ -50,15 +50,25 @@
   ${GLES2_INCLUDE_DIRS}
   )

+if (SA2_OPENGL)
 target_link_libraries(sa2 PRIVATE
   ${SDL2_LIBRARIES}
   ${SDL2_IMAGE_LIBRARIES}
-  ${GLES2_LIBRARIES}
   OpenGL::GL
   appleii
   common2
   windows
   )
+else()
+target_link_libraries(sa2 PRIVATE
+  ${SDL2_LIBRARIES}
+  ${SDL2_IMAGE_LIBRARIES}
+  ${GLES2_LIBRARIES}
+  appleii
+  common2
+  windows
+  )
+endif()

 target_link_directories(sa2 PRIVATE
   ${SDL2_LIBRARY_DIRS}
