--- a/source/frontends/sdl/CMakeLists.txt	2023-08-02 00:45:55.274179545 +0200
+++ b/source/frontends/sdl/CMakeLists.txt	2023-08-02 00:46:17.498625505 +0200
@@ -48,21 +48,38 @@ target_include_directories(sa2 PRIVATE
   ${GLES2_INCLUDE_DIRS}
   )
 
-target_link_libraries(sa2 PRIVATE
-  appleii
-  common2
-
-  ${YAML_LIBRARIES}
-  ${MINIZIP_LIBRARIES}
-  ${PCAP_LIBRARIES}
-  ${SLIRP_LIBRARIES}
-  ${ZLIB_LIBRARIES}
-
-  ${SDL2_LIBRARIES}
-  ${SDL2_IMAGE_LIBRARIES}
-  ${GLES2_LIBRARIES}
-  OpenGL::GL
-  )
+if (SA2_OPENGL)
+	target_link_libraries(sa2 PRIVATE
+	  appleii
+	  common2
+
+	  ${YAML_LIBRARIES}
+	  ${MINIZIP_LIBRARIES}
+	  ${PCAP_LIBRARIES}
+	  ${SLIRP_LIBRARIES}
+	  ${ZLIB_LIBRARIES}
+
+	  ${SDL2_LIBRARIES}
+	  ${SDL2_IMAGE_LIBRARIES}
+	  OpenGL::GL
+	  )
+else()
+	target_link_libraries(sa2 PRIVATE
+	  appleii
+	  common2
+
+	  ${YAML_LIBRARIES}
+	  ${MINIZIP_LIBRARIES}
+	  ${PCAP_LIBRARIES}
+	  ${SLIRP_LIBRARIES}
+	  ${ZLIB_LIBRARIES}
+
+	  ${SDL2_LIBRARIES}
+	  ${SDL2_IMAGE_LIBRARIES}
+	  ${GLES2_LIBRARIES}
+	  OpenGL::GL
+	  )
+endif()
 
 if (NOT WIN32)
   target_link_libraries(sa2 PUBLIC