--- a/arch/arm64/boot/dts/rockchip/px30.dtsi	2021-05-08 16:25:00.323305144 +0200
+++ b/arch/arm64/boot/dts/rockchip/px30.dtsi	2021-05-08 16:25:40.579970307 +0200
@@ -1094,6 +1094,17 @@
 		};
 	};
 
+	rk_rga: rk_rga@ff480000 {
+		compatible = "rockchip,rga2";
+		//dev_mode = <1>;
+		reg = <0x0 0xff480000 0x0 0x1000>;
+		interrupts = <GIC_SPI 76 IRQ_TYPE_LEVEL_HIGH>;
+		clocks = <&cru ACLK_RGA>, <&cru HCLK_RGA>, <&cru SCLK_RGA_CORE>;
+		clock-names = "aclk_rga", "hclk_rga", "clk_rga";
+		power-domains = <&power PX30_PD_VO>;
+		status = "disabled";
+	};
+
 	vopl_mmu: iommu@ff470f00 {
 		compatible = "rockchip,iommu";
 		reg = <0x0 0xff470f00 0x0 0x100>;
