diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index 5138f3e..ea782ec 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -322,8 +322,8 @@ endif(WIN32)
 
 if(SDL)
   include(FindPkgConfig)
-  pkg_check_modules(SDL REQUIRED sdl)
-  include_directories(${SDL_INCLUDE_DIRS})
+  pkg_check_modules(SDL REQUIRED sdl2)
+  include_directories(${SDL2_INCLUDE_DIRS})
   add_definitions(
 	-DUSE_SDL
   )
@@ -559,9 +559,9 @@ if(GLIDEN64_BUILD_TYPE STREQUAL "Debug")
 
   if(SDL)
         if (NOHQ)
-	  target_link_libraries(${GLideN64_DLL_NAME} PRIVATE ${OPENGL_LIBRARIES} ${SDL_LIBRARIES} ${FREETYPE_LIBRARIES} ${LOG_LIB} ${GLIDENUI_LIBRARIES} osald )
+	  target_link_libraries(${GLideN64_DLL_NAME} PRIVATE ${OPENGL_LIBRARIES} ${SDL2_LIBRARIES} ${FREETYPE_LIBRARIES} ${LOG_LIB} ${GLIDENUI_LIBRARIES} osald )
 	else (NOHQ)
-	  target_link_libraries(${GLideN64_DLL_NAME} PRIVATE ${OPENGL_LIBRARIES} ${SDL_LIBRARIES} ${FREETYPE_LIBRARIES} ${LOG_LIB} ${GLIDENUI_LIBRARIES} osald GLideNHQd )
+	  target_link_libraries(${GLideN64_DLL_NAME} PRIVATE ${OPENGL_LIBRARIES} ${SDL2_LIBRARIES} ${FREETYPE_LIBRARIES} ${LOG_LIB} ${GLIDENUI_LIBRARIES} osald GLideNHQd )
 	endif (NOHQ)
   else(SDL)
 	if (NOHQ)
@@ -573,9 +573,9 @@ if(GLIDEN64_BUILD_TYPE STREQUAL "Debug")
 else(GLIDEN64_BUILD_TYPE STREQUAL "Debug")
   if(SDL)
         if (NOHQ)
-          target_link_libraries(${GLideN64_DLL_NAME} ${OPENGL_LIBRARIES} ${SDL_LIBRARIES} ${FREETYPE_LIBRARIES} ${LOG_LIB} ${GLIDENUI_LIBRARIES} osal )
+          target_link_libraries(${GLideN64_DLL_NAME} ${OPENGL_LIBRARIES} ${SDL2_LIBRARIES} ${FREETYPE_LIBRARIES} ${LOG_LIB} ${GLIDENUI_LIBRARIES} osal )
         else (NOHQ)
-          target_link_libraries(${GLideN64_DLL_NAME} ${OPENGL_LIBRARIES} ${SDL_LIBRARIES} ${FREETYPE_LIBRARIES} ${LOG_LIB} ${GLIDENUI_LIBRARIES} osal GLideNHQ )
+          target_link_libraries(${GLideN64_DLL_NAME} ${OPENGL_LIBRARIES} ${SDL2_LIBRARIES} ${FREETYPE_LIBRARIES} ${LOG_LIB} ${GLIDENUI_LIBRARIES} osal GLideNHQ )
         endif (NOHQ)
   else(SDL)
         if (NOHQ)
