diff -ru a/src/RSP_LoadMatrixX86.cpp b/src/RSP_LoadMatrixX86.cpp
--- a/src/RSP_LoadMatrixX86.cpp	2020-05-30 09:55:43.955175321 +0200
+++ b/src/RSP_LoadMatrixX86.cpp	2020-05-30 10:03:20.370696021 +0200
@@ -1,3 +1,4 @@
+#include <stdint.h>
 #include "RSP.h"
 #include "GBI.h"
 
@@ -89,7 +90,7 @@
     "    loop    LoadLoop"                       "\n\t"
     ".att_syntax prefix"                         "\n\t"
     : /* no output */
-    : "f"(recip), "S"((int)RDRAM+address), "D"(mtx), "c"(4)
+    : "f"(recip), "S"((intptr_t)RDRAM+address), "D"(mtx), "c"(4)
     : "memory" );
 #endif // WIN32_ASM
 }
