diff --git a/src/r8168_n.c b/src/r8168_n.c
old mode 100644
new mode 100755
index 19ae62bec6..35318ff1e2
--- a/src/r8168_n.c
+++ b/src/r8168_n.c
@@ -6293,8 +6293,13 @@ rtl8168_get_drvinfo(struct net_device *dev,
         info->eedump_len = tp->eeprom_len;
         BUILD_BUG_ON(sizeof(info->fw_version) < sizeof(rtl_fw->version));
         if (rtl_fw)
+                #if (LINUX_VERSION_CODE < KERNEL_VERSION(6, 8, 0))
                 strlcpy(info->fw_version, rtl_fw->version,
                         sizeof(info->fw_version));
+                #else
+                strscpy(info->fw_version, rtl_fw->version,
+                        sizeof(info->fw_version));
+                #endif
 }
 
 static int
