diff --git a/Oxygen/oxygenengine/source/oxygen/application/EngineMain.cpp b/Oxygen/oxygenengine/source/oxygen/application/EngineMain.cpp
index ed423975..ce65be59 100644
--- a/Oxygen/oxygenengine/source/oxygen/application/EngineMain.cpp
+++ b/Oxygen/oxygenengine/source/oxygen/application/EngineMain.cpp
@@ -444,7 +444,7 @@ bool EngineMain::initConfigAndSettings(const std::wstring& argumentProjectPath)
 #if (defined(PLATFORM_MAC) || defined(PLATFORM_IOS)) && defined(ENDUSER)
 		config.loadConfiguration(config.mGameDataPath + L"/config.json");
 #else
-		config.loadConfiguration(L"config.json");
+		config.loadConfiguration(config.mAppDataPath + L"config.json");
 #endif
 	}
 
diff --git a/Oxygen/oxygenengine/source/oxygen/resources/ResourcesCache.cpp b/Oxygen/oxygenengine/source/oxygen/resources/ResourcesCache.cpp
index ad0383c1..ff645326 100644
--- a/Oxygen/oxygenengine/source/oxygen/resources/ResourcesCache.cpp
+++ b/Oxygen/oxygenengine/source/oxygen/resources/ResourcesCache.cpp
@@ -63,7 +63,7 @@ bool ResourcesCache::loadRom()
 	{
 		for (const GameProfile::RomInfo& romInfo : gameProfile.mRomInfos)
 		{
-			romPath = romInfo.mSteamRomName;
+			romPath = L"/userdata/roms/sonic3-air/Sonic_Knuckles_wSonic3.bin";
 			loaded = loadRomFile(romPath, romInfo);
 			if (loaded)
 				break;
diff --git a/Oxygen/oxygenserver/source/oxygenserver/main_server.cpp b/Oxygen/oxygenserver/source/oxygenserver/main_server.cpp
index 0d0f4bbe..7bfed082 100644
--- a/Oxygen/oxygenserver/source/oxygenserver/main_server.cpp
+++ b/Oxygen/oxygenserver/source/oxygenserver/main_server.cpp
@@ -77,7 +77,7 @@ int main(int argc, char** argv)
 #endif
 
 	Configuration config;
-	config.loadConfiguration(L"config.json");
+	config.loadConfiguration(config.mAppDataPath + L"config.json");
 
 	Sockets::startupSockets();
 	{
