diff --git a/Oxygen/sonic3air/build/_cmake/CMakeLists.txt b/Oxygen/sonic3air/build/_cmake/CMakeLists.txt
index 2a43f7db..de6e8b0d 100644
--- a/Oxygen/sonic3air/build/_cmake/CMakeLists.txt
+++ b/Oxygen/sonic3air/build/_cmake/CMakeLists.txt
@@ -9,7 +9,7 @@ project(Sonic3AIR)
 
 
 # Determine whether to support discord integration
-set(USE_DISCORD true)
+set(USE_DISCORD false)
 if (UNIX AND CMAKE_SYSTEM_PROCESSOR MATCHES "^arm")
 	link_directories(/opt/vc/lib)	# Needed when building on Raspberry Pi
 	set(USE_DISCORD false)
diff --git a/Oxygen/sonic3air/source/sonic3air/helper/DiscordIntegration.cpp b/Oxygen/sonic3air/source/sonic3air/helper/DiscordIntegration.cpp
index 205dc9a3..b98113d0 100644
--- a/Oxygen/sonic3air/source/sonic3air/helper/DiscordIntegration.cpp
+++ b/Oxygen/sonic3air/source/sonic3air/helper/DiscordIntegration.cpp
@@ -13,7 +13,7 @@
 //  - Windows
 //  - Linux, except if it's ARM architecture (like when building on a RasPi)
 //  - Mac. Sorry about having 3 checks. Command line build has issues finding the discord dylib since it can't be embedded. And temporarily ignore discord for ARM64 until they add support.
-#if (defined(PLATFORM_WINDOWS) && !defined(__GNUC__)) || (defined(PLATFORM_LINUX) && !defined(__arm__)) || (defined(PLATFORM_MAC) && !defined(NO_DISCORD))
+#if (defined(PLATFORM_WINDOWS) && !defined(__GNUC__)) || (!defined(PLATFORM_LINUX) && !defined(__arm__)) || (defined(PLATFORM_MAC) && !defined(NO_DISCORD))
 	#define SUPPORT_DISCORD
 #endif
 
