diff --git a/Makefile b/Makefile
index 3269005..a215089 100644
--- a/Makefile
+++ b/Makefile
@@ -58,6 +58,14 @@ else ifneq (,$(findstring rpi,$(platform)))
    override platform += unix
 else ifneq (,$(findstring odroid,$(platform)))
    override platform += unix
+else ifneq (,$(findstring rockpro64,$(platform)))
+   override platform += unix
+else ifneq (,$(findstring rk3588,$(platform)))
+   override platform += unix
+else ifneq (,$(findstring n2,$(platform)))
+   override platform += unix
+   else ifneq (,$(findstring h5,$(platform)))
+   override platform += unix
 endif
 
 # system platform
@@ -150,6 +156,10 @@ ifneq (,$(findstring unix,$(platform)))
          CPUFLAGS += -DNO_ASM -DARM -D__arm__ -DARM_ASM -D__NEON_OPT -DNOSSE
          CPUFLAGS += -mcpu=cortex-a53 -mfpu=neon-fp-armv8 -mfloat-abi=hard
          HAVE_NEON = 1
+      else ifneq (,$(findstring rpi4,$(platform)))
+         CPUFLAGS += -DNO_ASM -DARM -D__arm__ -DARM_ASM -D__NEON_OPT -DNOSSE
+         CPUFLAGS += -mcpu=cortex-a72 -mfpu=neon-fp-armv8 -mfloat-abi=hard
+         HAVE_NEON = 1
       # Raspberry pi 4 in 64bit mode with VULKAN
       else ifneq (,$(findstring rpi4_64,$(platform)))
          CPUFLAGS += -DNO_ASM -DARM -DARM_ASM -DDONT_WANT_ARM_OPTIMIZATIONS -DARM_FIX -DCLASSIC -DARM64
@@ -166,6 +176,47 @@ ifneq (,$(findstring unix,$(platform)))
 
    endif
    
+   # RockPro64
+   ifneq (,$(findstring rockpro64,$(platform)))
+      GLES = 1
+      GL_LIB := -lGLESv2
+      CPUFLAGS += -DNO_ASM -DARM -DARM_ASM -DDONT_WANT_ARM_OPTIMIZATIONS -DARM_FIX -DCLASSIC -DARM64
+ #    CPUFLAGS += -marm -mfloat-abi=hard -mfpu=neon
+      HAVE_NEON = 1
+      WITH_DYNAREC=aarch64
+      CPUFLAGS += -D__NEON_OPT -march=armv8-a+crc -mcpu=cortex-a72 -mtune=cortex-a72.cortex-a53
+   endif
+
+   # ODROIDN2
+   ifneq (,$(findstring n2,$(platform)))
+      GLES = 1
+      GL_LIB := -lGLESv2
+      CPUFLAGS += -DNO_ASM -DARM -DARM_ASM -DDONT_WANT_ARM_OPTIMIZATIONS -DARM_FIX -DCLASSIC -DARM64
+      HAVE_NEON = 1
+      WITH_DYNAREC=aarch64
+      CPUFLAGS += -D__NEON_OPT -march=armv8-a+crc -mcpu=cortex-a73 -mtune=cortex-a73.cortex-a53
+   endif
+
+   # RK3588
+   ifneq (,$(findstring rk3588,$(platform)))
+      GLES = 1
+      GL_LIB := -lmali_hook -Wl,--whole-archive -lmali_hook_injector -Wl,--no-whole-archive -lmali
+      CPUFLAGS += -DNO_ASM -DARM -DARM_ASM -DDONT_WANT_ARM_OPTIMIZATIONS -DARM_FIX -DCLASSIC -DARM64
+      HAVE_NEON = 1
+      WITH_DYNAREC=aarch64
+      CPUFLAGS += -D__NEON_OPT -march=armv8.2-a+crypto+dotprod+rcpc+sb+ssbs -mcpu=cortex-a76.cortex-a55 -mtune=cortex-a76.cortex-a55
+   endif
+   
+      # Allwinner H5
+   ifneq (,$(findstring h5,$(platform)))
+      GLES = 1
+      GL_LIB := -lGLESv2
+      CPUFLAGS += -DNO_ASM -DARM -DARM_ASM -DDONT_WANT_ARM_OPTIMIZATIONS -DARM_FIX -DCLASSIC -DARM64
+      HAVE_NEON = 1
+      WITH_DYNAREC=aarch64
+      CPUFLAGS += -D__NEON_OPT -march=armv8-a+crc -mcpu=cortex-a53 -mtune=cortex-a53
+   endif
+
    # ODROIDs
    ifneq (,$(findstring odroid,$(platform)))
       BOARD ?= $(shell cat /proc/cpuinfo | grep -i odroid | awk '{print $$3}')
@@ -181,6 +222,9 @@ ifneq (,$(findstring unix,$(platform)))
       else ifneq (,$(findstring ODROID-XU,$(BOARD)))
          # ODROID-XU3 XU4 and XU3-Lite
          CPUFLAGS += -mcpu=cortex-a15 -mtune=cortex-a15.cortex-a7 -mfpu=neon-vfpv4
+      # ODROIDGOA
+      else ifneq (,$(findstring ODROIDGOA,$(BOARD)))
+         CPUFLAGS += -march=armv8-a+crc -mfpu=neon-fp-armv8 -mcpu=cortex-a35 -mtune=cortex-a35
       else
          # ODROID-U3, U2, X2 & X
           CPUFLAGS += -mcpu=cortex-a9 -mfpu=neon
