diff --git a/Makefile b/Makefile
index bfc5a47..f986efe 100644
--- a/Makefile
+++ b/Makefile
@@ -58,6 +58,10 @@ else ifneq (,$(findstring rpi,$(platform)))
    override platform += unix
 else ifneq (,$(findstring odroid,$(platform)))
    override platform += unix
+else ifneq (,$(findstring rockpro64,$(platform)))
+   override platform += unix
+else ifneq (,$(findstring n2,$(platform)))
+   override platform += unix
 endif
 
 # system platform
@@ -159,6 +163,27 @@ ifneq (,$(findstring unix,$(platform)))
       CPUFLAGS += -DARM_FIX
    endif
 
+   # RockPro64
+   ifneq (,$(findstring rockpro64,$(platform)))
+      GLES = 1
+      GL_LIB := -lGLESv2
+      CPUFLAGS += -DNO_ASM -DARM -D__arm__ -DARM_ASM -D__NEON_OPT -DNOSSE -DARM_FIX
+      CPUFLAGS += -marm -mfloat-abi=hard -mfpu=neon
+      HAVE_NEON = 1
+      WITH_DYNAREC=arm
+      CPUFLAGS += -march=armv7ve -mcpu=cortex-a72 -mtune=cortex-a72.cortex-a53
+   endif
+
+   # ODROIDN2
+   ifneq (,$(findstring n2,$(platform)))
+      GLES = 1
+      GL_LIB := -lGLESv2
+      CPUFLAGS += -DNO_ASM -DARM -D__arm__ -DARM_ASM -D__NEON_OPT -DNOSSE -DARM_FIX
+      HAVE_NEON = 1
+      WITH_DYNAREC=aarch64
+      CPUFLAGS += -march=armv8-a+crc -mcpu=cortex-a73 -mtune=cortex-a73.cortex-a53
+   endif
+   
    # ODROIDs
    ifneq (,$(findstring odroid,$(platform)))
       BOARD ?= $(shell cat /proc/cpuinfo | grep -i odroid | awk '{print $$3}')
@@ -174,6 +199,9 @@ ifneq (,$(findstring unix,$(platform)))
       else ifneq (,$(findstring ODROID-XU,$(BOARD)))
          # ODROID-XU3 XU4 and XU3-Lite
          CPUFLAGS += -mcpu=cortex-a15 -mtune=cortex-a15.cortex-a7 -mfpu=neon-vfpv4
+      # ODROIDGOA
+      else ifneq (,$(findstring ODROIDGOA,$(BOARD)))
+        CPUFLAGS += -march=armv8-a+crc -mfpu=neon-fp-armv8 -mcpu=cortex-a35 -mtune=cortex-a35
       else
          # ODROID-U3, U2, X2 & X
          CPUFLAGS += -mcpu=cortex-a9 -mfpu=neon
