diff --git a/Makefile b/Makefile
index 14124d5..14e2910 100644
--- a/Makefile
+++ b/Makefile
@@ -56,6 +56,8 @@ else ifneq (,$(findstring rpi,$(platform)))
    override platform += unix
 else ifneq (,$(findstring odroid,$(platform)))
    override platform += unix
+else ifneq (,$(findstring rockpro64,$(platform)))
+   override platform += unix
 endif
 
 # system platform
@@ -155,6 +157,17 @@ ifneq (,$(findstring unix,$(platform)))
       endif
    endif
 
+   # RockPro64
+   ifneq (,$(findstring rockpro64,$(platform)))
+      GLES = 1
+      GL_LIB := -lGLESv2
+      CPUFLAGS += -DNO_ASM -DARM -D__arm__ -DARM_ASM -D__NEON_OPT -DNOSSE
+      CPUFLAGS += -marm -mfloat-abi=hard -mfpu=neon
+      HAVE_NEON = 1
+      WITH_DYNAREC=arm
+      CPUFLAGS += -march=armv7ve -mcpu=cortex-a72 -mtune=cortex-a72.cortex-a53
+   endif
+
    # ODROIDs
    ifneq (,$(findstring odroid,$(platform)))
       BOARD := $(shell cat /proc/cpuinfo | grep -i odroid | awk '{print $$3}')
