diff --git a/Makefile b/Makefile
index 14e2910..5d4d9d8 100644
--- a/Makefile
+++ b/Makefile
@@ -149,7 +149,7 @@ ifneq (,$(findstring unix,$(platform)))
          CPUFLAGS += -mcpu=cortex-a7 -mfpu=neon-vfpv4 -mfloat-abi=hard
          HAVE_NEON = 1
       else ifneq (,$(findstring rpi3,$(platform)))
-         CPUFLAGS += -DNO_ASM -DARM -D__arm__ -DARM_ASM -D__NEON_OPT -DNOSSE
+         CPUFLAGS += -DNO_ASM -DARM -D__arm__ -DARM_ASM -DARM_FIX -D__NEON_OPT -DNOSSE
          CPUFLAGS += -mcpu=cortex-a53 -mfpu=neon-fp-armv8 -mfloat-abi=hard
          HAVE_NEON = 1
       else
@@ -161,7 +161,7 @@ ifneq (,$(findstring unix,$(platform)))
    ifneq (,$(findstring rockpro64,$(platform)))
       GLES = 1
       GL_LIB := -lGLESv2
-      CPUFLAGS += -DNO_ASM -DARM -D__arm__ -DARM_ASM -D__NEON_OPT -DNOSSE
+      CPUFLAGS += -DNO_ASM -DARM -D__arm__ -DARM_ASM -DARM_FIX -D__NEON_OPT -DNOSSE
       CPUFLAGS += -marm -mfloat-abi=hard -mfpu=neon
       HAVE_NEON = 1
       WITH_DYNAREC=arm
@@ -173,7 +173,7 @@ ifneq (,$(findstring unix,$(platform)))
       BOARD := $(shell cat /proc/cpuinfo | grep -i odroid | awk '{print $$3}')
       GLES = 1
       GL_LIB := -lGLESv2
-      CPUFLAGS += -DNO_ASM -DARM -D__arm__ -DARM_ASM -D__NEON_OPT -DNOSSE
+      CPUFLAGS += -DNO_ASM -DARM -D__arm__ -DARM_ASM -DARM_FIX -D__NEON_OPT -DNOSSE
       CPUFLAGS += -marm -mfloat-abi=hard
       HAVE_NEON = 1
       WITH_DYNAREC=arm
