diff --git a/Makefile b/Makefile
index 9cf9b6c516..c4927c31f6 100644
--- a/Makefile
+++ b/Makefile
@@ -177,6 +177,16 @@ ifneq (,$(findstring unix,$(platform)))
          HAVE_OPENGL = 0
          GLES = 0
          GL_LIB :=
+      # Raspberry pi 5 in 64bit mode with VULKAN
+      else ifneq (,$(findstring rpi5_64,$(platform)))
+         CPUFLAGS += -DNO_ASM -DARM -DARM_ASM -DDONT_WANT_ARM_OPTIMIZATIONS -DARM_FIX -DCLASSIC -DARM64
+         CPUFLAGS += -march=armv8.2-a+fp16+simd -mcpu=cortex-a76 -mtune=cortex-a76
+         HAVE_PARALLEL = 1
+         WITH_DYNAREC = aarch64
+         HAVE_NEON = 0
+         HAVE_OPENGL = 0
+         GLES = 0
+         GL_LIB :=
       else
          CPUFLAGS += -DARMv5_ONLY -DNO_ASM
       endif
