diff --git a/Makefile b/Makefile
index e8eb343ea4..e9c8b5f0c3 100644
--- a/Makefile
+++ b/Makefile
@@ -64,6 +64,8 @@ else ifneq (,$(findstring rockpro64,$(platform)))
    override platform += unix
 else ifneq (,$(findstring rk3588,$(platform)))
    override platform += unix
+else ifneq (,$(findstring sm8250,$(platform)))
+   override platform += unix
 else ifneq (,$(findstring n2,$(platform)))
    override platform += unix
    else ifneq (,$(findstring h5,$(platform)))
@@ -233,6 +235,16 @@ ifneq (,$(findstring unix,$(platform)))
       WITH_DYNAREC=aarch64
       CPUFLAGS += -D__NEON_OPT -march=armv8-a+crc -mcpu=cortex-a53 -mtune=cortex-a53
    endif
+   
+   # Snapdragon SM8250
+   ifneq (,$(findstring sm8250,$(platform)))
+      GLES = 1
+      GL_LIB := -lGLESv2
+      CPUFLAGS += -DNO_ASM -DARM -DARM_ASM -DDONT_WANT_ARM_OPTIMIZATIONS -DARM_FIX -DCLASSIC -DARM64
+      HAVE_NEON = 1
+      WITH_DYNAREC=aarch64
+      CPUFLAGS += -D__NEON_OPT -march=armv8-a+crc -mcpu=cortex-a76 -mtune=cortex-a76
+   endif
 
    # ODROIDs
    ifneq (,$(findstring odroid,$(platform)))
