diff --git a/libfsemu/include/fs/ml.h b/libfsemu/include/fs/ml.h
index 9b97618..ae56859 100644
--- a/libfsemu/include/fs/ml.h
+++ b/libfsemu/include/fs/ml.h
@@ -289,6 +289,8 @@ void fs_ml_set_input_function(fs_ml_input_function function);
 #define FS_ML_MOUSE 1
 #define FS_ML_JOYSTICK 2
 
+#define FS_JOYSTICK_GUID_SIZE 64
+
 typedef struct fs_ml_input_device {
     int index;
     int type;
@@ -299,6 +301,7 @@ typedef struct fs_ml_input_device {
     int hats;
     int usage;
     int balls;
+    char guid[FS_JOYSTICK_GUID_SIZE];
 } fs_ml_input_device;
 
 int fs_ml_input_device_count();
diff --git a/libfsemu/src/emu/input.c b/libfsemu/src/emu/input.c
index 732326f..bdca511 100644
--- a/libfsemu/src/emu/input.c
+++ b/libfsemu/src/emu/input.c
@@ -511,6 +511,9 @@ static void map_custom_joystick_actions_2(
 
 static char *joystick_long_config_name(const fs_ml_input_device *device)
 {
+  char *resultBatocera = g_strdup_printf("%s_%s", device->guid, OS_NAME);
+  return resultBatocera;
+
     const char *in = device->name;
     char *result = g_malloc(strlen(device->name) + 1);
     char *out = result;
diff --git a/libfsemu/src/ml/input.c b/libfsemu/src/ml/input.c
index 4a282dd..2f849b4 100644
--- a/libfsemu/src/ml/input.c
+++ b/libfsemu/src/ml/input.c
@@ -315,6 +315,7 @@ void fs_ml_input_init()
         g_fs_ml_input_devices[k].buttons = SDL_JoystickNumButtons(joystick);
         g_fs_ml_input_devices[k].axes = SDL_JoystickNumAxes(joystick);
         g_fs_ml_input_devices[k].balls = SDL_JoystickNumBalls(joystick);
+	SDL_JoystickGetGUIDString(SDL_JoystickGetGUID(joystick), g_fs_ml_input_devices[k].guid, FS_JOYSTICK_GUID_SIZE);
 
         fs_log("[INPUT] Joystick device #%02d found: %s\n", i + 1, name);
         fs_log("[INPUT] %d buttons %d hats %d axes %d balls\n",
