From 51f1a1987c0f24f018ca76f7f15cf413410c5f30 Mon Sep 17 00:00:00 2001
From: Yang Deokgyu <secugyu@gmail.com>
Date: Thu, 5 Dec 2019 10:09:00 +0900
Subject: [PATCH 25/28] ODROID: arm/exynos: No need to use enynos_init_late

Signed-off-by: Yang Deokgyu <secugyu@gmail.com>
---
 arch/arm/mach-exynos/exynos.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/arch/arm/mach-exynos/exynos.c b/arch/arm/mach-exynos/exynos.c
index 1413837e1c44..c57032c68ac4 100644
--- a/arch/arm/mach-exynos/exynos.c
+++ b/arch/arm/mach-exynos/exynos.c
@@ -228,7 +228,7 @@ MACHINE_END
 		.init_early	= exynos_firmware_init,		\
 		.init_irq	= exynos_init_irq,		\
 		.init_machine	= exynos_dt_machine_init,	\
-		.init_late	= exynos_init_late,		\
+		.init_late	= exynos_pm_init,		\
 		.dt_compat	= compat,			\
 		.dt_fixup	= exynos_dt_fixup,		\
 	MACHINE_END
-- 
2.17.1

