From d91b8b8c7d58e6a26eaddf3014239f79f5fe869b Mon Sep 17 00:00:00 2001
From: MastaG <mastag@gmail.com>
Date: Tue, 17 Dec 2019 09:33:19 +0100
Subject: [PATCH 23/25] MEMEKA: fix ODROID-XU3 headphone-jack volume

---
 sound/soc/codecs/max98090.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/sound/soc/codecs/max98090.c b/sound/soc/codecs/max98090.c
index 032adc14562d..6448a4186618 100644
--- a/sound/soc/codecs/max98090.c
+++ b/sound/soc/codecs/max98090.c
@@ -651,7 +651,7 @@ static const struct snd_kcontrol_new max98090_snd_controls[] = {
 
 	SOC_DOUBLE_R_TLV("Headphone Volume", M98090_REG_LEFT_HP_VOLUME,
 		M98090_REG_RIGHT_HP_VOLUME, M98090_HPVOLL_SHIFT,
-		M98090_HPVOLL_NUM - 1, 0, max98090_hp_tlv),
+		M98090_HPVOLL_NUM - 1 + 24, 0, max98090_hp_tlv),
 
 	SOC_DOUBLE_R_RANGE_TLV("Speaker Volume",
 		M98090_REG_LEFT_SPK_VOLUME, M98090_REG_RIGHT_SPK_VOLUME,
-- 
2.17.1

