diff --git a/core/sdl/sdl.cpp b/core/sdl/sdl.cpp
index 15ace1b..fa2f7b4 100644
--- a/core/sdl/sdl.cpp
+++ b/core/sdl/sdl.cpp
@@ -501,11 +501,8 @@ void sdl_window_create()
 	int window_height = cfgLoadInt("x11","height", WINDOW_HEIGHT);
 
 	int flags = SDL_WINDOW_OPENGL;
-	#ifdef TARGET_PANDORA
-		flags |= SDL_FULLSCREEN;
-	#else
-		flags |= SDL_SWSURFACE;
-	#endif
+	flags |= SDL_WINDOW_FULLSCREEN;
+	flags |= SDL_SWSURFACE;
 
 	#ifdef GLES
 		SDL_GL_SetAttribute(SDL_GL_CONTEXT_PROFILE_MASK, SDL_GL_CONTEXT_PROFILE_ES);
@@ -520,7 +517,6 @@ void sdl_window_create()
 	SDL_GL_SetAttribute(SDL_GL_RED_SIZE, 8);
 	SDL_GL_SetAttribute(SDL_GL_GREEN_SIZE, 8);
 	SDL_GL_SetAttribute(SDL_GL_BLUE_SIZE, 8);
-	SDL_GL_SetAttribute(SDL_GL_ALPHA_SIZE, 8);
 	SDL_GL_SetAttribute(SDL_GL_DEPTH_SIZE, 24);
 	SDL_GL_SetAttribute(SDL_GL_STENCIL_SIZE, 8);
 	SDL_GL_SetAttribute(SDL_GL_DOUBLEBUFFER, 1);
