--- a/CMakeLists.txt	2023-08-10 16:42:52.112652740 +0200
+++ b/CMakeLists.txt	2023-08-10 16:43:37.116077776 +0200
@@ -423,11 +423,11 @@
 
 if(ENABLE_LAUNCHER OR ENABLE_EDITOR)
 	# Widgets finds its own dependencies (QtGui and QtCore).
-	find_package(QT NAMES Qt6 Qt5 REQUIRED COMPONENTS Widgets Network)
-	find_package(Qt${QT_VERSION_MAJOR} REQUIRED COMPONENTS Widgets Network)
+	find_package(Qt6 REQUIRED COMPONENTS Widgets Network)
+	find_package(Qt6 REQUIRED COMPONENTS Widgets Network)
 
 	if(ENABLE_TRANSLATIONS)
-		find_package(Qt${QT_VERSION_MAJOR} REQUIRED COMPONENTS LinguistTools)
+		find_package(Qt6 REQUIRED COMPONENTS LinguistTools)
 		add_definitions(-DENABLE_QT_TRANSLATIONS)
 	endif()
 endif()
@@ -649,7 +649,7 @@
 	endif()
 
 	if(ENABLE_LAUNCHER OR ENABLE_EDITOR)
-		get_target_property(QtCore_location Qt${QT_VERSION_MAJOR}::Core LOCATION)
+		get_target_property(QtCore_location Qt6::Core LOCATION)
 		get_filename_component(Qtbin_folder ${QtCore_location} PATH)
 		file(GLOB dep_files
 			${dep_files}
