diff --git a/src/core/system.cpp b/src/core/system.cpp
index 5007caa..56c7291 100644
--- a/src/core/system.cpp
+++ b/src/core/system.cpp
@@ -2008,9 +2008,6 @@ bool System::CreateGPU(GPURenderer renderer, bool is_switching)
   {
     Log_ErrorPrintf("Failed to initialize %s renderer, falling back to software renderer",
                     Settings::GetRendererName(renderer));
-    Host::AddFormattedOSDMessage(
-      30.0f, TRANSLATE("OSDMessage", "Failed to initialize %s renderer, falling back to software renderer."),
-      Settings::GetRendererName(renderer));
     g_gpu.reset();
     g_gpu = GPU::CreateSoftwareRenderer();
     if (!g_gpu)
diff --git a/src/util/opengl_device.cpp b/src/util/opengl_device.cpp
index 0c55094..e4b0763 100644
--- a/src/util/opengl_device.cpp
+++ b/src/util/opengl_device.cpp
@@ -325,9 +325,6 @@ bool OpenGLDevice::CreateDevice(const std::string_view& adapter, bool threaded_p
      (m_gl_context->IsGLES() && GLAD_GL_ES_VERSION_3_1));
   if (!opengl_is_available)
   {
-    Host::ReportErrorAsync(TRANSLATE_SV("GPUDevice", "Error"),
-                           TRANSLATE_SV("GPUDevice", "OpenGL renderer unavailable, your driver or hardware is not "
-                                                     "recent enough. OpenGL 3.1 or OpenGL ES 3.1 is required."));
     m_gl_context.reset();
     return false;
   }
