diff --git a/Makefile b/Makefile
index 5b99d34..3c9b42e 100644
--- a/Makefile
+++ b/Makefile
@@ -82,7 +82,11 @@ ifneq (,$(findstring unix,$(platform)))
       else ifneq (,$(findstring rpi2,$(platform)))
          FLAGS += -DARM -marm -mcpu=cortex-a7 -mfpu=neon-vfpv4 -mfloat-abi=hard
       else ifneq (,$(findstring rpi3,$(platform)))
-         FLAGS += -DARM -marm -mcpu=cortex-a53 -mfpu=neon-fp-armv8 -mfloat-abi=hard
+         ifneq (,$(findstring arm64,$(platform)))
+            FLAGS += -DARM -march=armv8-a+crc+simd -mcpu=cortex-a53
+         else
+            FLAGS += -DARM -marm -mcpu=cortex-a53 -mfpu=neon-fp-armv8 -mfloat-abi=hard
+         endif
       endif
    endif
    
