--- a/Makefile	2020-10-23 20:05:45.443191089 +0200
+++ b/Makefile	2020-10-23 20:06:06.451511089 +0200
@@ -134,6 +134,13 @@
     AARCH64 = 1
     endif
 
+# Amlogic S905D3/S905X3/S905Y3 (AMLSM1) e.g. HardKernel ODroid C4 & Khadas VIM3L (SDL2 64-bit)
+else ifneq (,$(findstring AMLSM1,$(PLATFORM)))
+    CPUFLAGS += -mcpu=cortex-a55
+    CPPFLAGS += -DCPU_AARCH64 -D_FILE_OFFSET_BITS=64
+    AARCH64 = 1
+
+
 # Odroid Go Advance target (SDL2, 64-bit)
 else ifeq ($(PLATFORM),go-advance)
     CPUFLAGS += -mcpu=cortex-a35
