diff --git a/CMakeLists.txt b/CMakeLists.txt
index 4c5d57073b..86191f9ba6 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -89,10 +89,7 @@ if( FORCE_CROSSCOMPILE )
 	set( CMAKE_CROSSCOMPILING TRUE )
 endif()
 
-if(CMAKE_CROSSCOMPILING)
-	set(IMPORT_EXECUTABLES "IMPORTFILE-NOTFOUND" CACHE FILEPATH "Export file from native build.")
-	include(${IMPORT_EXECUTABLES})
-endif()
+set( CMAKE_CROSSCOMPILING FALSE )
 
 # Replacement variables for a possible long list of C/C++ compilers compatible with GCC
 if( "${CMAKE_C_COMPILER_ID}" STREQUAL "GNU" OR "${CMAKE_C_COMPILER_ID}" STREQUAL "Clang" )
