diff --git a/makefile b/makefile
index 46c78d04dc0..d95e3e6c3b0 100644
--- a/makefile
+++ b/makefile
@@ -383,14 +383,14 @@ ifndef FORCE_DRC_C_BACKEND
 endif
 endif
 
-ifeq ($(findstring arm,$(UNAME)),arm)
+ifeq ($(findstring arm,$(PLATFORM)),arm)
 ARCHITECTURE :=
 ifndef FORCE_DRC_C_BACKEND
 	FORCE_DRC_C_BACKEND := 1
 endif
 endif
 
-ifeq ($(findstring aarch64,$(UNAME)),aarch64)
+ifeq ($(findstring aarch64,$(PLATFORM)),aarch64)
 ARCHITECTURE :=
 ifndef FORCE_DRC_C_BACKEND
 	FORCE_DRC_C_BACKEND := 1
@@ -406,7 +406,7 @@
 endif
 endif
 
-ifeq ($(findstring riscv64,$(UNAME)),riscv64)
+ifeq ($(findstring riscv64,$(PLATFORM)),riscv64)
 ARCHITECTURE :=
 ifndef FORCE_DRC_C_BACKEND
 	FORCE_DRC_C_BACKEND := 1
