diff --git a/3rdparty/bgfx/src/renderer_gl.h b/3rdparty/bgfx/src/renderer_gl.h
index d52efc76..9f5b2022 100644
--- a/3rdparty/bgfx/src/renderer_gl.h
+++ b/3rdparty/bgfx/src/renderer_gl.h
@@ -13,10 +13,12 @@
 	|| BX_PLATFORM_NX                                     \
 	|| BX_PLATFORM_RPI                                    \
 	|| BX_PLATFORM_WINDOWS                                \
+	|| BX_PLATFORM_LINUX_NO_GL							  \
 	) )
 
 #define BGFX_USE_HTML5 (BGFX_CONFIG_RENDERER_OPENGLES && (0 \
 	|| BX_PLATFORM_EMSCRIPTEN                               \
+	|| BX_PLATFORM_LINUX_NO_GL 								\
 	) )
 
 #define BGFX_USE_WGL (BGFX_CONFIG_RENDERER_OPENGL && (0 \
diff --git a/3rdparty/bx/include/bx/platform.h b/3rdparty/bx/include/bx/platform.h
index eea5fcb0..f57a3a79 100644
--- a/3rdparty/bx/include/bx/platform.h
+++ b/3rdparty/bx/include/bx/platform.h
@@ -60,6 +60,7 @@
 #define BX_PLATFORM_WINDOWS    0
 #define BX_PLATFORM_WINRT      0
 #define BX_PLATFORM_XBOXONE    0
+#define BX_PLATFORM_LINUX_NO_GL 0
 
 // http://sourceforge.net/apps/mediawiki/predef/index.php?title=Compilers
 #if defined(__clang__)
@@ -221,6 +222,13 @@
 #	define BX_PLATFORM_HAIKU 1
 #endif //
 
+#if BX_CPU_ARM
+#   undef  BX_PLATFORM_LINUX
+#   define BX_PLATFORM_LINUX 0
+#   undef  BX_PLATFORM_LINUX_NO_GL
+#   define BX_PLATFORM_LINUX_NO_GL 1
+#endif
+
 #if !BX_CRT_NONE
 // https://sourceforge.net/p/predef/wiki/Libraries/
 #	if defined(__BIONIC__)
@@ -269,6 +277,7 @@
 	||  BX_PLATFORM_OSX        \
 	||  BX_PLATFORM_PS4        \
 	||  BX_PLATFORM_RPI        \
+	||  BX_PLATFORM_LINUX_NO_GL \
 	)
 
 ///
@@ -287,6 +296,7 @@
 	||  BX_PLATFORM_WINDOWS    \
 	||  BX_PLATFORM_WINRT      \
 	||  BX_PLATFORM_XBOXONE    \
+	||  BX_PLATFORM_LINUX_NO_GL \
 	)
 
 ///
@@ -388,6 +398,8 @@
 #	define BX_PLATFORM_NAME "WinRT"
 #elif BX_PLATFORM_XBOXONE
 #	define BX_PLATFORM_NAME "Xbox One"
+#elif BX_PLATFORM_LINUX_NO_GL
+#	define BX_PLATFORM_NAME "Linux (GLES)"
 #else
 #	error "Unknown BX_PLATFORM!"
 #endif // BX_PLATFORM_
diff --git a/3rdparty/bx/src/mutex.cpp b/3rdparty/bx/src/mutex.cpp
index b3e9677e..f9d9ed95 100644
--- a/3rdparty/bx/src/mutex.cpp
+++ b/3rdparty/bx/src/mutex.cpp
@@ -18,7 +18,8 @@
 	|| BX_PLATFORM_OSX     \
 	|| BX_PLATFORM_PS4     \
 	|| BX_PLATFORM_RPI	   \
-	|| BX_PLATFORM_NX
+	|| BX_PLATFORM_NX      \
+	|| BX_PLATFORM_LINUX_NO_GL
 #	include <pthread.h>
 #elif  BX_PLATFORM_WINDOWS \
 	|| BX_PLATFORM_WINRT   \
diff --git a/3rdparty/bx/src/os.cpp b/3rdparty/bx/src/os.cpp
index 12586aa8..ae2836ad 100644
--- a/3rdparty/bx/src/os.cpp
+++ b/3rdparty/bx/src/os.cpp
@@ -27,7 +27,8 @@
 	|| BX_PLATFORM_NX         \
 	|| BX_PLATFORM_OSX        \
 	|| BX_PLATFORM_PS4        \
-	|| BX_PLATFORM_RPI
+	|| BX_PLATFORM_RPI		  \
+	|| BX_PLATFORM_LINUX_NO_GL
 #	include <sched.h> // sched_yield
 #	if BX_PLATFORM_BSD       \
 	|| BX_PLATFORM_HAIKU     \
@@ -45,7 +46,8 @@
 #	if BX_PLATFORM_ANDROID
 #		include <malloc.h> // mallinfo
 #	elif   BX_PLATFORM_LINUX     \
-		|| BX_PLATFORM_RPI
+		|| BX_PLATFORM_RPI       \
+		|| BX_PLATFORM_LINUX_NO_GL
 #		include <stdio.h>  // fopen
 #		include <unistd.h> // syscall
 #		include <sys/syscall.h>
@@ -98,7 +100,8 @@ namespace bx
 #if BX_PLATFORM_WINDOWS
 		return ::GetCurrentThreadId();
 #elif  BX_PLATFORM_LINUX \
-	|| BX_PLATFORM_RPI
+	|| BX_PLATFORM_RPI	 \
+	|| BX_PLATFORM_LINUX_NO_GL
 		return (pid_t)::syscall(SYS_gettid);
 #elif  BX_PLATFORM_IOS \
 	|| BX_PLATFORM_OSX
@@ -119,7 +122,8 @@ namespace bx
 		struct mallinfo mi = mallinfo();
 		return mi.uordblks;
 #elif  BX_PLATFORM_LINUX \
-	|| BX_PLATFORM_HURD
+	|| BX_PLATFORM_HURD	 \
+	|| BX_PLATFORM_LINUX_NO_GL
 		FILE* file = fopen("/proc/self/statm", "r");
 		if (NULL == file)
 		{
@@ -312,7 +316,8 @@ namespace bx
 	void* exec(const char* const* _argv)
 	{
 #if BX_PLATFORM_LINUX \
- || BX_PLATFORM_HURD
+ || BX_PLATFORM_HURD  \
+ || BX_PLATFORM_LINUX_NO_GL
 		pid_t pid = fork();
 
 		if (0 == pid)
diff --git a/3rdparty/bx/src/thread.cpp b/3rdparty/bx/src/thread.cpp
index 17ace092..75a51090 100644
--- a/3rdparty/bx/src/thread.cpp
+++ b/3rdparty/bx/src/thread.cpp
@@ -22,12 +22,14 @@
 	|| BX_PLATFORM_OSX     \
 	|| BX_PLATFORM_PS4     \
 	|| BX_PLATFORM_RPI     \
-	|| BX_PLATFORM_NX
+	|| BX_PLATFORM_NX      \
+	| BX_PLATFORM_LINUX_NO_GL
 #	include <pthread.h>
 #	if BX_PLATFORM_BSD
 #		include <pthread_np.h>
 #	endif // BX_PLATFORM_BSD
-#	if BX_PLATFORM_LINUX && (BX_CRT_GLIBC < 21200)
+#	if BX_PLATFORM_LINUX \
+        || BX_PLATFORM_LINUX_NO_GL && (BX_CRT_GLIBC < 21200)
 #		include <sys/prctl.h>
 #	endif // BX_PLATFORM_
 #elif  BX_PLATFORM_WINDOWS \
@@ -249,7 +251,8 @@ namespace bx
 		pthread_setname_np(_name);
 #elif (BX_CRT_GLIBC >= 21200) && ! BX_PLATFORM_HURD
 		pthread_setname_np(ti->m_handle, _name);
-#elif BX_PLATFORM_LINUX
+#elif BX_PLATFORM_LINUX \
+	|| BX_PLATFORM_LINUX_NO_GL
 		prctl(PR_SET_NAME,_name, 0, 0, 0);
 #elif BX_PLATFORM_BSD
 #	if defined(__NetBSD__)
