--- a/src/mame/layout/vgmplay.lay	2021-12-29 10:28:34.000000000 -0500
+++ b/src/mame/layout/vgmplay.lay	2022-03-20 19:47:09.153199980 -0400
@@ -105,23 +105,23 @@
 	</element>
 	<element name="text_l1">
 		<rect><color red="0.2" green="0.21" blue="0.23" /></rect>
-		<text string="&#x25A0;"><color red="0.9" green="0.9" blue="0.2" /></text>
+		<text string="[]"><color red="0.9" green="0.9" blue="0.2" /></text>
 	</element>
 	<element name="text_l2">
 		<rect><color red="0.2" green="0.21" blue="0.23" /></rect>
-		<text string="&#x25EB;"><color red="0.9" green="0.9" blue="0.2" /></text>
+		<text string="II"><color red="0.9" green="0.9" blue="0.2" /></text>
 	</element>
 	<element name="text_l3">
 		<rect><color red="0.2" green="0.21" blue="0.23" /></rect>
-		<text string="&#x25B6;"><color red="0.9" green="0.9" blue="0.2" /></text>
+		<text string=">"><color red="0.9" green="0.9" blue="0.2" /></text>
 	</element>
 	<element name="text_l4">
 		<rect><color red="0.2" green="0.21" blue="0.23" /></rect>
-		<text string="&#x25C4;"><color red="0.9" green="0.9" blue="0.2" /></text>
+		<text string=">>"><color red="0.9" green="0.9" blue="0.2" /></text>
 	</element>
 	<element name="text_l5">
 		<rect><color red="0.2" green="0.21" blue="0.23" /></rect>
-		<text string="&#x25CE;"><color red="0.9" green="0.9" blue="0.2" /></text>
+		<text string="R"><color red="0.9" green="0.9" blue="0.2" /></text>
 	</element>
 
 	<group name="activity">
