diff -Naur eduke32-9860bbe8c4a5a867e63571e1377f78db5f224d08.orig/Common.mak eduke32-9860bbe8c4a5a867e63571e1377f78db5f224d08/Common.mak
--- eduke32-9860bbe8c4a5a867e63571e1377f78db5f224d08.orig/Common.mak	2021-08-27 04:30:07.000000000 +0100
+++ eduke32-9860bbe8c4a5a867e63571e1377f78db5f224d08/Common.mak	2021-09-01 14:46:08.249088820 +0100
@@ -575,8 +575,8 @@
         else
             ifeq ($(PLATFORM),DARWIN)
                 OPTOPT := -march=core2 -mmmx -msse -msse2 -msse3 -mssse3
-            else
-                OPTOPT := -march=nocona
+#            else
+#                OPTOPT := -march=nocona
             endif
         endif
     endif
@@ -603,6 +603,9 @@
     ifeq ($(PLATFORM),WII)
         OPTOPT := -mtune=750
     endif
+    ifneq (0,$(RPI4))
+        OPTOPT := -mcpu=cortex-a72 -mtune=cortex-a72 -ffast-math
+    endif    
 endif
 
 ifeq ($(PACKAGE_REPOSITORY),0)
@@ -897,7 +900,7 @@
 
 ifeq ($(RENDERTYPE),SDL)
     ifeq ($(SDL_TARGET),2)
-        SDLCONFIG := sdl2-config
+        SDLCONFIG := sdl2
         SDLNAME := SDL2
     else ifeq ($(SDL_TARGET),1)
         SDLCONFIG := sdl-config
@@ -939,8 +942,8 @@
             SDLCONFIG_CFLAGS := $(strip $(subst -Dmain=SDL_main,,$(shell $(SDLCONFIG) --cflags)))
             SDLCONFIG_LIBS := $(strip $(subst -mwindows,,$(shell $(SDLCONFIG) --libs)))
 
-            COMPILERFLAGS += $(SDLCONFIG_CFLAGS)
-            LIBS += $(SDLCONFIG_LIBS)
+            COMPILERFLAGS += `pkg-config sdl2 --cflags`
+            LIBS += `pkg-config sdl2 --libs`
         else
             ifeq ($(SDL_TARGET),1)
                 COMPILERFLAGS += -D_GNU_SOURCE=1
diff -Naur eduke32-9860bbe8c4a5a867e63571e1377f78db5f224d08.orig/GNUmakefile eduke32-9860bbe8c4a5a867e63571e1377f78db5f224d08/GNUmakefile
--- eduke32-9860bbe8c4a5a867e63571e1377f78db5f224d08.orig/GNUmakefile	2021-08-27 04:30:07.000000000 +0100
+++ eduke32-9860bbe8c4a5a867e63571e1377f78db5f224d08/GNUmakefile	2021-09-01 14:30:18.343320946 +0100
@@ -1383,9 +1383,12 @@
 #### Targets
 
 all: \
+    duke3d \
     blood \
     rr \
+    sw \
     exhumed \
+	tekwar \
 
 start:
 	$(BUILD_STARTED)
