diff --git a/src/api/udev/JoystickUdev.cpp b/src/api/udev/JoystickUdev.cpp
index 2ef41a2..95697b4 100644
--- a/src/api/udev/JoystickUdev.cpp
+++ b/src/api/udev/JoystickUdev.cpp
@@ -217,10 +217,17 @@ bool CJoystickUdev::ScanEvents(void)
               const unsigned int axisIndex = it->second.axisIndex;
               const input_absinfo& info = it->second.axisInfo;
 
-              if (event.value >= 0)
-                SetAxisValue(axisIndex, event.value, info.maximum);
-              else
-                SetAxisValue(axisIndex, event.value, -info.minimum);
+	      int middle = (info.minimum+info.maximum)/2;
+	      int length = (info.maximum-info.minimum)/2;
+
+              if (event.value >= middle && event.value-middle > length/2) {
+                SetAxisValue(axisIndex, event.value-middle, length/2);
+	      }
+              else if (event.value <= middle && middle-event.value > length/2) {
+                SetAxisValue(axisIndex, -(middle-event.value), length/2);
+	      } else {
+                SetAxisValue(axisIndex, 0, length);
+	      }
             }
           }
           break;
