--- a/CMakeLists.txt	2023-09-24 16:25:48.183427774 +0200
+++ b/CMakeLists.txt	2023-09-24 16:26:01.563702737 +0200
@@ -1161,7 +1161,7 @@
 
 if(THEXTECH_BUILD_GL_ES_LEGACY)
     target_compile_definitions(thextech PRIVATE -DTHEXTECH_BUILD_GL_ES_LEGACY)
-    target_link_libraries(thextech PRIVATE EGL GLESv1_CM)
+    target_link_libraries(thextech PRIVATE EGL GLESv2)
 endif()
 
 if(THEXTECH_ENABLE_NEW_USER_PATHS)
