diff --git a/backends/platform/libretro/libco/sjlj.c b/backends/platform/libretro/libco/sjlj.c
index f074714..21ce614 100644
--- a/backends/platform/libretro/libco/sjlj.c
+++ b/backends/platform/libretro/libco/sjlj.c
@@ -59,7 +59,8 @@ cothread_t co_create(unsigned int size, void (*coentry)(void))
       stack_t stack;
       stack_t old_stack;
 
-      thread->coentry = thread->stack = 0;
+      thread->coentry = 0;
+      thread->stack = 0;
 
       stack.ss_flags = 0;
       stack.ss_size = size;
