diff --git a/es-app/src/ApiSystem.cpp b/es-app/src/ApiSystem.cpp
index 71bfcd3..ba26c41 100644
--- a/es-app/src/ApiSystem.cpp
+++ b/es-app/src/ApiSystem.cpp
@@ -904,7 +904,7 @@ bool ApiSystem::downloadGitRepository(const std::string& url, const std::string&
 			}
 		}
 
-		std::this_thread::sleep_for(std::chrono::milliseconds(20));
+		std::this_thread::sleep_for(std::chrono::milliseconds(1));
 	}
 
 	if (httpreq.status() != HttpReq::REQ_SUCCESS)
