diff --git a/es-app/src/services/HttpServerThread.cpp b/es-app/src/services/HttpServerThread.cpp
index cf56f97..41acbf0 100644
--- a/es-app/src/services/HttpServerThread.cpp
+++ b/es-app/src/services/HttpServerThread.cpp
@@ -31,6 +31,7 @@ Misc APIS
 -----------------
 GET  /caps                                                      -> capability info
 GET  /restart
+GET  /shutdown
 GET  /quit
 GET  /emukill
 GET  /reloadgames
@@ -248,6 +249,14 @@ void HttpServerThread::run()
 		Utils::Platform::quitES(Utils::Platform::QuitMode::REBOOT);
 	});
 
+	mHttpServer->Get("/shutdown", [](const httplib::Request& req, httplib::Response& res)
+	{
+		if (!isAllowed(req, res))
+			return;
+
+		Utils::Platform::quitES(Utils::Platform::QuitMode::SHUTDOWN);
+	});
+
 	mHttpServer->Get("/emukill", [](const httplib::Request& req, httplib::Response& res)
 	{
 		if (!isAllowed(req, res))
