diff --git a/m4/ul.m4 b/m4/ul.m4
index 9b3e73d..4d10051 100644
--- a/m4/ul.m4
+++ b/m4/ul.m4
@@ -455,42 +455,6 @@ AC_DEFUN([UL_NCURSES_CHECK], [
   m4_define([suffix], $1)
   m4_define([SUFFIX], m4_toupper($1))
 
-  # ncurses6-config
-  #
-  AS_IF([test "x$have_[]suffix" = xno], [
-    AC_CHECK_TOOL(SUFFIX[]6_CONFIG, suffix[]6-config)
-    if AC_RUN_LOG([$SUFFIX[]6_CONFIG --version >/dev/null]); then
-      have_[]suffix=yes
-      NCURSES_LIBS=`$SUFFIX[]6_CONFIG --libs`
-      NCURSES_CFLAGS=`$SUFFIX[]6_CONFIG --cflags`
-    else
-      have_[]suffix=no
-    fi
-  ])
-
-  # ncurses5-config
-  #
-  AS_IF([test "x$have_[]suffix" = xno], [
-    AC_CHECK_TOOL(SUFFIX[]5_CONFIG, suffix[]5-config)
-    if AC_RUN_LOG([$SUFFIX[]5_CONFIG --version >/dev/null]); then
-      have_[]suffix=yes
-      NCURSES_LIBS=`$SUFFIX[]5_CONFIG --libs`
-      NCURSES_CFLAGS=`$SUFFIX[]5_CONFIG --cflags`
-    else
-      have_[]suffix=no
-    fi
-  ])
-
-  # pkg-config (not supported by ncurses upstream by default)
-  #
-  AS_IF([test "x$have_[]suffix" = xno], [
-    PKG_CHECK_MODULES(SUFFIX, [$1], [
-      have_[]suffix=yes
-      NCURSES_LIBS=${SUFFIX[]_LIBS}
-      NCURSES_CFLAGS=${SUFFIX[]_CFLAGS}
-    ],[have_[]suffix=no])
-  ])
-
   # classic autoconf way
   #
   AS_IF([test "x$have_[]suffix" = xno], [
