diff --git a/libretro/main.cpp b/libretro/main.cpp
index d04d4effd7..c58e14d48b 100644
--- a/libretro/main.cpp
+++ b/libretro/main.cpp
@@ -2348,7 +2348,7 @@ void retro_init(void)
 		FileSystem::FindResultsArray results;
 		char bios_dir[PCSX2_PATH_MAX];
 
-		pcsx2_path_join(bios_dir, sizeof(bios_dir), system_base, "pcsx2/bios");
+		pcsx2_path_join(bios_dir, sizeof(bios_dir), system_base, "ps2");
 		if (FileSystem::FindFiles(bios_dir, "*", FILESYSTEM_FIND_FILES, &results))
 		{
 			u32 version, region;
diff --git a/pcsx2/Pcsx2Config.cpp b/pcsx2/Pcsx2Config.cpp
index a5e87e6e6d..e25b15203d 100644
--- a/pcsx2/Pcsx2Config.cpp
+++ b/pcsx2/Pcsx2Config.cpp
@@ -1029,7 +1029,7 @@ void Pcsx2Config::CopyRuntimeConfig(Pcsx2Config& cfg)
 
 void EmuFolders::SetDefaults(SettingsInterface& si)
 {
-	si.SetStringValue("Folders", "Bios", "bios");
+	si.SetStringValue("Folders", "Bios", "/userdata/bios/ps2");
 	si.SetStringValue("Folders", "MemoryCards", "memcards");
 	si.SetStringValue("Folders", "Cheats", "cheats");
 	si.SetStringValue("Folders", "CheatsWS", "cheats_ws");
