diff --git a/cmake/BuildParameters.cmake b/cmake/BuildParameters.cmake
index a653adf..d26cfb8 100644
--- a/cmake/BuildParameters.cmake
+++ b/cmake/BuildParameters.cmake
@@ -101,7 +101,7 @@ if(${PCSX2_TARGET_ARCHITECTURES} MATCHES "x86_64")
 			set(ARCH_FLAG "-msse -msse2 -msse4.1 -mfxsr")
 		else()
 			# Can't use march=native on Apple Silicon.
-			if(NOT APPLE OR "${CMAKE_HOST_SYSTEM_PROCESSOR}" STREQUAL "x86_64")
+			if("${CMAKE_HOST_SYSTEM_PROCESSOR}" STREQUAL "x86_64")
 				set(ARCH_FLAG "-march=native")
 			endif()
 		endif()
