--- a/libretro/CMakeLists.txt
+++ b/libretro/CMakeLists.txt
@@ -16,6 +16,9 @@
    ${CMAKE_SOURCE_DIR}/libretro/libretro-common/streams/file_stream.c
    ${CMAKE_SOURCE_DIR}/libretro/libretro-common/time/rtime.c
    ${CMAKE_SOURCE_DIR}/libretro/libretro-common/rthreads/rthreads.c
+   ${CMAKE_SOURCE_DIR}/libretro/libretro-common/rthreads/retro_eventcount.c
+   ${CMAKE_SOURCE_DIR}/libretro/libretro-common/rthreads/retro_asym_eventcount.c
+   ${CMAKE_SOURCE_DIR}/libretro/libretro-common/rthreads/retro_procbarrier.c
    ${CMAKE_SOURCE_DIR}/libretro/libretro-common/queues/retro_spsc.c
    ${CMAKE_SOURCE_DIR}/libretro/libretro-common/vfs/vfs_implementation.c
    ${CMAKE_SOURCE_DIR}/libretro/libretro-common/vfs/vfs_hybrid.c
