diff --git a/Makefile b/Makefile
index 8b0f9a6..8d6a559 100644
--- a/Makefile
+++ b/Makefile
@@ -86,11 +86,11 @@ YASM 	        = yasm
 CC  	        = $(LNXDEV)/$(PREFIX)gcc
 OBJTYPE         = elf
 INCLUDES        = $(SDKPATH)/include
-ifeq ($(findstring 64, $(GCC_TARGET)), 64)
+ifeq ($(findstring 64, $(BUILD_PLATFORM)), 64)
 TARGET_ARCH     = amd64
 BUILD_AMD64     = 1
 ARCHFLAGS       = -m64
-LIBRARIES       = $(SDKPATH)/lib/$(GCC_TARGET)
+#LIBRARIES       = $(SDKPATH)/lib/$(GCC_TARGET)
 CFLAGS          += -DAMD64
 else
 TARGET_ARCH     = x86
@@ -194,9 +194,6 @@ ifndef NO_STRIP
 ifdef BUILD_WIN
 STRIP 	        = $(WINDEV)/$(PREFIX)strip$(EXTENSION) $(TARGET) -o $(TARGET_FINAL)
 endif
-ifdef BUILD_LINUX
-STRIP 	        = $(LNXDEV)/$(PREFIX)strip $(TARGET) -o $(TARGET_FINAL)
-endif
 ifdef BUILD_DARWIN
 STRIP           = $(PREFIX)strip $(TARGET) -o $(TARGET_FINAL)
 endif
