--- a/Makefile	2015-02-04 19:40:51.949440875 +0100
+++ b/Makefile	2015-02-04 19:45:41.701936840 +0100
@@ -35,8 +35,8 @@
 endif
 
 TARGET_NAME := catsfc
-DEFS        :=
-CFLAGS      :=
+#DEFS        :=
+#CFLAGS      :=
 
 ifeq ($(platform), unix)
    TARGET := $(TARGET_NAME)_libretro.so
@@ -131,6 +131,24 @@
 	DEFS   +=  -DPSP -D_PSP_FW_VERSION=371
    INCLUDE     += -I$(shell psp-config --pspsdk-path)/include
    STATIC_LINKING := 1
+else ifneq (,$(findstring armv,$(platform)))
+       TARGET := $(TARGET_NAME)_libretro.so
+       fpic := -fPIC
+       SHARED := -shared -Wl,--no-undefined -Wl,--version-script=link.T
+       CC = gcc
+       CXX = g++
+       PLATFORM_DEFINES += -marm
+       ifneq (,$(findstring softfloat,$(platform)))
+               PLATFORM_DEFINES += -mfloat-abi=softfp
+       else ifneq (,$(findstring hardfloat,$(platform)))
+               PLATFORM_DEFINES += -mfloat-abi=hard
+       endif
+       ifneq (,$(findstring neon,$(platform)))
+               FLAGS += -mfpu=neon
+               ASFLAGS += -mfpu=neon
+               HAVE_NEON = 1
+       endif
+       PLATFORM_DEFINES += -DARM
 else
    TARGET := $(TARGET_NAME)_libretro.dll
    CC = gcc
@@ -143,7 +161,7 @@
           -DVAR_CYCLES -DCPU_SHUTDOWN -DSPC700_SHUTDOWN \
           -DNO_INLINE_SET_GET -DNOASM -DHAVE_MKSTEMP '-DACCEPT_SIZE_T=size_t' -DWANT_CHEATS
 
-DEFS  += -D__LIBRETRO__
+DEFS  += -D__LIBRETRO__ $(PLATFORM_DEFINES)
 
 CORE_DIR     := ./source
 LIBRETRO_DIR := .
