diff -Naur a/kernel/conftest.sh b/kernel/conftest.sh
--- a/kernel/conftest.sh
+++ b/kernel/conftest.sh
@@ -35,6 +35,9 @@ fi

 HEADERS_ARCH="$SOURCES/arch/$KERNEL_ARCH/include"

+# Create an empty file since the normal BUILD_MODULE_RULE is not working!
+touch .nv-kernel.o.cmd
+
 # VGX_BUILD parameter defined only for VGX builds

 test_xen() {
diff -Naur a/kernel/uvm/Makefile b/kernel/uvm/Makefile
index 043a08d..ec24244 100644
--- a/kernel/uvm/Makefile
+++ b/kernel/uvm/Makefile
@@ -206,7 +206,7 @@ EXTRA_CFLAGS += $(NV_DEFINES) $(NV_TARGET_DEFINES) $(NV_INCLUDES)
 ccflags-y += $(EXTRA_CFLAGS)

 RM_MODULE_SYMVERS:= $(RM_OUT_DIR)/Module.symvers
-UVM_MODULE_SYMVERS:= $(obj)/Module.symvers
+UVM_MODULE_SYMVERS:= $(obj)/../Module.symvers
 KBUILD_EXTRA_SYMBOLS:= $(UVM_MODULE_SYMVERS)

 module $(MODULE_NAME).ko: $(UVM_MODULE_SYMVERS) debug_diagnostics_printing
