diff --git a/src/Engine/EngineCore.cpp b/src/Engine/EngineCore.cpp
index 27f18256be..52dc114dec 100644
--- a/src/Engine/EngineCore.cpp
+++ b/src/Engine/EngineCore.cpp
@@ -103,7 +103,7 @@ EngineCore::EngineCore(IGame& game, const ISystem& system, PlayerInput& keyboard
     m_gameTimer.Reset();
 
     const fs::path filenamePath = m_system.GetConfigurationFilePath();
-    const fs::path savedGamesPath = filenamePath / m_game.GetSavedGamesPath();
+    const fs::path savedGamesPath = fs::path("/userdata/saves/CatacombGL") / m_game.GetSavedGamesPath();
     m_system.GetSavedGameNamesFromFolder(savedGamesPath, m_savedGames);
 
 #ifdef SAVEDGAMESINDOSFORMAT
@@ -3233,7 +3233,7 @@ bool EngineCore::StoreGameToFileWithFullPath(const fs::path filename) const
 bool EngineCore::StoreGameToFile(const std::string filename)
 {
     const fs::path filenamePath = m_system.GetConfigurationFilePath();
-    const fs::path filenamePathForGame = filenamePath / m_game.GetSavedGamesPath();
+    const fs::path filenamePathForGame = fs::path("/userdata/saves/CatacombGL") / m_game.GetSavedGamesPath();
     if (m_system.CreatePath(filenamePathForGame))
     {
         const fs::path fullPath = filenamePathForGame / ( filename + ".sav" );
@@ -3324,7 +3324,7 @@ void EngineCore::LoadGameFromFileWithFullPath(const fs::path filename)
 void EngineCore::LoadGameFromFile(const std::string filename)
 {
     const fs::path filenamePath = m_system.GetConfigurationFilePath();
-    const fs::path filenamePathAbyss = filenamePath / m_game.GetSavedGamesPath();
+    const fs::path filenamePathAbyss = fs::path("/userdata/saves/CatacombGL") / m_game.GetSavedGamesPath();
     const fs::path fullPath = filenamePathAbyss / ( filename + ".sav" );
     LoadGameFromFileWithFullPath(fullPath);
 }
