diff --git a/fceumm-code/src/drivers/libretro/libretro.c b/fceumm-code/src/drivers/libretro/libretro.c
index 8504acb..01a24e7 100644
--- a/fceumm-code/src/drivers/libretro/libretro.c
+++ b/fceumm-code/src/drivers/libretro/libretro.c
@@ -534,6 +534,8 @@ typedef struct
 static const keymap bindmap[] = {
    { RETRO_DEVICE_ID_JOYPAD_A, JOY_A },
    { RETRO_DEVICE_ID_JOYPAD_B, JOY_B },
+   { RETRO_DEVICE_ID_JOYPAD_X, JOY_B },
+   { RETRO_DEVICE_ID_JOYPAD_Y, JOY_A },
    { RETRO_DEVICE_ID_JOYPAD_SELECT, JOY_SELECT },
    { RETRO_DEVICE_ID_JOYPAD_START, JOY_START },
    { RETRO_DEVICE_ID_JOYPAD_UP, JOY_UP },
@@ -552,10 +554,10 @@ static void update_input(void)
 
    poll_cb();
 
-   for ( i = 0; i < 8; i++)
+   for ( i = 0; i < 10; i++)
       pad[0] |= input_cb(0, RETRO_DEVICE_JOYPAD, 0, bindmap[i].retro) ? bindmap[i].nes : 0;
 
-   for ( i = 0; i < 8; i++)
+   for ( i = 0; i < 10; i++)
       pad[1] |= input_cb(1, RETRO_DEVICE_JOYPAD, 0, bindmap[i].retro) ? bindmap[i].nes : 0;
 
    JSReturn[0] = pad[0] | (pad[1] << 8);
