diff --git a/board/rockchip/odroidgoa/odroidgoa.c b/board/rockchip/odroidgoa/odroidgoa.c
index 387125f..ac11983 100644
--- a/board/rockchip/odroidgoa/odroidgoa.c
+++ b/board/rockchip/odroidgoa/odroidgoa.c
@@ -196,8 +196,8 @@ int rk_board_late_init(void)
 	/* show boot logo and version */
 	lcd_show_logo();
 	lcd_setfg_color("grey");
-	lcd_printf(0, 18 + disp_offs, 1, " %s", U_BOOT_VERSION);
-	lcd_printf(0, 19 + disp_offs, 1, " %s %s", U_BOOT_DATE, U_BOOT_TIME);
+	lcd_printf(0, 18 + disp_offs, 1, " %s", "batocera.linux");
+	lcd_printf(0, 19 + disp_offs, 1, " %s", "only cores that matter");
 
 	if (!board_check_autotest()) {
 		board_run_autotest();
