diff --git a/hid-xpadneo/src/hid-xpadneo.c b/hid-xpadneo/src/hid-xpadneo.c
index 0417c2c996..7f0206fa2e 100644
--- a/hid-xpadneo/src/hid-xpadneo.c
+++ b/hid-xpadneo/src/hid-xpadneo.c
@@ -919,62 +919,6 @@ static int xpadneo_event(struct hid_device *hdev, struct hid_field *field,
 			xdata->last_abs_rz = value;
 			goto combine_z_axes;
 		}
-	} else if ((usage->type == EV_KEY) && (usage->code == BTN_XBOX)) {
-		/*
-		 * Handle the Xbox logo button: We want to cache the button
-		 * down event to allow for profile switching. The button will
-		 * act as a shift key and only send the input events when
-		 * released without pressing an additional button.
-		 */
-		if (!xdata->xbox_button_down && (value == 1)) {
-			/* cache this event */
-			xdata->xbox_button_down = true;
-		} else if (xdata->xbox_button_down && (value == 0)) {
-			xdata->xbox_button_down = false;
-			if (xdata->profile_switched) {
-				xdata->profile_switched = false;
-			} else {
-				/* replay cached event */
-				input_report_key(gamepad, BTN_XBOX, 1);
-				input_sync(gamepad);
-				/* synthesize the release to remove the scan code */
-				input_report_key(gamepad, BTN_XBOX, 0);
-				input_sync(gamepad);
-			}
-		}
-		goto stop_processing;
-	} else if ((usage->type == EV_KEY) && (usage->code == BTN_SHARE)) {
-		/* move the Share button to the keyboard device */
-		if (!keyboard)
-			goto keyboard_missing;
-		input_report_key(keyboard, BTN_SHARE, value);
-		input_sync(keyboard);
-		goto stop_processing;
-	} else if (xdata->xbox_button_down && (usage->type == EV_KEY)) {
-		if (!(xdata->quirks & XPADNEO_QUIRK_USE_HW_PROFILES)) {
-			switch (usage->code) {
-			case BTN_A:
-				if (value == 1)
-					xpadneo_switch_profile(xdata, 0, true);
-				goto stop_processing;
-			case BTN_B:
-				if (value == 1)
-					xpadneo_switch_profile(xdata, 1, true);
-				goto stop_processing;
-			case BTN_X:
-				if (value == 1)
-					xpadneo_switch_profile(xdata, 2, true);
-				goto stop_processing;
-			case BTN_Y:
-				if (value == 1)
-					xpadneo_switch_profile(xdata, 3, true);
-				goto stop_processing;
-			case BTN_SELECT:
-				if (value == 1)
-					xpadneo_toggle_mouse(xdata);
-				goto stop_processing;
-			}
-		}
 	}
 
 	/* Let hid-core handle the event */
