--- a/dlls/appwiz.cpl/appwiz.c	2021-01-07 10:27:57.313421164 +0100
+++ b/dlls/appwiz.cpl/appwiz.c	2021-01-07 10:28:09.260087386 +0100
@@ -932,12 +932,12 @@
         return FALSE;
 
     if(!wcscmp(params, L"install_gecko")) {
-        install_addon(ADDON_GECKO);
+        /* install_addon(ADDON_GECKO); */
         return TRUE;
     }
 
     if(!wcscmp(params, L"install_mono")) {
-        install_addon(ADDON_MONO);
+        /* install_addon(ADDON_MONO); */
         return TRUE;
     }
 
