diff --git a/driver/product/kernel/drivers/gpu/arm/midgard/mali_kbase_smc.c b/driver/product/kernel/drivers/gpu/arm/midgard/mali_kbase_smc.c
index 2176479..d0b9855 100644
--- a/driver/product/kernel/drivers/gpu/arm/midgard/mali_kbase_smc.c
+++ b/driver/product/kernel/drivers/gpu/arm/midgard/mali_kbase_smc.c
@@ -27,6 +27,10 @@
 
 #include <linux/compiler.h>
 
+#ifndef __asmeq
+#define __asmeq(x, y)  ".ifnc " x "," y " ; .err ; .endif\n\t"
+#endif
+
 static noinline u64 invoke_smc_fid(u64 function_id,
 		u64 arg0, u64 arg1, u64 arg2)
 {
