--- a/libswirl/utils/glinit/sdl/sdl.cpp	2020-08-13 20:12:25.672842767 +0200
+++ b/libswirl/utils/glinit/sdl/sdl.cpp	2020-08-13 20:12:36.753065401 +0200
@@ -39,7 +39,6 @@
 	SDL_GL_SetAttribute(SDL_GL_RED_SIZE, 8);
 	SDL_GL_SetAttribute(SDL_GL_GREEN_SIZE, 8);
 	SDL_GL_SetAttribute(SDL_GL_BLUE_SIZE, 8);
-	SDL_GL_SetAttribute(SDL_GL_ALPHA_SIZE, 8);
 	SDL_GL_SetAttribute(SDL_GL_DEPTH_SIZE, 24);
 	SDL_GL_SetAttribute(SDL_GL_STENCIL_SIZE, 8);
 	SDL_GL_SetAttribute(SDL_GL_DOUBLEBUFFER, 1);
