diff --git a/libretro/retro_events.c b/libretro/retro_events.c
index 670db3d..b20bea9 100644
--- a/libretro/retro_events.c
+++ b/libretro/retro_events.c
@@ -254,6 +254,15 @@ static void ev_events_joy()
          return;
       }
    }
+
+   if(input_state_cb(0, RETRO_DEVICE_JOYPAD, 0, RETRO_DEVICE_ID_JOYPAD_Y)){
+      if(!event_last){
+         event_last = do_action(&events_combo[4].action);
+         return;
+      }else{
+         release_emulated_key(RETRO_DEVICE_ID_JOYPAD_Y);
+      }
+   } 
    event_last = EVENT_NULL;
 }
 
@@ -300,7 +309,11 @@ void ev_joysticks() {
    if( input_state_cb(0, RETRO_DEVICE_JOYPAD, 0, RETRO_DEVICE_ID_JOYPAD_SELECT)) {
       ev_events_joy();
    } else {
-      ev_process_joy(ID_PLAYER1);
+      if( input_state_cb(0, RETRO_DEVICE_JOYPAD, 0, RETRO_DEVICE_ID_JOYPAD_Y)) {
+         ev_events_joy();
+      }else{
+         ev_process_joy(ID_PLAYER1);
+      }
    }
 
    ev_process_joy(ID_PLAYER2);
@@ -313,8 +326,9 @@ void ev_joysticks() {
  **/
 void ev_joy_vkeyboard(){
    uint32_t pressed = 0;
-   if ( (input_state_cb(0, RETRO_DEVICE_JOYPAD, 0, RETRO_DEVICE_ID_JOYPAD_SELECT))
-        && (input_state_cb(0, RETRO_DEVICE_JOYPAD, 0, RETRO_DEVICE_ID_JOYPAD_START)) ) {
+   if ( (input_state_cb(0, RETRO_DEVICE_JOYPAD, 0, RETRO_DEVICE_ID_JOYPAD_SELECT) 
+      && input_state_cb(0, RETRO_DEVICE_JOYPAD, 0, RETRO_DEVICE_ID_JOYPAD_START)) 
+      || (input_state_cb(0, RETRO_DEVICE_JOYPAD, 0, RETRO_DEVICE_ID_JOYPAD_Y)) ) {
       BIT_SET(pressed, 3);
       if(pressed != combo_last_event) {
          showkeyb=-showkeyb;
