From 5c6ff5cddc0d62ccab1a27ab7a9779640229da6f Mon Sep 17 00:00:00 2001
From: Ntemis <ierokipides.dem@gmail.com>
Date: Thu, 28 May 2020 19:03:39 +0300
Subject: [PATCH] M&T: Change the maximum frequency to 600Mhz for GPU

---
 arch/arm/boot/dts/rk3288.dtsi | 4 ----
 1 file changed, 4 deletions(-)

diff --git a/arch/arm/boot/dts/rk3288.dtsi b/arch/arm/boot/dts/rk3288.dtsi
index 77350494c..b96ba8261 100644
--- a/arch/arm/boot/dts/rk3288.dtsi
+++ b/arch/arm/boot/dts/rk3288.dtsi
@@ -1876,10 +1876,6 @@
 			opp-hz = /bits/ 64 <420000000>;
 			opp-microvolt = <1100000>;
 		};
-		opp-500000000 {
-			opp-hz = /bits/ 64 <500000000>;
-			opp-microvolt = <1200000>;
-		};
 		opp-600000000 {
 			opp-hz = /bits/ 64 <600000000>;
 			opp-microvolt = <1250000>;
-- 
2.17.1

