diff --git a/kernel/nvidia-drm/nvidia-drm-drv.c b/kernel/nvidia-drm/nvidia-drm-drv.c
index 070084cbea..ca4b3e658c 100644
--- a/kernel/nvidia-drm/nvidia-drm-drv.c
+++ b/kernel/nvidia-drm/nvidia-drm-drv.c
@@ -706,9 +706,9 @@ static struct drm_driver nv_drm_driver = {
 
     .ioctls                 = nv_drm_ioctls,
     .num_ioctls             = ARRAY_SIZE(nv_drm_ioctls),
-
+#if LINUX_VERSION_CODE < KERNEL_VERSION(6, 6, 0)
     .prime_handle_to_fd     = drm_gem_prime_handle_to_fd,
-
+#endif
 #if defined(NV_DRM_DRIVER_HAS_GEM_PRIME_CALLBACKS)
     .gem_prime_export       = nv_drm_gem_prime_export,
     .gem_prime_get_sg_table = nv_drm_gem_prime_get_sg_table,
