diff --git a/meson.build b/meson.build
index bb0826f..bf68306 100644
--- a/meson.build
+++ b/meson.build
@@ -564,7 +564,7 @@ if x11_enabled
   xfixes_dep     = dependency('xfixes', required: false)
   xcomposite_dep = dependency('xcomposite', required: false)
   fontconfig_dep = dependency('fontconfig', fallback: ['fontconfig', 'fontconfig_dep'])
-  atkbridge_dep  = dependency('atk-bridge-2.0', version: at_spi2_atk_req, required: false)
+  #atkbridge_dep  = dependency('atk-bridge-2.0', version: at_spi2_atk_req, required: false)
 
   backend_immodules += ['xim']
 
@@ -583,7 +583,7 @@ if x11_enabled
     x11_pkgs += ['xdamage']
   endif
 
-  atk_pkgs += ['atk-bridge-2.0']
+  #atk_pkgs += ['atk-bridge-2.0']
 
   cdata.set('HAVE_XDAMAGE', xdamage_dep.found() ? 1 : false)
   cdata.set('HAVE_XCURSOR', xcursor_dep.found() ? 1 : false)
