--- a/FindTinyXML2.cmake	2017-05-06 04:32:10.000000000 -0300
+++ b/FindTinyXML2.cmake	2019-08-16 11:39:09.838465312 -0300
@@ -6,17 +6,20 @@
 #   TINYXML2_LIBRARIES    - List of libraries when using TinyXML.
 #
 
-find_package(PkgConfig)
-if(PKG_CONFIG_FOUND)
-  pkg_check_modules (TINYXML2 tinyxml2)
-  list(APPEND TINYXML2_INCLUDE_DIRS ${TINYXML2_INCLUDEDIR})
-endif()
+find_path( TINYXML2_INCLUDE_DIRS "tinyxml2.h"
+           PATH_SUFFIXES "tinyxml2" )
+
+find_library( TINYXML2_LIBRARIES
+              NAMES "tinyxml2"
+              PATH_SUFFIXES "tinyxml2" )
+
+# batocera: look for debug version
 if(NOT TINYXML2_FOUND)
   find_path( TINYXML2_INCLUDE_DIRS "tinyxml2.h"
              PATH_SUFFIXES "tinyxml2" )
 
   find_library( TINYXML2_LIBRARIES
-                NAMES "tinyxml2"
+                NAMES "tinyxml2d"
                 PATH_SUFFIXES "tinyxml2" )
 endif()
 
