diff --git a/sway/ipc-server.c b/sway/ipc-server.c
index 00b01d7..8f60cfb 100644
--- a/sway/ipc-server.c
+++ b/sway/ipc-server.c
@@ -140,7 +140,7 @@ struct sockaddr_un *ipc_user_sockaddr(void) {
 		dir = "/tmp";
 	}
 	if (path_size <= snprintf(ipc_sockaddr->sun_path, path_size,
-			"%s/sway-ipc.%u.%i.sock", dir, getuid(), getpid())) {
+			"%s/sway-ipc.%u.sock", dir, getuid())) {
 		sway_abort("Socket path won't fit into ipc_sockaddr->sun_path");
 	}
 
diff --git a/sway/main.c b/sway/main.c
index 85bc2f1..032fd25 100644
--- a/sway/main.c
+++ b/sway/main.c
@@ -151,7 +151,8 @@ static void log_kernel(void) {
 }
 
 static bool detect_suid(void) {
-	if (geteuid() != 0 && getegid() != 0) {
+	if (geteuid() == 0 && getegid() == 0) {
+		sway_log(SWAY_INFO, "Running sway as root user");
 		return false;
 	}
 
@@ -309,11 +310,6 @@ int main(int argc, char **argv) {
 		}
 	}
 
-	// SUID operation is deprecated, so block it for now.
-	if (detect_suid()) {
-		exit(EXIT_FAILURE);
-	}
-
 	// Since wayland requires XDG_RUNTIME_DIR to be set, abort with just the
 	// clear error message (when not running as an IPC client).
 	if (!getenv("XDG_RUNTIME_DIR") && optind == argc) {
@@ -363,6 +359,11 @@ int main(int argc, char **argv) {
 		return 0;
 	}
 
+	// SUID operation is deprecated, so block it for now.
+	if (detect_suid()) {
+		exit(EXIT_FAILURE);
+	}
+
 	detect_proprietary(allow_unsupported_gpu);
 	increase_nofile_limit();
 
