diff --git a/projects/unix/Makefile b/projects/unix/Makefile
index 27a58b4..c80ade8 100755
--- a/projects/unix/Makefile
+++ b/projects/unix/Makefile
@@ -124,7 +124,7 @@ ifneq ("$(filter mips,$(HOST_CPU))","")
   $(warning Architecture "$(HOST_CPU)" not officially supported.')
 endif
 ifneq ("$(filter aarch64,$(HOST_CPU))","")
-    CPU := AARCH
+    CPU := AARCH64
     ARCH_DETECTED := 64BITS
     PIC ?= 1
     NEW_DYNAREC := 1
