diff --git a/plugin.video.filmon/default.py b/plugin.video.filmon/default.py
index 8d97edf..ee57804 100644
--- a/plugin.video.filmon/default.py
+++ b/plugin.video.filmon/default.py
@@ -794,7 +794,7 @@ def IIIiI11ii1I ( name , url , iconimage , description , favorites , deletefav ,
  O00OoOO0oo0 = xbmc . PlayList ( xbmc . PLAYLIST_VIDEO )
  O00OoOO0oo0 . clear ( )
  O00OoOO0oo0 . add ( url , iiIi1iI1iIii )
- xbmc . Player ( xbmc . PLAYER_CORE_MPLAYER ) . play ( O00OoOO0oo0 )
+ xbmc . Player ( ) . play ( O00OoOO0oo0 )
  IiiiI . close ( )
  if 96 - 96: OoOoOO00 . o0oOOo0O0Ooo - ooOoO0o
 def IiI111111IIII ( name , url , iconimage , description , favorites , deletefav , record , deleterecord , tvguide , programme_id , startdate_time ) :
