--- a/data/InputAutoCfg.ini	2015-07-08 07:33:14.000000000 +0200
+++ b/data/InputAutoCfg.ini	2015-07-25 18:08:26.285496091 +0200
@@ -721,15 +721,15 @@
 DPad L = hat(0 Left)
 DPad D = hat(0 Down)
 DPad U = hat(0 Up)
-Start = button(7)
-Z Trig = axis(2+)
+Start = button(9)
+Z Trig = button(6)
 B Button = button(2)
 A Button = button(0)
-C Button R = axis(4+)
-C Button L = axis(4-) button(3)
-C Button D = axis(3+) button(1)
+C Button R = axis(2+)
+C Button L = axis(2-)
+C Button D = axis(3+)
 C Button U = axis(3-)
-R Trig = button(5) axis(5+)
+R Trig = button(5)
 L Trig = button(4)
 Mempak switch = 
 Rumblepak switch = 
