diff --git a/SDL/SDLMain.cpp b/SDL/SDLMain.cpp
index f88d238..ed83e71 100644
--- a/SDL/SDLMain.cpp
+++ b/SDL/SDLMain.cpp
@@ -628,7 +628,8 @@ int main(int argc, char *argv[]) {
 #if !defined(MOBILE_DEVICE)
 			case SDL_WINDOWEVENT:
 				switch (event.window.event) {
-				case SDL_WINDOWEVENT_SIZE_CHANGED:  // better than RESIZED, more general
+				//case SDL_WINDOWEVENT_SIZE_CHANGED:  // better than RESIZED, more general
+				case SDL_WINDOWEVENT_RESIZED:
 				case SDL_WINDOWEVENT_MAXIMIZED:
 				case SDL_WINDOWEVENT_RESTORED:
 				{
