From 3b5df53cd096c760ab401be28625d5d95c47e966 Mon Sep 17 00:00:00 2001
From: orangepi-xunlong <258384131@qq.com>
Date: Tue, 12 Jul 2022 14:05:04 +0800
Subject: [PATCH] Set ethernet_phy variable based on the ethernet phy id

---
 drivers/net/phy/phy.c | 5 +++++
 include/phy.h         | 3 +++
 2 files changed, 8 insertions(+)

diff --git a/drivers/net/phy/phy.c b/drivers/net/phy/phy.c
index 69acb694..f4aea308 100644
--- a/drivers/net/phy/phy.c
+++ b/drivers/net/phy/phy.c
@@ -649,6 +649,11 @@ static struct phy_driver *get_phy_driver(struct phy_device *phydev,
 	int phy_id = phydev->phy_id;
 	struct phy_driver *drv = NULL;
 
+	if(phy_id == YT_8531C_PHY_ID)
+		env_set("ethernet_phy", "yt8531c");
+	if(phy_id == RTL_8211F_PHY_ID)
+		env_set("ethernet_phy", "rtl8211f");
+
 	list_for_each(entry, &phy_drivers) {
 		drv = list_entry(entry, struct phy_driver, list);
 		if ((drv->uid & drv->mask) == (phy_id & drv->mask))
diff --git a/include/phy.h b/include/phy.h
index 6b928636..4dd7ea21 100644
--- a/include/phy.h
+++ b/include/phy.h
@@ -21,6 +21,9 @@
 
 struct udevice;
 
+#define YT_8531C_PHY_ID 0x4f51e91b
+#define RTL_8211F_PHY_ID 0x001cc916
+
 #define PHY_FIXED_ID		0xa5a55a5a
 #define PHY_NCSI_ID            0xbeefcafe
 
-- 
2.37.1

