From b5c8dc48569a833a6ffcf1faa51753edae58eef7 Mon Sep 17 00:00:00 2001
From: Gleb Mazovetskiy <glex.spb@gmail.com>
Date: Sun, 21 Jan 2024 16:21:44 +0000
Subject: [PATCH] Enable VSync by default

---
 engine/client/ref_common.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/engine/client/ref_common.c b/engine/client/ref_common.c
index 89f1211b..0e3a6674 100644
--- a/engine/client/ref_common.c
+++ b/engine/client/ref_common.c
@@ -8,7 +8,7 @@
 struct ref_state_s ref;
 ref_globals_t refState;
 
-CVAR_DEFINE_AUTO( gl_vsync, "0", FCVAR_ARCHIVE,  "enable vertical syncronization" );
+CVAR_DEFINE_AUTO( gl_vsync, "1", FCVAR_ARCHIVE,  "enable vertical syncronization" );
 CVAR_DEFINE_AUTO( r_showtextures, "0", FCVAR_CHEAT, "show all uploaded textures" );
 CVAR_DEFINE_AUTO( r_adjust_fov, "1", FCVAR_ARCHIVE, "making FOV adjustment for wide-screens" );
 CVAR_DEFINE_AUTO( r_decals, "4096", FCVAR_ARCHIVE, "sets the maximum number of decals" );
-- 
2.40.1

