diff --git a/Makefile.common b/Makefile.common
index 0b2889f0b2..9cd9b9fe8a 100644
--- a/Makefile.common
+++ b/Makefile.common
@@ -1101,7 +1101,7 @@ OBJ += gfx/drivers_context/gfx_null_ctx.o
 ifeq ($(HAVE_KMS), 1)
    HAVE_AND_WILL_USE_DRM = 1
    OBJ += gfx/drivers_context/drm_ctx.o
-   DEF_FLAGS += $(GBM_CFLAGS) $(DRM_CFLAGS)
+   DEF_FLAGS += $(GBM_CFLAGS) $(DRM_CFLAGS) -D__GBM__
    LIBS += $(GBM_LIBS) $(DRM_LIBS)
 endif
