diff --git a/src/osdep/amiberry.cpp b/src/osdep/amiberry.cpp
index c747aca..9109fec 100644
--- a/src/osdep/amiberry.cpp
+++ b/src/osdep/amiberry.cpp
@@ -58,7 +58,7 @@ int input_default_mouse_speed = 100;
 bool input_keyboard_as_joystick_stop_keypresses = false;
 static char default_open_gui_key[128];
 static char default_quit_key[128];
-int rotation_angle = 0;
+int rotation_angle = -90;
 
 // Default Enter GUI key is F12
 int enter_gui_key = 0;
