diff --git a/src/osdep/amiberry.cpp b/src/osdep/amiberry.cpp
index 21312e8..faf5610 100644
--- a/src/osdep/amiberry.cpp
+++ b/src/osdep/amiberry.cpp
@@ -58,7 +58,7 @@ int input_default_mouse_speed = 100;
 bool input_keyboard_as_joystick_stop_keypresses = false;
 static char default_open_gui_key[128];
 static char default_quit_key[128];
-int rotation_angle = 0;
+int rotation_angle = -90;
 bool default_horizontal_centering = false;
 bool default_vertical_centering = false;
 int default_scaling_method = -1;
