Index: source/common/platform/posix/unix/i_specialpaths.cpp
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/source/common/platform/posix/unix/i_specialpaths.cpp b/source/common/platform/posix/unix/i_specialpaths.cpp
--- a/source/common/platform/posix/unix/i_specialpaths.cpp	(revision 486da6922dac5380b4e83cc4c680ad90afb91ca9)
+++ b/source/common/platform/posix/unix/i_specialpaths.cpp	(date 1657595110880)
@@ -45,59 +45,7 @@

 FString GetUserFile (const char *file)
 {
-	FString path;
-	struct stat info;
-
-	path = NicePath("$HOME/" GAME_DIR "/");
-
-	if (stat (path, &info) == -1)
-	{
-		struct stat extrainfo;
-
-		// Sanity check for $HOME/.config
-		FString configPath = NicePath("$HOME/.config/");
-		if (stat (configPath, &extrainfo) == -1)
-		{
-			if (mkdir (configPath, S_IRUSR | S_IWUSR | S_IXUSR) == -1)
-			{
-				I_FatalError ("Failed to create $HOME/.config directory:\n%s", strerror(errno));
-			}
-		}
-		else if (!S_ISDIR(extrainfo.st_mode))
-		{
-			I_FatalError ("$HOME/.config must be a directory");
-		}
-
-		// This can be removed after a release or two
-		// Transfer the old zdoom directory to the new location
-		bool moved = false;
-		FString oldpath = NicePath("$HOME/." GAMENAMELOWERCASE "/");
-		if (stat (oldpath, &extrainfo) != -1)
-		{
-			if (rename(oldpath, path) == -1)
-			{
-				I_Error ("Failed to move old " GAMENAMELOWERCASE " directory (%s) to new location (%s).",
-					oldpath.GetChars(), path.GetChars());
-			}
-			else
-				moved = true;
-		}
-
-		if (!moved && mkdir (path, S_IRUSR | S_IWUSR | S_IXUSR) == -1)
-		{
-			I_FatalError ("Failed to create %s directory:\n%s",
-				path.GetChars(), strerror (errno));
-		}
-	}
-	else
-	{
-		if (!S_ISDIR(info.st_mode))
-		{
-			I_FatalError ("%s must be a directory", path.GetChars());
-		}
-	}
-	path += file;
-	return path;
+    return FString("/userdata/saves/raze/") + file;
 }

 //===========================================================================
@@ -112,7 +60,7 @@
 {
 	// Don't use GAME_DIR and such so that ZDoom and its child ports can
 	// share the node cache.
-	FString path = NicePath("$HOME/.config/" GAMENAMELOWERCASE);
+	FString path = NicePath("/userdata/saves/raze");
 	if (create)
 	{
 		CreatePath(path);
@@ -132,7 +80,7 @@
 {
 	// Don't use GAME_DIR and such so that ZDoom and its child ports can
 	// share the node cache.
-	FString path = NicePath("$HOME/.config/zdoom/cache");
+	FString path = NicePath("/userdata/saves/raze/cache");
 	if (create)
 	{
 		CreatePath(path);
@@ -150,7 +98,7 @@

 FString M_GetAutoexecPath()
 {
-	return GetUserFile("autoexec.cfg");
+	return FString("/userdata/system/configs/raze/autoexec.cfg");
 }

 //===========================================================================
@@ -165,7 +113,7 @@

 FString M_GetConfigPath(bool for_reading)
 {
-	return GetUserFile(GAMENAMELOWERCASE ".ini");
+	return FString("/userdata/system/configs/raze/raze.ini");
 }

 //===========================================================================
@@ -178,7 +126,7 @@

 FString M_GetScreenshotsPath()
 {
-	return NicePath("$HOME/" GAME_DIR "/screenshots/");
+	return FString("/userdata/screenshots/");
 }

 //===========================================================================
@@ -191,7 +139,7 @@

 FString M_GetSavegamesPath()
 {
-	return NicePath("$HOME/" GAME_DIR "/");
+	return FString("/userdata/saves/raze/");
 }

 //===========================================================================
@@ -204,7 +152,7 @@

 FString M_GetDocumentsPath()
 {
-	return NicePath("$HOME/" GAME_DIR "/");
+	return M_GetSavegamesPath();
 }

 //===========================================================================
Index: source/common/platform/posix/sdl/i_main.cpp
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/source/common/platform/posix/sdl/i_main.cpp b/source/common/platform/posix/sdl/i_main.cpp
--- a/source/common/platform/posix/sdl/i_main.cpp	(revision 486da6922dac5380b4e83cc4c680ad90afb91ca9)
+++ b/source/common/platform/posix/sdl/i_main.cpp	(date 1657594523787)
@@ -175,19 +175,7 @@
 	Args = new FArgs(argc, argv);

 	// Should we even be doing anything with progdir on Unix systems?
-	char program[PATH_MAX];
-	if (realpath (argv[0], program) == NULL)
-		strcpy (program, argv[0]);
-	char *slash = strrchr (program, '/');
-	if (slash != NULL)
-	{
-		*(slash + 1) = '\0';
-		progdir = program;
-	}
-	else
-	{
-		progdir = "./";
-	}
+	progdir = "/usr/share/raze/";

 	I_StartupJoysticks();

Index: source/core/gameconfigfile.cpp
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/source/core/gameconfigfile.cpp b/source/core/gameconfigfile.cpp
--- a/source/core/gameconfigfile.cpp	(revision 486da6922dac5380b4e83cc4c680ad90afb91ca9)
+++ b/source/core/gameconfigfile.cpp	(date 1657595110893)
@@ -141,6 +141,7 @@
 		SetValueForKey ("Path", "/usr/local/share/games/eduke32", true);
 		SetValueForKey ("Path", "/usr/share/games/nblood", true);
 		SetValueForKey ("Path", "/usr/local/share/games/nblood", true);
+		SetValueForKey ("Path", "/usr/share/raze", true);
 #endif
 	}

Index: source/core/gamecontrol.cpp
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/source/core/gamecontrol.cpp b/source/core/gamecontrol.cpp
--- a/source/core/gamecontrol.cpp	(revision 486da6922dac5380b4e83cc4c680ad90afb91ca9)
+++ b/source/core/gamecontrol.cpp	(date 1657595110906)
@@ -971,7 +971,7 @@
 	FString logfile = Args->TakeValue("+logfile");

 	// As long as this engine is still in prerelease mode let's always write a log file.
-	if (logfile.IsEmpty()) logfile.Format("%s" GAMENAMELOWERCASE ".log", M_GetDocumentsPath().GetChars());
+	if (logfile.IsEmpty()) logfile.Format("/userdata/system/logs/raze.log");

 	if (logfile.IsNotEmpty())
 	{
