--- a/arch/arm64/boot/dts/allwinner/sun50i-h616-orangepi-zero2.dts	2023-07-07 21:51:08.954234047 +0200
+++ b/arch/arm64/boot/dts/allwinner/sun50i-h616-orangepi-zero2.dts	2023-07-07 21:54:44.446878817 +0200
@@ -37,19 +37,21 @@
 		};
 	};
 
-	leds: leds {
+	leds {
 		compatible = "gpio-leds";
 
 		led-red {
-			label = "red_led";
+			function = LED_FUNCTION_STATUS;
+			color = <LED_COLOR_ID_RED>;
 			gpios = <&pio 2 12 GPIO_ACTIVE_HIGH>; /* PC12 */
-			linux,default-trigger = "none";
+			linux,default-trigger = "heartbeat";
 		};
 
 		led-green {
-			label = "green_led";
+			function = LED_FUNCTION_POWER;
+			color = <LED_COLOR_ID_GREEN>;
 			gpios = <&pio 2 13 GPIO_ACTIVE_HIGH>; /* PC13 */
-			linux,default-trigger = "heartbeat";
+			default-state = "on";
 		};
 	};
 
