--- a/arch/arm64/boot/dts/allwinner/sun50i-h6-orangepi-3-lts.dts	2023-07-08 22:25:33.117853247 +0200
+++ b/arch/arm64/boot/dts/allwinner/sun50i-h6-orangepi-3-lts.dts	2023-07-08 22:26:25.418940556 +0200
@@ -103,6 +103,25 @@
 		reset-gpios = <&r_pio 1 3 GPIO_ACTIVE_LOW>; /* PM3 */
 		post-power-on-delay-ms = <200>;
 	};
+
+	sunxi-info {
+		compatible = "allwinner,sun50i-h6-sys-info";
+		status = "okay";
+	};
+
+	addr_mgt: addr-mgt {
+		compatible = "allwinner,sunxi-addr_mgt";
+		type_addr_wifi = <0x2>;
+		type_addr_bt = <0x2>;
+		type_addr_eth = <0x2>;
+		status = "okay";
+	};
+
+	dump_reg: dump_reg@20000 {
+		compatible = "allwinner,sunxi-dump-reg";
+		reg = <0x0 0x03001000 0x0 0x0f20>;
+		status = "okay";
+	};
 };
 
 &cpu0 {
