--- a/src/osd/retro/osinline.h	2022-06-14 23:05:20.000000000 +0200
+++ b/src/osd/retro/osinline.h	2023-09-19 01:51:07.715116332 +0200
@@ -35,7 +35,7 @@
 
 #endif
 
-#if defined(__aarch64__) || defined(_M_ARM64) || defined(__arm__) || defined(_M_ARM)
+#if defined(__aarch64__) || defined(_M_ARM64) || defined(__arm__) || defined(_M_ARM) || defined(__riscv)
 #ifndef YieldProcessor
 #define YieldProcessor() do {} while (0)
 #define osd_yield_processor() YieldProcessor()
