From 1821d2822783e8c24733b05138e06d9a9f4d309d Mon Sep 17 00:00:00 2001
From: Josh Cotton <jcotton42@users.noreply.github.com>
Date: Mon, 14 Nov 2022 07:57:40 -0800
Subject: [PATCH] Input: xpad - add support for 8BitDo Ultimate Wireless
 controller dongle

Signed-off-by: Josh Cotton <jcotton42@outlook.com>
---
 xpad.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/xpad.c b/xpad.c
index d61afde..6efc96a 100644
--- a/xpad.c
+++ b/xpad.c
@@ -371,6 +371,7 @@ static const struct xpad_device {
	{ 0x24c6, 0x5d04, "Razer Sabertooth", 0, XTYPE_XBOX360 },
	{ 0x24c6, 0xfafe, "Rock Candy Gamepad for Xbox 360", 0, XTYPE_XBOX360 },
	{ 0x2563, 0x058d, "OneXPlayer Gamepad", 0, XTYPE_XBOX360 },
+	{ 0x2dc8, 0x3106, "8BitDo Ultimate Wireless Controller", 0, XTYPE_XBOX360 },
 	{ 0x31e3, 0x1100, "Wooting One", 0, XTYPE_XBOX360 },
 	{ 0x31e3, 0x1200, "Wooting Two", 0, XTYPE_XBOX360 },
 	{ 0x31e3, 0x1210, "Wooting Lekker", 0, XTYPE_XBOX360 },
@@ -512,6 +513,7 @@ static const struct usb_device_id xpad_table[] = {
	XPAD_XBOX360_VENDOR(0x24c6),		/* PowerA Controllers */
	XPAD_XBOX360_VENDOR(0x2563),		/* OneXPlayer Gamepad */
	XPAD_XBOX360_VENDOR(0x260d),		/* Dareu H101 */
+	XPAD_XBOX360_VENDOR(0x2dc8),		/* 8BitDo Ultimate Wireless Controller */
	XPAD_XBOX360_VENDOR(0x2f24),		/* GameSir Controllers */
	XPAD_XBOX360_VENDOR(0x31e3),		/* Wooting Keyboards */
	XPAD_XBOX360_VENDOR(0x3285),		/* Nacon GC-100 */
