diff --git a/Sources/Jazz2/ContentResolver.cpp b/Sources/Jazz2/ContentResolver.cpp
index 1b3fa52ce9..7d79a9505f 100644
--- a/Sources/Jazz2/ContentResolver.cpp
+++ b/Sources/Jazz2/ContentResolver.cpp
@@ -227,17 +227,16 @@ namespace Jazz2
 			auto localStorage = fs::GetLocalStorage();
 			if (!localStorage.empty()) {
 				// Use "$XDG_DATA_HOME/Jazz² Resurrection/" if exists (for backward compatibility), otherwise "$XDG_DATA_HOME/{NCINE_LINUX_PACKAGE}/"
-				_cachePath = fs::CombinePath(localStorage, "Jazz² Resurrection/Cache/"_s);
+				_cachePath = "/userdata/system/cache/jazz2/";
 				if (!fs::DirectoryExists(_cachePath)) {
-					auto appData = fs::CombinePath(localStorage, NCINE_LINUX_PACKAGE);
-					_cachePath = fs::CombinePath(appData, "Cache/"_s);
+					fs::CreateDirectories(_cachePath);
 				}
 			} else {
 				_cachePath = "Cache/"_s;
 			}
 
 			// Prefer system-wide Source only if it exists and the local one doesn't exist
-			_sourcePath = fs::CombinePath(fs::GetDirectoryName(_cachePath), "Source/"_s);
+			_sourcePath = "/userdata/roms/jazz2/";
 			if (!fs::FindPathCaseInsensitive(fs::CombinePath(_sourcePath, "Anims.j2a"_s)) &&
 				!fs::FindPathCaseInsensitive(fs::CombinePath(_sourcePath, "AnimsSw.j2a"_s))) {
 				auto systemWideSource = NCINE_INSTALL_PREFIX "/share/" NCINE_LINUX_PACKAGE "/Source/";
index 5b5199b0b6..14a1be044a 100644
--- a/Sources/Shared/IO/FileSystem.cpp
+++ b/Sources/Shared/IO/FileSystem.cpp
@@ -1265,7 +1265,7 @@ namespace Death { namespace IO {
 #elif defined(DEATH_TARGET_UNIX) || defined(DEATH_TARGET_EMSCRIPTEN) || defined(DEATH_TARGET_VITA)
 		StringView config = ::getenv("XDG_CONFIG_HOME");
 		if (IsAbsolutePath(config)) {
-			return CombinePath(config, applicationName);
+			return CombinePath(config, "jazz2");
 		}
 
 		StringView home = ::getenv("HOME");