diff --git a/bin/pironman b/bin/pironman
index 02e1c837fe..0fdedf5cfb 100644
--- a/bin/pironman
+++ b/bin/pironman
@@ -178,24 +178,6 @@ while [[ $# -gt 0 ]]; do
       grep -E '=.' "${CONF}"
       exit 0
       ;;
-    -a|--auto)
-      case "$2" in
-        "on"|true)
-          echo "enable auto-start at boot"
-          systemctl enable pironman.service
-          shift 2
-          ;;
-        "off"|false)
-          echo "disable auto-start at boot"
-          systemctl disable pironman.service
-          shift 2
-        ;;
-        *)
-          echo "usage: pironman -a|--auto [on/off]"
-          exit 1
-          ;;
-      esac
-      ;;
     -u|--unit)
       if [ $# -lt 2 ] ;then
         echo "usage: pironman -c|--check [C/F]"