diff --git a/Source/Core/VideoBackends/Vulkan/RasterFont.cpp b/Source/Core/VideoBackends/Vulkan/RasterFont.cpp
index 7410ba5..03a2d27 100644
--- a/Source/Core/VideoBackends/Vulkan/RasterFont.cpp
+++ b/Source/Core/VideoBackends/Vulkan/RasterFont.cpp
@@ -15,12 +15,13 @@
 // Based on OGL RasterFont
 // TODO: We should move this to common.

+#define CHARACTER_WIDTH 8
+#define CHARACTER_HEIGHT 13
+#define CHARACTER_OFFSET 32
+#define CHARACTER_COUNT 95
+
 namespace Vulkan
 {
-constexpr int CHARACTER_WIDTH = 8;
-constexpr int CHARACTER_HEIGHT = 13;
-constexpr int CHARACTER_OFFSET = 32;
-constexpr int CHARACTER_COUNT = 95;

 static const u8 rasters[CHARACTER_COUNT][CHARACTER_HEIGHT] = {
     {0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00},
