diff --git a/UI/EmuScreen.cpp b/UI/EmuScreen.cpp
index bf90e06..e135ffc 100644
--- a/UI/EmuScreen.cpp
+++ b/UI/EmuScreen.cpp
@@ -698,24 +698,24 @@ void EmuScreen::onVKeyUp(int virtualKeyCode) {
 
 // Handles control rotation due to internal screen rotation.
 static void SetPSPAxis(char axis, float value, int stick) {
-	switch (g_Config.iInternalScreenRotation) {
-	case ROTATION_LOCKED_HORIZONTAL:
-		// Standard rotation.
-		break;
-	case ROTATION_LOCKED_HORIZONTAL180:
-		value = -value;
-		break;
-	case ROTATION_LOCKED_VERTICAL:
-		value = axis == 'Y' ? value : -value;
-		axis = (axis == 'X') ? 'Y' : 'X';
-		break;
-	case ROTATION_LOCKED_VERTICAL180:
-		value = axis == 'Y' ? -value : value;
-		axis = (axis == 'X') ? 'Y' : 'X';
-		break;
-	default:
-		break;
-	}
+	//switch (g_Config.iInternalScreenRotation) {
+	//case ROTATION_LOCKED_HORIZONTAL:
+	//	// Standard rotation.
+	//	break;
+	//case ROTATION_LOCKED_HORIZONTAL180:
+	//	value = -value;
+	//	break;
+	//case ROTATION_LOCKED_VERTICAL:
+	//	value = axis == 'Y' ? value : -value;
+	//	axis = (axis == 'X') ? 'Y' : 'X';
+	//	break;
+	//case ROTATION_LOCKED_VERTICAL180:
+	//	value = axis == 'Y' ? -value : value;
+	//	axis = (axis == 'X') ? 'Y' : 'X';
+	//	break;
+	//default:
+	//	break;
+	//}
 	if (axis == 'X')
 		__CtrlSetAnalogX(value, stick);
 	else if (axis == 'Y')
@@ -781,17 +781,17 @@ static int RotatePSPKeyCode(int x) {
 
 void EmuScreen::pspKey(int pspKeyCode, int flags) {
 	int rotations = 0;
-	switch (g_Config.iInternalScreenRotation) {
-	case ROTATION_LOCKED_HORIZONTAL180:
-		rotations = 2;
-		break;
-	case ROTATION_LOCKED_VERTICAL:
-		rotations = 1;
-		break;
-	case ROTATION_LOCKED_VERTICAL180:
-		rotations = 3;
-		break;
-	}
+	//switch (g_Config.iInternalScreenRotation) {
+	//case ROTATION_LOCKED_HORIZONTAL180:
+	//	rotations = 2;
+	//	break;
+	//case ROTATION_LOCKED_VERTICAL:
+	//	rotations = 1;
+	//	break;
+	//case ROTATION_LOCKED_VERTICAL180:
+	//	rotations = 3;
+	//	break;
+	//}
 
 	for (int i = 0; i < rotations; i++) {
 		pspKeyCode = RotatePSPKeyCode(pspKeyCode);
