--- a/src/osdep/amiberry.cpp	2024-04-27 14:43:54.983411996 +0200
+++ b/src/osdep/amiberry.cpp	2024-04-27 14:55:30.996331192 +0200
@@ -2914,22 +2914,22 @@
 
 std::string get_data_path()
 {
-	return fix_trailing(data_dir);
+	return std::string("/userdata/system/configs/amiberry/data/");
 }
 
 void get_saveimage_path(char* out, int size, int dir)
 {
-	_tcsncpy(out, fix_trailing(saveimage_dir).c_str(), size - 1);
+	strncpy(out, "/userdata/saves/amiga/", size - 1);
 }
 
 std::string get_configuration_path()
 {
-	return fix_trailing(config_path);
+	return std::string("/userdata/system/configs/amiberry/conf/");
 }
 
 void get_configuration_path(char* out, int size)
 {
-	_tcsncpy(out, fix_trailing(config_path).c_str(), size - 1);
+	strncpy(out, "/userdata/system/configs/amiberry/conf/", size - 1);
 }
 
 void set_configuration_path(const std::string& newpath)
@@ -2959,7 +2959,7 @@
 
 std::string get_controllers_path()
 {
-	return fix_trailing(controllers_path);
+	return std::string("/userdata/system/configs/amiberry/conf/retroarch/inputs/");
 }
 
 void set_controllers_path(const std::string& newpath)
@@ -3007,7 +3007,7 @@
 
 std::string get_whdbootpath()
 {
-	return fix_trailing(whdboot_path);
+	return std::string("/usr/share/amiberry/whdboot/");
 }
 
 void set_whdbootpath(const std::string& newpath)
@@ -3087,7 +3087,7 @@
 
 void get_savestate_path(char* out, int size)
 {
-	_tcsncpy(out, fix_trailing(savestate_dir).c_str(), size - 1);
+	strncpy(out, "/userdata/saves/amiga/", size - 1);
 }
 
 void fetch_ripperpath(TCHAR* out, int size)
@@ -3107,7 +3107,7 @@
 
 std::string get_screenshot_path()
 {
-	return fix_trailing(screenshot_dir);
+	return std::string("/userdata/screenshots/");
 }
 
 void get_video_path(char* out, int size)
@@ -3492,22 +3492,22 @@
 
 	// Paths
 	write_string_option("path", current_dir);
-	write_string_option("config_path", config_path);
-	write_string_option("controllers_path", controllers_path);
-	write_string_option("retroarch_config", retroarch_file);
+	write_string_option("config_path", "/userdata/system/configs/amiberry/conf/");
+	write_string_option("controllers_path", "/userdata/system/configs/amiberry/conf/retroarch/inputs/");
+	write_string_option("retroarch_config", "/userdata/system/configs/amiberry/conf/retroarch/retroarchcustom.cfg");
 	write_string_option("whdboot_path", whdboot_path);
 	write_string_option("whdload_arch_path", whdload_arch_path);
 	write_string_option("floppy_path", floppy_path);
 	write_string_option("harddrive_path", harddrive_path);
 	write_string_option("cdrom_path", cdrom_path);
-	write_string_option("logfile_path", logfile_path);
-	write_string_option("rom_path", rom_path);
+	write_string_option("logfile_path", "/userdata/system/logs/amiberry.log");
+	write_string_option("rom_path", "/userdata/bios/amiga/");
 	write_string_option("rp9_path", rp9_path);
 	write_string_option("floppy_sounds_dir", floppy_sounds_dir);
 	write_string_option("data_dir", data_dir);
-	write_string_option("saveimage_dir", saveimage_dir);
-	write_string_option("savestate_dir", savestate_dir);
-	write_string_option("screenshot_dir", screenshot_dir);
+	write_string_option("saveimage_dir", "/userdata/saves/amiga/");
+	write_string_option("savestate_dir", "/userdata/saves/amiga/");
+	write_string_option("screenshot_dir", "/userdata/screenshots/");
 	write_string_option("ripper_path", ripper_path);
 	write_string_option("inputrecordings_dir", input_dir);
 	write_string_option("nvram_dir", nvram_dir);
@@ -4127,7 +4127,7 @@
 		abort();
 	}
 
-	snprintf(savestate_fname, sizeof savestate_fname, "%s/default.ads", fix_trailing(savestate_dir).c_str());
+	snprintf(savestate_fname, sizeof savestate_fname, "%s/default.ads", "/userdata/saves/amiga");
 	logging_init();
 #if defined (CPU_arm)
 	memset(&action, 0, sizeof action);
