diff --git a/src/osdep/amiberry.cpp b/src/osdep/amiberry.cpp
index 6c66bf2..c747aca 100644
--- a/src/osdep/amiberry.cpp
+++ b/src/osdep/amiberry.cpp
@@ -623,7 +623,7 @@ void fetch_datapath(char* out, int size)
 
 void fetch_saveimagepath(char* out, int size, int dir)
 {
-	strncpy(out, start_path_data, size - 1);
+	strncpy(out, "/userdata/saves/amiga", size - 1);
 	strncat(out, "/savestates/", size - 1);
 }
 
@@ -678,13 +678,13 @@ void fetch_rp9path(char* out, int size)
 
 void fetch_savestatepath(char* out, int size)
 {
-	strncpy(out, start_path_data, size - 1);
+	strncpy(out, "/userdata/saves/amiga", size - 1);
 	strncat(out, "/savestates/", size - 1);
 }
 
 void fetch_screenshotpath(char* out, int size)
 {
-	strncpy(out, start_path_data, size - 1);
+	strncpy(out, "/userdata", size - 1);
 	strncat(out, "/screenshots/", size - 1);
 }
 
@@ -830,7 +830,7 @@ void save_amiberry_settings(void)
 {
 	char path[MAX_DPATH];
 
-	snprintf(path, MAX_DPATH, "%s/conf/amiberry.conf", start_path_data);
+	snprintf(path, MAX_DPATH, "%s/conf/amiberry.conf", "/userdata/system/configs/amiberry");
 	const auto f = fopen(path, "we");
 	if (!f)
 		return;
@@ -904,13 +904,13 @@ void save_amiberry_settings(void)
 	snprintf(buffer, MAX_DPATH, "path=%s\n", currentDir);
 	fputs(buffer, f);
 
-	snprintf(buffer, MAX_DPATH, "config_path=%s\n", config_path);
+	snprintf(buffer, MAX_DPATH, "/userdata/system/configs/amiberry/conf", config_path);
 	fputs(buffer, f);
 
-	snprintf(buffer, MAX_DPATH, "controllers_path=%s\n", controllers_path);
+	snprintf(buffer, MAX_DPATH, "/userdata/system/configs/amiberry/conf/retroarch/inputs", controllers_path);
 	fputs(buffer, f);
 
-	snprintf(buffer, MAX_DPATH, "retroarch_config=%s\n", retroarch_file);
+	snprintf(buffer, MAX_DPATH, "/userdata/system/configs/amiberry/conf/retroarch/retroarchcustom.cfg", retroarch_file);
 	fputs(buffer, f);
 
 	snprintf(buffer, MAX_DPATH, "rom_path=%s\n", rom_path);
@@ -976,9 +976,11 @@ void load_amiberry_settings(void)
 	char path[MAX_DPATH];
 	int i;
 	strncpy(currentDir, start_path_data, MAX_DPATH - 1);
-	snprintf(config_path, MAX_DPATH, "%s/conf/", start_path_data);
-	snprintf(controllers_path, MAX_DPATH, "%s/controllers/", start_path_data);
-	snprintf(retroarch_file, MAX_DPATH, "%s/conf/retroarch.cfg", start_path_data);
+
+	/* snprintf(config_path, MAX_DPATH, "%s/conf/", start_path_data);  */
+	snprintf(config_path, MAX_DPATH, "/userdata/system/configs/amiberry/conf");
+	snprintf(controllers_path, MAX_DPATH, "/userdata/system/configs/amiberry/conf/retroarch/inputs");
+	snprintf(retroarch_file, MAX_DPATH, "/userdata/system/configs/amiberry/conf/retroarch/retroarchcustom.cfg");
 
 #ifdef ANDROID
 	char afepath[MAX_DPATH];
@@ -989,12 +991,14 @@ void load_amiberry_settings(void)
 		closedir(afedir);
 	}
 	else
-		snprintf(rom_path, MAX_DPATH, "%s/kickstarts/", start_path_data);
+		snprintf(rom_path, MAX_DPATH, "/userdata/bios/");
+
 #else
-	snprintf(rom_path, MAX_DPATH, "%s/kickstarts/", start_path_data);
+	snprintf(rom_path, MAX_DPATH, "/userdata/bios/");
 #endif
 	snprintf(rp9_path, MAX_DPATH, "%s/rp9/", start_path_data);
-	snprintf(path, MAX_DPATH, "%s/conf/amiberry.conf", start_path_data);
+	snprintf(path, MAX_DPATH, "%s/conf/adfdir.conf", "/userdata/system/configs/amiberry");
+
 
 	const auto fh = zfile_fopen(path, _T("r"), ZFD_NORMAL);
 	if (fh)
@@ -1167,7 +1171,8 @@ int main(int argc, char* argv[])
 	load_amiberry_settings();
 	rp9_init();
 
-	snprintf(savestate_fname, sizeof savestate_fname, "%s/savestates/default.ads", start_path_data);
+	snprintf(savestate_fname, sizeof savestate_fname, "%s/savestates/default.ads", "/userdata/saves/amiga");
+
 	logging_init();
 
 	memset(&action, 0, sizeof action);
