diff --git a/Makefile b/Makefile
index e1fdd4d..4b1cba4 100644
--- a/Makefile
+++ b/Makefile
@@ -191,8 +191,9 @@ endif
 #
 ifdef USE_SDL2
 all: guisan $(PROG)
-export SDL_CFLAGS := $(shell sdl2-config --cflags)
-export SDL_LDFLAGS := $(shell sdl2-config --libs)
+SDL_CONFIG ?= sdl2-config
+SDL_CFLAGS = $(shell $(SDL_CONFIG) --cflags)
+SDL_LDFLAGS = $(shell $(SDL_CONFIG) --libs)
 
 CPPFLAGS += $(SDL_CFLAGS) -Iguisan-dev/include
 LDFLAGS += $(SDL_LDFLAGS) -lSDL2_image -lSDL2_ttf -lguisan -Lguisan-dev/lib
