diff --git a/Makefile b/Makefile.new
index 195ffd8b7e..5fc0ce12a6 100644
--- a/Makefile
+++ b/Makefile.new
@@ -181,6 +181,12 @@ else ifeq ($(PLATFORM),rpi4-64-dmx)
 	LDFLAGS += $(DISPMANX_LDFLAGS)
 	AARCH64 = 1
 
+# Raspberry Pi 5 (SDL2 64-bit)
+else ifeq ($(PLATFORM),rpi5-64-sdl2)
+	CPUFLAGS = -mcpu=cortex-a76+crc+simd+fp
+	CPPFLAGS += $(CPPFLAGS64)
+	AARCH64 = 1
+
 # Vero 4k (SDL2)
 else ifeq ($(PLATFORM),vero4k)
 	CPUFLAGS = -mcpu=cortex-a7 -mfpu=neon-vfpv4 -mfloat-abi=hard
