diff --git a/Makefile b/Makefile
index 1273d67..307ec76 100644
--- a/Makefile
+++ b/Makefile
@@ -145,6 +145,8 @@ else ifneq (,$(findstring RK,$(PLATFORM)))
         CPUFLAGS += -mcpu=cortex-a72
       else ifneq (,$(findstring RK3328,$(PLATFORM)))
         CPUFLAGS += -mcpu=cortex-a53
+      else ifneq (,$(findstring RK3326,$(PLATFORM)))
+        CPUFLAGS += -mcpu=cortex-a35
       endif
     else ifneq (,$(findstring RK3288,$(PLATFORM)))
       CPUFLAGS += -mcpu=cortex-a17 -mfloat-abi=hard -mfpu=neon-vfpv4
