diff --git a/CMakeLists.txt b/CMakeLists.txt
index e56611e..19cb371 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -138,7 +138,7 @@ if(UNIX AND NOT (APPLE OR ANDROID) AND VULKAN)
 			message(STATUS "Could not find Wayland libraries, disabling Wayland WSI support for Vulkan.")
 		else()
 			include_directories(${WAYLAND_INCLUDE_DIR})
-			add_definitions(-DVK_USE_PLATFORM_WAYLAND_KHR)
+			#add_definitions(-DVK_USE_PLATFORM_WAYLAND_KHR)
 		endif()
 	endif()
 endif()
