--- a/Makefile
+++ b/Makefile
@@ -355,6 +355,18 @@
    WITH_DYNAREC=arm
    COREFLAGS += -DUSE_GENERIC_GLESV2 -DOS_LINUX
    ASFLAGS = -f elf -d ELF_TYPE
+ 
+# AYN Odin
+else ifneq (,$(findstring odin,$(platform)))
+   TARGET := $(TARGET_NAME)_libretro.so
+   LDFLAGS += -shared -Wl,--version-script=$(LIBRETRO_DIR)/link.T -Wl,--no-undefined
+   GL_LIB := -lGLESv2
+   WITH_DYNAREC := aarch64
+   CPUFLAGS += -mcpu=cortex-a75.cortex-a55
+   GLES3= 1
+   GL_LIB := -lGLESv2
+   COREFLAGS += -DUSE_GENERIC_GLESV2 -DOS_LINUX
+   ASFLAGS = -f elf -d ELF_TYPE
 
 # OS X
 else ifneq (,$(findstring osx,$(platform)))
