diff --git a/acinclude.m4 b/acinclude.m4
index 9b85f7f..a2c7117 100644
--- a/acinclude.m4
+++ b/acinclude.m4
@@ -14,35 +14,13 @@ AC_ARG_ENABLE(sdl2,     [  --enable-sdl2           Enable SDL 2.x],
 
   SDL2_CONFIG=no
   if test x$enable_sdl2enable = xyes ; then
-    if test x$sdl2_exec_prefix != x ; then
-      sdl2_args="$sdl2_args --exec-prefix=$sdl2_exec_prefix"
-      if test x${SDL2_CONFIG+set} != xset ; then
-        SDL2_CONFIG=$sdl2_exec_prefix/bin/sdl2-config
-      fi
-    fi
-    if test x$sdl2_prefix != x ; then
-      sdl2_args="$sdl2_args --prefix=$sdl2_prefix"
-      if test x${SDL2_CONFIG+set} != xset ; then
-        SDL2_CONFIG=$sdl2_prefix/bin/sdl2-config
-      fi
-    fi
-
-    if test -x vs2015/sdl2/linux-host/bin/sdl2-config ; then
-      SDL2_CONFIG=vs2015/sdl2/linux-host/bin/sdl2-config
-      PATH=vs2015/sdl2/linux-host/bin:$PATH
-    fi
-
+    
+    SDL2_CONFIG=$sdl2_prefix/bin/sdl2-config
     AC_PATH_PROG(SDL2_CONFIG, sdl2-config, no)
-    min_sdl2_version=ifelse([$1], ,0.11.0,$1)
-    AC_MSG_CHECKING(for SDL2 - version >= $min_sdl2_version)
-    no_sdl2=""
-    if test "$SDL2_CONFIG" = "no" ; then
-      no_sdl2=yes
-    else
-      SDL2_CFLAGS=`$SDL2_CONFIG $sdl2conf_args --cflags`
-      SDL2_LIBS=`$SDL2_CONFIG $sdl2conf_args --libs`
-      AC_DEFINE(C_SDL2,1)
-    fi
+    SDL2_CFLAGS=`$SDL2_CONFIG $sdl2conf_args --cflags`
+    SDL2_LIBS=`$SDL2_CONFIG $sdl2conf_args --libs`
+    AC_DEFINE(C_SDL2,1)
+  
   fi
 
   AC_SUBST(SDL2_CFLAGS)
