diff --git a/.gitmodules b/.gitmodules
index 91bd312..8bf2137 100644
--- a/.gitmodules
+++ b/.gitmodules
@@ -1,11 +1,11 @@
 [submodule "deps/SDL"]
        path = deps/SDL
-       url = https://bitbucket.org/ecwolf/sdl.git
+       url = ../sdl.git

 [submodule "deps/SDL_mixer"]
        path = deps/SDL_mixer
-       url = https://bitbucket.org/ecwolf/sdl_mixer-for-ecwolf.git
+       url = ../sdl_mixer-for-ecwolf.git

 [submodule "deps/SDL_net"]
        path = deps/SDL_net
-       url = https://bitbucket.org/ecwolf/sdl_net.git
+       url = ../sdl_net.git
