diff --git a/unisocwcn/platform/wcn_boot.c b/unisocwcn/platform/wcn_boot.c
index ba93db7..43092c6 100644
--- a/unisocwcn/platform/wcn_boot.c
+++ b/unisocwcn/platform/wcn_boot.c
@@ -3363,7 +3363,11 @@ int chip_power_on(int subsys)
 	mem_pd_poweroff_deinit();
 #endif
 #ifdef CONFIG_WCN_SDIO
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(7, 1, 0)
+	if (!IS_ERR_OR_NULL(marlin_dev->int_ap))
+#else
 	if (marlin_dev->int_ap > 0)
+#endif
 		sdio_pub_int_poweron(true);
 #endif
 #endif
@@ -3392,7 +3396,11 @@ int chip_power_off(int subsys)
 #endif
 	loopcheck_first_boot_clear();
 #ifdef CONFIG_WCN_SDIO
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(7, 1, 0)
+	if (!IS_ERR_OR_NULL(marlin_dev->int_ap))
+#else
 	if (marlin_dev->int_ap > 0)
+#endif
 		sdio_pub_int_poweron(false);
 #endif
 
@@ -4059,9 +4067,14 @@ static int marlin_probe(struct platform_device *pdev)
 	sprdwcn_bus_preinit();
 #ifndef CONFIG_WCN_PCIE
 #ifdef CONFIG_WCN_SDIO
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(7, 1, 0)
+	if (!IS_ERR_OR_NULL(marlin_dev->int_ap))
+		sdio_pub_int_init(gpiod_to_irq(marlin_dev->int_ap));
+#else
 	if (marlin_dev->int_ap > 0)
 		sdio_pub_int_init(marlin_dev->int_ap);
 #endif
+#endif
 #ifdef CONFIG_MEM_PD
 	mem_pd_init();
 #endif
@@ -4137,7 +4150,11 @@ static int marlin_remove(struct platform_device *pdev)
 	log_dev_exit();
 	proc_fs_exit();
 #ifdef CONFIG_WCN_SDIO
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(7, 1, 0)
+	if (!IS_ERR_OR_NULL(marlin_dev->int_ap))
+#else
 	if (marlin_dev->int_ap > 0)
+#endif
 		sdio_pub_int_deinit();
 #endif
 #ifdef CONFIG_MEM_PD