diff --git a/arch/arm/mach-exynos/exynos-ss.c b/arch/arm/mach-exynos/exynos-ss.c
index 9ede727..ff2df6c 100644
--- a/arch/arm/mach-exynos/exynos-ss.c
+++ b/arch/arm/mach-exynos/exynos-ss.c
@@ -259,7 +259,7 @@ static unsigned int ess_size = 0;
 
 DEFINE_PER_CPU(struct exynos_ss_core_reg *, ess_core_reg);
 DEFINE_PER_CPU(struct exynos_ss_mmu_reg *, ess_mmu_reg);
-DEFINE_PER_CPU(enum ess_cause_emerg_events, ess_cause_emerg);
+//DEFINE_PER_CPU(enum ess_cause_emerg_events, ess_cause_emerg);
 
 static void exynos_ss_save_core(struct exynos_ss_core_reg *core_reg)
 {
@@ -508,7 +508,7 @@ static void exynos_ss_scratch_reg(unsigned int val)
 #if defined(CONFIG_EXYNOS_SNAPSHOT_FORCE_DUMP_MODE) || defined(CONFIG_EXYNOS_SNAPSHOT_PANIC_REBOOT)
 static void exynos_ss_report_cause_emerg(enum ess_cause_emerg_events event)
 {
-	per_cpu(ess_cause_emerg, smp_processor_id()) = event;
+//	per_cpu(ess_cause_emerg, smp_processor_id()) = event;
 }
 #endif
 
