diff --git a/sdl2/Makefile.libretro b/sdl2/Makefile.libretro
index 3174232..6356bb2 100755
--- a/sdl2/Makefile.libretro
+++ b/sdl2/Makefile.libretro
@@ -92,6 +92,23 @@ else ifneq (,$(findstring rpi,$(platform)))
 		CFLAGS += -mcpu=cortex-a72 -fdata-sections -ffunction-sections -Wl,--gc-sections -fmerge-all-constants
 	endif
 
+# ROCKPRO64
+else ifneq (,$(findstring RK3399,$(platform)))
+	TARGET := $(TARGET_NAME)_libretro.so
+	fpic := -fPIC
+	SHARED := -shared -Wl,-version-script=link.T
+	CFLAGS += -fomit-frame-pointer -ffast-math -DARM -mtune=cortex-a72.cortex-a53
+	CPUFLAGS += -march=armv8-a+crc -mfloat-abi=hard -mfpu=neon-fp-armv8 -mtune=cortex-a72.cortex-a53
+
+# ODROIDN2 / VIM3
+else ifneq (,$(findstring odroidn2,$(platform)))
+	TARGET := $(TARGET_NAME)_libretro.so
+	fpic := -fPIC
+	SHARED := -shared -Wl,-version-script=link.T
+	ARCHS = arm64
+	CFLAGS += -fomit-frame-pointer -ffast-math -DARM
+	CPUFLAGS += -march=armv8-a+crc -mfpu=neon-fp-armv8 -mcpu=cortex-a73 -mtune=cortex-a73.cortex-a53
+
 # ODROIDs
 else ifneq (,$(findstring odroid,$(platform)))
 	TARGET := $(TARGET_NAME)_libretro.so
