diff --git a/sdl2/Makefile.libretro b/sdl2/Makefile.libretro
index 3174232..07b58cf 100755
--- a/sdl2/Makefile.libretro
+++ b/sdl2/Makefile.libretro
@@ -78,7 +78,7 @@ else ifneq (,$(findstring rpi,$(platform)))
 
 	# Pi (original)
 	ifneq (,$(findstring rpi1,$(platform)))
-		CFLAGS += -mcpu=cortex-a6j
+		CFLAGS += -mcpu=arm1176jzf-s -mfpu=vfp -mfloat-abi=hard
 	# Pi2
 	else ifneq (,$(findstring rpi2,$(platform)))
 		CFLAGS += -mcpu=cortex-a7 -mfpu=neon-vfpv4 -mfloat-abi=hard
