diff --git a/src/GLideNHQ/TxInternal.h b/src/GLideNHQ/TxInternal.h
index b46acfa..45e7b7c 100644
--- a/src/GLideNHQ/TxInternal.h
+++ b/src/GLideNHQ/TxInternal.h
@@ -44,6 +44,14 @@
 #define GL_RGBA8 0x8058
 #endif
 #define GL_COLOR_INDEX8_EXT  0x80E5
+#elif defined(VC)
+#include <GLES2/gl2.h>
+#include <GLES2/gl2ext.h>
+#define GL_COLOR_INDEX8_EXT  0x80E5
+#elif defined(GLES2)
+#include <GLES2/gl2.h>
+#include <GLES2/gl2ext.h>
+#define GL_COLOR_INDEX8_EXT  0x80E5
 #elif defined(GLES3)
 #include <GLES3/gl3.h>
 #include <GLES3/gl3ext.h>
