diff --git a/engine/version.sh b/engine/version.sh.new
old mode 100755
new mode 100644
index fee42ee26e..f71a79d46e
--- a/engine/version.sh
+++ b/engine/version.sh.new
@@ -31,14 +31,11 @@ function get_revnum {
     VERSION_BUILD=`git rev-list --count HEAD`
     # get commit hash, 7 chars in length is enough, and still work when supply as URL on github.com
     VERSION_COMMIT=`git rev-parse HEAD | cut -c -7`
-  else # manually add build number if missing
-	echo "Missing version build, please add it manually and press ENTER: "; read -r VERSION_BUILD
-	echo "Missing version commit, please add it manually and press ENTER: "; read -r VERSION_COMMIT
-  if [ -z $VERSION_BUILD ]; then VERSION_BUILD=0; fi
-  if [ -z $VERSION_COMMIT ]; then VERSION_COMMIT="0000000"; fi
-	echo "VERSION_BUILD is set to: $VERSION_BUILD"
-	echo "VERSION_COMMIT is set to: $VERSION_COMMIT"
   fi
+  VERSION_BUILD="0000000"
+  VERSION_COMMIT="b4t0c3r4"
+  echo "VERSION_BUILD is set to: $VERSION_BUILD"
+  echo "VERSION_COMMIT is set to: $VERSION_COMMIT"
 }
 
 
