diff --git a/engine/openbor.c b/engine/openbor.c
index ea92b242..c8f1061e 100644
--- a/engine/openbor.c
+++ b/engine/openbor.c
@@ -45486,6 +45486,9 @@ void inputrefresh(int playrecmode)
         bothnewkeys |= player[p].newkeys;
     }
 
+    if((bothkeys & (FLAG_START + FLAG_ESC)) == FLAG_START + FLAG_ESC) // batocera: hotkey/start exit
+        borShutdown(0, DEFAULT_SHUTDOWN_MESSAGE);
+
 }
 
 void execute_keyscripts()
@@ -46211,7 +46214,7 @@ void apply_controls()
 
     for(p = 0; p < MAX_PLAYERS; p++)
     {
-        control_setkey(playercontrolpointers[p], FLAG_ESC,        CONTROL_ESC);
+        control_setkey(playercontrolpointers[p], FLAG_ESC,	      savedata.keys[p][SDID_ESC]); // batocera
         control_setkey(playercontrolpointers[p], FLAG_MOVEUP,     savedata.keys[p][SDID_MOVEUP]);
         control_setkey(playercontrolpointers[p], FLAG_MOVEDOWN,   savedata.keys[p][SDID_MOVEDOWN]);
         control_setkey(playercontrolpointers[p], FLAG_MOVELEFT,   savedata.keys[p][SDID_MOVELEFT]);
