--- a/src/Makefile	2021-01-03 20:23:00.000000000 +0100
+++ b/src/Makefile	2021-01-05 19:39:43.890103318 +0100
@@ -1,26 +1,14 @@
 # Project: SDLPoP
 
-CC = gcc
 RM = rm -f
 
 HFILES = common.h config.h data.h proto.h types.h
 OBJ = main.o data.o seg000.o seg001.o seg002.o seg003.o seg004.o seg005.o seg006.o seg007.o seg008.o seg009.o seqtbl.o replay.o options.o lighting.o screenshot.o menu.o midi.o opl3.o stb_vorbis.o
 BIN = ../prince
 
-OS      := $(shell uname)
-
 CPPFLAGS += -Wall -D_GNU_SOURCE=1
 CFLAGS += -std=gnu99 -O2
 
-ifeq ($(OS),Darwin)
-LIBS := $(shell sdl2-config --libs) -lSDL2_image
-CFLAGS += -I/opt/local/include
-CPPFLAGS += -D_THREAD_SAFE -DOSX
-else
-LIBS := $(shell pkg-config --libs   sdl2 SDL2_image)
-CFLAGS += $(shell pkg-config --cflags sdl2 SDL2_image)
-endif
-
 all: $(BIN)
 
 clean:
