--- a/CMakeLists.txt	2023-07-23 11:26:15.637554089 +0200
+++ b/CMakeLists.txt	2023-07-23 11:27:32.028678103 +0200
@@ -163,7 +163,7 @@
 include_directories(src src/cdogs)
 
 # Tests
-enable_testing()
+#enable_testing()
 
 add_subdirectory(src)
 
--- a/src/CMakeLists.txt	2023-05-11 13:56:55.000000000 +0200
+++ b/src/CMakeLists.txt	2023-07-23 11:32:57.673464000 +0200
@@ -12,7 +12,7 @@
 endif()
 add_subdirectory(json)
 add_subdirectory(proto)
-add_subdirectory(tests)
+#add_subdirectory(tests)
 
 set(CDOGS_SDL_SOURCES
 	ammo_menu.c
