--- a/menu/menu_entry.c	2015-12-17 03:02:48.000000000 +0100
+++ b/menu/menu_entry.c	2015-12-17 16:39:44.833487570 +0100
@@ -402,7 +402,7 @@
             ret = cbs->action_select(entry->path, entry->label, entry->type, i);
          break;
       case MENU_ACTION_SEARCH:
-         menu_input_ctl(MENU_INPUT_CTL_SEARCH_START, NULL);
+         //menu_input_ctl(MENU_INPUT_CTL_SEARCH_START, NULL);
          break;
 
       case MENU_ACTION_SCAN:
