diff --git a/arch/arm64/boot/dts/allwinner/sun50i-h616.dtsi b/arch/arm64/boot/dts/allwinner/sun50i-h616.dtsi
index 06907a77bf..23bd5f576b 100644
--- a/arch/arm64/boot/dts/allwinner/sun50i-h616.dtsi
+++ b/arch/arm64/boot/dts/allwinner/sun50i-h616.dtsi
@@ -1040,7 +1040,7 @@
 		hdmi_audio: hdmi-audio@1 {
 			compatible = "allwinner,sunxi-hdmi-machine";
 			sunxi,cpudai-controller = <&ahub_i2s1>;
-			sunxi,snddaudio-codec = "hdmi-audio-codec.4.auto";
+			sunxi,snddaudio-codec = "hdmi-audio-codec.2.auto";
 			status = "disabled";
 		};
 
