diff --git a/common/Pcsx2Defs.h b/common/Pcsx2Defs.h
index 111db24..4c4ad62 100644
--- a/common/Pcsx2Defs.h
+++ b/common/Pcsx2Defs.h
@@ -119,7 +119,7 @@ static const int __pagesize = PCSX2_PAGESIZE;
 	#define __vectorcall __fastcall
 	#define _inline __inline__ __attribute__((unused))
 	#ifdef NDEBUG
-		#define __forceinline __attribute__((always_inline, unused))
+		#define __forceinline __attribute__((unused))
 	#else
 		#define __forceinline __attribute__((unused))
 	#endif
