diff --git a/Makefile b/Makefile
index 28170c3..04236d3 100644
--- a/Makefile
+++ b/Makefile
@@ -73,6 +73,11 @@ ifneq (,$(findstring unix,$(platform)))
             CFLAGS = -mcpu=cortex-a53
         endif
 
+    # ODROIDN2
+    else ifneq (,$(findstring CortexA73_G12B,$(platform)))
+         CFLAGS += -fomit-frame-pointer -ffast-math
+         CPUFLAGS += -march=armv8-a+crc -mfpu=neon-fp-armv8 -mcpu=cortex-a73 -mtune=cortex-a73.cortex-a53
+
     # ODROIDs
     else ifneq (,$(findstring odroid,$(platform)))
         CFLAGS += -fomit-frame-pointer -ffast-math -DARM -marm -mfloat-abi=hard
