diff --git a/Makefile b/Makefile
index db17991..bbdc67e 100644
--- a/Makefile
+++ b/Makefile
@@ -3,7 +3,6 @@ include config.mk
 NAME=wvkbd
 BIN=${NAME}-${LAYOUT}
 SRC=.
-MAN1 = ${NAME}.1
 
 PKGS = wayland-client xkbcommon pangocairo
 
@@ -21,12 +20,9 @@ HDRS = $(WAYLAND_HEADERS:.xml=-client-protocol.h)
 WAYLAND_SRC = $(HDRS:.h=.c)
 SOURCES = $(WVKBD_SOURCES) $(WAYLAND_SRC)
 
-SCDOC=scdoc
-DOCS = wvkbd.1
-
 OBJECTS = $(SOURCES:.c=.o)
 
-all: ${BIN} ${DOCS}
+all: ${BIN}
 
 config.h:
 	cp config.def.h config.h
@@ -43,18 +39,12 @@ wvkbd-${LAYOUT}: config.h $(OBJECTS) layout.${LAYOUT}.h
 	$(CC) -o wvkbd-${LAYOUT} $(OBJECTS) $(LDFLAGS)
 
 clean:
-	rm -f $(OBJECTS) $(HDRS) $(WAYLAND_SRC) ${BIN} ${DOCS}
+	rm -f $(OBJECTS) $(HDRS) $(WAYLAND_SRC) ${BIN}
 
 format:
 	clang-format -i $(WVKBD_SOURCES) $(WVKBD_HEADERS)
 
-%: %.scd
-	$(SCDOC) < $< > $@
-
 install: all
 	mkdir -p ${DESTDIR}${PREFIX}/bin
 	cp -f ${NAME}-${LAYOUT} ${DESTDIR}${PREFIX}/bin
 	chmod 755 ${DESTDIR}${PREFIX}/bin/${NAME}-${LAYOUT}
-	mkdir -p "${DESTDIR}${MANPREFIX}/man1"
-	sed "s/VERSION/${VERSION}/g" < ${MAN1} > ${DESTDIR}${MANPREFIX}/man1/${MAN1}
-	chmod 644 ${DESTDIR}${MANPREFIX}/man1/${MAN1}
