From 80965d8afd3241017b7dde8dcbd56ab2fd879973 Mon Sep 17 00:00:00 2001
From: Elimar Riesebieter <riesebie@lxtec.de>
Date: Wed, 11 Jan 2017 16:16:14 +0100
Subject: [PATCH] Enabled extended namehints in alsa.conf

---
 src/conf/alsa.conf | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

Index: alsa-lib/src/conf/alsa.conf
===================================================================
--- alsa-lib.orig/src/conf/alsa.conf
+++ alsa-lib/src/conf/alsa.conf
@@ -53,11 +53,11 @@ cards.@hooks [
 #
 
 # show all name hints also for definitions without hint {} section
-defaults.namehint.showall off
+defaults.namehint.showall on
 # show just basic name hints
 defaults.namehint.basic on
 # show extended name hints
-defaults.namehint.extended off
+defaults.namehint.extended on
 #
 defaults.ctl.card 0
 defaults.pcm.card 0
