diff -purN unclutter-10fd337bb77e4e93c3380f630a0555372778a948/Makefile unclutter/Makefile
--- unclutter-10fd337bb77e4e93c3380f630a0555372778a948/Makefile 2018-12-11 18:44:10.000000000 -0200
+++ unclutter/Makefile  2019-04-22 23:40:45.526303870 -0300
@@ -37,7 +37,7 @@ all: clean $(TARGET) mans

 .PHONY: $(TARGET)
 $(TARGET): $(OBJS)
-	$(LD) $(OBJS) $(LDFLAGS) -o "$(TARGET)"
+	$(CC) $(OBJS) $(LDFLAGS) -o "$(TARGET)"

 $(ODIR)/%.o: $(SDIR)/%.c $(INCS)
 	$(CC) $(CPPFLAGS) $(CFLAGS) -o "$@" -c "$<"
