diff --git a/include/solarus/containers/Quadtree.h b/include/solarus/containers/Quadtree.h
index 2f6e352..16abaf9 100644
--- a/include/solarus/containers/Quadtree.h
+++ b/include/solarus/containers/Quadtree.h
@@ -29,6 +29,7 @@
 #include <memory>
 #include <stack>
 #include <vector>
+#include <optional>
 
 namespace Solarus {
 
diff --git a/include/solarus/entities/Entities.h b/include/solarus/entities/Entities.h
index 42c1c86..650f705 100644
--- a/include/solarus/entities/Entities.h
+++ b/include/solarus/entities/Entities.h
@@ -300,7 +300,7 @@ std::set<std::shared_ptr<const T>> Entities::get_entities_by_type() const {
   }
 
   for (const auto& kvp : it->second) {
-    for (const ConstEntityPtr& entity : kvp.second) {
+    for (const ConstEntityPtr entity : kvp.second) {
       result.insert(std::static_pointer_cast<const T>(entity));
     }
   }
diff --git a/src/core/DialogResources.cpp b/src/core/DialogResources.cpp
index 960f90f..e08acf9 100644
--- a/src/core/DialogResources.cpp
+++ b/src/core/DialogResources.cpp
@@ -305,12 +305,12 @@ bool DialogResources::import_from_lua(lua_State* l) {
  */
 bool DialogResources::export_to_lua(std::ostream& out) const {
 
-  for (const auto kvp : dialogs) {
+  for (const auto& kvp : dialogs) {
     const std::string& id = kvp.first;
     const DialogData& dialog = kvp.second;
 
     out << "dialog{\n  id = \"" << escape_string(id) << "\",\n";
-    for (const auto pkvp : dialog.get_properties()) {
+    for (const auto& pkvp : dialog.get_properties()) {
       out << "  " << pkvp.first << " = \"" << pkvp.second << "\",\n";
     }
     const std::string& text = dialog.get_text();
diff --git a/src/core/StringResources.cpp b/src/core/StringResources.cpp
index 59b6828..1550f74 100644
--- a/src/core/StringResources.cpp
+++ b/src/core/StringResources.cpp
@@ -188,7 +188,7 @@ bool StringResources::import_from_lua(lua_State* l) {
  */
 bool StringResources::export_to_lua(std::ostream& out) const {
 
-  for (const auto kvp : strings) {
+  for (const auto& kvp : strings) {
     const std::string& key = kvp.first;
     const std::string& value = kvp.second;
 
diff --git a/src/lua/MainApi.cpp b/src/lua/MainApi.cpp
index 8ee5f3a..6b0526e 100644
--- a/src/lua/MainApi.cpp
+++ b/src/lua/MainApi.cpp
@@ -389,7 +389,7 @@ int LuaContext::main_api_get_resource_ids(lua_State* l) {
     lua_settop(l, 0);
     lua_newtable(l);
     int i = 1;
-    for (const std::pair<std::string, std::string>& kvp : elements) {
+    for (const std::pair<std::string, std::string> kvp : elements) {
       const std::string& id = kvp.first;
       push_string(l, id);
       lua_rawseti(l, 1, i);
