diff --git a/cmake/AddDefines.cmake b/cmake/AddDefines.cmake
index 6330da7..35e80e3 100644
--- a/cmake/AddDefines.cmake
+++ b/cmake/AddDefines.cmake
@@ -8,16 +8,14 @@ endif()
 
 # Base directory where to write files.
 # If blank it will be set depending on the OS (typically the user's home directory).
-set(SOLARUS_BASE_WRITE_DIR "" CACHE STRING "Base directory where to write files, if blank it will be set depending on the OS (typically the user's home directory).")
-if(SOLARUS_BASE_WRITE_DIR)
-  add_definitions(-DSOLARUS_BASE_WRITE_DIR=\"${SOLARUS_BASE_WRITE_DIR}\")
-endif()
+set(SOLARUS_BASE_WRITE_DIR "/userdata/saves" CACHE STRING "Base directory where to write files, if blank it will be set depending on the OS (typically the user's home directory).")
+add_definitions(-DSOLARUS_BASE_WRITE_DIR=\"${SOLARUS_BASE_WRITE_DIR}\")
 
 # Directory where to write savegames and other files saved by quests.
 if(${CMAKE_SYSTEM_NAME} MATCHES "Darwin")
   set(SOLARUS_INITIAL_WRITE_DIR "Solarus")
 else()
-  set(SOLARUS_INITIAL_WRITE_DIR ".solarus")
+  set(SOLARUS_INITIAL_WRITE_DIR "solarus")
 endif()
 set(SOLARUS_WRITE_DIR ${SOLARUS_INITIAL_WRITE_DIR} CACHE STRING "Directory where Solarus savegames are stored, relative to the base write directory.")
 if(SOLARUS_WRITE_DIR)
