diff --git a/yabause/src/libretro/Makefile b/yabause/src/libretro/Makefile
index 961154a..78c2a6a 100644
--- a/yabause/src/libretro/Makefile
+++ b/yabause/src/libretro/Makefile
@@ -154,12 +154,13 @@ else ifneq (,$(findstring rockpro64,$(platform)))
 	fpic := -fPIC
 	SHARED := -shared -Wl,--no-undefined -Wl,--version-script=link.T
 	LDFLAGS += -lpthread
+	FLAGS += -DAARCH64
 	ARCH_IS_LINUX = 1
 	HAVE_SSE = 0
 	FORCE_GLES = 1
-	USE_ARM_DRC = 1
-	DYNAREC_DEVMIYAX = 1
-	FLAGS += -march=armv8-a+crc -mcpu=cortex-a72 -mtune=cortex-a72.cortex-a53 -mfloat-abi=hard -mfpu=neon-vfpv4 -mvectorize-with-neon-quad
+	# USE_AARCH64_DRC = 1
+	# DYNAREC_DEVMIYAX = 1
+	FLAGS += -march=armv8-a+crc -mcpu=cortex-a72 -mtune=cortex-a72.cortex-a53
 
 # Rpi4 (64-bit)
 else ifeq ($(platform), rpi4)
