diff --git a/yabause/src/libretro/Makefile b/yabause/src/libretro/Makefile
index 961154a..94a05c3 100644
--- a/yabause/src/libretro/Makefile
+++ b/yabause/src/libretro/Makefile
@@ -127,6 +127,21 @@ else ifneq (,$(findstring AMLG12B,$(platform)))
 	DYNAREC_DEVMIYAX = 1
 	FLAGS += -march=armv8-a+crc -mtune=cortex-a73.cortex-a53 -mfloat-abi=hard -mfpu=neon-fp-armv8 -mvectorize-with-neon-quad

+# RK3588 (64-bit)
+else ifneq (,$(findstring RK3588,$(platform)))
+	override platform += unix
+	TARGET := $(TARGET_NAME)_libretro.so
+	fpic := -fPIC
+	SHARED := -shared -Wl,--no-undefined -Wl,--version-script=link.T
+	LDFLAGS += -lpthread
+	FLAGS += -DAARCH64
+	ARCH_IS_LINUX = 1
+	HAVE_SSE = 0
+	FORCE_GLES = 1
+	USE_AARCH64_DRC = 1
+	DYNAREC_DEVMIYAX = 1
+	FLAGS += -mcpu=cortex-a76.cortex-a55 -mtune=cortex-a76.cortex-a55
+
 # Rockchip RK3288 / RK3399 - 32-bit userspace
 else ifneq (,$(findstring RK,$(platform)))
 	override platform += unix
@@ -147,19 +162,20 @@ else ifneq (,$(findstring RK,$(platform)))
 	endif
 	FLAGS += -mfloat-abi=hard -mvectorize-with-neon-quad

-# RockPro64
+# RockPro64 (64-bit)
 else ifneq (,$(findstring rockpro64,$(platform)))
 	override platform += unix
 	TARGET := $(TARGET_NAME)_libretro.so
 	fpic := -fPIC
 	SHARED := -shared -Wl,--no-undefined -Wl,--version-script=link.T
 	LDFLAGS += -lpthread
+	FLAGS += -DAARCH64
 	ARCH_IS_LINUX = 1
 	HAVE_SSE = 0
 	FORCE_GLES = 1
-	USE_ARM_DRC = 1
+	USE_AARCH64_DRC = 1
 	DYNAREC_DEVMIYAX = 1
-	FLAGS += -march=armv8-a+crc -mcpu=cortex-a72 -mtune=cortex-a72.cortex-a53 -mfloat-abi=hard -mfpu=neon-vfpv4 -mvectorize-with-neon-quad
+	FLAGS += -mcpu=cortex-a72 -mtune=cortex-a72.cortex-a53

 # Rpi4 (64-bit)
 else ifeq ($(platform), rpi4)
