--- a/yabause/src/sh2_dynarec_devmiyax/dynalib_arm64.s
+++ b/yabause/src/sh2_dynarec_devmiyax/dynalib_arm64.s
@@ -770,11 +770,11 @@ CALL_SETMEM_LONG
 opdesc MOVBL,	28,0,4,0xff,0xff,0xff
 opfunc MOVBL
 mov  w0, #0  // m
-mov  w4, #0  // n
+mov  w22, #0  // n
 ldr  w0, [x19, x0] // w0 = R[m] 
 CALL_GETMEM_BYTE
 sxtb w0, w0
-str  w0, [x19, x4]
+str  w0, [x19, x22]
 
 
 opdesc MOVWL,		28,0,4,0xff,0xff,0xff
