diff --git a/src/action.c b/src/action.c
index 265ff6e..964a1e9 100644
--- a/src/action.c
+++ b/src/action.c
@@ -139,6 +139,7 @@ struct action_arg_list {
 	X(TOGGLE_SHOW_DESKTOP, "ToggleShowDesktop") \
 	X(WARP_CURSOR, "WarpCursor") \
 	X(HIDE_CURSOR, "HideCursor") \
+	X(SHOW_CURSOR, "ShowCursor") \
 	X(DEBUG_TOGGLE_KEY_STATE_INDICATOR, "DebugToggleKeyStateIndicator")
 
 /*
@@ -1594,7 +1595,10 @@ run_action(struct view *view, struct action *action,
 		break;
 	}
 	case ACTION_TYPE_HIDE_CURSOR:
-		cursor_set_visible(&server.seat, false);
+		cursor_set_permanent_hide(&server.seat, true);
+		break;
+	case ACTION_TYPE_SHOW_CURSOR:
+		cursor_set_permanent_hide(&server.seat, false);
 		break;
 	case ACTION_TYPE_DEBUG_TOGGLE_KEY_STATE_INDICATOR:
 		key_state_indicator_toggle();
diff --git a/include/labwc.h b/include/labwc.h
index 87a4219..88392cf 100644
--- a/include/labwc.h
+++ b/include/labwc.h
@@ -35,6 +35,7 @@ struct seat {
 	 */
 	enum lab_cursors server_cursor;
 	bool cursor_visible;
+	bool cursor_force_hidden;
 	struct wlr_cursor *cursor;
 	struct wlr_xcursor_manager *xcursor_manager;
 	struct accumulated_scroll {
diff --git a/src/input/cursor.c b/src/input/cursor.c
index b0f5058..e15a409 100644
--- a/src/input/cursor.c
+++ b/src/input/cursor.c
@@ -394,9 +394,21 @@ cursor_set(struct seat *seat, enum lab_cursors cursor)
 	seat->server_cursor = cursor;
 }
 
+void
+cursor_set_permanent_hide(struct seat *seat, bool enable)
+{
+	seat->cursor_force_hidden = enable;
+	cursor_set_visible(seat, !enable);
+}
+
 void
 cursor_set_visible(struct seat *seat, bool visible)
 {
+	/* Do not show the cursor if it is permanently/force hidden */
+	if (seat->cursor_force_hidden && visible) {
+		return;
+	}
+
 	if (seat->cursor_visible == visible) {
 		return;
 	}
diff --git a/include/input/cursor.h b/include/input/cursor.h
index 3d7aa71..fff2f4c 100644
--- a/include/input/cursor.h
+++ b/include/input/cursor.h
@@ -73,6 +73,8 @@ void cursor_set(struct seat *seat, enum lab_cursors cursor);
 
 void cursor_set_visible(struct seat *seat, bool visible);
 
+void cursor_set_permanent_hide(struct seat *seat, bool enable);
+
 /*
  * Safely store a cursor context to saved_ctx. saved_ctx is cleared when either
  * of its node, surface and view is destroyed.