diff --git a/ui/xui/main.cc b/ui/xui/main.cc
index db84ed658d..96a229cc70 100644
--- a/ui/xui/main.cc
+++ b/ui/xui/main.cc
@@ -520,19 +520,6 @@ void xemu_hud_update(void)
     if (!ImGui::IsWindowFocused(ImGuiFocusedFlags_AnyWindow) &&
         !g_scene_mgr.IsDisplayingScene()) {
 
-        // If the guide button is pressed, wake the ui
-        bool menu_button = false;
-        uint32_t buttons = g_input_mgr.CombinedButtons();
-        if (buttons & CONTROLLER_BUTTON_GUIDE) {
-            menu_button = true;
-        }
-
-        // Allow controllers without a guide button to also work
-        if ((buttons & CONTROLLER_BUTTON_BACK) &&
-            (buttons & CONTROLLER_BUTTON_START)) {
-            menu_button = true;
-        }
-
         if (HotkeyPressed(g_config.input.hotkeys.lightgun_mode)) {
             g_config.chihiro.settings.lightgun_mode =
                 !g_config.chihiro.settings.lightgun_mode;
@@ -542,8 +529,7 @@ void xemu_hud_update(void)
             g_scene_mgr.PushScene(g_main_menu);
         } else if (HotkeyPressed(g_config.input.hotkeys.quick_menu)) {
             g_scene_mgr.PushScene(g_popup_menu);
-        } else if (menu_button ||
-                   (!GunTakesMouse() &&
+        } else if ((!GunTakesMouse() &&
                     ImGui::IsMouseClicked(ImGuiMouseButton_Right) &&
                     !ImGui::IsAnyItemFocused() && !ImGui::IsAnyItemHovered())) {
             g_scene_mgr.PushScene(g_popup_menu);
