diff --git a/configure.ac b/configure.ac
index 79bf05f..0326320 100644
--- a/configure.ac
+++ b/configure.ac
@@ -779,7 +779,7 @@ AC_ARG_ENABLE(
 if test $ac_lib_freetype = auto; then
 	AC_PATH_PROGS(
 		[ac_lib_freetype_config],
-		[${ac_tool_prefix}freetype-config],
+		[freetype-config],
 		[],
 		[${ac_lib_freetype_path}]
 	)
@@ -814,12 +814,12 @@ if test $ac_lib_freetype = auto; then
 elif test $ac_lib_freetype = yes; then
 	AC_PATH_PROGS(
 		[ac_lib_freetype_config],
-		[${ac_tool_prefix}freetype-config],
+		[freetype-config],
 		[],
 		[${ac_lib_freetype_path}]
 	)
 	if test -z $ac_lib_freetype_config ; then
-		AC_MSG_ERROR([the ${ac_tool_prefix}freetype-config tool is missing])
+		AC_MSG_ERROR([the freetype-config tool is missing])
 	else
 		ac_save_CFLAGS="$CFLAGS"
 		ac_save_LIBS="$LIBS"
@@ -994,7 +994,7 @@ if test $ac_lib_sdl = auto; then
 		else
 			AC_PATH_PROGS(
 				[ac_lib_sdl_config],
-				[${ac_tool_prefix}sdl-config],
+				[sdl-config],
 				[],
 				[${ac_lib_sdl_path}]
 			)
@@ -1030,12 +1030,12 @@ elif test $ac_lib_sdl = yes; then
 	fi
 	AC_PATH_PROGS(
 		[ac_lib_sdl_config],
-		[${ac_tool_prefix}sdl-config],
+		[sdl-config],
 		[],
 		[${ac_lib_sdl_path}]
 	)
 	if test -z $ac_lib_sdl_config ; then
-		AC_MSG_ERROR([the ${ac_tool_prefix}sdl-config tool is missing])
+		AC_MSG_ERROR([the sdl-config tool is missing])
 	else
 		ac_save_CFLAGS="$CFLAGS"
 		ac_save_LIBS="$LIBS"
