--- a/shell/linux/Makefile	2021-02-25 15:25:19.863682760 +0100
+++ b/shell/linux/Makefile	2021-02-25 20:55:29.119615303 +0100
@@ -146,7 +146,7 @@
 endif
     ARM64_REC := 1
     ISARM64 := 1
-    CFLAGS += -fno-builtin-sqrtf
+    CFLAGS += -fno-builtin-sqrtf -DEGL_NO_X11
     CXXFLAGS += -fexceptions
 
 # Generic 32 bit ARMhf (a.k.a. ARMv7h)
@@ -155,7 +155,8 @@
         MFLAGS += -mfpu=neon
         ASFLAGS += -mfpu=neon
     endif
-    USE_GLES := 1
+    CFLAGS += -DARM_HARDFP -fsingle-precision-constant -DEGL_NO_X11
+  USE_GLES := 1
 
 # LinCPP
 else ifneq (,$(findstring lincpp,$(platform)))
@@ -168,6 +168,17 @@
     CFLAGS += -D TARGET_NO_JIT
     CXXFLAGS += -fexceptions -std=gnu++11
 
+# Raspberry Pi 4 aarch64 + mesa vulkan
+else ifneq (,$(findstring rpi4-64-mesa,$(platform)))
+    NOT_ARM := 1
+    ARM64_REC := 1
+    ISARM64 := 1
+    USE_GLES := 1
+    USE_SDL := 1
+    MFLAGS += -mcpu=cortex-a72 -mtune=cortex-a72
+    CFLAGS += -fno-builtin-sqrtf  -fsingle-precision-constant -fpermissive -DGLES3 -DEGL_NO_X11 
+    CXXFLAGS += -fexceptions
+
 # Raspberry Pi
 else ifneq (,$(findstring rpi,$(platform)))
     CFLAGS += -DARM_HARDFP -fsingle-precision-constant
@@ -177,6 +188,7 @@
     else ifneq (,$(findstring rpi3,$(platform)))
         MFLAGS += -march=armv8-a+crc -mtune=cortex-a53 -mfpu=neon-fp-armv8 -mfloat-abi=hard -marm
         ASFLAGS += -march=armv8-a+crc -mtune=cortex-a53 -mfpu=neon-fp-armv8 -mfloat-abi=hard -marm
+        CFLAGS += -DEGL_NO_X11
     else ifneq (,$(findstring rpi4,$(platform)))
         MFLAGS += -march=armv8-a+crc -mtune=cortex-a72 -mfpu=neon-fp-armv8 -mfloat-abi=hard -marm
         ASFLAGS += -march=armv8-a+crc -mtune=cortex-a72 -mfpu=neon-fp-armv8 -mfloat-abi=hard -marm
@@ -214,6 +226,53 @@
     ASFLAGS += -march=armv7-a -mfpu=neon -mfloat-abi=softfp
     CFLAGS += -D TARGET_PANDORA  -D WEIRD_SLOWNESS -fsingle-precision-constant
 
+# ODROID c2 aarch64
+else ifneq (,$(findstring odroidc2,$(platform)))
+    NOT_ARM := 1
+    ARM64_REC := 1
+    ISARM64 := 1
+    USE_GLES := 1
+    USE_EVDEV := 1
+    USE_UDEV := 1
+    FOR_LINUX := 1
+    LIBS += -ldrm -lGLESv2
+    MFLAGS += -mcpu=cortex-a53 -mtune=cortex-a53
+    CFLAGS += -fno-builtin-sqrtf -fpermissive -DEGL_NO_X11 -DARM_HARDFP -fsingle-precision-constant
+    CXXFLAGS += -fexceptions
+    undefine USE_X11
+    undefine USE_VULKAN
+
+# ODROID c4 aarch64
+else ifneq (,$(findstring odroidc4,$(platform)))
+    NOT_ARM := 1
+    ARM64_REC := 1
+    ISARM64 := 1
+    USE_GLES := 1
+    USE_EVDEV := 1
+    USE_UDEV := 1
+    FOR_LINUX := 1
+    LIBS += -ldrm -lGLESv2
+    MFLAGS += -mcpu=cortex-a55 -mtune=cortex-a55
+    CFLAGS += -fno-builtin-sqrtf -fpermissive -DGLES3 -DEGL_NO_X11 -fsingle-precision-constant
+    CXXFLAGS += -fexceptions
+    undefine USE_X11
+    undefine USE_VULKAN
+
+# ODROID n2 aarch64
+else ifneq (,$(findstring odroidn2,$(platform)))
+    NOT_ARM := 1
+    ARM64_REC := 1
+    ISARM64 := 1
+    USE_GLES := 1
+    USE_EVDEV := 1
+    USE_UDEV := 1
+    FOR_LINUX := 1
+    LIBS += -ldrm -lGLESv2
+    CFLAGS += -fno-builtin-sqrtf -fpermissive -DGLES3 -DEGL_NO_X11 -fsingle-precision-constant
+    CXXFLAGS += -fexceptions
+    undefine USE_X11
+    undefine USE_VULKAN
+
 # ODROIDs
 else ifneq (,$(findstring odroid,$(platform)))
     MFLAGS += -marm -mfpu=neon -mfloat-abi=hard -funroll-loops
