diff --git a/core/linux-dist/main.cpp b/core/linux-dist/main.cpp
index 4e76c2cf..d4ef4993 100644
--- a/core/linux-dist/main.cpp
+++ b/core/linux-dist/main.cpp
@@ -117,9 +117,8 @@ void common_linux_setup();
 
 // Find the user config directory.
 // The following folders are checked in this order:
-// $HOME/.reicast
+// $HOME/.flycast
 // $HOME/.config/flycast
-// $HOME/.config/reicast
 // If no folder exists, $HOME/.config/flycast is created and used.
 std::string find_user_config_dir()
 {
@@ -127,10 +126,10 @@ std::string find_user_config_dir()
 	std::string xdg_home;
 	if (nowide::getenv("HOME") != NULL)
 	{
-		// Support for the legacy config dir at "$HOME/.reicast"
-		std::string legacy_home = (std::string)nowide::getenv("HOME") + "/.reicast/";
+		// Support for the legacy config dir at "$HOME/.flycast"
+		std::string legacy_home = (std::string)nowide::getenv("HOME") + "/.flycast/";
 		if (flycast::stat(legacy_home.c_str(), &info) == 0 && (info.st_mode & S_IFDIR))
-			// "$HOME/.reicast" already exists, let's use it!
+			// "$HOME/.flycast" already exists, let's use it!
 			return legacy_home;
 
 		/* If $XDG_CONFIG_HOME is not set, we're supposed to use "$HOME/.config" instead.
@@ -149,10 +148,6 @@ std::string find_user_config_dir()
 		if (flycast::stat(fullpath.c_str(), &info) == 0 && (info.st_mode & S_IFDIR))
 			// Found .config/flycast
 			return fullpath;
-		fullpath = xdg_home + "/reicast/";
-		if (flycast::stat(fullpath.c_str(), &info) == 0 && (info.st_mode & S_IFDIR))
-			// Found .config/reicast
-			return fullpath;
 
 		// Create .config/flycast
 		fullpath = xdg_home + "/flycast/";
@@ -167,9 +162,8 @@ std::string find_user_config_dir()
 
 // Find the user data directory.
 // The following folders are checked in this order:
-// $HOME/.reicast/data
+// $HOME/.flycast/data
 // $HOME/.local/share/flycast
-// $HOME/.local/share/reicast
 // If no folder exists, $HOME/.local/share/flycast is created and used.
 std::string find_user_data_dir()
 {
@@ -177,10 +171,10 @@ std::string find_user_data_dir()
 	std::string xdg_home;
 	if (nowide::getenv("HOME") != NULL)
 	{
-		// Support for the legacy config dir at "$HOME/.reicast/data"
-		std::string legacy_data = (std::string)nowide::getenv("HOME") + "/.reicast/data/";
+		// Support for the legacy config dir at "$HOME/.flycast/data"
+		std::string legacy_data = (std::string)nowide::getenv("HOME") + "/.flycast/data/";
 		if (flycast::stat(legacy_data.c_str(), &info) == 0 && (info.st_mode & S_IFDIR))
-			// "$HOME/.reicast/data" already exists, let's use it!
+			// "$HOME/.flycast/data" already exists, let's use it!
 			return legacy_data;
 
 		/* If $XDG_DATA_HOME is not set, we're supposed to use "$HOME/.local/share" instead.
@@ -199,10 +193,6 @@ std::string find_user_data_dir()
 		if (flycast::stat(fullpath.c_str(), &info) == 0 && (info.st_mode & S_IFDIR))
 			// Found .local/share/flycast
 			return fullpath;
-		fullpath = xdg_home + "/reicast/";
-		if (flycast::stat(fullpath.c_str(), &info) == 0 && (info.st_mode & S_IFDIR))
-			// Found .local/share/reicast
-			return fullpath;
 
 		// Create .local/share/flycast
 		fullpath = xdg_home + "/flycast/";
@@ -217,12 +207,10 @@ std::string find_user_data_dir()
 
 // Find a file in the user and system config directories.
 // The following folders are checked in this order:
-// $HOME/.reicast
+// $HOME/.flycast
 // $HOME/.config/flycast
-// $HOME/.config/reicast
 // if XDG_CONFIG_DIRS is defined:
 //   <$XDG_CONFIG_DIRS>/flycast
-//   <$XDG_CONFIG_DIRS>/reicast
 // else
 //   /etc/flycast/
 //   /etc/xdg/flycast/
@@ -234,8 +222,8 @@ std::vector<std::string> find_system_config_dirs()
 	std::string xdg_home;
 	if (nowide::getenv("HOME") != NULL)
 	{
-		// Support for the legacy config dir at "$HOME/.reicast"
-		dirs.push_back((std::string)nowide::getenv("HOME") + "/.reicast/");
+		// Support for the legacy config dir at "$HOME/.flycast"
+		dirs.push_back((std::string)nowide::getenv("HOME") + "/.flycast/");
 		xdg_home = (std::string)nowide::getenv("HOME") + "/.config";
 	}
 	if (nowide::getenv("XDG_CONFIG_HOME") != NULL)
@@ -245,7 +233,6 @@ std::vector<std::string> find_system_config_dirs()
 	{
 		// XDG config locations
 		dirs.push_back(xdg_home + "/flycast/");
-		dirs.push_back(xdg_home + "/reicast/");
 	}
 
 	if (nowide::getenv("XDG_CONFIG_DIRS") != NULL)
@@ -296,8 +277,8 @@ std::vector<std::string> find_system_data_dirs()
 	std::string xdg_home;
 	if (nowide::getenv("HOME") != NULL)
 	{
-		// Support for the legacy data dir at "$HOME/.reicast/data"
-		dirs.push_back((std::string)nowide::getenv("HOME") + "/.reicast/data/");
+		// Support for the legacy data dir at "$HOME/.flycast/data"
+		dirs.push_back((std::string)nowide::getenv("HOME") + "/.flycast/data/");
 		xdg_home = (std::string)nowide::getenv("HOME") + "/.local/share";
 	}
 	if (nowide::getenv("XDG_DATA_HOME") != NULL)
@@ -307,8 +288,7 @@ std::vector<std::string> find_system_data_dirs()
 	{
 		// XDG data locations
 		dirs.push_back(xdg_home + "/flycast/");
-		dirs.push_back(xdg_home + "/reicast/");
-		dirs.push_back(xdg_home + "/reicast/data/");
+		dirs.push_back(xdg_home + "/flycast/data/");
 	}
 
 	if (nowide::getenv("XDG_DATA_DIRS") != NULL)
@@ -368,7 +344,7 @@ int main(int argc, char* argv[])
 	settings.profile.run_counts=0;
 
 	if (reicast_init(argc, argv))
-		die("Reicast initialization failed\n");
+		die("Flycast initialization failed\n");
 
 	mainui_loop();
 
