diff --git a/libretro/Makefile b/libretro/Makefile
index eaab210..d264dab 100644
--- a/libretro/Makefile
+++ b/libretro/Makefile
@@ -207,6 +207,13 @@ else ifeq ($(platform), rpi3)
 	PLATFORM_DEFINES += -marm -mcpu=cortex-a53 -mfpu=neon-fp-armv8 -mfloat-abi=hard -ffast-math
 	PLATFORM_DEFINES += -DARM
 
+# Raspberry Pi 3 (64-bit)
+else ifeq ($(platform), rpi3_64)
+	TARGET := $(TARGET_NAME)_libretro.so
+	fpic := -fPIC
+	SHARED := -shared -Wl,-version-script=link.T -Wl,-no-undefined
+	PLATFORM_DEFINES += -mcpu=cortex-a53 -mtune=cortex-a53 -ffast-math
+
 # Raspberry Pi 4 (64-bit)
 else ifeq ($(platform), rpi4_64)
 	TARGET := $(TARGET_NAME)_libretro.so
