diff --git a/libretro/Makefile b/libretro/Makefile
index 7cfd5b09dd..807a731408 100644
--- a/libretro/Makefile
+++ b/libretro/Makefile
@@ -231,6 +231,13 @@ else ifeq ($(platform), rpi4_64)
 	SHARED := -shared -Wl,-version-script=link.T -Wl,-no-undefined
 	PLATFORM_DEFINES += -mcpu=cortex-a72 -mtune=cortex-a72 -ffast-math
 
+# Raspberry Pi 5 (64-bit)
+else ifeq ($(platform), rpi5_64)
+	TARGET := $(TARGET_NAME)_libretro.so
+	fpic := -fPIC
+	SHARED := -shared -Wl,-version-script=link.T -Wl,-no-undefined
+	PLATFORM_DEFINES += -mcpu=cortex-a76 -mtune=cortex-a76 -ffast-math
+
 # Lightweight PS3 Homebrew SDK
 else ifneq (,$(filter $(platform), ps3 psl1ght))
 	TARGET := $(TARGET_NAME)_libretro_$(platform).a
