diff --git a/kernel-open/nvidia/nv.c b/kernel-open/nvidia/nv.c
index 83705a0513..f2d021a9bd 100644
--- a/kernel-open/nvidia/nv.c
+++ b/kernel-open/nvidia/nv.c
@@ -127,7 +127,7 @@ MODULE_ALIAS_CHARDEV_MAJOR(NV_MAJOR_DEVICE_NUMBER);
  * DMA_BUF namespace is added by commit id 16b0314aa746
  * ("dma-buf: move dma-buf symbols into the DMA_BUF module namespace") in 5.16
  */
-MODULE_IMPORT_NS(DMA_BUF);
+MODULE_INFO(import_ns, "DMA_BUF");
 #endif  // defined(MODULE_IMPORT_NS)
 
 const NvBool nv_is_rm_firmware_supported_os = NV_TRUE;
