diff --git a/CMakeLists.txt b/CMakeLists.txt.new
index c7c6558654..af59cd86c2 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt.new
@@ -25,10 +25,10 @@ list(APPEND CMAKE_MODULE_PATH ${TINYCMMC_MODULE_PATH})
 option(WARNINGS "Switch on extra warnings" OFF)
 option(WERROR "Turn warnings into errors" OFF)
 option(BUILD_TESTS "Build test cases" OFF)
-option(BUILD_SDL_JSTEST "Build sdl-jstest" ON)
+option(BUILD_SDL_JSTEST "Build sdl-jstest" OFF)
 option(BUILD_SDL2_JSTEST "Build sdl2-jstest" ON)
 
-include(GetProjectVersion)
+#include(GetProjectVersion)
 include(GNUInstallDirs)
 
 add_definitions(-DSDL_JSTEST_VERSION="${PROJECT_VERSION}")
@@ -105,7 +105,7 @@ if(BUILD_SDL2_JSTEST)
   file(COPY sdl2-jstest.1
     DESTINATION ${CMAKE_CURRENT_BINARY_DIR})
 
-  file(COPY SDL_GameControllerDB/gamecontrollerdb.txt
+  file(COPY gamecontrollerdb.txt
     DESTINATION ${CMAKE_CURRENT_BINARY_DIR})
 
   install(FILES ${CMAKE_CURRENT_BINARY_DIR}/gamecontrollerdb.txt
