diff --git a/sdl2-jstest.c b/sdl2-jstest.c
index 5c76a8c..0a1eb2f 100644
--- a/sdl2-jstest.c
+++ b/sdl2-jstest.c
@@ -150,6 +150,26 @@ void list_joysticks()
   }
 }
 
+void infopath()
+{
+  int num_joysticks = SDL_NumJoysticks();
+
+  for(int joy_idx = 0; joy_idx < num_joysticks; ++joy_idx) {
+    SDL_Joystick* joy = SDL_JoystickOpen(joy_idx);
+    if (!joy) {
+        fprintf(stderr, "Unable to open joystick %d\n", joy_idx);
+    } else {
+      SDL_GameController* gamepad = SDL_GameControllerOpen(joy_idx);
+      printf("%s %i\n", SDL_JoystickDevicePathById(joy_idx), joy_idx);
+      if (gamepad)
+        {
+          SDL_GameControllerClose(gamepad);
+        }
+      SDL_JoystickClose(joy);
+    }
+  }
+}
+
 void test_joystick(int joy_idx)
 {
   SDL_Joystick* joy = SDL_JoystickOpen(joy_idx);
@@ -579,7 +599,7 @@ int main(int argc, char** argv)
   SDL_SetHint(SDL_HINT_JOYSTICK_ALLOW_BACKGROUND_EVENTS, "1");
 
   // FIXME: We don't need video, but without it SDL will fail to work in SDL_WaitEvent()
-  if(SDL_Init(SDL_INIT_TIMER | SDL_INIT_VIDEO | SDL_INIT_JOYSTICK | SDL_INIT_GAMECONTROLLER | SDL_INIT_HAPTIC) < 0)
+  if(SDL_Init(SDL_INIT_JOYSTICK) < 0)
   {
     fprintf(stderr, "Unable to init SDL: %s\n", SDL_GetError());
     exit(1);
@@ -603,6 +623,10 @@ int main(int argc, char** argv)
     {
       list_joysticks();
     }
+    else if (argc == 2 && (strcmp(argv[1], "--infopath") == 0))
+      {
+	infopath();
+      }
     else if (argc == 3 && (strcmp(argv[1], "--gamecontroller") == 0 ||
                            strcmp(argv[1], "-g") == 0))
     {
