diff --git a/Makefile.libretro b/Makefile.libretro
index 1e07d50..a664b4a 100644
--- a/Makefile.libretro
+++ b/Makefile.libretro
@@ -54,6 +54,15 @@ ifeq ($(platform), unix)
 	CFLAGS += -DFAMEC_NO_GOTOS
 	use_sh2drc = 1
 
+# AARCH64 generic
+else ifeq ($(platform), aarch64)
+	TARGET := $(TARGET_NAME)_libretro.so
+	ARCH = aarch64
+        fpic := -fPIC
+	SHARED := -shared
+	DONT_COMPILE_IN_ZLIB = 1
+	CFLAGS += -DFAMEC_NO_GOTOS
+	
 # Portable Linux
 else ifeq ($(platform), linux-portable)
 	EXT ?= so
