--- a/Makefile.libretro		2020-06-15 18:38:21.640704933 +0200
+++ b/Makefile.libretro		2020-06-15 18:40:01.094048398 +0200
@@ -54,6 +54,24 @@
 	CFLAGS += -DFAMEC_NO_GOTOS
 	use_sh2drc = 1
 
+# x86/x86_64 generic
+else ifneq (,$(findstring x86,$(platform)))
+	TARGET := $(TARGET_NAME)_libretro.so
+	ARCH := 86
+        fpic := -fPIC
+	SHARED := -shared
+	DONT_COMPILE_IN_ZLIB = 1
+	CFLAGS += -DFAMEC_NO_GOTOS
+
+# AARCH64 generic
+else ifeq ($(platform), aarch64)
+	TARGET := $(TARGET_NAME)_libretro.so
+	ARCH = aarch64
+        fpic := -fPIC
+	SHARED := -shared
+	DONT_COMPILE_IN_ZLIB = 1
+	CFLAGS += -DFAMEC_NO_GOTOS
+	
 # Portable Linux
 else ifeq ($(platform), linux-portable)
 	EXT ?= so
