diff -Naur openjazz-8108082394a2ca1007b2444b4099d9f18afcb530.orig/src/setup.cpp openjazz-8108082394a2ca1007b2444b4099d9f18afcb530/src/setup.cpp
--- openjazz-8108082394a2ca1007b2444b4099d9f18afcb530.orig/src/setup.cpp	2021-08-28 21:23:57.000000000 +0100
+++ openjazz-8108082394a2ca1007b2444b4099d9f18afcb530/src/setup.cpp	2021-08-30 12:46:18.821708517 +0100
@@ -44,7 +44,7 @@
 #elif __vita__
     #define CONFIG_FILE "ux0:data/jazz/openjazz.cfg"
 #else
-    #define CONFIG_FILE "openjazz.cfg"
+    #define CONFIG_FILE "/userdata/system/configs/openjazz.cfg"
 #endif
 
 