diff -Naur openjazz-8108082394a2ca1007b2444b4099d9f18afcb530.orig/src/main.cpp openjazz-8108082394a2ca1007b2444b4099d9f18afcb530/src/main.cpp
--- openjazz-8108082394a2ca1007b2444b4099d9f18afcb530.orig/src/main.cpp	2021-08-28 21:23:57.000000000 +0100
+++ openjazz-8108082394a2ca1007b2444b4099d9f18afcb530/src/main.cpp	2021-08-30 12:48:02.126669468 +0100
@@ -86,28 +86,29 @@
  */
 void startUp (int argc, char *argv[]) {
 
+    SDL_ShowCursor(SDL_DISABLE);
 	File* file;
 	unsigned char* pixels = NULL;
 	int count;
 	int screenW = DEFAULT_SCREEN_WIDTH;
 	int screenH = DEFAULT_SCREEN_HEIGHT;
 	int scaleFactor = 1;
-#ifdef FULLSCREEN_ONLY
+//#ifdef FULLSCREEN_ONLY
 	bool fullscreen = true;
-#else
-	bool fullscreen = false;
-#endif
+//#else
+//	bool fullscreen = false;
+//#endif
 
 
 	// Determine paths
 
 	// Use hard-coded paths, if available
 
-#ifdef DATAPATH
-	firstPath = new Path(NULL, createString(DATAPATH));
-#else
-	firstPath = NULL;
-#endif
+//#ifdef DATAPATH
+//	firstPath = new Path(NULL, createString(DATAPATH));
+//#else
+	firstPath = new Path(NULL, createString("/userdata/roms/openjazz"));
+//#endif
 
 #ifdef __HAIKU__
 	dev_t volume = dev_for_path("/boot");
