diff --git a/usbmount b/usbmount
index c09a865..7d89dbf 100755
--- a/usbmount
+++ b/usbmount
@@ -123,7 +123,7 @@ if [ "$1" = add ]; then
 
 		# Mount the filesystem.
 		log info "executing command: mount -t$fstype ${options:+-o$options} $DEVNAME $mountpoint"
-		mount "-t$fstype" "${options:+-o$options}" "$DEVNAME" "$mountpoint"
+		batocera-mount "$fstype" 0 "${DEVNAME}" "${mountpoint}"
 
 		# Determine vendor and model.
 		vendor=
