--- a/drivers/input/joystick/xpad.c	2015-04-14 01:05:54.000000000 +0200
+++ b/drivers/input/joystick/xpad.c	2015-05-17 19:34:00.974403905 +0200
@@ -1190,7 +1190,8 @@
 					 usb_sndintpipe(udev,
 							ep_irq_in->bEndpointAddress),
 					 xpad->bdata, XPAD_PKT_LEN,
-					 xpad_bulk_out, xpad, 0);
+					 xpad_bulk_out, xpad, 
+                                         ep_irq_in->bInterval);
 		}
 
 		/*
