diff --git a/Makefile.libretro b/Makefile.libretro
index 0cd567cf17..37754c232c 100644
--- a/Makefile.libretro
+++ b/Makefile.libretro
@@ -256,6 +256,14 @@ else ifneq (,$(findstring rpi4,$(platform)))
 	CFLAGS += -mcpu=cortex-a72 -mtune=cortex-a72 -funsafe-math-optimizations
 	CFLAGS += -fomit-frame-pointer -fstrict-aliasing -ffast-math
 
+# Raspberry Pi 5 (64-bit)
+else ifneq (,$(findstring rpi5,$(platform)))
+	TARGET := $(TARGET_NAME)_libretro.so
+	fpic := -fPIC
+	SHARED := -shared -Wl,--version-script=src/drivers/libretro/link.T -Wl,-no-undefined
+	CFLAGS += -mcpu=cortex-a76 -mtune=cortex-a76 -funsafe-math-optimizations
+	CFLAGS += -fomit-frame-pointer -fstrict-aliasing -ffast-math
+
 # Lightweight PS3 Homebrew SDK
 else ifneq (,$(filter $(platform), ps3 psl1ght))
     ENDIANNESS_DEFINES =  -DMSB_FIRST
