diff --git a/meson.build b/meson.build
index f1be50c777..79bcd7e3df 100644
--- a/meson.build
+++ b/meson.build
@@ -610,7 +610,7 @@ gir_init_section = [ '--add-init-section=extern void gst_init(gint*,gchar**);' +
     'g_setenv("GST_PLUGIN_PATH_1_0", "", TRUE);' + \
     'g_setenv("GST_PLUGIN_SYSTEM_PATH_1_0", "", TRUE);' + \
     'g_setenv("GST_TRACERS", "", TRUE);' + \
-    'gst_init(NULL,NULL);', '--quiet']
+    'gst_init(NULL,NULL);', '--quiet', '--extra-library=gstreamer-1.0' ]
 
 gst_c_args = ['-DHAVE_CONFIG_H']
 