diff --git a/common.h b/common.h
index 96e921d..ebd4f3d 100755
--- a/common.h
+++ b/common.h
@@ -34,12 +34,12 @@ struct uart_test {
 	unsigned int random_enable;				/* determines if data packets sent are fixed or "random" size */
 };
 
-struct termios oldtio;
+extern struct termios oldtio;
 
-int fd1,fd2;
-struct uart_test ut;
-char tx_rx;
-int read_flag;
+extern int fd1,fd2;
+extern struct uart_test ut;
+extern char tx_rx;
+extern int read_flag;
 
 int writeport(int *fd, unsigned char *chars,int len);
 
diff --git a/ts_uart.c b/ts_uart.c
index 6b7d068..ecf6841 100644
--- a/ts_uart.c
+++ b/ts_uart.c
@@ -1,6 +1,14 @@
 #include "common.h"
 #include "uinput-ctl.h"
 
+struct termios oldtio;
+int fd1,fd2;
+struct uart_test ut;
+char tx_rx;
+int read_flag;
+
+struct joypad joy;
+
 /*
 *	v1.0 start to use version to manageme the code
 *
diff --git a/uinput-ctl.h b/uinput-ctl.h
index fffab3f..daf9b4e 100755
--- a/uinput-ctl.h
+++ b/uinput-ctl.h
@@ -32,7 +32,7 @@ struct joypad {
 	int deadzone;
 	struct bt_adc adcs[2];
 };
-struct joypad  joy;
+extern struct joypad joy;
 
 void joystic_init(void);
 void joystick_move(int xVal, int yVal);
