diff --git a/qb/qb.comp.sh b/qb/qb.comp.sh
index 1477154..eaec4be 100644
--- a/qb/qb.comp.sh
+++ b/qb/qb.comp.sh
@@ -14,17 +14,12 @@ cc_works=0
 if [ "$CC" ]; then
 	"$CC" -o "$TEMP_EXE" "$TEMP_C" >/dev/null 2>&1 && cc_works=1
 else
-	for cc in gcc cc clang; do
-		CC="$(exists "${CROSS_COMPILE}${cc}")" || CC=""
-		if [ "$CC" ]; then
-			"$CC" -o "$TEMP_EXE" "$TEMP_C" >/dev/null 2>&1 && {
-				cc_works=1; break
-			}
-		fi
+	for CC in $(exists ${CROSS_COMPILE}gcc ${CROSS_COMPILE}cc ${CROSS_COMPILE}clang) ''; do
+		"$CC" -o "$TEMP_EXE" "$TEMP_C" >/dev/null 2>&1 && cc_works=1 && break
 	done
 fi
 
-rm -f -- "$TEMP_C" "$TEMP_EXE"
+rm -f "$TEMP_C" "$TEMP_EXE"
 
 cc_status='does not work'
 if [ "$cc_works" = '1' ]; then
@@ -49,17 +44,12 @@ cxx_works=0
 if [ "$CXX" ]; then
 	"$CXX" -o "$TEMP_EXE" "$TEMP_CXX" >/dev/null 2>&1 && cxx_works=1
 else
-	for cxx in g++ c++ clang++; do
-		CXX="$(exists "${CROSS_COMPILE}${cxx}")" || CXX=""
-		if [ "$CXX" ]; then
-			"$CXX" -o "$TEMP_EXE" "$TEMP_CXX" >/dev/null 2>&1 && {
-				cxx_works=1; break
-			}
-		fi
+	for CXX in $(exists ${CROSS_COMPILE}g++ ${CROSS_COMPILE}c++ ${CROSS_COMPILE}clang++) ''; do
+		"$CXX" -o "$TEMP_EXE" "$TEMP_CXX" >/dev/null 2>&1 && cxx_works=1 && break
 	done
 fi
 
-rm -f -- "$TEMP_CXX" "$TEMP_EXE"
+rm -f "$TEMP_CXX" "$TEMP_EXE"
 
 cxx_status='does not work'
 if [ "$cxx_works" = '1' ]; then
@@ -77,22 +67,23 @@ fi
 if [ "$OS" = "Win32" ]; then
 	echobuf="Checking for windres"
 	if [ -z "$WINDRES" ]; then
-		WINDRES="$(exists "${CROSS_COMPILE}windres")" || WINDRES=""
-		[ -z "$WINDRES" ] && die 1 "$echobuf ... Not found. Exiting."
+		WINDRES=$(exists ${CROSS_COMPILE}windres)
+		[ "$WINDRES" ] || die 1 "$echobuf ... Not found. Exiting."
 	fi
 	echo "$echobuf ... $WINDRES"
 fi
 
-if [ -z "$PKG_CONF_PATH" ]; then
+[ -n "$PKG_CONF_PATH" ] || {
 	PKG_CONF_PATH="none"
-	for pkgconf in pkgconf pkg-config; do
-		PKGCONF="$(exists "${CROSS_COMPILE}${pkgconf}")" || PKGCONF=""
-		[ "$PKGCONF" ] && {
-			PKG_CONF_PATH="$PKGCONF"
-			break
+
+	for p in $(exists "${CROSS_COMPILE}pkg-config") ''; do
+		[ -n "$p" ] && {
+			PKG_CONF_PATH=$p;
+			break;
 		}
 	done
-fi
+
+}
 
 echo "Checking for pkg-config ... $PKG_CONF_PATH"
 
