diff --git a/qb/qb.comp.sh b/qb/qb.comp.sh
index 902afac..c99d126 100644
--- a/qb/qb.comp.sh
+++ b/qb/qb.comp.sh
@@ -107,16 +107,15 @@ if [ "$HAVE_QT" != "no" ]; then
 	fi
 fi
 
-if [ -z "$PKG_CONF_PATH" ]; then
+[ -n "$PKG_CONF_PATH" ] || {
 	PKG_CONF_PATH="none"
-	for pkgconf in pkgconf pkg-config; do
-		PKGCONF="$(exists "${CROSS_COMPILE}${pkgconf}")" || PKGCONF=""
-		[ "$PKGCONF" ] && {
-			PKG_CONF_PATH="$PKGCONF"
-			break
-		}
+	for p in $(exists "${CROSS_COMPILE}pkg-config") ''; do
+		[ -n "$p" ] && {
+			PKG_CONF_PATH=$p;
+			break;
+ 		}
 	done
-fi
+}
 
 echo "Checking for pkg-config ... $PKG_CONF_PATH"
 
