--- ./include/drm/drm_fourcc.h	2017-09-04 10:45:05.812784940 -0700
+++ ./include/drm/drm_fourcc.h	2017-09-04 10:45:27.831516420 -0700
@@ -116,6 +116,13 @@
 #define DRM_FORMAT_NV24		fourcc_code('N', 'V', '2', '4') /* non-subsampled Cr:Cb plane */
 #define DRM_FORMAT_NV42		fourcc_code('N', 'V', '4', '2') /* non-subsampled Cb:Cr plane */
 
+#define DRM_FORMAT_NV12_10      fourcc_code('N', 'A', '1', '2') /* 2x2 subsampled Cr:Cb plane */
+#define DRM_FORMAT_NV21_10      fourcc_code('N', 'A', '2', '1') /* 2x2 subsampled Cb:Cr plane */
+#define DRM_FORMAT_NV16_10      fourcc_code('N', 'A', '1', '6') /* 2x1 subsampled Cr:Cb plane */
+#define DRM_FORMAT_NV61_10      fourcc_code('N', 'A', '6', '1') /* 2x1 subsampled Cb:Cr plane */
+#define DRM_FORMAT_NV24_10      fourcc_code('N', 'A', '2', '4') /* non-subsampled Cr:Cb plane */
+#define DRM_FORMAT_NV42_10      fourcc_code('N', 'A', '4', '2') /* non-subsampled Cb:Cr plane */
+
 /*
  * 3 plane YCbCr
  * index 0: Y plane, [7:0] Y
