--- a/yabause/src/libretro/Makefile	2021-06-07 16:02:37.493918789 +0200
+++ b/yabause/src/libretro/Makefile	2021-06-07 16:03:28.467250253 +0200
@@ -257,6 +257,20 @@
         LDFLAGS += -static-libgcc -static-libstdc++
 #######################################
 
+# ODROID C4
+else ifneq (,$(findstring c4,$(platform)))
+        override platform += unix
+        TARGET := $(TARGET_NAME)_libretro.so
+        fpic := -fPIC
+        SHARED := -shared -Wl,--no-undefined -Wl,--version-script=link.T
+        LDFLAGS += -lpthread
+        ARCH_IS_LINUX = 1
+        HAVE_SSE = 0
+        FORCE_GLES = 1
+        USE_AARCH64_DRC = 1
+        DYNAREC_DEVMIYAX = 1
+        FLAGS += -DAARCH64 -mcpu=cortex-a55 -mtune=cortex-a55
+
 # QNX
 else ifneq (,$(findstring qnx,$(platform)))
 	TARGET := $(TARGET_NAME)_libretro_$(platform).so
