diff --git a/runloop.c b/runloop.c
index 16cf8fe..7add5be 100644
--- a/runloop.c
+++ b/runloop.c
@@ -2534,17 +2534,17 @@ bool runloop_environment_cb(unsigned cmd, void *data)
 
       case RETRO_ENVIRONMENT_GET_PERF_INTERFACE:
       {
-         struct retro_perf_callback *cb = (struct retro_perf_callback*)data;
-
-         RARCH_LOG("[Environ]: GET_PERF_INTERFACE.\n");
-         cb->get_time_usec    = cpu_features_get_time_usec;
-         cb->get_cpu_features = cpu_features_get;
-         cb->get_perf_counter = cpu_features_get_perf_counter;
-
-         cb->perf_register    = runloop_performance_counter_register;
-         cb->perf_start       = core_performance_counter_start;
-         cb->perf_stop        = core_performance_counter_stop;
-         cb->perf_log         = runloop_perf_log;
+         //struct retro_perf_callback *cb = (struct retro_perf_callback*)data;
+	 //
+         //RARCH_LOG("[Environ]: GET_PERF_INTERFACE.\n");
+         //cb->get_time_usec    = cpu_features_get_time_usec;
+         //cb->get_cpu_features = cpu_features_get;
+         //cb->get_perf_counter = cpu_features_get_perf_counter;
+	 //
+         //cb->perf_register    = runloop_performance_counter_register;
+         //cb->perf_start       = core_performance_counter_start;
+         //cb->perf_stop        = core_performance_counter_stop;
+         //cb->perf_log         = runloop_perf_log;
          break;
       }
 
