diff --git a/Makefile b/Makefile
index deaaa9a..9854940 100644
--- a/Makefile
+++ b/Makefile
@@ -315,7 +315,7 @@ else ifneq (,$(findstring rpi,$(platform)))
    SHARED := -shared -Wl,--no-undefined
    fpic = -fPIC
    LDFLAGS += $(SHARED)
-   LIBS += -lstdc++ -lpthread
+   LIBS += -lstdc++ -lpthread -lm
    CCOMFLAGS += -fomit-frame-pointer -ffast-math -fsigned-char
    ARM_ENABLED = 1
    X86_SH2DRC = 0
