diff --git a/Makefile b/Makefile
index d21dd56..d447f73 100644
--- a/Makefile
+++ b/Makefile
@@ -25,21 +25,13 @@ ifeq ($(HAVE_CLANG),1)
 	CXX      = ${CC_PREFIX}clang++
 	CC       = ${CC_PREFIX}clang
 	SHARED   := -fuse-ld=lld
-else
-	CXX      = ${CC_PREFIX}g++
-	CC       = ${CC_PREFIX}gcc
-	SHARED   :=
 endif
-CC_AS    = ${CC_PREFIX}as
 
 MFLAGS   := 
 ASFLAGS  := 
-LDFLAGS  :=
 LDFLAGS_END :=
 INCFLAGS :=
 LIBS     :=
-CFLAGS   := 
-CXXFLAGS :=
 
 GIT_VERSION := " $(shell git rev-parse --short HEAD || echo unknown)"
 ifneq ($(GIT_VERSION)," unknown")
@@ -857,7 +849,7 @@ LIBS     += -lm
 PREFIX        ?= /usr/local
 
 ifneq (,$(findstring arm, $(ARCH)))
-	CC_AS    = ${CC_PREFIX}gcc #The ngen_arm.S must be compiled with gcc, not as
+	CC_AS    = ${CC_PREFIX}${CC} #The ngen_arm.S must be compiled with gcc, not as
 	ASFLAGS  += $(CFLAGS)
 endif
 
