--- a/src/GLideNHQ/CMakeLists.txt	2015-10-21 17:57:48.613858250 +0200
+++ b/src/GLideNHQ/CMakeLists.txt	2015-10-20 21:08:10.000000000 +0200
@@ -57,13 +57,28 @@
 	)
 endif( CMAKE_BUILD_TYPE STREQUAL "Debug")
 
-find_package(OpenGL REQUIRED)
-include_directories(${OpenGL_INCLUDE_DIRS})
-link_directories(${OpenGL_LIBRARY_DIRS})
-add_definitions(${OpenGL_DEFINITIONS})
-if(NOT OPENGL_FOUND)
-	message(ERROR " OPENGL not found!")
-endif(NOT OPENGL_FOUND)
+if(GLES2)
+  add_definitions(
+    -DGLES2
+   )
+   list(APPEND GLideN64_SOURCES 
+    GLES2/GLSLCombiner_gles2.cpp
+    GLES2/UniformSet.cpp
+   )
+   SET(OPENGL_LIBRARIES -lGLESv2 -lEGL)
+else(GLES2)
+  find_package(OpenGL REQUIRED)
+  include_directories(${OpenGL_INCLUDE_DIRS})
+  link_directories(${OpenGL_LIBRARY_DIRS})
+  add_definitions(${OpenGL_DEFINITIONS})
+  if(NOT OPENGL_FOUND)
+  	message(ERROR " OPENGL not found!")
+  endif(NOT OPENGL_FOUND)
+  list(APPEND GLideN64_SOURCES 
+    OGL3X/GLSLCombiner_ogl3x.cpp
+    OGL3X/UniformBlock.cpp
+  )
+endif(GLES2)
 
 if("${CMAKE_CXX_COMPILER_ID}" STREQUAL "MSVC")
 	add_definitions( -D__MSC__)
