From 7d0e79a58f9aeb54ec679d19ff87710f25c95c8a Mon Sep 17 00:00:00 2001
From: Max Kellermann <max@duempel.org>
Date: Mon, 24 Apr 2017 14:38:06 +0200
Subject: [PATCH 09/62] JsonSchemaBuilder/Makefile: check "cd" success

Signed-off-by: memeka <mihailescu2m@gmail.com>
---
 tools/depends/native/JsonSchemaBuilder/Makefile | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/tools/depends/native/JsonSchemaBuilder/Makefile b/tools/depends/native/JsonSchemaBuilder/Makefile
index 3951342..3abd729 100644
--- a/tools/depends/native/JsonSchemaBuilder/Makefile
+++ b/tools/depends/native/JsonSchemaBuilder/Makefile
@@ -25,10 +25,10 @@ $(PLATFORM)/autogen.sh: $(SOURCE)/*
 	cp -a $(SOURCE) $(PLATFORM)
 
 $(PLATFORM)/configure: $(PLATFORM)/autogen.sh
-	cd $(PLATFORM); ./autogen.sh
+	cd $(PLATFORM) && ./autogen.sh
 
 $(PLATFORM)/config.log: $(PLATFORM)/configure
-	cd $(PLATFORM); $(CONFIGURE)
+	cd $(PLATFORM) && $(CONFIGURE)
 
 $(PLATFORM)/JsonSchemaBuilder: $(PLATFORM)/config.log
 	$(MAKE) -C $(PLATFORM)
-- 
2.7.4

