diff -Naur a/kernel/nv-drm.c b/kernel/nv-drm.c
--- a/kernel/nv-drm.c	2023-11-05 03:19:26.555143512 +0000
+++ b/kernel/nv-drm.c	2023-11-05 03:19:26.555143512 +0000
@@ -520,8 +520,9 @@
 #elif LINUX_VERSION_CODE <= KERNEL_VERSION(5, 10, 0)
     .gem_free_object_unlocked = nv_gem_free,
 #endif
-
+#if LINUX_VERSION_CODE < KERNEL_VERSION(6, 6, 0)
     .prime_handle_to_fd = drm_gem_prime_handle_to_fd,
+#endif
 #if LINUX_VERSION_CODE <= KERNEL_VERSION(5, 10, 0)
     .gem_prime_export = drm_gem_prime_export,
     .gem_prime_get_sg_table = nv_gem_prime_get_sg_table,
