--- a/kernel/conftest.sh 2025-05-14 11:44:20.223439900 +0300
+++ b/kernel/conftest.sh  2025-05-14 11:44:29.546716534 +0300
@@ -196,7 +196,7 @@
 }

 build_cflags() {
-    BASE_CFLAGS="-O2 -Wno-implicit-function-declaration -Wno-incompatible-pointer-types -D__KERNEL__ \
+    BASE_CFLAGS="-std=gnu17 -O2 -Wno-implicit-function-declaration -Wno-incompatible-pointer-types -D__KERNEL__ \
 -DKBUILD_BASENAME=\"#conftest$$\" -DKBUILD_MODNAME=\"#conftest$$\" \
 -nostdinc -isystem $ISYSTEM"

--- a/kernel/nvidia-modules-common.mk    2025-05-14 11:44:20.176773517 +0300
+++ b/kernel/nvidia-modules-common.mk 2025-05-14 11:44:28.033392404 +0300
@@ -116,7 +116,7 @@
 # NVIDIA specific CFLAGS and #define's.
 #

-EXTRA_CFLAGS += -D__KERNEL__ -DMODULE -DNVRM -DNV_VERSION_STRING=\"340.108\" -Wno-unused-function -Wuninitialized -fno-strict-aliasing -mno-red-zone -mcmodel=kernel -DNV_UVM_ENABLE -D__linux__ -DNV_DEV_NAME=\"$(MODULE_NAME)\"
+EXTRA_CFLAGS += -std=gnu17 -D__KERNEL__ -DMODULE -DNVRM -DNV_VERSION_STRING=\"340.108\" -Wno-unused-function -Wuninitialized -fno-strict-aliasing -mno-red-zone -mcmodel=kernel -DNV_UVM_ENABLE -D__linux__ -DNV_DEV_NAME=\"$(MODULE_NAME)\"

 #
 # Detect SGI UV systems and apply system-specific optimizations.
--- a/kernel/uvm/conftest.sh 2025-05-14 11:44:20.223439900 +0300
+++ b/kernel/uvm/conftest.sh  2025-05-14 11:44:28.816720973 +0300
@@ -196,7 +196,7 @@
 }

 build_cflags() {
-    BASE_CFLAGS="-O2 -Wno-implicit-function-declaration -Wno-incompatible-pointer-types -D__KERNEL__ \
+    BASE_CFLAGS="-std=gnu17 -O2 -Wno-implicit-function-declaration -Wno-incompatible-pointer-types -D__KERNEL__ \
 -DKBUILD_BASENAME=\"#conftest$$\" -DKBUILD_MODNAME=\"#conftest$$\" \
 -nostdinc -isystem $ISYSTEM"
