From 9e8f8bc79ae845bfd250b085151d684cf4d8c135 Mon Sep 17 00:00:00 2001
From: Max Kellermann <max@duempel.org>
Date: Mon, 24 Apr 2017 14:38:05 +0200
Subject: [PATCH 03/62] xbmc/gen-compileinfo.mk: set GIT_DIR, fixes out-of-tree
 build

Signed-off-by: memeka <mihailescu2m@gmail.com>
---
 xbmc/gen-compileinfo.mk | 1 +
 1 file changed, 1 insertion(+)

diff --git a/xbmc/gen-compileinfo.mk b/xbmc/gen-compileinfo.mk
index 11ded2a..4d93bdd 100644
--- a/xbmc/gen-compileinfo.mk
+++ b/xbmc/gen-compileinfo.mk
@@ -13,6 +13,7 @@ $(FILEPATH)/.GitRevision:
 	@if test -f $(GITVERFILE); then \
           GITREV=$$(cat $(GITVERFILE)) ;\
         elif test "$(GIT)" = "git" && test -d $(FILEPATH)/../.git ; then \
+          export GIT_DIR="$(FILEPATH)/../.git"; \
           if ! git diff-files --ignore-submodules --quiet -- || ! git diff-index --cached --ignore-submodules --quiet HEAD --; then \
             BUILD_DATE=$$(date -u "+%F"); \
             BUILD_SCMID=$$(git --no-pager log --abbrev=7 -n 1 --pretty=format:"%h-dirty"); \
-- 
2.7.4

