--- a/SConstruct	2023-08-30 17:25:09.494021248 +0200
+++ b/SConstruct	2023-08-30 17:29:17.680353388 +0200
@@ -1596,11 +1596,11 @@
 		error_text_opengl_mismatch = f'Rebirth configured with OpenGL enabled, but SDL{sdl2} configured with OpenGL disabled.  Disable Rebirth OpenGL or install an SDL{sdl2} with OpenGL enabled.'
 		test_opengl = (f'''
 #if !((SDL_MAJOR_VERSION == 1) && (SDL_MINOR_VERSION == 2) && (SDL_PATCHLEVEL >= 50))
-#ifndef SDL_VIDEO_OPENGL
+#if (!defined(SDL_VIDEO_OPENGL)) && (!defined(SDL_VIDEO_OPENGL_ES))
 #error "{error_text_opengl_mismatch}"
 #endif
 #endif
-''') if user_settings.opengl else ''
+''') if user_settings.opengl or user_settings.opengles else ''
 		main = '''
 	SDL_RWops *ops = reinterpret_cast<SDL_RWops *>(argv);
 #if DXX_MAX_JOYSTICKS
