diff --git a/similar/misc/physfsx.cpp b/similar/misc/physfsx.cpp
index acb655f9dd..8890ca413b 100644
--- a/similar/misc/physfsx.cpp
+++ b/similar/misc/physfsx.cpp
@@ -143,7 +143,7 @@ bool PHYSFSX_init(int argc, char *argv[])
 		path = getenv(&home_environ_var[4]);
 		if (!path)
 # if !(defined(__APPLE__) && defined(__MACH__))
-	path = "~/.d" DESCENT_PATH_NUMBER "x-rebirth/";
+	path = "/userdata/system/configs/d" DESCENT_PATH_NUMBER "x-rebirth/";
 # else
 	path = "~/Library/Preferences/D" DESCENT_PATH_NUMBER "X Rebirth/";
 # endif
