diff --git a/Makefile b/Makefile
index cada1b8..7c9cc23 100644
--- a/Makefile
+++ b/Makefile
@@ -61,7 +61,7 @@ else ifeq ($(platform), linux-portable)
 	LDFLAGS += -nostdlib
 else ifneq (,$(findstring rpi,$(platform)))
    TARGET := $(TARGET_NAME)_libretro.so
-   LDFLAGS += -shared -Wl,--version-script=libretro/link.T
+   #LDFLAGS += -shared -Wl,--version-script=libretro/link.T
    fpic = -fPIC
    SHARED := -shared -Wl,-version-script=link.T -Wl,-no-undefined
 else ifeq ($(platform), osx)
