diff --git a/Makefile.am b/Makefile.am
index 80b4fa8f8c..f25e52f169 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -106,15 +106,15 @@ endif
 # endif
 endif
 
-if STATIC_BUILD
-AM_LDFLAGS += -static
-AM_CPPFLAGS += -DGLEW_STATIC
-AM_CPPFLAGS += -DAL_LIBTYPE_STATIC
+#if STATIC_BUILD
+#AM_LDFLAGS += -static
+#AM_CPPFLAGS += -DGLEW_STATIC
+#AM_CPPFLAGS += -DAL_LIBTYPE_STATIC
 # Hack for OpenAL static missing libcommon.a
 # LIBS += ../openal-soft/libcommon.a
 # One or more .pc files are missing -liconv
-LIBS += -liconv
-endif
+#LIBS += -liconv
+#endif
 
 MSGFMT = @MSGFMT@
 ZIP = @ZIP@