diff --git a/src/fsuae/fsuae-path.c b/src/fsuae/fsuae-path.c
index 6f12da6..0b1b13c 100644
--- a/src/fsuae/fsuae-path.c
+++ b/src/fsuae/fsuae-path.c
@@ -482,11 +482,12 @@ const char *fsuae_path_themes_dir()
 
 const char *fsuae_path_plugins_dir(void)
 {
-    static const char *path;
-    if (!path) {
-        path = get_default_dir("Plugins", "plugins_dir", NULL, "plugins-dir");
-    }
-    return path;
+    return "/usr/share/fs-uae/Plugins";
+    //static const char *path;
+    //if (!path) {
+    //    path = get_default_dir("Plugins", "plugins_dir", NULL, "plugins-dir");
+    //}
+    //return path;
 }
 
 const char *fsuae_path_system_dir(void)
