diff --git a/CMakeLists.txt b/CMakeLists.txt
index e26ebe2..a1c8fd2 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -1,5 +1,5 @@
 # vim:noexpandtab:
-cmake_minimum_required(VERSION 3.6)
+cmake_minimum_required(VERSION 3.5)
 project(PPSSPP)
 
 #This is supposed to work but doesn't!
@@ -75,14 +75,15 @@ if(${CMAKE_SYSTEM_NAME} MATCHES "Android")
 endif()
 
 # We only support Vulkan on Unix, Android and Windows.
-if(ANDROID OR WIN32 OR (UNIX AND NOT APPLE))
-  set(VULKAN ON)
+if(DISABLE_VULKAN)
+  set(VULKAN OFF)
+  message("Vulkan disabled")
 else()
-  add_definitions(-DNO_VULKAN)
-endif()
-
-if(NOT DEFINED HEADLESS)
-	set(HEADLESS OFF)
+  if(ANDROID OR WIN32 OR (UNIX AND NOT APPLE))
+    set(VULKAN ON)
+  else()
+    add_definitions(-DNO_VULKAN)
+  endif()
 endif()
 
 if(GOLD)
@@ -122,6 +123,7 @@ option(USE_SYSTEM_FFMPEG "Dynamically link against system FFMPEG" ${USE_SYSTEM_F
 option(USE_SYSTEM_LIBZIP "Dynamically link against system libzip" ${USE_SYSTEM_LIBZIP})
 option(USE_WAYLAND_WSI "Set to ON to require Wayland support for Vulkan" ${USE_WAYLAND_WSI})
 option(USE_ADDRESS_SANITIZER "Use Clang memory sanitizer" ${USE_ADDRESS_SANITIZER})
+option(DISABLE_VULKAN "Set to ON to if building for Vulkan is unsupported" ${DISABLE_VULKAN})
 
 if(UNIX AND NOT (APPLE OR ANDROID) AND VULKAN)
 	add_definitions(-DVK_USE_PLATFORM_XLIB_KHR)
