diff --git a/Core/Util/PathUtil.cpp b/Core/Util/PathUtil.cpp
index 96be0ea..38b19a9 100644
--- a/Core/Util/PathUtil.cpp
+++ b/Core/Util/PathUtil.cpp
@@ -43,38 +43,33 @@ Path FindConfigFile(const Path &searchPath, std::string_view baseFilename, bool
 }
 
 Path GetSysDirectory(PSPDirectories directoryType) {
-	const Path &memStickDirectory = g_Config.memStickDirectory;
-	Path pspDirectory;
-	if (!strcasecmp(memStickDirectory.GetFilename().c_str(), "PSP")) {
-		// Let's strip this off, to easily allow choosing a root directory named "PSP" on Android.
-		pspDirectory = memStickDirectory;
-	} else {
-		pspDirectory = memStickDirectory / "PSP";
-	}
+	// batocera - set config directories
+	Path memStickDirectory = Path("/userdata/saves/psp/");
+	Path pspDirectory = Path("/userdata/system/configs/ppsspp/PSP/");
 
 	switch (directoryType) {
 	case DIRECTORY_PSP:
 		return pspDirectory;
 	case DIRECTORY_CHEATS:
-		return pspDirectory / "Cheats";
+		return Path("/userdata/cheats/psp/");
 	case DIRECTORY_GAME:
 		return pspDirectory / "GAME";
 	case DIRECTORY_SAVEDATA:
-		return pspDirectory / "SAVEDATA";
+		return memStickDirectory / "SAVEDATA";
 	case DIRECTORY_SCREENSHOT:
-		return pspDirectory / "SCREENSHOT";
+		return Path("/userdata/screenshots/");
 	case DIRECTORY_SYSTEM:
 		return pspDirectory / "SYSTEM";
 	case DIRECTORY_PAUTH:
-		return memStickDirectory / "PAUTH";  // This one's at the root...
+		return memStickDirectory / "PAUTH";
 	case DIRECTORY_EXDATA:
-		return memStickDirectory / "EXDATA";  // This one's traditionally at the root...
+		return memStickDirectory / "EXDATA";
 	case DIRECTORY_DUMP:
 		return pspDirectory / "SYSTEM/DUMP";
 	case DIRECTORY_SAVESTATE:
-		return pspDirectory / "PPSSPP_STATE";
+		return memStickDirectory;
 	case DIRECTORY_CACHE:
-		return pspDirectory / "SYSTEM/CACHE";
+		return Path("/userdata/system/cache/ppsspp/");
 	case DIRECTORY_TEXTURES:
 		return pspDirectory / "TEXTURES";
 	case DIRECTORY_PLUGINS:
@@ -83,22 +78,21 @@ Path GetSysDirectory(PSPDirectories directoryType) {
 		if (!g_Config.appCacheDirectory.empty()) {
 			return g_Config.appCacheDirectory;
 		}
-		return pspDirectory / "SYSTEM/CACHE";
+		return Path("/userdata/system/cache/ppsspp/");
 	case DIRECTORY_VIDEO:
 		return pspDirectory / "VIDEO";
 	case DIRECTORY_AUDIO:
-		return pspDirectory / "AUDIO";
+		return Path("/userdata/music/");
 	case DIRECTORY_CUSTOM_SHADERS:
 		return pspDirectory / "shaders";
 	case DIRECTORY_CUSTOM_THEMES:
 		return pspDirectory / "themes";
 
 	case DIRECTORY_MEMSTICK_ROOT:
-		return g_Config.memStickDirectory;
-		// Just return the memory stick root if we run into some sort of problem.
+		return memStickDirectory;
 	default:
 		ERROR_LOG(Log::FileSystem, "Unknown directory type.");
-		return g_Config.memStickDirectory;
+		return pspDirectory;
 	}
 }
 
diff --git a/SDL/SDLJoystick.cpp b/SDL/SDLJoystick.cpp
index 290cf57..82378bd 100644
--- a/SDL/SDLJoystick.cpp
+++ b/SDL/SDLJoystick.cpp
@@ -24,7 +24,7 @@ SDLJoystick::SDLJoystick(bool init_SDL ) : registeredAsEventHandler(false) {
 		SDL_Init(SDL_INIT_JOYSTICK | SDL_INIT_VIDEO | SDL_INIT_GAMECONTROLLER);
 	}
 
-	const char *dbPath = "gamecontrollerdb.txt";
+	const char *dbPath = "/userdata/system/configs/ppsspp/gamecontrollerdb.txt";
 	INFO_LOG(Log::System, "loading control pad mappings from %s:", dbPath);
 
 	size_t size;
diff --git a/UI/NativeApp.cpp b/UI/NativeApp.cpp
index 4a26924..69373dd 100644
--- a/UI/NativeApp.cpp
+++ b/UI/NativeApp.cpp
@@ -367,7 +367,7 @@ void NativeInit(int argc, const char *argv[], const char *savegame_dir, const ch
 	g_VFS.Register("", new DirectoryReader(File::GetExeDirectory()));
 	g_VFS.Register("", new DirectoryReader(Path("/usr/local/share/ppsspp/assets")));
 	g_VFS.Register("", new DirectoryReader(Path("/usr/local/share/games/ppsspp/assets")));
-	g_VFS.Register("", new DirectoryReader(Path("/usr/share/ppsspp/assets")));
+	g_VFS.Register("", new DirectoryReader(Path("/usr/share/ppsspp/PPSSPP")));
 	g_VFS.Register("", new DirectoryReader(Path("/usr/share/games/ppsspp/assets")));
 #endif
 
@@ -479,7 +479,7 @@ void NativeInit(int argc, const char *argv[], const char *savegame_dir, const ch
 	else // Just in case
 		config = "./config";
 
-	g_Config.memStickDirectory = Path(config) / "ppsspp";
+	g_Config.memStickDirectory = Path("/userdata/saves/psp/");
 	g_Config.flash0Directory = File::GetExeDirectory() / "assets/flash0";
 	if (getenv("HOME") != nullptr) {
 		g_Config.defaultCurrentDirectory = Path(getenv("HOME"));
