diff --git a/Core/System.cpp b/Core/System.cpp
index 67a4e6f..3b99408 100644
--- a/Core/System.cpp
+++ b/Core/System.cpp
@@ -517,7 +517,7 @@ std::string GetSysDirectory(PSPDirectories directoryType) {
 	case DIRECTORY_DUMP:
 		return g_Config.memStickDirectory + "PSP/SYSTEM/DUMP/";
 	case DIRECTORY_SAVESTATE:
-		return g_Config.memStickDirectory + "PSP/PPSSPP_STATE/";
+		return "/userdata/saves/psp/";
 	case DIRECTORY_CACHE:
 		return g_Config.memStickDirectory + "PSP/SYSTEM/CACHE/";
 	case DIRECTORY_TEXTURES:
