diff --git a/SDL/SDLJoystick.cpp b/SDL/SDLJoystick.cpp
index b9e0a83..45e470c 100644
--- a/SDL/SDLJoystick.cpp
+++ b/SDL/SDLJoystick.cpp
@@ -19,7 +19,7 @@ static int SDLJoystickEventHandlerWrapper(void* userdata, SDL_Event* event)
 	return 0;
 }
 
-SDLJoystick::SDLJoystick(bool init_SDL ) : registeredAsEventHandler(false) {
+SDLJoystick::SDLJoystick(bool init_SDL, int njoy) : registeredAsEventHandler(false) {
 	SDL_SetHint(SDL_HINT_JOYSTICK_ALLOW_BACKGROUND_EVENTS, "1");
 	if (init_SDL) {
 		SDL_Init(SDL_INIT_JOYSTICK | SDL_INIT_VIDEO | SDL_INIT_GAMECONTROLLER);
@@ -41,16 +41,18 @@ SDLJoystick::SDLJoystick(bool init_SDL ) : registeredAsEventHandler(false) {
 		cout << "gamecontrollerdb.txt missing" << endl;
 	}
 	cout << "SUCCESS!" << endl;
-	setUpControllers();
+	_njoy = njoy;
+	//setUpControllers(); // ADD are called automatically
 }
 
 void SDLJoystick::setUpControllers() {
 	int numjoys = SDL_NumJoysticks();
-	for (int i = 0; i < numjoys; i++) {
-		setUpController(i);
-	}
-	if (controllers.size() > 0) {
-		cout << "pad 1 has been assigned to control pad: " << SDL_JoystickNameForIndex(0) << endl;
+
+	if(_njoy < numjoys) {
+          setUpController(_njoy);
+	  if (controllers.size() > 0) {
+		cout << "pad 1 has been assigned to control pad: " << SDL_JoystickNameForIndex(_njoy) << " (" << _njoy << ")" << endl;
+	  }
 	}
 }
 
@@ -200,11 +202,9 @@ void SDLJoystick::ProcessInput(SDL_Event &event){
 		}
 		break;
 	case SDL_CONTROLLERDEVICEADDED:
-		// for add events, "which" is the device index!
-		int prevNumControllers = controllers.size();
-		setUpController(event.cdevice.which);
-		if (prevNumControllers == 0 && controllers.size() > 0) {
-                       cout << "pad 1 has been assigned to control pad: " << SDL_JoystickNameForIndex(0) << endl;
+ 	        if(event.cdevice.which == _njoy) {
+		  setUpController(_njoy);
+                  cout << "pad 1 has been assigned to control pad: " << SDL_JoystickNameForIndex(_njoy) << endl;
 		}
 		break;
 	}
diff --git a/SDL/SDLJoystick.h b/SDL/SDLJoystick.h
index d7494f8..173a865 100644
--- a/SDL/SDLJoystick.h
+++ b/SDL/SDLJoystick.h
@@ -16,7 +16,7 @@
 
 class SDLJoystick{
 public:
-	SDLJoystick(bool init_SDL = false);
+	SDLJoystick(bool init_SDL = false, int njoy = 0);
 	~SDLJoystick();
 
 	void registerEventHandler();
@@ -30,4 +30,5 @@ class SDLJoystick{
 	bool registeredAsEventHandler;
 	std::vector<SDL_GameController *> controllers;
 	std::map<int, int> controllerDeviceMap;
+	int _njoy;
 };
diff --git a/SDL/SDLMain.cpp b/SDL/SDLMain.cpp
index d7a1b3a..b251904 100644
--- a/SDL/SDLMain.cpp
+++ b/SDL/SDLMain.cpp
@@ -646,6 +646,7 @@ int main(int argc, char *argv[]) {
 	bool set_ipad = false;
 	float set_dpi = 1.0f;
 	float set_scale = 1.0f;
+	int set_njoy = 0;
 
 	// Produce a new set of arguments with the ones we skip.
 	int remain_argc = 1;
@@ -664,6 +665,8 @@ int main(int argc, char *argv[]) {
 			set_dpi = parseFloat(argv[i]);
 		else if (set_scale == -2)
 			set_scale = parseFloat(argv[i]);
+		else if (set_njoy == -2)
+			set_njoy = parseInt(argv[i]);
 		else if (!strcmp(argv[i],"--xres"))
 			set_xres = -2;
 		else if (!strcmp(argv[i],"--yres"))
@@ -676,6 +679,8 @@ int main(int argc, char *argv[]) {
 			set_ipad = true;
 		else if (!strcmp(argv[i],"--portrait"))
 			portrait = true;
+		else if (!strcmp(argv[i],"--njoy"))
+			set_njoy = -2;
 		else {
 			remain_argv[remain_argc++] = argv[i];
 		}
@@ -906,7 +911,7 @@ int main(int argc, char *argv[]) {
 	InitSDLAudioDevice();
 
 	if (joystick_enabled) {
-		joystick = new SDLJoystick();
+		joystick = new SDLJoystick(false, set_njoy);
 	} else {
 		joystick = nullptr;
 	}
