--- a/os_dep/linux/ioctl_cfg80211.c
+++ b/os_dep/linux/ioctl_cfg80211.c
@@ -17,6 +17,10 @@
 #include <drv_types.h>
 #include <hal_data.h>
 
+/* Force version to 6.3.0 for compatibility with this kernel */
+#undef  LINUX_VERSION_CODE
+#define LINUX_VERSION_CODE 393984
+
 #ifdef CONFIG_IOCTL_CFG80211
 
 #ifndef DBG_RTW_CFG80211_STA_PARAM
--- a/os_dep/osdep_service.c
+++ b/os_dep/osdep_service.c
@@ -18,6 +18,8 @@
 
 #include <drv_types.h>
 
+MODULE_IMPORT_NS(VFS_internal_I_am_really_a_filesystem_and_am_NOT_a_driver);
+
 #define RT_TAG	'1178'
 
 #ifdef DBG_MEMORY_LEAK
