diff --git a/CMake/Dependencies.cmake b/CMake/Dependencies.cmake
index e591174..c6fcaf1 100644
--- a/CMake/Dependencies.cmake
+++ b/CMake/Dependencies.cmake
@@ -31,7 +31,7 @@ else()
   if(DEVILUTIONX_SYSTEM_SDL2)
     find_package(SDL2 REQUIRED)
     if(TARGET SDL2::SDL2)
-      set(SDL2_MAIN SDL2::SDL2main)
+      message("SDL2_MAIN SDL2::SDL2main : ok, don't worry about me")
     elseif(TARGET SDL2::SDL2-static)
       # On some distros, such as vitasdk, only the SDL2::SDL2-static target is available.
       # Alias to SDL2::SDL2 because some finder scripts may refer to SDL2::SDL2.
