diff --git a/src/intel/compiler/jay/jay_ir.h b/src/intel/compiler/jay/jay_ir.h
index 8307044..abf175a 100644
--- a/src/intel/compiler/jay/jay_ir.h
+++ b/src/intel/compiler/jay/jay_ir.h
@@ -579,7 +579,7 @@ jay_type_is_any_float(enum jay_type t)
    return jay_base_type(t) == JAY_TYPE_F || jay_base_type(t) == JAY_TYPE_BF;
 }
 
-enum jay_predication : uint8_t {
+enum jay_predication {
    /** No predication. */
    JAY_NOT_PREDICATED = 0,
 
@@ -634,7 +634,7 @@ typedef struct jay_inst {
    bool decrement_dep:1;
    unsigned padding  :12;
 
-   enum jay_predication predication;
+   uint8_t predication;
    enum jay_conditional_mod conditional_mod;
 
    jay_def cond_flag; /**< conditional flag */
