diff --git a/drivers/net/wireless/ew-7611ulb/hal/phydm/halrf/halrf.h b/drivers/net/wireless/ew-7611ulb/hal/phydm/halrf/halrf.h
index 5b63a00b29..462447214c 100644
--- a/drivers/net/wireless/ew-7611ulb/hal/phydm/halrf/halrf.h
+++ b/drivers/net/wireless/ew-7611ulb/hal/phydm/halrf/halrf.h
@@ -206,8 +206,8 @@ halrf_cmn_info_init(
 void
 halrf_cmn_info_hook(
 	void		*p_dm_void,
-	u32		cmn_info,
-	void		*p_value
+	enum halrf_cmninfo_hook_e cmn_info,
+	void		*value
 );
 
 void
@@ -298,7 +298,6 @@ halrf_psd_log2base(
 	IN u32 val
 );
 
-
 #if (RTL8822B_SUPPORT == 1 || RTL8821C_SUPPORT == 1)
 void halrf_iqk_dbg(void	*p_dm_void);
 #endif
diff --git a/drivers/net/wireless/rtl88x2bu/hal/phydm/halrf/halrf.h b/drivers/net/wireless/rtl88x2bu/hal/phydm/halrf/halrf.h
index 7240397263..58c6ca25a4 100644
--- a/drivers/net/wireless/rtl88x2bu/hal/phydm/halrf/halrf.h
+++ b/drivers/net/wireless/rtl88x2bu/hal/phydm/halrf/halrf.h
@@ -354,7 +354,7 @@ halrf_cmn_info_init(
 void
 halrf_cmn_info_hook(
 	void		*dm_void,
-	u32		cmn_info,
+	enum halrf_cmninfo_hook cmn_info,
 	void		*value
 );
 
