diff --git a/Makefile b/Makefile
index b4cfd80..5eb3894 100644
--- a/Makefile
+++ b/Makefile
@@ -667,10 +667,18 @@ KBUILD_CFLAGS	+= $(call cc-disable-warning, format-overflow)
 KBUILD_CFLAGS	+= $(call cc-disable-warning, int-in-bool-context)
 KBUILD_CFLAGS	+= $(call cc-disable-warning, address-of-packed-member)
 KBUILD_CFLAGS	+= $(call cc-disable-warning, attribute-alias)
-KBUILD_CFLAGS += $(call cc-disable-warning, missing-attributes)
-KBUILD_CFLAGS += $(call cc-disable-warning, array-bounds)
-KBUILD_CFLAGS += $(call cc-disable-warning, bool-operation)
-KBUILD_CFLAGS += $(call cc-disable-warning, restrict)
+KBUILD_CFLAGS	+= $(call cc-disable-warning, missing-attributes)
+KBUILD_CFLAGS	+= $(call cc-disable-warning, array-bounds)
+KBUILD_CFLAGS	+= $(call cc-disable-warning, bool-operation)
+KBUILD_CFLAGS	+= $(call cc-disable-warning, restrict)
+KBUILD_CFLAGS	+= $(call cc-disable-warning,zero-length-bounds,)
+KBUILD_CFLAGS	+= $(call cc-disable-warning,stringop-overflow,)
+KBUILD_CFLAGS	+= $(call cc-disable-warning,uninitialized,)
+KBUILD_CFLAGS	+= $(call cc-disable-warning,misleading-indentation)
+KBUILD_CFLAGS	+= $(call cc-disable-warning,array-compare)
+KBUILD_CFLAGS	+= $(call cc-disable-warning,stringop-overread)
+KBUILD_CFLAGS	+= $(call cc-disable-warning,dangling-pointer)
+KBUILD_CFLAGS	+= $(call cc-disable-warning,address)
 
 ifdef CONFIG_CC_OPTIMIZE_FOR_SIZE
 KBUILD_CFLAGS	+= -Os
