diff --git a/arch/arm64/boot/dts/rockchip/rk3326-powkiddy-rgb10x.dts b/arch/arm64/boot/dts/rockchip/rk3326-powkiddy-rgb10x.dts
new file mode 100644
index 000000000000..70e642907531
--- /dev/null
+++ b/arch/arm64/boot/dts/rockchip/rk3326-powkiddy-rgb10x.dts
@@ -0,0 +1,209 @@
+// SPDX-License-Identifier: (GPL-2.0+ OR MIT)
+/*
+ * Copyright (c) 2024 ROCKNIX
+ */
+
+/dts-v1/;
+#include "rk3326-odroid-go.dtsi"
+#include "retrogame_joypad_s2_f1.dtsi"
+
+/ {
+	model = "Powkiddy RGB10X";
+	compatible = "powkiddy,rk3326-rgb10x", "rockchip,rk3326";
+
+	aliases {
+		mmc1 = &sdio;
+	};
+
+	battery: battery {
+		compatible = "simple-battery";
+		charge-full-design-microamp-hours = <2800000>;
+		charge-term-current-microamp = <280000>;
+		constant-charge-current-max-microamp = <2000000>;
+		constant-charge-voltage-max-microvolt = <4200000>;
+		factory-internal-resistance-micro-ohms = <180000>;
+		voltage-max-design-microvolt = <4100000>;
+		voltage-min-design-microvolt = <3500000>;
+
+		ocv-capacity-celsius = <20>;
+		ocv-capacity-table-0 =  <4046950 100>, <4001920 95>, <3967900 90>, <3919950 85>,
+					<3888450 80>, <3861850 75>, <3831540 70>, <3799130 65>,
+					<3768190 60>, <3745650 55>, <3726610 50>, <3711630 45>,
+					<3696720 40>, <3685660 35>, <3674950 30>, <3663050 25>,
+					<3649470 20>, <3635260 15>, <3616920 10>, <3592440 5>,
+					<3574170 0>;
+	};
+
+	gpio-keys-vol {
+		compatible = "gpio-keys";
+		autorepeat;
+		pinctrl-0 = <&btn_pins_vol>;
+		pinctrl-names = "default";
+
+		button-vol-down {
+			gpios = <&gpio2 RK_PA1 GPIO_ACTIVE_LOW>;
+			label = "VOLUMEDOWN";
+			linux,code = <KEY_VOLUMEDOWN>;
+		};
+
+		button-volume-up {
+			gpios = <&gpio2 RK_PA0 GPIO_ACTIVE_LOW>;
+			label = "VOLUMEUP";
+			linux,code = <KEY_VOLUMEUP>;
+		};
+	};
+};
+
+&joypad {
+	compatible = "rocknix-singleadc-joypad";
+	joypad-name = "retrogame_joypad_s1_f2"; /* distinct name to indicate there is 1 stick and 2 fn buttons */
+	joypad-product = <0x1112>;  /* 11 inherited, 1 for number of sticks, 2 for number of FN */
+	status = "okay";
+
+	/* gpio pincontrol setup */
+	pinctrl-names = "default";
+	pinctrl-0 = <&btn_pins>;
+
+	/* Analog mux define */
+	io-channel-names = "amux_adc";
+	io-channels = <&saradc 1>;
+
+	/* adc mux channel count */
+	/* despite there is only left stick, amux-count needs to be 4, */
+	/* because amux 0 and 1 are hardcoded as RX and RY in the driver */
+	amux-count = <4>;
+	/* adc mux select(a,b) gpio */
+	amux-a-gpios = <&gpio3 RK_PB3 GPIO_ACTIVE_LOW>;
+	amux-b-gpios = <&gpio3 RK_PB0 GPIO_ACTIVE_LOW>;
+	/* adc mux enable gpio */
+	amux-en-gpios = <&gpio3 RK_PB5 GPIO_ACTIVE_LOW>;
+
+	/*
+	  Analog Stick data tuning value(precent)
+	  p = positive direction, n = negative direction
+	  report value = (real_adc_data * tuning_value) / 100
+	*/
+	abs_x-p-tuning = <200>;
+	abs_x-n-tuning = <200>;
+
+	abs_y-p-tuning = <200>;
+	abs_y-n-tuning = <200>;
+
+	abs_rx-p-tuning = <200>;
+	abs_rx-n-tuning = <200>;
+
+	abs_ry-p-tuning = <200>;
+	abs_ry-n-tuning = <200>;
+
+	/* required for RGB10X */
+	invert-absx;
+	invert-absy;
+
+	/*
+	    joypad driver is poll-device driver.
+	    poll-device is does not support wakeup-source.
+	*/
+	up    { gpios = <&gpio1 RK_PB4 GPIO_ACTIVE_LOW>; };
+	down  { gpios = <&gpio1 RK_PB5 GPIO_ACTIVE_LOW>; };
+	left  { gpios = <&gpio1 RK_PB6 GPIO_ACTIVE_LOW>; };
+	right { gpios = <&gpio1 RK_PB7 GPIO_ACTIVE_LOW>; };
+
+	a     { gpios = <&gpio1 RK_PA2 GPIO_ACTIVE_LOW>; };
+	b     { gpios = <&gpio1 RK_PA5 GPIO_ACTIVE_LOW>; };
+	x     { gpios = <&gpio1 RK_PA7 GPIO_ACTIVE_LOW>; };
+	y     { gpios = <&gpio1 RK_PA6 GPIO_ACTIVE_LOW>; };
+
+	tl    { gpios = <&gpio2 RK_PA6 GPIO_ACTIVE_LOW>; };
+	tr    { gpios = <&gpio2 RK_PA7 GPIO_ACTIVE_LOW>; };
+	tr2   { gpios = <&gpio3 RK_PB7 GPIO_ACTIVE_LOW>; };
+	tl2   { gpios = <&gpio3 RK_PB2 GPIO_ACTIVE_LOW>; };
+
+	select{ gpios = <&gpio3 RK_PB1 GPIO_ACTIVE_LOW>; };
+	start { gpios = <&gpio3 RK_PB4 GPIO_ACTIVE_LOW>; };
+
+	thumbl{ gpios = <&gpio2 RK_PA2 GPIO_ACTIVE_LOW>; };
+
+	/* Plus and Minus face buttons are mapped to mode and thumbr */
+	mode {
+		gpios = <&gpio2 RK_PA4 GPIO_ACTIVE_LOW>;
+		label = "GPIO MINUS";
+	};
+	thumbr {
+		gpios = <&gpio2 RK_PA3 GPIO_ACTIVE_LOW>;
+		label = "GPIO PLUS";
+	};
+};
+
+&internal_display {
+	compatible = "powkiddy,rk2023-panel", "newvision,nv3051d";
+	vdd-supply = <&vcc_lcd>;
+};
+
+&io_domains {
+	vccio1-supply = <&vccio_sd>;
+};
+
+/delete-node/ &pwm3;
+/delete-node/ &pwm_led;
+
+&red_led {
+	gpios = <&gpio2 RK_PB5 GPIO_ACTIVE_HIGH>;
+};
+
+&red_led_pin {
+	rockchip,pins = <2 RK_PB5 RK_FUNC_GPIO &pcfg_pull_none>;
+};
+
+&rk817_charger {
+	monitored-battery = <&battery>;
+};
+
+&sdio {
+	cap-sd-highspeed;
+	card-detect-delay = <200>;
+	cd-gpios = <&gpio3 RK_PB6 GPIO_ACTIVE_LOW>;
+	sd-uhs-sdr12;
+	sd-uhs-sdr25;
+	sd-uhs-sdr50;
+	sd-uhs-sdr104;
+	vmmc-supply = <&vcc_sd>;
+	vqmmc-supply = <&vccio_sd>;
+	status = "okay";
+};
+
+&vcc_sd {
+	regulator-max-microvolt = <3000000>;
+	regulator-min-microvolt = <1800000>;
+};
+
+&vccio_sd {
+	regulator-max-microvolt = <1800000>;
+};
+
+&pinctrl {
+	btns {
+		btn_pins: btn-pins {
+			rockchip,pins = <1 RK_PA2 RK_FUNC_GPIO &pcfg_pull_up>,
+					<1 RK_PA5 RK_FUNC_GPIO &pcfg_pull_up>,
+					<1 RK_PA6 RK_FUNC_GPIO &pcfg_pull_up>,
+					<1 RK_PA7 RK_FUNC_GPIO &pcfg_pull_up>,
+					<1 RK_PB4 RK_FUNC_GPIO &pcfg_pull_up>,
+					<1 RK_PB5 RK_FUNC_GPIO &pcfg_pull_up>,
+					<1 RK_PB6 RK_FUNC_GPIO &pcfg_pull_up>,
+					<1 RK_PB7 RK_FUNC_GPIO &pcfg_pull_up>,
+					<2 RK_PA2 RK_FUNC_GPIO &pcfg_pull_up>,
+					<2 RK_PA3 RK_FUNC_GPIO &pcfg_pull_up>,
+					<2 RK_PA4 RK_FUNC_GPIO &pcfg_pull_up>,
+					<2 RK_PA6 RK_FUNC_GPIO &pcfg_pull_up>,
+					<2 RK_PA7 RK_FUNC_GPIO &pcfg_pull_up>,
+					<3 RK_PB1 RK_FUNC_GPIO &pcfg_pull_up>,
+					<3 RK_PB2 RK_FUNC_GPIO &pcfg_pull_up>,
+					<3 RK_PB4 RK_FUNC_GPIO &pcfg_pull_up>,
+					<3 RK_PB7 RK_FUNC_GPIO &pcfg_pull_up>;
+		};
+		btn_pins_vol: btn-pins-vol {
+			rockchip,pins = <2 RK_PA0 RK_FUNC_GPIO &pcfg_pull_up>,
+					<2 RK_PA1 RK_FUNC_GPIO &pcfg_pull_up>;
+		};
+	};
+};
