diff --git a/drivers/net/wireless/esp8089/esp_sip.c b/drivers/net/wireless/esp8089/esp_sip.c
index 95172cb98..45f7a69dc 100644
--- a/drivers/net/wireless/esp8089/esp_sip.c
+++ b/drivers/net/wireless/esp8089/esp_sip.c
@@ -81,7 +81,7 @@ static struct sip_trace str;
 #define STRACE_TX_ONE_SHOT_INC() (str.tx_one_shot_overflow++)
 #define STRACE_SHOW(sip)
 #else
-#define esp_sip_dbg(...)
+#define esp_sip_dbg(...) do {} while (0)
 #define STRACE_TX_DATA_INC()
 #define STRACE_TX_CMD_INC()
 #define STRACE_RX_DATA_INC()
