--- a/tools/lib/traceevent/Makefile	2021-02-03 01:55:41.000000000 +0100
+++ b/tools/lib/traceevent/Makefile.patched	2021-03-09 19:30:54.508423866 +0100
@@ -251,7 +251,7 @@
 
 define do_generate_dynamic_list_file
 	(echo '{';							\
-	$(NM) -u -D $1 | awk 'NF>1 {print "\t"$$2";"}' | sort -u;	\
+	$(NM) -u -D $1 | awk 'NF>1 {sub("@.*", "", $$2); print "\t"$$2";"}' | sort -u;\
 	echo '};';							\
 	) > $2
 endef
