--- a/libswirl/sdl/sdl.cpp	2020-08-13 20:10:37.078783737 +0200
+++ b/libswirl/sdl/sdl.cpp	2020-08-13 20:10:29.154634612 +0200
@@ -241,11 +241,8 @@
 	int window_height = cfgLoadInt("x11","height", WINDOW_HEIGHT);
 
 	int flags = 0;
-	#ifdef TARGET_PANDORA
-		flags |= SDL_FULLSCREEN;
-	#else
-		flags |= SDL_SWSURFACE | SDL_WINDOW_RESIZABLE;
-	#endif
+	flags |= SDL_WINDOW_FULLSCREEN;
+	flags |= SDL_SWSURFACE;
 
 	// try full GL first
 	if (!sdlgl_CreateWindow(false, flags, window_width, window_height, wind, disp)) {
