From f5f78dda7b1161779ce7a0c495f54e30f971776a Mon Sep 17 00:00:00 2001
From: Stefan Saraev <stefan@saraev.ca>
Date: Sat, 18 Apr 2015 14:59:29 +0300
Subject: [PATCH 07/13] disable minimize

---
 xbmc/Application.cpp | 1 -
 1 file changed, 1 deletion(-)

diff --git a/xbmc/Application.cpp b/xbmc/Application.cpp
index 95ae98c..9aca1e3 100644
--- a/xbmc/Application.cpp
+++ b/xbmc/Application.cpp
@@ -5044,7 +5044,6 @@ bool CApplication::SwitchToFullScreen(bool force /* = false */)
 
 void CApplication::Minimize()
 {
-  CServiceBroker::GetWinSystem()->Minimize();
 }
 
 PLAYERCOREID CApplication::GetCurrentPlayer()
-- 
2.5.0

