diff --git a/src/gui/render.cpp b/src/gui/render.cpp
index d6be555..e3a0fc4 100644
--- a/src/gui/render.cpp
+++ b/src/gui/render.cpp
@@ -419,6 +419,7 @@ forcenormal:
 		}
 	}
 /* Setup the scaler variables */
+	gfx_scalew = 320/480.0 - 0.04;
 	gfx_flags=GFX_SetSize(width,height,gfx_flags,gfx_scalew,gfx_scaleh,&RENDER_CallBack);
 	if (gfx_flags & GFX_CAN_8)
 		render.scale.outMode = scalerMode8;
diff --git a/src/gui/sdlmain.cpp b/src/gui/sdlmain.cpp
index 95b94ee..c3fcc4b 100644
--- a/src/gui/sdlmain.cpp
+++ b/src/gui/sdlmain.cpp
@@ -1514,7 +1514,10 @@ void GFX_EndUpdate( const Bit16u *changedLines ) {
 	case SCREEN_TEXTURE:
 		SDL_UnlockTexture(sdl.texture.texture);
 		SDL_RenderClear(sdl.renderer);
-		SDL_RenderCopy(sdl.renderer, sdl.texture.texture, NULL, &sdl.clip);
+		//SDL_RenderCopy(sdl.renderer, sdl.texture.texture, NULL, &sdl.clip);
+ 		static SDL_Point x = {160, 160};
+ 		static int n = -90;
+ 		SDL_RenderCopyEx(sdl.renderer, sdl.texture.texture, NULL, &sdl.clip, n, &x, SDL_FLIP_NONE);
 		SDL_RenderPresent(sdl.renderer);
 		break;
 #else	// !SDL_VERSION_ATLEAST(2,0,0)
