diff --git a/drivers/media/usb/uvc/uvc_driver.c b/drivers/media/usb/uvc/uvc_driver.c
index e289cc71b..fc9f66e8d 100644
--- a/drivers/media/usb/uvc/uvc_driver.c
+++ b/drivers/media/usb/uvc/uvc_driver.c
@@ -37,6 +37,7 @@ static unsigned int uvc_quirks_param = -1;
 unsigned int uvc_dbg_param;
 unsigned int uvc_timeout_param = UVC_CTRL_STREAMING_TIMEOUT;
 bool uvc_allow_privacy_override_param;
+unsigned int uvc_bandwidth_cap_param;
 
 static struct usb_driver uvc_driver;
 
@@ -2508,6 +2509,8 @@ MODULE_PARM_DESC(timeout, "Streaming control requests timeout");
 module_param_named(allow_privacy_override, uvc_allow_privacy_override_param, bool, 0644);
 MODULE_PARM_DESC(allow_privacy_override,
 		 "Allow access to privacy related controls");
+module_param_named(bandwidth_cap, uvc_bandwidth_cap_param, uint, S_IRUGO|S_IWUSR);
+MODULE_PARM_DESC(bandwidth_cap, "Maximum bandwidth per device");
 
 /* ------------------------------------------------------------------------
  * Driver initialization and cleanup
diff --git a/drivers/media/usb/uvc/uvc_video.c b/drivers/media/usb/uvc/uvc_video.c
index fc3536a43..af1d8218e 100644
--- a/drivers/media/usb/uvc/uvc_video.c
+++ b/drivers/media/usb/uvc/uvc_video.c
@@ -271,6 +271,11 @@ static void uvc_fixup_video_ctrl(struct uvc_streaming *stream,
 				     stream->maxpsize);
 		ctrl->dwMaxPayloadTransferSize = stream->maxpsize;
 	}
+
+	if (uvc_bandwidth_cap_param && ctrl->dwMaxPayloadTransferSize > uvc_bandwidth_cap_param) {
+	  dev_printk(KERN_WARNING, &(stream->dev)->udev->dev, "Bandwidth capped from %u to %u B/frame.\n", ctrl->dwMaxPayloadTransferSize, uvc_bandwidth_cap_param);
+	  ctrl->dwMaxPayloadTransferSize = uvc_bandwidth_cap_param;
+	}
 }
 
 static size_t uvc_video_ctrl_size(struct uvc_streaming *stream)
diff --git a/drivers/media/usb/uvc/uvcvideo.h b/drivers/media/usb/uvc/uvcvideo.h
index b6bcee4a2..54a1d25a4 100644
--- a/drivers/media/usb/uvc/uvcvideo.h
+++ b/drivers/media/usb/uvc/uvcvideo.h
@@ -668,6 +668,7 @@ extern unsigned int uvc_dbg_param;
 extern unsigned int uvc_timeout_param;
 extern unsigned int uvc_hw_timestamps_param;
 extern bool uvc_allow_privacy_override_param;
+extern unsigned int uvc_bandwidth_cap_param;
 
 #define uvc_dbg(_dev, flag, fmt, ...)					\
 do {									\
