--- a/makefile	2021-01-19 15:07:57.734211143 +0100
+++ b/makefile	2021-01-19 15:08:21.464210324 +0100
@@ -29,7 +29,7 @@
 # Linux
 ifeq  ($(PLATFORM),Linux)
 EXTRA_LIBS = libdrm
-CPPFLAGS += $(shell $(PKG_CONFIG) --cflags $(EXTRA_LIBS))
+CPPFLAGS += $(shell $(PKG_CONFIG) --define-prefix --cflags $(EXTRA_LIBS))
 SRC += display_linux.cpp custom_video_xrandr.cpp custom_video_drmkms.cpp
 CPPFLAGS += -fPIC
 LIBS = -ldl
