diff --git a/makefile b/makefile
index 89f52e0a78..cf2682972e 100644
--- a/makefile
+++ b/makefile
@@ -150,8 +150,8 @@ install:
 	$(INSTALL) -Dm644 switchres.h $(INCDIR)/switchres/switchres.h
 	$(INSTALL) -Dm644 switchres.pc $(PKGDIR)/switchres.pc
 ifneq ($(SO_NAME),)
-	$(LN) -s $(REAL_SO_NAME) $(LIBDIR)/$(SO_NAME)
-	$(LN) -s $(SO_NAME) $(LIBDIR)/$(LINKER_NAME)
+	$(LN) -sf $(REAL_SO_NAME) $(LIBDIR)/$(SO_NAME)
+	$(LN) -sf $(SO_NAME) $(LIBDIR)/$(LINKER_NAME)
 endif
 
 uninstall:
