diff --git a/makefile b/makefile
index 93f215f..051a4c1 100644
--- a/makefile
+++ b/makefile
@@ -10,7 +10,8 @@ OBJS = $(SRC:.cpp=.o)
 CROSS_COMPILE ?=
 CXX ?= g++
 AR ?= ar
-LDFLAGS = -shared
+LDFLAGS ?=
+LDFLAGS += -shared
 FINAL_CXX=$(CROSS_COMPILE)$(CXX)
 FINAL_AR=$(CROSS_COMPILE)$(AR)
 CPPFLAGS = -O3 -Wall -Wextra
@@ -20,9 +21,9 @@ INSTALL=install
 
 DESTDIR ?=
 PREFIX ?= /usr
-INCDIR = $(DESTDIR)$(PREFIX)/include
-LIBDIR = $(DESTDIR)$(PREFIX)/lib
-BINDIR = $(DESTDIR)$(PREFIX)/bin
+INCDIR = $(PREFIX)/include
+LIBDIR = $(PREFIX)/lib
+BINDIR = $(PREFIX)/bin
 PKGDIR = $(LIBDIR)/pkgconfig
 
 ifneq ($(DEBUG),)
@@ -68,8 +69,8 @@ CPPFLAGS += $(shell $(PKG_CONFIG) --cflags $(EXTRA_LIBS))
 CPPFLAGS += $(shell $(PKG_CONFIG) --libs $(EXTRA_LIBS))
 endif
 
-CPPFLAGS += -fPIC
-LIBS += -ldl
+CPPFLAGS += -fPIC -I$(INCDIR) -I$(INCDIR)/libdrm
+LIBS = -ldl -lm -lrt -lstdc++
 
 REMOVE = rm -f
 STATIC_LIB_EXT = a
