diff --color -uri switchres-lbrpdx/makefile switchres-a8af8e8f7f54d24414b4a8595a84c0f0f23e14a1/makefile
--- switchres-lbrpdx/makefile	2021-06-28 14:51:06.000000000 -0700
+++ switchres-a8af8e8f7f54d24414b4a8595a84c0f0f23e14a1/makefile	2021-09-10 03:36:52.380008814 -0700
@@ -10,7 +10,8 @@
 CROSS_COMPILE ?=
 CXX ?= g++
 AR ?= ar
-LDFLAGS = -shared
+LDFLAGS ?=
+LDFLAGS += -shared
 FINAL_CXX=$(CROSS_COMPILE)$(CXX)
 FINAL_AR=$(CROSS_COMPILE)$(AR)
 CPPFLAGS = -O3 -Wall -Wextra
@@ -20,9 +21,9 @@

 DESTDIR ?=
 PREFIX ?= /usr
-INCDIR = $(DESTDIR)$(PREFIX)/include
-LIBDIR = $(DESTDIR)$(PREFIX)/lib
-BINDIR = $(DESTDIR)$(PREFIX)/bin
+INCDIR = $(PREFIX)/include
+LIBDIR = $(PREFIX)/lib
+BINDIR = $(PREFIX)/bin
 PKGDIR = $(LIBDIR)/pkgconfig

 # Linux
@@ -48,12 +49,12 @@
     SRC += custom_video_drmkms.cpp
 endif

-ifneq (,$(EXTRA_LIBS))
-CPPFLAGS += $(shell $(PKG_CONFIG) --cflags $(EXTRA_LIBS))
-endif
+#ifneq (,$(EXTRA_LIBS))
+#CPPFLAGS += $(shell $(PKG_CONFIG) --cflags $(EXTRA_LIBS))
+#endif

-CPPFLAGS += -fPIC
-LIBS = -ldl
+CPPFLAGS += -fPIC -I$(INCDIR) -I$(INCDIR)/libdrm
+LIBS = -ldl -lm -lrt -lstdc++
 REMOVE = rm -f
 STATIC_LIB_EXT = a
 DYNAMIC_LIB_EXT = so
