--- a/makefile	2021-01-19 15:20:22.240850354 +0100
+++ b/makefile	2021-01-19 15:20:30.257516731 +0100
@@ -64,13 +64,7 @@
 clean:
 	$(REMOVE) $(OBJS) $(STANDALONE) $(TARGET_LIB).*
 
-prepare_pkg_config:
-	$(SED) -e "s+@prefix@+$(PREFIX)+g" \
-	  -e"s+@libdir@+$(LIBDIR)+g" \
-	  -e"s+@includedir@+$(INCDIR)+g" \
-	  switchres.pc.in > switchres.pc
-
-install: prepare_pkg_config
+install:
 	$(INSTALL) -Dm644 $(TARGET_LIB).$(DYNAMIC_LIB_EXT) $(LIBDIR)/$(TARGET_LIB).$(DYNAMIC_LIB_EXT)
 	$(INSTALL) -Dm644 switchres_wrapper.h $(INCDIR)/switchres/switchres_wrapper.h
 	$(INSTALL) -Dm644 switchres.h $(INCDIR)/switchres/switchres.h
