--- a/CPCCore/CPCCoreEmu/CMakeLists.txt	2023-08-18 19:48:03.571622053 +0200
+++ b/CPCCore/CPCCoreEmu/CMakeLists.txt	2023-08-18 19:48:26.812139963 +0200
@@ -49,5 +49,5 @@
 endif()
 
 if(CMAKE_COMPILER_IS_GNUCC OR CMAKE_CXX_COMPILER_ID MATCHES "Clang")
-	set ( CMAKE_CXX_FLAGS ${CMAKE_CXX_FLAGS} --std=c++1z )
+	set_property(TARGET CPCCoreEmu PROPERTY CXX_STANDARD 17)
 endif()
--- a/CPCCore/UnitTests/CMakeLists.txt	2023-08-18 19:48:12.307816760 +0200
+++ b/CPCCore/UnitTests/CMakeLists.txt	2023-08-18 19:48:34.604313568 +0200
@@ -56,7 +56,7 @@
 ENDIF(MSVC)
 
 if(CMAKE_COMPILER_IS_GNUCC OR CMAKE_CXX_COMPILER_ID MATCHES "Clang")
-	set ( CMAKE_CXX_FLAGS ${CMAKE_CXX_FLAGS} --std=c++1z)
+	set_property(TARGET unitTests PROPERTY CXX_STANDARD 17)
    if (CMAKE_CXX_COMPILER_ID MATCHES "Clang")
       set (LIBZA libz.a )
    else()
--- a/CPCCore/AmstradCore/CMakeLists.txt	2024-03-04 12:48:31.829479684 +0100
+++ b/CPCCore/AmstradCore/CMakeLists.txt	2024-03-04 12:49:41.227197343 +0100
@@ -48,5 +48,5 @@
 endif()
 
 if(CMAKE_COMPILER_IS_GNUCC OR CMAKE_CXX_COMPILER_ID MATCHES "Clang")
-	set ( CMAKE_CXX_FLAGS ${CMAKE_CXX_FLAGS} --std=c++1z )
+  set_property(TARGET AmstradCore PROPERTY CXX_STANDARD 17)
 endif()
--- a/CPCCore/AmstradCoreTests/CMakeLists.txt	2024-03-04 12:50:48.936852893 +0100
+++ b/CPCCore/AmstradCoreTests/CMakeLists.txt	2024-03-04 12:51:41.838134002 +0100
@@ -42,7 +42,7 @@
 ENDIF(MSVC)
 
 if(CMAKE_COMPILER_IS_GNUCC OR CMAKE_CXX_COMPILER_ID MATCHES "Clang")
-	set ( CMAKE_CXX_FLAGS ${CMAKE_CXX_FLAGS} --std=c++1z)
+   set_property(TARGET AmstradCoreTests PROPERTY CXX_STANDARD 17)
    if (CMAKE_CXX_COMPILER_ID MATCHES "Clang")
       set (LIBZA libz.a )
    else()
