diff --git a/src/duckstation-nogui/nogui_host_interface.cpp b/src/duckstation-nogui/nogui_host_interface.cpp
index 45c88625..bb938df2 100644
--- a/src/duckstation-nogui/nogui_host_interface.cpp
+++ b/src/duckstation-nogui/nogui_host_interface.cpp
@@ -209,6 +209,8 @@ bool NoGUIHostInterface::CreateDisplay()
     return false;
   }
 
+  m_display->SetDisplayRotation(HostDisplay::Rotation::R270Degrees);
+
   if (!m_display->CreateImGuiContext() ||
       (m_fullscreen_ui_enabled && !FullscreenUI::Initialize(this, m_settings_interface.get())) ||
       !m_display->UpdateImGuiFontTexture())
