--- a/drivers/hid/hid-input.c	2014-08-04 13:16:23.000000000 +0200
+++ b/drivers/hid/hid-input.c	2015-05-17 17:25:20.418536214 +0200
@@ -1100,8 +1100,10 @@
 	    (field->logical_minimum < field->logical_maximum) &&
 	    (value < field->logical_minimum ||
 	     value > field->logical_maximum)) {
-		dbg_hid("Ignoring out-of-range value %x\n", value);
-		return;
+            if(value < field->logical_minimum)
+                value = field->logical_minimum;
+            else
+                value = field->logical_maximum;
 	}
 
 	/* report the usage code as scancode if the key status has changed */
