--- a/3rdparty/FAudio/src/FAudio_internal.h	2021-06-30 21:14:13.000000000 +0200
+++ b/3rdparty/FAudio/src/FAudio_internal.h	2021-08-27 21:25:50.304110669 +0200
@@ -115,10 +115,10 @@ extern void FAudio_Log(char const *msg);
 #include <SDL3/SDL_endian.h>
 #include <SDL3/SDL_log.h>
 #else
-#include <SDL_stdinc.h>
-#include <SDL_assert.h>
-#include <SDL_endian.h>
-#include <SDL_log.h>
+#include <SDL2/SDL_stdinc.h>
+#include <SDL2/SDL_assert.h>
+#include <SDL2/SDL_endian.h>
+#include <SDL2/SDL_log.h>
 #endif
 
 #define FAudio_malloc SDL_malloc
--- a/3rdparty/FAudio/src/FAudio_platform_sdl2.c  2021-06-30 21:14:13.000000000 +0200
+++ b/3rdparty/FAudio/src/FAudio_platform_sdl2.c 2021-08-27 21:28:01.430818783 +0200
@@ -26,7 +26,7 @@
 
 #include "FAudio_internal.h"
 
-#include <SDL.h>
+#include <SDL2/SDL.h>
 
 #if !SDL_VERSION_ATLEAST(2, 0, 9)
 #error "SDL version older than 2.0.9"
