diff --git a/3rdparty/rtmidi/rtmidi/CMakeLists.txt b/3rdparty/rtmidi/rtmidi/CMakeLists.txt
index 3a85a03704..bb5f72cbb3 100644
--- a/3rdparty/rtmidi/rtmidi/CMakeLists.txt
+++ b/3rdparty/rtmidi/rtmidi/CMakeLists.txt
@@ -2,7 +2,7 @@
 # CopyPolicy: RtMidi license.
 
 # Set minimum CMake required version for this project.
-cmake_minimum_required(VERSION 3.24 FATAL_ERROR)
+cmake_minimum_required(VERSION 3.22 FATAL_ERROR)
 
 # Define a C++ project.
 project(RtMidi LANGUAGES CXX C)
