diff --git a/rpcs3/CMakeLists.txt b/rpcs3/CMakeLists.txt
index 70f4a83..6f4f98b 100644
--- a/rpcs3/CMakeLists.txt
+++ b/rpcs3/CMakeLists.txt
@@ -11,7 +11,7 @@
 set(CMAKE_CXX_STANDARD 20)
 
 set(ADDITIONAL_LIBS "")
-if(CMAKE_SYSTEM_NAME STREQUAL "Linux")
+if(TRUE)
     #on some Linux distros shm_unlink and similar functions are in librt only
     set(ADDITIONAL_LIBS ${ADDITIONAL_LIBS} "rt")
 elseif(NOT WIN32 AND NOT CMAKE_CXX_FLAGS MATCHES "LIBICONV_PLUG")
