diff --git a/rpcs3/rpcs3qt/main_window.cpp b/rpcs3/rpcs3qt/main_window.cpp
index 8a67c71..d4eecbb 100644
--- a/rpcs3/rpcs3qt/main_window.cpp
+++ b/rpcs3/rpcs3qt/main_window.cpp
@@ -1224,12 +1224,12 @@
 	}
 	else
 	{
-		if (QMessageBox::question(this, tr("RPCS3 Firmware Installer"), tr("Install firmware: %1?").arg(file_path),
-			QMessageBox::Yes | QMessageBox::No, QMessageBox::No) != QMessageBox::Yes)
-		{
-			gui_log.notice("Firmware: Cancelled installation from drop. File: %s", file_path);
-			return;
-		}
+		//if (QMessageBox::question(this, tr("RPCS3 Firmware Installer"), tr("Install firmware: %1?").arg(file_path),
+		//	QMessageBox::Yes | QMessageBox::No, QMessageBox::No) != QMessageBox::Yes)
+		//{
+		//	gui_log.notice("Firmware: Cancelled installation from drop. File: %s", file_path);
+		//	return;
+		//}
 	}
 
 	if (!file_path.isEmpty())
