diff --git a/rpcs3/Emu/savestate_utils.cpp b/rpcs3/Emu/savestate_utils.cpp
index 5add3e8..5ff1193 100644
--- a/rpcs3/Emu/savestate_utils.cpp
+++ b/rpcs3/Emu/savestate_utils.cpp
@@ -280,12 +280,12 @@ std::string get_savestate_file(std::string_view title_id, std::string_view boot_
 	if (aggregate_file_size == umax && rel_id == -1)
 	{
 		// Return directory
-		return fs::get_config_dir() + "savestates/" + title + "/";
+		return "/userdata/saves/ps3/" + title + "/";
 	}
 
 	if (rel_id >= 0)
 	{
-		std::string dir_path = fs::get_config_dir() + "savestates/" + title + "/";
+		std::string dir_path = "/userdata/saves/ps3/" + title + "/";
 
 		fs::dir dir_view{dir_path};
 
@@ -417,7 +417,7 @@ std::string get_savestate_file(std::string_view title_id, std::string_view boot_
 	// While not needing to keep a 59 chars long suffix at all times for this purpose
 	const char prefix = ::at32("0123456789ABCDEFGHJKLMNPQRSTUVWXYZabcdefghijkmnopqrstuvwxyz"sv, save_id.size());
 
-	std::string path = fs::get_config_dir() + "/savestates/" + title + "/" + title + '_' + prefix + '_' + save_id + ".SAVESTAT";
+	std::string path = "/userdata/saves/ps3/" + title + "/" + title + '_' + prefix + '_' + save_id + ".SAVESTAT";
 
 	if (std::string path_compressed = path + ".zst"; fs::is_file(path_compressed))
 	{