diff --git a/CMakeLists.txt b/CMakeLists.txt
index f1cdda8..9d2f06e 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -78,6 +78,7 @@ option(ENABLE_DC_PROFILER "Build with support for target machine (SH4) profiler"
 option(ENABLE_FC_PROFILER "Build with support for host app (Flycast) profiler" OFF)
 option(USE_DISCORD "Use Discord Presence API" OFF)
 option(USE_LIBCDIO "Use libcdio for CDROM access" OFF)
+option(USE_MALI "Use mali" OFF)
 
 if(IOS AND NOT LIBRETRO)
 	set(USE_VULKAN OFF CACHE BOOL "Force vulkan off" FORCE)
@@ -114,6 +115,10 @@ else()
 	set(GIT_HASH "000000000")
 endif()
 
+if(USE_MALI)
+  target_link_libraries(${PROJECT_NAME} PRIVATE "-lmali")
+endif()
+
 if(WINDOWS_STORE)
 	string(REGEX REPLACE "[Vv]" "" MS_VERSION ${GIT_VERSION})
 	string(REPLACE "-" "." MS_VERSION ${MS_VERSION})
