diff --git a/shell/linux/Makefile b/shell/linux/Makefile
index 522d45d..75d2935 100644
--- a/shell/linux/Makefile
+++ b/shell/linux/Makefile
@@ -166,6 +166,17 @@ else ifneq (,$(findstring lincpp,$(platform)))
     CFLAGS += -D TARGET_NO_JIT
     CXXFLAGS += -fexceptions -std=gnu++11
 
+# Raspberry Pi 3 aarch64 + mesa
+else ifneq (,$(findstring rpi3-64-mesa,$(platform)))
+    NOT_ARM := 1
+    ARM64_REC := 1
+    ISARM64 := 1
+    USE_GLES := 1
+    USE_SDL := 1
+    MFLAGS += -mcpu=cortex-a53 -mtune=cortex-a53
+    CFLAGS += -fno-builtin-sqrtf  -fsingle-precision-constant -fpermissive -DGLES3 -DEGL_NO_X11 
+    CXXFLAGS += -fexceptions
+
 # Raspberry Pi 4 aarch64 + mesa vulkan
 else ifneq (,$(findstring rpi4-64-mesa,$(platform)))
     NOT_ARM := 1
