diff --git a/CMakeLists.txt b/CMakeLists.txt
index 4517d3f09c..e63fb53e2d 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -300,7 +300,6 @@ if(PKG_CONFIG_FOUND AND NOT ANDROID AND NOT APPLE AND NOT LIBRETRO)
         target_link_libraries(${PROJECT_NAME} PRIVATE PkgConfig::AO)
     endif()
 
-	if(NOT SDL2_FOUND)
 	    pkg_check_modules(LIBEVDEV IMPORTED_TARGET libevdev)
 	    if(LIBEVDEV_FOUND)
 	        target_compile_definitions(${PROJECT_NAME} PRIVATE USE_EVDEV)
@@ -312,7 +311,6 @@ if(PKG_CONFIG_FOUND AND NOT ANDROID AND NOT APPLE AND NOT LIBRETRO)
 	            target_link_libraries(${PROJECT_NAME} PRIVATE PkgConfig::LIBUDEV)
 	        endif()
 	    endif()
-	endif()
 
     pkg_check_modules(LIBPULSE IMPORTED_TARGET libpulse)
     if(LIBPULSE_FOUND)
