diff --git a/core/input/gamepad_device.cpp b/core/input/gamepad_device.cpp
index e1bf9ea..d623ff8 100644
--- a/core/input/gamepad_device.cpp
+++ b/core/input/gamepad_device.cpp
@@ -37,6 +37,8 @@ s8 joyry[4];
 u8 rt[4];
 u8 lt[4];
 
+static int hotkey_pressed = 0;
+
 std::vector<std::shared_ptr<GamepadDevice>> GamepadDevice::_gamepads;
 std::mutex GamepadDevice::_gamepads_mutex;
 bool fast_forward_mode;
@@ -93,6 +95,17 @@ bool GamepadDevice::gamepad_btn_input(u32 code, bool pressed)
 			case DC_DPAD2_RIGHT:
 				kcode[_maple_port] |= (u16)DC_DPAD2_LEFT;
 				break;
+			case DC_BTN_START:
+				if (hotkey_pressed == 1)
+					dc_exit();
+				break;
+			case DC_BTN_A:
+				if (hotkey_pressed == 1)
+					gui_open_settings();
+				break;
+			case DC_BTN_D:
+				hotkey_pressed = 1;
+				break;
 			default:
 				break;
 			}
@@ -112,7 +112,9 @@
 			}
 		}
-		else
+		else {
+			hotkey_pressed = 0;
 			kcode[_maple_port] |= (u16)key;
+		}
 #ifdef TEST_AUTOMATION
 		if (record_input != NULL)
 			fprintf(record_input, "%ld button %x %04x\n", sh4_sched_now64(), _maple_port, kcode

