--- a/core/hw/arm7/arm7_rec_arm32.cpp	2021-06-20 21:42:35.482984094 +0200
+++ b/core/hw/arm7/arm7_rec_arm32.cpp	2021-06-20 21:43:10.866316062 +0200
@@ -55,13 +55,15 @@
 	ass.Str(cc, host_reg, MemOperand(r8, (u8*)&arm_Reg[guest_reg].I - (u8*)&arm_Reg[0].I));
 }
 
+#define MAX_REGS 6
+
 const std::array<Register, 6> alloc_regs{
 	r5, r6, r7, r9, r10, r11
 };
 
-class Arm32ArmRegAlloc : public ArmRegAlloc<alloc_regs.size(), Arm32ArmRegAlloc>
+class Arm32ArmRegAlloc : public ArmRegAlloc<MAX_REGS, Arm32ArmRegAlloc>
 {
-	using super = ArmRegAlloc<alloc_regs.size(), Arm32ArmRegAlloc>;
+	using super = ArmRegAlloc<MAX_REGS, Arm32ArmRegAlloc>;
 
 	void LoadReg(int host_reg, Arm7Reg armreg, ArmOp::Condition cc = ArmOp::AL)
 	{
