diff --git a/acinclude.m4 b/acinclude.m4.new
index 171e275f59..21b3fd589c 100644
--- a/acinclude.m4
+++ b/acinclude.m4.new
@@ -14,45 +14,11 @@ AC_ARG_ENABLE(sdl2,     [  --enable-sdl2           Enable SDL 2.x],
 
   SDL2_CONFIG=no
   if test x$enable_sdl2enable = xyes ; then
-    if test x$sdl2_exec_prefix != x ; then
-      sdl2_args="$sdl2_args --exec-prefix=$sdl2_exec_prefix"
-      if test x${SDL2_CONFIG+set} != xset ; then
-        SDL2_CONFIG=$sdl2_exec_prefix/bin/sdl2-config
-      fi
-    fi
-    if test x$sdl2_prefix != x ; then
-      sdl2_args="$sdl2_args --prefix=$sdl2_prefix"
-      if test x${SDL2_CONFIG+set} != xset ; then
-        SDL2_CONFIG=$sdl2_prefix/bin/sdl2-config
-      fi
-    fi
-
-    if test -x vs/sdl2/linux-host/bin/sdl2-config ; then
-      SDL2_CONFIG=vs/sdl2/linux-host/bin/sdl2-config
-      PATH=vs/sdl2/linux-host/bin:$PATH
-    fi
-
-    manual_sdl2config="$SDL2_CONFIG"
-
+    SDL2_CONFIG=$sdl2_prefix/bin/sdl2-config
     AC_PATH_PROG(SDL2_CONFIG, sdl2-config, no)
-    AC_MSG_CHECKING(for SDL2)
-
-    # Prefer pkg-config unless sdl2-config was
-    # manually specified before we looked for it
-    if test x$manual_sdl2config = xno && \
-        pkg-config --exists sdl2 ; then
-      SDL2_CFLAGS=`pkg-config sdl2 --cflags`
-      SDL2_LIBS=`pkg-config sdl2 --libs`
-      AC_DEFINE(C_SDL2,1)
-      AC_MSG_RESULT(found using pkg-config)
-    elif test x$SDL2_CONFIG != xno; then
-      SDL2_CFLAGS=`$SDL2_CONFIG $sdl2conf_args --cflags`
-      SDL2_LIBS=`$SDL2_CONFIG $sdl2conf_args --libs`
-      AC_DEFINE(C_SDL2,1)
-      AC_MSG_RESULT(found using sdl2-config)
-    else
-      AC_MSG_RESULT(not found)
-    fi
+    SDL2_CFLAGS=`$SDL2_CONFIG $sdl2conf_args --cflags`
+    SDL2_LIBS=`$SDL2_CONFIG $sdl2conf_args --libs`
+    AC_DEFINE(C_SDL2,1)
   fi
 
   AC_SUBST(SDL2_CFLAGS)
