--- a/src/gui/sdlmain.cpp	2020-08-03 06:33:31.000000000 +0200
+++ b/src/gui/sdlmain.cpp	2020-08-28 13:21:18.681580955 +0200
@@ -907,8 +907,8 @@
     PrintScreenSizeInfo();
 }
 
-#define MAPPERFILE_SDL1         "mapper-" VERSION ".map"
-#define MAPPERFILE_SDL2         "mapper-" VERSION ".sdl2.map"
+#define MAPPERFILE_SDL1         "mapper-sdl2.map"
+#define MAPPERFILE_SDL2         "mapper-sdl2.map"
 #if defined(C_SDL2)
 # define MAPPERFILE             MAPPERFILE_SDL2
 #else
@@ -954,7 +954,7 @@
 #elif defined(HAIKU)
 #define DEFAULT_CONFIG_FILE "~/config/settings/dosbox-x/dosbox-x.conf"
 #else /*linux freebsd*/
-# define DEFAULT_CONFIG_FILE            "/.dosboxrc"
+# define DEFAULT_CONFIG_FILE            "/dosboxx.conf"
 #endif
 
 #if C_SET_PRIORITY
--- a/src/misc/cross.cpp	2020-08-03 06:33:31.000000000 +0200
+++ b/src/misc/cross.cpp	2020-08-28 13:24:35.977009442 +0200
@@ -89,7 +89,7 @@
 #elif defined(RISCOS)
 	in = "/<Choices$Write>/DosBox-X";
 #elif !defined(HX_DOS)
-	in = "~/.config/dosbox-x";
+	in = "/userdata/system/configs/dosbox";
 	ResolveHomedir(in);
 #endif
 	in += CROSS_FILESPLIT;
@@ -101,7 +101,7 @@
 #elif defined(MACOSX)
 #define DEFAULT_CONFIG_FILE "DOSBox-X " VERSION " Preferences"
 #else /*linux freebsd*/
-#define DEFAULT_CONFIG_FILE "dosbox-x-" VERSION ".conf"
+#define DEFAULT_CONFIG_FILE "dosboxx.conf"
 #endif
 	in = DEFAULT_CONFIG_FILE;
 }
@@ -123,7 +123,7 @@
 	in = "/<Choices$Write>/DosBox-X";
 	mkdir(in.c_str(),0700);
 #elif !defined(HX_DOS)
-	in = "~/.config/dosbox-x";
+	in = "/userdata/system/configs/dosbox";;
 	ResolveHomedir(in);
 	mkdir(in.c_str(),0700);
 #endif
