diff --git a/arch/arm64/boot/dts/qcom/sm8250-retroidpocket-common.dtsi b/arch/arm64/boot/dts/qcom/sm8250-retroidpocket-common.dtsi
index 8b107da7e82e..6ca92ac0be96 100644
--- a/arch/arm64/boot/dts/qcom/sm8250-retroidpocket-common.dtsi
+++ b/arch/arm64/boot/dts/qcom/sm8250-retroidpocket-common.dtsi
@@ -50,6 +50,8 @@ fan: pwm-fan {
 
 	gpio-keys {
 		compatible = "gpio-keys";
+		pinctrl-names = "default";
+		pinctrl-0 = <&volume_up_default>;
 
 		key-vol-up {
 			label = "Volume Up";
@@ -902,6 +904,14 @@ wifi-therm@1 {
 	};
 };
 
+&pm8150_gpios {
+	volume_up_default: volume-up-default {
+		pins = "gpio6";
+		function = "normal";
+		bias-pull-up;
+	};
+};
+
 &pm8150b_adc {
 	channel@4f {
 		reg = <ADC5_AMUX_THM3_100K_PU>;
