diff -ru a/source/common.h b/source/common.h
--- a/source/common.h	2019-07-21 14:41:22.000000000 +0200
+++ b/source/common.h	2021-01-21 08:53:32.428520688 +0100
@@ -30,14 +30,14 @@
 #define I2C          42
 #define PWM          43
 
-int gpio_mode;
-const int pin_to_gpio_rev1[41];
-const int pin_to_gpio_rev2[41];
-const int pin_to_gpio_rev3[41];
-const int (*pin_to_gpio)[41];
-int gpio_direction[54];
-rpi_info rpiinfo;
-int setup_error;
-int module_setup;
-int check_gpio_priv(void);
-int get_gpio_number(int channel, unsigned int *gpio);
+extern int gpio_mode;
+extern const int pin_to_gpio_rev1[41];
+extern const int pin_to_gpio_rev2[41];
+extern const int pin_to_gpio_rev3[41];
+extern const int (*pin_to_gpio)[41];
+extern int gpio_direction[54];
+extern rpi_info rpiinfo;
+extern int setup_error;
+extern int module_setup;
+extern int check_gpio_priv(void);
+extern int get_gpio_number(int channel, unsigned int *gpio);
diff -ru a/source/constants.h source/constants.h
--- a/source/constants.h	2019-07-21 14:41:22.000000000 +0200
+++ b/source/constants.h	2021-01-21 08:54:06.105186261 +0100
@@ -23,22 +23,22 @@
 #define PY_PUD_CONST_OFFSET 20
 #define PY_EVENT_CONST_OFFSET 30
 
-PyObject *high;
-PyObject *low;
-PyObject *input;
-PyObject *output;
-PyObject *pwm;
-PyObject *serial;
-PyObject *i2c;
-PyObject *spi;
-PyObject *unknown;
-PyObject *board;
-PyObject *bcm;
-PyObject *pud_off;
-PyObject *pud_up;
-PyObject *pud_down;
-PyObject *rising_edge;
-PyObject *falling_edge;
-PyObject *both_edge;
+extern PyObject *high;
+extern PyObject *low;
+extern PyObject *input;
+extern PyObject *output;
+extern PyObject *pwm;
+extern PyObject *serial;
+extern PyObject *i2c;
+extern PyObject *spi;
+extern PyObject *unknown;
+extern PyObject *board;
+extern PyObject *bcm;
+extern PyObject *pud_off;
+extern PyObject *pud_up;
+extern PyObject *pud_down;
+extern PyObject *rising_edge;
+extern PyObject *falling_edge;
+extern PyObject *both_edge;
 
 void define_constants(PyObject *module);
diff -ru a/source/py_pwm.h b/source/py_pwm.h
--- a/source/py_pwm.h	2019-07-21 14:41:22.000000000 +0200
+++ b/source/py_pwm.h	2021-01-21 08:58:02.118511397 +0100
@@ -20,5 +20,5 @@
 SOFTWARE.
 */
 
-PyTypeObject PWMType;
+extern PyTypeObject PWMType;
 PyTypeObject *PWM_init_PWMType(void);
diff -ru a/source/soft_pwm.c b/source/soft_pwm.c
--- a/source/soft_pwm.c	2019-07-21 14:41:22.000000000 +0200
+++ b/source/soft_pwm.c	2021-01-21 08:57:44.028512017 +0100
@@ -25,7 +25,7 @@
 #include <time.h>
 #include "c_gpio.h"
 #include "soft_pwm.h"
-pthread_t threads;
+extern pthread_t threads;
 
 struct pwm
 {
