diff --git a/Makefile b/Makefile
index 35aca329b6..a873c4973a 100644
--- a/Makefile
+++ b/Makefile
@@ -1,17 +1,17 @@
 SHELL := /bin/bash
-EXTRA_CFLAGS += $(USER_EXTRA_CFLAGS)
-EXTRA_CFLAGS += -O1
+ccflags-y += $(USER_EXTRA_CFLAGS)
+ccflags-y += -O1
 
-EXTRA_CFLAGS += -Wno-unused-variable
-EXTRA_CFLAGS += -Wno-unused-value
-EXTRA_CFLAGS += -Wno-unused-label
-EXTRA_CFLAGS += -Wno-unused-parameter
-EXTRA_CFLAGS += -Wno-unused-function
-EXTRA_CFLAGS += -Wno-unused
+ccflags-y += -Wno-unused-variable
+ccflags-y += -Wno-unused-value
+ccflags-y += -Wno-unused-label
+ccflags-y += -Wno-unused-parameter
+ccflags-y += -Wno-unused-function
+ccflags-y += -Wno-unused
 
-EXTRA_CFLAGS += -Wno-uninitialized
+ccflags-y += -Wno-uninitialized
 
-EXTRA_CFLAGS += -I$(src)/include
+ccflags-y += -I$(src)/include
 
 ccflags-y += -D__CHECK_ENDIAN__
 
@@ -87,11 +87,11 @@ $(shell cp $(TopDIR)/autoconf_rtl8188e_usb_linux.h $(TopDIR)/include/autoconf.h)
 endif
 
 ifeq ($(CONFIG_BT_COEXIST), y)
-EXTRA_CFLAGS += -DCONFIG_BT_COEXIST
+ccflags-y += -DCONFIG_BT_COEXIST
 endif
 
 ifeq ($(CONFIG_WOWLAN), y)
-EXTRA_CFLAGS += -DCONFIG_WOWLAN
+ccflags-y += -DCONFIG_WOWLAN
 endif
 
 SUBARCH := $(shell uname -m | sed -e "s/i.86/i386/; s/ppc.*/powerpc/; s/armv.l/arm/; s/aarch64/arm64/;")
diff --git a/include/osdep_service.h b/include/osdep_service.h
index 627f17b7ea..921e406e85 100644
--- a/include/osdep_service.h
+++ b/include/osdep_service.h
@@ -112,7 +112,11 @@ static inline void _set_timer(struct timer_list *ptimer,u32 delay_time)
 
 static inline void _cancel_timer(struct timer_list *ptimer,u8 *bcancelled)
 {
+#if (LINUX_VERSION_CODE >= KERNEL_VERSION(6, 15, 0))
+    timer_delete_sync(ptimer);
+#else
 	del_timer_sync(ptimer);
+#endif
 	*bcancelled=  true;/* true ==1; false==0 */
 }
 
@@ -286,7 +290,11 @@ void rtw_yield_os(void);
 
 static inline unsigned char _cancel_timer_ex(struct timer_list *ptimer)
 {
+#if (LINUX_VERSION_CODE >= KERNEL_VERSION(6, 15, 0))
+    return timer_delete_sync(ptimer);
+#else
 	return del_timer_sync(ptimer);
+#endif
 }
 
 static __inline void thread_enter(char *name)
