diff --git a/Makefile.libretro b/Makefile.libretro
index 854b4f1..53a0cd8 100644
--- a/Makefile.libretro
+++ b/Makefile.libretro
@@ -87,6 +87,23 @@ else ifeq ($(platform), classic_armv7_a7)
 	endif
 #######################################
 
+# ARMV
+else ifneq (,$(findstring armv,$(platform)))
+   TARGET := $(TARGET_NAME)_libretro.so
+   fpic := -fPIC
+   SHARED := -shared -Wl,--no-undefined 
+   LDFLAGS += -shared -Wl,--version-script=./libretro/link.T
+   ifneq (,$(findstring neon,$(platform)))
+      CFLAGS += -mfpu=neon
+      HAVE_NEON = 1
+   endif
+   ifneq (,$(findstring softfloat,$(platform)))
+      CFLAGS += -mfloat-abi=softfp
+   else ifneq (,$(findstring hardfloat,$(platform)))
+      CFLAGS += -mfloat-abi=hard
+   endif
+      CFLAGS += -DARM -marm -DALIGN_DWORD -mstructure-size-boundary=32 -mthumb-interwork -falign-functions=16 -pipe -fstack-protector
+      
 # Nintendo Switch (libnx)
 else ifeq ($(platform), libnx)
    include $(DEVKITPRO)/libnx/switch_rules
