diff --git a/bsnes/GNUmakefile b/bsnes/GNUmakefile
index f017341..81fb6f3 100644
--- a/bsnes/GNUmakefile
+++ b/bsnes/GNUmakefile
@@ -1,9 +1,9 @@
 target := bsnes
 binary := application
 build := performance
-openmp := true
-local := true
-flags += -I. -I..
+openmp := false
+local := false
+flags += -I. -I.. -Wno-narrowing -Wno-multichar -g -fPIC ${CXXFLAGS}
 
 # in order for this to work, obj/lzma.o must be omitted or bsnes will hang on startup.
 # further, only the X-Video driver works reliably. OpenGL 3.2, OpenGL 2.0, and XShm crash bsnes.
diff --git a/bsnes/target-libretro/GNUmakefile b/bsnes/target-libretro/GNUmakefile
index 4c4cf7f..8d248ec 100644
--- a/bsnes/target-libretro/GNUmakefile
+++ b/bsnes/target-libretro/GNUmakefile
@@ -1,5 +1,5 @@
 name := libretro.so
-flags += -Wno-narrowing -Wno-multichar -fopenmp -g -fPIC
+flags += -Wno-narrowing -Wno-multichar -g -fPIC
 
 objects := libretro $(objects)
 objects := $(patsubst %,obj/%.o,$(objects))
@@ -8,9 +8,9 @@ obj/libretro.o: target-libretro/libretro.cpp
 
 all: $(objects)
 ifeq ($(platform),linux)
-	$(strip $(compiler) -o out/bsnes_libretro.so -shared $(objects) -Wl,--no-undefined -Wl,--version-script=target-libretro/link.T -Wl,-Bdynamic -lpthread -ldl -lgomp)
+	$(strip $(compiler) -o out/bsnes_libretro.so -shared $(objects) -Wl,--no-undefined -Wl,--version-script=target-libretro/link.T -Wl,-Bdynamic -lpthread -ldl)
 else ifeq ($(platform),windows)
-	$(strip $(compiler) -o out/bsnes_libretro.dll -shared $(objects) -Wl,--no-undefined -Wl,--version-script=target-libretro/link.T -static-libgcc -static-libstdc++ -Wl,-Bstatic -lws2_32 -lpthread -lgomp -Wl,-Bdynamic)
+	$(strip $(compiler) -o out/bsnes_libretro.dll -shared $(objects) -Wl,--no-undefined -Wl,--version-script=target-libretro/link.T -static-libgcc -static-libstdc++ -Wl,-Bstatic -lws2_32 -lpthread -Wl,-Bdynamic)
 else ifeq ($(platform),macos)
 	$(strip $(compiler) -o out/bsnes_libretro.dylib -shared $(objects) -lpthread -ldl)
 endif
