diff --git a/src/Cafe/Filesystem/FST/KeyCache.cpp b/src/Cafe/Filesystem/FST/KeyCache.cpp
index 5d8d51c..92678e4 100644
--- a/src/Cafe/Filesystem/FST/KeyCache.cpp
+++ b/src/Cafe/Filesystem/FST/KeyCache.cpp
@@ -59,7 +59,15 @@ void KeyCache_Prepare()
 	sKeyCachePrepared = true;
 	g_keyCache.clear();
 	// load keys
-	auto keysPath = ActiveSettings::GetUserDataPath("keys.txt");
+	// Use /userdata/bios/cemu for the keys.txt directory, to match batocera
+	// documentation at https://wiki.batocera.org/systems:wiiu#keystxt
+
+	std::string keysDir = "/userdata/bios/cemu";
+	// Let's first make sure the parent directory exists.
+	std::error_code err;
+	fs::create_directories(keysDir, err);
+
+	auto keysPath = keysDir + "/keys.txt";
 	FileStream* fs_keys = FileStream::openFile2(keysPath);
 	if( !fs_keys )
 	{
