--- a/xbmc/cores/VideoPlayer/VideoRenderers/HwDecRender/VideoLayerBridgeDRMPRIME.h	2021-03-16 21:50:30.832274276 +0100
+++ b/xbmc/cores/VideoPlayer/VideoRenderers/HwDecRender/VideoLayerBridgeDRMPRIME.h	2021-03-16 21:50:46.648940708 +0100
@@ -15,34 +15,6 @@
 
 #include <drm_mode.h>
 
-#ifndef HAVE_HDR_OUTPUT_METADATA
-// HDR structs is copied from linux include/linux/hdmi.h
-struct hdr_metadata_infoframe
-{
-  uint8_t eotf;
-  uint8_t metadata_type;
-  struct
-  {
-    uint16_t x, y;
-  } display_primaries[3];
-  struct
-  {
-    uint16_t x, y;
-  } white_point;
-  uint16_t max_display_mastering_luminance;
-  uint16_t min_display_mastering_luminance;
-  uint16_t max_cll;
-  uint16_t max_fall;
-};
-struct hdr_output_metadata
-{
-  uint32_t metadata_type;
-  union {
-    struct hdr_metadata_infoframe hdmi_metadata_type1;
-  };
-};
-#endif
-
 namespace KODI
 {
 namespace WINDOWING
