diff -ur ./3rdparty/minimp3/minimp3.h /home/romain/batocera.linux/output/odroidn2/build/mame-mame0226/3rdparty/minimp3/minimp3.h
--- a/3rdparty/minimp3/minimp3.h	2020-10-27 02:00:36.000000000 +0100
+++ b/3rdparty/minimp3/minimp3.h	2020-11-09 18:39:21.731361951 +0100
@@ -84,6 +84,10 @@
 #define MINIMP3_MIN(a, b)           ((a) > (b) ? (b) : (a))
 #define MINIMP3_MAX(a, b)           ((a) < (b) ? (b) : (a))
 
+#ifndef HAVE_SSE
+#define HAVE_SSE 0
+#endif
+
 #if !defined(MINIMP3_NO_SIMD)
 
 #if !defined(MINIMP3_ONLY_SIMD) && (defined(_M_X64) || defined(_M_ARM64) || defined(__x86_64__) || defined(__aarch64__))
diff -ur ./makefile /home/romain/batocera.linux/output/odroidn2/build/mame-mame0226/makefile
--- a/makefile	2020-10-27 02:00:36.000000000 +0100
+++ b/makefile	2020-11-09 18:56:15.179960037 +0100
@@ -33,6 +33,7 @@
 # SDL2_MULTIAPI = 1
 # NO_USE_MIDI = 1
 # NO_USE_PORTAUDIO = 1
+# NO_USE_BGFX_KHRONOS =
 # USE_TAPTUN = 1
 # USE_PCAP = 1
 # USE_QTDEBUG = 1
@@ -86,10 +87,13 @@
 # BUILDDIR = build
 # TARGETOS = windows
 # CROSS_BUILD = 1
+# CROSS_ARCH =
 # TOOLCHAIN =
 # OVERRIDE_CC = cc
 # OVERRIDE_CXX = c++
 # OVERRIDE_LD = ld
+# OVERRIDE_AR = ar
+# OVERRIDE_STRIP = strip
 
 # DEPRECATED = 0
 # LTO = 1
@@ -329,6 +333,10 @@
 
 else
 CROSS_BUILD := 1
+ifdef CROSS_ARCH
+PLATFORM := $(CROSS_ARCH)
+UNAME := $(CROSS_ARCH)
+endif
 endif # TARGET_OS
 
 ifdef PTR64
@@ -573,6 +581,18 @@
 LD := $(OVERRIDE_LD)
 endif
 endif
+ifdef OVERRIDE_AR
+PARAMS += --AR='$(OVERRIDE_AR)'
+ifndef CROSS_BUILD
+AR := $(OVERRIDE_AR)
+endif
+endif
+ifdef OVERRIDE_STRIP
+PARAMS += --STRIP='$(OVERRIDE_STRIP)'
+ifndef CROSS_BUILD
+STRIP := $(OVERRIDE_STRIP)
+endif
+endif
 
 #-------------------------------------------------
 # sanity check the configuration
@@ -740,6 +760,10 @@
 PARAMS += --NO_USE_MIDI='$(NO_USE_MIDI)'
 endif
 
+ifdef NO_USE_BGFX_KHRONOS
+PARAMS += --NO_USE_BGFX_KHRONOS='$(NO_USE_BGFX_KHRONOS)'
+endif
+
 ifdef NO_USE_PORTAUDIO
 PARAMS += --NO_USE_PORTAUDIO='$(NO_USE_PORTAUDIO)'
 endif
diff -ur ./scripts/genie.lua /home/romain/batocera.linux/output/odroidn2/build/mame-mame0226/scripts/genie.lua
--- a/scripts/genie.lua	2020-10-27 02:00:36.000000000 +0100
+++ b/scripts/genie.lua	2020-11-09 19:09:38.958845876 +0100
@@ -195,6 +195,16 @@
 }
 
 newoption {
+	trigger = "AR",
+	description = "AR replacement",
+}
+
+newoption {
+	trigger = "STRIP",
+	description = "STRIP replacement",
+}
+
+newoption {
 	trigger = "TOOLCHAIN",
 	description = "Toolchain prefix"
 }
@@ -390,6 +400,11 @@
 }
 
 newoption {
+	trigger = "NO_USE_BGFX_KHRONOS",
+	description = "Skip including bgfx 3rdparty khronos headers (rely on system ones)",
+}
+
+newoption {
 	trigger = "DEBUG_DIR",
 	description = "Default directory for debugger.",
 }
diff -ur ./scripts/src/3rdparty.lua /home/romain/batocera.linux/output/odroidn2/build/mame-mame0226/scripts/src/3rdparty.lua
--- a/scripts/src/3rdparty.lua	2020-10-27 02:00:36.000000000 +0100
+++ b/scripts/src/3rdparty.lua	2020-11-09 18:03:17.288620257 +0100
@@ -1327,9 +1327,11 @@
 	}
 
 	configuration { "not steamlink"}
-		includedirs {
-			MAME_DIR .. "3rdparty/bgfx/3rdparty/khronos",
-		}
+	        if not _OPTIONS["NO_USE_BGFX_KHRONOS"] then
+			includedirs {
+				MAME_DIR .. "3rdparty/bgfx/3rdparty/khronos",
+			}
+		end
 
 	configuration { "android-*"}
 		buildoptions {
@@ -2121,8 +2123,12 @@
 	configuration { "mingw*"}
 		includedirs {
 			MAME_DIR .. "3rdparty/SDL2-override/mingw",
-			MAME_DIR .. "3rdparty/bgfx/3rdparty/khronos",
 		}
+                if not _OPTIONS["NO_USE_BGFX_KHRONOS"] then
+			includedirs {
+				MAME_DIR .. "3rdparty/bgfx/3rdparty/khronos",
+			}
+		end
 		buildoptions_c {
 			"-Wno-undef",
 			"-Wno-format",
diff -ur ./scripts/src/osd/modules.lua /home/romain/batocera.linux/output/odroidn2/build/mame-mame0226/scripts/src/osd/modules.lua
--- a/scripts/src/osd/modules.lua	2020-10-27 02:00:36.000000000 +0100
+++ b/scripts/src/osd/modules.lua	2020-11-09 17:59:35.068394429 +0100
@@ -255,11 +255,16 @@
 		MAME_DIR .. "3rdparty/bgfx/examples/common",
 		MAME_DIR .. "3rdparty/bgfx/include",
 		MAME_DIR .. "3rdparty/bgfx/3rdparty",
-		MAME_DIR .. "3rdparty/bgfx/3rdparty/khronos",
 		MAME_DIR .. "3rdparty/bx/include",
 		ext_includedir("rapidjson")
 	}
 
+	if not _OPTIONS["NO_USE_BGFX_KHRONOS"] then
+		includedirs {
+			MAME_DIR .. "3rdparty/bgfx/3rdparty/khronos",
+		}
+	end
+
 	if _OPTIONS["NO_USE_PORTAUDIO"]=="1" then
 		defines {
 			"NO_USE_PORTAUDIO",
diff -ur ./scripts/toolchain.lua /home/romain/batocera.linux/output/odroidn2/build/mame-mame0226/scripts/toolchain.lua
--- a/scripts/toolchain.lua	2020-10-27 02:00:36.000000000 +0100
+++ b/scripts/toolchain.lua	2020-11-09 19:09:29.890677516 +0100
@@ -207,7 +207,11 @@
 				premake.gcc.cc   = "@gcc -V 4.2"
 				premake.gcc.cxx  = "@g++-4.2"
 			end
-			premake.gcc.ar  = "ar"
+			if _OPTIONS["AR"]~=nil then
+				premake.gcc.ar = _OPTIONS["AR"]
+			else
+				premake.gcc.ar  = "ar"
+			end
 			location (_buildDir .. "projects/" .. _subDir .. "/".. _ACTION .. "-linux")
 		end
 
@@ -737,6 +741,18 @@
 	configuration { "linux-gcc", "x64", "Debug" }
 		targetdir (_buildDir .. "linux_gcc" .. "/bin/x64/Debug")
 
+	configuration { "linux-gcc", "arm64" }
+		objdir (_buildDir .. "linux_gcc" .. "/obj")
+		buildoptions {
+			"-march=armv8-a",
+		}
+
+	configuration { "linux-gcc", "arm64", "Release" }
+		targetdir (_buildDir .. "linux_gcc" .. "/bin/arm64/Release")
+
+	configuration { "linux-gcc", "arm64", "Debug" }
+		targetdir (_buildDir .. "linux_gcc" .. "/bin/arm64/Debug")
+
 	configuration { "linux-clang", "x32" }
 		objdir (_buildDir .. "linux_clang" .. "/obj")
 		buildoptions {
@@ -860,8 +876,12 @@
 
 	configuration { "android-*" }
 		objdir (_buildDir .. "android/obj/" .. _OPTIONS["PLATFORM"])
+	        if not _OPTIONS["NO_USE_BGFX_KHRONOS"] then
+			includedirs {
+				MAME_DIR .. "3rdparty/bgfx/3rdparty/khronos",
+			}
+		end
 		includedirs {
-			MAME_DIR .. "3rdparty/bgfx/3rdparty/khronos",
 			"$(ANDROID_NDK_ROOT)/sources/cxx-stl/llvm-libc++/libcxx/include",
 			"$(ANDROID_NDK_ROOT)/sources/cxx-stl/llvm-libc++/include",
 			"$(ANDROID_NDK_ROOT)/sysroot/usr/include",
@@ -1112,10 +1132,17 @@
 		}
 
 	configuration { "linux-* or rpi" }
-		postbuildcommands {
-			"$(SILENT) echo Stripping symbols.",
-			"$(SILENT) strip -s \"$(TARGET)\""
-		}
+		if _OPTIONS['STRIP']~=nil then
+			postbuildcommands {
+				"$(SILENT) echo Stripping symbols.",
+				"$(SILENT) " .. _OPTIONS['STRIP'] .. " -s \"$(TARGET)\""
+			}
+		else
+			postbuildcommands {
+				"$(SILENT) echo Stripping symbols.",
+				"$(SILENT) strip -s \"$(TARGET)\""
+			}
+		end
 
 	configuration { "mingw*", "x64" }
 		postbuildcommands {
diff -ur ./src/osd/modules/render/drawbgfx.cpp /home/romain/batocera.linux/output/odroidn2/build/mame-mame0226/src/osd/modules/render/drawbgfx.cpp
--- a/src/osd/modules/render/drawbgfx.cpp	2020-10-27 02:00:36.000000000 +0100
+++ b/src/osd/modules/render/drawbgfx.cpp	2020-11-09 18:45:54.759139263 +0100
@@ -169,7 +169,11 @@
 	}
 
 #   if BX_PLATFORM_LINUX || BX_PLATFORM_BSD || BX_PLATFORM_RPI
+#ifdef SDLMAME_NO_X11
+	return nullptr;
+#else
 	return (void*)wmi.info.x11.window;
+#endif
 #   elif BX_PLATFORM_OSX
 	return wmi.info.cocoa.window;
 #   elif BX_PLATFORM_WINDOWS
@@ -192,8 +196,10 @@
 
 	bgfx::PlatformData pd;
 #   if BX_PLATFORM_LINUX || BX_PLATFORM_BSD
+#ifndef SDLMAME_NO_X11
 	pd.ndt          = wmi.info.x11.display;
 	pd.nwh          = (void*)(uintptr_t)wmi.info.x11.window;
+#endif
 #   elif BX_PLATFORM_OSX
 	pd.ndt          = NULL;
 	pd.nwh          = wmi.info.cocoa.window;
