diff --git a/src/frontend/mame/ui/ui.cpp b/src/frontend/mame/ui/ui.cpp.new
index dab4a5c84a..55a6ebeda1 100644
--- a/src/frontend/mame/ui/ui.cpp
+++ b/src/frontend/mame/ui/ui.cpp.new
@@ -643,6 +643,8 @@ void mame_ui_manager::display_startup_screens(bool first_time)
 	// or if we are debugging, or if there's no mame window to send inputs to
 	if (!first_time || (str > 0 && str < 60*5) || &machine().system() == &GAME_NAME(___empty) || (machine().debug_flags & DEBUG_FLAG_ENABLED) || video_none)
 		show_gameinfo = show_warnings = false;
+	
+	show_gameinfo = show_warnings = false;
 
 #if defined(__EMSCRIPTEN__)
 	// also disable for the JavaScript port since the startup screens do not run asynchronously
