--- a/src/mame/arcade.flt	2020-10-27 02:00:36.000000000 +0100
+++ b/src/mame/arcade.flt	2020-10-28 03:45:15.000000000 +0100
@@ -1,65 +1,65 @@
-1942.cpp
-1943.cpp
-1945kiii.cpp
-20pacgal.cpp
-24cdjuke.cpp
-2mindril.cpp
-30test.cpp
-39in1.cpp
-3do.cpp
-3x3puzzl.cpp
-40love.cpp
-4enlinea.cpp
-4enraya.cpp
-4roses.cpp
-5clown.cpp
-8080bw.cpp
-88games.cpp
-a1supply.cpp
-ace.cpp
-acefruit.cpp
-aces1.cpp
-acesp.cpp
-acommand.cpp
-actfancr.cpp
-adp.cpp
-aeroboto.cpp
-aerofgt.cpp
-age_candy.cpp
+1942.cpp           // ok
+1943.cpp           // ok
+1945kiii.cpp       // ok
+20pacgal.cpp       // ok
+//24cdjuke.cpp     // mechanical
+2mindril.cpp       // ok
+30test.cpp         // ok
+39in1.cpp          // ok
+//3do.cpp          // chd
+3x3puzzl.cpp       // ok
+40love.cpp         // ok
+//4enlinea.cpp     // check
+4enraya.cpp        // ok
+//4roses.cpp       // check
+//5clown.cpp       // poker
+8080bw.cpp         // ok
+88games.cpp        // ok
+//a1supply.cpp     // check
+ace.cpp            // ok
+//acefruit.cpp     // fruit
+//aces1.cpp        // fruit
+//acesp.cpp        // fruit
+acommand.cpp       // ok
+actfancr.cpp       // ok
+//adp.cpp            // ok
+aeroboto.cpp       // ok
+aerofgt.cpp        // ok
+//age_candy.cpp    // mechanical
 airbustr.cpp
-airraid.cpp
+//airraid.cpp
 ajax.cpp
 akkaarrh.cpp
 albazc.cpp
-albazg.cpp
+//albazg.cpp
 aleck64.cpp
-aleisttl.cpp
-alg.cpp
-alien.cpp
+//aleisttl.cpp     // ttl
+//alg.cpp
+//alien.cpp
 aliens.cpp
 alinvade.cpp
-allied.cpp
+//allied.cpp       // pinball
 alpha68k.cpp
 alpha68k_i.cpp
 alpha68k_n.cpp
-alvg.cpp
-amaticmg.cpp
-ambush.cpp
-amerihok.cpp
-ampoker2.cpp
+//alvg.cpp         // pinball
+//amaticmg.cpp
+ambush.cpp         // ok
+//amerihok.cpp     // check
+//ampoker2.cpp     // poker
 amspdwy.cpp
-amusco.cpp
-anes.cpp
+//amusco.cpp       // poker
+//anes.cpp         // check
 angelkds.cpp
 appoooh.cpp
 aquarium.cpp
 arabian.cpp
-arachnid.cpp
+//arachnid.cpp
 arcadecl.cpp
 argus.cpp
-aristmk4.cpp
-aristmk5.cpp
-aristmk6.cpp
+//aristmk4.cpp     // fruit
+//aristmk5.cpp     // fruit
+//aristmk6.cpp     // fruit
 arkanoid.cpp
 armedf.cpp
 arsystems.cpp
@@ -68,11 +68,11 @@
 asterix.cpp
 asteroid.cpp
 astinvad.cpp
-astrafr.cpp
-astrcorp.cpp
+//astrafr.cpp      // fruit
+//astrcorp.cpp
 astrocde.cpp
 astrof.cpp
-astropc.cpp
+//astropc.cpp      // check
 asuka.cpp
 atarifb.cpp
 atarig1.cpp
@@ -81,28 +81,28 @@
 atarigx2.cpp
 atarisy1.cpp
 atarisy2.cpp
-atarisy4.cpp
-atarittl.cpp
-atari_s1.cpp
-atari_s2.cpp
+//atarisy4.cpp
+//atarittl.cpp     // ttl
+//atari_s1.cpp     // pinball
+//atari_s2.cpp     // pinball
 atetris.cpp
-atlantis.cpp
-atronic.cpp
+//atlantis.cpp     // chd
+//atronic.cpp      // check
 attckufo.cpp
-atvtrack.cpp
+//atvtrack.cpp     // check
 avalnche.cpp
-avt.cpp
+//avt.cpp
 aztarac.cpp
 backfire.cpp
 badlands.cpp
 badlandsbl.cpp
 bagman.cpp
-bailey.cpp
+//bailey.cpp       // check
 balsente.cpp
 bankp.cpp
 baraduke.cpp
 barata.cpp
-barni.cpp
+//barni.cpp        // pinball
 bartop52.cpp
 batman.cpp
 battlane.cpp
@@ -113,34 +113,34 @@
 beaminv.cpp
 beathead.cpp
 beezer.cpp
-belatra.cpp
+//belatra.cpp      // check
 berzerk.cpp
 bestleag.cpp
 bfcobra.cpp
-bfmsys83.cpp
-bfmsys85.cpp
-bfm_ad5sw.cpp
-bfm_sc1.cpp
-bfm_sc2.cpp
-bfm_sc4.cpp
-bfm_sc5sw.cpp
-bfm_swp.cpp
-bgt.cpp
+//bfmsys83.cpp     // fruit
+//bfmsys85.cpp     // fruit
+//bfm_ad5sw.cpp      // fruit
+//bfm_sc1.cpp      // fruit
+//bfm_sc2.cpp      // fruit
+//bfm_sc4.cpp      // fruit
+//bfm_sc5sw.cpp      // fruit
+//bfm_swp.cpp      // fruit
+//bgt.cpp          // fruit, skeleton
 big10.cpp
 bigevglf.cpp
 bigkarnk_ms.cpp
 bigstrkb.cpp
-bingo.cpp
-bingoc.cpp
-bingoman.cpp
-bingor.cpp
-bingowav.cpp
+//bingo.cpp        // check
+//bingoc.cpp       // check
+//bingoman.cpp     // check
+//bingor.cpp       // check
+//bingowav.cpp     // check
 bionicc.cpp
 bishi.cpp
 bking.cpp
 blackt96.cpp
 bladestl.cpp
-blitz68k.cpp
+//blitz68k.cpp     // fruit&poker
 blktiger.cpp
 blktiger_ms.cpp
 blmbycar.cpp
@@ -160,7 +160,7 @@
 bombjack.cpp
 boogwing.cpp
 bottom9.cpp
-bowltry.cpp
+//bowltry.cpp      // check
 boxer.cpp
 brkthru.cpp
 bsktball.cpp
@@ -168,40 +168,40 @@
 btoads.cpp
 bublbobl.cpp
 buggychl.cpp
-buster.cpp
+//buster.cpp       // fruit
 bwidow.cpp
 bwing.cpp
-by17.cpp
-by35.cpp
-by6803.cpp
-by68701.cpp
-byvid.cpp
 bzone.cpp
+//by17.cpp         // pinball
+//by35.cpp         // pinball
+//by6803.cpp       // pinball
+//by68701.cpp      // pinball
+//byvid.cpp          // pinball-video
 cabal.cpp
-cabaret.cpp
+//cabaret.cpp      // poker
 calchase.cpp
-calcune.cpp
-calomega.cpp
+calcune.cpp             // ok
+//calomega.cpp     // poker
 calorie.cpp
 canyon.cpp
 capbowl.cpp
-capcom.cpp
-capr1.cpp
-caprcyc.cpp
+//capcom.cpp       // check
+//capr1.cpp        // check
+//caprcyc.cpp      // check
 cardline.cpp
 carjmbre.cpp
 carpolo.cpp
 carrera.cpp
-castle.cpp
+//castle.cpp       // check
 caswin.cpp
 cave.cpp
 cavepc.cpp
-cb2001.cpp
+//cb2001.cpp       // fruit
 cball.cpp
 cbasebal.cpp
 cbuster.cpp
 ccastles.cpp
-cchance.cpp
+//cchance.cpp      // fruit
 cchasm.cpp
 cclimber.cpp
 cdi.cpp
@@ -217,15 +217,15 @@
 chanbara.cpp
 chance32.cpp
 changela.cpp
-changyu.cpp
+//changyu.cpp      // skeleton gambling
 cheekyms.cpp
-chexx.cpp
-chicago.cpp
-chihiro.cpp
+//chexx.cpp        // mechanical
+//chicago.cpp      // ttl
+//chihiro.cpp      // chd
 chinagat.cpp
 chinsan.cpp
 chqflag.cpp
-chsuper.cpp
+//chsuper.cpp      // poker
 cidelsa.cpp
 cinemat.cpp
 circus.cpp
@@ -236,8 +236,8 @@
 cliffhgr.cpp
 cloak.cpp
 cloud9.cpp
-clowndwn.cpp
-clpoker.cpp
+//clowndwn.cpp     // mechanical
+//clpoker.cpp      // poker
 clshroad.cpp
 cmmb.cpp
 cninja.cpp
@@ -245,7 +245,7 @@
 cobra.cpp
 cocoloco.cpp
 coinmstr.cpp
-coinmvga.cpp
+//coinmvga.cpp     // check
 combatsc.cpp
 comebaby.cpp
 commando.cpp
@@ -254,32 +254,32 @@
 coolpool.cpp
 coolridr.cpp
 cop01.cpp
-cops.cpp
+//cops.cpp         // check
 copsnrob.cpp
 corona.cpp
 cosmic.cpp
-cowtipping.cpp
+//cowtipping.cpp     // skeleton
 cps1.cpp
 cps1bl_5205.cpp
 cps1bl_pic.cpp
 cps2.cpp
 cps3.cpp
 cpzodiac.cpp
-crazybal.cpp
+//crazybal.cpp     // TTL
 crbaloon.cpp
 crgolf.cpp
 crimfght.cpp
-cromptons.cpp
+//cromptons.cpp    // mechanical
 crospang.cpp
 crospuzl.cpp
 crshrace.cpp
 crystal.cpp
-csplayh5.cpp
+//csplayh5.cpp     // chd
 cswat.cpp
 cubeqst.cpp
 cubo.cpp
 cultures.cpp
-cupidon.cpp
+//cupidon.cpp      // check
 cv1k.cpp
 cvs.cpp
 cwheel.cpp
@@ -308,21 +308,23 @@
 ddragon.cpp
 ddragon3.cpp
 ddribble.cpp
-ddz.cpp
+//ddz.cpp          // skeleton
 deadang.cpp
 dec0.cpp
 dec8.cpp
+deco_ld.cpp        // needed by combram3
+deco_mlc.cpp
 deco156.cpp
 deco32.cpp
 decocass.cpp
 deco_ld.cpp
 deco_mlc.cpp
 deniam.cpp
-deshoros.cpp
+//deshoros.cpp     // check
 destroyr.cpp
-de_2.cpp
-de_3.cpp
-de_3b.cpp
+//de_2.cpp         // pinball
+//de_3.cpp         // pinball
+//de_3b.cpp        // pinball
 dfruit.cpp
 dgpix.cpp
 dietgo.cpp
@@ -334,8 +336,8 @@
 djmain.cpp
 dkmb.cpp
 dkong.cpp
-dlair.cpp
-dlair2.cpp
+//dlair.cpp        // laser-disk
+//dlair2.cpp       // laser-disk
 dmndrby.cpp
 docastle.cpp
 dogfgt.cpp
@@ -351,20 +353,20 @@
 dribling.cpp
 drmicro.cpp
 drtomy.cpp
-drw80pkr.cpp
+//drw80pkr.cpp     // poker
 dunhuang.cpp
-dwarfd.cpp
+//dwarfd.cpp       // poker
 dynadice.cpp
 dynamoah.cpp
 dynax.cpp
 dynduke.cpp
-ecoinf1.cpp
-ecoinf2.cpp
-ecoinf3.cpp
-ecoinfr.cpp
-efdt.cpp
+//ecoinf1.cpp      // fruit
+//ecoinf2.cpp      // fruit
+//ecoinf3.cpp      // fruit
+//ecoinfr.cpp      // fruit
+efdt.cpp           // ok
 egghunt.cpp
-electra.cpp
+//electra.cpp      // ttl
 embargo.cpp
 enigma2.cpp
 eolith.cpp
@@ -374,7 +376,7 @@
 equites.cpp
 ertictac.cpp
 esd16.cpp
-esh.cpp
+//esh.cpp          // chd
 espial.cpp
 esripsys.cpp
 ettrivia.cpp
@@ -382,11 +384,11 @@
 exerion.cpp
 exidy.cpp
 exidy440.cpp
-exidyttl.cpp
+//exidyttl.cpp     // ttl
 expro02.cpp
 exprraid.cpp
 exterm.cpp
-extrema.cpp
+//extrema.cpp      // check
 exzisus.cpp
 ez2d.cpp
 f-32.cpp
@@ -398,18 +400,18 @@
 fastlane.cpp
 fcombat.cpp
 fcrash.cpp
-feversoc.cpp
-ffantasy_ms.cpp
+//feversoc.cpp     // fruit
+//ffantasy_ms.cpp    // skeleton
 fgoal.cpp
 finalizr.cpp
-fireball.cpp
+//fireball.cpp     // mechanical
 firebeat.cpp
 firefox.cpp
 firetrap.cpp
 firetrk.cpp
 fitfight.cpp
-flashbeats.cpp
-flicker.cpp
+//flicker.cpp      // pinball
+//flashbeats.cpp     // not working
 flipjack.cpp
 flkatck.cpp
 flower.cpp
@@ -417,30 +419,30 @@
 flyball.cpp
 foodf.cpp
 forte2.cpp
-fortecar.cpp
+//fortecar.cpp     // poker
 freekick.cpp
-freeway.cpp
-fresh.cpp
+//fresh.cpp        // fruit
+//freeway.cpp      // black screen; driver has no info about this system
 fromanc2.cpp
 fromance.cpp
 fruitpc.cpp
-fungames.cpp
+//fungames.cpp     // ttl
 funkball.cpp
 funkybee.cpp
 funkyjet.cpp
-funtech.cpp
-funworld.cpp
+//funtech.cpp      // poker
+//funworld.cpp     // poker
 funybubl.cpp
 fuukifg2.cpp
 fuukifg3.cpp
-g627.cpp
+//g627.cpp         // pinball
 gaelco.cpp
 gaelco2.cpp
 gaelco3d.cpp
 gaelcof3.cpp
 gaelcopc.cpp
 gaiden.cpp
-gal3.cpp
+//gal3.cpp         // check
 galaga.cpp
 galastrm.cpp
 galaxi.cpp
@@ -455,12 +457,12 @@
 galpanic.cpp
 galpanic_ms.cpp
 galspnbl.cpp
-gambl186.cpp
-gamecstl.cpp
-gamemasters.cpp
+//gambl186.cpp     // check
+//gamecstl.cpp     // chd
+//gamemasters.cpp  // mechanical
 gameplan.cpp
-gammagic.cpp
-gamtor.cpp
+//gammagic.cpp    // check
+//gamtor.cpp      // check
 gaplus.cpp
 gatron.cpp
 gauntlet.cpp
@@ -472,21 +474,21 @@
 giclassic.cpp
 gijoe.cpp
 ginganin.cpp
-gkigt.cpp
+//gkigt.cpp        // check
 gladiatr.cpp
 glass.cpp
-globalfr.cpp
-globalvr.cpp
+//globalfr.cpp     // mechanical
+//globalvr.cpp     // chd
 gluck2.cpp
 gng.cpp
 go2000.cpp
 goal92.cpp
 goindol.cpp
-gokidetor.cpp
-goldart.cpp
-goldngam.cpp
-goldnpkr.cpp
-goldstar.cpp
+//gokidetor.cpp     // check
+//goldart.cpp       // mechanical
+//goldngam.cpp      // poker
+//goldnpkr.cpp      // poker
+//goldstar.cpp      // fruit
 gomoku.cpp
 good.cpp
 goodejan.cpp
@@ -495,8 +497,8 @@
 gottlieb.cpp
 gotya.cpp
 gpworld.cpp
-gp_1.cpp
-gp_2.cpp
+//gp_1.cpp          // pinball
+//gp_2.cpp          // pinball
 gradius3.cpp
 grchamp.cpp
 gridlee.cpp
@@ -506,12 +508,12 @@
 gstriker.cpp
 gsword.cpp
 gticlub.cpp
-gts1.cpp
-gts3.cpp
-gts3a.cpp
-gts80.cpp
-gts80a.cpp
-gts80b.cpp
+//gts1.cpp          // pinball
+//gts3.cpp          // pinball
+//gts3a.cpp         // pinball
+//gts80.cpp         // pinball
+//gts80a.cpp        // pinball
+//gts80b.cpp        // pinball
 guab.cpp
 gumbo.cpp
 gunbustr.cpp
@@ -521,72 +523,72 @@
 gyruss.cpp
 halleys.cpp
 hanaawas.cpp
-hankin.cpp
-hapyfish.cpp
+//hankin.cpp        // pinball
+//hapyfish.cpp      // too big to fit in memory
 harddriv.cpp
-hazelgr.cpp
+//hazelgr.cpp       // fruit
 hcastle.cpp
 headonb.cpp
 hexion.cpp
-hideseek.cpp
+//hideseek.cpp     // check
 higemaru.cpp
-highvdeo.cpp
-hikaru.cpp
+//highvdeo.cpp     // poker
+//hikaru.cpp       // check
 himesiki.cpp
 hitme.cpp
-hitpoker.cpp
+//hitpoker.cpp     // check
 hnayayoi.cpp
 hng64.cpp
-hobbyplay.cpp
+//hobbyplay.cpp         // fruit
 holeland.cpp
 homedata.cpp
 homerun.cpp
 hornet.cpp
 hotblock.cpp
-hotstuff.cpp
-hshavoc.cpp
+//hotstuff.cpp     // poker
+//hshavoc.cpp      // check
 hvyunit.cpp
 hyhoo.cpp
 hyperspt.cpp
 hyprduel.cpp
-icecold.cpp
-ice_bozopail.cpp
-ice_tbd.cpp
-ichiban.cpp
-idsa.cpp
+//icecold.cpp      // mechanical
+//ice_bozopail.cpp // mechanical
+//ice_tbd.cpp      // mechanical
+//ichiban.cpp      // check
+//idsa.cpp         // pinball
 igs009.cpp
 igs011.cpp
 igs017.cpp
-igspc.cpp
-igspoker.cpp
-igs_fear.cpp
-igs_m027.cpp
-igs_m036.cpp
+//igspc.cpp        // chd
+//igspoker.cpp     // poker
+//igs_fear.cpp     // check
+//igs_m027.cpp     // check
+//igs_m036.cpp     // check
 ikki.cpp
 imolagp.cpp
-inder.cpp
-inderp.cpp
-instantm.cpp
-intrscti.cpp
+//inder.cpp        // pinball
+//inderp.cpp       // pinball
+//instantm.cpp     // mechanical
+//intrscti.cpp     // check
 inufuku.cpp
 invqix.cpp
 iqblock.cpp
 irobot.cpp
 ironhors.cpp
-island.cpp
-istellar.cpp
+//island.cpp       // mechanical
+//istellar.cpp     // check, mnw
 istrebiteli.cpp
 iteagle.cpp
 itech32.cpp
 itech8.cpp
-itgambl2.cpp
-itgambl3.cpp
-itgamble.cpp
+//itgambl2.cpp     // fruit
+//itgambl3.cpp     // fruit
+//itgamble.cpp     // fruit
 jack.cpp
 jackal.cpp
 jackie.cpp
-jackpool.cpp
-jackpot.cpp
+//jackpool.cpp     // poker
+//jackpot.cpp      // mnw, not stated, probably junk
 jaguar.cpp
 jailbrek.cpp
 jalmah.cpp
@@ -596,26 +598,26 @@
 jchan.cpp
 jclub2.cpp
 jedi.cpp
-jeutel.cpp
-joctronic.cpp
-jokrwild.cpp
+//jeutel.cpp       // pinball
+//joctronic.cpp    // mechanical
+//jokrwild.cpp     // poker
 jollyjgr.cpp
 jongkyo.cpp
 joystand.cpp
-jp.cpp
-jpmimpct.cpp
-jpmimpctsw.cpp
-jpmmps.cpp
-jpms80.cpp
-jpmsru.cpp
-jpmsys5.cpp
-jpmsys5sw.cpp
-jpmsys7.cpp
+//jp.cpp           // pinball
+//jpmimpct.cpp     // check
+//jpmimpctsw.cpp   // fruit
+//jpmmps.cpp       // fruit
+//jpms80.cpp       // fruit
+//jpmsru.cpp       // fruit
+//jpmsys5.cpp      // fruit
+//jpmsys5sw.cpp    // fruit
+//jpmsys7.cpp      // fruit
 jrpacman.cpp
 jubilee.cpp
-jungleyo.cpp
+//jungleyo.cpp     // check
 junofrst.cpp
-jvh.cpp
+//jvh.cpp          // pinball
 kaneko16.cpp
 kangaroo.cpp
 karnov.cpp
@@ -628,7 +630,7 @@
 kingobox.cpp
 kingpin.cpp
 kinst.cpp
-kissproto.cpp
+//kissproto.cpp    // pinball
 klax.cpp
 kncljoe.cpp
 koftball.cpp
@@ -638,10 +640,10 @@
 konamigv.cpp
 konamigx.cpp
 konamim2.cpp
-konblands.cpp
-konendev.cpp
-konmedal.cpp
-konmedal68k.cpp
+//konblands.cpp    // not working; check back later
+//konendev.cpp     // fruit
+//konmedal.cpp     // check
+//konmedal68k.cpp  // check
 kontest.cpp
 kopunch.cpp
 kpython.cpp
@@ -660,33 +662,33 @@
 lastduel.cpp
 lastfght.cpp
 lazercmd.cpp
-laz_aftrshok.cpp
-laz_ribrac.cpp
+//laz_aftrshok.cpp // mechanical
+//laz_ribrac.cpp   // mechanical
 lbeach.cpp
-lckydraw.cpp
+//lckydraw.cpp     // pinball
 legionna.cpp
 leland.cpp
 lemmings.cpp
 lethal.cpp
 lethalj.cpp
-lgp.cpp
+//lgp.cpp          // check
 liberate.cpp
 liberatr.cpp
 limenko.cpp
-lindbergh.cpp
+//lindbergh.cpp    // fatal memory handler error at start
 littlerb.cpp
 lkage.cpp
 lockon.cpp
 looping.cpp
 lordgun.cpp
 lsasquad.cpp
-ltcasino.cpp
-ltd.cpp
-luckgrln.cpp
-lucky37.cpp
-lucky74.cpp
-luckybal.cpp
-lvcards.cpp
+//ltcasino.cpp     // gambling
+//ltd.cpp          // pinball
+//luckgrln.cpp     // gambling
+//lucky37.cpp      // poker
+//lucky74.cpp      // poker
+//luckybal.cpp     // gambling
+//lvcards.cpp      // poker
 lwings.cpp
 m10.cpp
 m107.cpp
@@ -700,15 +702,15 @@
 m79amb.cpp
 m90.cpp
 m92.cpp
-macp.cpp
+//macp.cpp         // pinball
 macrossp.cpp
 macs.cpp
 madalien.cpp
 madmotor.cpp
-magic10.cpp
-magicard.cpp
-magicfly.cpp
-magictg.cpp
+//magic10.cpp      // poker
+//magicard.cpp     // poker
+//magicfly.cpp     // poker
+//magictg.cpp      // check
 magmax.cpp
 magreel.cpp
 magtouch.cpp
@@ -716,29 +718,29 @@
 mainsnk.cpp
 majorpkr.cpp
 malzak.cpp
-manohman.cpp
+//manohman.cpp     // check
 mappy.cpp
 marineb.cpp
 marinedt.cpp
 mario.cpp
 markham.cpp
-marywu.cpp
+//marywu.cpp       // check
 mastboy.cpp
 mastboyo.cpp
 matmania.cpp
-matrix.cpp
+//matrix.cpp         // fruit
 maxaflex.cpp
-maygay1bsw.cpp
-maygayep.cpp
-maygayew.cpp
-maygayv1.cpp
+//maygay1bsw.cpp     // fruit
+//maygayep.cpp     // fruit
+//maygaysw.cpp     // fruit
+//maygayv1.cpp     // fruit
 mazerbla.cpp
 mcatadv.cpp
 mcr.cpp
 mcr3.cpp
 mcr68.cpp
 meadows.cpp
-meadwttl.cpp
+//meadwttl.cpp     // ttl
 mediagx.cpp
 megadriv_acbl.cpp
 megaphx.cpp
@@ -747,27 +749,27 @@
 megatech.cpp
 megazone.cpp
 meijinsn.cpp
-menghong.cpp
-mephistp.cpp
+//menghong.cpp     // gambling
+//mephistp.cpp     // pinball
 merit.cpp
 merit3xx.cpp
 meritm.cpp
 merits.cpp
 meritsdx.cpp
 mermaid.cpp
-metalmx.cpp
+//metalmx.cpp      // check
 metlclsh.cpp
 metlfrzr.cpp
 metro.cpp
-meyc8080.cpp
-meyc8088.cpp
-mgames.cpp
-mgavegas.cpp
+//meyc8080.cpp     // poker
+//meyc8088.cpp     // fruit
+//mgames.cpp       // poker
+//mgavegas.cpp     // check
 mgolf.cpp
 mhavoc.cpp
 micro3d.cpp
-microdar.cpp
-micropin.cpp
+//microdar.cpp     // mechanical
+//micropin.cpp     // pinball
 midas.cpp
 midqslvr.cpp
 midtunit.cpp
@@ -777,7 +779,7 @@
 midyunit.cpp
 midzeus.cpp
 mikie.cpp
-mil4000.cpp
+//mil4000.cpp      // poker
 miniboy7.cpp
 minivadr.cpp
 minivideo.cpp
@@ -792,57 +794,57 @@
 mjsister.cpp
 mlanding.cpp
 mmagic.cpp
-mmm.cpp
+//mmm.cpp          // mechanical
 model1.cpp
 model2.cpp
 model3.cpp
 mogura.cpp
 mole.cpp
 momoko.cpp
-monacogp.cpp
+//monacogp.cpp     // ttl
 monzagp.cpp
 moo.cpp
 mosaic.cpp
 mouser.cpp
-mpu12wbk.cpp
-mpu2.cpp
-mpu3.cpp
-mpu4.cpp
-mpu4avan.cpp
-mpu4bwb.cpp
-mpu4concept.cpp
-mpu4crystal.cpp
-mpu4dealem.cpp
-mpu4empire.cpp
-mpu4mdm.cpp
-mpu4misc.cpp
-mpu4mod2sw.cpp
-mpu4mod4yam.cpp
-mpu4plasma.cpp
-mpu4sw.cpp
-mpu4union.cpp
-mpu4vid.cpp
-mpu5sw.cpp
-mquake.cpp
+//mpu12wbk.cpp     // fruit
+//mpu2.cpp         // fruit
+//mpu3.cpp         // fruit
+//mpu4.cpp         // fruit
+//mpu4avan.cpp     // fruit
+//mpu4bwb.cpp      // fruit
+//mpu4concept.cpp  // fruit
+//mpu4crystal.cpp  // fruit
+//mpu4dealem.cpp   // fruit
+//mpu4empire.cpp   // fruit
+//mpu4mdm.cpp      // fruit
+//mpu4misc.cpp     // fruit
+//mpu4mod2sw.cpp   // fruit
+//mpu4mod4yam.cpp  // fruit
+//mpu4plasma.cpp   // fruit
+//mpu4sw.cpp       // fruit
+//mpu4union.cpp    // fruit
+//mpu4vid.cpp      // fruit
+//mpu5sw.cpp         // fruit
+//mquake.cpp       // test rom
 mrdo.cpp
 mrflea.cpp
-mrgame.cpp
+//mrgame.cpp       // pinball
 mrjong.cpp
 ms32.cpp
 msisaac.cpp
-mtouchxl.cpp
+//mtouchxl.cpp     // chd
 mugsmash.cpp
-multfish.cpp
-multfish_boot.cpp
+//multfish.cpp     // fruit
+//multfish_boot.cpp // fruit
 multigam.cpp
 munchmo.cpp
-murogem.cpp
-murogmbl.cpp
+//murogem.cpp      // poker
+//murogmbl.cpp     // poker
 mustache.cpp
 mw18w.cpp
 mw8080bw.cpp
 mwarr.cpp
-mwsub.cpp
+//mwsub.cpp        // mechanical
 mystston.cpp
 mystwarr.cpp
 n8080.cpp
@@ -850,9 +852,9 @@
 namcona1.cpp
 namconb1.cpp
 namcond1.cpp
-namcops2.cpp
+//namcops2.cpp     // check
 namcos1.cpp
-namcos10.cpp
+//namcos10.cpp     // check
 namcos11.cpp
 namcos12.cpp
 namcos1b.cpp
@@ -861,7 +863,7 @@
 namcos21_c67.cpp
 namcos21_de.cpp
 namcos22.cpp
-namcos23.cpp
+//namcos23.cpp     // check
 namcos86.cpp
 naomi.cpp
 naughtyb.cpp
@@ -874,11 +876,11 @@
 neogeo.cpp
 neopcb.cpp
 neoprint.cpp
-neptunp2.cpp
+//neptunp2.cpp    // check
 news.cpp
-nexus3d.cpp
-nibble.cpp
-nichild.cpp
+//nexus3d.cpp     // check
+//nibble.cpp      // check
+//nichild.cpp     // check (black screen)
 nightgal.cpp
 nightmare.cpp
 ninjakd2.cpp
@@ -887,13 +889,13 @@
 niyanpai.cpp
 nmg5.cpp
 nmk16.cpp
-nmkmedal.cpp
-norautp.cpp
-notechan.cpp
+//nmkmedal.cpp     // mechanical
+//norautp.cpp      // poker
+//notechan.cpp     // check
 nova2001.cpp
-nsg6809.cpp
-nsm.cpp
-nsmpoker.cpp
+//nsg6809.cpp      // mechanical
+//nsm.cpp          // pinball
+//nsmpoker.cpp     // check
 nss.cpp
 nwk-tr.cpp
 nycaptor.cpp
@@ -916,7 +918,7 @@
 pacland.cpp
 pacman.cpp
 pandoras.cpp
-pangofun.cpp
+//pangofun.cpp     // check
 panicr.cpp
 paradise.cpp
 paranoia.cpp
@@ -932,18 +934,18 @@
 pcxt.cpp
 pengadvb.cpp
 pengo.cpp
-peplus.cpp
-peyper.cpp
+//peplus.cpp       // fruit
+//peyper.cpp       // pinball
 pgm.cpp
 pgm2.cpp
-pgm3.cpp
+//pgm3.cpp         // chd
 phoenix.cpp
 photon.cpp
 photon2.cpp
-photoply.cpp
-photoplysx.cpp
-piggypas.cpp
-pinball2k.cpp
+//photoply.cpp     // chd
+//photoplysx.cpp           // skeleton
+//piggypas.cpp     // mechanical
+//pinball2k.cpp    // pinball
 pingpong.cpp
 pinkiri8.cpp
 pipedrm.cpp
@@ -955,14 +957,14 @@
 pktgaldx.cpp
 playch10.cpp
 playmark.cpp
-play_1.cpp
-play_2.cpp
-play_3.cpp
-pluto5.cpp
-plygonet.cpp
+//play_1.cpp       // pinball
+//play_2.cpp       // pinball
+//play_3.cpp       // pinball
+//pluto5.cpp       // fruit
+//plygonet.cpp     // check
 pntnpuzl.cpp
 pokechmp.cpp
-poker72.cpp
+//poker72.cpp      // poker
 pokerout.cpp
 polepos.cpp
 policetr.cpp
@@ -974,15 +976,15 @@
 popobear.cpp
 popper.cpp
 portrait.cpp
-potgoldu.cpp
+//potgoldu.cpp     // check
 powerbal.cpp
 powerins.cpp
 ppmast93.cpp
 prehisle.cpp
-proconn.cpp
+//proconn.cpp      // fruit
 progolf.cpp
-psattack.cpp
-pse.cpp
+//psattack.cpp     // skeleton
+//pse.cpp          // ttl
 psikyo.cpp
 psikyo4.cpp
 psikyosh.cpp
@@ -1002,7 +1004,7 @@
 quizdna.cpp
 quizo.cpp
 quizpani.cpp
-quizpun2.cpp
+//quizpun2.cpp     // check
 quizshow.cpp
 r2dtank.cpp
 r2dx_v33.cpp
@@ -1012,58 +1014,58 @@
 raiden_ms.cpp
 rallyx.cpp
 rampart.cpp
-ramtek.cpp
+//ramtek.cpp       // ttl
 rastan.cpp
 rastersp.cpp
 rbisland.cpp
 rbmk.cpp
-rcorsair.cpp
+//rcorsair.cpp     // check
 re900.cpp
 realbrk.cpp
 redalert.cpp
 redclash.cpp
-regama.cpp
+//regama.cpp       // pinball
 relief.cpp
 renegade.cpp
 retofinv.cpp
-rfslots8085.cpp
-rfslotspcpent.cpp
-rgum.cpp
+//rfslots8085.cpp  // skeleton mechanical
+//rfslotspcpent.cpp          // mechanical
+//rgum.cpp         // check
 rltennis.cpp
 rmhaihai.cpp
 rockrage.cpp
 rocnrope.cpp
 rohga.cpp
 rollerg.cpp
-rollext.cpp
+//rollext.cpp      // check
 rollrace.cpp
 ron.cpp
 rotaryf.cpp
 roul.cpp
 route16.cpp
-rowamet.cpp
+//rowamet.cpp      // pinball
 royalmah.cpp
 rpunch.cpp
 rulechan.cpp
 runaway.cpp
 rungun.cpp
-s11.cpp
-s11a.cpp
-s11b.cpp
-s11c.cpp
-s3.cpp
-s4.cpp
-s6.cpp
-s6a.cpp
-s7.cpp
-s8.cpp
-s8a.cpp
-s9.cpp
+//s11.cpp          // pinball
+//s11a.cpp         // pinball
+//s11b.cpp         // pinball
+//s11c.cpp         // pinball
+//s3.cpp           // pinball
+//s4.cpp           // pinball
+//s6.cpp           // pinball
+//s6a.cpp          // pinball
+//s7.cpp           // pinball
+//s8.cpp           // pinball
+//s8a.cpp          // pinball
+//s9.cpp           // pinball
 safarir.cpp
-sam.cpp
+//sam.cpp          // pinball
 sandscrp.cpp
 sangho.cpp
-sanremmg.cpp
+//sanremmg.cpp     // check
 sanremo.cpp
 saturn.cpp
 sauro.cpp
@@ -1072,38 +1074,38 @@
 sbowling.cpp
 sbrkout.cpp
 sbugger.cpp
-schick.cpp
-scm_500.cpp
+schick.cpp         // playable, bad gfx, no sound
+//scm_500.cpp      // mechanical
 scobra.cpp
 scotrsht.cpp
 scramble.cpp
 scregg.cpp
 scyclone.cpp
 sderby.cpp
-sderby2.cpp
+//sderby2.cpp      // check
 seabattl.cpp
 sealy.cpp
-sealy_fr.cpp
+//sealy_fr.cpp     // skeleton
 seattle.cpp
 segaatom.cpp
 segac2.cpp
-segacoin.cpp
+//segacoin.cpp     // check
 segae.cpp
 segag80r.cpp
 segag80v.cpp
 segahang.cpp
 segajw.cpp
-segald.cpp
-segam1.cpp
+//segald.cpp       // check
+//segam1.cpp       // check
 segaorun.cpp
 segas16a.cpp
 segas16b.cpp
 segas18.cpp
 segas24.cpp
 segas32.cpp
-segasp.cpp
-segattl.cpp
-segaufo.cpp
+//segasp.cpp       // check
+//segattl.cpp      // ttl
+//segaufo.cpp      // mechanical
 segaxbd.cpp
 segaybd.cpp
 seibucats.cpp
@@ -1115,8 +1117,8 @@
 seta.cpp
 seta2.cpp
 sf.cpp
-sfbonus.cpp
-sfcbox.cpp
+//sfbonus.cpp      // fruit
+//sfcbox.cpp       // check
 sfkick.cpp
 sg1000a.cpp
 shadfrce.cpp
@@ -1125,23 +1127,23 @@
 shangkid.cpp
 shaolins.cpp
 shisen.cpp
-shootaway2.cpp
+//shootaway2.cpp   // check
 shootout.cpp
 shougi.cpp
 shtzone.cpp
 shuuz.cpp
 sidearms.cpp
 sidepckt.cpp
-sigma21.cpp
-sigmab31.cpp
-sigmab52.cpp
+//sigma21.cpp      // skeleton, probably gambling
+//sigmab31.cpp     // gambling
+//sigmab52.cpp     // poker
 sigmab98.cpp
 silkroad.cpp
 silvmil.cpp
 simpl156.cpp
 simple_st0016.cpp
 simpsons.cpp
-skeetsht.cpp
+//skeetsht.cpp     // check
 skimaxx.cpp
 skopro.cpp
 skullxbo.cpp
@@ -1153,10 +1155,10 @@
 skyraid.cpp
 slapfght.cpp
 slapshot.cpp
-sleic.cpp
+//sleic.cpp        // pinball
 sliver.cpp
-slotcarn.cpp
-smotor.cpp
+//slotcarn.cpp     // fruit-poker
+smotor.cpp           // ok
 smsmcorp.cpp
 sms_bootleg.cpp
 snesb.cpp
@@ -1176,19 +1178,19 @@
 spcforce.cpp
 spdheat.cpp
 spdodgeb.cpp
-special_gambl.cpp
-spectra.cpp
+//special_gambl.cpp  // gambling
+//spectra.cpp      // pinball
 speedatk.cpp
 speedbal.cpp
 speedspn.cpp
 speglsht.cpp
 spiders.cpp
-spinb.cpp
-spirit76.cpp
+//spinb.cpp        // pinball
+//spirit76.cpp     // pinball
 splash.cpp
-splash_ms.cpp
-splus.cpp
-spoker.cpp
+splash_ms.cpp      // adult
+//splus.cpp        // fruit
+//spoker.cpp       // poker
 spool99.cpp
 sprcros2.cpp
 sprint2.cpp
@@ -1214,36 +1216,36 @@
 stadhero.cpp
 starcrus.cpp
 starfire.cpp
-stargame.cpp
-starrider.cpp
+//stargame.cpp     // pinball
+//starrider.cpp    // skeleton
 starshp1.cpp
 startouch.cpp
 starwars.cpp
 statriv2.cpp
-stellafr.cpp
+//stellafr.cpp     // fruit
 stfight.cpp
 stlforce.cpp
 strkzn.cpp
 stuntair.cpp
 stv.cpp
-st_mp100.cpp
-st_mp200.cpp
+//st_mp100.cpp     // pinball
+//st_mp200.cpp     // pinball
 su2000.cpp
 sub.cpp
-subhuntr.cpp
+//subhuntr.cpp     // check
 subs.cpp
-subsino.cpp
+//subsino.cpp      // fruit/poker
 subsino2.cpp
-summit.cpp
-sumt8035.cpp
+//summit.cpp       // fruit
+//sumt8035.cpp     // mechanical
 suna16.cpp
 suna8.cpp
 supbtime.cpp
 supdrapo.cpp
 supduck.cpp
 superchs.cpp
-supercrd.cpp
-superdq.cpp
+//supercrd.cpp     // fruit
+//superdq.cpp      // check
 superqix.cpp
 supertnk.cpp
 superwng.cpp
@@ -1252,29 +1254,29 @@
 suprnova.cpp
 suprridr.cpp
 suprslam.cpp
-supstarf.cpp
+//supstarf.cpp     // mechanical
 surpratk.cpp
 sweetland.cpp
 system1.cpp
 system16.cpp
 tagteam.cpp
 tail2nos.cpp
-taito.cpp
+//taito.cpp        // pinball
 taitoair.cpp
 taitogn.cpp
 taitojc.cpp
 taitopjc.cpp
 taitosj.cpp
-taitottl.cpp
+//taitottl.cpp     // ttl
 taitotx.cpp
 taitotz.cpp
-taitowlf.cpp
+//taitowlf.cpp     // check
 taito_b.cpp
 taito_f2.cpp
 taito_f3.cpp
 taito_h.cpp
 taito_l.cpp
-taito_o.cpp
+//taito_o.cpp      // poker
 taito_x.cpp
 taito_z.cpp
 tank8.cpp
@@ -1283,24 +1285,24 @@
 taotaido.cpp
 tapatune.cpp
 targeth.cpp
-tasman.cpp
+//tasman.cpp       // check
 tatsumi.cpp
 tattack.cpp
 taxidriv.cpp
 tbowl.cpp
 tceptor.cpp
-teamjocs.cpp
-techno.cpp
+//teamjocs.cpp     // unknown skeleton
+//techno.cpp       // pinball
 tecmo.cpp
 tecmo16.cpp
 tecmosys.cpp
-tecnodar.cpp
+//tecnodar.cpp     // mechanical
 tehkanwc.cpp
 tempest.cpp
 terracre.cpp
 tetrisp2.cpp
 tgtpanic.cpp
-thayers.cpp
+//thayers.cpp      // laserdisk
 thedealr.cpp
 thedeep.cpp
 thepit.cpp
@@ -1313,11 +1315,11 @@
 tigeroad.cpp
 timelimt.cpp
 timeplt.cpp
-timetrv.cpp
+//timetrv.cpp      // check
 tmaster.cpp
 tmmjprd.cpp
 tmnt.cpp
-tmspoker.cpp
+//tmspoker.cpp     // check
 tnzs.cpp
 toaplan1.cpp
 toaplan2.cpp
@@ -1336,7 +1338,7 @@
 travrusa.cpp
 triforce.cpp
 triplhnt.cpp
-triviaquiz.cpp
+//triviaquiz.cpp    // skeleton
 trivrus.cpp
 truco.cpp
 trucocl.cpp
@@ -1352,7 +1354,7 @@
 turbo.cpp
 turrett.cpp
 tutankhm.cpp
-tvcapcom.cpp
+//tvcapcom.cpp     // skeleton
 tvg01.cpp
 twin16.cpp
 twincobr.cpp
@@ -1365,12 +1367,12 @@
 ultrsprt.cpp
 umipoker.cpp
 undrfire.cpp
-unianapc.cpp
+//unianapc.cpp     // chd
 unico.cpp
 unkhorse.cpp
-unkpoker.cpp
+//unkpoker.cpp     // not working poker
 upscope.cpp
-usbilliards.cpp
+//usbilliards.cpp  // ttl
 usgames.cpp
 v0bowl.cpp
 vamphalf.cpp
@@ -1378,7 +1380,7 @@
 vastar.cpp
 vball.cpp
 vcombat.cpp
-vd.cpp
+//vd.cpp           // pinball
 vectrex.cpp
 vega.cpp
 vegaeo.cpp
@@ -1387,18 +1389,18 @@
 vertigo.cpp
 vicdual.cpp
 victory.cpp
-video21.cpp
+//video21.cpp      // gambling
 videopin.cpp
-videopkr.cpp
-videosaa.cpp
+//videopkr.cpp     // poker
+//videosaa.cpp     // gambling skeletons
 vigilant.cpp
 vindictr.cpp
 viper.cpp
-vlc.cpp
+//vlc.cpp          // check
 volfied.cpp
 voyager.cpp
 vp101.cpp
-vpoker.cpp
+//vpoker.cpp       // poker
 vroulet.cpp
 vsnes.cpp
 vulgus.cpp
@@ -1414,25 +1416,25 @@
 welltris.cpp
 wgp.cpp
 wheelfir.cpp
-whitestar.cpp
-white_mod.cpp
-wico.cpp
-wildpkr.cpp
+//whitestar.cpp    // pinball
+//white_mod.cpp    // pinball
+//wico.cpp         // pinball
+//wildpkr.cpp      // poker
 williams.cpp
 wink.cpp
 wiping.cpp
 witch.cpp
 wiz.cpp
 wmg.cpp
-wms.cpp
+//wms.cpp          // fruit
 wolfpack.cpp
-wpc_95.cpp
-wpc_an.cpp
-wpc_dcs.cpp
-wpc_dot.cpp
-wpc_flip1.cpp
-wpc_flip2.cpp
-wpc_s.cpp
+//wpc_95.cpp       // pinball
+//wpc_an.cpp       // pinball
+//wpc_dcs.cpp      // pinball
+//wpc_dot.cpp      // pinball
+//wpc_flip1.cpp    // pinball
+//wpc_flip2.cpp    // pinball
+//wpc_s.cpp        // pinball
 wrally.cpp
 wwfsstar.cpp
 wyvernf0.cpp
@@ -1441,7 +1443,7 @@
 xmen.cpp
 xorworld.cpp
 xtheball.cpp
-xtom3d.cpp
+//xtom3d.cpp       // check
 xxmissio.cpp
 xybots.cpp
 xyonix.cpp
@@ -1449,17 +1451,17 @@
 yiear.cpp
 yunsun16.cpp
 yunsung8.cpp
-yuvomz80.cpp
+//yuvomz80.cpp          // mechanical
 zac2650.cpp
 zaccaria.cpp
-zac_1.cpp
-zac_2.cpp
-zac_proto.cpp
+//zac_1.cpp             // pinball
+//zac_2.cpp             // pinball
+//zac_proto.cpp         // pinball
 zaxxon.cpp
 zerozone.cpp
 zn.cpp
 zodiack.cpp
-zpinball.cpp
+//zpinball.cpp          // pinball, skeleton
 zr107.cpp
 zwackery.cpp
 //exceptions + to add - to remove from imported list
@@ -1473,10 +1475,10 @@
 -cdimono1
 -cdimono2
 //cps1.cpp
--sfach
--sfzbch
--sfzch
--wofch
+//-sfach
+//-sfzbch
+//-sfzch
+//-wofch
 //jaguar.cpp
 -jaguar
 -jaguarcd
