--- a/src/mame/layout/vgmplay.lay	2022-02-01 05:26:54.000000000 -0500
+++ b/src/mame/layout/vgmplay.lay	2022-02-14 15:38:02.548182991 -0500
@@ -109,19 +109,19 @@
 	</element>
 	<element name="text_l2">
 		<rect><color red="0.2" green="0.21" blue="0.23" /></rect>
-		<text string="&#x25EB;"><color red="0.9" green="0.9" blue="0.2" /></text>
+		<text string="ll"><color red="0.9" green="0.9" blue="0.2" /></text>
 	</element>
 	<element name="text_l3">
 		<rect><color red="0.2" green="0.21" blue="0.23" /></rect>
-		<text string="&#x25B6;"><color red="0.9" green="0.9" blue="0.2" /></text>
+		<text string="&gt;"><color red="0.9" green="0.9" blue="0.2" /></text>
 	</element>
 	<element name="text_l4">
 		<rect><color red="0.2" green="0.21" blue="0.23" /></rect>
-		<text string="&#x25C4;"><color red="0.9" green="0.9" blue="0.2" /></text>
+		<text string="&lt;&lt;"><color red="0.9" green="0.9" blue="0.2" /></text>
 	</element>
 	<element name="text_l5">
 		<rect><color red="0.2" green="0.21" blue="0.23" /></rect>
-		<text string="&#x25CE;"><color red="0.9" green="0.9" blue="0.2" /></text>
+		<text string="R"><color red="0.9" green="0.9" blue="0.2" /></text>
 	</element>
 
 	<group name="activity">
