diff --git a/src/mame/misc/sliver.cpp b/src/mame/misc/sliver.cpp
index e3b1b2afab..db63e517b8 100644
--- a/src/mame/misc/sliver.cpp
+++ b/src/mame/misc/sliver.cpp
@@ -281,7 +281,7 @@ void sliver_state::render_jpeg()
 
 		jpeg_mem_src(&cinfo, memregion("user2")->base()+addr, memregion("user2")->bytes()-addr);
 
-		jpeg_read_header(&cinfo, true);
+		jpeg_read_header(&cinfo, boolean(true));
 		jpeg_start_decompress(&cinfo);
 
 		int row_stride = cinfo.output_width * cinfo.output_components;
