diff --git a/Makefile.libretro b/Makefile.libretro
index 8d0bc9e..3a987ef 100644
--- a/Makefile.libretro
+++ b/Makefile.libretro
@@ -108,6 +108,34 @@ else ifeq ($(platform), wincross64)
 	SHARED := -shared -Wl,--no-undefined -Wl,--version-script=link.T
 	LDFLAGS += -static-libgcc -static-libstdc++ 
 	EXTRA_LDF := -lwinmm -Wl,--export-all-symbols
+# ARM
+else ifneq (,$(findstring armv,$(platform)))
+	TARGET := $(TARGET_NAME)_libretro.so
+	SHARED := -shared -Wl,--no-undefined -Wl,--version-script=link.T
+	fpic := -fPIC -fsigned-char
+	CFLAGS += -D_GNU_SOURCE=1
+	CC = gcc
+ifneq (,$(findstring cortexa8,$(platform)))
+	CFLAGS += -marm -mcpu=cortex-a8
+	ASFLAGS += -mcpu=cortex-a8
+else ifneq (,$(findstring cortexa9,$(platform)))
+	CFLAGS += -marm -mcpu=cortex-a9
+	ASFLAGS += -mcpu=cortex-a9
+endif
+	CFLAGS += -marm
+ifneq (,$(findstring neon,$(platform)))
+	CFLAGS += -mfpu=neon
+	ASFLAGS += -mfpu=neon
+	HAVE_NEON = 1
+endif
+ifneq (,$(findstring softfloat,$(platform)))
+	CFLAGS += -mfloat-abi=softfp
+	ASFLAGS += -mfloat-abi=softfp
+else ifneq (,$(findstring hardfloat,$(platform)))
+	CFLAGS += -mfloat-abi=hard
+	ASFLAGS += -mfloat-abi=hard
+endif
+	CFLAGS += -DARM
 else
 	TARGET :=  $(TARGET_NAME)_libretro.dll
 	CC = gcc
