--- a/yabause/src/libretro/Makefile	2020-09-25 14:51:36.738293717 +0200
+++ b/yabause/src/libretro/Makefile	2020-09-25 14:53:36.277216508 +0200
@@ -107,6 +107,21 @@
 	DYNAREC_DEVMIYAX = 1
 	ALLOW_POLYGON_MODE = 0
 
+# S912
+else ifneq (,$(findstring s912,$(platform)))
+       override platform += unix
+       TARGET := $(TARGET_NAME)_libretro.so
+       fpic := -fPIC
+       SHARED := -shared -Wl,--no-undefined -Wl,--version-script=link.T
+       LDFLAGS += -lpthread
+       ARCH_IS_LINUX = 1
+       HAVE_SSE = 0
+       FORCE_GLES = 1
+       USE_AARCH64_DRC = 1
+       DYNAREC_DEVMIYAX = 1
+       FLAGS += -DAARCH64 -march=armv8-a+crc+fp+simd -mcpu=cortex-a53 -mtune=cortex-a53
+
+
 # Amlogic S922X Odroid-N2 / A311D Khadas VIM3 (AMLG12B) - 32-bit userspace
 else ifneq (,$(findstring AMLG12B,$(platform)))
 	override platform += unix
