--- a/qb/config.libs.sh	2023-06-07 12:52:16.722566249 +0200
+++ b/qb/config.libs.sh	2023-06-07 12:52:40.511186426 +0200
@@ -520,7 +520,7 @@
 check_lib '' XRANDR -lXrandr
 check_header '' XSHM X11/Xlib.h X11/extensions/XShm.h
 check_val '' XKBCOMMON -lxkbcommon '' xkbcommon 0.3.2 '' false
-check_val '' WAYLAND '-lwayland-egl -lwayland-client' '' wayland-egl 10.1.0 '' false
+check_val '' WAYLAND '-lwayland-egl -lwayland-client' '' wayland-egl 1.0 '' false
 check_val '' WAYLAND_CURSOR -lwayland-cursor '' wayland-cursor 1.12 '' false
 check_pkgconf WAYLAND_PROTOS wayland-protocols 1.15
 check_pkgconf WAYLAND_SCANNER wayland-scanner '1.15 1.12'
