diff -Naur a/arch/arm64/Makefile b/arch/arm64/Makefile
--- a/arch/arm64/Makefile	2018-01-15 23:42:00.806287795 +0800
+++ b/arch/arm64/Makefile	2018-01-15 23:41:03.794515482 +0800
@@ -35,14 +35,21 @@
 
 CHECKFLAGS	+= -D__aarch64__
 
-ifeq ($(CONFIG_ARM64_ERRATUM_843419), y)
-KBUILD_CFLAGS_MODULE	+= $(call cc-option,-mfix-cortex-a53-843419)
+ifeq ($(CONFIG_ARM64_ERRATUM_843419),y)
+  ifeq ($(call ld-option, --fix-cortex-a53-843419),)
+$(warning ld does not support --fix-cortex-a53-843419; kernel may be susceptible to erratum)
+  else
+LDFLAGS_vmlinux	+= --fix-cortex-a53-843419
+  endif
 endif
 
-ifeq ($(CONFIG_ARM64_ERRATUM_835769), y)
-KBUILD_CFLAGS_MODULE	+= $(call cc-option,-mfix-cortex-a53-835769)
+ifeq ($(CONFIG_ARM64_ERRATUM_835769),y)
+  ifeq ($(call ld-option, --fix-cortex-a53-835769),)
+$(warning ld does not support --fix-cortex-a53-835769; kernel may be susceptible to erratum)
+  else
+LDFLAGS_vmlinux	+= --fix-cortex-a53-843419
+  endif
 endif
-$(info KBUILD_CFLAGS_MODULE:$(KBUILD_CFLAGS_MODULE))
 
 # Default value
 head-y		:= arch/arm64/kernel/head.o
@@ -79,9 +86,7 @@
 	$(Q)$(MAKE) $(build)=$(boot) $@
 
 %.dtb: scripts
-	$(if $(filter $@,$(mesondtb)),\
-	$(Q)$(MAKE) $(build)=$(boot)/dts/amlogic $(boot)/dts/amlogic/$@,\
-	$(Q)$(MAKE) $(build)=$(boot)/dts $(boot)/dts/$@)
+	$(Q)$(MAKE) $(build)=$(boot)/dts $(boot)/dts/$@
 
 dtbs: scripts
 	$(Q)$(MAKE) $(build)=$(boot)/dts dtbs
