diff --git a/share/wine/wine.inf b/share/wine/wine.inf
index da9c9b7..6afc4a8 100644
--- a/share/wine/wine.inf
+++ b/share/wine/wine.inf
@@ -88,7 +89,8 @@ WineFakeDlls=FakeDllsWin32
 AddReg=\
     Packages,\
     VCRuntime.ntx86,\
-    VersionInfo
+    VersionInfo, \
+    WineBatoOverrides
 
 [DefaultInstall.ntarm]
 Needs=BaseInstall
@@ -1538,3 +1540,7 @@ HKLM,Software\Khronos\OpenXR\1,"ActiveRuntime",,"C:\openxr\wineopenxr64.json"
 ;;Other app-specific overrides
 HKCU,Software\Wine\AppDefaults\Pentiment.exe\DllOverrides,"SpeechSynthesisWrapper",4,
 HKCU,Software\Wine\AppDefaults\Maine-Win64-Shipping.exe\DllOverrides,"SpeechSynthWrapper",4,
+
+[WineBatoOverrides]
+HKLM,System\CurrentControlSet\Services\winebus,"DisableHidraw",0x10001,"1"
+HKLM,System\CurrentControlSet\Services\winebus,"Enable SDL",0x10001,"1"
